Index: /Zend Framework/library/Zend/Http/UserAgent.php
===================================================================
--- /Zend Framework/library/Zend/Http/UserAgent.php	(revision 23621)
+++ /Zend Framework/library/Zend/Http/UserAgent.php	(working copy)
@@ -528,7 +528,12 @@
     }
 
     /**
-     * @return the $device
+     * Get the User-Agent Device object
+     *
+     * This is the object that will contain the various discovered device
+     * capabilities.
+     *
+     * @return Zend_Http_UserAgent_Device $device
      */
     public function getDevice()
     {
