Index: Application.php
===================================================================
--- Application.php	(revision 16086)
+++ Application.php	(working copy)
@@ -289,6 +289,18 @@
         if (!class_exists($class, false)) {
             require_once $path;
         }
+        
+        try {
+            $reflectionClass = new ReflectionClass($class);
+        } catch (ReflectionException $e) {
+            throw new Zend_Application_Exception("Class $class not found in $path");
+        }
+        
+        if (!$reflectionClass->implementsInterface('Zend_Application_Bootstrap_Bootstrapper')) {
+            throw new Zend_Application_Exception("Class $class must implement Zend_Application_Bootstrap_Bootstrapper");
+        }
+        unset($reflectionClass);
+        
         $this->_bootstrap = new $class($this);
         
         return $this;
