Index: library/Zend/Form.php
===================================================================
--- library/Zend/Form.php	(revisione 14377)
+++ library/Zend/Form.php	(copia locale)
@@ -1346,7 +1346,7 @@
     {
         $origName = $this->getElementsBelongTo();
         $name = $this->filterName($array, true);
-        if (empty($name)) {
+        if ($name == '') {
             $name = null;
         }
         $this->_elementsBelongTo = $name;
