Index: library/Zend/Form.php
===================================================================
--- library/Zend/Form.php	(revision 23858)
+++ library/Zend/Form.php	(working copy)
@@ -1142,24 +1142,21 @@
                     }
                     $this->addElement($type, $name, $options);
                 } else {
-                    switch ($argc) {
-                        case 0:
-                            continue;
-                        case (1 <= $argc):
-                            $type = array_shift($spec);
-                        case (2 <= $argc):
-                            if (null === $name) {
-                                $name = array_shift($spec);
-                            } else {
-                                $options = array_shift($spec);
-                            }
-                        case (3 <= $argc):
-                            if (empty($options)) {
-                                $options = array_shift($spec);
-                            }
-                        default:
-                            $this->addElement($type, $name, $options);
+                    if(0 == $argc) {
+                        continue;
                     }
+                    $type = array_shift($spec);
+                    if (2 <= $argc) {
+                        if (null === $name) {
+                            $name = array_shift($spec);
+                        } else {
+                            $options = array_shift($spec);
+                        }
+                    }
+                    if (3 <= $argc && empty($options)) {
+                        $options = array_shift($spec);
+                    }
+                    $this->addElement($type, $name, $options);
                 }
             }
         }
@@ -1651,17 +1648,17 @@
             if (is_array($spec)) {
                 $argc  = count($spec);
                 $order = null;
-                switch ($argc) {
-                    case 0:
-                        continue;
-                    case (1 <= $argc):
+                if (1 <= $argc) {
+                    if(1 <= $argc) {
                         $subForm = array_shift($spec);
-                    case (2 <= $argc):
-                        $name  = array_shift($spec);
-                    case (3 <= $argc):
-                        $order = array_shift($spec);
-                    default:
-                        $this->addSubForm($subForm, $name, $order);
+                    }
+                    if(2 <= $argc)  {
+                        $name    = array_shift($spec);
+                    }
+                    if(3 <= $argc) {
+                        $order   = array_shift($spec);
+                    }
+                    $this->addSubForm($subForm, $name, $order);
                 }
             }
         }
@@ -1911,26 +1908,27 @@
                 }
                 $this->addDisplayGroup($elements, $name, $options);
             } else {
-                switch ($argc) {
-                    case (1 <= $argc):
-                        $elements = array_shift($spec);
-                        if (!is_array($elements) && (null !== $name)) {
-                            $elements = array_merge((array) $elements, $spec);
-                            $this->addDisplayGroup($elements, $name);
-                            break;
-                        }
-                    case (2 <= $argc):
-                        if (null !== $name) {
-                            $options = array_shift($spec);
-                            $this->addDisplayGroup($elements, $name, $options);
-                            break;
-                        }
-                        $name = array_shift($spec);
-                    case (3 <= $argc):
+                if(1 == $spec) {
+                    continue;
+                }
+                $elements = array_shift($spec);
+                if (!is_array($elements) && null !== $name) {
+                    $elements = array_merge((array) $elements, $spec);
+                    $this->addDisplayGroup($elements, $name);
+                    continue;
+                }
+                if (2 <= $argc) {
+                    if (null !== $name) {
                         $options = array_shift($spec);
-                    default:
                         $this->addDisplayGroup($elements, $name, $options);
+                        continue;
+                    }
+                    $name = array_shift($spec);
                 }
+                if (3 <= $argc) {
+                    $options = array_shift($spec);
+                }
+                $this->addDisplayGroup($elements, $name, $options);
             }
         }
         return $this;
@@ -2701,16 +2699,12 @@
                     }
                     $this->addDecorator($decorator, $options);
                 } else {
-                    switch (true) {
-                        case (0 == $argc):
-                            break;
-                        case (1 <= $argc):
-                            $decorator  = array_shift($decoratorInfo);
-                        case (2 <= $argc):
+                    if (1 <= $argc) {
+                        $decorator  = array_shift($decoratorInfo);
+                        if (2 <= $argc){
                             $options = array_shift($decoratorInfo);
-                        default:
-                            $this->addDecorator($decorator, $options);
-                            break;
+                        }
+                        $this->addDecorator($decorator, $options);
                     }
                 }
             } else {
Index: library/Zend/File/Transfer/Adapter/Abstract.php
===================================================================
--- library/Zend/File/Transfer/Adapter/Abstract.php	(revision 23858)
+++ library/Zend/File/Transfer/Adapter/Abstract.php	(working copy)
@@ -409,22 +409,18 @@
                         $this->addValidator($validator, $breakChainOnFailure, $options, $files);
                     } else {
                         $file = $files;
-                        switch (true) {
-                            case (0 == $argc):
-                                break;
-                            case (1 <= $argc):
-                                $validator  = array_shift($validatorInfo);
-                            case (2 <= $argc):
+                        if (1 <= $argc) {
+                            $validator  = array_shift($validatorInfo);
+                            if (2 <= $argc) {
                                 $breakChainOnFailure = array_shift($validatorInfo);
-                            case (3 <= $argc):
+                            }
+                            if (3 <= $argc) {
                                 $options = array_shift($validatorInfo);
-                            case (4 <= $argc):
-                                if (!empty($validatorInfo)) {
-                                    $file = array_shift($validatorInfo);
-                                }
-                            default:
-                                $this->addValidator($validator, $breakChainOnFailure, $options, $file);
-                                break;
+                            }
+                            if (4 <= $argc && !empty($validatorInfo)) {
+                                $file = array_shift($validatorInfo);
+                            }
+                            $this->addValidator($validator, $breakChainOnFailure, $options, $file);
                         }
                     }
                 }
Index: library/Zend/Form/DisplayGroup.php
===================================================================
--- library/Zend/Form/DisplayGroup.php	(revision 23858)
+++ library/Zend/Form/DisplayGroup.php	(working copy)
@@ -763,16 +763,12 @@
                     }
                     $this->addDecorator($decorator, $options);
                 } else {
-                    switch (true) {
-                        case (0 == $argc):
-                            break;
-                        case (1 <= $argc):
-                            $decorator  = array_shift($decoratorInfo);
-                        case (2 <= $argc):
+                    if (1 <= $argc) {
+                        $decorator  = array_shift($decoratorInfo);
+                        if (2 <= $argc) {
                             $options = array_shift($decoratorInfo);
-                        default:
-                            $this->addDecorator($decorator, $options);
-                            break;
+                        }
+                        $this->addDecorator($decorator, $options);
                     }
                 }
             } else {
Index: library/Zend/Form/Element.php
===================================================================
--- library/Zend/Form/Element.php	(revision 23858)
+++ library/Zend/Form/Element.php	(working copy)
@@ -1182,18 +1182,15 @@
                     }
                     $this->addValidator($validator, $breakChainOnFailure, $options);
                 } else {
-                    switch (true) {
-                        case (0 == $argc):
-                            break;
-                        case (1 <= $argc):
-                            $validator  = array_shift($validatorInfo);
-                        case (2 <= $argc):
+                    if (1 <= $argc) {
+                        $validator  = array_shift($validatorInfo);
+                        if (2 <= $argc) {
                             $breakChainOnFailure = array_shift($validatorInfo);
-                        case (3 <= $argc):
+                        }
+                        if (3 <= $argc) {
                             $options = array_shift($validatorInfo);
-                        default:
-                            $this->addValidator($validator, $breakChainOnFailure, $options);
-                            break;
+                        }
+                        $this->addValidator($validator, $breakChainOnFailure, $options);
                     }
                 }
             } else {
@@ -1644,16 +1641,12 @@
                     }
                     $this->addFilter($filter, $options);
                 } else {
-                    switch (true) {
-                        case (0 == $argc):
-                            break;
-                        case (1 <= $argc):
-                            $filter  = array_shift($filterInfo);
-                        case (2 <= $argc):
+                    if (1 <= $argc) {
+                        $filter  = array_shift($filterInfo);
+                        if (2 <= $argc) {
                             $options = array_shift($filterInfo);
-                        default:
-                            $this->addFilter($filter, $options);
-                            break;
+                        }
+                        $this->addFilter($filter, $options);
                     }
                 }
             } else {
@@ -1884,16 +1877,12 @@
                     }
                     $this->addDecorator($decorator, $options);
                 } else {
-                    switch (true) {
-                        case (0 == $argc):
-                            break;
-                        case (1 <= $argc):
-                            $decorator  = array_shift($decoratorInfo);
-                        case (2 <= $argc):
+                    if (1 <= $argc) {
+                        $decorator  = array_shift($decoratorInfo);
+                        if (2 <= $argc) {
                             $options = array_shift($decoratorInfo);
-                        default:
-                            $this->addDecorator($decorator, $options);
-                            break;
+                        }
+                        $this->addDecorator($decorator, $options);
                     }
                 }
             } else {
