Index: library/Zend/File/Transfer/Adapter/Http.php
===================================================================
--- library/Zend/File/Transfer/Adapter/Http.php	(revision 19683)
+++ library/Zend/File/Transfer/Adapter/Http.php	(working copy)
@@ -355,12 +355,13 @@
                     $status['total']   = $status['bytes_total'];
                     $status['current'] = $status['bytes_uploaded'];
                     $status['rate']    = $status['speed_average'];
-                    if ($status['total'] == $status['current']) {
-                        $status['done'] = true;
-                    }
                 }
             }
 
+            if ($status['total'] == $status['current']) {
+                $status['done'] = true;
+            }
+
             if (!is_array($call)) {
                 $status['done']    = true;
                 $status['message'] = 'Failure while retrieving the upload progress';

