Index: Config.php
===================================================================
--- Config.php	(revision 13150)
+++ Config.php	(working copy)
@@ -82,6 +82,10 @@
      * @var string
      */
     protected $_loadFileErrorStr = null;
+    
+    protected $_cache;
+    
+    protected static $_defaultCache;
 
     /**
      * Zend_Config provides a property based interface to
@@ -388,6 +392,16 @@
         }
     }
     
+    public function setCache(Zend_Cache_Core $cache)
+    {
+        $this->_cache = $cache;
+    }
+    
+    public static function setDefaultCache(Zend_Cache_Core $cache)
+    {
+        self::$_defaultCache = $cache;
+    }
+    
     /**
      * Throws an exception if $extendingSection may not extend $extendedSection,
      * and tracks the section extension if it is valid.
@@ -430,4 +444,8 @@
         }
     }
 
+    protected function _cacheEnabled()
+    {
+        return !is_null($this->_cache);
+    }
 }
