Index: Zend/Service/Twitter.php
===================================================================
--- Zend/Service/Twitter.php	(revision 21108)
+++ Zend/Service/Twitter.php	(working copy)
@@ -484,6 +484,9 @@
                 case 'page':
                     $_params['page'] = (int) $value;
                     break;
+                case 'cursor':
+                    $_params['cursor'] = $value;
+                    break;
                 default:
                     break;
             }
