Index: incubator/Zend/Http/Client/Adapter/Socket.php
===================================================================
--- incubator/Zend/Http/Client/Adapter/Socket.php	(revision 847)
+++ incubator/Zend/Http/Client/Adapter/Socket.php	(revision 848)
@@ -134,7 +134,11 @@
             throw new Zend_Http_Client_Adapter_Exception("Trying to write but we are connected to the wrong host");
 
             // Build request headers
-        $request = "{$method} {$uri->__toString()} HTTP/{$http_ver}\r\n";
+        $requestUri = $uri->getPath();
+        if (strlen($uri->getQuery())) {
+            $requestUri .= '?' . $uri->getQuery();
+        }
+        $request = "{$method} {$requestUri} HTTP/{$http_ver}\r\n";
         foreach ($headers as $k => $v) {
             if (is_string($k)) $v = ucfirst($k) . ": $v";
             $request .= "$v\r\n";
