Index: HtmlList.php
===================================================================
--- HtmlList.php	(revision 19365)
+++ HtmlList.php	(working copy)
@@ -53,6 +53,10 @@
             require_once 'Zend/View/Exception.php';
             throw new Zend_View_Exception('First param must be an array', $this);
         }
+		
+		if (count($items) < 1){
+			return '';
+		}
 
         $list = '';
 
