Index: Adapter.php
===================================================================
--- Adapter.php	(revision 219)
+++ Adapter.php	(working copy)
@@ -50,7 +50,7 @@
      * @param  mixed $element
      * @return string
      */
-    public function render(Zend_View_Interface $view, $element = null);
+    public function render($view = null, $element = null);
 
     /**
      * Set captcha name
Index: Dumb.php
===================================================================
--- Dumb.php	(revision 219)
+++ Dumb.php	(working copy)
@@ -43,7 +43,7 @@
      * @param  mixed $element
      * @return string
      */
-    public function render(Zend_View_Interface $view, $element = null)
+    public function render($view = null, $element = null)
     {
         return 'Please type this word backwards: <b>'
              . strrev($this->getWord())
Index: Figlet.php
===================================================================
--- Figlet.php	(revision 219)
+++ Figlet.php	(working copy)
@@ -76,7 +76,7 @@
      * @param mixed $element
      * @return string
      */
-    public function render(Zend_View_Interface $view, $element = null)
+    public function render($view = null, $element = null)
     {
         return '<pre>'
              . $this->_figlet->render($this->getWord())
Index: Image.php
===================================================================
--- Image.php	(revision 219)
+++ Image.php	(working copy)
@@ -587,7 +587,7 @@
      * @param mixed $element
      * @return string
      */
-    public function render(Zend_View_Interface $view, $element = null)
+    public function render($view = null, $element = null)
     {
         return '<img alt="'.$this->getImgAlt().'" src="' . $this->getImgUrl() . $this->getId() . $this->getSuffix() . '"/><br/>';
     }
Index: ReCaptcha.php
===================================================================
--- ReCaptcha.php	(revision 219)
+++ ReCaptcha.php	(working copy)
@@ -260,7 +260,7 @@
      * @param  mixed $element 
      * @return string
      */
-    public function render(Zend_View_Interface $view, $element = null)
+    public function render($view = null, $element = null)
     {
         return $this->getService()->getHTML();
     }
