Index: Log.php
===================================================================
--- Log.php	(revision 936)
+++ Log.php	(working copy)
@@ -225,6 +225,10 @@
             throw new Zend_Log_Exception("Cannot unregister, no instance of log named \"$logName\".");
         }
 
+        if(self::isDefaultLogger($logName)) {
+            throw new Zend_Log_Exception("You cannot unregister the default logger");
+        }
+
         unset(self::$_instances[$logName]);
         return true;
     }
@@ -546,5 +550,16 @@
                         self::LEVEL_INFO, self::LEVEL_DEBUG));
     }
 
+    /**
+     * Check if a logger is set as default
+     *
+     * @param string    $logName    Name of logger instance
+     * @return boolean              Is this the default logger?
+     */
+    public static function isDefaultLogger($logName = null)
+    {
+        return ($logName == self::$_defaultLogName);
+    }
+
 }
 
