diff --git a/library/Zend/View/Helper/Partial.php b/library/Zend/View/Helper/Partial.php
old mode 100644
new mode 100755
index ab6c16f..76f92e3
--- a/library/Zend/View/Helper/Partial.php
+++ b/library/Zend/View/Helper/Partial.php
@@ -71,6 +71,10 @@ class Zend_View_Helper_Partial extends Zend_View_Helper_Abstract
         if (isset($this->partialCounter)) {
             $view->partialCounter = $this->partialCounter;
         }
+        if (isset($this->partialTotalCount)) {
+            $view->partialTotalCount = $this->partialTotalCount;
+        }
+
         if ((null !== $module) && is_string($module)) {
             require_once 'Zend/Controller/Front.php';
             $moduleDir = Zend_Controller_Front::getInstance()->getControllerDirectory($module);
diff --git a/library/Zend/View/Helper/PartialLoop.php b/library/Zend/View/Helper/PartialLoop.php
old mode 100644
new mode 100755
index 021e577..550afe4
--- a/library/Zend/View/Helper/PartialLoop.php
+++ b/library/Zend/View/Helper/PartialLoop.php
@@ -84,6 +84,8 @@ class Zend_View_Helper_PartialLoop extends Zend_View_Helper_Partial
         $content = '';
         // reset the counter if it's call again
         $this->partialCounter = 0;
+        $this->partialTotalCount = count($model);
+
         foreach ($model as $item) {
             // increment the counter variable
             $this->partialCounter++;
