Index: library/Zend/Cache/Backend/Memcached.php
===================================================================
--- library/Zend/Cache/Backend/Memcached.php    (revision 20567)
+++ library/Zend/Cache/Backend/Memcached.php    (working copy)
@@ -384,11 +384,12 @@
         $memUsed = 0;
         foreach ($mems as $key => $mem) {
             if ($mem === false) {
-                Zend_Cache::throwException('can\'t get stat from ' . $key);
+                $this->_log('can\'t get stat from ' . $key);
             } else {
                 $eachSize = $mem['limit_maxbytes'];
                 if ($eachSize == 0) {
-                    Zend_Cache::throwException('can\'t get memory size from ' . $key);
+                    $this->_log('can\'t get stat from ' . $key);
+                    continue;
                 }

                 $eachUsed = $mem['bytes'];
@@ -401,6 +402,10 @@
             }
         }

+        if (!$memSize || !$memUsed) {
+            Zend_Cache::throwException('Can\'t get filling percentage');
+        }
+
         return ((int) (100. * ($memUsed / $memSize)));
     }

