Index: File.php
===================================================================
--- File.php	(revision 6822)
+++ File.php	(working copy)
@@ -213,17 +213,17 @@
         $file = $this->_file($id, $expire);
         $path = $this->_path($id);
         $firstTry = true;
-        $result = false;      
+        $result = false;
         if ($this->_options['hashed_directory_level'] > 0) {
             if (!is_writable($path)) {
                 // maybe, we just have to build the directory structure
                 @mkdir($this->_path($id), $this->_options['hashed_directory_umask'], true);
-                @chmod($this->_path($id), $this->_options['hashed_directory_umask']); // see #ZF-320 (this line is required in some configurations)      
+                @chmod($this->_path($id), $this->_options['hashed_directory_umask']); // see #ZF-320 (this line is required in some configurations)
             }
             if (!is_writable($path)) {
-                return false;    
+                return false;
             }
-        }        
+        }
         $fp = @fopen($file, "wb");
         if (!($fp)) {
             return false;
@@ -255,7 +255,7 @@
     {
         $result1 = true;
         $files = @glob($this->_file($id, '*'));
-        if (count($files) == 0) {
+        if ($files === false || count($files) == 0) {
             return false;
         }
         foreach ($files as $file) {
@@ -375,6 +375,9 @@
         $result = true;
         $prefix = $this->_options['file_name_prefix'];
         $glob = @glob($dir . $prefix . '--*');
+        if ($glob === false || count($glob) == 0) {
+            return $result;
+        }
         foreach ($glob as $file)  {
             if (is_file($file)) {
                 if ($mode==Zend_Cache::CLEANING_MODE_ALL) {
@@ -452,6 +455,9 @@
     {
         $filesToRemove = @glob($this->_file($this->_tagCacheId($id, '*'), '*'));
         $result = true;
+        if ($filesToRemove === false || count($filesToRemove) < 1) {
+            return $result;
+        }
         foreach ($filesToRemove as $file) {
             $result = $result && ($this->_remove($file));
         }
@@ -621,6 +627,9 @@
         $path = $this->_path($id);
         if (is_null($expire)) {
             $glob = @glob($path . $this->_idToFileName($id, '*'));
+            if ($glob === false || count($glob) == 0) {
+                return null;
+            }
             $nbr = count($glob);
             if ($nbr == 1) {
                 return $glob[0];
