Index: /home/maners/workspace/zend-framework/library/Zend/View/Helper/FormElement.php
===================================================================
--- /home/maners/workspace/zend-framework/library/Zend/View/Helper/FormElement.php	(revision 5223)
+++ /home/maners/workspace/zend-framework/library/Zend/View/Helper/FormElement.php	(working copy)
@@ -84,7 +84,7 @@
         $info = array(
             'name'    => is_array($name) ? '' : $name,
             'id'      => is_array($name) ? '' : $name,
-            'value'   => $value,
+            'value'   => is_array($value) ? '' : $value,
             'attribs' => $attribs,
             'options' => $options,
             'listsep' => $listsep,
@@ -100,6 +100,9 @@
                 }
             }
         }
+        else if (is_array($value) && isset($value[$name])) {
+        	$info['value'] = $value[$name];
+        }
         
         // force attribs to an array, per note from Orjan Persson.
         settype($info['attribs'], 'array');
