Index: tests/Zend/Form/Decorator/FileTest.php
===================================================================
--- tests/Zend/Form/Decorator/FileTest.php	(revision 25022)
+++ tests/Zend/Form/Decorator/FileTest.php	(working copy)
@@ -170,6 +170,91 @@
         $this->assertRegexp('#<input[^>]*>.*?(content)#s', $file, $file);
     }
 
+    /**
+     * @group ZF-12348
+     */
+    public function testNameWithBelongsTo()
+    {
+        $element = new Zend_Form_Element_File(
+            'foo',
+            array(
+                 'belongsTo' => 'bar',
+                 'view'      => $this->getView(),
+            )
+        );
+        $this->decorator->setElement($element);
+
+        $this->assertSame(
+            PHP_EOL . '<input type="file" name="bar[foo]" id="bar-foo">',
+            $this->decorator->render(null)
+        );
+    }
+
+    /**
+     * @group ZF-12348
+     */
+    public function testNameAsArray()
+    {
+        $element = new Zend_Form_Element_File(
+            'foo',
+            array(
+                 'isArray' => true,
+                 'view'    => $this->getView(),
+            )
+        );
+        $this->decorator->setElement($element);
+
+        $this->assertSame(
+            PHP_EOL . '<input type="file" name="foo[]" id="foo-0">',
+            $this->decorator->render(null)
+        );
+    }
+
+    /**
+     * @group ZF-12348
+     */
+    public function testMultifileElement()
+    {
+        $element = new Zend_Form_Element_File(
+            'foo',
+            array(
+                 'multiFile' => 4,
+                 'view'      => $this->getView(),
+            )
+        );
+        $this->decorator->setElement($element);
+
+        $expected = PHP_EOL . '<input type="file" name="foo[]" id="foo-0">'
+                  . PHP_EOL . '<input type="file" name="foo[]" id="foo-1">'
+                  . PHP_EOL . '<input type="file" name="foo[]" id="foo-2">'
+                  . PHP_EOL . '<input type="file" name="foo[]" id="foo-3">';
+
+        $this->assertSame($expected, $this->decorator->render(null));
+    }
+
+    /**
+     * @group ZF-12348
+     */
+    public function testMultifileElementWithBelongsTo()
+    {
+        $element = new Zend_Form_Element_File(
+            'foo',
+            array(
+                 'multiFile' => 4,
+                 'belongsTo' => 'bar',
+                 'view'      => $this->getView(),
+            )
+        );
+        $this->decorator->setElement($element);
+
+        $expected = PHP_EOL . '<input type="file" name="bar[foo][]" id="bar-foo-0">'
+                  . PHP_EOL . '<input type="file" name="bar[foo][]" id="bar-foo-1">'
+                  . PHP_EOL . '<input type="file" name="bar[foo][]" id="bar-foo-2">'
+                  . PHP_EOL . '<input type="file" name="bar[foo][]" id="bar-foo-3">';
+
+        $this->assertSame($expected, $this->decorator->render(null));
+    }
+
     private function _convertIniToInteger($setting)
     {
         if (!is_numeric($setting)) {
