--- ZendFramework-1.7.5/library/Zend/Dom/Query.php	2008-08-24 23:06:20.000000000 +0200
+++ ZendFramework-1.7.5-new/library/Zend/Dom/Query.php	2009-02-26 10:56:53.000000000 +0100
@@ -177,6 +177,10 @@
         switch ($type) {
             case self::DOC_XML:
                 $success = @$domDoc->loadXML($document);
+                // parsing as html
+                $domDoc = new DOMDocument;
+                $purgedDocument = ereg_replace('<[?]xml(.*)[?]>', '', $document);
+                $success = $success && @$domDoc->loadHTML($purgedDocument);
                 break;
             case self::DOC_HTML:
             case self::DOC_XHTML:
