Index: tests/Zend/View/Helper/FormImageTest.php
===================================================================
--- tests/Zend/View/Helper/FormImageTest.php	(revision 24270)
+++ tests/Zend/View/Helper/FormImageTest.php	(working copy)
@@ -92,6 +92,29 @@
         $this->assertRegexp('/<input[^>]*?name="foo"/', $button);
         $this->assertRegexp('/<input[^>]*?type="image"/', $button);
     }
+    
+    /**
+     * @group ZF-11477
+     */
+    public function testRendersAsHtmlByDefault()
+    {
+        $test = $this->helper->formImage(array(
+            'name' => 'foo',
+        ));
+        $this->assertNotContains(' />', $test);
+    }
+
+    /**
+     * @group ZF-11477
+     */
+    public function testCanRendersAsXHtml()
+    {
+        $this->view->doctype('XHTML1_STRICT');
+        $test = $this->helper->formImage(array(
+            'name' => 'foo',
+        ));
+        $this->assertContains(' />', $test);
+    }
 }
 
 // Call Zend_View_Helper_FormImageTest::main() if this source file is executed directly.
Index: tests/Zend/View/Helper/FormRadioTest.php
===================================================================
--- tests/Zend/View/Helper/FormRadioTest.php	(revision 24270)
+++ tests/Zend/View/Helper/FormRadioTest.php	(working copy)
@@ -429,6 +429,46 @@
         $this->assertRegExp('/<input([^>]*)(id="'.$name.'--1")([^>]*)(checked="checked")/', $html);
     }
     
+    /**
+     * @group ZF-11477
+     */
+    public function testRendersAsHtmlByDefault()
+    {
+        $this->view->doctype('HTML4_STRICT');
+        $options = array(
+            'foo' => 'Foo',
+            'bar' => 'Bar',
+            'baz' => 'Baz'
+        );
+        $html = $this->helper->formRadio(array(
+            'name'    => 'foo',
+            'options' => $options,
+        ));
+
+        $this->assertContains('value="foo">', $html);
+        $this->assertContains('value="bar">', $html);
+        $this->assertContains('value="baz">', $html);
+    }
+
+    /**
+     * @group ZF-11477
+     */
+    public function testCanRendersAsXHtml()
+    {
+        $this->view->doctype('XHTML1_STRICT');
+        $options = array(
+            'foo' => 'Foo',
+            'bar' => 'Bar',
+            'baz' => 'Baz'
+        );
+        $html = $this->helper->formRadio(array(
+            'name'    => 'foo',
+            'options' => $options,
+        ));
+        $this->assertContains('value="foo" />', $html);
+        $this->assertContains('value="bar" />', $html);
+        $this->assertContains('value="baz" />', $html);
+    }
 }
 
 // Call Zend_View_Helper_FormRadioTest::main() if this source file is executed directly.
