Index: library/Zend/Date/DateObject.php
===================================================================
--- library/Zend/Date/DateObject.php	(revision 24665)
+++ library/Zend/Date/DateObject.php	(working copy)
@@ -179,9 +179,10 @@
 
         // correct months > 12 and months < 1
         if ($month > 12) {
-            $overlap = floor($month / 12);
+            $overlap = floor(($month-1) / 12);
             $year   += $overlap;
-            $month  -= $overlap * 12;
+            $month  = ($month % 12);
+            if(!$month) $month = 12;  // December
         } else {
             $overlap = ceil((1 - $month) / 12);
             $year   -= $overlap;
Index: tests/Zend/DateTest.php
===================================================================
--- tests/Zend/DateTest.php	(revision 24665)
+++ tests/Zend/DateTest.php	(working copy)
@@ -5678,6 +5678,25 @@
         $date = new Zend_Date($t, 'yyyy-MM-dd HH:mm:ss');
         $this->assertEquals($t, $date->toString('yyyy-MM-dd HH:mm:ss'));
     }
+    
+    /**
+     * @group ZF-11951
+     */
+    public function testMonthManipulationOfDatesBeforeUnixEpoch()
+    {
+        date_default_timezone_set('Etc/UTC');
+        
+        $year = 1899;
+        foreach ( range(1,50) as $month ) {
+            if ($month % 12 == 1) $year++;
+            $actualMonth = ($month % 12);
+            if (!$actualMonth) $actualMonth = 12;
+                        
+            $date = new Zend_Date(array('year'=>1900, 'month'=>12, 'day'=>19, 'hour'=>21, 'minute'=>58, 'second'=>10));
+            $date->set($month, Zend_Date::MONTH);
+            $this->assertEquals("{$year}-{$actualMonth}", $date->get('YYYY-M'));
+        }
+    }
 
 }
 
