--- tests/Zend/CodeGenerator/Php/PropertyTest.php	(Revision 18385)
+++ tests/Zend/CodeGenerator/Php/PropertyTest.php	(Arbeitskopie)
@@ -67,13 +67,19 @@
             5, 
             'one' => 1,
             'two' => '2',
+            'null' => null,
+            'true' => true,
+            "bar's" => "bar's",
             );
         
         $expectedSource = <<<EOS
     public \$myFoo = array(
         5,
         'one' => 1,
-        'two' => '2'
+        'two' => '2',
+        'null' => null,
+        'true' => true,
+        'bar\'s' => 'bar\'s'
         );
 EOS;
 
