Index: tests/Zend/JsonTest.php
===================================================================
--- tests/Zend/JsonTest.php	(revision 22212)
+++ tests/Zend/JsonTest.php	(working copy)
@@ -755,6 +755,15 @@
     {
         Zend_Json::decode(' some string ');
     }
+
+    /**
+     * @group ZF-9416
+     */
+    public function testIteratorWithoutKey()
+    {
+		$value = new ArrayIterator(array('foo'));
+        $this->_testEncodeDecode(array($value));
+    }
 }
 
 /**
