Index: Zend/Http/Client.php
===================================================================
--- Zend/Http/Client.php	(.../branches/zend-framework/library/incubator)	(revision 791)
+++ Zend/Http/Client.php	(.../trunk/library/incubator)	(revision 791)
@@ -696,6 +696,10 @@
             $this->adapter->write($this->method, $uri, $this->config['httpversion'], $headers, $body);
             $response = Zend_Http_Response::factory($this->adapter->read());
             
+            if ($response->getStatus() == 100) {
+                $response = Zend_Http_Response::factory($response->getBody());
+            }
+
             // Load cookies into cookie jar
             if (isset($this->Cookiejar)) $this->Cookiejar->addCookiesFromResponse($response, $uri);
 
