Index: DijitElement.php
===================================================================
--- DijitElement.php	(revision 12360)
+++ DijitElement.php	(working copy)
@@ -38,7 +38,7 @@
  * @subpackage Form_Decorator
  * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id: $
+ * @version    $Id$
  */
 class Zend_Dojo_Form_Decorator_DijitElement extends Zend_Form_Decorator_ViewHelper
 {
@@ -181,7 +181,13 @@
             } while ($view->dojo()->hasDijit($id));
         }
         $attribs['id'] = $id;
-
+        if(array_key_exists('options', $attribs)){
+        	if(is_array($options)){
+        		$options = array_merge($options, $attribs['options']);
+        	}else{
+        		$options = $attribs['options'];
+        	}
+        }
         $elementContent = $view->$helper($name, $value, $dijitParams, $attribs, $options);
         switch ($this->getPlacement()) {
             case self::APPEND:
@@ -192,4 +198,4 @@
                 return $elementContent;
         }
     }
-}
+}
\ No newline at end of file
