--- library/Zend/Tool/Framework/Client/Console/HelpSystem.php	(revision 16111)
+++ library/Zend/Tool/Framework/Client/Console/HelpSystem.php	(working copy)
@@ -160,6 +160,8 @@
      */
     protected function _respondWithHeader()
     {
+        /** @see Zend_Version */
+        require_once 'Zend/Version.php';
         $this->_response->appendContent('Zend Framework', array('color' => array('hiWhite'), 'separator' => false));
         $this->_response->appendContent(' Command Line Console Tool v' . Zend_Version::VERSION . '');
         return $this;
