Index: library/Zend/Dojo/Form/Decorator/DijitElement.php
===================================================================
--- library/Zend/Dojo/Form/Decorator/DijitElement.php	(revision 19842)
+++ library/Zend/Dojo/Form/Decorator/DijitElement.php	(working copy)
@@ -164,9 +164,8 @@
         $name      = $element->getFullyQualifiedName();
 
         $dijitParams = $this->getDijitParams();
-        if ($element->isRequired()) {
-            $dijitParams['required'] = true;
-        }
+        $dijitParams['required'] = $element->isRequired();
+
 
         $id = $element->getId();
         if ($view->dojo()->hasDijit($id)) {
Index: tests/Zend/Dojo/Form/Decorator/DijitElementTest.php
===================================================================
--- tests/Zend/Dojo/Form/Decorator/DijitElementTest.php	(revision 19842)
+++ tests/Zend/Dojo/Form/Decorator/DijitElementTest.php	(working copy)
@@ -213,6 +213,14 @@
         $html = $this->decorator->render('');
         $this->assertContains('required="', $html);
     }
+
+    public function testRenderingShouldRenderRequiredAttributeWhenElementIsNotRequired()
+    {
+        $this->element->setRequired(false);
+        $html = $this->decorator->render('');
+        $this->assertContains('required="false', $html);
+
+    }
 }
 
 // Call Zend_Dojo_Form_Decorator_DijitElementTest::main() if this source file is executed directly.

