Index: library/Zend/View/Helper/UserAgent.php
===================================================================
--- library/Zend/View/Helper/UserAgent.php	(revision 23462)
+++ library/Zend/View/Helper/UserAgent.php	(working copy)
@@ -33,6 +33,13 @@
 class Zend_View_Helper_UserAgent extends Zend_View_Helper_Abstract
 {
     /**
+     * UserAgent instance
+     * 
+     * @var Zend_Http_UserAgent
+     */
+    protected $_userAgent = null;
+
+    /**
      * Helper method: retrieve or set UserAgent instance
      * 
      * @param  null|Zend_Http_UserAgent $userAgent 
