Index: library/Zend/Loader/Autoloader.php
===================================================================
--- library/Zend/Loader/Autoloader.php	(revision 23865)
+++ library/Zend/Loader/Autoloader.php	(working copy)
@@ -352,7 +352,13 @@
         }
 
         // Add non-namespaced autoloaders
-        $autoloaders = $autoloaders + $this->getNamespaceAutoloaders('');
+        $autoloadersNonNamespace = $this->getNamespaceAutoloaders('');
+        if (count($autoloadersNonNamespace)) {
+            foreach ($autoloadersNonNamespace as $ns) {
+                $autoloaders[] = $ns;
+            }
+            unset($autoloadersNonNamespace);
+        }
 
         // Add fallback autoloader
         if (!$namespace && $this->isFallbackAutoloader()) {
Index: tests/Zend/Loader/AutoloaderTest.php
===================================================================
--- tests/Zend/Loader/AutoloaderTest.php	(revision 23865)
+++ tests/Zend/Loader/AutoloaderTest.php	(working copy)
@@ -405,6 +405,20 @@
         $this->assertEquals(array($this, 'autoloadSecondLevel'), $al);
     }
 
+    /**
+     * @group ZF-10136
+     */
+    public function testMergedAutoloadersWithoutNamespace()
+    {
+        $this->autoloader
+             ->pushAutoloader('autoloadOne')
+             ->pushAutoloader('autoloadSecond');
+
+        $class = 'Zend_Autoloader_Test';
+        $autoloaders = $this->autoloader->getClassAutoloaders($class);
+        $this->assertEquals(3, count($autoloaders));
+    }
+
     public function addTestIncludePath()
     {
         set_include_path(dirname(__FILE__) . '/_files/' . PATH_SEPARATOR . $this->includePath);
