--- Layout.php (saved version)
+++ (current document)
@@ -72,6 +72,12 @@
     protected $_inflectorTarget = ':script.:suffix';
 
     /**
+     * Instance registered with MVC, if any
+     * @var Zend_Layout
+     */
+    protected static $_instance;
+
+    /**
      * Layout view
      * @var string
      */
@@ -90,12 +96,6 @@
     protected $_mvcEnabled = true;
 
     /**
-     * Instance registered with MVC, if any
-     * @var Zend_Layout
-     */
-    protected static $_mvcInstance;
-
-    /**
      * Flag: is MVC successful action only flag set?
      * @var bool
      */
@@ -169,23 +169,34 @@
      */
     public static function startMvc($options = null)
     {
-        if (null === self::$_mvcInstance) {
-            self::$_mvcInstance = new self($options, true);
+        if (null === self::$_instance) {
+            self::$_instance = new self($options, true);
         } else {
-            self::$_mvcInstance->setOptions($options);
+            self::$_instance->setOptions($options);
         }
 
-        return self::$_mvcInstance;
+        return self::$_instance;
+    }
+
+    /**
+     * Retrieve MVC instance of Zend_Layout object
+     * 
+     * @return Zend_Layout|null
+     */
+    public static function getInstance()
+    {
+        return self::$_instance;
     }
 
     /**
      * Retrieve MVC instance of Zend_Layout object
+     * For backwards compatibility
      * 
      * @return Zend_Layout|null
      */
     public static function getMvcInstance()
     {
-        return self::$_mvcInstance;
+        return self::$_instance;
     }
 
     /**
