Index: library/Zend/Feed.php
===================================================================
--- library/Zend/Feed.php	(revision 10737)
+++ library/Zend/Feed.php	(working copy)
@@ -377,8 +377,8 @@
          * @see Zend_Loader
          */
         require_once 'Zend/Loader.php';
-        Zend_Loader::loadClass($obj);
-        Zend_Loader::loadClass('Zend_Feed_Builder');
+        Zend_Loader::loadClass($obj);
+        require_once 'Zend/Feed/Builder.php';
 
         return new $obj(null, null, new Zend_Feed_Builder($data));
     }
Index: library/Zend/Feed/Builder/Header.php
===================================================================
--- library/Zend/Feed/Builder/Header.php	(revision 10737)
+++ library/Zend/Feed/Builder/Header.php	(working copy)
@@ -174,8 +174,8 @@
      * @throws Zend_Feed_Builder_Exception
      */
     public function setEmail($email)
-    {
-        Zend_Loader::loadClass('Zend_Validate_EmailAddress');
+    {
+        require_once 'Zend/Validate/EmailAddress.php';
         $validate = new Zend_Validate_EmailAddress();
         if (!$validate->isValid($email)) {
             /**
@@ -246,7 +246,7 @@
      */
     public function setWebmaster($webmaster)
     {
-        Zend_Loader::loadClass('Zend_Validate_EmailAddress');
+        require_once 'Zend/Validate/EmailAddress.php';
         $validate = new Zend_Validate_EmailAddress();
         if (!$validate->isValid($webmaster)) {
             /**
@@ -269,7 +269,7 @@
      */
     public function setTtl($ttl)
     {
-        Zend_Loader::loadClass('Zend_Validate_Int');
+        require_once 'Zend/Validate/Int.php';
         $validate = new Zend_Validate_Int();
         if (!$validate->isValid($ttl)) {
             /**
Index: library/Zend/Feed/Builder/Header/Itunes.php
===================================================================
--- library/Zend/Feed/Builder/Header/Itunes.php	(revision 10737)
+++ library/Zend/Feed/Builder/Header/Itunes.php	(working copy)
@@ -111,8 +111,8 @@
      */
     public function setOwner($name = '', $email = '')
     {
-        if (!empty($email)) {
-            Zend_Loader::loadClass('Zend_Validate_EmailAddress');
+        if (!empty($email)) {
+            require_once 'Zend/Validate/EmailAddress.php';
             $validate = new Zend_Validate_EmailAddress();
             if (!$validate->isValid($email)) {
                 /**
