--- library/Zend/Application/Resource/Modules.php	(revision 22445)
+++ library/Zend/Application/Resource/Modules.php	(working copy)
@@ -70,6 +70,11 @@
         $default = $front->getDefaultModule();
         $curBootstrapClass = get_class($bootstrap);
         foreach ($modules as $module => $moduleDirectory) {
+            if ($front->getParam('prefixDefaultModule') == 'true' && $module == 'default')
+            {
+                $module = $default;
+                $moduleDirectory = $modules[$module];
+            }
             $bootstrapClass = $this->_formatModuleName($module) . '_Bootstrap';
             if (!class_exists($bootstrapClass, false)) {
                 $bootstrapPath  = dirname($moduleDirectory) . '/Bootstrap.php';
