Index: CheckBoxTest.php
===================================================================
--- CheckBoxTest.php	(revision 12339)
+++ CheckBoxTest.php	(working copy)
@@ -17,7 +17,7 @@
  * @subpackage UnitTests
  * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id:$
+ * @version    $Id$
  */
 
 // Call Zend_Dojo_Form_Element_CheckBoxTest::main() if this source file is executed directly.
@@ -252,6 +252,20 @@
         $html = $this->element->render();
         $this->assertNotContains('options="', $html, $html);
     }
+    
+    /**
+     * @group ZF-4274
+     */
+    public function testCheckedValuesCanBePassedInConstructor()
+    {
+    	$element = new Zend_Dojo_Form_Element_CheckBox('myCheckbox',    array(
+			        'checkedValue' => 'checkedVal',
+					'unCheckedValue' => 'UNCHECKED',
+			    ));
+		$element->setView(new Zend_View());
+		$html = $element->render();
+        $this->assertContains('value="checkedVal"', $html, $html);
+    }
 }
 
 // Call Zend_Dojo_Form_Element_CheckBoxTest::main() if this source file is executed directly.
