--- library/Zend/Form.php	(revision 22797)
+++ library/Zend/Form.php	(working copy)
@@ -2222,7 +2222,7 @@
         }
         $context = $data;
         foreach ($this->getElements() as $key => $element) {
-            if (null !== $translator && !$element->hasTranslator()) {
+            if ($this->hasTranslator() && !$element->hasTranslator()) {
                 $element->setTranslator($translator);
             }
             $check = $data;
