Index: library/Zend/Controller/Request/Http.php
===================================================================
--- library/Zend/Controller/Request/Http.php	(revision 18173)
+++ library/Zend/Controller/Request/Http.php	(working copy)
@@ -505,8 +505,13 @@
                 return $this;
             }
 
+            $truncatedRequestUri = $requestUri;
+            if (($pos = strpos($requestUri, '?')) !== false) {
+                $truncatedRequestUri = substr($requestUri, 0, $pos);
+            }
+            
             $basename = basename($baseUrl);
-            if (empty($basename) || !strpos($requestUri, $basename)) {
+            if (empty($basename) || !strpos($truncatedRequestUri, $basename)) {
                 // no match whatsoever; set it blank
                 $this->_baseUrl = '';
                 return $this;
Index: tests/Zend/Controller/Request/HttpTest.php
===================================================================
--- tests/Zend/Controller/Request/HttpTest.php	(revision 18173)
+++ tests/Zend/Controller/Request/HttpTest.php	(working copy)
@@ -498,6 +498,19 @@
 
         $this->assertEquals('/index.php', $request->getBaseUrl());
     }
+    /**
+     * @group ZF-7824
+     */
+    public function testSetBaseUrlWithScriptNameAsGetParam()
+    {
+        $request = new Zend_Controller_Request_Http;
+        
+        $_SERVER['REQUEST_URI']     = '/article/archive?foo=index.php';
+        $_SERVER['QUERY_STRING']    = 'foo=index.php';
+        $_SERVER['SCRIPT_FILENAME'] = '/var/www/zftests/index.php';
+        
+        $this->assertEquals('/article/archive', $request->getPathInfo());
+    }
 
     public function testGetSetBasePath()
     {
