Index: Json.php
===================================================================
--- Json.php	(Revision 1561)
+++ Json.php	(Arbeitskopie)
@@ -46,9 +46,13 @@
      */
     static public function decode($encodedValue, $objectDecodeType = null)
     {
-        // @todo Zend::loadClass()
-        include_once 'Zend/Json/Decoder.php';
-        return Zend_Json_Decoder::decode($encodedValue, $objectDecodeType);
+        if (function_exists('json_decode')) {
+            return json_decode($encodedValue);
+        } else {
+            // @todo Zend::loadClass()
+            include_once 'Zend/Json/Decoder.php';
+            return Zend_Json_Decoder::decode($encodedValue, $objectDecodeType);
+        }
     }
 
 
@@ -65,9 +69,13 @@
      */
     static public function encode($valueToEncode)
     {
-        // @todo Zend::loadClass()
-        include_once 'Zend/Json/Encoder.php';
-    	return Zend_Json_Encoder::encode($valueToEncode);
+        if (function_exists('json_encode')) {
+            return json_encode($valueToEncode);
+        } else {
+            // @todo Zend::loadClass()
+            include_once 'Zend/Json/Encoder.php';
+            return Zend_Json_Encoder::encode($valueToEncode);
+        }
     }
 }
 
