Index: library/Zend/Application/Resource/Translate.php
===================================================================
--- library/Zend/Application/Resource/Translate.php	(revision 22797)
+++ library/Zend/Application/Resource/Translate.php	(working copy)
@@ -67,7 +67,7 @@
         if (null === $this->_translate) {
             $options = $this->getOptions();
 
-            if (!isset($options['data'])) {
+            if (!isset($options['content']) && !isset($options['data'])) {
                 require_once 'Zend/Application/Resource/Exception.php';
                 throw new Zend_Application_Resource_Exception('No translation source data provided.');
             }
@@ -77,6 +77,7 @@
             }
 
             if (!empty($options['data'])) {
+            	trigger_error(__METHOD__ . ': the option "data" is deprecated as of 1.10.x and will be removed with 2.0.0; use "content" instead', E_USER_NOTICE);
                 $options['content'] = $options['data'];
                 unset($options['data']);
             }
