Index: Wsdl.php
===================================================================
--- Wsdl.php	(revision 21292)
+++ Wsdl.php	(working copy)
@@ -424,7 +424,7 @@
         }
 
         $doc = $this->_dom->createElement('documentation');
-        $doc_cdata = $this->_dom->createTextNode($documentation);
+        $doc_cdata = $this->_dom->createTextNode(str_replace(array("\r\n", "\r"), "\n", $documentation));
         $doc->appendChild($doc_cdata);
 
         if($node->hasChildNodes()) {
