Index: Select.php
===================================================================
--- Select.php	(revision 18738)
+++ Select.php	(working copy)
@@ -798,6 +798,7 @@
             list($schema, $tableName) = explode('.', $tableName);
         }
 
+        $lastFromCorrelationName = null;
         if (!empty($correlationName)) {
             if (array_key_exists($correlationName, $this->_parts[self::FROM])) {
                 /**
@@ -807,7 +808,6 @@
                 throw new Zend_Db_Select_Exception("You cannot define a correlation name '$correlationName' more than once");
             }
             
-            $lastFromCorrelationName = null;
             if ($type == self::FROM) {
                 // append this from after the last from joinType
                 $tmpFromParts = $this->_parts[self::FROM];
