Index: library/Zend/Cache/Manager.php
===================================================================
--- library/Zend/Cache/Manager.php	(revision 21139)
+++ library/Zend/Cache/Manager.php	(working copy)
@@ -164,6 +164,11 @@
                 $this->_optionTemplates[$name]['backend']['options']['tag_cache']
                     = $this->getCache(self::PAGETAGCACHE );
             }
+
+            /**
+             * @see Zend_Cache
+             */
+            require_once 'Zend/Cache.php';
             $this->_caches[$name] = Zend_Cache::factory(
                 $this->_optionTemplates[$name]['frontend']['name'],
                 $this->_optionTemplates[$name]['backend']['name'],
