Index: library/Zend/Db/Table/Abstract.php
===================================================================
--- library/Zend/Db/Table/Abstract.php	(Revision 9236)
+++ library/Zend/Db/Table/Abstract.php	(Arbeitskopie)
@@ -750,16 +750,16 @@
             self::DEPENDENT_TABLES => $this->_dependentTables,
             self::SEQUENCE         => $this->_sequence
         );
-        
+
         if ($key === null) {
             return $info;
         }
-        
+
         if (!array_key_exists($key, $info)) {
             require_once 'Zend/Db/Table/Exception.php';
             throw new Zend_Db_Table_Exception('There is no table information for the key "' . $key . '"');
         }
-        
+
         return $info[$key];
     }
 
@@ -895,6 +895,18 @@
      */
     public function delete($where)
     {
+        $args = func_get_args();
+        if (count($args) > 1) {
+            $currentArg = 1;
+            while (strpos($text, '?') != false && isset($args[$currentArg])) {
+                $where = substr_replace($where,
+                                        $this->_db->quote($args[$currentArg]),
+                                        strpos($where, '?'),
+                                        1);
+                $currentArg++;
+            }
+        }
+
         $tableSpec = ($this->_schema ? $this->_schema . '.' : '') . $this->_name;
         return $this->_db->delete($tableSpec, $where);
     }
