Index: tests/Zend/File/Transfer/Adapter/HttpTest.php
===================================================================
--- tests/Zend/File/Transfer/Adapter/HttpTest.php	(revision 25051)
+++ tests/Zend/File/Transfer/Adapter/HttpTest.php	(working copy)
@@ -148,6 +148,18 @@
         }
     }
 
+    /**
+     * @group ZF-12451
+     */
+    public function testReceiveEmptyArray()
+    {
+        $_SERVER['CONTENT_LENGTH'] = 10;
+        $_FILES = array();
+
+        $adapter = new Zend_File_Transfer_Adapter_Http();
+        $this->assertFalse($adapter->receive(array()));
+    }
+
     public function testReceiveValidatedFile()
     {
         $_FILES = array(
