--- /C/wamp/apache/htdocs/zendframework/library/Zend/Translate/Plural.php.orig	Wed Jul 22 09:05:46 2009
+++ /C/wamp/apache/htdocs/zendframework/library/Zend/Translate/Plural.php	Wed Aug  5 22:26:54 2009
@@ -138,7 +138,7 @@
 
             case 'cs':
             case 'sk':
-                return ($number == 1) ? 0 : (($number >= 2) && ($number <= 4)) ? 1 : 2;
+                return ($number == 1) ? 0 : (($number >= 2 && $number <= 4) ? 1 : 2);
 
             case 'ga':
                 return ($number == 1) ? 0 : ($number == 2) ? 1 : 2;
