Index: IsCompressed.php
===================================================================
--- IsCompressed.php	(revision 15)
+++ IsCompressed.php	(working copy)
@@ -81,6 +81,6 @@
             );
         }
 
-        $this->setMimeType($mimetype);
+        parent::__construct($mimetype);
     }
 }
Index: IsImage.php
===================================================================
--- IsImage.php	(revision 15)
+++ IsImage.php	(working copy)
@@ -85,6 +85,6 @@
             );
         }
 
-        $this->setMimeType($mimetype);
+        parent::__construct($mimetype);
     }
 }
