--- library/Zend/View/Helper/Navigation.php	(revision 22945)
+++ library/Zend/View/Helper/Navigation.php	(working copy)
@@ -156,10 +156,13 @@
             return $this->_helpers[$proxy];
         }
 
-        if (!$this->view->getPluginLoader('helper')->getPaths(self::NS)) {
-            $this->view->addHelperPath(
-                    str_replace('_', '/', self::NS),
-                    self::NS);
+        $paths = $this->view->getPluginLoader('helper')->getPaths();
+        foreach($paths AS $ns => $path) {
+            if (!$this->view->getPluginLoader('helper')->getPaths($ns . 'Navigation_')) {
+                $this->view->addHelperPath(
+                        str_replace('_', '/', $ns . 'Navigation_'),
+                        $ns . 'Navigation');
+            }
         }
 
         if ($strict) {
