diff --git a/Zend/Form/Element.php b/Zend/Form/Element.php
index 16333c8..625998c 100644
--- a/Zend/Form/Element.php
+++ b/Zend/Form/Element.php
@@ -338,7 +338,11 @@ class Zend_Form_Element implements Zend_Validate_Interface
             unset($options['disableTranslator']);
         }
 
-        unset($options['options']);
+        if (array_key_exists('options', $options)) {
+            $this->options = $options['options'];
+            unset($options['options']);
+        }
+
         unset($options['config']);
 
         foreach ($options as $key => $value) {
