Index: library/Zend/Translate/Adapter.php
===================================================================
--- library/Zend/Translate/Adapter.php	(revision 23369)
+++ library/Zend/Translate/Adapter.php	(working copy)
@@ -464,7 +464,7 @@
     /**
      * Returns the available languages from this adapter
      *
-     * @return array
+     * @return array|null
      */
     public function getList()
     {
