Index: Config.php
===================================================================
--- Config.php	(revision 6362)
+++ Config.php	(working copy)
@@ -280,8 +280,34 @@
     {
         return $this->_loadedSection === null;
     }
-
+    
+    
     /**
+     * Merge another Zend_Config with this one. The items
+     * in $merge will override the same named items in
+     * the current config.
+     *
+     * @param Zend_Config $merge
+     * @return Zend_Config
+     */
+    public function merge(Zend_Config $merge)
+    {
+        foreach($merge as $key => $item) {
+            if(array_key_exists($key, $this->_data)) {
+                if(get_class($item) == 'Zend_Config' && get_class($this->$key) == 'Zend_Config') {
+                    $this->$key = $this->$key->merge($item);
+                } else {
+                    $this->$key = $item;
+                }
+            } else {
+                $this->$key = $item;
+            }
+        }
+        
+        return $this;
+    }
+    
+    /**
      * Throws an exception if $extendingSection may not extend $extendedSection,
      * and tracks the section extension if it is valid.
      *
