Index: library/Zend/Service/Amazon/Ec2/Instance.php
===================================================================
--- library/Zend/Service/Amazon/Ec2/Instance.php	(revision 21805)
+++ library/Zend/Service/Amazon/Ec2/Instance.php	(working copy)
@@ -282,9 +282,9 @@
                 $item['amiLaunchIndex'] = $xpath->evaluate('string(ec2:amiLaunchIndex/text())', $is_node);
                 $item['monitoringState'] = $xpath->evaluate('string(ec2:monitoring/ec2:state/text())', $is_node);
 
+                $return['instances'][] = $item;
                 unset($is_node);
             }
-            $return['instances'][] = $item;
             unset($item);
             unset($is);
         }
