Index: library/Zend/Config/Writer/Xml.php
===================================================================
--- library/Zend/Config/Writer/Xml.php	(revision 18173)
+++ library/Zend/Config/Writer/Xml.php	(working copy)
@@ -184,7 +184,7 @@
 
             if ($branchType === 'numeric') {
                 if ($value instanceof Zend_Config) {
-                    $child = $parent->addChild($branchName, (string) $value);
+                    $child = $parent->addChild($branchName);
 
                     $this->_addBranch($value, $child, $parent);
                 } else {
