--- library/Zend/Queue/Adapter/Memcacheq.php	(revision 23351)
+++ library/Zend/Queue/Adapter/Memcacheq.php	(working copy)
@@ -218,7 +218,8 @@
         $response = $this->_sendCommand('stats queue', array('END'));
 
         foreach ($response as $i => $line) {
-            $this->_queues[] = str_replace('STAT ', '', $line);
+            // ZF-9969: output is changed since memcacheq 0.2.0
+            $this->_queues[] = preg_replace('~STAT (\w+)(?: [0-9]+/[0-9]+)?~', '\\1', $line);
         }
 
         return $this->_queues;
