Index: DefaultValue.php
===================================================================
--- DefaultValue.php	(revision 18037)
+++ DefaultValue.php	(working copy)
@@ -266,12 +266,14 @@
             case self::TYPE_STRING:
                 $output .= "'" . $value . "'";
                 break;
+            case self::TYPE_NULL:
+                $output .= 'NULL';
+                break;
             case self::TYPE_NUMBER:
             case self::TYPE_INTEGER:
             case self::TYPE_INT:
             case self::TYPE_FLOAT:
             case self::TYPE_DOUBLE:
-            case self::TYPE_NULL:
             case self::TYPE_CONSTANT:
                 $output .= $value;
                 break;
