Index: Zend/Form/Element/RadioTest.php
===================================================================
--- Zend/Form/Element/RadioTest.php	(revision 21857)
+++ Zend/Form/Element/RadioTest.php	(working copy)
@@ -186,6 +186,21 @@
     }
 
     /**
+     * @group ZF-9682
+     */
+    public function testCustomLabelDecorator()
+    {
+        $form = new Zend_Form();
+        $form->addElementPrefixPath('My_Decorator', dirname(__FILE__) . '/../_files/decorators/', 'decorator');
+
+        $form->addElement($this->element);
+
+        $element = $form->getElement('foo');
+
+        $this->assertType('My_Decorator_Label', $element->getDecorator('Label'));
+    }
+
+    /**
      * Used by test methods susceptible to ZF-2794, marks a test as incomplete
      *
      * @link   http://framework.zend.com/issues/browse/ZF-2794
