Index: Dojo/View/Helper/Dijit.php
===================================================================
--- Dojo/View/Helper/Dijit.php	(revision 11549)
+++ Dojo/View/Helper/Dijit.php	(working copy)
@@ -212,7 +212,7 @@
                         if (!is_string($value)) {
                             continue;
                         }
-                        $values[] = $value;
+                        $values[$key] = $value;
                     }
                 } elseif (is_string($params[$param])) {
                     $values = (array) $params[$param];
