Index: HeadTitle.php
===================================================================
--- HeadTitle.php	(revision 22029)
+++ HeadTitle.php	(working copy)
@@ -58,7 +58,6 @@
      *
      * @param  string $title
      * @param  string $setType
-     * @param  string $separator
      * @return Zend_View_Helper_HeadTitle
      */
     public function headTitle($title = null, $setType = Zend_View_Helper_Placeholder_Container_Abstract::APPEND)
