Index: tests/Zend/Test/PHPUnit/_files/application/views/scripts/zend-test-php-unit-foo/baz.phtml
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/views/scripts/zend-test-php-unit-foo/baz.phtml	(revision 21561)
+++ tests/Zend/Test/PHPUnit/_files/application/views/scripts/zend-test-php-unit-foo/baz.phtml	(working copy)
@@ -12,6 +12,9 @@
     <fieldset>
         <legend class="bat">La di da</legend>
     </fieldset>
+    <fieldset>
+        <legend class="numeric">42</legend>
+    </fieldset>
 </div>
 </body>
 </html>
Index: tests/Zend/Test/PHPUnit/ControllerTestCaseTest.php
===================================================================
--- tests/Zend/Test/PHPUnit/ControllerTestCaseTest.php	(revision 21561)
+++ tests/Zend/Test/PHPUnit/ControllerTestCaseTest.php	(working copy)
@@ -305,6 +305,8 @@
         $this->testCase->assertQuery('div#foo legend.bat', $body);
         $this->testCase->assertNotQuery('div#foo legend.bogus', $body);
         $this->testCase->assertQueryContentContains('legend.bat', 'La di da', $body);
+        $this->testCase->assertQueryContentContains('legend.numeric', 42, $body);
+        $this->testCase->assertNotQueryContentContains('legend.numeric', 31, $body);
         $this->testCase->assertNotQueryContentContains('legend.bat', 'La do da', $body);
         $this->testCase->assertQueryContentRegex('legend.bat', '/d[a|i]/i', $body);
         $this->testCase->assertNotQueryContentRegex('legend.bat', '/d[o|e]/i', $body);
Index: library/Zend/Test/PHPUnit/Constraint/DomQuery.php
===================================================================
--- library/Zend/Test/PHPUnit/Constraint/DomQuery.php	(revision 21561)
+++ library/Zend/Test/PHPUnit/Constraint/DomQuery.php	(working copy)
@@ -275,6 +275,8 @@
      */
     protected function _matchContent($result, $match)
     {
+        $match = (string) $match;
+
         if (0 == count($result)) {
             return false;
         }
