Index: library/Zend/Navigation/Page/Mvc.php
===================================================================
--- library/Zend/Navigation/Page/Mvc.php	(revision 24692)
+++ library/Zend/Navigation/Page/Mvc.php	(working copy)
@@ -233,9 +233,7 @@
         $fragment = $this->getFragment();       
         if (null !== $fragment) {
             $url .= '#' . $fragment;
-        }         
-        
-         return $this->_hrefCache = $url;
+        }
 
         return $this->_hrefCache = $url;
     }
