Index: tests/Zend/Form/ElementTest.php
===================================================================
--- tests/Zend/Form/ElementTest.php	(revision 20213)
+++ tests/Zend/Form/ElementTest.php	(working copy)
@@ -1692,7 +1692,7 @@
 
         $options = $this->getOptions();
         $options['validators'] = array(
-            array('notEmpty', true, array('bar')),
+            array('notEmpty', true, array(Zend_Validate_NotEmpty::ALL)),
             array('digits', true, array('bar')),
         );
         $this->element->setOptions($options);
@@ -1711,7 +1711,7 @@
         $options = $this->getOptions();
         $options['validators'] = array(
             array(
-                'options'             => array('bar'),
+                'options'             => array(Zend_Validate_NotEmpty::ALL),
                 'breakChainOnFailure' => true,
                 'validator'           => 'notEmpty',
             ),

