Index: tests/Zend/Form/Decorator/FieldsetTest.php
===================================================================
--- tests/Zend/Form/Decorator/FieldsetTest.php	(revision 25022)
+++ tests/Zend/Form/Decorator/FieldsetTest.php	(working copy)
@@ -159,7 +159,7 @@
     /**#@+
      * @group ZF-3731
      */
-    public function testIdShouldBePrefixedWithFieldset()
+    public function testIdShouldBeAppendedWithFieldset()
     {
         $form = new Zend_Form();
         $form->setAction('/foo/bar')
@@ -168,7 +168,7 @@
              ->setView($this->getView());
         $this->decorator->setElement($form);
         $test = $this->decorator->render('content');
-        $this->assertContains('id="fieldset-foobar"', $test);
+        $this->assertContains('id="foobar-fieldset"', $test);
     }
 
     public function testElementWithNoIdShouldNotCreateFieldsetId()
@@ -255,7 +255,7 @@
         
         $html = $this->decorator->setElement($form)->render('content');
         
-        $this->assertContains('<fieldset id="fieldset-form-id"', $html);
+        $this->assertContains('<fieldset id="form-id-fieldset"', $html);
     }
 }
 
