Index: Zend/Uri/Http.php
===================================================================
--- Zend/Uri/Http.php	(revision 4954)
+++ Zend/Uri/Http.php	(working copy)
@@ -253,7 +253,11 @@
 
         $password = strlen($this->_password) > 0 ? ":$this->_password" : '';
         $auth     = strlen($this->_username) > 0 ? "$this->_username$password@" : '';
-        $port     = strlen($this->_port) > 0 ? ":$this->_port" : '';
+        if((int)$this->_port !== $this->getDefaultPort()) {
+            $port = strlen($this->_port) > 0 ? ":$this->_port" : '';
+        } else {
+            $port = '';
+        }
         $query    = strlen($this->_query) > 0 ? "?$this->_query" : '';
         $fragment = strlen($this->_fragment) > 0 ? "#$this->_fragment" : '';
 
@@ -267,7 +271,25 @@
              . $fragment;
     }
 
+
     /**
+     * Returns the default port for the current scheme.
+     * 
+     * @return bool|int
+     */
+    public function getDefaultPort()
+    {
+        if($this->_scheme == 'http') {
+            return 80;
+        } elseif($this->_scheme = 'https') {
+            return 443;
+        } else {
+            return false;
+        }
+    }
+
+
+    /**
      * Validate the current URI from the instance variables. Returns true if and only if all
      * parts pass validation.
      *
