Index: library/Zend/Pdf/Page.php
===================================================================
--- library/Zend/Pdf/Page.php	(revision 19744)
+++ library/Zend/Pdf/Page.php	(working copy)
@@ -284,7 +284,7 @@
                     $param1 = Zend_Pdf_Page::SIZE_LETTER_LANDSCAPE;
                     break;
                 default:
-                    // should be in "x:y" form
+                    // should be in "x:y:" form
             }
 
             $pageDim = explode(':', $param1);
