Index: CloudWatch.php
===================================================================
--- CloudWatch.php	(revision 18719)
+++ CloudWatch.php	(working copy)
@@ -280,7 +280,8 @@
                 $options[$dimKey . '.member.' . $x++] = $dimVal;
             }
         }
-
+        unset( $options['Dimensions'] );
+        
         $params = array_merge($params, $options);
 
         $response = $this->sendRequest($params);
