--- tests/Zend/Queue/Adapter/DbTest.php	(revision 23443)
+++ tests/Zend/Queue/Adapter/DbTest.php	(working copy)
@@ -138,5 +138,46 @@
             }
         }
     }
+
+    /**
+     * Confirm that default table names are 'queue' and 'message'
+     */
+    public function testDefaultTableNames()
+    {
+        if (TESTS_ZEND_QUEUE_DB === false) {
+            return $this->markTestSkipped();
+        }
+
+        $config = $this->getTestConfig();
+        $queue  = $this->createQueue(__FUNCTION__, $config);
+
+        $queueTable = $queue->getAdapter()->getQueueTable();
+        $messageTable = $queue->getAdapter()->getMessageTable();
+
+        $this->assertEquals('queue', $queueTable->info(Zend_Queue_Adapter_Db_Queue::NAME));
+        $this->assertEquals('message', $messageTable->info(Zend_Queue_Adapter_Db_Message::NAME));
+    }
+
+    /**
+     * @group ZF-10597
+     */
+    public function testCustomTableNames()
+    {
+        if (TESTS_ZEND_QUEUE_DB === false) {
+            return $this->markTestSkipped();
+        }
+
+        $config = $this->getTestConfig();
+        $config['queueTableName'] = 'foo_queue';
+        $config['messageTableName'] = 'foo_message';
+
+        $queue = $this->createQueue(__FUNCTION__, $config);
+
+        $queueTable = $queue->getAdapter()->getQueueTable();
+        $messageTable = $queue->getAdapter()->getMessageTable();
+
+        $this->assertEquals('foo_queue', $queueTable->info(Zend_Queue_Adapter_Db_Queue::NAME));
+        $this->assertEquals('foo_message', $messageTable->info(Zend_Queue_Adapter_Db_Message::NAME));
+    }
 }
 
Index: library/Zend/Queue/Adapter/Db.php
===================================================================
--- library/Zend/Queue/Adapter/Db.php	(revision 23443)
+++ library/Zend/Queue/Adapter/Db.php	(working copy)
@@ -99,14 +99,27 @@
             $db = $this->_initDbAdapter();
         }
 
+        if (isset($this->_options['queueTableName'])) {
+            $queueTableName = $this->_options['queueTableName'];
+        } else {
+            $queueTableName = 'queue';
+        }
+
+        if (isset($this->_options['messageTableName'])) {
+            $messageTableName = $this->_options['messageTableName'];
+        } else {
+            $messageTableName = 'message';
+        }
+
         $this->_queueTable = new Zend_Queue_Adapter_Db_Queue(array(
-            'db' => $db,
+            'db'   => $db,
+            'name' => $queueTableName
         ));
 
         $this->_messageTable = new Zend_Queue_Adapter_Db_Message(array(
-            'db' => $db,
+            'db'   => $db,
+            'name' => $messageTableName
         ));
-
     }
 
     /**
@@ -533,4 +546,24 @@
 
         return $this->_queues[$name];
     }
+
+    /**
+     * Get queue table instance
+     *
+     * @return Zend_Queue_Adapter_Db_Queue
+     */
+    public function getQueueTable()
+    {
+        return $this->_queueTable;
+    }
+
+    /**
+     * Get message table instance
+     *
+     * @return Zend_Queue_Adapter_Db_Message
+     */
+    public function getMessageTable()
+    {
+        return $this->_messageTable;
+    }
 }
Index: documentation/manual/en/module_specs/Zend_Queue-Adapters.xml
===================================================================
--- documentation/manual/en/module_specs/Zend_Queue-Adapters.xml	(revision 23443)
+++ documentation/manual/en/module_specs/Zend_Queue-Adapters.xml	(working copy)
@@ -172,7 +172,9 @@
     'options' => array(
         // use Zend_Db_Select for update, not all databases can support this
         // feature.
-        Zend_Db_Select::FOR_UPDATE => true
+        Zend_Db_Select::FOR_UPDATE => true,
+        'queueTableName'           => 'queue', // optional parameter, default 'queue'
+        'messageTableName'         => 'message' // optional parameter, default 'message'
     )
 );
 
