Index: tests/Zend/Db/Statement/Pdo/TestCommon.php
===================================================================
--- tests/Zend/Db/Statement/Pdo/TestCommon.php	(revision 20278)
+++ tests/Zend/Db/Statement/Pdo/TestCommon.php	(working copy)
@@ -94,8 +94,8 @@
         } catch (Zend_Exception $e) {
             $this->assertType('Zend_Db_Statement_Exception', $e,
                 'Expecting object of type Zend_Db_Statement_Exception, got '.get_class($e));
-            $this->assertTrue($e->hasChainedException());
-            $this->assertType('PDOException', $e->getChainedException());
+            $this->assertTrue($e->hasChainedException(), 'Missing Chaired Exception');
+            $this->assertType('PDOException', $e->getChainedException(), 'Wrong type of Exception');
         }
     }
 
Index: tests/Zend/Db/TestSetup.php
===================================================================
--- tests/Zend/Db/TestSetup.php	(revision 20278)
+++ tests/Zend/Db/TestSetup.php	(working copy)
@@ -117,7 +117,9 @@
     public function tearDown()
     {
         $this->_util->tearDown();
+        $this->_db->closeConnection();
         $this->_db = null;
+        parent::tearDown();
     }
 
 }
Index: tests/Zend/Db/Table/Select/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/Select/TestCommon.php	(revision 20278)
+++ tests/Zend/Db/Table/Select/TestCommon.php	(working copy)
@@ -62,6 +62,7 @@
         if ($this->_runtimeIncludePath) {
             $this->_restoreIncludePath();
         }
+        parent::tearDown();
     }
 
     protected function _getTable($tableClass, $options = array())
Index: tests/Zend/Db/Table/TestSetup.php
===================================================================
--- tests/Zend/Db/Table/TestSetup.php	(revision 20278)
+++ tests/Zend/Db/Table/TestSetup.php	(working copy)
@@ -64,6 +64,7 @@
         if ($this->_runtimeIncludePath) {
             $this->_restoreIncludePath();
         }
+        parent::tearDown();
     }
 
     protected function _getTable($tableClass, $options = array())
Index: library/Zend/Db/Statement/Exception.php
===================================================================
--- library/Zend/Db/Statement/Exception.php	(revision 20278)
+++ library/Zend/Db/Statement/Exception.php	(working copy)
@@ -43,7 +43,7 @@
      */
     public function hasChainedException()
     {
-        return ($this->_previous !== null);
+        return ($this->getPrevious() !== null);
     }
 
     /**
@@ -51,6 +51,6 @@
      */
     public function getChainedException()
     {
-        return $this->_previous;
+        return $this->getPrevious();
     }
 }
