Index: ProgressBar.php
===================================================================
--- ProgressBar.php	(revision 165)
+++ ProgressBar.php	(working copy)
@@ -196,14 +196,15 @@
     /**
      * Call the adapters finish() behaviour
      *
+     * @param array $parameter
      * @return void
      */
-    public function finish()
+    public function finish( $parameter = null )
     {
         if ($this->_persistenceNamespace !== null) {
             unset($this->_persistenceNamespace->isSet);
         }
 
-        $this->_adapter->finish();
+        $this->_adapter->finish( $parameter );
     }
 }
Index: ProgressBar/Adapter.php
===================================================================
--- ProgressBar/Adapter.php	(revision 165)
+++ ProgressBar/Adapter.php	(working copy)
@@ -107,7 +107,8 @@
     /**
      * Called when the progress is explicitly finished
      *
+     * @param array $parameter
      * @return void
      */
-    abstract public function finish();
+    abstract public function finish($parameter = null);
 }
Index: ProgressBar/Adapter/JsPush.php
===================================================================
--- ProgressBar/Adapter/JsPush.php	(revision 165)
+++ ProgressBar/Adapter/JsPush.php	(working copy)
@@ -112,16 +112,17 @@
     /**
      * Defined by Zend_ProgressBar_Adapter_Interface
      *
+     * @param array $parameter
      * @return void
      */
-    public function finish()
+    public function finish($parameter = null)
     {
         if ($this->_finishMethodName === null) {
             return;
         }
-
+        
         $data = '<script type="text/javascript">'
-              . 'parent.' . $this->_finishMethodName . '();'
+              . 'parent.' . $this->_finishMethodName . '(' . Zend_Json::encode($parameter) . ');'
               . '</script>';
 
         $this->_outputData($data);
