Index: manual-print2.xml.in
===================================================================
--- manual-print2.xml.in	(revision 24879)
+++ manual-print2.xml.in	(working copy)
@@ -197,7 +197,6 @@
         <xi:include href="module_specs/Zend_Service_Flickr.xml" />
         <xi:include href="module_specs/Zend_Service_Nirvanix.xml" />
         <xi:include href="module_specs/Zend_Service-ReCaptcha.xml" />
-        <xi:include href="module_specs/Zend_Service_Simpy.xml" />
         <xi:include href="module_specs/Zend_Service_SlideShare.xml" />
         <xi:include href="module_specs/Zend_Service_StrikeIron-Overview.xml" />
         <xi:include href="module_specs/Zend_Service_StrikeIron-BundledServices.xml" />
Index: manual.xml.in
===================================================================
--- manual.xml.in	(revision 24879)
+++ manual.xml.in	(working copy)
@@ -2006,11 +2006,6 @@
                     <xi:include href="../en/module_specs/Zend_Service_ShortUrl.xml" />
                 </xi:fallback>
             </xi:include>
-            <xi:include href="module_specs/Zend_Service_Simpy.xml">
-                <xi:fallback>
-                    <xi:include href="../en/module_specs/Zend_Service_Simpy.xml" />
-                </xi:fallback>
-            </xi:include>
             <xi:include href="module_specs/Zend_Service_SlideShare.xml">
                 <xi:fallback>
                     <xi:include href="../en/module_specs/Zend_Service_SlideShare.xml" />
