--- library/Zend/Service/Rackspace/Files.php	(revision 25005)
+++ library/Zend/Service/Rackspace/Files.php	(working copy)
@@ -43,13 +43,13 @@
     const ERROR_PARAM_NO_METADATA              = 'You must specify the metadata array';
     const ERROR_CDN_TTL_OUT_OF_RANGE           = 'TTL must be a number in seconds, min is 900 sec and maximum is 1577836800 (50 years)';
     const ERROR_PARAM_UPDATE_CDN               = 'You must specify at least one the parameters: ttl, cdn_enabled or log_retention';
-    const HEADER_CONTENT_TYPE                  = 'Content-type';
+    const HEADER_CONTENT_TYPE                  = 'Content-Type';
     const HEADER_HASH                          = 'Etag';
-    const HEADER_LAST_MODIFIED                 = 'Last-modified';
-    const HEADER_CONTENT_LENGTH                = 'Content-length';
+    const HEADER_LAST_MODIFIED                 = 'Last-Modified';
+    const HEADER_CONTENT_LENGTH                = 'Content-Length';
     const HEADER_COPY_FROM                     = 'X-Copy-From';
-    const METADATA_OBJECT_HEADER               = "X-object-meta-";
-    const METADATA_CONTAINER_HEADER            = "X-container-meta-";
+    const METADATA_OBJECT_HEADER               = "X-Object-Meta-";
+    const METADATA_CONTAINER_HEADER            = "X-Container-Meta-";
     const CDN_URI                              = "X-CDN-URI";
     const CDN_SSL_URI                          = "X-CDN-SSL-URI";
     const CDN_ENABLED                          = "X-CDN-Enabled";
