diff -ur /home/dolly/installer/Zend-framework/ZendFramework-1.9.1/library/Zend/Tool/Project/Context/Zf/ControllerFile.php Tool/Project/Context/Zf/ControllerFile.php
--- /home/dolly/installer/Zend-framework/ZendFramework-1.9.1/library/Zend/Tool/Project/Context/Zf/ControllerFile.php	2009-07-23 01:44:24.000000000 +0700
+++ Tool/Project/Context/Zf/ControllerFile.php	2009-08-28 16:57:11.000000000 +0700
@@ -26,6 +26,11 @@
 require_once 'Zend/Tool/Project/Context/Filesystem/File.php';
 
 /**
+ * @see Zend_Tool_Project_Context_Filesystem_File
+ */
+require_once 'Zend/Tool/Project/Context/Filesystem/File.php';
+
+/**
  * @see Zend_CodeGenerator_Php_File
  */
 require_once 'Zend/CodeGenerator/Php/File.php';
@@ -111,10 +116,16 @@
      */
     public function getContents()
     {
-
-        $filter = new Zend_Filter_Word_DashToCamelCase();
+        $this->_controllerName = $this->_resource->getAttribute('controllerName');
+        $filter  = new Zend_Filter_Word_DashToCamelCase();
+        $moduleName = $this->_resource->getAttribute('moduleName');
         
-        $className = $filter->filter($this->_controllerName) . 'Controller';
+        if (! is_null($moduleName)) {
+        	$className = ucfirst($moduleName) . '_' . $filter->filter($this->_controllerName)
+                           . 'Controller';
+        } else {
+        	$className = $filter->filter($this->_controllerName) . 'Controller';
+        }
         
         $codeGenFile = new Zend_CodeGenerator_Php_File(array(
             'fileName' => $this->getPath(),
Only in Tool/Project/Profile/Iterator: EnabledResource.php
diff -ur /home/dolly/installer/Zend-framework/ZendFramework-1.9.1/library/Zend/Tool/Project/Provider/Controller.php Tool/Project/Provider/Controller.php
--- /home/dolly/installer/Zend-framework/ZendFramework-1.9.1/library/Zend/Tool/Project/Provider/Controller.php	2009-07-23 01:05:45.000000000 +0700
+++ Tool/Project/Provider/Controller.php	2009-08-28 16:52:48.000000000 +0700
@@ -81,7 +81,9 @@
             throw new Zend_Tool_Project_Provider_Exception($exceptionMessage);
         }
 
-        $newController = $controllersDirectory->createResource('controllerFile', array('controllerName' => $controllerName));
+        $newController = $controllersDirectory->createResource('controllerFile', array(
+									'controllerName' => $controllerName,
+									'moduleName' => $moduleName));
 
         return $newController;
     }
Only in Tool/Project/Provider: DbTable.php
Only in Tool/Project: Structure
