Index: incubator/tools/http_server/src/Request.php
===================================================================
--- incubator/tools/http_server/src/Request.php	(revision 1546)
+++ incubator/tools/http_server/src/Request.php	(working copy)
@@ -19,6 +19,7 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 
+require_once 'Zend/Uri/Http.php';
 
 /**
  * @category   Zend
@@ -183,28 +184,21 @@
 	 */
 	protected function setURI( $uri )
 	{
-		preg_match( "!^(.*)/([^/?]*)(\?.*$|$)!", $uri, $matches );
-		list( $this->uri, $this->path, $this->file, $this->query_string ) = $matches;
-
-		$query_string = preg_replace( "|^\?|", "", $this->query_string );
-
-		if( trim( $query_string ) == "" )
-			return;
-		
-		$params = explode( "&", trim( $query_string ) );
-		foreach( $params as $param )
-		{
-			$parts = explode( "=", $param );
-
-			if( count( $parts ) > 1 )
-			{
-				$this->get[ $parts[ 0 ] ] = urldecode( $parts[ 1 ] );
-			}
-			else
-			{
-				$this->get[ $parts[ 0 ] ] = 1;
-			}
+		$this->uri = $uri;
+		if (Zend_Uri_Http::check($uri)) {
+			$uri = Zend_Uri::factory($uri);
+			$this->query_string = $uri->getQuery();
+			$this->path = dirname($uri->getPath());
+			$this->file = basename($uri->getPath());
+		} else {
+			list($this->path, $this->query_string) = explode('?', $uri, 2);
+			$this->file = basename($this->path);
+			$this->path = dirname($this->path);
 		}
+		
+		if (! $this->path) $this->path = '/';
+		
+		parse_str($this->query_string, $this->get);
 	}
 
 	protected function setData( $data )
