Index: library/Zend/Validate/File/MimeType.php
===================================================================
--- library/Zend/Validate/File/MimeType.php	(revision 21852)
+++ library/Zend/Validate/File/MimeType.php	(working copy)
@@ -343,8 +343,6 @@
             if ($this->_finfo !== false) {
                 $this->_type = finfo_file($this->_finfo, $value);
             }
-
-            unset($this->_finfo);
         }
 
         if (empty($this->_type) &&
