Index: PluginLoader.php
===================================================================
--- PluginLoader.php	(Revision 20695)
+++ PluginLoader.php	(Arbeitskopie)
@@ -433,15 +433,19 @@
             return;
         }
 
-        if (!file_exists($file) && !file_exists(dirname($file))) {
+        $directory         = dirname($file);
+        $fileExists        = file_exists($file);
+        $directoryExists   = file_exists($directory);
+
+        if (!$fileExists && !$directoryExists) {
             require_once 'Zend/Loader/PluginLoader/Exception.php';
             throw new Zend_Loader_PluginLoader_Exception('Specified file does not exist and/or directory does not exist (' . $file . ')');
         }
-        if (file_exists($file) && !is_writable($file)) {
+        if ($fileExists && !is_writable($file)) {
             require_once 'Zend/Loader/PluginLoader/Exception.php';
             throw new Zend_Loader_PluginLoader_Exception('Specified file is not writeable (' . $file . ')');
         }
-        if (!file_exists($file) && file_exists(dirname($file)) && !is_writable(dirname($file))) {
+        if (!$fileExists && $directoryExists && !is_writable($directory)) {
             require_once 'Zend/Loader/PluginLoader/Exception.php';
             throw new Zend_Loader_PluginLoader_Exception('Specified file is not writeable (' . $file . ')');
         }
