Index: View.php
===================================================================
--- View.php	(revision 21170)
+++ View.php	(working copy)
@@ -61,14 +61,19 @@
     /**
      * Retrieve view object
      *
-     * @return Zend_View
+     * @return Zend_View_Abstract
      */
     public function getView()
     {
         if (null === $this->_view) {
             $options = $this->getOptions();
-            $this->_view = new Zend_View($options);
 
+            $className = 'Zend_View';
+            if (isset($options['className'])) {
+                $className = $options['className'];
+            }
+            $this->_view = new $className($options);
+
             if(isset($options['doctype'])) {
                 $this->_view->doctype()->setDoctype(strtoupper($options['doctype']));
             }
