Index: library/Zend/View/Helper/Navigation/Menu.php
===================================================================
--- library/Zend/View/Helper/Navigation/Menu.php	(revision 24182)
+++ library/Zend/View/Helper/Navigation/Menu.php	(working copy)
@@ -231,9 +231,10 @@
 
         // does page have a href?
         if ($href = $page->getHref()) {
-            $element = 'a';
-            $attribs['href'] = $href;
-            $attribs['target'] = $page->getTarget();
+            $element              = 'a';
+            $attribs['href']      = $href;
+            $attribs['target']    = $page->getTarget();
+            $attribs['accesskey'] = $page->getAccesskey();
         } else {
             $element = 'span';
         }
