Index: Zend/Http/Response.php
===================================================================
--- Zend/Http/Response.php	(.../branches/zend-framework/library/incubator)	(revision 791)
+++ Zend/Http/Response.php	(.../trunk/library/incubator)	(revision 791)
@@ -485,7 +485,7 @@
                     }
 
                     $headers[$h_name][] = $h_value;
-                    $last_header = &end($headers[$h_name]);
+                    $last_header = end($headers[$h_name]);
                 } else {
                     $headers[$h_name] = $h_value;
                     $last_header = &$headers[$h_name];
