--- library/Zend/View/Helper/FormLabel.php	(revision 8757)
+++ library/Zend/View/Helper/FormLabel.php	(working copy)
@@ -43,6 +43,11 @@
      */
     public function formLabel($name, $value = null, array $attribs = array())
     {
+        if ($name instanceof Zend_Form_Element) {
+            $value = $name->getLabel();
+            $name  = $name->getName();
+        }
+        
         $info = $this->_getInfo($name, $value, $attribs);
         extract($info); // name, value, attribs, options, listsep, disable, escape
 
