Index: library/Zend/Controller/Plugin/Broker.php
===================================================================
--- library/Zend/Controller/Plugin/Broker.php	(revision 19920)
+++ library/Zend/Controller/Plugin/Broker.php	(working copy)
@@ -336,9 +336,10 @@
             } catch (Exception $e) {
                 if (Zend_Controller_Front::getInstance()->throwExceptions()) {
                     require_once 'Zend/Controller/Exception.php';
-                    throw new Zend_Controller_Exception($e->getMessage(), $e->getCode(), $e);
+                    throw new Zend_Controller_Exception($e->getMessage() . $e->getTraceAsString(), $e->getCode(), $e);
                 } else {
                     $this->getResponse()->setException($e);
+                    $this->getRequest()->setDispatched(false);
                 }
             }
         }
