Index: Abstract.php
===================================================================
--- Abstract.php	(revision 3815)
+++ Abstract.php	(working copy)
@@ -207,7 +232,7 @@
      * @param array $bind Column-value pairs.
      * @return int The number of affected rows.
      */
-    public function insert($table, $bind)
+    public function insert($table, array $bind)
     {
         // col names come from the array keys
         $cols = array_keys($bind);
@@ -229,10 +254,10 @@
      *
      * @param string $table The table to udpate.
      * @param array $bind Column-value pairs.
-     * @param string $where UPDATE WHERE clause.
+     * @param string $where Optional UPDATE WHERE clause.
      * @return int The number of affected rows.
      */
-    public function update($table, $bind, $where)
+    public function update($table, array $bind, $where = '')
     {
         // build "col = ?" pairs for the statement
         $set = array();
@@ -255,10 +280,10 @@
      * Deletes table rows based on a WHERE clause.
      *
      * @param string $table The table to udpate.
-     * @param string $where DELETE WHERE clause.
+     * @param string $where Optional DELETE WHERE clause.
      * @return int The number of affected rows.
      */
-    public function delete($table, $where)
+    public function delete($table, $where = '')
     {
         // build the statement
         $sql = "DELETE FROM $table"
