Index: SessionTest.php
===================================================================
--- SessionTest.php	(révision 23688)
+++ SessionTest.php	(copie de travail)
@@ -1047,5 +1047,18 @@
             $this->assertEquals($v, $s->getNamespace());
         }
     }
+    
+    public function testArrayAccessInterface()
+    {
+        $ns = new Zend_Session_Namespace('foo');
+        $ns['bar'] = 'baz';
+        $this->assertEquals('baz', $ns['bar']);
+        $this->assertEquals('baz', $ns->bar);
+        $this->assertTrue(isset($ns['bar']));
 
+        $value = &$ns['bar'];
+        $value = 'reference';
+        
+        $this->assertEquals('reference', $ns['bar']);        
+    }
 }
Index: Namespace.php
===================================================================
--- Namespace.php	(révision 23688)
+++ Namespace.php	(copie de travail)
@@ -41,7 +41,7 @@
  * @copyright  Copyright (c) 2005-2010 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Session_Namespace extends Zend_Session_Abstract implements IteratorAggregate
+class Zend_Session_Namespace extends Zend_Session_Abstract implements IteratorAggregate, ArrayAccess
 {
 
     /**
@@ -526,4 +526,54 @@
     {
         return $this->_namespace;
     }
+
+    /** offsetExists - Proxies to __isset()
+     * ArrayAccess interface
+     * 
+     * @param string $offset
+     * @throws Zend_Session_Exception
+     * @return true
+     */
+    public function offsetExists($offset)
+    {
+        return $this->__isset($offset);        
+    }
+
+    /** offsetGet - Proxies to __get()
+     * ArrayAccess interface
+     * 
+     * @param string $offset
+     * @param mixed $value
+     * @throws Zend_Session_Exception
+     * @return true
+     */
+    public function &offsetGet($offset)
+    {
+        return $this->__get($offset);
+    }
+
+    /** offsetSet - Proxies to __set()
+     * ArrayAccess interface
+     * 
+     * @param string $offset
+     * @param mixed $value
+     * @throws Zend_Session_Exception
+     * @return true
+     */
+    public function offsetSet($offset, $value)
+    {
+        return $this->__set($offset, $value);
+        
+    }
+
+    /** offsetUnset - Proxies to __unset()
+     * ArrayAccess interface
+     * 
+     * @param string $offset
+     * @return true
+     */
+    public function offsetUnset($offset)
+    {
+        return $this->__unset($offset);
+    }
 }
