Index: tests/Zend/Loader/AutoloaderTest.php
===================================================================
--- tests/Zend/Loader/AutoloaderTest.php	(revision 19083)
+++ tests/Zend/Loader/AutoloaderTest.php	(working copy)
@@ -382,6 +382,19 @@
         $this->assertFalse(class_exists($class, false));
     }
 
+    /**
+     * @group ZF-7191
+     */
+    public function testShouldBeAbleToUseAnonymousCallbacksAsAutoloader()
+    {
+        $push = function ($class) {return 'pushed ' . $class;};
+        $unshift = function ($class) {return 'unshifted ' . $class;};
+        $this->autoloader->pushAutoloader($push, 'foo_');
+        $this->autoloader->unshiftAutoloader($unshift, 'bar_');
+        $this->assertEquals('pushed foo_wat', $this->autoloader->autoload('foo_wat'));
+        $this->assertEquals('unshifted bar_wat', $this->autoloader->autoload('bar_wat'));
+    }
+
     public function addTestIncludePath()
     {
         set_include_path(dirname(__FILE__) . '/_files/' . PATH_SEPARATOR . $this->includePath);
