diff -Naur ZendFramework-1.9.2/library/Zend/Db.php ZendFramework-1.9.2_/library/Zend/Db.php
--- ZendFramework-1.9.2/library/Zend/Db.php	2009-06-21 20:56:17.000000000 +0200
+++ ZendFramework-1.9.2_/library/Zend/Db.php	2009-09-09 22:20:28.000000000 +0200
@@ -240,9 +240,13 @@
                 $adapterNamespace = $config['adapterNamespace'];
             }
             unset($config['adapterNamespace']);
+            // if adapterNamespace was set in config, I do not change them
+            // beacuse under Linux systems this change for ZendX give exception
+            $adapterName = $adapterNamespace . '_' . ucwords(strtolower($adapter));
+        } else {
+            $adapterName = strtolower($adapterNamespace . '_' . $adapter);
+            $adapterName = str_replace(' ', '_', ucwords(str_replace('_', ' ', $adapterName)));
         }
-        $adapterName = strtolower($adapterNamespace . '_' . $adapter);
-        $adapterName = str_replace(' ', '_', ucwords(str_replace('_', ' ', $adapterName)));
 
         /*
          * Load the adapter class.  This throws an exception
