--- library/Zend/Dojo/Form/Element/ValidationTextBox.php	(revision 23872)
+++ library/Zend/Dojo/Form/Element/ValidationTextBox.php	(working copy)
@@ -132,6 +132,8 @@
      */
     public function setConstraints(array $constraints)
     {
+        $tmp = $this->getConstraints();
+        $constraints = array_merge($tmp, $constraints);
         array_walk_recursive($constraints, array($this, '_castBoolToString'));
         $this->setDijitParam('constraints', $constraints);
         return $this;
