Index: Zend/Session/SessionTest.php
===================================================================
--- Zend/Session/SessionTest.php	(revision 23334)
+++ Zend/Session/SessionTest.php	(working copy)
@@ -783,6 +783,7 @@
     /**
      * test expiration of namespaces and namespace variables by seconds; expect expiration of specified keys/namespace
      *
+     * @runInSeparateProcess
      * @return void
      */
     public function testSetExpirationSeconds()
@@ -842,6 +843,7 @@
     /**
      * test expiration of namespaces by hops; expect expiration of specified namespace in the proper number of hops
      *
+     * @runInSeparateProcess
      * @return void
      */
     public function testSetExpireSessionHops()
@@ -874,6 +876,7 @@
     /**
      * test expiration of namespace variables by hops; expect expiration of specified keys in the proper number of hops
      *
+     * @runInSeparateProcess
      * @return void
      */
     public function testSetExpireSessionVarsByHops1()
@@ -884,6 +887,7 @@
     /**
      * sanity check .. we should be able to repeat this test without problems
      *
+     * @runInSeparateProcess
      * @return void
      */
     public function testSetExpireSessionVarsByHops2()
@@ -949,6 +953,7 @@
     /**
      * test expiration of namespace variables by hops; expect expiration of specified keys in the proper number of hops
      *
+     * @runInSeparateProcess
      * @return void
      */
     public function testSetExpireSessionVarsByHopsOnUse()
