diff --git a/tests/Zend/Validate/DateTest.php b/tests/Zend/Validate/DateTest.php
index ecc395b..00db7c5 100644
--- a/tests/Zend/Validate/DateTest.php
+++ b/tests/Zend/Validate/DateTest.php
@@ -204,7 +204,7 @@ class Zend_Validate_DateTest extends PHPUnit_Framework_TestCase
     }
 
     /**
-     * @ZF-4352
+     * @group ZF-4352
      */
     public function testNonStringValidation()
     {
@@ -212,7 +212,7 @@ class Zend_Validate_DateTest extends PHPUnit_Framework_TestCase
     }
 
     /**
-     * @ZF-6374
+     * @group ZF-6374
      */
     public function testUsingApplicationLocale()
     {
@@ -222,7 +222,7 @@ class Zend_Validate_DateTest extends PHPUnit_Framework_TestCase
     }
 
     /**
-     * ZF-7630
+     * @group ZF-7630
      */
     public function testDateObjectVerification()
     {
@@ -231,7 +231,7 @@ class Zend_Validate_DateTest extends PHPUnit_Framework_TestCase
     }
 
     /**
-     * ZF-6457
+     * @group ZF-6457
      */
     public function testArrayVerification()
     {
