Index: library/Zend/Translate/Adapter/Tbx.php
===================================================================
--- library/Zend/Translate/Adapter/Tbx.php	(Revision 7028)
+++ library/Zend/Translate/Adapter/Tbx.php	(Arbeitskopie)
@@ -93,10 +93,11 @@
         xml_set_character_data_handler($this->_file, "_contentElement");
 
         if (!xml_parse($this->_file, file_get_contents($filename))) {
-            throw new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
+            $exception = new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
                       xml_error_string(xml_get_error_code($this->_file)),
                       xml_get_current_line_number($this->_file)));
             xml_parser_free($this->_file);
+            throw $exception;
         }
     }
 
Index: library/Zend/Translate/Adapter/XmlTm.php
===================================================================
--- library/Zend/Translate/Adapter/XmlTm.php	(Revision 7028)
+++ library/Zend/Translate/Adapter/XmlTm.php	(Arbeitskopie)
@@ -88,10 +88,11 @@
         xml_set_character_data_handler($this->_file, "_contentElement");
 
         if (!xml_parse($this->_file, file_get_contents($filename))) {
-            throw new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
+            $exception = new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
                       xml_error_string(xml_get_error_code($this->_file)),
                       xml_get_current_line_number($this->_file)));
             xml_parser_free($this->_file);
+            throw $exception;
         }
     }
 
Index: library/Zend/Translate/Adapter/Qt.php
===================================================================
--- library/Zend/Translate/Adapter/Qt.php	(Revision 7028)
+++ library/Zend/Translate/Adapter/Qt.php	(Arbeitskopie)
@@ -93,10 +93,11 @@
         xml_set_character_data_handler($this->_file, "_contentElement");
 
         if (!xml_parse($this->_file, file_get_contents($filename))) {
-            throw new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
+            $exception = new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
                       xml_error_string(xml_get_error_code($this->_file)),
                       xml_get_current_line_number($this->_file)));
             xml_parser_free($this->_file);
+            throw $exception;
         }
     }
 
Index: library/Zend/Translate/Adapter/Tmx.php
===================================================================
--- library/Zend/Translate/Adapter/Tmx.php	(Revision 7028)
+++ library/Zend/Translate/Adapter/Tmx.php	(Arbeitskopie)
@@ -93,10 +93,11 @@
         xml_set_character_data_handler($this->_file, "_contentElement");
 
         if (!xml_parse($this->_file, file_get_contents($filename))) {
-            throw new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
+            $exception = new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
                       xml_error_string(xml_get_error_code($this->_file)),
                       xml_get_current_line_number($this->_file)));
             xml_parser_free($this->_file);
+            throw $exception;
         }
     }
 
Index: library/Zend/Translate/Adapter/Xliff.php
===================================================================
--- library/Zend/Translate/Adapter/Xliff.php	(Revision 7028)
+++ library/Zend/Translate/Adapter/Xliff.php	(Arbeitskopie)
@@ -91,10 +91,11 @@
         xml_set_character_data_handler($this->_file, "_contentElement");
 
         if (!xml_parse($this->_file, file_get_contents($filename))) {
-            throw new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
+            $exception = new Zend_Translate_Exception(sprintf('XML error: %s at line %d',
                       xml_error_string(xml_get_error_code($this->_file)),
                       xml_get_current_line_number($this->_file)));
             xml_parser_free($this->_file);
+            throw $exception;
         }
     }
 
