--- Gdata/Calendar.php	(revision 23623)
+++ Gdata/Calendar.php	(working copy)
@@ -110,7 +110,8 @@
   } else {
     $protocol = 'http://';
   }
-  $host = $_SERVER['HTTP_HOST'];
+  $tmp = explode(':', $_SERVER['HTTP_HOST']);
+  $host = $tmp[0];
   if ($_SERVER['SERVER_PORT'] != '' &&
      (($protocol == 'http://' && $_SERVER['SERVER_PORT'] != '80') ||
      ($protocol == 'https://' && $_SERVER['SERVER_PORT'] != '443'))) {
Index: Gdata/Photos.php
===================================================================
--- Gdata/Photos.php	(revision 23623)
+++ Gdata/Photos.php	(working copy)
@@ -356,7 +356,8 @@
     } else {
         $protocol = 'http://';
     }
-    $host = $_SERVER['HTTP_HOST'];
+    $tmp = explode(':', $_SERVER['HTTP_HOST']);
+    $host = $tmp[0];
     if ($_SERVER['SERVER_PORT'] != '' &&
         (($protocol == 'http://' && $_SERVER['SERVER_PORT'] != '80') ||
         ($protocol == 'https://' && $_SERVER['SERVER_PORT'] != '443'))) {
Index: Gdata/Health.php
===================================================================
--- Gdata/Health.php	(revision 23623)
+++ Gdata/Health.php	(working copy)
@@ -294,7 +294,8 @@
   } else {
     $protocol = 'http://';
   }
-  $host = $_SERVER['HTTP_HOST'];
+  $tmp = explode(':', $_SERVER['HTTP_HOST']);
+  $host = $tmp[0];
   if ($_SERVER['SERVER_PORT'] != '' &&
      (($protocol == 'http://' && $_SERVER['SERVER_PORT'] != '80') ||
      ($protocol == 'https://' && $_SERVER['SERVER_PORT'] != '443'))) {
