--- Transfer.old	2009-12-30 08:30:07.000000000 -0500
+++ Transfer.php	2009-12-30 08:30:37.000000000 -0500
@@ -51,9 +51,9 @@
         }
 
         $this->_adapter = new $adapter();
-        if (!$this->_adapter instanceof Zend_File_Transfer_Adapter) {
+        if (!$this->_adapter instanceof Zend_File_Transfer_Adapter_Abstract) {
             require_once 'Zend/File/Transfer/Exception.php';
-            throw new Zend_File_Transfer_Exception("Adapter " . $adapter . " does not extend Zend_File_Transfer_Adapter'");
+            throw new Zend_File_Transfer_Exception("Adapter " . $adapter . " does not extend Zend_File_Transfer_Adapter_Abstract'");
         }
 
         return $this->_adapter;
