Index: tests/Zend/Service/Audioscrobbler/ArtistTest.php
===================================================================
--- tests/Zend/Service/Audioscrobbler/ArtistTest.php	(revision 12418)
+++ tests/Zend/Service/Audioscrobbler/ArtistTest.php	(working copy)
@@ -224,4 +224,13 @@
             $this->fail("Exception: [" . $e->getMessage() . "] thrown by test");
         }
     }
+
+    public function testSetArtist() 
+    { 
+          $as = new Zend_Service_Audioscrobbler();
+	  $as->set('artist', 'Kooks');
+	  $this->assertEquals('Kooks', $as->get('artist'));
+	  $as->setArtist('SystemofaDown');
+	  $this->assertEquals('SystemofaDown', $as->get('artist'));	
+    }
 }
Index: library/Zend/Service/Audioscrobbler.php
===================================================================
--- library/Zend/Service/Audioscrobbler.php	(revision 12418)
+++ library/Zend/Service/Audioscrobbler.php	(working copy)
@@ -436,11 +436,14 @@
     //////////////////////////////////////////////////////////
 
     /**
-     * Public functions for retrieveing artist-specific information
+     * Public functions for retrieving artist-specific information
      *
      */
+    
 
-
+     public function setArtist($artistsName) {
+        $this->set('artist', $artistsName);
+     }
     /**
      * Utility function that returns a list of artists similiar to this artist
      * @return SimpleXML object containing result set
