--- library/Zend/Db/Table/Row/Abstract.php	(revision 5451)
+++ library/Zend/Db/Table/Row/Abstract.php	(working copy)
@@ -864,7 +864,7 @@
          * findParent<Class>()
          * findParent<Class>By<Rule>()
          */
-        if (preg_match('/^findParent(\w+)(?:By(\w+))?$/', $method, $matches)) {
+        if (preg_match('/^findParent(\w+?)(?:By(\w+))?$/', $method, $matches)) {
             $class    = $matches[1];
             $ruleKey1 = isset($matches[2]) ? $matches[2] : null;
             return $this->findParentRow($class, $ruleKey1);
