Index: library/Zend/Form.php
===================================================================
--- library/Zend/Form.php	(revision 19781)
+++ library/Zend/Form.php	(working copy)
@@ -846,8 +846,14 @@
             require_once 'Zend/Form/Exception.php';
             throw new Zend_Form_Exception('Invalid name provided; must contain only valid variable characters and be non-empty');
         }
-
-        return $this->setAttrib('name', $name);
+        
+        $this->setAttrib('name', $name);
+        
+        if($this->isArray()) {
+            $this->_setElementsBelongTo();
+        }
+        
+        return $this;
     }
 
     /**
Index: tests/Zend/Form/FormTest.php
===================================================================
--- tests/Zend/Form/FormTest.php	(revision 19782)
+++ tests/Zend/Form/FormTest.php	(working copy)
@@ -1121,6 +1121,42 @@
             $this->assertEquals('foo', $element->getBelongsTo(), $message);
         }
     }
+    
+    public function testAddSubFormSetsBelongsToWhenSubFormHasNone()
+    {
+        $subform = new Zend_Form_SubForm();
+        $subform->addElement(
+            $subformElement = new Zend_Form_Element_Text('foo')
+        );
+        $this->form->addSubForm($subform, "FoobarSubForm");
+
+        $this->assertEquals('FoobarSubForm', $subform->getElementsBelongTo());
+        $this->assertEquals('FoobarSubForm', $subformElement->getBelongsTo());
+    }
+    
+    public function testElementsBelongToBeforeChangingName()
+    {
+        $this->form->setIsArray(true);
+        $this->form->addElement(
+            new Zend_Form_Element_Text('bar')
+        );
+        $this->form->setName('baz');
+        $this->assertEquals('baz', $this->form->getElementsBelongTo());
+        $this->assertEquals('baz',$this->form->getElement('bar')->getBelongsTo());
+    }
+    
+    public function testElementsBelongToAfterChangingName()
+    {
+        $this->form->setIsArray(true);
+        $this->form->setName('baz');
+        $this->assertEquals('baz', $this->form->getElementsBelongTo());
+        
+        $this->form->addElement(
+            $barElement = new Zend_Form_Element_Text('bar')
+        );
+        
+        $this->assertEquals('baz',$barElement->getBelongsTo());
+    }
 
     /**
      * @group ZF-3742

