--- Zend/Loader/Autoloader/Resource.php	(revision 19152)
+++ Zend/Loader/Autoloader/Resource.php	(working copy)
@@ -173,7 +173,14 @@
 
         $final = substr($class, strlen($lastMatch) + 1);
         $path = $this->_components[$lastMatch];
-        return $path . '/' . str_replace('_', '/', $final) . '.php';
+
+        $classPath = $path . '/' . str_replace('_', '/', $final) . '.php';
+
+        if (!Zend_Loader::isReadable($classPath)) {
+            return false;
+        }
+
+        return $classPath;
     }
 
     /**
@@ -184,7 +191,11 @@
      */
     public function autoload($class)
     {
-        return include $this->getClassPath($class);
+        $classPath = $this->getClassPath($class);
+        if (false === $classPath) {
+            return false;
+        }
+        return include $classPath;
     }
 
     /**
