Index: Zend/Form/Element.php
===================================================================
--- Zend/Form/Element.php	(revision 17325)
+++ Zend/Form/Element.php	(working copy)
@@ -2097,7 +2097,7 @@
             if (null !== $translator) {
                 $message = $translator->translate($message);
             }
-            if ($this->isArray() || is_array($value)) {
+            if (($this->isArray() || is_array($value)) && !empty($value)) {
                 $aggregateMessages = array();
                 foreach ($value as $val) {
                     $aggregateMessages[] = str_replace('%value%', $val, $message);
