Index: library/Zend/Currency.php
===================================================================
--- library/Zend/Currency.php	(revision 21588)
+++ library/Zend/Currency.php	(working copy)
@@ -823,7 +823,7 @@
 
         foreach ($options as $name => $value) {
             $name = strtolower($name);
-            if ($name !== 'format') {
+            if ($name !== 'locale') {
                 if (gettype($value) === 'string') {
                     $value = strtolower($value);
                 }
@@ -838,7 +838,7 @@
 
                     break;
 
-                case 'format':
+                case 'locale':
                     if ((empty($value) === false) and (Zend_Locale::isLocale($value, null, false) === false)) {
                         require_once 'Zend/Currency/Exception.php';
                         throw new Zend_Currency_Exception("'" .
