Index: Loader.php
===================================================================
--- Loader.php	(revision 5577)
+++ Loader.php	(working copy)
@@ -97,6 +97,44 @@
     }
 
     /**
+     * Operates a lot like Zend_Loader::loadClass(), but instead of taking a 
+     * full class name and a list of directories, it takes a class base name and 
+     * a list of namespaces. It includes the first matching class, and returns 
+     * the full name of the found class. If no such class is found, an exception 
+     * is thrown.
+     *
+     * @param string $basename   Class base name
+     * @param array $namespaces  Optional list of namespace class prefixes
+     * @param string|array $dirs Optional list of directories to search in
+     * @return string            Full name of the first matching class
+     * @throws Zend_Exception
+     */
+    public function namespaceLoad($basename, array $namespaces = null, $dirs = null)
+    {
+        $basename = ucfirst($basename);
+
+        if (null == $namespaces) {
+           self::loadClass($basename, $dirs); 
+           return $basename;
+        }
+
+        foreach ($namespaces as $ns) {
+            try {
+                $fullClass = rtrim($ns, '_').'_'.$basename;
+                self::loadClass($fullClass, $dirs);
+
+                // If the class was successfully loaded, return the name
+                return $fullClass;
+
+            } catch (Zend_Exception $e) {
+                continue;
+            }
+        }
+
+        throw new Zend_Exception("No class with base name '$basename' was found within any of the given namespaces, within the include path. Namespaces searched: ".implode(', ', (array)$namespaces));
+    }
+
+    /**
      * Loads a PHP file.  This is a wrapper for PHP's include() function.
      *
      * $filename must be the complete filename, including any
