Index: ../trunk/library/Zend/Form/Element/Captcha.php
===================================================================
--- ../trunk/library/Zend/Form/Element/Captcha.php	(revision 19479)
+++ ../trunk/library/Zend/Form/Element/Captcha.php	(working copy)
@@ -168,12 +168,11 @@
     }
 
     /**
-     * Render form element
+     * Prepend additional decorators and generate value for the captcha key
      *
-     * @param  Zend_View_Interface $view
-     * @return string
+     * @return void
      */
-    public function render(Zend_View_Interface $view = null)
+    protected function _preRender()
     {
         $captcha    = $this->getCaptcha();
         $captcha->setName($this->getFullyQualifiedName());
@@ -191,10 +190,9 @@
         $this->setDecorators($decorators);
 
         $this->setValue($this->getCaptcha()->generate());
-
-        return parent::render($view);
     }
 
+
     /**
      * Retrieve plugin loader for validator or filter chain
      *
