Index: Zend/Db/Select/StaticTest.php
===================================================================
--- Zend/Db/Select/StaticTest.php	(revision 17353)
+++ Zend/Db/Select/StaticTest.php	(working copy)
@@ -387,7 +387,7 @@
     {
         $select = $this->_selectWhereOr();
         $sql = preg_replace('/\\s+/', ' ', $select->__toString());
-        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE ("product_id" = 1) OR ("product_id" = 2)', $sql);
+        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE (("product_id" = 1) OR ("product_id" = 2))', $sql);
     }
 
     /**
@@ -399,7 +399,7 @@
     {
         $select = $this->_selectWhereOrWithParameter();
         $sql = preg_replace('/\\s+/', ' ', $select->__toString());
-        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE ("product_id" = 1) OR ("product_id" = 2)', $sql);
+        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE (("product_id" = 1) OR ("product_id" = 2))', $sql);
     }
 
     /**
@@ -653,6 +653,43 @@
         $this->assertEquals('SELECT "bug_id" AS "id", "bug_status" AS "name" FROM "zfbugs" UNION SELECT "product_id" AS "id", "product_name" AS "name" FROM "zfproducts" ORDER BY "id" ASC', $sql);
     }
 
+    public function testSelectTwoAndThenOr()
+    {
+        $select = $this->_db->select();
+        $select->from("zfproducts")
+               ->where("name = Foo")
+               ->where("name = Bar")
+               ->orWhere("name = Baz");
+        $sql = preg_replace('/\\s+/', ' ', $select->__toString());
+        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE (((name = Foo) AND (name = Bar)) OR (name = Baz))', $sql);
+    }
+
+    public function testSelectWhereTwoNamedCombineOr()
+    {
+        $select = $this->_db->select();
+        $select->from("zfproducts")
+               ->where("name = Foo", null, null, 'cond1')
+               ->where("name = Bar", null, null, 'cond2')
+               ->combine(array('cond1', 'cond2'), 'or');
+        $sql = preg_replace('/\\s+/', ' ', $select->__toString());
+        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE ((name = Foo) OR (name = Bar))', $sql);
+    }
+
+    public function testSelectWhereCombineNamedCombine()
+    {
+        $select = $this->_db->select();
+        $select->from("zfproducts")
+               ->where("name = Foo", null, null, 'cond1')
+               ->where("name = Bar", null, null, 'cond2')
+               ->combine(array('cond1', 'cond2'), 'and', 'cond3')
+               ->where("name = Baz", null, null, 'cond4')
+               ->where('name = Lala', null, null, 'cond5')
+               ->combine(array('cond4', 'cond5'), 'and', 'cond6')
+               ->combine(array('cond3', 'cond6'), 'or');
+        $sql = preg_replace('/\\s+/', ' ', $select->__toString());
+        $this->assertEquals('SELECT "zfproducts".* FROM "zfproducts" WHERE (((name = Foo) AND (name = Bar)) OR ((name = Baz) AND (name = Lala)))', $sql);
+    }
+
     public function getDriver()
     {
         return 'Static';
