Index: Zend/Cache/Backend.php
===================================================================
--- Zend/Cache/Backend.php	(revision 10737)
+++ Zend/Cache/Backend.php	(working copy)
@@ -193,8 +193,8 @@
         if (isset($this->_directives['logger']) && $this->_directives['logger'] instanceof Zend_Log) {
             return;
         }
-        // Create a default logger to the standard output stream
-        Zend_Loader::loadClass('Zend_Log_Writer_Stream');
+        // Create a default logger to the standard output stream
+        require_once 'Zend/Log/Writer/Stream.php';
         $logger = new Zend_Log(new Zend_Log_Writer_Stream('php://output'));
         $this->_directives['logger'] = $logger;
     }
Index: Zend/Cache/Core.php
===================================================================
--- Zend/Cache/Core.php	(revision 10737)
+++ Zend/Cache/Core.php	(working copy)
@@ -457,8 +457,8 @@
         if (isset($this->_options['logger']) && $this->_options['logger'] instanceof Zend_Log) {
             return;
         }
-        // Create a default logger to the standard output stream
-        Zend_Loader::loadClass('Zend_Log_Writer_Stream');
+        // Create a default logger to the standard output stream
+        require_once 'Zend/Log/Writer/Stream.php';
         $logger = new Zend_Log(new Zend_Log_Writer_Stream('php://output'));
         $this->_options['logger'] = $logger;
     }
