Index: Select.php
===================================================================
--- Select.php	(Revision 17003)
+++ Select.php	(working copy)
@@ -1085,7 +1085,7 @@
                 if ($target instanceof Zend_Db_Select) {
                     $target = $target->assemble();
                 }
-                $sql .= $target;
+                $sql .= ' ( ' . $target . ' ) ';
                 if ($cnt < $parts - 1) {
                     $sql .= ' ' . $type . ' ';
                 }
