Index: tests/Zend/Service/Rackspace/Files/OfflineTest.php
===================================================================
--- tests/Zend/Service/Rackspace/Files/OfflineTest.php	(revision 25298)
+++ tests/Zend/Service/Rackspace/Files/OfflineTest.php	(working copy)
@@ -212,6 +212,7 @@
             'zf-object-test' . '-copy'
         );
         $this->assertTrue($result);
+        $this->assertNotContains('application/x-www-form-urlencoded', $this->rackspace->getHttpClient()->getLastRequest());
     }
 
     public function testGetObjects()
