Index: ArgumentParser.php
===================================================================
--- ArgumentParser.php	(revision 24170)
+++ ArgumentParser.php	(working copy)
@@ -469,7 +469,7 @@
 
         // if non-option arguments exist, attempt to process them before processing options
         $wordStack = array();
-        while (($wordOnTop = array_shift($this->_argumentsWorking))) {
+        while (($wordOnTop = array_shift($this->_argumentsWorking)) !== false) {
             if (substr($wordOnTop, 0, 1) != '-') {
                 array_push($wordStack, $wordOnTop);
             } else {
