--- library/Zend/Controller/Router/Route/Chain.php	(revision 23064)
+++ library/Zend/Controller/Router/Route/Chain.php	(working copy)
@@ -77,7 +77,7 @@
         $values  = array();
 
         foreach ($this->_routes as $key => $route) {
-            if ($key > 0 && $matchedPath !== null) {
+            if ($key > 0 && $matchedPath !== null && $subPath !== '' && $subPath !== false) {
                 $separator = substr($subPath, 0, strlen($this->_separators[$key]));
 
                 if ($separator !== $this->_separators[$key]) {
