Index: library/Zend/Form/Element.php
===================================================================
--- library/Zend/Form/Element.php	(revision 18996)
+++ library/Zend/Form/Element.php	(working copy)
@@ -2050,7 +2050,7 @@
             $instance = new $name;
         } else {
             $messages = false;
-            if (isset($validator['options']['messages'])) {
+            if (is_array($validator['options']) && isset($validator['options']['messages'])) {
                 $messages = $validator['options']['messages'];
                 unset($validator['options']['messages']);
             }