Index: library/Zend/Loader/PluginLoader.php
===================================================================
--- library/Zend/Loader/PluginLoader.php	(revision 162)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -126,6 +126,12 @@
         if($prefix == "") {
             return $prefix;
         }
+
+        $last = strlen($prefix) - 1;
+        if ($prefix{$last} == '\\') {
+            return $prefix;
+        }
+
         return rtrim($prefix, '_') . '_';
     }
 
