Index: Zend/Wildfire/Plugin/FirePhp/TableMessage.php
===================================================================
--- Zend/Wildfire/Plugin/FirePhp/TableMessage.php	(revision 52)
+++ Zend/Wildfire/Plugin/FirePhp/TableMessage.php	(working copy)
@@ -94,5 +94,43 @@
         return $table;
     }
 
+    /**
+     * Returns the row at the given index
+     *
+     * @param integer $i
+     * @return array
+     */
+    public function getRow($i) {
+        return $this->_rows[$i];
+    }
+
+    /**
+     * Returns the last row of the table
+     *
+     * @return array
+     */
+    public function getLastRow() {
+        return $this->_rows[count($this->_rows)-1];
+    }
+
+    /**
+     * Sets the row on the given index to a new row
+     *
+     * @param integer $i
+     * @param array $row
+     */
+    public function setRow($i, $row) {
+        $this->_rows[$i] = $row;
+    }
+
+    /**
+     * Returns the row count
+     *
+     * @return integer
+     */
+    public function getRowCount() {
+        return count($this->_rows);
+    }
+    
 }
 
