Index: library/Zend/XmlRpc/Response/Http.php
===================================================================
--- library/Zend/XmlRpc/Response/Http.php	(revision 9106)
+++ library/Zend/XmlRpc/Response/Http.php	(working copy)
@@ -43,7 +43,7 @@
     public function __toString()
     {
         if (!headers_sent()) {
-            header('Content-Type: application/xml; charset=' . strtolower($this->getEncoding()));
+            header('Content-Type: text/xml; charset=' . strtolower($this->getEncoding()));
         }
 
         return parent::__toString();
