--- library/Zend/Form.php	(revision 24131)
+++ library/Zend/Form.php	(working copy)
@@ -2242,6 +2242,9 @@
         }
         $context = $data;
         foreach ($this->getElements() as $key => $element) {
+            if ($element->getIgnore()) {
+                continue;
+            }
             if (null !== $translator && $this->hasTranslator()
                     && !$element->hasTranslator()) {
                 $element->setTranslator($translator);
@@ -2300,6 +2303,9 @@
         $context    = $data;
 
         foreach ($this->getElements() as $key => $element) {
+            if ($element->getIgnore()) {
+                continue;
+            }
             $check = $data;
             if (($belongsTo = $element->getBelongsTo()) !== $eBelongTo) {
                 $check = $this->_dissolveArrayValue($data, $belongsTo);
Index: tests/Zend/Form/FormTest.php
===================================================================
--- tests/Zend/Form/FormTest.php	(revision 24131)
+++ tests/Zend/Form/FormTest.php	(working copy)
@@ -4461,6 +4461,27 @@
         $form = new Zend_Form();
         $form->addElement(NULL);
     }
+
+    /** @group ZF-10056 */
+    public function testIgnoredElementsShouldNotGenerateErrorMessages()
+    {
+        $element = new Zend_Form_Element_Text('foo', array(
+            'validators' => array('alpha')
+        ));
+        $element2 = new Zend_Form_Element_Text('bar', array(
+            'validators' => array('alpha'),
+            'ignore' => true
+        ));
+        $this->form->addElements(array($element, $element2));
+
+        $this->form->isValid(array('foo' => 1, 'bar' => 2));
+        $result = $this->form->getErrors();
+        $this->assertSame(array(), $result['bar']);
+
+        $this->form->isValidPartial(array('bar' => 2));
+        $result = $this->form->getErrors();
+        $this->assertSame(array(), $result['bar']);
+    }
 }
 
 class Zend_Form_FormTest_DisplayGroup extends Zend_Form_DisplayGroup
