Index: StripTags.php
===================================================================
--- StripTags.php	(revision 22536)
+++ StripTags.php	(working copy)
@@ -244,7 +244,7 @@
             $pos   = strrpos($value, '<!--');
             $start = substr($value, 0, $pos);
             $value = substr($value, $pos);
-            $value = preg_replace('/<(?:!(?:--[\s\S]*?--\s*)?(>))/us', '',  $value);
+            $value = preg_replace('/<(?:!(?:--[\s\S]*?--\s*)?(>))/s', '',  $value);
             $value = $start . $value;
         }
 
