Index: /usr/share/zend/Zend/View/Helper/FormSelect.php
===================================================================
--- /usr/share/zend/Zend/View/Helper/FormSelect.php	(revision 1616)
+++ /usr/share/zend/Zend/View/Helper/FormSelect.php	(working copy)
@@ -118,7 +118,7 @@
 
                 if (is_array($opt_label)) {
                     $list[] = '<optgroup '
-                            . 'name="' . htmlspecialchars($opt_value, ENT_COMPAT, 'UTF-8') .'">';
+                            . 'label="' . htmlspecialchars($opt_value, ENT_COMPAT, 'UTF-8') .'">';
                     foreach ($opt_label as $val => $lab) {
                         $list[] = $this->_build($val, $lab, $value);
                     }
