Index: Zend/View/Helper/_files/scripts/testBaseUrl.phtml
===================================================================
--- Zend/View/Helper/_files/scripts/testBaseUrl.phtml	(revision )
+++ Zend/View/Helper/_files/scripts/testBaseUrl.phtml	(revision )
@@ -0,0 +1,1 @@
+<?php echo $this->baseUrl(); ?>
\ No newline at end of file
Index: Zend/View/Helper/BaseUrlTest.php
===================================================================
--- Zend/View/Helper/BaseUrlTest.php	(revision 23772)
+++ Zend/View/Helper/BaseUrlTest.php	(revision )
@@ -36,6 +36,11 @@
 require_once 'Zend/Controller/Front.php';
 
 /**
+ * @see Zend_View
+ */
+require_once 'Zend/View.php';
+
+/**
  * @category   Zend
  * @package    Zend_View
  * @subpackage UnitTests
@@ -204,7 +209,17 @@
         $helper = new Zend_View_Helper_BaseUrl();
         $this->assertEquals('/mybar', $helper->getBaseUrl());
     }
+
+    public function testZF9162()
+    {
+        $baseUrl = 'http://foobar';
+        Zend_Controller_Front::getInstance()->setBaseUrl($baseUrl);
+
+        $view = new Zend_View();
+        $view->addBasePath(dirname(__FILE__) . '/_files');
+        $this->assertEquals($baseUrl, $view->render('testBaseUrl.phtml'));
-}
+    }
+}
 
 // Call Zend_View_Helper_BaseUrlTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == 'Zend_View_Helper_BaseUrlTest::main') {
