--- library/Zend/Tool/Project/Provider/Layout.php	(revision 24131)
+++ library/Zend/Tool/Project/Provider/Layout.php	(working copy)
@@ -70,7 +70,7 @@
 
         $zc = $applicationConfigResource->getAsZendConfig();
 
-        if (isset($zc->resources) && isset($zf->resources->layout)) {
+        if (isset($zc->resources) && isset($zc->resources->layout)) {
             $this->_registry->getResponse()->appendContent('A layout resource already exists in this project\'s application configuration file.');
             return;
         }
@@ -95,15 +95,11 @@
             $this->_registry->getResponse()->appendContent('A layout entry has been added to the application config file.');
         }
 
-
-
+        $this->_storeProfile();
     }
 
     public function disable()
     {
         // @todo
     }
-
-
-
 }
