Index: library/Zend/Service/Amazon/Ec2/Ebs.php
===================================================================
--- library/Zend/Service/Amazon/Ec2/Ebs.php	(revision 16131)
+++ library/Zend/Service/Amazon/Ec2/Ebs.php	(working copy)
@@ -212,9 +212,9 @@
         $params = array();
         $params['Action']       = 'DetachVolume';
         $params['VolumeId']     = $volumeId;
-        $params['InstanceId']   = $instanceId;
-        $params['Device']       = $device;
-        $params['Force']        = $force;
+        $params['InstanceId']   = strval($instanceId);
+        $params['Device']       = strval($device);
+        $params['Force']        = intval($force);
 
         $response = $this->sendRequest($params);
 
Index: library/Zend/Service/Amazon/Ec2/Abstract.php
===================================================================
--- library/Zend/Service/Amazon/Ec2/Abstract.php	(revision 16131)
+++ library/Zend/Service/Amazon/Ec2/Abstract.php	(working copy)
@@ -78,7 +78,7 @@
         try {
             /* @var $request Zend_Http_Client */
             $request = self::getHttpClient();
-			$request->resetParameters();
+            $request->resetParameters();
 
             $request->setConfig(array(
                 'timeout' => self::HTTP_TIMEOUT
@@ -125,7 +125,7 @@
     {
         $parameters['AWSAccessKeyId']   = $this->_getAccessKey();
         $parameters['SignatureVersion'] = self::EC2_SIGNATURE_VERSION;
-        $parameters['Expires']          = gmdate('c');
+        $parameters['Timestamp']        = gmdate('Y-m-d\TH:i:s\Z');
         $parameters['Version']          = self::EC2_API_VERSION;
         $parameters['SignatureMethod']  = self::EC2_SIGNATURE_METHOD;
         $parameters['Signature']        = $this->signParameters($parameters);
