Index: Pdo/Mssql.php
===================================================================
--- Pdo/Mssql.php	(revision 759)
+++ Pdo/Mssql.php	(working copy)
@@ -35,12 +35,11 @@
  */
 class Zend_Db_Adapter_Pdo_Mssql extends Zend_Db_Adapter_Pdo_Abstract
 {
-    /**
-     * PDO type.
-     *
-     * @var string
-     */
-    protected $_pdoType = 'mssql';
+    public function __construct($config)
+    {
+        $config['dsnprefix'] = 'mssql';
+        return parent::__construct($config);
+    }
 
 
     /**
@@ -157,6 +156,7 @@
         return $sql;
     }
 
+
     /**
      * Gets the last inserted ID.
      *
Index: Pdo/Sqlite.php
===================================================================
--- Pdo/Sqlite.php	(revision 759)
+++ Pdo/Sqlite.php	(working copy)
@@ -38,12 +38,35 @@
 class Zend_Db_Adapter_Pdo_Sqlite extends Zend_Db_Adapter_Pdo_Abstract
 {
     /**
+     * Constructor.
+     *
+     * $config is an array of key/value pairs containing configuration
+     * options.  These options are common to most adapters:
+     *
+     * dbname   => (string) The name of the database to user (required,
+     *                      use :memory: for memory-based database)
+     * username => (string) Connect to the database as this username (optional).
+     * password => (string) Password associated with the username (optional).
+     * host     => (string) What host to connect to (default 127.0.0.1).
+     * dsnprefix => (string) The PDO driver's DSN prefix (PDO only, optional).
+     *
+     * @param array $config An array of configuration keys.
+     */
+    public function __construct($config)
+    {
+        if (!isset($config['dsnprefix'])) // allows use of sqlite2 DSN prefix
+            $config['dsnprefix'] = 'sqlite';
+
+        return parent::__construct($config);
+    }
+
+
+    /**
      * DSN builder
      */
     protected function _dsn()
     {
-        $name = $this->_config['dbname'];
-        return "sqlite:$name";
+        return $this->_config['dsnprefix'].':'.$this->_config['dbname'];
     }
 
 
Index: Pdo/Abstract.php
===================================================================
--- Pdo/Abstract.php	(revision 759)
+++ Pdo/Abstract.php	(working copy)
@@ -46,6 +46,7 @@
         $dsn = $this->_config;
 
         // don't pass the username and password in the DSN
+        unset($dsn['dsnprefix']); // allows possibility of adapter servicing multiple drivers
         unset($dsn['username']);
         unset($dsn['password']);
 
@@ -54,7 +55,7 @@
             $dsn[$key] = "$key=$val";
         }
 
-        return $this->_pdoType . ':' . implode(';', $dsn);
+        return $this->_config['dsnprefix'] . ':' . implode(';', $dsn);
     }
 
 
@@ -62,6 +63,7 @@
      * Creates a PDO object and connects to the database.
      *
      * @return void
+     * @todo move PDO extension and driver checks out of this class (should use Zend_Environment).
      */
     protected function _connect()
     {
@@ -76,8 +78,8 @@
         }
 
         // check the PDO driver is available
-        if (!in_array($this->_pdoType, PDO::getAvailableDrivers())) {
-            throw new Zend_DB_Adapter_Exception('The ' . $this->_pdoType . ' driver is not currently installed');
+        if (!in_array($this->_config['dsnprefix'], PDO::getAvailableDrivers())) {
+            throw new Zend_DB_Adapter_Exception('The ' . $this->_config['dsnprefix'] . ' driver is not currently installed');
         }
 
         // create PDO connection
Index: Pdo/Mysql.php
===================================================================
--- Pdo/Mysql.php	(revision 759)
+++ Pdo/Mysql.php	(working copy)
@@ -37,15 +37,13 @@
  */
 class Zend_Db_Adapter_Pdo_Mysql extends Zend_Db_Adapter_Pdo_Abstract
 {
+    public function __construct($config)
+    {
+        $config['dsnprefix'] = 'mysql';
+        return parent::__construct($config);
+    }
 
-    /**
-     * PDO type.
-     *
-     * @var string
-     */
-    protected $_pdoType = 'mysql';
 
-
     /**
      * Quotes an identifier.
      *
Index: Pdo/Pgsql.php
===================================================================
--- Pdo/Pgsql.php	(revision 759)
+++ Pdo/Pgsql.php	(working copy)
@@ -37,14 +37,12 @@
  */
 class Zend_Db_Adapter_Pdo_Pgsql extends Zend_Db_Adapter_Pdo_Abstract
 {
-    /**
-     * PDO type.
-     *
-     * @var string
-     */
-    protected $_pdoType = 'pgsql';
+    public function __construct($config)
+    {
+        $config['dsnprefix'] = 'pgsql';
+        return parent::__construct($config);
+    }
 
-
     /**
      * Quotes an identifier.
      *
Index: Abstract.php
===================================================================
--- Abstract.php	(revision 759)
+++ Abstract.php	(working copy)
@@ -77,10 +77,11 @@
      * $config is an array of key/value pairs containing configuration
      * options.  These options are common to most adapters:
      *
-     * username => (string) Connect to the database as this username.
-     * password => (string) Password associated with the username.
-     * host     => (string) What host to connect to (default 127.0.0.1)
-     * dbname   => (string) The name of the database to user
+     * dbname   => (string) The name of the database to user (required)
+     * username => (string) Connect to the database as this username (optional).
+     * password => (string) Password associated with the username (optional).
+     * host     => (string) What host to connect to (default 127.0.0.1).
+     * dsnprefix => (string) The PDO driver's DSN prefix (PDO only, optional).
      *
      * @param array $config An array of configuration keys.
      */
