Index: Abstract.php
===================================================================
--- Abstract.php	(revision 9624)
+++ Abstract.php	(working copy)
@@ -1052,7 +1052,39 @@
         @Zend_Loader::loadClass($this->_rowsetClass);
         return new $this->_rowsetClass($data);
     }
+    
+    /**
+     * Fetches all result rows as a Rowset-wrapped array of key-value pairs.
+     *
+     * The first column of the select is the key, the second column is the
+     * value.
+     *
+     * Honors the Zend_Db_Adapter fetch mode.
+     *
+     * @param Zend_Db_Table_Select $where      Zend_Db_Table_Select object.
+     * @return Zend_Db_Table_Rowset_Abstract The row results per the Zend_Db_Adapter fetch mode.
+     */
+    public function fetchPairs(Zend_Db_Table_Select $select)
+    {
+        if (count($select->getPart(Zend_Db_Table_Select::COLUMNS)) != 2) {
+            require_once "Zend/Db/Table/Exception.php";
+            throw new Zend_Db_Table_Exception('Exactly 2 columns must be provided to build key => value pairs');
+        }
 
+        $rows = $this->getAdapter()->fetchPairs($select);
+
+        $data  = array(
+            'table'    => $this,
+            'data'     => $rows,
+            'readOnly' => $select->isReadOnly(),
+            'rowClass' => $this->_rowClass,
+            'stored'   => true
+        );
+
+        Zend_Loader::loadClass($this->_rowsetClass);
+        return new $this->_rowsetClass($data);
+    }
+
     /**
      * Fetches one row in an object of type Zend_Db_Table_Row_Abstract,
      * or returns Boolean false if no row matches the specified criteria.
