Index: tests/Zend/Feed/ImportTest.php
===================================================================
--- tests/Zend/Feed/ImportTest.php	(revision 7764)
+++ tests/Zend/Feed/ImportTest.php	(working copy)
@@ -407,6 +407,7 @@
 
         $feed = Zend_Feed::import('http://localhost');
         $this->assertTrue($feed instanceof Zend_Feed_Rss);
+        $this->assertTrue($feed->count() > 0);
     }
 
     /**
Index: library/Zend/Feed/Rss.php
===================================================================
--- library/Zend/Feed/Rss.php	(revision 7764)
+++ library/Zend/Feed/Rss.php	(working copy)
@@ -80,7 +80,11 @@
         parent::__wakeup();
 
         // Find the base channel element and create an alias to it.
-        $this->_element = $this->_element->getElementsByTagName('channel')->item(0);
+        if ($this->_element->firstChild->nodeName == 'rdf:RDF') {
+            $this->_element = $this->_element->firstChild; 
+        } else {
+            $this->_element = $this->_element->getElementsByTagName('channel')->item(0);
+        }
         if (!$this->_element) {
             /** 
              * @see Zend_Feed_Exception
