Index: library/Zend/View/Helper/FormRadio.php
===================================================================
--- library/Zend/View/Helper/FormRadio.php	(revision 24270)
+++ library/Zend/View/Helper/FormRadio.php	(working copy)
@@ -179,6 +179,11 @@
             // add to the array of radio buttons
             $list[] = $radio;
         }
+        
+        // XHTML or HTML for standard list separator?
+        if (!$this->_isXhtml() && false !== strpos($listsep, '<br />')) {
+            $listsep = str_replace('<br />', '<br>', $listsep);
+        }
 
         // done!
         $xhtml .= implode($listsep, $list);
