diff --git a/includes/Zend/Log/Writer/Syslog.php b/includes/Zend/Log/Writer/Syslog.php
index 3409ae5..67409f0 100644
--- a/includes/Zend/Log/Writer/Syslog.php
+++ b/includes/Zend/Log/Writer/Syslog.php
@@ -94,6 +94,8 @@ class Zend_Log_Writer_Syslog extends Zend_Log_Writer_Abstract
             $this->_facility = $params['facility'];
         }
         $this->_initializeSyslog();
+
+		$this->_formatter = new Zend_Log_Formatter_Simple('%message%');
     }
 
     /**
@@ -170,6 +172,8 @@ class Zend_Log_Writer_Syslog extends Zend_Log_Writer_Abstract
             $this->_initializeSyslog();
         }
 
-        syslog($priority, $event['message']);
+		$message = $this->_formatter->format($event);
+
+        syslog($priority, $message);
     }
 }
