Index: tests/Zend/Auth/Adapter/DbTable/BasicSqliteTest.php
===================================================================
--- tests/Zend/Auth/Adapter/DbTable/BasicSqliteTest.php	(revision 22206)
+++ tests/Zend/Auth/Adapter/DbTable/BasicSqliteTest.php	(working copy)
@@ -342,6 +342,27 @@
         $this->_adapter->authenticate();
     }
 
+    /**
+     * Ensures expected behavior for setting default adapter
+     *
+     * @return void
+     */
+    public function testAuthenticateWithDefaultAdapter()
+    {
+        Zend_Auth_Adapter_DbTable::setDefaultAdapter($this->_db);
+        $this->_adapter = new Zend_Auth_Adapter_DbTable();
+        $this->_adapter
+            ->setTableName('users')
+            ->setIdentityColumn('username')
+            ->setCredentialColumn('password')
+            ->setTableName('users')
+            ->setIdentity('my_username')
+            ->setCredential('my_password');
+        $result = $this->_adapter->authenticate();
+        $this->assertTrue($result->isValid());
+    }
+
+
     protected function _setupDbAdapter($optionalParams = array())
     {
         $params = array('dbname' => TESTS_ZEND_AUTH_ADAPTER_DBTABLE_PDO_SQLITE_DATABASE);
@@ -366,6 +387,7 @@
 
     protected function _setupAuthAdapter()
     {
+        Zend_Auth_Adapter_DbTable::setDefaultAdapter(null);
         $this->_adapter = new Zend_Auth_Adapter_DbTable($this->_db, 'users', 'username', 'password');
     }
 
Index: library/Zend/Auth/Adapter/DbTable.php
===================================================================
--- library/Zend/Auth/Adapter/DbTable.php	(revision 22206)
+++ library/Zend/Auth/Adapter/DbTable.php	(working copy)
@@ -48,6 +48,13 @@
 {
 
     /**
+     * Default Zend_Db_Adapter_Abstract object.
+     *
+     * @var Zend_Db_Adapter_Abstract
+     */
+    protected static $_defaultDb = null;
+
+    /**
      * Database Connection
      *
      * @var Zend_Db_Adapter_Abstract
@@ -125,10 +132,10 @@
      * @param  string                   $credentialTreatment
      * @return void
      */
-    public function __construct(Zend_Db_Adapter_Abstract $zendDb, $tableName = null, $identityColumn = null,
+    public function __construct(Zend_Db_Adapter_Abstract $zendDb = null, $tableName = null, $identityColumn = null,
                                 $credentialColumn = null, $credentialTreatment = null)
     {
-        $this->_zendDb = $zendDb;
+        $this->_zendDb = (null !== $zendDb) ? $zendDb : self::$_defaultDb;
 
         if (null !== $tableName) {
             $this->setTableName($tableName);
@@ -148,6 +155,18 @@
     }
 
     /**
+     * Sets the default Zend_Db_Adapter_Abstract.
+     *
+     * @param  Zend_Db_Adapter_Abstract $db
+     * @return void
+     */
+    public static function setDefaultAdapter(Zend_Db_Adapter_Abstract $db = null)
+    {
+        self::$_defaultDb = $db;
+    }
+
+
+    /**
      * setTableName() - set the table name to be used in the select query
      *
      * @param  string $tableName
