Index: Zend/Db/Adapter/Pdo/Abstract.php
===================================================================
--- Zend/Db/Adapter/Pdo/Abstract.php	(revision 1018)
+++ Zend/Db/Adapter/Pdo/Abstract.php	(working copy)
@@ -87,7 +87,8 @@
             $this->_connection = new PDO(
                 $this->_dsn(),
                 $this->_config['username'],
-                $this->_config['password']
+                $this->_config['password'],
+                $this->_config['driver_options']
             );
 
             $this->_profiler->queryEnd($q);
