Index: library/Zend/Application/Bootstrap/BootstrapAbstract.php
===================================================================
--- library/Zend/Application/Bootstrap/BootstrapAbstract.php	(revision 23324)
+++ library/Zend/Application/Bootstrap/BootstrapAbstract.php	(working copy)
@@ -414,7 +414,8 @@
     {
         if ($this->_pluginLoader === null) {
             $options = array(
-                'Zend_Application_Resource' => 'Zend/Application/Resource'
+                'Zend_Application_Resource'  => 'Zend/Application/Resource',
+                'ZendX_Application_Resource' => 'ZendX/Application/Resource'
             );
 
             $this->_pluginLoader = new Zend_Loader_PluginLoader($options);
Index: tests/Zend/Application/Bootstrap/BootstrapAbstractTest.php
===================================================================
--- tests/Zend/Application/Bootstrap/BootstrapAbstractTest.php	(revision 23324)
+++ tests/Zend/Application/Bootstrap/BootstrapAbstractTest.php	(working copy)
@@ -770,6 +770,18 @@
         ));
         $this->assertEquals('baz', $application->getBootstrap()->getOption('FooBar'));
     }
+
+    /**
+     * @group ZF-8751
+     */
+    public function testPathDefaultZendXToPluginsResources()
+    {
+        $application = $this->application
+                            ->getBootstrap()
+                            ->getPluginLoader();
+
+        $this->assertEquals(1, count($application->getPaths('ZendX_Application_Resource_')));
+    }
 }
 
 class Zend_Application_Bootstrap_BootstrapAbstractTest_View
