Index: tests/Zend/Service/Rackspace/Files/OfflineTest.php
===================================================================
--- tests/Zend/Service/Rackspace/Files/OfflineTest.php	(revision 25298)
+++ tests/Zend/Service/Rackspace/Files/OfflineTest.php	(working copy)
@@ -222,7 +222,16 @@
         $this->assertEquals($objects[0]->getName(), 'zf-object-test');
         $this->assertEquals($objects[1]->getName(), 'zf-object-test' . '-copy');
     }
+    
+    public function testGetObjectsPseudoDirs()
+    {
+        $objects = $this->rackspace->getObjects('zf-unit-test', array('delimiter' => '/', 'prefix' => 'dir/'));
+        $this->assertTrue($objects !== false);
 
+        $this->assertEquals($objects[0]->getName(), 'dir/subdir1/');
+        $this->assertEquals($objects[1]->getName(), 'dir/subdir2/');
+    }
+
     public function testGetSizeContainers()
     {
         $size = $this->rackspace->getSizeContainers();
Index: tests/Zend/Service/Rackspace/Files/_files/testGetObjectsPseudoDirs.response
===================================================================
--- tests/Zend/Service/Rackspace/Files/_files/testGetObjectsPseudoDirs.response	(revision 0)
+++ tests/Zend/Service/Rackspace/Files/_files/testGetObjectsPseudoDirs.response	(working copy)
@@ -0,0 +1,12 @@
+HTTP/1.1 200 OK
+X-Timestamp: 1340815553.90579
+Content-Type: application/json; charset=utf-8
+Date: Mon, 25 Mar 2013 16:00:03 GMT
+X-Container-Bytes-Used: 22
+X-Container-Object-Count: 2
+Accept-Ranges: bytes
+X-Trans-Id: tx1f63df08fcc646a5874388de8f197b72
+Connection: close
+Content-Length: 56
+
+[{"subdir": "dir/subdir1/"}, {"subdir": "dir/subdir2/"}]
\ No newline at end of file
