Index: Interface.php
===================================================================
--- Interface.php	(revision 10596)
+++ Interface.php	(working copy)
@@ -28,9 +28,32 @@
  * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-interface Zend_Controller_Router_Route_Interface {
+interface Zend_Controller_Router_Route_Interface
+{
+    /**
+     * Matches a user submitted path against the route. Assigns and
+     * returns an array of variables on a successful match. Returns false if
+     * the route does not match the path.
+     *
+     * @param string $path Path used to match against this routing map
+     * @return array|false An array of assigned values or a false on a mismatch
+     */
     public function match($path, $partial = null);
+
+    /**
+     * Assembles a URL path defined by this route
+     *
+     * @param  array $data An array of variable and value pairs used as parameters
+     * @return string Route path with user submitted parameters
+     */
     public function assemble($data = array(), $reset = false, $encode = false);
+
+    /**
+     * Instantiates route based on passed Zend_Config structure
+     *
+     * @param Zend_Config $config Configuration object
+     * @return Zend_Controller_Router_Route_Interface
+     */
     public static function getInstance(Zend_Config $config);
 }
 
