Index: Zend/View/Helper/Translate.php
===================================================================
--- Zend/View/Helper/Translate.php	(Revision 15593)
+++ Zend/View/Helper/Translate.php	(Arbeitskopie)
@@ -79,9 +79,8 @@
         $options = func_get_args();
         array_shift($options);
 
-        $count  = count($options);
         $locale = null;
-        if ($count > 0) {
+        if (count($options) > 0) {
             if (Zend_Locale::isLocale($options[($count - 1)], null, false) !== false) {
                 $locale = array_pop($options);
             }
@@ -92,7 +91,7 @@
         }
 
         $message = $translate->translate($messageid, $locale);
-        if ($count === 0) {
+        if (count($options) === 0) {
             return $message;
         }
 
