--- library/ZendX/Application/Resource/Jquery.php	(revision 23775)
+++ library/ZendX/Application/Resource/Jquery.php	(working copy)
@@ -105,8 +105,6 @@
      */
     protected function _parseOptions(array $options)
     {
-        $options = array_merge($options, array('cdn_ssl' => false));
-
         foreach ($options as $key => $value) {
             switch(strtolower($key)) {
                 case 'noconflictmode':
