Index: Multi.php
===================================================================
--- Multi.php	(revision 15065)
+++ Multi.php	(working copy)
@@ -278,7 +278,7 @@
     protected function _translateOption($option, $value)
     {
         if ($this->translatorIsDisabled()) {
-            return true;
+            return false;
         }
 
         if (!isset($this->_translated[$option]) && !empty($value)) {
