Index: .
===================================================================
--- .	(revision 11689)
+++ .	(working copy)
@@ -176,6 +176,22 @@
 
         unlink(dirname(__FILE__).'/_files/addfunction2.wsdl');
     }
+
+    public function testChangeWsdlUri()
+    {
+        $scriptUri = 'http://' . $_SERVER['HTTP_HOST'] . $_SERVER['SCRIPT_NAME'];
+
+        $server = new Zend_Soap_AutoDiscover(true, "http://example.com/service.php");
+        $server->addFunction('Zend_Soap_AutoDiscover_TestFunc');
+
+        ob_start();
+        $server->handle();
+        $wsdlOutput = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertNotContains($scriptUri, $wsdlOutput);
+        $this->assertContains("http://example.com/service.php", $wsdlOutput);
+    }
 }
 
 /* Test Functions */
