Index: tests/Zend/Cache/AllTests.php
===================================================================
--- tests/Zend/Cache/AllTests.php	(revision 21752)
+++ tests/Zend/Cache/AllTests.php	(working copy)
@@ -73,6 +73,7 @@
         $suite->addTestSuite('Zend_Cache_ClassFrontendTest');
         $suite->addTestSuite('Zend_Cache_FileFrontendTest');
         $suite->addTestSuite('Zend_Cache_PageFrontendTest');
+        $suite->addTestSuite('Zend_Cache_StaticBackendTest');
 
         /*
          * Check if SQLite tests are enabled, and if extension and driver are available.
