Index: Form.php
===================================================================
--- Form.php	(revision 20600)
+++ Form.php	(working copy)
@@ -2437,11 +2437,15 @@
      */
     public function addDecorators(array $decorators)
     {
-        foreach ($decorators as $decoratorInfo) {
+        foreach ($decorators as $decoratorName => $decoratorInfo) {
             if (is_string($decoratorInfo)) {
                 $this->addDecorator($decoratorInfo);
             } elseif ($decoratorInfo instanceof Zend_Form_Decorator_Interface) {
-                $this->addDecorator($decoratorInfo);
+                if (is_numeric($decoratorName)) {
+                    $this->addDecorator($decoratorInfo);
+                } else {
+                    $this->addDecorator(array($decoratorName => $decoratorInfo));
+                }
             } elseif (is_array($decoratorInfo)) {
                 $argc    = count($decoratorInfo);
                 $options = array();
Index: Form/DisplayGroup.php
===================================================================
--- Form/DisplayGroup.php	(revision 20600)
+++ Form/DisplayGroup.php	(working copy)
@@ -702,11 +702,15 @@
      */
     public function addDecorators(array $decorators)
     {
-        foreach ($decorators as $decoratorInfo) {
+        foreach ($decorators as $decoratorName => $decoratorInfo) {
             if (is_string($decoratorInfo)) {
                 $this->addDecorator($decoratorInfo);
             } elseif ($decoratorInfo instanceof Zend_Form_Decorator_Interface) {
-                $this->addDecorator($decoratorInfo);
+                if (is_numeric($decoratorName)) {
+                    $this->addDecorator($decoratorInfo);
+                } else {
+                    $this->addDecorator(array($decoratorName => $decoratorInfo));
+                }
             } elseif (is_array($decoratorInfo)) {
                 $argc    = count($decoratorInfo);
                 $options = array();
Index: Form/Element.php
===================================================================
--- Form/Element.php	(revision 20600)
+++ Form/Element.php	(working copy)
@@ -1824,11 +1824,15 @@
      */
     public function addDecorators(array $decorators)
     {
-        foreach ($decorators as $decoratorInfo) {
+        foreach ($decorators as $decoratorName => $decoratorInfo) {
             if (is_string($decoratorInfo)) {
                 $this->addDecorator($decoratorInfo);
             } elseif ($decoratorInfo instanceof Zend_Form_Decorator_Interface) {
-                $this->addDecorator($decoratorInfo);
+                if (is_numeric($decoratorName)) {
+                    $this->addDecorator($decoratorInfo);
+                } else {
+                    $this->addDecorator(array($decoratorName => $decoratorInfo));
+                }
             } elseif (is_array($decoratorInfo)) {
                 $argc    = count($decoratorInfo);
                 $options = array();
