Index: library/Zend/Form/Element/Multi.php
===================================================================
--- library/Zend/Form/Element/Multi.php	(revision 24718)
+++ library/Zend/Form/Element/Multi.php	(working copy)
@@ -97,16 +97,16 @@
     /**
      * Add an option
      *
-     * @param  string $option
-     * @param  string $value
+     * @param  string $value   Specifies the value to be sent to a server
+     * @param  string $content Specifies the content which the user can see
      * @return Zend_Form_Element_Multi
      */
-    public function addMultiOption($option, $value = '')
+    public function addMultiOption($value, $content = '')
     {
-        $option  = (string) $option;
+        $value = (string) $value;
         $this->_getMultiOptions();
-        if (!$this->_translateOption($option, $value)) {
-            $this->options[$option] = $value;
+        if (!$this->_translateOption($value, $content)) {
+            $this->options[$value] = $content;
         }
 
         return $this;
