Index: library/Zend/Form.php
===================================================================
--- library/Zend/Form.php	(revision 23428)
+++ library/Zend/Form.php	(revision )
@@ -1036,7 +1036,9 @@
                 }
             }
 
-            $this->_elements[$name] = $this->createElement($element, $name, $options);
+            $element = $this->createElement($element, $name, $options);
+            $name = $element->getName();
+            $this->_elements[$name] = $element;
         } elseif ($element instanceof Zend_Form_Element) {
             $prefixPaths              = array();
             $prefixPaths['decorator'] = $this->getPluginLoader('decorator')->getPaths();
Index: tests/Zend/Form/FormTest.php
===================================================================
--- tests/Zend/Form/FormTest.php	(revision 23427)
+++ tests/Zend/Form/FormTest.php	(revision )
@@ -4417,7 +4417,19 @@
         }
         $this->assertNotEquals($result,'');
     }
+
+    public function testAddElementNameIsSetCorrectInInternalArray()
+    {
+        $form = new Zend_Form();
+        $form->addElement('text', 'my-element');
+        $form->addElement(new Zend_Form_Element_Text('my-element-2'));
+
+        $this->assertNull($form->getElement('my-element'));
+        $this->assertNull($form->getElement('my-element-2'));
+        $this->assertNotNull($form->getElement('myelement'));
+        $this->assertNotNull($form->getElement('myelement2'));
-}
+    }
+}
 
 class Zend_Form_FormTest_DisplayGroup extends Zend_Form_DisplayGroup
 {
