Index: library/Zend/Controller/Action/Helper/ContextSwitch.php
===================================================================
--- library/Zend/Controller/Action/Helper/ContextSwitch.php	(revision 24535)
+++ library/Zend/Controller/Action/Helper/ContextSwitch.php	(working copy)
@@ -1304,14 +1304,14 @@
         if (null === $controller) {
             return array();
         }
-        $action     = (string) $action;
         $contextKey = $this->_contextKey;
 
         if (!isset($controller->$contextKey)) {
             return array();
         }
 
-        if (null !== $action) {
+        if (!empty($action)) {
+            $action = (string) $action;
             if (isset($controller->{$contextKey}[$action])) {
                 return $controller->{$contextKey}[$action];
             } else {
Index: tests/Zend/Controller/Action/Helper/ContextSwitchTest.php
===================================================================
--- tests/Zend/Controller/Action/Helper/ContextSwitchTest.php	(revision 24535)
+++ tests/Zend/Controller/Action/Helper/ContextSwitchTest.php	(working copy)
@@ -919,6 +919,19 @@
         $this->assertContains('foo', $suffix, $suffix);
         $this->assertNotContains('foo.foo', $suffix, $suffix);
     }
+    
+    /**
+     * @group ZF-11793
+     */
+    public function testGetActionContextsReturnsAllConfiguredActionContextsWhenNoArgumentProvided()
+    {
+        $this->helper->initContext();
+        $ctx = $this->helper->getActionContexts();
+        $this->assertArrayHasKey('foo', $ctx);
+        $this->assertEquals(array('xml'), $ctx['foo']);
+        $this->assertArrayHasKey('bar', $ctx);
+        $this->assertEquals(array('xml','json'), $ctx['bar']);
+    }
 }
 
 class Zend_Controller_Action_Helper_ContextSwitchTestController extends Zend_Controller_Action
@@ -1004,3 +1017,4 @@
 if (PHPUnit_MAIN_METHOD == "Zend_Controller_Action_Helper_ContextSwitchTest::main") {
     Zend_Controller_Action_Helper_ContextSwitchTest::main();
 }
+
