Index: tests/Zend/Controller/Action/Helper/ViewRendererTest.php
===================================================================
--- tests/Zend/Controller/Action/Helper/ViewRendererTest.php    (revision 24251)
+++ tests/Zend/Controller/Action/Helper/ViewRendererTest.php    (working copy)
@@ -827,19 +827,6 @@
         $body = $this->response->getBody();
         $this->assertContains('fooUseHelper invoked', $body, 'Received ' . $body);
     }
-    /**
-     * @group ZF-10725
-     */
-    public function testThatCharactersStrippedFromActionNameByDispatcherAreAlsoStrippedFromViewScriptName()
-    {
-        $this->request->setModuleName('default')
-                      ->setControllerName('foo')
-                      ->setActionName('-myBar-');
-        $controller = new Bar_IndexController($this->request, $this->response, array());
-        $this->helper->setActionController($controller);
-        $scriptName = $this->helper->getViewScript();
-        $this->assertEquals('foo/my-bar.phtml', $scriptName);
-    }

     public function testCorrectViewHelperPathShouldBePropagatedWhenSubControllerInvokedInDefaultModule()
     {
Index: library/Zend/Controller/Action/Helper/ViewRenderer.php
===================================================================
--- library/Zend/Controller/Action/Helper/ViewRenderer.php      (revision 24251)
+++ library/Zend/Controller/Action/Helper/ViewRenderer.php      (working copy)
@@ -630,14 +630,6 @@
         $replacePattern = array('/[^a-z0-9]+$/i', '/^[^a-z0-9]+/i');
         $vars['action'] = preg_replace($replacePattern, '', $vars['action']);

-        // Remove non-alphanumeric characters from action name
-        // see ZF-10725
-        $vars['action'] = preg_replace(
-            array('/[^a-z0-9]+$/', '/^[^a-z0-9]+/'),
-            array('', ''),
-            $vars['action']
-        );
-
         $inflector = $this->getInflector();
         if ($this->getNoController() || $this->getNeverController()) {
             $this->_setInflectorTarget($this->getViewScriptPathNoControllerSpec());
