Index: C:/WebServices/JadSVN/php_common/library/Zend/Soap/Server.php
===================================================================
--- C:/WebServices/JadSVN/php_common/library/Zend/Soap/Server.php	(revision 1582)
+++ C:/WebServices/JadSVN/php_common/library/Zend/Soap/Server.php	(revision 1720)
@@ -913,9 +913,10 @@
      * @link   http://www.w3.org/TR/soap12-part1/#faultcodes
      * @param  string|Exception $fault
      * @param  string $code SOAP Fault Codes
+     * @param  boolean $issueFault Issue SoapServer fault
      * @return SoapFault
      */
-    public function fault($fault = null, $code = "Receiver")
+    public function fault($fault = null, $code = "Receiver", $issueFault = false)
     {
         if ($fault instanceof Exception) {
             $class = get_class($fault);
@@ -940,6 +941,10 @@
             $code = "Receiver";
         }
 
+        if ($issueFault) {
+            $this->_getSoap()->fault($code, $message);
+        }
+
         return new SoapFault($code, $message);
     }
 
