Index: .
===================================================================
--- .	(revision 11496)
+++ .	(working copy)

@@ -63,7 +63,20 @@
         $this->_reflection = new Zend_Server_Reflection();
         $this->_extractComplexTypes = $extractComplexTypes;
     }
-    
+
+    /**
+     * Set an Object that will be Autodiscovered
+     *
+     * @param object $object
+     * @param string $namespace
+     */
+    public function setObject($object, $namespace = '')
+    {
+        $class = get_class($object);
+        unset($object);
+        $this->setClass($class, $namespace, null);
+    }
+
     /**
      * Set the Class the SOAP server will use
      *