Index: library/Zend/Service/ReCaptcha.php
===================================================================
--- library/Zend/Service/ReCaptcha.php	(revision 24672)
+++ library/Zend/Service/ReCaptcha.php	(working copy)
@@ -105,6 +105,7 @@
     protected $_options = array(
         'theme' => 'red',
         'lang' => 'en',
+        'custom_translations' => array()
     );
 
     /**
Index: tests/Zend/Captcha/ReCaptchaTest.php
===================================================================
--- tests/Zend/Captcha/ReCaptchaTest.php	(revision 24672)
+++ tests/Zend/Captcha/ReCaptchaTest.php	(working copy)
@@ -185,6 +185,18 @@
         $captcha = new Zend_Captcha_ReCaptcha;
         $this->assertEquals('Captcha_ReCaptcha', $captcha->getDecorator());
     }
+
+    /** @group ZF-12086 */
+    public function testAllowsSettingCustomTranslationsOptionOnServiceObject()
+    {
+        $captcha = new Zend_Captcha_ReCaptcha;
+        $captcha->setOption('custom_translations', 
+                            array('instructions_visual' => 'Wpisz dwa wyrazy:', 
+                                  'instructions_audio' => 'Wpisz uslyszany tekst:'));
+        $this->assertEquals(array('instructions_visual' => 'Wpisz dwa wyrazy:', 
+                                  'instructions_audio' => 'Wpisz uslyszany tekst:'), 
+                            $captcha->getService()->getOption('custom_translations'));
+    }
 }
 
 class Zend_Captcha_ReCaptchaTest_SessionContainer
