Index: ControllerTestCase.php
===================================================================
--- ControllerTestCase.php	(revision 18161)
+++ ControllerTestCase.php	(working copy)
@@ -243,6 +243,8 @@
      */
     public function resetRequest()
     {
+        $this->getRequest()->clearPost();
+        $this->getRequest()->clearQuery();
         $this->_request = null;
         return $this;
     }
