--- library/Zend/Gdata/Docs.php	(revision 16177)
+++ library/Zend/Gdata/Docs.php	(working copy)
@@ -221,8 +221,7 @@
 
         // Set the mime type of the data.
         if ($mimeType === null) {
-          $slugHeader =  $fs->getSlug();
-          $filenameParts = explode('.', $slugHeader);
+          $filenameParts = explode('.', $fileLocation);
           $fileExtension = end($filenameParts);
           $mimeType = self::lookupMimeType($fileExtension);
         }
