Index: Filesystem.php
===================================================================
--- Filesystem.php	(revision 8547)
+++ Filesystem.php	(working copy)
@@ -26,6 +26,8 @@
 /** Zend_Search_Lucene_Storage_File_Filesystem */
 require_once 'Zend/Search/Lucene/Storage/File/Filesystem.php';
 
+/** Zend_Search_Lucene_Exception */
+require_once 'Zend/Search/Lucene/Exception.php';
 
 /**
  * FileSystem implementation of Directory abstraction.
@@ -119,10 +121,8 @@
         if (!is_dir($path)) {
             if (file_exists($path)) {
                 throw new Zend_Search_Lucene_Exception('Path exists, but it\'s not a directory');
-            } else {
-                if (!self::mkdirs($path)) {
+            } elseif(!self::mkdirs($path)) {
                     throw new Zend_Search_Lucene_Exception("Can't create directory '$path'.");
-                }
             }
         }
         $this->_dirPath = $path;
@@ -182,13 +182,12 @@
         unset($this->_fileHandlers[$filename]);
         $this->_fileHandlers[$filename] = new Zend_Search_Lucene_Storage_File_Filesystem($this->_dirPath . '/' . $filename, 'w+b');
 
-        global $php_errormsg;
-        $trackErrors = ini_get('track_errors'); ini_set('track_errors', '1');
+        ini_set('track_errors', 1);
         if (!@chmod($this->_dirPath . '/' . $filename, self::$_defaultFilePermissions)) {
-            ini_set('track_errors', $trackErrors);
+            ini_restore('track_errors');
             throw new Zend_Search_Lucene_Exception($php_errormsg);
         }
-        ini_set('track_errors', $trackErrors);
+        ini_restore('track_errors');
 
         return $this->_fileHandlers[$filename];
     }
@@ -208,32 +207,29 @@
         }
         unset($this->_fileHandlers[$filename]);
 
-        global $php_errormsg;
-        $trackErrors = ini_get('track_errors'); ini_set('track_errors', '1');
+        ini_set('track_errors', 1);
         if (!@unlink($this->_dirPath . '/' . $filename)) {
-            ini_set('track_errors', $trackErrors);
-            throw $e;
-            
+            ini_restore('track_errors');
             throw new Zend_Search_Lucene_Exception('Can\'t delete file: ' . $php_errormsg);
         }
-        ini_set('track_errors', $trackErrors);
+        ini_restore('track_errors');
     }
 
     /**
@@ -286,8 +282,6 @@
      */
     public function renameFile($from, $to)
     {
-        global $php_errormsg;
-
         if (isset($this->_fileHandlers[$from])) {
             $this->_fileHandlers[$from]->close();
         }
@@ -304,16 +298,14 @@
             }
         }
 
-        $trackErrors = ini_get('track_errors');
-        ini_set('track_errors', '1');
-
+        ini_set('track_errors', 1);
         $success = @rename($this->_dirPath . '/' . $from, $this->_dirPath . '/' . $to);
         if (!$success) {
-            ini_set('track_errors', $trackErrors);
+            ini_restore('track_errors');
             throw new Zend_Search_Lucene_Exception($php_errormsg);
         }
 
-        ini_set('track_errors', $trackErrors);
+        ini_restore('track_errors');
 
         return $success;
     }
