Index: ResponseHeader.php
===================================================================
--- ResponseHeader.php	(revision 22)
+++ ResponseHeader.php	(working copy)
@@ -65,6 +65,11 @@
      * @var int Response code
      */
     protected $_code              = 200;
+    
+    /**
+     * @var int Actual response code
+     */
+    protected $_actualCode        = null;
 
     /**
      * @var string Header
@@ -197,6 +202,9 @@
                     $failure = 'Failed asserting response code IS NOT "%s"';
                 }
                 $failure = sprintf($failure, $this->_code);
+                if (!$this->_negate && $this->_actualCode) {
+                    $failure .= sprintf(PHP_EOL . 'Was "%s"', $this->_actualCode);
+                }
                 break;
             case self::ASSERT_HEADER:
                 $failure = 'Failed asserting response header "%s" found';
@@ -250,6 +258,7 @@
     protected function _code(Zend_Controller_Response_Abstract $response, $code)
     {
         $test = $this->_getCode($response);
+        $this->_actualCode = $test;
         return ($test == $code);
     }
 
