Index: tests/Zend/Validate/DigitsTest.php
===================================================================
--- tests/Zend/Validate/DigitsTest.php	(revision 20802)
+++ tests/Zend/Validate/DigitsTest.php	(working copy)
@@ -120,6 +120,7 @@
     }
 
     /**
+	 * @group  ZF-9032
      * @return void
      */
     public function testInvalidValueResultsInProperValidationFailureMessages()
@@ -127,7 +128,7 @@
         $this->assertFalse($this->_validator->isValid('#'));
         $messages = $this->_validator->getMessages();
         $arrayExpected = array(
-            Zend_Validate_Digits::NOT_DIGITS => '\'#\' contains not only digit characters'
+            Zend_Validate_Digits::NOT_DIGITS => '\'#\' contains characters that are not digits; only digits are allowed'
             );
         $this->assertThat($messages, $this->identicalTo($arrayExpected));
     }
Index: library/Zend/Validate/Digits.php
===================================================================
--- library/Zend/Validate/Digits.php	(revision 20802)
+++ library/Zend/Validate/Digits.php	(working copy)
@@ -49,7 +49,7 @@
      * @var array
      */
     protected $_messageTemplates = array(
-        self::NOT_DIGITS   => "'%value%' contains not only digit characters",
+        self::NOT_DIGITS   => "'%value%' contains characters that are not digits; only digits are allowed",
         self::STRING_EMPTY => "'%value%' is an empty string",
         self::INVALID      => "Invalid type given, value should be string, integer or float",
     );
