Index: library/Zend/Mail/Storage/Pop3.php
===================================================================
--- library/Zend/Mail/Storage/Pop3.php	(revision 6076)
+++ library/Zend/Mail/Storage/Pop3.php	(working copy)
@@ -128,7 +128,7 @@
             throw new Zend_Mail_Storage_Exception('not implemented');
         }
 
-        $content = $this->_protocol->retrive($id);
+        $content = $this->_protocol->retrieve($id);
         // TODO: find a way to avoid decoding the headers
         Zend_Mime_Decode::splitMessage($content, $null, $body);
         return $body;
