Index: Abstract.php
===================================================================
--- Abstract.php	(revision 909)
+++ Abstract.php	(working copy)
@@ -306,6 +306,16 @@
     }
 
     /**
+     * Clear all the assigned variables
+     *
+     * @return void
+     */
+    public function clearParams()
+    {
+        $this->_vars = array();
+    }
+
+    /**
      * Assigns variables to the view script via differing strategies.
      *
      * Zend_View::assign('name', $value) assigns a variable called 'name'
@@ -434,9 +444,9 @@
         foreach ((array) $path as $dir) {
         	// attempt to strip any possible separator and
         	// append the system directory separator
-            $dir = rtrim($dir, '\\/' . DIRECTORY_SEPARATOR) 
+            $dir = rtrim($dir, '\\/' . DIRECTORY_SEPARATOR)
                  . DIRECTORY_SEPARATOR;
-            
+
             // add to the top of the stack.
             array_unshift($this->_path[$type], $dir);
         }
@@ -472,22 +482,22 @@
         if (class_exists($class, false)) {
         	return $class;
         }
-        
+
         // only look for "$Name.php"
         $file = ucfirst($name) . '.php';
         foreach ($this->_path[$type] as $dir) {
             if (is_readable($dir . $file)) {
                 include $dir . $file;
-                
+
                 if (! class_exists($class, false)) {
                 	$msg = "$type '$name' loaded but class '$class' not found within";
                 	throw new Zend_View_Exception($msg);
                 }
-                
+
                 return $class;
             }
         }
-        
+
         throw new Zend_View_Exception("$type '$name' not found in path.");
     }
 }
