Index: /library/Zend/Controller/Plugin/Broker.php
===================================================================
--- /library/Zend/Controller/Plugin/Broker.php	(revision 7160)
+++ /library/Zend/Controller/Plugin/Broker.php	(working copy)
@@ -71,7 +71,6 @@
             }
             $this->_plugins[$stackIndex] = $plugin;
         } else {
-            $stackIndex = count($this->_plugins);
             while (isset($this->_plugins[$stackIndex])) {
                 ++$stackIndex;
             }
