--- tests/Zend/Controller/Action/Helper/RedirectorTest.php	(revision 23864)
+++ tests/Zend/Controller/Action/Helper/RedirectorTest.php	(working copy)
@@ -533,6 +533,28 @@
         $this->assertEquals($expected, $result);
     }
 
+    /** @group ZF-6025 */
+    public function testGotoSimpleDoesNotHardcodeControllerActionModuleKeys()
+    {
+        $this->request->setControllerKey('foo')
+                      ->setActionKey('bar')
+                      ->setModuleKey('baz');
+
+        $this->router->removeRoute('default');
+        $this->router->addRoute('default', new Zend_Controller_Router_Route(
+            ':baz/:foo/:bar/*', array(
+                'baz' => 'default',
+                'foo' => 'index',
+                'bar' => 'index'
+            )
+        ));
+
+        $this->redirector->gotoSimple('babar', 'barbapapa', 'barbazoo', array('asd' => 1));
+        $result = $this->redirector->getRedirectUrl();
+        $expected = '/barbazoo/barbapapa/babar/asd/1';
+        $this->assertEquals($expected, $result);
+    }
+
     /**#@-*/
 
     protected function _parseLocationHeaderValue()
