Index: tests/Zend/Dom/QueryTest.php
===================================================================
--- tests/Zend/Dom/QueryTest.php	(revision 16361)
+++ tests/Zend/Dom/QueryTest.php	(working copy)
@@ -78,6 +78,20 @@
         $this->assertNull($this->query->getDocument());
     }
 
+    public function testDocShouldBeNullByEmptyStringConstructor()
+    {
+        $emptyStr = "";
+        $query = new Zend_Dom_Query($emptyStr);
+        $this->assertNull($this->query->getDocument());
+    }
+
+    public function testDocShouldBeNullByEmptyStringSet()
+    {
+        $emptyStr = "";
+        $this->query->setDocument($emptyStr);
+        $this->assertNull($this->query->getDocument());
+    }
+
     public function testDocTypeShouldBeNullByDefault()
     {
         $this->assertNull($this->query->getDocumentType());
Index: library/Zend/Dom/Query.php
===================================================================
--- library/Zend/Dom/Query.php	(revision 16361)
+++ library/Zend/Dom/Query.php	(working copy)
@@ -65,9 +65,7 @@
      */
     public function __construct($document = null)
     {
-        if (null !== $document) {
-            $this->setDocument($document);
-        }
+        $this->setDocument($document);
     }
 
     /**
@@ -78,6 +76,9 @@
      */
     public function setDocument($document)
     {
+        if (0 === strlen($document)) {
+            return $this;
+        }
         if ('<?xml' == substr(trim($document), 0, 5)) {
             return $this->setDocumentXml($document);
         }
