Index: library/Zend/View/Helper/FormCheckbox.php
===================================================================
--- library/Zend/View/Helper/FormCheckbox.php	(revision 13200)
+++ library/Zend/View/Helper/FormCheckbox.php	(working copy)
@@ -105,10 +105,10 @@
 
     /**
      * Determine checkbox information
-     * 
-     * @param  string $value 
-     * @param  bool $checked 
-     * @param  array|null $checkedOptions 
+     *
+     * @param  string $value
+     * @param  bool $checked
+     * @param  array|null $checkedOptions
      * @return array
      */
     public static function determineCheckboxInfo($value, $checked, array $checkedOptions = null)
@@ -140,7 +140,7 @@
 
         // is the element checked?
         $checkedString = '';
-        if ($checked || ($value === $checkedValue)) {
+        if ($checked || ((string) $value === $checkedValue)) {
             $checkedString = ' checked="checked"';
             $checked = true;
         } else {
Index: tests/Zend/View/Helper/FormCheckboxTest.php
===================================================================
--- tests/Zend/View/Helper/FormCheckboxTest.php	(revision 13200)
+++ tests/Zend/View/Helper/FormCheckboxTest.php	(working copy)
@@ -247,6 +247,12 @@
         $test = $this->helper->formCheckbox('foo', 'bar');
         $this->assertContains(' />', $test);
     }
+
+    public function testZF5256()
+    {
+        $test = $this->helper->formCheckbox('myfield', 1, array(), array('checked' => 1, 'unChecked' => 0));
+        $this->assertContains('checked', $test);
+    }
 }
 
 // Call Zend_View_Helper_FormCheckboxTest::main() if this source file is executed directly.
