Index: Select.php
===================================================================
--- Select.php	(revision 9302)
+++ Select.php	(working copy)
@@ -778,8 +778,8 @@
             throw new Zend_Db_Select_Exception("You can only perform a joinUsing after specifying a FROM table");
         }
 
-        $join  = $this->_adapter->quoteIdentifier(key($this->_parts[self::FROM]));
-        $from  = $this->_adapter->quoteIdentifier($this->_uniqueCorrelation($name));
+        $join  = $this->_adapter->quoteIdentifier(key($this->_parts[self::FROM]), true);
+        $from  = $this->_adapter->quoteIdentifier($this->_uniqueCorrelation($name), true);
 
         $cond1 = $from . '.' . $cond;
         $cond2 = $join . '.' . $cond;
