--- library/Zend/Controller/Action/Helper/Redirector.php	(revision 5159)
+++ library/Zend/Controller/Action/Helper/Redirector.php	(working copy)
@@ -303,7 +303,7 @@
         }
 
         // If relative URL, decide if we should prepend base URL
-        if (!preg_match('|^[a-z]+://|', $url)) {
+        if (!preg_match('|^[a-z]+://|', $url) && true === $prependBase) {
             $url = $this->_prependBase($url);
         }
 
