--- library/Zend/Service/Rackspace/Abstract.php	(revision 25009)
+++ library/Zend/Service/Rackspace/Abstract.php	(working copy)
@@ -325,6 +325,13 @@
      */
     public function authenticate()
     {
+        if (empty($this->user)) {
+            /**
+             * @see Zend_Service_Rackspace_Exception
+             */
+            require_once 'Zend/Service/Rackspace/Exception.php';
+            throw new Zend_Service_Rackspace_Exception("User has not been set");
+        }
         $headers = array (
             self::AUTHUSER_HEADER => $this->user,
             self::AUTHKEY_HEADER => $this->key
