Index: Abstract.php
===================================================================
--- Abstract.php	(revision 7327)
+++ Abstract.php	(working copy)
@@ -1043,7 +1043,12 @@
      */
     public function createRow(array $data = array())
     {
-        $defaults = array_combine($this->_cols, array_fill(0, count($this->_cols), null));
+        $defaults = array();
+        foreach ($this->_cols as $col) {
+            if (array_key_exists($this->_metadata[$col]['DEFAULT'])) {
+                $defaults[$col] = new Zend_Db_Expr('DEFAULT');
+            }
+        }
         $keys = array_flip($this->_cols);
         $data = array_intersect_key($data, $keys);
         $data = array_merge($defaults, $data);
