Index: library/Zend/Feed/Writer/Feed/FeedAbstract.php
===================================================================
--- library/Zend/Feed/Writer/Feed/FeedAbstract.php	(revision 24636)
+++ library/Zend/Feed/Writer/Feed/FeedAbstract.php	(working copy)
@@ -176,7 +176,7 @@
         $zdate = null;
         if ($date === null) {
             $zdate = new Zend_Date;
-        } elseif (ctype_digit($date) && strlen($date) == 10) {
+        } elseif (ctype_digit($date)) {
             $zdate = new Zend_Date($date, Zend_Date::TIMESTAMP);
         } elseif ($date instanceof Zend_Date) {
             $zdate = $date;
@@ -197,7 +197,7 @@
         $zdate = null;
         if ($date === null) {
             $zdate = new Zend_Date;
-        } elseif (ctype_digit($date) && strlen($date) == 10) {
+        } elseif (ctype_digit($date)) {
             $zdate = new Zend_Date($date, Zend_Date::TIMESTAMP);
         } elseif ($date instanceof Zend_Date) {
             $zdate = $date;
@@ -218,7 +218,7 @@
         $zdate = null;
         if ($date === null) {
             $zdate = new Zend_Date;
-        } elseif (ctype_digit($date) && strlen($date) == 10) {
+        } elseif (ctype_digit($date)) {
             $zdate = new Zend_Date($date, Zend_Date::TIMESTAMP);
         } elseif ($date instanceof Zend_Date) {
             $zdate = $date;
Index: tests/Zend/Feed/Writer/FeedTest.php
===================================================================
--- tests/Zend/Feed/Writer/FeedTest.php	(revision 24636)
+++ tests/Zend/Feed/Writer/FeedTest.php	(working copy)
@@ -195,6 +195,17 @@
         $myDate = new Zend_Date('1234567890', Zend_Date::TIMESTAMP);
         $this->assertTrue($myDate->equals($writer->getDateCreated()));
     }
+    
+    /**
+     * @group ZF-12023
+     */
+    public function testSetDateCreatedUsesGivenUnixTimestampThatIsLessThanTenDigits()
+    {
+        $writer = new Zend_Feed_Writer_Feed;
+        $writer->setDateCreated(123456789);
+        $myDate = new Zend_Date('123456789', Zend_Date::TIMESTAMP);
+        $this->assertTrue($myDate->equals($writer->getDateCreated()));
+    }
 
     public function testSetDateCreatedUsesZendDateObject()
     {
@@ -220,6 +231,17 @@
         $this->assertTrue($myDate->equals($writer->getDateModified()));
     }
 
+    /**
+     * @group ZF-12023
+     */
+    public function testSetDateModifiedUsesGivenUnixTimestampThatIsLessThanTenDigits()
+    {
+        $writer = new Zend_Feed_Writer_Feed;
+        $writer->setDateModified(123456789);
+        $myDate = new Zend_Date('123456789', Zend_Date::TIMESTAMP);
+        $this->assertTrue($myDate->equals($writer->getDateModified()));
+    }
+
     public function testSetDateModifiedUsesZendDateObject()
     {
         $writer = new Zend_Feed_Writer_Feed;
@@ -276,6 +298,17 @@
         $this->assertTrue($myDate->equals($writer->getLastBuildDate()));
     }
 
+    /**
+     * @group ZF-12023
+     */
+    public function testSetLastBuildDateUsesGivenUnixTimestampThatIsLessThanTenDigits()
+    {
+        $writer = new Zend_Feed_Writer_Feed;
+        $writer->setLastBuildDate(123456789);
+        $myDate = new Zend_Date('123456789', Zend_Date::TIMESTAMP);
+        $this->assertTrue($myDate->equals($writer->getLastBuildDate()));
+    }
+
     public function testSetLastBuildDateUsesZendDateObject()
     {
         $writer = new Zend_Feed_Writer_Feed;
