Index: library/Zend/Application/Resource/Modules.php
===================================================================
--- library/Zend/Application/Resource/Modules.php       (revision 16898)
+++ library/Zend/Application/Resource/Modules.php       (working copy)
@@ -61,11 +61,7 @@
         $front = $bootstrap->getResource('FrontController');

         $modules = $front->getControllerDirectory();
-        $default = $front->getDefaultModule();
         foreach (array_keys($modules) as $module) {
-            if ($module === $default) {
-                continue;
-            }

             $bootstrapClass = $this->_formatModuleName($module) . '_Bootstrap';
             if (!class_exists($bootstrapClass, false)) {
