Index: library/Zend/Db/Table/Row/Abstract.php
===================================================================
--- library/Zend/Db/Table/Row/Abstract.php	(revision 5066)
+++ library/Zend/Db/Table/Row/Abstract.php	(working copy)
@@ -823,7 +823,7 @@
             'table'    => $matchTable,
             'data'     => $stmt->fetchAll(Zend_Db::FETCH_ASSOC),
             'rowClass' => $matchTable->getRowClass(),
-            'clean'    => true
+            'stored'   => true
         );
 
         $rowsetClass = $matchTable->getRowsetClass();
