--- ZendFramework-1.9.7/library/Zend/Pdf/Font.php~	2010-01-13 01:38:25.537700969 +0100
+++ ZendFramework-1.9.7/library/Zend/Pdf/Font.php	2010-01-13 01:46:38.550555791 +0100
@@ -551,7 +551,7 @@
                 break;
 
             default:
-                require_once 'Zend/Pdf/Excaption.php';
+                require_once 'Zend/Pdf/Exception.php';
                 throw new Zend_Pdf_Exception("Unknown font name: $name",
                                              Zend_Pdf_Exception::BAD_FONT_NAME);
         }
