--- .svn/text-base/DebugTest.php.svn-base	2010-06-22 22:14:19.702487412 +0200
+++ DebugTest.php	2010-08-19 22:38:12.951616962 +0200
@@ -118,4 +118,14 @@
         $this->assertContains("</pre>", $result);
     }
 
+    public function testDebugDumpBacktrace()
+    {
+        Zend_Debug::setSapi('cli');
+        $result = Zend_Debug::dump_backtrace(null, false);
+        $result = explode(PHP_EOL, $result);
+		$result = $result[2];
+		$this->assertContains('Zend_Debug::dump_backtrace', $result);
+		$this->assertContains('called at', $result);
+    }
+
 }
