Index: Zend/Loader/PluginLoader.php
===================================================================
--- Zend/Loader/PluginLoader.php	(revision 21797)
+++ Zend/Loader/PluginLoader.php	(working copy)
@@ -471,11 +471,13 @@
      */
     protected static function _appendIncFile($incFile)
     {
-        if (!file_exists(self::$_includeFileCache)) {
-            $file = '<?php';
-        } else {
+        if (file_exists(self::$_includeFileCache))
+        {
             $file = file_get_contents(self::$_includeFileCache);
         }
+        if (!isset($file) || (!$file)) {
+            $file = '<?php';
+        }
         if (!strstr($file, $incFile)) {
             $file .= "\ninclude_once '$incFile';";
             file_put_contents(self::$_includeFileCache, $file);
