--- library/Zend/Json/Server.php	(revision 16013)
+++ library/Zend/Json/Server.php	(working copy)
@@ -521,6 +521,22 @@
         $serviceMap    = $this->getServiceMap();
         $service       = $serviceMap->getService($method);
         $serviceParams = $service->getParams();
+        
+        // Only reorder a named parameter field
+        if(array_keys($params) != range(0, count($params) - 1)) {
+            $nParams = array();
+            foreach($serviceParams as $serviceParam) {
+                if(isset($params[$serviceParam['name']])) {
+                    $nParams[] = $params[$serviceParam['name']];
+                }
+                else {
+                    if(!$serviceParam['optional']) {
+                        return $this->fault('Invalid params', -32602);
+                    }
+                }
+            }
+            $params = $nParams;
+        }
 
         if (count($params) < count($serviceParams)) {
             $params = $this->_getDefaultParams($params, $serviceParams);
