Index: Dispatcher.php
===================================================================
--- Dispatcher.php	(revision 2958)
+++ Dispatcher.php	(working copy)
@@ -581,8 +581,13 @@
                 throw new Zend_Controller_Dispatcher_Exception('Invalid controller class ("' . $moduleClass . '")');
             }
             $className = $moduleClass;
-        } else {
-            Zend::loadClass($className, $dispatchDir);
+        } else {
+            try {
+                Zend::loadClass($className, $dispatchDir);
+            } catch (Zend_Exception $e) {
+                settype($e, 'Zend_Controller_Dispatcher_Exception');
+                throw $e;
+            }            
         }
         
         /**
