Index: library/Zend/Gdata/Spreadsheets/SpreadsheetEntry.php
===================================================================
--- library/Zend/Gdata/Spreadsheets/SpreadsheetEntry.php	(revision 6247)
+++ library/Zend/Gdata/Spreadsheets/SpreadsheetEntry.php	(working copy)
@@ -20,7 +20,7 @@
  */
 
 /**
- * @see Zend_Gdata_EntryAtom
+ * @see Zend_Gdata_Entry
  */
 require_once 'Zend/Gdata/Entry.php';
 
Index: library/Zend/Gdata/Spreadsheets/CellEntry.php
===================================================================
--- library/Zend/Gdata/Spreadsheets/CellEntry.php	(revision 6247)
+++ library/Zend/Gdata/Spreadsheets/CellEntry.php	(working copy)
@@ -20,7 +20,7 @@
  */
 
 /**
- * @see Zend_Gdata_EntryAtom
+ * @see Zend_Gdata_Entry
  */
 require_once 'Zend/Gdata/Entry.php';
 
Index: library/Zend/Gdata/Calendar/ListEntry.php
===================================================================
--- library/Zend/Gdata/Calendar/ListEntry.php	(revision 6247)
+++ library/Zend/Gdata/Calendar/ListEntry.php	(working copy)
@@ -20,7 +20,7 @@
  */
 
 /**
- * @see Zend_Gdata_EntryAtom
+ * @see Zend_Gdata_Entry
  */
 require_once 'Zend/Gdata/Entry.php';
 
Index: library/Zend/Gdata/Calendar/EventEntry.php
===================================================================
--- library/Zend/Gdata/Calendar/EventEntry.php	(revision 6247)
+++ library/Zend/Gdata/Calendar/EventEntry.php	(working copy)
@@ -20,7 +20,7 @@
  */
 
 /**
- * @see Zend_Gdata_EntryAtom
+ * @see Zend_Gdata_Entry
  */
 require_once 'Zend/Gdata/Entry.php';
 
