--- tests/Zend/Controller/Request/Apache404Test.php	(revision 23864)
+++ tests/Zend/Controller/Request/Apache404Test.php	(working copy)
@@ -83,11 +83,12 @@
 
     /**
      * @group ZF-3057
+     * @group ZF-9776 changed REDIRECT_QUERYSTRING to REDIRECT_QUERY_STRING
      */
     public function testRedirectQueryStringShouldBeParsedIntoGetVars()
     {
         $_SERVER['REDIRECT_URL']         = '/foo/bar';
-        $_SERVER['REDIRECT_QUERYSTRING'] = 'baz=bat&bat=delta';
+        $_SERVER['REDIRECT_QUERY_STRING'] = 'baz=bat&bat=delta';
         $_SERVER['REQUEST_URI']          = '/baz/bat';
 
         $request = new Zend_Controller_Request_Apache404();
