Index: Validate/File/MimeType.php
===================================================================
--- Validate/File/MimeType.php	(revision 5763)
+++ Validate/File/MimeType.php	(working copy)
@@ -334,6 +334,7 @@

         $types = explode('/', $this->_type);
         $types = array_merge($types, explode('-', $this->_type));
+        $types = array_merge($types, explode(';', $this->_type));
         foreach($mimetype as $mime) {
             if (in_array($mime, $types)) {
                 return true;
