Index: AutoComplete.php
===================================================================
--- AutoComplete.php	(revision 22445)
+++ AutoComplete.php	(working copy)
@@ -58,15 +58,16 @@
         }
 
         $data = (array) $data;
-        $output = "";
+        $output = "[ ";
         foreach($data AS $k => $v) {
             if(is_numeric($k)) {
-                $output .= $v."\n";
+                $output .= '"' . $v . '", ';
             } else {
                 $output .= $k."|".$v."\n";
             }
         }
-
+	
+	$output .= " ]";
         if (!$keepLayouts) {
             $this->disableLayouts();
         }
