Index: library/Zend/Soap/Wsdl.php
===================================================================
--- library/Zend/Soap/Wsdl.php	(revision 23986)
+++ library/Zend/Soap/Wsdl.php	(working copy)
@@ -255,6 +255,8 @@
         if (is_string($fault) && (strlen(trim($fault)) >= 1)) {
             $node = $this->_dom->createElement('fault');
             $node->setAttribute('message', $fault);
+            //remove tns:
+            $node->setAttribute('name', substr(strchr($fault, ':'), 1));
             $operation->appendChild($node);
         }
 
@@ -560,7 +562,7 @@
                 return 'soap-enc:Array';
                 break;
             case 'object':
-                return 'xsd:struct';
+                return 'soap-enc:Struct';
                 break;
             case 'mixed':
                 return 'xsd:anyType';
