*** Zend/ProgressBar/Adapter/Console.php	2009-03-06 10:20:19.000000000 -0600
--- Zend/ProgressBar/Adapter/Console.php	2009-03-06 12:46:13.000000000 -0600
***************
*** 148,155 ****
       */
      public function __construct($options = null)
      {
          // If STDOUT isn't defined, open a local resource
!         if (!defined('STDOUT')) {
              $this->_stdout = fopen('php://stdout', 'w');
          }
  
--- 148,161 ----
       */
      public function __construct($options = null)
      {
+         // If output override set, use it.
+         if (isset($options['output']) and is_string($options['output'])) {
+             $this->_stdout = fopen($options['output'], 'w');
+             if ($this->_stdout === false) {
+               throw new Zend_ProgressBar_Exception("Unable to open output stream");
+             }
          // If STDOUT isn't defined, open a local resource
!         } elseif (!defined('STDOUT')) {
              $this->_stdout = fopen('php://stdout', 'w');
          }
  
