Index: Mssql.php
===================================================================
--- Mssql.php	(revision 13128)
+++ Mssql.php	(working copy)
@@ -107,6 +107,13 @@
                 case 'mssql':
                     $this->_pdoType = 'mssql';
                     break;
+                case 'odbc':
+                    $this->_pdoType = 'odbc';
+                    $dsn['DATABASE']    = $dsn['dbname'];
+                    unset($dsn['dbname']);
+                    $dsn['SERVER']    = $dsn['host'];
+                    unset($dsn['host']);
+                    break;
                 case 'dblib':
                 default:
                     $this->_pdoType = 'dblib';
