Index: Parameter.php
===================================================================
--- Parameter.php	(revision 21377)
+++ Parameter.php	(working copy)
@@ -109,13 +109,16 @@
      */
     public function getType()
     {
-        if ($docblock = $this->getDeclaringFunction()->getDocblock()) {
-            $params = $docblock->getTags('param');
+        $function = $this->getDeclaringFunction();
 
-            if (isset($params[$this->getPosition()])) {
-                return $params[$this->getPosition()]->getType();
-            }
+        if (!$function->hasDocblock()) {
+            return null;
+        }
+
+        $params = $function->getDocblock()->getTags('param');
 
+        if (isset($params[$this->getPosition()])) {
+            return $params[$this->getPosition()]->getType();
         }
 
         return null;
Index: Function.php
===================================================================
--- Function.php	(revision 21377)
+++ Function.php	(working copy)
@@ -40,11 +40,11 @@
      */
     public function getDocblock($reflectionClass = 'Zend_Reflection_Docblock')
     {
-        if ('' == ($comment = $this->getDocComment())) {
+        if (!$this->hasDocblock()) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception($this->getName() . ' does not have a docblock');
         }
-        $instance = new $reflectionClass($comment);
+        $instance = new $reflectionClass($this->getDocDomment());
         if (!$instance instanceof Zend_Reflection_Docblock) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception('Invalid reflection class provided; must extend Zend_Reflection_Docblock');
@@ -53,6 +53,16 @@
     }
 
     /**
+     * Whether the function has a docblock
+     *
+     * @return bool
+     */
+    public function hasDocblock()
+    {
+        return ('' == $this->getDocDomment());
+    }
+
+    /**
      * Get start line (position) of function
      *
      * @param  bool $includeDocComment
