Index: /home/matt/Documents/Projects/ZendFramework/library/Zend/Controller/Action/Helper/ContextSwitch.php
===================================================================
--- /home/matt/Documents/Projects/ZendFramework/library/Zend/Controller/Action/Helper/ContextSwitch.php	(revision 11000)
+++ /home/matt/Documents/Projects/ZendFramework/library/Zend/Controller/Action/Helper/ContextSwitch.php	(working copy)
@@ -140,7 +140,7 @@
                 ),
                 'xml'  => array(
                     'suffix'    => 'xml',
-                    'headers'   => array('Content-Type' => 'text/xml'),
+                    'headers'   => array('Content-Type' => 'application/xml'),
                 )
             ));
         }
