--- library/Zend/Config/Ini.php	(revision 23872)
+++ library/Zend/Config/Ini.php	(working copy)
@@ -277,6 +277,10 @@
     {
         if (strpos($key, $this->_nestSeparator) !== false) {
             $pieces = explode($this->_nestSeparator, $key, 2);
+
+            // Allow whitespace between pieces e.g. 'elements   .key   .value = "text"'
+            $pieces = array_map('trim', $pieces);
+
             if (strlen($pieces[0]) && strlen($pieces[1])) {
                 if (!isset($config[$pieces[0]])) {
                     if ($pieces[0] === '0' && !empty($config)) {
