Index: tests/Zend/Validate/EmailAddressTest.php
===================================================================
--- tests/Zend/Validate/EmailAddressTest.php	(revision 25051)
+++ tests/Zend/Validate/EmailAddressTest.php	(working copy)
@@ -165,7 +165,7 @@
 
         $messages = $this->_validator->getMessages();
 
-        $this->assertType('array', $messages);
+        $this->assertTrue(is_array($messages));
         $this->assertEquals(0, count($messages));
     }
 
