Index: AllTests.php
===================================================================
--- AllTests.php	(revision 12830)
+++ AllTests.php	(working copy)
@@ -1,4 +1,9 @@
 <?php
+/**
+ * Test helper
+ */
+require_once dirname(dirname(dirname(__FILE__))) . DIRECTORY_SEPARATOR . 'TestHelper.php';
+
 if (!defined('PHPUnit_MAIN_METHOD')) {
     define('PHPUnit_MAIN_METHOD', 'Zend_Mail_AllTests::main');
 }
