Index: Zend_Framework/library/Zend/Amf/Response/Http.php
===================================================================
--- Zend_Framework/library/Zend/Amf/Response/Http.php	(Revision 21958)
+++ Zend_Framework/library/Zend/Amf/Response/Http.php	(Arbeitskopie)
@@ -41,8 +41,9 @@
     public function getResponse()
     {
         if (!headers_sent()) {
-            header('Cache-Control: cache, must-revalidate');
-            header('Pragma: public');
+            header('Cache-Control: no-cache, must-revalidate');
+            header('Expires: Thu, 19 Nov 1981 08:52:00 GMT');
+            header('Pragma: no-cache');
             header('Content-Type: application/x-amf');
         }
         return parent::getResponse();
