--- a/vendor/Zend/Validate/CreditCard.php
+++ b/vendor/Zend/Validate/CreditCard.php
@@ -99,7 +99,7 @@ class Zend_Validate_CreditCard extends Zend_Validate_Abstract
      */
     protected $_cardType = array(
         self::AMERICAN_EXPRESS => array('34', '37'),
-        self::DINERS_CLUB      => array('300', '301', '302', '303', '304', '305', '36'),
+        self::DINERS_CLUB      => array('300', '301', '302', '303', '304', '305', '36', '38'),
         self::DINERS_CLUB_US   => array('54', '55'),
         self::DISCOVER         => array('6011', '622126', '622127', '622128', '622129', '62213',
                                         '62214', '62215', '62216', '62217', '62218', '62219',
