Index: Log.php
===================================================================
--- Log.php	(revision 929)
+++ Log.php	(working copy)
@@ -204,6 +204,11 @@
             throw new Zend_Log_Exception("Cannot register, \"$logName\" already exists.");
         }
 
+        //make sure we have a default logger
+        if(sizeof(self::$_instances == 1)) {
+            self::setDefaultLogger($logName);
+        }
+
         return true;
     }
 
