Index: library/Zend/Form/Decorator/Fieldset.php
===================================================================
--- library/Zend/Form/Decorator/Fieldset.php	(revision 25022)
+++ library/Zend/Form/Decorator/Fieldset.php	(working copy)
@@ -133,7 +133,7 @@
         $id      = (string)$element->getId();
 
         if ((!array_key_exists('id', $attribs) || $attribs['id'] == $id) && '' !== $id) {
-            $attribs['id'] = 'fieldset-' . $id;
+            $attribs['id'] = $id . '-fieldset';
         }
 
         if (null !== $legend) {
