Index: library/Zend/View/Helper/HeadMeta.php
===================================================================
--- library/Zend/View/Helper/HeadMeta.php	(revision 13992)
+++ library/Zend/View/Helper/HeadMeta.php	(working copy)
@@ -329,6 +329,7 @@
                 : $this->getIndent();
 
         $items = array();
+        $this->getContainer()->ksort();
         foreach ($this as $item) {
             $items[] = $this->itemToString($item);
         }
Index: library/Zend/View/Helper/HeadStyle.php
===================================================================
--- library/Zend/View/Helper/HeadStyle.php	(revision 13992)
+++ library/Zend/View/Helper/HeadStyle.php	(working copy)
@@ -362,6 +362,7 @@
                 : $this->getIndent();
 
         $items = array();
+        $this->getContainer()->ksort();
         foreach ($this as $item) {
             if (!$this->_isValid($item)) {
                 continue;
Index: library/Zend/View/Helper/HeadScript.php
===================================================================
--- library/Zend/View/Helper/HeadScript.php	(revision 13992)
+++ library/Zend/View/Helper/HeadScript.php	(working copy)
@@ -446,6 +446,7 @@
         $escapeEnd   = ($useCdata) ? '//]]>'       : '//-->';
 
         $items = array();
+        $this->getContainer()->ksort();
         foreach ($this as $item) {
             if (!$this->_isValid($item)) {
                 continue;
Index: library/Zend/View/Helper/HeadLink.php
===================================================================
--- library/Zend/View/Helper/HeadLink.php	(revision 13992)
+++ library/Zend/View/Helper/HeadLink.php	(working copy)
@@ -307,6 +307,7 @@
                 : $this->getIndent();
 
         $items = array();
+        $this->getContainer()->ksort();
         foreach ($this as $item) {
             $items[] = $this->itemToString($item);
         }
Index: tests/Zend/View/Helper/HeadMetaTest.php
===================================================================
--- tests/Zend/View/Helper/HeadMetaTest.php	(revision 13992)
+++ tests/Zend/View/Helper/HeadMetaTest.php	(working copy)
@@ -372,6 +372,24 @@
             );
     }
 
+    public function testZF5435()
+    {
+
+        $this->helper->offsetSetName(1, 'keywords', 'foo');
+        $this->helper->offsetSetName(10, 'description', 'foo');
+        $this->helper->offsetSetHttpEquiv(20, 'pragma', 'baz');
+        $this->helper->offsetSetHttpEquiv(5, 'Cache-control', 'baz');
+
+        $test = $this->helper->toString();
+
+        $expected = '<meta name="keywords" content="foo" />
+<meta http-equiv="Cache-control" content="baz" />
+<meta name="description" content="foo" />
+<meta http-equiv="pragma" content="baz" />';
+
+        $this->assertEquals($expected, $test);
+    }
+
 }
 
 // Call Zend_View_Helper_HeadMetaTest::main() if this source file is executed directly.
Index: tests/Zend/View/Helper/HeadStyleTest.php
===================================================================
--- tests/Zend/View/Helper/HeadStyleTest.php	(revision 13992)
+++ tests/Zend/View/Helper/HeadStyleTest.php	(working copy)
@@ -385,7 +385,42 @@
         $test = $this->helper->toString();
         $this->assertContains('<!--[if lt IE 7]>', $test);
     }
+
+    public function testZF5435()
+    {
+
+        $this->helper->offsetSetStyle(10, '
+a {
+    display: none;
+}');
+        $this->helper->offsetSetStyle(5, '
+h1 {
+    font-weight: bold
+}');
+
+
+        $test = $this->helper->toString();
+
+        $expected = '<style type="text/css" media="screen">
+<!--
+
+h1 {
+    font-weight: bold
 }
+-->
+</style>
+<style type="text/css" media="screen">
+<!--
+
+a {
+    display: none;
+}
+-->
+</style>';
+
+        $this->assertEquals($expected, $test);
+    }
+}
 
 // Call Zend_View_Helper_HeadStyleTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == "Zend_View_Helper_HeadStyleTest::main") {
Index: tests/Zend/View/Helper/HeadScriptTest.php
===================================================================
--- tests/Zend/View/Helper/HeadScriptTest.php	(revision 13992)
+++ tests/Zend/View/Helper/HeadScriptTest.php	(working copy)
@@ -398,6 +398,25 @@
         $test = $this->helper->headScript()->toString();
         $this->assertContains('<!--[if lt IE 7]>', $test);
     }
+
+    public function testZF5435()
+    {
+
+        $this->helper->offsetSetFile(1, 'test1.js');
+        $this->helper->offsetSetFile(20, 'test2.js');
+        $this->helper->offsetSetFile(10, 'test3.js');
+        $this->helper->offsetSetFile(5, 'test4.js');
+
+
+        $test = $this->helper->toString();
+
+        $expected = '<script type="text/javascript" src="test1.js"></script>
+<script type="text/javascript" src="test4.js"></script>
+<script type="text/javascript" src="test3.js"></script>
+<script type="text/javascript" src="test2.js"></script>';
+
+        $this->assertEquals($expected, $test);
+    }
 }
 
 // Call Zend_View_Helper_HeadScriptTest::main() if this source file is executed directly.
Index: tests/Zend/View/Helper/HeadLinkTest.php
===================================================================
--- tests/Zend/View/Helper/HeadLinkTest.php	(revision 13992)
+++ tests/Zend/View/Helper/HeadLinkTest.php	(working copy)
@@ -418,6 +418,22 @@
         $test = $this->helper->toString();
         $this->assertContains(' media="screen,print"', $test);
     }
+    public function testZF5435()
+    {
+        $this->helper->headLink()->offsetSetStylesheet(1,'/test1.css');
+        $this->helper->headLink()->offsetSetStylesheet(10,'/test2.css');
+        $this->helper->headLink()->offsetSetStylesheet(20,'/test3.css');
+        $this->helper->headLink()->offsetSetStylesheet(5,'/test4.css');
+
+        $test = $this->helper->toString();
+
+        $expected = '<link href="/test1.css" media="screen" rel="stylesheet" type="text/css" >
+<link href="/test4.css" media="screen" rel="stylesheet" type="text/css" >
+<link href="/test2.css" media="screen" rel="stylesheet" type="text/css" >
+<link href="/test3.css" media="screen" rel="stylesheet" type="text/css" >';
+
+        $this->assertEquals($expected, $test);
+    }
 }
 
 // Call Zend_View_Helper_HeadLinkTest::main() if this source file is executed directly.
