Index: library/Zend/Mail/Protocol/Pop3.php
===================================================================
--- library/Zend/Mail/Protocol/Pop3.php	(revision 6076)
+++ library/Zend/Mail/Protocol/Pop3.php	(working copy)
@@ -349,7 +349,7 @@
     {
         if ($this->hasTop === false) {
             if ($fallback) {
-                return $this->retrive($msgno);
+                return $this->retrieve($msgno);
             } else {
                 throw new Zend_Mail_Protocol_Exception('top not supported and no fallback wanted');
             }
@@ -363,7 +363,7 @@
         } catch (Zend_Mail_Protocol_Exception $e) {
             $this->hasTop = false;
             if ($fallback) {
-                $result = $this->retrive($msgno);
+                $result = $this->retrieve($msgno);
             } else {
                 throw $e;
             }
@@ -380,7 +380,7 @@
      * @return string message
      * @throws Zend_Mail_Protocol_Exception
      */
-    public function retrive($msgno)
+    public function retrieve($msgno)
     {
         $result = $this->request("RETR $msgno", true);
         return $result;
