--- Server.php	2007-11-13 16:50:53.000000000 -0500
+++ Server_new2.php	2008-01-31 18:08:54.000000000 -0500
@@ -136,6 +136,9 @@
                         if (isset($request[strtolower($arg->getName())])) {
                             $calling_args[] = $request[strtolower($arg->getName())];
                         }
+                        else if ($arg->isOptional()) {
+                            $calling_args[] = $arg->getDefaultValue();
+                        }
                     }
 
                     foreach ($request as $key => $value) {
