Index: tests/Zend/Validate/PostCodeTest.php
===================================================================
--- tests/Zend/Validate/PostCodeTest.php	(revision 22484)
+++ tests/Zend/Validate/PostCodeTest.php	(working copy)
@@ -131,7 +131,7 @@
             $this->_validator->setLocale('gez_ER');
             $this->fail();
         } catch (Zend_Validate_Exception $e) {
-            $this->assertContains('Unable to detect a format', $e->getMessage());
+            $this->assertContains('Unable to detect a postcode', $e->getMessage());
         }
     }
 
@@ -164,14 +164,14 @@
             $this->_validator->setFormat(null);
             $this->fail();
         } catch (Zend_Validate_Exception $e) {
-            $this->assertContains('a not empty string', $e->getMessage());
+            $this->assertContains('string has to be given for validation', $e->getMessage());
         }
 
         try {
             $this->_validator->setFormat('');
             $this->fail();
         } catch (Zend_Validate_Exception $e) {
-            $this->assertContains('a not empty string', $e->getMessage());
+            $this->assertContains('string has to be given for validation', $e->getMessage());
         }
     }
 
@@ -182,7 +182,7 @@
     {
         $this->assertFalse($this->_validator->isValid('hello'));
         $message = $this->_validator->getMessages();
-        $this->assertContains('not appear to be an postal code', $message['postcodeNoMatch']);
+        $this->assertContains('not appear to be a postal code', $message['postcodeNoMatch']);
     }
 }
 
