Index: Abstract.php
===================================================================
--- Abstract.php	(revision 19044)
+++ Abstract.php	(working copy)
@@ -902,7 +902,7 @@
             return str_replace('?', $this->quote($value, $type), $text);
         } else {
             while ($count > 0) {
-                if (strpos($text, '?') != false) {
+                if (strpos($text, '?') !== false) {
                     $text = substr_replace($text, $this->quote($value, $type), strpos($text, '?'), 1);
                 }
                 --$count;
