Index: CodeGenerator/Php/Parameter.php
===================================================================
--- CodeGenerator/Php/Parameter.php	(revision 11)
+++ CodeGenerator/Php/Parameter.php	(working copy)
@@ -79,7 +79,7 @@
         } else {
             $typeClass = $reflectionParameter->getClass();
             if($typeClass !== null) {
-                $param->setType($typeClass->getName());
+                $param->setType(is_string($typeClass)?$typeClass:$typeClass->getName());
             }
         }
         
Index: Reflection/Parameter.php
===================================================================
--- Reflection/Parameter.php	(revision 11)
+++ Reflection/Parameter.php	(working copy)
@@ -58,7 +58,19 @@
      */
     public function getClass($reflectionClass = 'Zend_Reflection_Class')
     {
-        $phpReflection  = parent::getClass();
+        try {
+            $phpReflection  = parent::getClass();
+        } catch(ReflectionException $e) {
+            preg_match_all('|Class(.*)does not exist|U', $e->getMessage(), $matches);
+            if (isset($matches[1]) && isset($matches[1][0])) {
+                $phpReflection = trim($matches[1][0]);    
+            }
+            if (!is_string($phpReflection)) {
+                throw $e;
+            } else {
+                return $phpReflection;
+            }
+        }
         if($phpReflection == null) {
             return null;
         }
