Index: library/Zend/Filter/HtmlEntities.php
===================================================================
--- library/Zend/Filter/HtmlEntities.php	(revision 20756)
+++ library/Zend/Filter/HtmlEntities.php	(working copy)
@@ -62,7 +62,9 @@
      */
     public function __construct($options = array())
     {
-        if (!is_array($options)) {
+    	if ($options instanceof Zend_Config) {
+    		$options = $options->toArray(); 	
+    	} elseif (!is_array($options)) {
             $options = func_get_args();
             $temp['quotestyle'] = array_shift($options);
             if (!empty($options)) {
