--- tests/Zend/View/Helper/HeadLinkTest.php	(revision 23226)
+++ tests/Zend/View/Helper/HeadLinkTest.php	(working copy)
@@ -463,6 +463,15 @@
 
         $this->assertEquals($expected, $test);
     }
+
+    /**
+     * @issue ZF-10345
+     */
+    public function testIdAttributeIsSupported()
+    {
+        $this->helper->appendStylesheet(array('href' => '/bar/baz', 'id' => 'foo'));
+        $this->assertContains('id="foo"', $this->helper->toString());
+    }
 }
 
 // Call Zend_View_Helper_HeadLinkTest::main() if this source file is executed directly.
Index: library/Zend/View/Helper/HeadLink.php
===================================================================
--- library/Zend/View/Helper/HeadLink.php	(revision 23226)
+++ library/Zend/View/Helper/HeadLink.php	(working copy)
@@ -40,7 +40,7 @@
      *
      * @var array
      */
-    protected $_itemKeys = array('charset', 'href', 'hreflang', 'media', 'rel', 'rev', 'type', 'title', 'extras');
+    protected $_itemKeys = array('charset', 'href', 'hreflang', 'id', 'media', 'rel', 'rev', 'type', 'title', 'extras');
 
     /**
      * @var string registry key
