Index: Http.php
===================================================================
--- Http.php	(revision 12360)
+++ Http.php	(working copy)
@@ -553,6 +553,9 @@
                 $basePath = $baseUrl;
             }
         }
+        if (substr(PHP_OS, 0, 3) === 'WIN') {
+            $basePath = rtrim($basePath, '\\');
+        }
 
         $this->_basePath = rtrim($basePath, '/');
         return $this;
