Index: Zend/Form/Element/File.php
===================================================================
--- Zend/Form/Element/File.php	(revision 11209)
+++ Zend/Form/Element/File.php	(working copy)
@@ -258,9 +258,9 @@
         $adapter = $this->getTransferAdapter();
         $this->setValue($adapter->getFileName($this->getName()));
 
-        if (!$this->isRequired() && empty($value)) {
-            return true;
-        }
+//        if (!$this->isRequired() && empty($value)) {
+//            return true;
+//        }
 
         if ($this->isRequired() 
             && $this->autoInsertNotEmptyValidator() 
@@ -272,9 +272,15 @@
             $this->setValidators($validators);
         }
 
-        if($adapter->receive($value)) {
+        if($adapter->receive($this->getName())) {
+            return true;
+        }
+        
+        if (!$this->isRequired() && $adapter->getErrors() == array(Zend_Validate_File_Upload::NO_FILE)){
+            $adapter->clearErrors();
             return true;
         }
+        
         return false;
     }
