Index: Date.php
===================================================================
--- Date.php	(revision xxx)
+++ Date.php	(working copy)
@@ -2544,7 +2544,7 @@
                             $part .= Zend_Locale_Format::getTimeFormat($locale);
                         }
                         $parsed = Zend_Locale_Format::getDate($date, array('date_format' => $part, 'locale' => $locale, 'fix_date' => true, 'format_type' => 'iso'));
-                        if ((strpos(strtoupper($part), 'YY') !== false) and (strpos(strtoupper($part), 'YYYY') === false)) {
+                        if ((strpos(strtoupper($part), 'YY') !== false) and (strpos(strtoupper($part), 'YYYY') === false or strlen($parsed['year']) < 4)) {
                             $parsed['year'] = self::getFullYear($parsed['year']);
                         }
                         if (($calc == 'set') || ($calc == 'cmp')) {
