Index: /library/Zend/Form/Decorator/Abstract.php
===================================================================
--- /library/Zend/Form/Decorator/Abstract.php	(revision 14151)
+++ /library/Zend/Form/Decorator/Abstract.php	(working copy)
@@ -240,6 +240,18 @@
     }
 
     /**
+     * Set separator
+     *
+     * @param mixed $separator
+     * @return self
+     */
+    public function setSeparator($separator)
+    {
+        $this->_separator = $separator;
+        return $this;
+    }
+
+    /**
      * Decorate content and/or element
      * 
      * @param  string $content
