Index: Oci.php
===================================================================
--- Oci.php	(revision 6628)
+++ Oci.php	(working copy)
@@ -78,20 +78,20 @@
         // baseline of DSN parts
         $dsn = $this->_config;
 
-        $tns = 'dbname=';
+        $tns = 'dbname=(DESCRIPTION=';
         if (isset($dsn['host'])) {
-            $tns .= '//' . $dsn['host'];
+            $tns .= '(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)(HOST=' . $dsn['host'] . ')';
             if (isset($dsn['port'])) {
-                $tns .= ':' . $dsn['port'];
+                $tns .= '(PORT=' . $dsn['port'] . ')';
             }
-            $tns .= '/';
+            $tns .= '))';
         }
-        $tns .= $dsn['dbname'];
+        $tns .= '(CONNECT_DATA=(SID=' . $dsn['dbname'] . ')))';
 
         if (isset($dsn['charset']))
         {
             $tns .= ';charset=' . $dsn['charset'];
-        }       
+        }
 
         return $this->_pdoType . ':' . $tns;
     }
