Index: Http.php
===================================================================
--- Http.php	(.../branches/release-1.5/library/Zend/Auth/Adapter/Http.php)	(revision 10487)
+++ Http.php	(.../trunk/library/Zend/Auth/Adapter/Http.php)	(revision 10487)
@@ -394,7 +394,7 @@
             return new Zend_Auth_Result(
                 Zend_Auth_Result::FAILURE_UNCATEGORIZED,
                 array(),
-                array('Client requested an unsupported authentication scheme')
+                array('Client requested an incorrect or unsupported authentication scheme')
             );
         }
 
@@ -410,13 +410,13 @@
                 break;
             case 'digest':
                 $result = $this->_digestAuth($authHeader);
-                break;
+            break;
             default:
                 /**
-                 * @see Zend_Auth_Adapter_Exception
-                 */
-                require_once 'Zend/Auth/Adapter/Exception.php';
-                throw new Zend_Auth_Adapter_Exception('Unsupported authentication scheme');
+                * @see Zend_Auth_Adapter_Exception
+                */
+ 	            require_once 'Zend/Auth/Adapter/Exception.php';
+ 	            throw new Zend_Auth_Adapter_Exception('Unsupported authentication scheme');
         }
 
         return $result;
