Index: library/Zend/Mail.php
===================================================================
--- library/Zend/Mail.php	(revision 21568)
+++ library/Zend/Mail.php	(working copy)
@@ -77,7 +77,7 @@
      * Mail character set
      * @var string
      */
-    protected $_charset = null;
+    protected $_charset = 'iso-8859-1';
 
     /**
      * Mail headers
@@ -210,9 +210,11 @@
      *
      * @param string $charset
      */
-    public function __construct($charset = 'iso-8859-1')
+    public function __construct($charset = null)
     {
-        $this->_charset = $charset;
+        if (!is_null($charset)) {
+            $this->setCharset($charset);
+        }
     }
 
     /**
@@ -226,6 +228,18 @@
     }
 
     /**
+     * Set charset string
+     *
+     * @param  string $charset
+     * @return Zend_Mail Provides a fluent interface
+     */
+    public function setCharset($charset)
+    {
+        $this->_charset = (string) $charset;
+        return $this;
+    }
+
+    /**
      * Set content type
      *
      * Should only be used for manually setting multipart content types.

