--- Abstract.php.old	2009-12-15 10:06:55.000000000 +0000
+++ Abstract.php	2009-12-15 10:39:44.037582341 +0000
@@ -138,6 +138,13 @@
      * @var boolean
      */
     private $_strictVars = false;
+    
+    /**
+     * Corresponds to the forth htmlentities() argument
+     *
+     * @var boolean
+     */
+    private $_doubleEncode = true;
 
     /**
      * Constructor.
@@ -162,6 +169,11 @@
             $this->setEncoding($config['encoding']);
         }
 
+        // double encode
+        if (array_key_exists('doubleencode', $config)) {
+            $this->setDoubleEncode($config['doubleencode']);
+        }
+
         // base path
         if (array_key_exists('basePath', $config)) {
             $prefix = 'Zend_View';
@@ -845,7 +857,7 @@
     public function escape($var)
     {
         if (in_array($this->_escape, array('htmlspecialchars', 'htmlentities'))) {
-            return call_user_func($this->_escape, $var, ENT_COMPAT, $this->_encoding);
+            return call_user_func($this->_escape, $var, ENT_COMPAT, $this->_encoding, $this->_doubleEncode);
         }
 
         return call_user_func($this->_escape, $var);
@@ -874,6 +886,28 @@
     }
 
     /**
+     * Returns the doubleEncode option
+     *
+     * @return boolean
+     */
+    public function getDoubleEncode()
+    {
+        return $this->_doubleEncode;
+    }
+
+    /**
+     * Sets the doubleEncode option
+     *
+     * @param boolean $doubleEncode
+     * @return Zend_View_Abstract
+     */
+    public function setDoubleEncode($doubleEncode)
+    {
+        $this->_doubleEncode = (boolean) $doubleEncode;
+        return $this;
+    }
+
+    /**
      * Enable or disable strict vars
      *
      * If strict variables are enabled, {@link __get()} will raise a notice
