Index: RedirectorTest.php
===================================================================
--- RedirectorTest.php	(revision 14153)
+++ RedirectorTest.php	(working copy)
@@ -161,6 +161,13 @@
         $this->assertFalse($this->redirector->getPrependBase());
     }
 
+    public function testCloseSessionOnExit()
+    {
+        $this->assertTrue($this->redirector->getCloseSessionOnExit());
+        $this->redirector->setCloseSessionOnExit(false);
+        $this->assertFalse($this->redirector->getCloseSessionOnExit());
+    }
+
     public function testGetRedirectUrlNullByDefault()
     {
         $this->assertNull($this->redirector->getRedirectUrl());
