--- library/Zend/Pdf/FileParser/Font/OpenType/TrueType.php	(revision 20335)
+++ library/Zend/Pdf/FileParser/Font/OpenType/TrueType.php	(working copy)
@@ -58,6 +58,9 @@
 
             case 0x74727565:    // 'true' - Macintosh TrueType signature
                 break;
+				
+			case 0x4f54544f:    // OpenType CFF (PostScript outlines)
+				break;					
 
             default:
                 require_once 'Zend/Pdf/Exception.php';
