--- Amf-1.9.6/Response.php	2009-11-12 10:26:19.000000000 -0600
+++ Amf/Response.php	2010-01-05 00:47:41.000000000 -0600
@@ -105,11 +105,12 @@
             $stream->writeUTF($body->getTargetURI());
             $stream->writeUTF($body->getResponseURI());
             $stream->writeLong(Zend_Amf_Constants::UNKNOWN_CONTENT_LENGTH);
+            $bodyData = $body->getData();
             if($this->_objectEncoding == Zend_Amf_Constants::AMF0_OBJECT_ENCODING) {
-                $serializer->writeTypeMarker($body->getData());
+                $serializer->writeTypeMarker($bodyData);
             } else {
                 // Content is AMF3
-                $serializer->writeTypeMarker($body->getData(),Zend_Amf_Constants::AMF0_AMF3);
+                $serializer->writeTypeMarker($bodyData,Zend_Amf_Constants::AMF0_AMF3);
             }
         }
 
