Index: library/Zend/Mime.php
===================================================================
--- library/Zend/Mime.php	(revision 7623)
+++ library/Zend/Mime.php	(working copy)
@@ -196,16 +196,17 @@
      *
      * @param string $str
      * @param string $encoding
+     * @param string $EOL EOL string; defaults to {@link Zend_Mime::LINEEND}
      * @return string
      */
-    public static function encode($str, $encoding)
+    public static function encode($str, $encoding, $EOL = self::LINEEND)
     {
         switch ($encoding) {
             case self::ENCODING_BASE64:
-                return self::encodeBase64($str);
+                return self::encodeBase64($str, self::LINELENGTH, $EOL);
 
             case self::ENCODING_QUOTEDPRINTABLE:
-                return self::encodeQuotedPrintable($str);
+                return self::encodeQuotedPrintable($str, self::LINELENGTH, $EOL);
 
             default:
                 /**
