Index: library/Zend/Cache/Manager.php
===================================================================
--- library/Zend/Cache/Manager.php	(revision 21139)
+++ library/Zend/Cache/Manager.php	(working copy)
@@ -168,7 +168,10 @@
                 $this->_optionTemplates[$name]['frontend']['name'],
                 $this->_optionTemplates[$name]['backend']['name'],
                 isset($this->_optionTemplates[$name]['frontend']['options']) ? $this->_optionTemplates[$name]['frontend']['options'] : array(),
-                isset($this->_optionTemplates[$name]['backend']['options']) ? $this->_optionTemplates[$name]['backend']['options'] : array()
+                isset($this->_optionTemplates[$name]['backend']['options']) ? $this->_optionTemplates[$name]['backend']['options'] : array(),
+                isset($this->_optionTemplates[$name]['frontend']['custom_naming']) ? $this->_optionTemplates[$name]['frontend']['custom_naming'] : false,
+                isset($this->_optionTemplates[$name]['backend']['custom_naming']) ? $this->_optionTemplates[$name]['backend']['custom_naming'] : false,
+                isset($this->_optionTemplates[$name]['autoload']) ? $this->_optionTemplates[$name]['autoload'] : false
             );
             return $this->_caches[$name];
         }
