Index: Translate/Adapter/Qt.php
===================================================================
--- Translate/Adapter/Qt.php	(revision 8)
+++ Translate/Adapter/Qt.php	(working copy)
@@ -104,7 +104,9 @@
     {
         switch(strtolower($name)) {
             case 'ts':
-                $this->_translate[$this->_target] = array();
+                if (!array_key_exists($this->_target, $this->_translate)) {
+                    $this->_translate[$this->_target] = array();
+                }
                 break;
             case 'message':
                 $this->_source = null;
