--- 1.7.0/Zend/Amf/Parse/Amf3/Deserializer.php~	2008-11-21 11:52:03.000000000 +0000
+++ 1.7.0/Zend/Amf/Parse/Amf3/Deserializer.php	2008-11-21 11:52:03.000000000 +0000
@@ -166,7 +166,7 @@
      * Strings are encoding using UTF-8 - however the header may either
      * describe a string literal or a string reference.
      *
-     * - string = 0×06 string-data
+     * - string = 0(0xd7)06 string-data
      * - string-data = integer-data [ modified-utf-8 ]
      * - modified-utf-8 = *OCTET
      *
@@ -205,7 +205,7 @@
      * of midnight, 1st Jan 1970 in the UTC time zone.
      * Local time zone information is not sent to flash.
      *
-     * - date = 0×08 integer-data [ number-data ]
+     * - date = 0(0xd7)08 integer-data [ number-data ]
      *
      * @return Zend_Date
      */
@@ -232,7 +232,7 @@
     /**
      * Read amf array to PHP array
      *
-     * - array = 0×09 integer-data ( [ 1OCTET *amf3-data ] | [OCTET *amf3-data 1] | [ OCTET *amf-data ] )
+     * - array = 0(0xd7)09 integer-data ( [ 1OCTET *amf3-data ] | [OCTET *amf3-data 1] | [ OCTET *amf-data ] )
      *
      * @return array
      */
