Index: library/Zend/Session/Abstract.php
===================================================================
--- library/Zend/Session/Abstract.php	(revision 174)
+++ library/Zend/Session/Abstract.php	(working copy)
@@ -120,7 +120,7 @@
             unset(self::$_expiringData[$namespace]);
         } else {
             unset($_SESSION[$namespace][$name]);
-            unset(self::$_expiringData[$namespace]);
+            unset(self::$_expiringData[$namespace][$name]);
         }
 
         // if we remove the last value, remove namespace.
