Index: Abstract.php
===================================================================
--- Abstract.php	(Revision 3830)
+++ Abstract.php	(Arbeitskopie)
@@ -283,9 +283,9 @@
 
             // Get return type(s) and description
             $return     = 'void';
-            if (preg_match('/@return ([^\s]*)/', $docBlock, $matches)) {
+            if (preg_match('/@return +([^\s]*)/', $docBlock, $matches)) {
                 $return = explode('|', $matches[1]);
-                if (preg_match('/@return [^\s]*\s+(.*?)(@|\*\/)/s', $docBlock, $matches))
+                if (preg_match('/@return +[^\s]*\s+(.*?)(@|\*\/)/s', $docBlock, $matches))
                 {
                     $value = $matches[1];
                     $value = preg_replace('/\s?\* /m', '', $value);
@@ -295,9 +295,9 @@
             }
 
             // Get param types and description
-            if (preg_match_all('/@param ([^ ]*) /', $docBlock, $matches)) {
+            if (preg_match_all('/@param +([^ ]*) /', $docBlock, $matches)) {
                 $paramTypesTmp = $matches[1];
-                if (preg_match_all('/@param [^\s]*\s+(\$[^ ]*)\s+(.*?)(@|\*\/)/s', $docBlock, $matches))
+                if (preg_match_all('/@param +[^\s]*\s+(\$[^ ]*)\s+(.*?)(@|\*\/)/s', $docBlock, $matches))
                 {
                     $paramDesc = $matches[2];
                     foreach ($paramDesc as $key => $value) {
