Index: tests/Zend/Uri/HttpTest.php
===================================================================
--- tests/Zend/Uri/HttpTest.php	(revision 23530)
+++ tests/Zend/Uri/HttpTest.php	(working copy)
@@ -430,4 +430,45 @@
         ), $uri->getQueryAsArray());
         $this->assertEquals('a=1&c=3', $uri->getQuery());
     }
+    
+    
+    /**
+     * Test that URIs using HTTP scheme on port 80 omit port from toString
+     * @group ZF-10812
+     */
+    public function testShouldNotReturnPortWhenSchemeIsHttpAndPortIs80()
+    {
+        $uri = Zend_Uri::factory('http://example.com:80', 'Zend_Uri_Http');
+        $this->assertNotContains(':80', (string)$uri);
+    }
+    
+    /**
+     * Test that URIs using HTTPS scheme on port 443 omit port from toString
+     * @group ZF-10812
+     */
+    public function testShouldNotReturnPortWhenSchemeIsHttpsAndPortIs443()
+    {
+        $uri = Zend_Uri::factory('https://example.com:443', 'Zend_Uri_Http');
+        $this->assertNotContains(':443', (string)$uri);
+    }
+    
+    /**
+     * Test that URIs using HTTP scheme on port != 80 includes port in toString
+     * @group ZF-10812
+     */
+    public function testShouldReturnPortWhenSchemeIsHttpAndPortIsNot80()
+    {
+        $uri = Zend_Uri::factory('http://example.com:8080', 'Zend_Uri_Http');
+        $this->assertContains(':8080', (string)$uri);
+    }
+    
+    /**
+     * Test that URIs using HTTPS scheme on != 443 includes port in toString
+     * @group ZF-10812
+     */
+    public function testShouldReturnPortWhenSchemeIsHttpsAndPortIsNot443()
+    {
+        $uri = Zend_Uri::factory('https://example.com:8080', 'Zend_Uri_Http');
+        $this->assertContains(':8080', (string)$uri);
+    }   
 }
\ No newline at end of file
Index: library/Zend/Uri/Http.php
===================================================================
--- library/Zend/Uri/Http.php	(revision 23530)
+++ library/Zend/Uri/Http.php	(working copy)
@@ -253,7 +253,7 @@
 
         $password = strlen($this->_password) > 0 ? ":$this->_password" : '';
         $auth     = strlen($this->_username) > 0 ? "$this->_username$password@" : '';
-        $port     = strlen($this->_port) > 0 ? ":$this->_port" : '';
+        $port     = strlen($this->_port) > 0 && !('http' == $this->_scheme && 80 == $this->_port) && !('https' == $this->_scheme && 443 == $this->_port) ? ":{$this->_port}" : '';
         $query    = strlen($this->_query) > 0 ? "?$this->_query" : '';
         $fragment = strlen($this->_fragment) > 0 ? "#$this->_fragment" : '';
 
