Index: library/Zend/Search/Lucene/Interface/MultiSearcher.php
===================================================================
--- library/Zend/Search/Lucene/Interface/MultiSearcher.php	(revision 0)
+++ library/Zend/Search/Lucene/Interface/MultiSearcher.php	(revision 0)
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Search_Lucene
+ * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id: MultiSearcher.php 24593 2012-01-05 20:35:02Z matthew $
+ */
+
+
+/** Zend_Search_Lucene_MultiSearcher */
+require_once 'Zend/Search/Lucene/MultiSearcher.php';
+
Index: library/Zend/Search/Lucene/MultiSearcher.php
===================================================================
--- library/Zend/Search/Lucene/MultiSearcher.php	(revision 24794)
+++ library/Zend/Search/Lucene/MultiSearcher.php	(working copy)
@@ -24,6 +24,12 @@
 require_once 'Zend/Search/Lucene/Interface.php';
 
 /**
+ * Import Zend_Search_Lucene_Interface_MultiSearcher for BC (see ZF-12067)
+ * @see Zend_Search_Lucene_Interface_MultiSearcher 
+ */
+require_once 'Zend/Search/Lucene/Interface/MultiSearcher.php';
+
+/**
  * Multisearcher allows to search through several independent indexes.
  *
  * @category   Zend
@@ -31,7 +37,7 @@
  * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Search_Lucene_Interface_MultiSearcher implements Zend_Search_Lucene_Interface
+class Zend_Search_Lucene_MultiSearcher implements Zend_Search_Lucene_Interface
 {
     /**
      * List of indices for searching.
@@ -971,3 +977,16 @@
         // Do nothing, since it's never referenced by indices
     }
 }
+
+/**
+ * This class is provided for backwards-compatibility (See ZF-12067)
+ *
+ * @category   Zend
+ * @package    Zend_Search_Lucene
+ * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_Search_Lucene_Interface_MultiSearcher
+    extends Zend_Search_Lucene_MultiSearcher
+{
+}
