Index: library/Zend/Controller/Action/Helper/FlashMessenger.php
===================================================================
--- library/Zend/Controller/Action/Helper/FlashMessenger.php	(revision 1270)
+++ library/Zend/Controller/Action/Helper/FlashMessenger.php	(revision 1270)
@@ -130,7 +130,9 @@
             self::$_session->{$this->_namespace} = array();
         }
         
-        self::$_session->{$this->_namespace}[] = $message;
+        $sessionMessages = self::$_session->{$this->_namespace};
+        $sessionMessages[] = $message;
+        self::$_session->{$this->_namespace} = $sessionMessages;
         
         return;
     }
