Index: /media/www/ZF/trunk/library/Zend/Loader/Autoloader.php
===================================================================
--- /media/www/ZF/trunk/library/Zend/Loader/Autoloader.php	(révision 22477)
+++ /media/www/ZF/trunk/library/Zend/Loader/Autoloader.php	(copie de travail)
@@ -120,16 +120,16 @@
                 if ($autoloader->autoload($class)) {
                     return true;
                 }
-            } elseif (is_string($autoloader)) {
-                if ($autoloader($class)) {
-                    return true;
-                }
             } elseif (is_array($autoloader)) {
                 $object = array_shift($autoloader);
                 $method = array_shift($autoloader);
                 if (call_user_func(array($object, $method), $class)) {
                     return true;
                 }
+            } elseif (is_string($autoloader) || is_callable($autoloader)) {
+                if ($autoloader($class)) {
+                    return true;
+                }
             }
         }
 
