--- tests/Zend/Validate/EmailAddressTest.php	(revision 23237)
+++ tests/Zend/Validate/EmailAddressTest.php	(working copy)
@@ -156,7 +156,7 @@
         $this->assertContains('quoted-string', current($messages));
 
         $this->assertContains('Some User', next($messages));
-        $this->assertContains('no valid local part', current($messages));
+        $this->assertContains('not a valid local part', current($messages));
     }
 
     /**
@@ -184,7 +184,7 @@
         $this->assertFalse($this->_validator->isValid('username@ example . com'));
         $messages = $this->_validator->getMessages();
         $this->assertThat(count($messages), $this->greaterThanOrEqual(1));
-        $this->assertContains('no valid hostname', current($messages));
+        $this->assertContains('not a valid hostname', current($messages));
     }
 
     /**
@@ -219,7 +219,7 @@
         $this->assertFalse($this->_validator->isValid('User Name <username@example.com>'));
         $messages = $this->_validator->getMessages();
         $this->assertThat(count($messages), $this->greaterThanOrEqual(3));
-        $this->assertContains('no valid hostname', current($messages));
+        $this->assertContains('not a valid hostname', current($messages));
         $this->assertContains('cannot match TLD', next($messages));
         $this->assertContains('does not appear to be a valid local network name', next($messages));
     }
Index: tests/Zend/Validate/Sitemap/LocTest.php
===================================================================
--- tests/Zend/Validate/Sitemap/LocTest.php	(revision 23237)
+++ tests/Zend/Validate/Sitemap/LocTest.php	(working copy)
@@ -96,7 +96,7 @@
         foreach ($values as $value) {
             $this->assertSame(false, $this->_validator->isValid($value));
             $messages = $this->_validator->getMessages();
-            $this->assertContains('is no valid', current($messages));
+            $this->assertContains('is not a valid', current($messages));
         }
     }
 
Index: tests/Zend/Validate/Sitemap/PriorityTest.php
===================================================================
--- tests/Zend/Validate/Sitemap/PriorityTest.php	(revision 23237)
+++ tests/Zend/Validate/Sitemap/PriorityTest.php	(working copy)
@@ -88,7 +88,7 @@
         foreach ($values as $value) {
             $this->assertSame(false, $this->_validator->isValid($value));
             $messages = $this->_validator->getMessages();
-            $this->assertContains('is no valid', current($messages));
+            $this->assertContains('is not a valid', current($messages));
         }
     }
 
Index: tests/Zend/Validate/Sitemap/ChangefreqTest.php
===================================================================
--- tests/Zend/Validate/Sitemap/ChangefreqTest.php	(revision 23237)
+++ tests/Zend/Validate/Sitemap/ChangefreqTest.php	(working copy)
@@ -89,7 +89,7 @@
         foreach ($values as $value) {
             $this->assertSame(false, $this->_validator->isValid($value));
             $messages = $this->_validator->getMessages();
-            $this->assertContains('is no valid', current($messages));
+            $this->assertContains('is not a valid', current($messages));
         }
     }
 
Index: tests/Zend/Validate/Sitemap/LastmodTest.php
===================================================================
--- tests/Zend/Validate/Sitemap/LastmodTest.php	(revision 23237)
+++ tests/Zend/Validate/Sitemap/LastmodTest.php	(working copy)
@@ -101,7 +101,7 @@
         foreach ($values as $value) {
             $this->assertSame(false, $this->_validator->isValid($value));
             $messages = $this->_validator->getMessages();
-            $this->assertContains('is no valid', current($messages));
+            $this->assertContains('is not a valid', current($messages));
         }
     }
 
Index: library/Zend/Validate/EmailAddress.php
===================================================================
--- library/Zend/Validate/EmailAddress.php	(revision 23237)
+++ library/Zend/Validate/EmailAddress.php	(working copy)
@@ -52,13 +52,13 @@
      */
     protected $_messageTemplates = array(
         self::INVALID            => "Invalid type given. String expected",
-        self::INVALID_FORMAT     => "'%value%' is no valid email address in the basic format local-part@hostname",
-        self::INVALID_HOSTNAME   => "'%hostname%' is no valid hostname for email address '%value%'",
+        self::INVALID_FORMAT     => "'%value%' is not a valid email address in the basic format local-part@hostname",
+        self::INVALID_HOSTNAME   => "'%hostname%' is not a valid hostname for email address '%value%'",
         self::INVALID_MX_RECORD  => "'%hostname%' does not appear to have a valid MX record for the email address '%value%'",
         self::INVALID_SEGMENT    => "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network",
         self::DOT_ATOM           => "'%localPart%' can not be matched against dot-atom format",
         self::QUOTED_STRING      => "'%localPart%' can not be matched against quoted-string format",
-        self::INVALID_LOCAL_PART => "'%localPart%' is no valid local part for email address '%value%'",
+        self::INVALID_LOCAL_PART => "'%localPart%' is not a valid local part for email address '%value%'",
         self::LENGTH_EXCEEDED    => "'%value%' exceeds the allowed length",
     );
 
Index: library/Zend/Validate/Isbn.php
===================================================================
--- library/Zend/Validate/Isbn.php	(revision 23237)
+++ library/Zend/Validate/Isbn.php	(working copy)
@@ -45,7 +45,7 @@
      */
     protected $_messageTemplates = array(
         self::INVALID => "Invalid type given. String or integer expected",
-        self::NO_ISBN => "'%value%' is no valid ISBN number",
+        self::NO_ISBN => "'%value%' is not a valid ISBN number",
     );
 
     /**
Index: library/Zend/Validate/Sitemap/Loc.php
===================================================================
--- library/Zend/Validate/Sitemap/Loc.php	(revision 23237)
+++ library/Zend/Validate/Sitemap/Loc.php	(working copy)
@@ -56,7 +56,7 @@
      * @var array
      */
     protected $_messageTemplates = array(
-        self::NOT_VALID => "'%value%' is no valid sitemap location",
+        self::NOT_VALID => "'%value%' is not a valid sitemap location",
         self::INVALID   => "Invalid type given. String expected",
     );
 
Index: library/Zend/Validate/Sitemap/Priority.php
===================================================================
--- library/Zend/Validate/Sitemap/Priority.php	(revision 23237)
+++ library/Zend/Validate/Sitemap/Priority.php	(working copy)
@@ -51,7 +51,7 @@
      * @var array
      */
     protected $_messageTemplates = array(
-        self::NOT_VALID => "'%value%' is no valid sitemap priority",
+        self::NOT_VALID => "'%value%' is not a valid sitemap priority",
         self::INVALID   => "Invalid type given. Numeric string, integer or float expected",
     );
 
Index: library/Zend/Validate/Sitemap/Changefreq.php
===================================================================
--- library/Zend/Validate/Sitemap/Changefreq.php	(revision 23237)
+++ library/Zend/Validate/Sitemap/Changefreq.php	(working copy)
@@ -51,7 +51,7 @@
      * @var array
      */
     protected $_messageTemplates = array(
-        self::NOT_VALID => "'%value%' is no valid sitemap changefreq",
+        self::NOT_VALID => "'%value%' is not a valid sitemap changefreq",
         self::INVALID   => "Invalid type given. String expected",
     );
 
Index: library/Zend/Validate/Sitemap/Lastmod.php
===================================================================
--- library/Zend/Validate/Sitemap/Lastmod.php	(revision 23237)
+++ library/Zend/Validate/Sitemap/Lastmod.php	(working copy)
@@ -57,7 +57,7 @@
      * @var array
      */
     protected $_messageTemplates = array(
-        self::NOT_VALID => "'%value%' is no valid sitemap lastmod",
+        self::NOT_VALID => "'%value%' is not a valid sitemap lastmod",
         self::INVALID   => "Invalid type given. String expected",
     );
 
Index: library/Zend/Application/Resource/Session.php
===================================================================
--- library/Zend/Application/Resource/Session.php	(revision 23237)
+++ library/Zend/Application/Resource/Session.php	(working copy)
@@ -50,7 +50,7 @@
      *
      * @param  array|string|Zend_Session_SaveHandler_Interface $saveHandler
      * @return Zend_Application_Resource_Session
-     * @throws Zend_Application_Resource_Exception When $saveHandler is no valid save handler
+     * @throws Zend_Application_Resource_Exception When $saveHandler is not a valid save handler
      */
     public function setSaveHandler($saveHandler)
     {
Index: resources/languages/uk/Zend_Validate.php
===================================================================
--- resources/languages/uk/Zend_Validate.php	(revision 23237)
+++ resources/languages/uk/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "Неприпустимий тип даних, значення повинно рядком",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' неприпустима адреса електронної пошти для формату ім'я@домен",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' неприпустиме ім'я хоста для адреси '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' неприпустима адреса електронної пошти для формату ім'я@домен",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' неприпустиме ім'я хоста для адреси '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' не має коректного MX-запису про адресу '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' не є маршрутизованим сегментом мережі. Адреса електронної пошти '%value%' не може бути отримана з публічної мережі.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart% не відповідає формату dot-atom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' не відповідає формату quoted-string",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' неприпустиме ім'я для адреси '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' неприпустиме ім'я для адреси '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' перевищує дозволену довжину",
 
     // Zend_Validate_File_Count
@@ -223,7 +223,7 @@
     "'%value%' does not appear to be a valid IP address" => "'%value%' - некоректна IP-адреса",
 
     // Zend_Validate_Isbn
-    "'%value%' is no valid ISBN number" => "'%value%' - некоректний номер ISBN",
+    "'%value%' is not a valid ISBN number" => "'%value%' - некоректний номер ISBN",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' не менше ніж '%max%'",
@@ -241,16 +241,16 @@
     "'%value%' does not match against pattern '%pattern%'" => "'%value%' не відповідає шаблону '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' неприпустиме значення для sitemap changefreq",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' неприпустиме значення для sitemap changefreq",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' неприпустиме значення для sitemap lastmod",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' неприпустиме значення для sitemap lastmod",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' неприпустиме значення для sitemap location",
+    "'%value%' is not a valid sitemap location" => "'%value%' неприпустиме значення для sitemap location",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' неприпустиме значення для sitemap priority",
+    "'%value%' is not a valid sitemap priority" => "'%value%' неприпустиме значення для sitemap priority",
 
     // Zend_Validate_StringLength
     "Invalid type given, value should be a string" => "Неприпустимий тип даних, значення повинно бути рядком",
Index: resources/languages/en/Zend_Validate.php
===================================================================
--- resources/languages/en/Zend_Validate.php	(revision 23237)
+++ resources/languages/en/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given. String expected" => "Invalid type given. String expected",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' is no valid email address in the basic format local-part@hostname",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' is no valid hostname for email address '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' is not a valid email address in the basic format local-part@hostname",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' is not a valid hostname for email address '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' does not appear to have a valid MX record for the email address '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network" => "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' can not be matched against dot-atom format",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' can not be matched against quoted-string format",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' is no valid local part for email address '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' is not a valid local part for email address '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' exceeds the allowed length",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given. String or integer expected" => "Invalid type given. String or integer expected",
-    "'%value%' is no valid ISBN number" => "'%value%' is no valid ISBN number",
+    "'%value%' is not a valid ISBN number" => "'%value%' is not a valid ISBN number",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' is not less than '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "There was an internal error while using the pattern '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' is no valid sitemap changefreq",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' is not a valid sitemap changefreq",
     "Invalid type given. String expected" => "Invalid type given. String expected",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' is no valid sitemap lastmod",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' is not a valid sitemap lastmod",
     "Invalid type given. String expected" => "Invalid type given. String expected",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' is no valid sitemap location",
+    "'%value%' is not a valid sitemap location" => "'%value%' is not a valid sitemap location",
     "Invalid type given. String expected" => "Invalid type given. String expected",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' is no valid sitemap priority",
+    "'%value%' is not a valid sitemap priority" => "'%value%' is not a valid sitemap priority",
     "Invalid type given. Numeric string, integer or float expected" => "Invalid type given. Numeric string, integer or float expected",
 
     // Zend_Validate_StringLength
Index: resources/languages/sr/Zend_Validate.php
===================================================================
--- resources/languages/sr/Zend_Validate.php	(revision 23237)
+++ resources/languages/sr/Zend_Validate.php	(working copy)
@@ -78,13 +78,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "Nevalidan tip, vrednost treba da bude tekst",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' nije validna adresa elektronske pošte u formatu adresa@imehosta",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' nije validno ime hosta za adresu elektronske pošte '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' nije validna adresa elektronske pošte u formatu adresa@imehosta",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' nije validno ime hosta za adresu elektronske pošte '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' nema validan MX zapis za adresu elektronske pošte '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' nije rutabilan mrežni segment. Adresa elektronske pošte '%value%' ne treba da bude razrešena sa javne mreže",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' se ne poklapa sa dot-atom formatom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' se ne poklapa sa quoted-string formatom",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' nije validan deo adrese elektronske pošte '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' nije validan deo adrese elektronske pošte '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' prelazi dozvoljenu dužinu",
 
     // Zend_Validate_File_Count
@@ -225,7 +225,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given, value should be string or integer" => "Nevalidan tip, vrednost treba da bude tekst ili ceo broj",
-    "'%value%' is no valid ISBN number" => "'%value%' nije validan ISBN broj",
+    "'%value%' is not a valid ISBN number" => "'%value%' nije validan ISBN broj",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' je manje od '%max%'",
@@ -244,19 +244,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Dogodila se greška pri korišćenju formata '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' nije validna frekvencija promene mape sajta",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' nije validna frekvencija promene mape sajta",
     "Invalid type given, the value should be a string" => "Nevalidan tip, vrednost treba da bude tekst",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' nije validan datum izmene mape sajta",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' nije validan datum izmene mape sajta",
     "Invalid type given, the value should be a string" => "Nevalidan tip, vrednost treba da bude tekst",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' nije validna lokacija mape sajta",
+    "'%value%' is not a valid sitemap location" => "'%value%' nije validna lokacija mape sajta",
     "Invalid type given, the value should be a string" => "Nevalidan tip, vrednost treba da bude tekst",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' nije validan prioritet mape sajta",
+    "'%value%' is not a valid sitemap priority" => "'%value%' nije validan prioritet mape sajta",
     "Invalid type given, the value should be a integer, a float or a numeric string" => "Nevalidan tip, vrednost treba da bude broj ili numerički niz",
 
     // Zend_Validate_StringLength
Index: resources/languages/pt_BR/Zend_Validate.php
===================================================================
--- resources/languages/pt_BR/Zend_Validate.php	(revision 23237)
+++ resources/languages/pt_BR/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "O tipo especificado é inválido, o valor deve ser uma string",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' não é um endereço de e-mail válido no formato local-part@hostname",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' não é um nome de host válido para o endereço de e-mail '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' não é um endereço de e-mail válido no formato local-part@hostname",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' não é um nome de host válido para o endereço de e-mail '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' não parece ter um registro MX válido para o endereço de e-mail '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' não é um segmento de rede roteável. O endereço de e-mail '%value%' não deve ser resolvido a partir de um rede pública.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' não corresponde com o formato dot-atom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' não corresponde com o formato quoted-string",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' não é uma parte local válida para o endereço de e-mail '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' não é uma parte local válida para o endereço de e-mail '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' excede o comprimento permitido",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given, value should be string or integer" => "O tipo especificado é inválido, o valor deve ser string ou inteiro",
-    "'%value%' is no valid ISBN number" => "'%value%' não é um número ISBN válido",
+    "'%value%' is not a valid ISBN number" => "'%value%' não é um número ISBN válido",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' não é menor que '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Houve um erro interno durante o uso do padrão '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' não é um changefreq de sitemap válido",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' não é um changefreq de sitemap válido",
     "Invalid type given, the value should be a string" => "O tipo especificado é inválido, o valor deve ser uma string",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' não é um lastmod de sitemap válido",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' não é um lastmod de sitemap válido",
     "Invalid type given, the value should be a string" => "O tipo especificado é inválido, o valor deve ser uma string",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' não é uma localização de sitemap válida",
+    "'%value%' is not a valid sitemap location" => "'%value%' não é uma localização de sitemap válida",
     "Invalid type given, the value should be a string" => "O tipo especificado é inválido, o valor deve ser uma string",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' não é uma prioridade de sitemap válida",
+    "'%value%' is not a valid sitemap priority" => "'%value%' não é uma prioridade de sitemap válida",
     "Invalid type given, the value should be a integer, a float or a numeric string" => "O tipo especificado é inválido, o valor deve ser um inteiro, um float ou uma string numérica",
 
     // Zend_Validate_StringLength
Index: resources/languages/ru/Zend_Validate.php
===================================================================
--- resources/languages/ru/Zend_Validate.php	(revision 23237)
+++ resources/languages/ru/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "Недопустимый тип данных, значение должно быть строкой",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' недопустимый адрес электронной почты. Введите его в формате имя@домен",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' недопустимое имя хоста для адреса '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' недопустимый адрес электронной почты. Введите его в формате имя@домен",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' недопустимое имя хоста для адреса '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' не имеет корректной MX-записи об адресе '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' не является маршрутизируемым сегментом сети. Адрес электронной почты '%value%' не может быть получен из публичной сети.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart% не соответствует формату dot-atom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' не соответствует формату quoted-string",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' недопустимое имя для адреса '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' недопустимое имя для адреса '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' превышает допустимую длину",
 
     // Zend_Validate_File_Count
@@ -223,7 +223,7 @@
     "'%value%' does not appear to be a valid IP address" => "'%value%' не является корректным IP-адресом",
 
     // Zend_Validate_Isbn
-    "'%value%' is no valid ISBN number" => "'%value%' не является корректным номером ISBN",
+    "'%value%' is not a valid ISBN number" => "'%value%' не является корректным номером ISBN",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' не меньше, чем '%max%'",
@@ -241,16 +241,16 @@
     "'%value%' does not match against pattern '%pattern%'" => "'%value%' не соответствует шаблону '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' недопустимое значение для sitemap changefreq",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' недопустимое значение для sitemap changefreq",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' недопустимое значение для sitemap lastmod",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' недопустимое значение для sitemap lastmod",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' недопустимое значение для sitemap location",
+    "'%value%' is not a valid sitemap location" => "'%value%' недопустимое значение для sitemap location",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' недопустимое значение для sitemap priority",
+    "'%value%' is not a valid sitemap priority" => "'%value%' недопустимое значение для sitemap priority",
 
     // Zend_Validate_StringLength
     "Invalid type given, value should be a string" => "Недопустимый тип данных, значение должно быть строкой",
Index: resources/languages/es/Zend_Validate.php
===================================================================
--- resources/languages/es/Zend_Validate.php	(revision 23237)
+++ resources/languages/es/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "El tipo especificado es incorrecto, el valor debería ser una cadena de texto",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' no es una dirección de correo electrónico válida en el formato básico parte-local@dominio",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' no es un nombre de dominio válido para la dirección de correo electrónico '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' no es una dirección de correo electrónico válida en el formato básico parte-local@dominio",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' no es un nombre de dominio válido para la dirección de correo electrónico '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' no parece tener un registro MX válido para la dirección de correo electrónico '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' no es un segmento de red enrutable. La dirección de correo electrónico '%value%' no se debe poder resolver desde una red pública.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' no se corresponde con el formato dot-atom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' no se corresponde con el formato de cadena de texto entrecomillada",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' no es una parte local válida para la dirección de correo electrónico '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' no es una parte local válida para la dirección de correo electrónico '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' excede la longitud permitida",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given, value should be string or integer" => "El tipo especificado es inválido, el valor debería ser una cadena de texto o un entero",
-    "'%value%' is no valid ISBN number" => "El número ISBN especificado ('%value%') no es válido",
+    "'%value%' is not a valid ISBN number" => "El número ISBN especificado ('%value%') no es válido",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' no es menor que '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Se ha producido un error interno al usar el patrón '%pattern%' especificado",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' no es una especificación válida de frecuencia de cambio",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' no es una especificación válida de frecuencia de cambio",
     "Invalid type given, the value should be a string" => "El tipo especificado es inválido, el valor debería ser una cadena de texto",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' no es un lastmod de mapa web válido",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' no es un lastmod de mapa web válido",
     "Invalid type given, the value should be a string" => "El tipo especificado es inválido, el valor debería ser una cadena de texto",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' no es una ubicación de mapa web válida",
+    "'%value%' is not a valid sitemap location" => "'%value%' no es una ubicación de mapa web válida",
     "Invalid type given, the value should be a string" => "El tipo especificado es inválido, el valor debería ser una cadena de texto",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' no es una prioridad de mapa web válida",
+    "'%value%' is not a valid sitemap priority" => "'%value%' no es una prioridad de mapa web válida",
     "Invalid type given, the value should be a integer, a float or a numeric string" => "El tipo especificado es inválido, el valor debería ser un entero, un float o una cadena de texto numérica",
 
     // Zend_Validate_StringLength
Index: resources/languages/fr/Zend_Validate.php
===================================================================
--- resources/languages/fr/Zend_Validate.php	(revision 23237)
+++ resources/languages/fr/Zend_Validate.php	(working copy)
@@ -25,11 +25,11 @@
  */
 return array (
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' ne semble pas avoir d'enregistrement MX valide pour l'adresse email '%value%'",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' n'est pas un nom d'hôte valide pour l'adresse email '%value%'",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' n'est pas un nom d'hôte valide pour l'adresse email '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network" => "'%hostname%' n'est pas dans un segment réseau routable. L'adresse email '%value%' ne devrait pas être résolue depuis un réseau public.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' ne correspond pas au format dot-atom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' ne correspond pas au format quoted-string",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' n'est pas une partie locale valide pour l'adresse email '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' n'est pas une partie locale valide pour l'adresse email '%value%'",
     "'%value%' appears to be a DNS hostname but cannot extract TLD part" => "'%value%' semble être un nom d'hôte DNS mais l'extension TLD ne peut être extraite",
     "'%value%' appears to be a DNS hostname but cannot match TLD against known list" => "'%value%' semble être un nom d'hôte DNS mais son extension TLD semble inconnue",
     "'%value%' appears to be a DNS hostname but cannot match against hostname schema for TLD '%tld%'" => "'%value%' semble être un nom d'hôte DNS valide mais ne correspond pas au schéma de l'extension TLD '%tld%'",
@@ -58,12 +58,12 @@
     "'%value%' is an empty string" => "'%value%' est une chaîne vide",
     "'%value%' is less than %min% characters long" => "La taille de '%value%' est inférieure à %min% caractères",
     "'%value%' is more than %max% characters long" => "La taille de '%value%' est supérieure à %max% caractères",
-    "'%value%' is no valid ISBN number" => "'%value%' n'est pas un ISBN valide",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' n'est pas un email valide dans le format local-part@hostname",
-    "'%value%' is no valid sitemap changefreq" => "'%value%' n'est pas une valeur de fréquence de sitemap valide",
-    "'%value%' is no valid sitemap lastmod" => "'%value%' n'est pas une date de modification de sitemap valide",
-    "'%value%' is no valid sitemap location" => "'%value%' n'est pas un emplacement valide pour une sitemap",
-    "'%value%' is no valid sitemap priority" => "'%value%' n'est pas une priorité sitemap valide",
+    "'%value%' is not a valid ISBN number" => "'%value%' n'est pas un ISBN valide",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' n'est pas un email valide dans le format local-part@hostname",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' n'est pas une valeur de fréquence de sitemap valide",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' n'est pas une date de modification de sitemap valide",
+    "'%value%' is not a valid sitemap location" => "'%value%' n'est pas un emplacement valide pour une sitemap",
+    "'%value%' is not a valid sitemap priority" => "'%value%' n'est pas une priorité sitemap valide",
     "'%value%' is not between '%min%' and '%max%', inclusively" => "'%value%' n'est pas comprise entre '%min%' et '%max%', inclusivement",
     "'%value%' is not from an allowed institute" => "'%value%' ne provient pas d'une institution autorisée",
     "'%value%' is not greater than '%min%'" => "'%value%' n'est pas plus grand que '%min%'",
Index: resources/languages/de/Zend_Validate.php
===================================================================
--- resources/languages/de/Zend_Validate.php	(revision 23237)
+++ resources/languages/de/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given. String expected" => "Ungültiger Typ angegeben. String erwartet",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' ist keine gültige Emailadresse im Basisformat local-part@hostname",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' ist kein gültiger Hostname für die Emailadresse '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' ist keine gültige Emailadresse im Basisformat local-part@hostname",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' ist kein gültiger Hostname für die Emailadresse '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' scheint keinen gültigen MX Eintrag für die Emailadresse '%value%' zu haben",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network" => "'%hostname%' ist in keinem routebaren Netzwerksegment. Die Emailadresse '%value%' sollte nicht vom öffentlichen Netz aus aufgelöst werden",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' passt nicht auf das dot-atom Format",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' passt nicht auf das quoted-string Format",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' ist kein gültiger lokaler Teil für die Emailadresse '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' ist kein gültiger lokaler Teil für die Emailadresse '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' ist länger als erlaubt",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given. String or integer expected" => "Ungültiger Typ angegeben. String oder Integer erwartet",
-    "'%value%' is no valid ISBN number" => "'%value%' ist keine gültige ISBN Nummer",
+    "'%value%' is not a valid ISBN number" => "'%value%' ist keine gültige ISBN Nummer",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' ist nicht weniger als '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Es gab einen internen Fehler bei der Verwendung des Patterns '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' ist keine gültige Changefreq für Sitemap",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' ist keine gültige Changefreq für Sitemap",
     "Invalid type given. String expected" => "Ungültiger Typ angegeben. String erwartet",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' ist keine gültige Lastmod für Sitemap",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' ist keine gültige Lastmod für Sitemap",
     "Invalid type given. String expected" => "Ungültiger Typ angegeben. String erwartet",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' ist keine gültige Location für Sitemap",
+    "'%value%' is not a valid sitemap location" => "'%value%' ist keine gültige Location für Sitemap",
     "Invalid type given. String expected" => "Ungültiger Typ angegeben. String erwartet",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' ist keine gültige Priority für Sitemap",
+    "'%value%' is not a valid sitemap priority" => "'%value%' ist keine gültige Priority für Sitemap",
     "Invalid type given. Numeric string, integer or float expected" => "Ungültiger Typ angegeben. Nummerischer String, Integer oder Float erwartet",
 
     // Zend_Validate_StringLength
Index: resources/languages/hr/Zend_Validate.php
===================================================================
--- resources/languages/hr/Zend_Validate.php	(revision 23237)
+++ resources/languages/hr/Zend_Validate.php	(working copy)
@@ -76,13 +76,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "Neispravan tip, vrijednost bi trebala biti niz",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' nije ispravna email adresa u osnovnom formatu lokalni-dio@ime-poslužitelja",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' nije ispravno ime poslužitelja za email adresu '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' nije ispravna email adresa u osnovnom formatu lokalni-dio@ime-poslužitelja",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' nije ispravno ime poslužitelja za email adresu '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' nema ispravan MX zapis za email adresu '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' nije u rutabilnom mrežnom segmentu. Email adresa '%value%' ne bi smjela biti razlučiva iz javne mreže.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' se ne podudara s dot-atom formatom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' se ne podudara s 'quoted-string' formatom",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' nije ispravan lokalni dio za email adresu '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' nije ispravan lokalni dio za email adresu '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' je duža od dozvoljene dužine",
 
     // Zend_Validate_File_Count
@@ -223,7 +223,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given, value should be string or integer" => "Neispravan tip, vrijednost mora biti niz ili cijeli broj",
-    "'%value%' is no valid ISBN number" => "'%value%' nije ispravan ISBN broj",
+    "'%value%' is not a valid ISBN number" => "'%value%' nije ispravan ISBN broj",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' nije manje od '%max%'",
@@ -242,19 +242,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Došlo je do interne pogreške prilikom korištenja uzorka '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' nije ispravna vrijednost za sitemap 'changefreq'",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' nije ispravna vrijednost za sitemap 'changefreq'",
     "Invalid type given, the value should be a string" => "Neispravan tip, vrijednost mora biti niz",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' nije ispravna vrijednost za sitemap 'lastmod'",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' nije ispravna vrijednost za sitemap 'lastmod'",
     "Invalid type given, the value should be a string" => "Neispravan tip, vrijednost mora biti niz",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' nije ispravna lokacija za 'sitemap'",
+    "'%value%' is not a valid sitemap location" => "'%value%' nije ispravna lokacija za 'sitemap'",
     "Invalid type given, the value should be a string" => "Neispravan tip, vrijednost mora biti niz",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' nije ispravna vrijednost za sitemap 'priority'",
+    "'%value%' is not a valid sitemap priority" => "'%value%' nije ispravna vrijednost za sitemap 'priority'",
     "Invalid type given, the value should be a integer, a float or a numeric string" => "Neispravan tip, vrijednost mora biti cijeli broj, realni broj ili niz znamenki",
 
     // Zend_Validate_StringLength
Index: resources/languages/nl/Zend_Validate.php
===================================================================
--- resources/languages/nl/Zend_Validate.php	(revision 23237)
+++ resources/languages/nl/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "Ongeldig type opgegeven, waarde moet een string zijn",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' is geen geldig e-mail adres in het basis formaat lokaal-gedeelte@hostname",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' is geen geldige hostnaam voor e-mail adres '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' is geen geldig e-mail adres in het basis formaat lokaal-gedeelte@hostname",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' is geen geldige hostnaam voor e-mail adres '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' lijkt geen geldig MX record te hebben voor e-mail adres '%value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' bevindt zich niet in een routeerbaar netwerk segment. Het e-mail adres '%value%' zou niet naar mogen worden verwezen vanaf een publiek netwerk.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' kan niet worden gematched met het dot-atom formaat",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' kan niet worden gematched met het quoted-string formaat",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' is geen geldig lokaal gedeelte voor e-mail adres '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' is geen geldig lokaal gedeelte voor e-mail adres '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' overschrijdt de toegestane lengte",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given, value should be string or integer" => "Ongeldig type opgegeven, waarde moet een string of integer zijn",
-    "'%value%' is no valid ISBN number" => "'%value%' is geen geldig ISBN nummer",
+    "'%value%' is not a valid ISBN number" => "'%value%' is geen geldig ISBN nummer",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' is niet minder dan '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Er is een interne fout opgetreden tijdens het gebruik van het patroon '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' is geen geldige sitemap changefreq",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' is geen geldige sitemap changefreq",
     "Invalid type given, the value should be a string" => "Ongeldig type opgegeven, waarde dient een string te zijn",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' is geen geldige sitemap lastmod",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' is geen geldige sitemap lastmod",
     "Invalid type given, the value should be a string" => "Ongeldig type opgegeven, waarde dient een string te zijn",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' is geen geldige sitemap locatie",
+    "'%value%' is not a valid sitemap location" => "'%value%' is geen geldige sitemap locatie",
     "Invalid type given, the value should be a string" => "Ongeldig type opgegeven, waarde dient een string te zijn",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' is geen geldige sitemap prioriteit",
+    "'%value%' is not a valid sitemap priority" => "'%value%' is geen geldige sitemap prioriteit",
     "Invalid type given, the value should be a integer, a float or a numeric string" => "Ongeldig type opgegeven, waarde dient een integer, float of een numerieke string te zijn",
 
     // Zend_Validate_StringLength
Index: resources/languages/ja/Zend_Validate.php
===================================================================
--- resources/languages/ja/Zend_Validate.php	(revision 23237)
+++ resources/languages/ja/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given. String expected" => "不正な形式です。文字列が期待されています",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' はメールアドレスの基本的な形式 local-part@hostname ではありません",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "メールアドレス '%value%' 内の '%hostname%' は有効なホスト名ではありません",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' はメールアドレスの基本的な形式 local-part@hostname ではありません",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "メールアドレス '%value%' 内の '%hostname%' は有効なホスト名ではありません",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "メールアドレス '%value%' 内の '%hostname%' は有効な MX レコードではないようです",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network" => "'%hostname%' はネットワークセグメントにありません。メールアドレス '%value%' はパブリックなネットワークから名前解決できませんでした",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' はドットアトム形式ではありません",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' は引用文字列形式ではありません",
-    "'%localPart%' is no valid local part for email address '%value%'" => "メールアドレス '%value%' 内の '%localPart%' は有効なローカルパートではありません",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "メールアドレス '%value%' 内の '%localPart%' は有効なローカルパートではありません",
     "'%value%' exceeds the allowed length" => "'%value%' は許された長さを超えています",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
      "Invalid type given. String or integer expected" => "不正な形式です。文字列または整数が期待されています",
-    "'%value%' is no valid ISBN number" => " '%value%' は ISBN 番号ではありません",
+    "'%value%' is not a valid ISBN number" => " '%value%' は ISBN 番号ではありません",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => " '%value%' は '%max%' 未満ではありません",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "正規表現パターン '%pattern%' を使用中に内部エラーが発生しました。",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => " '%value%' は正しいサイトマップの更新頻度ではありません",
+    "'%value%' is not a valid sitemap changefreq" => " '%value%' は正しいサイトマップの更新頻度ではありません",
     "Invalid type given. String expected" => "不正な形式です。文字列が期待されています",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => " '%value%' は正しいサイトマップの最終更新日ではありません",
+    "'%value%' is not a valid sitemap lastmod" => " '%value%' は正しいサイトマップの最終更新日ではありません",
     "Invalid type given. String expected" => "不正な形式です。文字列が期待されています",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => " '%value%' は正しいサイトマップの位置ではありません",
+    "'%value%' is not a valid sitemap location" => " '%value%' は正しいサイトマップの位置ではありません",
     "Invalid type given. String expected" => "不正な形式です。文字列が期待されています",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => " '%value%' は正しいサイトマップの優先度ではありません",
+    "'%value%' is not a valid sitemap priority" => " '%value%' は正しいサイトマップの優先度ではありません",
     "Invalid type given. Numeric string, integer or float expected" => "不正な形式です。数字、整数もしくは小数が期待されています",
 
     // Zend_Validate_StringLength
Index: resources/languages/it/Zend_Validate.php
===================================================================
--- resources/languages/it/Zend_Validate.php	(revision 23237)
+++ resources/languages/it/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given, value should be a string" => "Tipo di dato non valido, il dato dev'essere una stringa",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' non è un indirizzo email valido nel formato base local-part@hostname",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' non è un hostname valido nell'indirizzo email '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' non è un indirizzo email valido nel formato base local-part@hostname",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' non è un hostname valido nell'indirizzo email '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "'%hostname%' non sembra avere un record MX DNS valido nell'indirizzo email %value%'",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network." => "'%hostname%' non è in un segmento di rete routabile. L'indirizzo email '%value%' non può essere risolto nella rete pubblica.",
     "'%localPart%' can not be matched against dot-atom format" => "'%localPart%' non può essere validato nel formato dot-atom",
     "'%localPart%' can not be matched against quoted-string format" => "'%localPart%' non può essere validato nel formato quoted-string",
-    "'%localPart%' is no valid local part for email address '%value%'" => "'%localPart%' non è una local part valida nell'indirizzo email '%value%'",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "'%localPart%' non è una local part valida nell'indirizzo email '%value%'",
     "'%value%' exceeds the allowed length" => "'%value%' supera la lunghezza consentita",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given, value should be string or integer" => "Tipo di dato non valido, il dato dev'essere una stringa o un intero",
-    "'%value%' is no valid ISBN number" => "'%value%' non è un numero ISBN valido",
+    "'%value%' is not a valid ISBN number" => "'%value%' non è un numero ISBN valido",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' non è minore di '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Si è verificato un errore interno usando il pattern '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' non è una sitemap changefreq valida",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' non è una sitemap changefreq valida",
     "Invalid type given, the value should be a string" => "Invalid type given, the value should be a string",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' non è un sitemap lastmod valido",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' non è un sitemap lastmod valido",
     "Invalid type given, the value should be a string" => "Tipo di dato non valido, il dato dev'essere una stringa",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' non è una sitemap location valida",
+    "'%value%' is not a valid sitemap location" => "'%value%' non è una sitemap location valida",
     "Invalid type given, the value should be a string" => "Tipo di dato non valido, il dato dev'essere una stringa",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' non è una sitemap priority valida",
+    "'%value%' is not a valid sitemap priority" => "'%value%' non è una sitemap priority valida",
     "Invalid type given, the value should be a integer, a float or a numeric string" => "Tipo di dato non valido, il dato dev'essere di tipo intero, float o una stringa numerica",
 
     // Zend_Validate_StringLength
Index: resources/languages/fi/Zend_Validate.php
===================================================================
--- resources/languages/fi/Zend_Validate.php	(revision 23237)
+++ resources/languages/fi/Zend_Validate.php	(working copy)
@@ -77,13 +77,13 @@
 
     // Zend_Validate_EmailAddress
     "Invalid type given. String expected" => "Epäkelpo syöte. Pitäisi olla merkkijono",
-    "'%value%' is no valid email address in the basic format local-part@hostname" => "'%value%' on virheellinen sähköpostiosoite, ei vastaa muotoa paikallisosa@alue",
-    "'%hostname%' is no valid hostname for email address '%value%'" => "'%hostname%' on virheellinen verkkotunnus osoitteelle '%value%'",
+    "'%value%' is not a valid email address in the basic format local-part@hostname" => "'%value%' on virheellinen sähköpostiosoite, ei vastaa muotoa paikallisosa@alue",
+    "'%hostname%' is not a valid hostname for email address '%value%'" => "'%hostname%' on virheellinen verkkotunnus osoitteelle '%value%'",
     "'%hostname%' does not appear to have a valid MX record for the email address '%value%'" => "Osoitteen '%value%' verkkotunnukselle '%hostname%' ei löydy MX-tietuetta",
     "'%hostname%' is not in a routable network segment. The email address '%value%' should not be resolved from public network" => "'%hostname%' ei ole reititettävän verkon osa. Sähköpostiosoitetta '%value%' ei pitäisi selvittää julkisesta verkosta.",
     "'%localPart%' can not be matched against dot-atom format" => "Virheellinen paikallisosa, '%localPart%' ei ole verrattavissa dot-atom -muotoon",
     "'%localPart%' can not be matched against quoted-string format" => "Virheellinen paikallisosa, '%localPart%' ei ole verrattavissa quoted-string -muotoon",
-    "'%localPart%' is no valid local part for email address '%value%'" => "Sähköpostiosoitteen '%value%' paikallisosa '%localPart%' on virheellinen",
+    "'%localPart%' is not a valid local part for email address '%value%'" => "Sähköpostiosoitteen '%value%' paikallisosa '%localPart%' on virheellinen",
     "'%value%' exceeds the allowed length" => "Osoite '%value%' on liian pitkä",
 
     // Zend_Validate_File_Count
@@ -224,7 +224,7 @@
 
     // Zend_Validate_Isbn
     "Invalid type given. String or integer expected" => "Epäkelpo syöte. Pitäisi olla merkkijono tai kokonaisluku",
-    "'%value%' is no valid ISBN number" => "'%value%' ei ole käypä ISBN-numero",
+    "'%value%' is not a valid ISBN number" => "'%value%' ei ole käypä ISBN-numero",
 
     // Zend_Validate_LessThan
     "'%value%' is not less than '%max%'" => "'%value%' ei ole pienempi kuin '%max%'",
@@ -243,19 +243,19 @@
     "There was an internal error while using the pattern '%pattern%'" => "Sisäinen virhe käytettäessa muotoa '%pattern%'",
 
     // Zend_Validate_Sitemap_Changefreq
-    "'%value%' is no valid sitemap changefreq" => "'%value%' ei ole käypä sivukartan muutosnopeus",
+    "'%value%' is not a valid sitemap changefreq" => "'%value%' ei ole käypä sivukartan muutosnopeus",
     "Invalid type given. String expected" => "Epäkelpo syöte. Pitäisi olla merkkijono",
 
     // Zend_Validate_Sitemap_Lastmod
-    "'%value%' is no valid sitemap lastmod" => "'%value%' ei ole käypä arvo sivukartan viimeksimuokatuksi arvoksi",
+    "'%value%' is not a valid sitemap lastmod" => "'%value%' ei ole käypä arvo sivukartan viimeksimuokatuksi arvoksi",
     "Invalid type given. String expected" => "Epäkelpo syöte. Pitäisi olla merkkijono",
 
     // Zend_Validate_Sitemap_Loc
-    "'%value%' is no valid sitemap location" => "'%value%' ei ole käypä sivukartan sijainti",
+    "'%value%' is not a valid sitemap location" => "'%value%' ei ole käypä sivukartan sijainti",
     "Invalid type given. String expected" => "Epäkelpo syöte. Pitäisi olla merkkijono",
 
     // Zend_Validate_Sitemap_Priority
-    "'%value%' is no valid sitemap priority" => "'%value%' ei ole käypä sivukartan prioriteetti",
+    "'%value%' is not a valid sitemap priority" => "'%value%' ei ole käypä sivukartan prioriteetti",
     "Invalid type given. Numeric string, integer or float expected" => "Epäkelpo syöte. Pitäisi olla kokonaisluku tai liukuluku",
 
     // Zend_Validate_StringLength
