Index: tests/Zend/Service/Twitter/TwitterTest.php
===================================================================
--- tests/Zend/Service/Twitter/TwitterTest.php	(revision 23686)
+++ tests/Zend/Service/Twitter/TwitterTest.php	(working copy)
@@ -647,6 +647,27 @@
         $timeline = $twitter->statusUserTimeline($params);
     }
 
+    /**
+     * @group ZF-11014
+     */
+    public function testStatusFriendsTimelineShouldHonorAllFlags()
+    {
+        $params = array(
+            'page'             => 3,
+            'include_rts'      => '1',
+            'trim_user'        => '1',
+            'include_entities' => '1',
+        );
+        $twitter = new Zend_Service_Twitter();
+        $twitter->setLocalHttpClient($this->_stubTwitter(
+            'statuses/friends_timeline.xml', Zend_Http_Client::GET, 'statuses.friends_timeline.page.xml',
+            $params
+        ));  
+
+        // Assertions are part of mocking
+        $timeline = $twitter->statusFriendsTimeline($params);
+    }    
+          
 }
 
 if (PHPUnit_MAIN_METHOD == 'Zend_Service_TwitterTest2::main') {
Index: library/Zend/Service/Twitter.php
===================================================================
--- library/Zend/Service/Twitter.php	(revision 23686)
+++ library/Zend/Service/Twitter.php	(working copy)
@@ -358,6 +358,14 @@
                 case 'page':
                     $_params['page'] = (int) $value;
                     break;
+                case 'max_id':
+                    $_params['max_id'] = $this->_validInteger($value);
+                    break;
+                case 'include_rts':
+                case 'trim_user':
+                case 'include_entities':
+                    $_params[strtolower($key)] = $value ? '1' : '0';
+                    break;                    
                 default:
                     break;
             }
