Index: tests/Zend/Form/Decorator/HtmlTagTest.php
===================================================================
--- tests/Zend/Form/Decorator/HtmlTagTest.php	(revision 25022)
+++ tests/Zend/Form/Decorator/HtmlTagTest.php	(working copy)
@@ -63,6 +63,7 @@
     public function setUp()
     {
         $this->decorator = new Zend_Form_Decorator_HtmlTag();
+        $this->decorator->setElement(new Zend_Form_Element('foo'));
     }
 
     /**
@@ -234,6 +235,63 @@
         $this->decorator->setOption('tag', 'dl');
         $this->assertEquals('dl', $this->decorator->getTag());
     }
+
+    /**
+     * @group ZF-6364
+     */
+    public function testCanSetErrorClass()
+    {
+        $this->decorator->setErrorClass('error');
+        $this->assertEquals('error', $this->decorator->getErrorClass());
+    }
+
+    /**
+     * @group ZF-6364
+     */
+    public function testCanSetErrorClassViaOption()
+    {
+        $this->decorator->setOption('errorClass', 'error');
+        $this->assertEquals('error', $this->decorator->getErrorClass());
+    }
+
+    /**
+     * @group ZF-6364
+     */
+    public function testRenderingWithoutErrorClass()
+    {
+        $this->decorator->getElement()->markAsError();
+        $this->assertEquals(
+            '<div></div>',
+            $this->decorator->render(null)
+        );
+    }
+
+    /**
+     * @group ZF-6364
+     */
+    public function testRenderingWithErrorClass()
+    {
+        $this->decorator->setOption('errorClass', 'invalid');
+        $this->decorator->getElement()->markAsError();
+        $this->assertEquals(
+            '<div class="invalid"></div>',
+            $this->decorator->render(null)
+        );
+    }
+
+    /**
+     * @group ZF-6364
+     */
+    public function testRenderingWithClassAndErrorClass()
+    {
+        $this->decorator->setOption('errorClass', 'invalid');
+        $this->decorator->setOption('class', 'foo');
+        $this->decorator->getElement()->markAsError();
+        $this->assertEquals(
+            '<div class="foo invalid"></div>',
+            $this->decorator->render(null)
+        );
+    }
 }
 
 // Call Zend_Form_Decorator_HtmlTagTest::main() if this source file is executed directly.
