Index: Zend_Framework/library/Zend/Amf/Parse/Amf0/Deserializer.php
===================================================================
--- Zend_Framework/library/Zend/Amf/Parse/Amf0/Deserializer.php	(Revision 15103)
+++ Zend_Framework/library/Zend/Amf/Parse/Amf0/Deserializer.php	(Arbeitskopie)
@@ -48,13 +48,6 @@
     protected $_objectEncoding = Zend_Amf_Constants::AMF0_OBJECT_ENCODING;
 
     /**
-     * refrence to AMF3 deserializer
-     *
-     * @var Zend_Amf_Parse_Amf3_Deserializer
-     */
-    protected $_deserializer = null;
-
-    /**
      * Read AMF markers and dispatch for deserialization
      *
      * Checks for AMF marker types and calls the appropriate methods
@@ -289,7 +282,8 @@
      */
     public function readAmf3TypeMarker()
     {
-        $deserializer = $this->getDeserializer();
+    	require_once 'Zend/Amf/Parse/Amf3/Deserializer.php';
+        $deserializer = new Zend_Amf_Parse_Amf3_Deserializer($this->_stream);
         $this->_objectEncoding = Zend_Amf_Constants::AMF3_OBJECT_ENCODING;
         return $deserializer->readTypeMarker();
     }
@@ -304,18 +298,4 @@
     {
         return $this->_objectEncoding;
     }
-
-    /**
-     * Get deserializer
-     *
-     * @return Zend_Amf_Parse_Amf3_Deserializer
-     */
-    public function getDeserializer()
-    {
-        if (null === $this->_deserializer) {
-            require_once 'Zend/Amf/Parse/Amf3/Deserializer.php';
-            $this->_deserializer = new Zend_Amf_Parse_Amf3_Deserializer($this->_stream);
-        }
-        return $this->_deserializer;
-    }
 }
