Index: tests/Zend/MimeTest.php
===================================================================
--- tests/Zend/MimeTest.php	(revision 23462)
+++ tests/Zend/MimeTest.php	(working copy)
@@ -97,6 +97,7 @@
         $mail->addTo('test@email.com');
 
         // test with generic transport
+        require_once('Mail/MailTest.php');
         $mock = new Zend_Mail_Transport_Sendmail_Mock();
         $mail->send($mock);
         $body = quoted_printable_decode($mock->body);
