--- Zend\Mail\Transport\Smtp.original.php	Tue Jul 11 00:01:06 2006
+++ Zend\Mail\Transport\Smtp.php	Fri Aug 18 10:45:34 2006
@@ -188,9 +188,9 @@ class Zend_Mail_Transport_Smtp extends Z
         $this->_send('DATA');
         $this->_expect(354);
         foreach(explode($this->EOL, $data) as $line) {
-            if ($line=='.') {
-                // important. replace single dot on a line
-                $line='..';
+            if ($line!='' && $line[0]=='.') {
+                // important. the first leading dot is always stripped by the receiver
+                $line='.'.$line;
             }
             $this->_send($line);
         }
