Index: Zend/Http/Client/Adapter/Socket.php
===================================================================
--- Zend/Http/Client/Adapter/Socket.php	(revision 11875)
+++ Zend/Http/Client/Adapter/Socket.php	(working copy)
@@ -295,8 +295,8 @@
             
         // If the connection is set to close, just read until socket closes
         } elseif (isset($headers['connection']) && $headers['connection'] == 'close') {
-            while (($buff = @fread($this->socket, 8192)) !== false) {
-                $response .= $buff;
+            while (!feof($this->socket)) {
+                $response .= @fread($this->socket, 8192);
             }
 
             $this->close();
