Index: tests/Zend/Service/DeveloperGarden/OfflineBaseUserServiceTest.php
===================================================================
--- tests/Zend/Service/DeveloperGarden/OfflineBaseUserServiceTest.php	(revision 24790)
+++ tests/Zend/Service/DeveloperGarden/OfflineBaseUserServiceTest.php	(working copy)
@@ -156,6 +156,9 @@
 
     public function testGetCredentialOnSoapObject()
     {
+        if (!extension_loaded('soap')) {
+            $this->markTestSkipped('SOAP extension is not loaded');
+        }
         $this->assertType(
             'Zend_Service_DeveloperGarden_Credential',
             $this->service->getSoapClient()->getCredential()
@@ -164,6 +167,9 @@
 
     public function testGetTokenServiceOnSoapObject()
     {
+        if (!extension_loaded('soap')) {
+            $this->markTestSkipped('SOAP extension is not loaded');
+        }
         $this->assertType(
             'Zend_Service_DeveloperGarden_SecurityTokenServer',
             $this->service->getSoapClient()->getTokenService()
Index: tests/Zend/Service/DeveloperGarden/OfflineSecurityTokenServerTest.php
===================================================================
--- tests/Zend/Service/DeveloperGarden/OfflineSecurityTokenServerTest.php	(revision 24790)
+++ tests/Zend/Service/DeveloperGarden/OfflineSecurityTokenServerTest.php	(working copy)
@@ -84,6 +84,9 @@
 
     public function testWsdlCache()
     {
+        if (!extension_loaded('soap')) {
+            $this->markTestSkipped('SOAP extension is not loaded');
+        }
         $this->assertNull(
             Zend_Service_DeveloperGarden_SecurityTokenServer_Cache::getWsdlCache()
         );
@@ -126,6 +129,9 @@
 
     public function testDisableWsdlCache()
     {
+        if (!extension_loaded('soap')) {
+            $this->markTestSkipped('SOAP extension is not loaded');
+        }
         Zend_Service_DeveloperGarden_SecurityTokenServer_Cache::setWsdlCache(WSDL_CACHE_BOTH);
         $this->assertEquals(
             WSDL_CACHE_BOTH,
Index: tests/Zend/Service/DeveloperGarden/OfflineClientTest.php
===================================================================
--- tests/Zend/Service/DeveloperGarden/OfflineClientTest.php	(revision 24790)
+++ tests/Zend/Service/DeveloperGarden/OfflineClientTest.php	(working copy)
@@ -130,6 +130,9 @@
 
     public function testGetSoapClient()
     {
+        if (!extension_loaded('soap')) {
+            $this->markTestSkipped('SOAP extension is not loaded');
+        }
         $options = array(
             'Username' => 'Zend',
             'Password' => 'Framework',
@@ -243,6 +246,9 @@
 
     public function testGetClientOptionsWithWsdlCache()
     {
+        if (!extension_loaded('soap')) {
+            $this->markTestSkipped('SOAP extension is not loaded');
+        }
         $this->assertNull(
             Zend_Service_DeveloperGarden_SecurityTokenServer_Cache::setWsdlCache(WSDL_CACHE_BOTH)
         );
