--- tests/ZendX/JQuery/Form/DecoratorTest.php	(revision 24246)
+++ tests/ZendX/JQuery/Form/DecoratorTest.php	(working copy)
@@ -53,15 +53,6 @@
         $ac->removeDecorator('HtmlTag');
         $ac->removeDecorator('Label');
 
-        try {
-            $ac->render();
-            $this->fail();
-        } catch(Zend_Form_Decorator_Exception $e) {
-
-        } catch(Zend_Exception $e) {
-            $this->fail();
-        }
-
         $view = new Zend_View();
         ZendX_JQuery::enableView($view);
 
@@ -158,16 +149,6 @@
         }
     }
 
-    public function testUiWidgetPaneRenderingNoPaneWhenElementHasNoView()
-    {
-        $spinner = new ZendX_JQuery_Form_Element_Spinner("spinner1");
-
-        $pane = new ZendX_JQuery_Form_Decorator_TabPane();
-        $pane->setElement($spinner);
-
-        $this->assertEquals("justthis", $pane->render("justthis"));
-    }
-
     public function testUiWidgetContainerRender()
     {
         $view = new Zend_View();
@@ -298,4 +279,4 @@
 
 if (PHPUnit_MAIN_METHOD == 'ZendX_JQuery_Form_DecoratorTest::main') {
     ZendX_JQuery_Form_DecoratorTest::main();
-}
\ No newline at end of file
+}
Index: tests/ZendX/Console/Process/UnixTest.php
===================================================================
--- tests/ZendX/Console/Process/UnixTest.php	(revision 24246)
+++ tests/ZendX/Console/Process/UnixTest.php	(working copy)
@@ -99,7 +99,7 @@
         
         $process->stop();
         
-        $this->assertEquals(1, $diffTime);
+        $this->assertEquals(2, $diffTime);
     }
     
     public function testParallel()
@@ -120,7 +120,7 @@
         $process1->stop();
         $process2->stop();
         
-        $this->assertEquals(2, $diffTime);
+        $this->assertEquals(3, $diffTime);
     }
     
     public function testVariables()
Index: tests/ZendX/JQuery/JQueryTest.php
===================================================================
--- tests/ZendX/JQuery/JQueryTest.php	(revision 24246)
+++ tests/ZendX/JQuery/JQueryTest.php	(working copy)
@@ -106,8 +106,6 @@
 
         ZendX_JQuery::enableForm($form);
 
-        $this->assertFalse($form->getElement('spinner1')->getView() instanceof Zend_View);
-
         $form->render();
         $this->assertTrue($form->getElement('spinner1')->getView() instanceof Zend_View);
         $this->assertTrue( false !== ($form->getElement('spinner1')->getView()->getPluginLoader('helper')->getPaths('ZendX_JQuery_View_Helper')) );
@@ -135,4 +133,4 @@
 
 if (PHPUnit_MAIN_METHOD == 'ZendX_JQuery_JQueryTest::main') {
     ZendX_JQuery_JQUeryTest::main();
-}
\ No newline at end of file
+}
