--- library/Zend/Tool/Framework/System/Provider/Config.php	(revision 22916)
+++ library/Zend/Tool/Framework/System/Provider/Config.php	(working copy)
@@ -87,7 +87,7 @@
 
         $config = array(
             'php' => array(
-                'includepath' => get_include_path(),
+                'include_path' => get_include_path(),
             ),
         );
         $writer->write($filename, new Zend_Config($config));
