--- Zend/Soap/AutoDiscover.php	2009-04-15 18:07:53.000000000 +0200
+++ AutoDiscover.php	2009-06-09 15:14:46.000000000 +0200
@@ -294,6 +294,8 @@
                 if ($prototype->getReturnType() != "void") {
                     $returnName = 'return';
                     $message = $wsdl->addMessage($method->getName() . 'Response', array($returnName => $wsdl->getType($prototype->getReturnType())));
+                } else {
+                    $message = $wsdl->addMessage($method->getName() . 'Response', array());
                 }
 
                 /* <wsdl:binding>'s */
