===================================================================
--- library/Zend/Mail/Transport/Smtp.php        (revision 1084)
+++ library/Zend/Mail/Transport/Smtp.php        (working copy)
@@ -187,10 +187,10 @@
     {
         $this->_send('DATA');
         $this->_expect(354);
-        foreach(explode($this->EOL, $data) as $line) {
-            if ($line=='.') {
+        foreach(explode(Zend_Mime::LINEEND, $data) as $line) {
+            if (strpos($line, '.') === 0) {
                 // important. replace single dot on a line
-                $line='..';
+                $line = '.' . $line;
             }
             $this->_send($line);
         }
