--- Simpy.php	(revision 18130)
+++ Simpy.php	(working copy)
@@ -48,7 +48,7 @@
      *
      * @var Zend_Http_Client
      */
-    protected $_http;
+    protected $_httpClient;
 
     /**
      * Constructs a new Simpy (free) REST API Client
@@ -59,12 +59,8 @@
      */
     public function __construct($username, $password)
     {
-        /**
-         * @see Zend_Service_Rest
-         */
-        require_once 'Zend/Rest/Client.php';
-        $this->_http = new Zend_Http_Client;
-        $this->_http->setAuth($username, $password);
+        $this->_httpClient = new Zend_Http_Client();
+        $this->_httpClient->setAuth($username, $password);
     }
 
     /**
@@ -75,7 +71,7 @@
      */
     public function getHttpClient()
     {
-        return $this->_http;
+        return $this->_httpClient;
     }
 
     /**
@@ -94,8 +90,8 @@
             $query = '?' . http_build_query($query);
         }
 
-        $this->_http->setUri($this->_baseUri . $op . '.do' . $query);
-        $response = $this->_http->request('GET');
+        $this->_httpClient->setUri($this->_baseUri . $op . '.do' . $query);
+        $response = $this->_httpClient->request('GET');
 
         if ($response->isSuccessful()) {
             $doc = new DOMDocument();
