--- tests/Zend/OpenId/AllTests.php	(revision 7623)
+++ tests/Zend/OpenId/AllTests.php	(working copy)
@@ -56,6 +56,6 @@
     }
 }
 
-if (PHPUnit_MAIN_METHOD == 'Zend_Soap_AllTests::main') {
+if (PHPUnit_MAIN_METHOD == 'Zend_OpenId_AllTests::main') {
     Zend_OpenId_AllTests::main();
 }
Index: tests/Zend/OpenId/ConsumerTest.php
===================================================================
--- tests/Zend/OpenId/ConsumerTest.php	(revision 7623)
+++ tests/Zend/OpenId/ConsumerTest.php	(working copy)
@@ -37,6 +37,16 @@
 require_once 'Zend/OpenId/ResponseHelper.php';
 
 /**
+ * Zend_OpenId_Consumer_Storage_File
+ */
+require_once 'Zend/OpenId/Consumer/Storage/File.php';
+
+/**
+ * Zend_OpenId_Extensions_Sreg
+ */
+require_once 'Zend/OpenId/Extension/Sreg.php';
+
+/**
  * Zend_Http_Client_Adapter_Test
  */
 require_once 'Zend/Http/Client/Adapter/Test.php';
