Index: library/Zend/View/Helper/FormElement.php
===================================================================
--- library/Zend/View/Helper/FormElement.php	(revision 24793)
+++ library/Zend/View/Helper/FormElement.php	(working copy)
@@ -37,14 +37,14 @@
 abstract class Zend_View_Helper_FormElement extends Zend_View_Helper_HtmlElement
 {
     /**
-     * @var Zend_Translate
+     * @var Zend_Translate_Adapter|null
      */
     protected $_translator;
 
     /**
      * Get translator
      *
-     * @return Zend_Translate
+     * @return Zend_Translate_Adapter|null
      */
     public function getTranslator()
     {
@@ -54,7 +54,7 @@
     /**
      * Set translator
      *
-     * @param  Zend_Translate $translator
+     * @param  Zend_Translate|Zend_Translate_Adapter|null $translator
      * @return Zend_View_Helper_FormElement
      */
     public function setTranslator($translator = null)
