Index: library/Zend/Loader.php
===================================================================
--- library/Zend/Loader.php	(revision 7558)
+++ library/Zend/Loader.php	(working copy)
@@ -61,7 +61,7 @@
         }
 
         // autodiscover the path from the class name
-        $file = str_replace('_', DIRECTORY_SEPARATOR, $class) . '.php';
+        $file = str_replace(array('_', '::'), DIRECTORY_SEPARATOR, $class) . '.php';
         if (!empty($dirs)) {
             // use the autodiscovered path
             $dirPath = dirname($file);
@@ -263,7 +263,7 @@
         /**
          * Security check
          */
-        if (preg_match('/[^a-z0-9\\/\\\\_.-]/i', $filename)) {
+        if (preg_match('/[^a-z0-9\\/\\\\_:.-]/i', $filename)) {
             require_once 'Zend/Exception.php';
             throw new Zend_Exception('Security check: Illegal character in filename');
         }
