--- library/Zend/Dojo/View/Helper/Dojo/Container.php	(revision 24416)
+++ library/Zend/Dojo/View/Helper/Dojo/Container.php	(working copy)
@@ -479,7 +479,8 @@
     }
 
     /**
-     * Set path to local dojo
+     * Set path to the local dojo.js file
+     * (Example: /js/dojo/dojo.js)
      *
      * @param  string $path
      * @return Zend_Dojo_View_Helper_Dojo_Container
@@ -492,7 +493,7 @@
     }
 
     /**
-     * Get local path to dojo
+     * Get path to the local dojo.js file
      *
      * @return string
      */
