--- library/Zend/Cache/Frontend/Page.php	(revision 15351)
+++ library/Zend/Cache/Frontend/Page.php	(working copy)
@@ -217,6 +217,17 @@
             if (!is_array($conf)) {
                 Zend_Cache::throwException('regexps option must be an array of arrays !');
             }
+            // Convert regexp option (if set) to key for Zend_Config_Ini configs...
+            if (isset( $conf['regexp'])) {
+                // Remove old expression
+                unset($regexps[ $regexp ]);
+                // Get actual expression
+                $regexp = $conf['regexp'];
+                // Remove expression value - not needed anymore
+                unset($conf['regexp']);
+                // Add the conf item to the regexps array with the new key...
+                $regexps[ $regexp ] = $conf;
+            }
             $validKeys = array_keys($this->_specificOptions['default_options']);
             foreach ($conf as $key=>$value) {
                 if (!is_string($key)) {
