Index: library/Zend/Validate/File/ImageSize.php
===================================================================
--- library/Zend/Validate/File/ImageSize.php	(revision 23889)
+++ library/Zend/Validate/File/ImageSize.php	(working copy)
@@ -307,6 +307,9 @@
      */
     public function isValid($value, $file = null)
     {
+        if (is_null($file)) {
+            $file = array('name' => basename($value));
+        }
         // Is file readable ?
         require_once 'Zend/Loader.php';
         if (!Zend_Loader::isReadable($value)) {
Index: tests/Zend/Validate/File/ImageSizeTest.php
===================================================================
--- tests/Zend/Validate/File/ImageSizeTest.php	(revision 23889)
+++ tests/Zend/Validate/File/ImageSizeTest.php	(working copy)
@@ -240,4 +240,18 @@
             $this->assertContains("less than or equal", $e->getMessage());
         }
     }
+    
+    /**
+     * @ZF-11258
+     */
+    public function testFileNamePresenceInGetMessages()
+    {
+        $file = 'picture.jpg';
+        $validator = new Zend_Validate_File_ImageSize(array('minwidth' => 100, 'minheight' => 10, 'maxwidth' => 500, 'maxheight' => 50));
+        $this->assertEquals(false, $validator->isValid(dirname(__FILE__) . '/_files/' . $file));
+        $failures = $validator->getMessages();
+        foreach ($failures as $message) {
+            $this->assertContains("'" . $file . "'", $message);
+        }
+    }
 }
