Index: .
===================================================================
--- .	(revision 7156)
+++ .	(working copy)
@@ -132,6 +132,13 @@
     protected $_throwExceptions = false;
 
     /**
+     * ErrorHandler class
+     * 
+     * @var string
+     */
+    protected $_errorHandlerClass = 'Zend_Controller_Plugin_ErrorHandler';
+    
+    /**
      * Constructor
      *
      * Instantiate using {@link getInstance()}; front controller is a singleton
@@ -624,6 +631,27 @@
     {
         return $this->_response;
     }
+    
+    /**
+     * Set ErrorHandler class
+     *
+     * @param string $classname
+     * @return Zend_Controller_Front
+     */
+    public function setErrorHandler ($classname)
+    {
+        $this->_errorHandlerClass = $classname;
+    }
+    
+    /**
+     * Get ErrorHandler class
+     * 
+     * @return string classname
+     */
+    public function getErrorHandler ()
+    {
+        return $this->_errorHandlerClass;
+    }
 
     /**
      * Add or modify a parameter to use when instantiating an action controller
@@ -813,10 +841,17 @@
      */
     public function dispatch(Zend_Controller_Request_Abstract $request = null, Zend_Controller_Response_Abstract $response = null)
     {
-        if (!$this->getParam('noErrorHandler') && !$this->_plugins->hasPlugin('Zend_Controller_Plugin_ErrorHandler')) {
+        if (!$this->getParam('noErrorHandler') && !$this->_plugins->hasPlugin($this->_errorHandlerClass)) {
             // Register with stack index of 100
+            require_once 'Zend/Loader.php';
             require_once 'Zend/Controller/Plugin/ErrorHandler.php';
-            $this->_plugins->registerPlugin(new Zend_Controller_Plugin_ErrorHandler(), 100);
+            Zend_Loader::loadClass ($this->_errorHandlerClass);
+            $errorHandler = new $this->_errorHandlerClass();
+            if (!$errorHandler instanceof Zend_Controller_Plugin_ErrorHandler) {
+                require_once 'Zend/Controller/Exception.php';
+                throw new Zend_Controller_Exception('Illegal ErrorHandler type \''.get_cfg_var($errorHandler) . '\'');
+            }
+            $this->_plugins->registerPlugin($errorHandler, 100);
         }
 
         if (!$this->getParam('noViewRenderer') && !Zend_Controller_Action_HelperBroker::hasHelper('viewRenderer')) {
