Index: tests/Zend/View/Helper/Navigation/_files/expected/menu/css_active.html
===================================================================
--- tests/Zend/View/Helper/Navigation/_files/expected/menu/css_active.html	(revision 0)
+++ tests/Zend/View/Helper/Navigation/_files/expected/menu/css_active.html	(revision 0)
@@ -0,0 +1,11 @@
+<ul class="navigation">
+    <li>
+        <a href="site1">Site 1</a>
+    </li>
+    <li class="current">
+        <a href="site2">Site 2</a>
+    </li>
+    <li>
+        <a href="site3">Site 3</a>
+    </li>
+</ul>
\ No newline at end of file
Index: tests/Zend/View/Helper/Navigation/MenuTest.php
===================================================================
--- tests/Zend/View/Helper/Navigation/MenuTest.php	(revision 24447)
+++ tests/Zend/View/Helper/Navigation/MenuTest.php	(working copy)
@@ -190,6 +190,21 @@
         $expected = $this->_getExpected('menu/css.html');
         $this->assertEquals($expected, $this->_helper->render($this->_nav2));
     }
+    
+    /**
+     * @group ZF-9543
+     */
+    public function testSetActiveClass()
+    {
+        $this->_helper->setActiveClass('current');
+        
+        // Test getter
+        $this->assertEquals('current', $this->_helper->getActiveClass());
+        
+        // Test rendering
+        $expected = $this->_getExpected('menu/css_active.html');
+        $this->assertEquals($expected, $this->_helper->render($this->_nav2));
+    }
 
     public function testTranslationUsingZendTranslate()
     {
