Index: library/Zend/Oauth/Client.php
===================================================================
--- library/Zend/Oauth/Client.php	(revision 25165)
+++ library/Zend/Oauth/Client.php	(working copy)
@@ -196,10 +196,12 @@
             $this->setRequestMethod(self::POST);
         } elseif($method == self::PUT) {
             $this->setRequestMethod(self::PUT);
-        }  elseif($method == self::DELETE) {
+        } elseif($method == self::DELETE) {
             $this->setRequestMethod(self::DELETE);
-        }   elseif($method == self::HEAD) {
+        } elseif($method == self::HEAD) {
             $this->setRequestMethod(self::HEAD);
+        } elseif($method == self::OPTIONS) {
+            $this->setRequestMethod(self::OPTIONS);
         }
         return parent::setMethod($method);
     }
Index: library/Zend/Oauth/Config.php
===================================================================
--- library/Zend/Oauth/Config.php	(revision 25165)
+++ library/Zend/Oauth/Config.php	(working copy)
@@ -581,6 +581,7 @@
                 Zend_Oauth::POST,
                 Zend_Oauth::PUT,
                 Zend_Oauth::DELETE,
+                Zend_Oauth::OPTIONS,
             ))
         ) {
             require_once 'Zend/Oauth/Exception.php';
Index: library/Zend/Oauth.php
===================================================================
--- library/Zend/Oauth.php	(revision 25165)
+++ library/Zend/Oauth.php	(working copy)
@@ -38,6 +38,7 @@
     const PUT                        = 'PUT';
     const DELETE                     = 'DELETE';
     const HEAD                       = 'HEAD';
+    const OPTIONS                    = 'OPTIONS';
 
     /**
      * Singleton instance if required of the HTTP client
