Index: /library/Zend/Auth/Adapter/Http.php
===================================================================
--- /library/Zend/Auth/Adapter/Http.php	(revision 7190)
+++ /library/Zend/Auth/Adapter/Http.php	(working copy)
@@ -168,6 +168,11 @@
      */
     public function __construct(array $config)
     {
+        if (!extension_loaded('hash')) {
+            require_once 'Zend/Auth/Adapter/Exception.php';
+            throw new Zend_Auth_Adapter_Exception(__CLASS__  . ' requires the \'hash\' extension');
+        }
+        
         $this->_request  = null;
         $this->_response = null;
         $this->_ieNoOpaque = false;
