diff --git a/Zend/Application/Resource/Frontcontroller.php b/Zend/Application/Resource/Frontcontroller.php
index 0cad5c3..28d81d0 100644
--- a/Zend/Application/Resource/Frontcontroller.php
+++ b/Zend/Application/Resource/Frontcontroller.php
@@ -108,6 +108,10 @@ class Zend_Application_Resource_Frontcontroller extends Zend_Application_Resourc
                             }
                         }
 
+                        if (!class_exists($pluginClass, true)) {
+                            Zend_Loader::loadClass($pluginClass);
+                        }
+
                         $plugin = new $pluginClass();
                         $front->registerPlugin($plugin, $stackIndex);
                     }
