Index: Zend/Translate/Adapter.php
===================================================================
--- Zend/Translate/Adapter.php	(revision 10896)
+++ Zend/Translate/Adapter.php	(working copy)
@@ -284,7 +284,7 @@
         }
 
         $locale = (string) $locale;
-        if (!isset($this->_translate[$locale])) {
+        if (empty($this->_translate[$locale])) {
             $temp = explode('_', $locale);
             if (!isset($this->_translate[$temp[0]])) {
                 /**
