Index: library/Zend/XmlRpc/Client.php
===================================================================
--- library/Zend/XmlRpc/Client.php	(revision 17681)
+++ library/Zend/XmlRpc/Client.php	(working copy)
@@ -336,7 +336,7 @@
             }
         }
 
-        $request = new Zend_XmlRpc_Request($method, $params);
+        $request = $this->_createRequest($method, $params);
 
         $this->doRequest($request);
 
@@ -353,4 +353,14 @@
 
         return $this->_lastResponse->getReturnValue();
     }
+
+    /**
+     * Create request object
+     *
+     * @return Zend_XmlRpc_Request
+     */
+    protected function _createRequest($method, $params)
+    {
+        return new Zend_XmlRpc_Request($method, $params);
+    }
 }
