diff --git a/sipgate20_zend_library/Zend/Currency.php b/sipgate20_zend_library/Zend/Currency.php
index 8e25f4f..c14e8bb 100644
--- a/sipgate20_zend_library/Zend/Currency.php
+++ b/sipgate20_zend_library/Zend/Currency.php
@@ -151,6 +151,10 @@ class Zend_Currency
                                                                'number_format' => $format,
                                                               'precision'     => $options['precision']));
 
+        if($options['currency'] != $this->_options['currency']) {
+        	$options['symbol'] = $this->getSymbol($options['currency']);
+        }
+        
         if ($options['position'] !== self::STANDARD) {
             $value = str_replace('¤', '', $value);
             $space = '';
@@ -301,6 +305,10 @@ class Zend_Currency
         if (($currency === null) and ($locale === null)) {
             return $this->_options['symbol'];
         }
+        
+        if($locale === null) {
+        	$locale = $this->getLocale();
+        }
 
         $params = self::_checkParams($currency, $locale);
 
