diff -ru Amf.orig/Parse/Amf3/Serializer.php Amf.noref-writeString/Parse/Amf3/Serializer.php
--- Amf.orig/Parse/Amf3/Serializer.php	2009-11-12 10:26:19.000000000 -0600
+++ Amf.noref-writeString/Parse/Amf3/Serializer.php	2009-12-30 10:22:02.000000000 -0600
@@ -220,7 +220,7 @@
             return $this;
         }
 
-        $ref = array_search($string, $this->_referenceStrings, true);
+        $ref = false;
         if($ref === false){
             $this->_referenceStrings[] = $string;
             $this->writeBinaryString($string);
