Index: library/Zend/Application/Resource/Multidb.php
===================================================================
--- library/Zend/Application/Resource/Multidb.php	(revision 21071)
+++ library/Zend/Application/Resource/Multidb.php	(working copy)
@@ -76,19 +76,18 @@
     public function init() 
     {
         $options = $this->getOptions();
-        
         foreach ($options as $id => $params) {
         	$adapter = $params['adapter'];
             $default = isset($params['default'])?(int)$params['default']:false;
-            unset($params['adapter'], $params['default']);
+            $default = (!$default && isset($params['isDefaultTableAdapter'])) 
+            		 ? (int) $params['isDefaultTableAdapter'] 
+            		 : $default;
+              
+            unset($params['adapter'], $params['default'], $params['isDefaultTableAdapter']);
         	
             $this->_dbs[$id] = Zend_Db::factory($adapter, $params);
 
-            if ($default
-                // For consistency with the Db Resource Plugin
-                || (isset($params['isDefaultTableAdapter']) 
-                    && $params['isDefaultTableAdapter'] == true)
-            ) {
+            if ($default) {
                 $this->_setDefault($this->_dbs[$id]);
             }
         }
