Index: library/Zend/Db/Adapter/Oracle.php
===================================================================
--- library/Zend/Db/Adapter/Oracle.php	(revision 861)
+++ library/Zend/Db/Adapter/Oracle.php	(working copy)
@@ -17,7 +17,7 @@
  * @subpackage Adapter
  * @copyright  Copyright (c) 2006 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- */ 
+ */
 
 
 /** Zend_Db_Adapter_Abstract */
@@ -97,7 +97,7 @@
 
         $this->_profiler = new Zend_Db_Profiler($enabled);
     }
-	
+
     /**
      * Creates a connection resource.
      *
@@ -111,7 +111,7 @@
         if ($this->_connection) {
             return;
         }
-		
+
 		if (isset($this->_config['database'])) {
 			$this->_connection = oci_connect($this->_config['username'], $this->_config['password'], $this->_config['database']);
 		} else {
@@ -164,7 +164,7 @@
     public function listTables()
     {
         $this->_connect();
-		$data = $stmt->fetchCol('SELECT table_name FROM all_tables');
+		$data = $this->fetchCol('SELECT table_name FROM all_tables');
 		return $data;
     }
 
