--- Route.php.orig	2010-01-06 03:05:10.000000000 +0100
+++ Route.php	2010-02-21 20:02:33.000000000 +0100
@@ -129,10 +129,11 @@
         $this->_request = $request;
         $this->_setRequestKeys();
 
-        $path   = $request->getPathInfo();
-        $params = $request->getParams();
-        $values = array();
-        $path   = trim($path, self::URI_DELIMITER);
+        $path     = $request->getPathInfo();
+        $params   = $request->getParams();
+        $values   = array();
+        $path     = trim($path, self::URI_DELIMITER);
+        $pathOrig = $path;
 
         if ($path != '') {
 
@@ -223,14 +224,22 @@
             }
 
         }
-        $this->_values = $values + $params;
 
-        $result = $this->_values + $this->_defaults;
+        if ($pathOrig != '' || $this->_allRestful()) {
 
-        if ($partial && $result)
-            $this->setMatchedPath($request->getPathInfo());
+            $this->_values = $values + $params;
 
-        return $result;
+            $result = $this->_values + $this->_defaults;
+
+            if ($partial && $result) {
+                $this->setMatchedPath($request->getPathInfo());
+            }
+
+            return $result;
+
+        }
+
+        return false;
     }
 
     /**
