Index: Zend/Cache/Backend/APC.php
===================================================================
--- Zend/Cache/Backend/APC.php	(revision 1315)
+++ Zend/Cache/Backend/APC.php	(working copy)
@@ -143,7 +143,7 @@
      * @param tags array $tags array of tags
      * @return boolean true if no problem
      */
-    public function clean($mode = 'all', $tags = array()) 
+    public function clean($mode = Zend_Cache::CLEANING_MODE_ALL, $tags = array()) 
     {
         if ($mode==Zend_Cache::CLEANING_MODE_ALL) {
             return apc_clear_cache('user');
Index: Zend/Cache/Backend/File.php
===================================================================
--- Zend/Cache/Backend/File.php	(revision 1315)
+++ Zend/Cache/Backend/File.php	(working copy)
@@ -268,7 +268,7 @@
      * @param tags array $tags array of tags
      * @return boolean true if no problem
      */
-    public function clean($mode = 'all', $tags = array()) 
+    public function clean($mode = Zend_Cache::CLEANING_MODE_ALL, $tags = array()) 
     {
         // We use this private method to hide the recursive stuff
         clearstatcache();
Index: Zend/Cache/Backend/Memcached.php
===================================================================
--- Zend/Cache/Backend/Memcached.php	(revision 1315)
+++ Zend/Cache/Backend/Memcached.php	(working copy)
@@ -207,7 +207,7 @@
      * @param tags array $tags array of tags
      * @return boolean true if no problem
      */
-    public function clean($mode = 'all', $tags = array()) 
+    public function clean($mode = Zend_Cache::CLEANING_MODE_ALL, $tags = array()) 
     {
         if ($mode==Zend_Cache::CLEANING_MODE_ALL) {
             return $this->_memcache->flush();
