Index: Cache.php
===================================================================
--- Cache.php	(revision 177)
+++ Cache.php	(working copy)
@@ -133,7 +133,7 @@
             require_once str_replace('_', DIRECTORY_SEPARATOR, $backendClass) . '.php';
         } else {
             // we use a custom backend
-            if (!preg_match('~^[\w]+$~D', $backend)) {
+            if (!preg_match('~^[\w\\\\]+$~D', $backend)) {
                 Zend_Cache::throwException("Invalid backend name [$backend]");
             }
             if (!$customBackendNaming) {
@@ -143,7 +143,7 @@
                 $backendClass = $backend;
             }
             if (!$autoload) {
-                $file = str_replace('_', DIRECTORY_SEPARATOR, $backendClass) . '.php';
+                $file = str_replace(array('_', '\\'), DIRECTORY_SEPARATOR, $backendClass) . '.php';
                 if (!(self::_isReadable($file))) {
                     self::throwException("file $file not found in include_path");
                 }
