Index: Zend/Feed/Builder/Entry.php
===================================================================
--- Zend/Feed/Builder/Entry.php	(revision 27815)
+++ Zend/Feed/Builder/Entry.php	(working copy)
@@ -100,6 +100,18 @@
             $this->offsetUnset($key);
         }
     }
+    
+    /**
+     * Sets the author of the entry
+     *
+     * @param  string $author
+     * @return Zend_Feed_Builder_Entry
+     */
+    public function setAuthor($author)
+    {
+        $this->offsetSet('author', $author);
+        return $this;
+    }
 
     /**
      * Sets the id/guid of the entry
Index: Zend/Feed/Builder.php
===================================================================
--- Zend/Feed/Builder.php	(revision 27815)
+++ Zend/Feed/Builder.php	(working copy)
@@ -354,6 +354,9 @@
                 }
             }
             $entry = new Zend_Feed_Builder_Entry($row['title'], $row['link'], $row['description']);
+            if (isset($row['author'])) {
+                $entry->setAuthor($row['author']);
+            }
             if (isset($row['guid'])) {
                 $entry->setId($row['guid']);
             }
Index: Zend/Feed/Rss.php
===================================================================
--- Zend/Feed/Rss.php	(revision 27815)
+++ Zend/Feed/Rss.php	(working copy)
@@ -384,6 +384,11 @@
             $title->appendChild($this->_element->createCDATASection($dataentry->title));
             $item->appendChild($title);
 
+            if (isset($dataentry->author)) {
+                $author = $this->_element->createElement('author', $dataentry->author);
+                $item->appendChild($author);
+            }
+            
             $link = $this->_element->createElement('link', $dataentry->link);
             $item->appendChild($link);
 
