--- incubator/library/Zend/Mail.php	2006-11-18 20:37:12.000000000 +0100
+++ library/Zend/Mail.php	2006-11-18 19:30:23.000000000 +0100
@@ -183,7 +183,6 @@
     public function setMimeBoundary($boundary)
     {
       $this->_mimeBoundary = $boundary;
-      return $this;
     }
 
     /**
@@ -216,7 +215,7 @@
         $mp->charset = $charset;
 
         $this->_bodyText = $mp;
-        return $this;
+        return $mp;
     }
 
     /**
@@ -250,7 +249,7 @@
         $mp->charset = $charset;
 
         $this->_bodyHtml = $mp;
-        return $this;
+        return $mp;
     }
 
     /**
@@ -389,8 +388,6 @@
     public function addTo($email, $name='')
     {
         $this->_addRecipientAndHeader('To', $name, $email);
-
-        return $this;
     }
 
     /**
@@ -402,8 +399,6 @@
     public function addCc($email, $name='')
     {
         $this->_addRecipientAndHeader('Cc', $name, $email);
-
-        return $this;
     }
 
     /**
@@ -414,8 +409,6 @@
     public function addBcc($email)
     {
         $this->_addRecipientAndHeader('Bcc', '', $email);
-
-        return $this;
     }
 
     /**
@@ -444,8 +437,6 @@
         } else {
             throw new Zend_Mail_Exception('From Header set twice');
         }
-        
-        return $this;
     }
 
     /**
@@ -475,8 +466,6 @@
         } else {
             throw new Zend_Mail_Exception('Return-Path Header set twice');
         }
-
-        return $this;
     }
 
     /**
@@ -510,8 +499,6 @@
         } else {
             throw new Zend_Mail_Exception('Subject set twice');
         }
-
-        return $this;
     }
 
     /**
@@ -541,8 +528,6 @@
         $value = strtr($value,"\r\n\t",'???');
         $value = $this->_encodeHeader($value);
         $this->_storeHeader($name, $value, $append);
-
-        return $this;
     }
 
     /**
