--- ZF-1.7.0/tests/Zend/Session/SessionTest.php	(revision 30)
+++ ZF-1.7.0/tests/Zend/Session/SessionTest.php	(working copy)
@@ -61,13 +61,35 @@
     protected $_savePath;
 
     /**
+     * Retrieve a path to the loaded php.ini file
+     *
+     * @return string
+     */
+    private static function getPhpIniLoadedFile()
+    {
+        if (function_exists('php_ini_loaded_file')) {
+            return php_ini_loaded_file();
+        } else {
+            ob_start();
+            phpinfo(INFO_GENERAL);
+            $stuff = ob_get_clean();
+            if (preg_match('@Loaded Configuration File => (.+)@', $stuff, $climatch)) {
+                return $climatch[1];
+            } elseif (preg_match('@(?<="v">).+php\.ini@', $stuff, $htmlmatch)) {
+                return $htmlmatch[0];
+            }
+        }
+    }
+
+
+    /**
      * Initializes instance data
      *
      * @return void
      */
     public function __construct()
     {
-        $this->_script = 'php -c \'' . php_ini_loaded_file() . '\' '
+        $this->_script = 'php -c \'' . self::getPhpIniLoadedFile() . '\' '
             . escapeshellarg(dirname(__FILE__) . DIRECTORY_SEPARATOR . 'SessionTestHelper.php');
 
         $this->_savePath = ini_get('session.save_path');
