Index: ControllerTestCaseTest.php
===================================================================
--- ControllerTestCaseTest.php	(revision 18161)
+++ ControllerTestCaseTest.php	(working copy)
@@ -634,12 +634,15 @@
      */
     public function testResetRequestShouldClearRequestObject()
     {
-        $this->testCase->getFrontController()->setControllerDirectory(dirname(__FILE__) . '/_files/application/controllers');
-        $this->testCase->dispatch('/zend-test-php-unit-foo/baz');
         $request = $this->testCase->getRequest();
+
+        $postArray = array('foo'=>'bar');
+        $request->setPost($postArray);
+
         $this->testCase->resetRequest();
         $test = $this->testCase->getRequest();
         $this->assertNotSame($request, $test);
+        $this->assertNotEquals($postArray, $test->getPost());
     }
 
     public function testResetResponseShouldClearAllViewPlaceholders()
