+ Fix for Zend PriorityQueue clone

diff -r fca982957171 -r d2cfbab57895 vendor/Zend/Stdlib/PriorityQueue.php
--- a/vendor/Zend/Stdlib/PriorityQueue.php	Thu Jan 19 19:35:48 2012 +0200
+++ b/vendor/Zend/Stdlib/PriorityQueue.php	Sat Jan 21 19:03:41 2012 +0200
@@ -302,4 +302,15 @@
         }
         return $this->queue;
     }
+
+    /**
+     * Clone handler
+     * @return void
+     */
+    public function __clone()
+    {
+        if ($this->queue !== null) {
+            $this->queue = clone $this->queue;
+        }
+    }
 }
