Index: tests/Zend/Loader/Autoloader/ResourceTest.php
===================================================================
--- tests/Zend/Loader/Autoloader/ResourceTest.php	(revision 23558)
+++ tests/Zend/Loader/Autoloader/ResourceTest.php	(working copy)
@@ -418,6 +418,32 @@
         ));
         $path = $this->loader->autoload('Something_Totally_Wrong');
         $this->assertFalse($path);
+    }    
+    
+    /**
+     * @group ZF-10852
+     * @expectedException Zend_Loader_Exception
+     */
+    public function testAutoloaderResourceConstructorThrowsExceptionOnMissingNamespaceInResourceType()
+    {
+        $data = array(
+            'basePath'      => 'path/to/some/directory',
+            'namespace'     => 'My',
+            'resourceTypes' => array(
+                'acl' => array(
+                    'path'      => 'acls/',
+                    'namespace' => 'Acl',
+                ),
+                'form' => array(
+                    'path'      => 'forms/',
+                    'namespace' => 'Form',
+                ),
+                'model' => array(
+                    'path'      => 'models/',
+                )
+            )
+        );
+        $zlar1 = new Zend_Loader_Autoloader_Resource($data);
     }
 }
 
