Index: library/Zend/Soap/Server.php
===================================================================
--- library/Zend/Soap/Server.php	(Revision 24046)
+++ library/Zend/Soap/Server.php	(Arbeitskopie)
@@ -829,7 +829,7 @@
             $soap->fault("Sender", $setRequestException->getMessage());
         } else {
             try {
-                $soap->handle($request);
+                $soap->handle($this->_request);
             } catch (Exception $e) {
                 $fault = $this->fault($e);
                 $soap->fault($fault->faultcode, $fault->faultstring);
