Index: library/Zend/Config/Ini.php
===================================================================
--- library/Zend/Config/Ini.php	(revision 24270)
+++ library/Zend/Config/Ini.php	(working copy)
@@ -129,10 +129,11 @@
             // Load entire file
             $dataArray = array();
             foreach ($iniArray as $sectionName => $sectionData) {
-                if(!is_array($sectionData)) {
-                    $dataArray = $this->_arrayMergeRecursive($dataArray, $this->_processKey(array(), $sectionName, $sectionData));
+                if(is_array($sectionData) && array_key_exists(';extends', $sectionData)) {
+                    $dataArray[$sectionName] = $this->_processSection($iniArray, $sectionName);
                 } else {
-                    $dataArray[$sectionName] = $this->_processSection($iniArray, $sectionName);
+                    $dataArray = $this->_arrayMergeRecursive($dataArray,
+                                                             $this->_processKey(array(), $sectionName, $sectionData));
                 }
             }
             parent::__construct($dataArray, $allowModifications);
