Index: library/Zend/Tool/Project/Context/Zf/ViewScriptFile.php
===================================================================
--- library/Zend/Tool/Project/Context/Zf/ViewScriptFile.php	(revision 15883)
+++ library/Zend/Tool/Project/Context/Zf/ViewScriptFile.php	(working copy)
@@ -146,7 +146,7 @@
 </html>
             
 EOS;
-        } elseif ($this->_forActionName == 'index' && $this->_resource->getParentResource()->getAttribute('forControllerName') == 'index') {
+        } elseif ($this->_forActionName == 'index' && $this->_resource->getParentResource()->getAttribute('forControllerName') == 'index' && !$this->_resource->getParentResource()->getAttribute('forModuleName')) {
             
             $contents =<<<EOS
 <style>
Index: library/Zend/Tool/Project/Provider/View.php
===================================================================
--- library/Zend/Tool/Project/Provider/View.php	(revision 15883)
+++ library/Zend/Tool/Project/Provider/View.php	(working copy)
@@ -76,7 +76,7 @@
         
         // XXXXXXXXX below is failing b/c of above search params
         if (($viewControllerScriptsDirectory = $viewScriptsDirectory->search($profileSearchParams)) === false) {
-            $viewControllerScriptsDirectory = $viewScriptsDirectory->createResource('viewControllerScriptsDirectory', array('forControllerName' => $controllerName));
+            $viewControllerScriptsDirectory = $viewScriptsDirectory->createResource('viewControllerScriptsDirectory', array('forControllerName' => $controllerName, 'forModuleName' => $moduleName));
         }
         
         $newViewScriptFile = $viewControllerScriptsDirectory->createResource('ViewScriptFile', array('forActionName' => $actionName));
