Index: Float.php
===================================================================
--- Float.php	(wersja 11344)
+++ Float.php	(kopia robocza)
@@ -64,7 +64,7 @@
         $valueFiltered = str_replace($locale['thousands_sep'], '', $valueString);
         $valueFiltered = str_replace($locale['decimal_point'], '.', $valueFiltered);
 
-        if (strval(floatval($valueFiltered)) != $valueFiltered) {
+        if (strval(floatval($valueFiltered)) != $valueString) {
             $this->_error();
             return false;
         }
