Index: InfoCard.php
===================================================================
--- InfoCard.php	(revision 10697)
+++ InfoCard.php	(working copy)
@@ -138,8 +138,8 @@
      */
     public function getAdapter()
     {
-        if(is_null($this->_adapter)) {
-            Zend_Loader::loadClass('Zend_InfoCard_Adapter_Default');
+        if(is_null($this->_adapter)) {
+            require_once 'Zend/InfoCard/Adapter/Default.php';
             $this->setAdapter(new Zend_InfoCard_Adapter_Default());
         }
 
