Index: tests/Zend/Cache/ManagerTest.php
===================================================================
--- tests/Zend/Cache/ManagerTest.php	(revision 21065)
+++ tests/Zend/Cache/ManagerTest.php	(working copy)
@@ -18,6 +18,11 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  * @version    $Id$
  */
+
+/**
+ * Test helper
+ */
+require_once dirname(__FILE__) . '/../../TestHelper.php';
  
 require_once 'Zend/Cache.php';
 require_once 'Zend/Cache/Manager.php';
@@ -58,20 +63,20 @@
     public function testLazyLoadsDefaultPageCache()
     {
         $manager = new Zend_Cache_Manager;
-        $manager->setTemplateOptions('tagCache',array(
+        $manager->setTemplateOptions('pagetag',array(
             'backend' => array(
                 'options' => array(
                     'cache_dir' => $this->_cache_dir
                 )
             )
         ));
-        $this->assertTrue($manager->getCache('page') instanceof Zend_Cache_Frontend_Output);
+        $this->assertTrue($manager->getCache('page') instanceof Zend_Cache_Frontend_Capture);
     }
 
     public function testCanOverrideCacheFrontendNameConfiguration()
     {
         $manager = new Zend_Cache_Manager;
-        $manager->setTemplateOptions('tagCache',array(
+        $manager->setTemplateOptions('pagetag',array(
             'backend' => array(
                 'options' => array(
                     'cache_dir' => $this->_cache_dir
@@ -96,15 +101,15 @@
                 )
             )
         ));
-        $manager->setTemplateOptions('tagCache', $config);
-        $options = $manager->getCacheTemplate('tagCache');
+        $manager->setTemplateOptions('pagetag', $config);
+        $options = $manager->getCacheTemplate('pagetag');
         $this->assertEquals($this->_cache_dir, $options['backend']['options']['cache_dir']);
     }
 
     public function testCanOverrideCacheBackendendNameConfiguration()
     {
         $manager = new Zend_Cache_Manager;
-        $manager->setTemplateOptions('tagCache',array(
+        $manager->setTemplateOptions('pagetag',array(
             'backend' => array(
                 'options' => array(
                     'cache_dir' => $this->_cache_dir
@@ -216,9 +221,9 @@
     public function testGettingPageCacheAlsoCreatesTagCache()
     {
         $manager = new Zend_Cache_Manager;
-        $tagCacheConfig = $manager->getCacheTemplate('tagCache');
+        $tagCacheConfig = $manager->getCacheTemplate('pagetag');
         $tagCacheConfig['backend']['options']['cache_dir'] = $this->getTmpDir();
-        $manager->setCacheTemplate('tagCache', $tagCacheConfig);
+        $manager->setCacheTemplate('pagetag', $tagCacheConfig);
         $tagCache = $manager->getCache('page')->getBackend()->getOption('tag_cache');
         $this->assertTrue($tagCache instanceof Zend_Cache_Core);
     }
