Index: File.php
===================================================================
--- File.php	(revision 16773)
+++ File.php	(working copy)
@@ -266,7 +266,7 @@
             return $wordLow;
         }
 
-        return $wordHigh*(float)0x100000000/* 0x00000001 00000000 */ + $wordLow;
+        return $wordHigh*(float)0x10000000*(float)0x10/* 0x00000001 00000000 */ + $wordLow;
     }
 
 
@@ -287,8 +287,8 @@
             $wordHigh = (int)0xFFFFFFFF;
             $wordLow  = (int)$value;
         } else {
-            $wordHigh = (int)($value/(float)0x100000000/* 0x00000001 00000000 */);
-            $wordLow  = $value - $wordHigh*(float)0x100000000/* 0x00000001 00000000 */;
+            $wordHigh = (int)($value/((float)0x10000000*(float)0x10)/* 0x00000001 00000000 */);
+            $wordLow  = $value - $wordHigh*(float)0x10000000*(float)0x10/* 0x00000001 00000000 */;
 
             if ($wordLow > 0x7FFFFFFF) {
                 // Highest bit of low word is set. Translate it to the corresponding negative integer value
Index: File/Memory.php
===================================================================
--- File/Memory.php	(revision 16773)
+++ File/Memory.php	(working copy)
@@ -378,7 +378,7 @@
             return $wordLow;
         }
 
-        return $wordHigh*(float)0x100000000/* 0x00000001 00000000 */ + $wordLow;
+        return $wordHigh*(float)0x10000000*(float)0x10/* 0x00000001 00000000 */ + $wordLow;
     }
 
 
@@ -399,8 +399,8 @@
             $wordHigh = (int)0xFFFFFFFF;
             $wordLow  = (int)$value;
         } else {
-            $wordHigh = (int)($value/(float)0x100000000/* 0x00000001 00000000 */);
-            $wordLow  = $value - $wordHigh*(float)0x100000000/* 0x00000001 00000000 */;
+            $wordHigh = (int)($value/((float)0x10000000*(float)0x10)/* 0x00000001 00000000 */);
+            $wordLow  = $value - $wordHigh*(float)0x10000000*(float)0x10/* 0x00000001 00000000 */;
 
             if ($wordLow > 0x7FFFFFFF) {
                 // Highest bit of low word is set. Translate it to the corresponding negative integer value
