Index: Zend/Locale.php
===================================================================
--- Zend/Locale.php	(revision 10525)
+++ Zend/Locale.php	(working copy)
@@ -240,20 +240,20 @@
         if ($searchorder === self::ENVIRONMENT) {
             $languages = $this->getEnvironment();
             if ((empty($languages) === true) or ($fastsearch === false)) {
-                $languages = $this->getFramework() + $languages;
-                $languages = $this->getBrowser() + $languages;
+                $languages += $this->getFramework();
+                $languages += $this->getBrowser();
             }
         } else if ($searchorder === self::FRAMEWORK) {
             $languages = $this->getFramework();
             if ((empty($languages) === true) or ($fastsearch === false)) {
-                $languages = $this->getEnvironment() + $languages;
-                $languages = $this->getBrowser() + $languages;
+                $languages += $this->getEnvironment();
+                $languages += $this->getBrowser();
             }
         } else {
             $languages = $this->getBrowser();
             if ((empty($languages) === true) or ($fastsearch === false)) {
-                $languages = $this->getEnvironment() + $languages;
-                $languages = $this->getFramework() + $languages;
+                $languages += $this->getEnvironment();
+                $languages += $this->getFramework();
             }
         }
 
