--- tests/ZendX/Application/Resource/JqueryTest.php	(revision 23775)
+++ tests/ZendX/Application/Resource/JqueryTest.php	(working copy)
@@ -127,7 +127,7 @@
         $this->assertEquals('/foo/bar/', $res->getLocalPath());
         $this->assertEquals('2.3.4', $res->getUiVersion());
         $this->assertEquals('/bar/foo/', $res->getUiLocalPath());
-//        $this->assertTrue($res->getCdnSsl()); // Not implemented yet
+        $this->assertTrue($res->getCdnSsl());
         $this->assertEquals(192, $res->getRenderMode());
         $this->assertEquals(array('/fooBar.css', 'johndoe.css', 'janedoe.css'),
                             $res->getStylesheets());
