Index: Request/Http.php
===================================================================
--- Request/Http.php	(revision 12339)
+++ Request/Http.php	(working copy)
@@ -549,6 +549,10 @@
 
             if (basename($baseUrl) === $filename) {
                 $basePath = dirname($baseUrl);
+                /* fix for windows, relies on $_SERVER['WINDIR'] being set on no other OS than windows */
+                if( isset($_SERVER['WINDIR']) ){
+                	$basePath = str_replace('\\','/',$basePath);
+                }
             } else {
                 $basePath = $baseUrl;
             }
@@ -569,7 +573,6 @@
         if (null === $this->_basePath) {
             $this->setBasePath();
         }
-
         return $this->_basePath;
     }
 
