--- trunk/library/Zend/Soap/AutoDiscover.php	2008-11-06 19:56:55.000000000 +0100
+++ tomdev/library/Zend/Soap/AutoDiscover.php	2008-11-06 20:01:28.000000000 +0100
@@ -117,7 +117,7 @@
                 $schema = 'https';
             }
             $host = (isset($_SERVER['HTTP_HOST'])?$_SERVER['HTTP_HOST']:$_SERVER['SERVER_NAME']);
-            $uri = Zend_Uri::factory($schema . '://' . $host . $_SERVER['SCRIPT_NAME']);
+            $uri = Zend_Uri::factory($schema . '://' . $host . $_SERVER['REQUEST_URI']);
             $this->setUri($uri);
         }
         return $uri;
