Index: tests/Zend/Log/Formatter/SimpleTest.php
===================================================================
--- tests/Zend/Log/Formatter/SimpleTest.php	(revision 22824)
+++ tests/Zend/Log/Formatter/SimpleTest.php	(working copy)
@@ -104,6 +104,27 @@
         $line = $f->format($fields);
         $this->assertContains('object', $line);
     }
+
+    /**
+     * @group ZF-103xx
+     */
+    public function testDefaultFormatShouldDisplayExtraInformations()
+    {
+    	$message = 'custom message';
+    	$exception = new RuntimeException($message);
+    	$event = array(
+    	    'timestamp' => date('c'),
+    		'message' => 'Application error',
+    	    'priority' => 2,
+    	    'priorityName' => 'CRIT',
+    	    'info' => $exception,
+    	);
+
+        $formatter = new Zend_Log_Formatter_Simple();
+        $ouput = $formatter->format($event);
+
+        $this->assertContains($message, $ouput);
+    }
 }
 
 class Zend_Log_Formatter_SimpleTest_TestObject1 {
@@ -115,4 +136,4 @@
 }
 
 class Zend_Log_Formatter_SimpleTest_TestObject2 {
-}
+}
\ No newline at end of file
