--- orig-Yahoo.php      Wed Jul 12 16:59:53 2006
+++ Yahoo.php   Thu Jul 13 16:37:59 2006
@@ -150,10 +150,6 @@

         $this->_validateImageSearch($options);

-        $this->_rest = new Zend_Service_Rest;
-
-        $this->_rest->setUri('api.search.yahoo.com');
-
         $response = $this->_rest->restGet('/ImageSearchService/V1/imageSearch',
 $options);

         if ($response->isError()) {
@@ -207,7 +203,7 @@

         $this->_validateLocalSearch($options);

-        $this->_uri->setHost('api.local.yahoo.com');
+        $this->_rest->setUri('http://api.local.yahoo.com');

         $response = $this->_rest->restGet('/LocalSearchService/V1/localSearch',
 $options);

@@ -250,8 +246,6 @@

         $this->_validateNewsSearch($options);

-        $this->_uri->setHost('api.search.yahoo.com');
-
         $response = $this->_rest->restGet('/NewsSearchService/V1/newsSearch', $
options);

         if ($response->isError()) {
@@ -298,8 +292,6 @@

         $options = $this->_prepareOptions($query, $options, $default_options);
         $this->_validateWebSearch($options);
-
-        $this->_uri->setHost('api.search.yahoo.com');

         $response = $this->_rest->restGet('/WebSearchService/V1/webSearch', $op
tions);

