Index: library/Zend/Oauth/Client.php
===================================================================
--- library/Zend/Oauth/Client.php	(revision 11201)
+++ library/Zend/Oauth/Client.php	(working copy)
@@ -156,9 +156,18 @@
             $this->setRawData($raw);
             $this->paramsPost = array();
         } elseif ($requestScheme == Zend_Oauth::REQUEST_SCHEME_QUERYSTRING) {
-            $this->getUri()->setQuery('');
+            $params = array();
+            $query = $this->getUri()->getQuery();
+            if ($query) {
+                $queryParts = split('&', $this->getUri()->getQuery());
+                foreach ($queryParts as $queryPart) {
+                    $kvTuple = split('=', $queryPart);
+                    $params[$kvTuple[0]] = 
+                        (array_key_exists(1, $kvTuple) ? $kvTuple[1] : NULL);
+                }
+            }
             $query = $this->getToken()->toQueryString(
-                $this->getUri(true), $this->_config, $this->paramsGet
+                $this->getUri(true), $this->_config, $params
             );
             $this->getUri()->setQuery($query);
             $this->paramsGet = array();
