--- Rewrite.php.orig	2009-06-20 19:57:43.000000000 +0200
+++ Rewrite.php	2009-06-22 21:01:30.000000000 +0200
@@ -66,6 +66,13 @@
     protected $_globalParams = array();
     
     /**
+     * Separator to use with chain names
+     * 
+     * @var string
+     */
+    protected $_chainNameSeparator = '-';
+    
+    /**
      * Add default routes which are used to mimic basic router behaviour
      * 
      * @return Zend_Controller_Router_Rewrite
@@ -235,7 +242,7 @@
                 $chainRoute = $route->chain($childRoute);
             }
             
-            $chainName = $name . '-' . $childRouteName;
+            $chainName = $name . $this->_chainNameSeparator . $childRouteName;
             
             if (isset($childRouteInfo->chains)) {
                 $this->_addChainRoutesFromConfig($chainName, $chainRoute, $childRouteInfo->chains);
@@ -452,4 +459,25 @@
     
         return $this;
     }
+    
+    /**
+     * Set the separator to use with chain names
+     * 
+     * @param  string $separator The separator to use
+     * @return Zend_Controller_Router_Rewrite
+     */
+    public function setChainNameSeparator($separator) {
+    	$this->_chainNameSeparator = $separator;
+    	
+    	return $this;
+    }
+    
+    /**
+     * Get the separator to use for chain names
+     * 
+     * @return string
+     */
+    public function getChainNameSeparator() {
+        return $this->_chainNameSeparator;
+    }
 }
