Index: StaticRoute.php
===================================================================
--- StaticRoute.php	(revision 2914)
+++ StaticRoute.php	(working copy)
@@ -54,11 +54,15 @@
      * Matches a user submitted path with a previously defined route. 
      * Assigns and returns an array of defaults on a successful match.  
      *
-     * @param string Path used to match against this routing map 
+     * @param Zend_Controller_Request_Http Request used to match against this 
+     * routing map
      * @return array|false An array of assigned values or a false on a mismatch
      */
-    public function match($path)
+    public function match(Zend_Controller_Request_Http $request)
     {
+        // This matcher only needs the path info
+        $path = $request->getPathInfo();
+
         if (trim($path, '/') == $this->_route) {
             return $this->_defaults;
         }
