Index: Controller/Action/HelperBroker.php
===================================================================
--- Controller/Action/HelperBroker.php	(revision 5182)
+++ Controller/Action/HelperBroker.php	(working copy)
@@ -232,6 +232,17 @@
 
         throw new Zend_Controller_Action_Exception('Action helper "' . $name . '" has not been registered with the helper broker');
     }
+
+    /**
+     * return existing helpers
+     *
+     * @return array
+     */
+    public static function getExistingHelpers()
+    {
+        return self::$_helpers;
+    }
+
      /**
      * Is a particular helper loaded in the broker?
      * 
