Index: ArrayOfTypeComplex.php
===================================================================
--- ArrayOfTypeComplex.php	(revision 20686)
+++ ArrayOfTypeComplex.php	(working copy)
@@ -48,7 +48,7 @@
     {
         if(in_array($type, $this->_inProcess)) {
             require_once "Zend/Soap/Wsdl/Exception.php";
-            throw new Zend_Soap_Wsdl_Exception("Infinite recursion, cannot nest '".$type."' into itsself.");
+            throw new Zend_Soap_Wsdl_Exception("Infinite recursion, cannot nest '".$type."' into itself.");
         }
         $this->_inProcess[$type] = $type;
 
