Index: library/Zend/Loader/ClassMapAutoloader.php
===================================================================
--- library/Zend/Loader/ClassMapAutoloader.php	(revision 24790)
+++ library/Zend/Loader/ClassMapAutoloader.php	(working copy)
@@ -204,16 +204,17 @@
      */
     public static function realPharPath($path)
     {
-        if (strpos($path, 'phar:///') !== 0) {
+        if (strpos($path, 'phar://') !== 0) {
             return;
         }
         
-        $parts = explode('/', str_replace(array('/','\\'), '/', substr($path, 8)));
+        // 7 is the length of "phar://"
+        $parts = explode('/', str_replace(array('/','\\'), '/', substr($path, 7)));
         $parts = array_values(array_filter($parts, array(__CLASS__, 'concatPharParts')));
 
         array_walk($parts, array(__CLASS__, 'resolvePharParentPath'), $parts);
 
-        if (file_exists($realPath = 'phar:///' . implode('/', $parts))) {
+        if (file_exists($realPath = 'phar://' . implode('/', $parts))) {
             return $realPath;
         }
     }
