--- incubator/library/Zend/Mail/Imap.php	(revision 2961)
+++ incubator/library/Zend/Mail/Imap.php	(working copy)
@@ -142,7 +142,7 @@
             try {
                 $this->selectFolder('INBOX');
             } catch(Zend_Mail_Exception $e) {
-                throw Zend_Mail_Exception('cannot select INBOX, is this a valid transport?');
+                throw new Zend_Mail_Exception('cannot select INBOX, is this a valid transport?');
             }
             return;
         }
