Index: Html.php
===================================================================
--- Html.php	(revision 20162)
+++ Html.php	(working copy)
@@ -157,6 +157,16 @@
         }
         $this->_links = array_unique($this->_links);
 
+        $linkNodes = $this->_doc->getElementsByTagName('area');
+        foreach ($linkNodes as $linkNode) {
+            if (($href = $linkNode->getAttribute('href')) != '' &&
+                (!self::$_excludeNoFollowLinks  ||  strtolower($linkNode->getAttribute('rel')) != 'nofollow' )
+               ) {
+                $this->_links[] = $href;
+            }
+        }
+        $this->_links = array_unique($this->_links);
+
         $linkNodes = $xpath->query('/html/head/link');
         foreach ($linkNodes as $linkNode) {
             if (($href = $linkNode->getAttribute('href')) != '') {
