Index: /home/jordan/workspace/zend-framework/library/Zend/Controller/Dispatcher/Standard.php
===================================================================
--- /home/jordan/workspace/zend-framework/library/Zend/Controller/Dispatcher/Standard.php	(revision 7102)
+++ /home/jordan/workspace/zend-framework/library/Zend/Controller/Dispatcher/Standard.php	(working copy)
@@ -348,7 +348,7 @@
     public function isValidModule($module)
     {
         $controllerDir = $this->getControllerDirectory();
-        return ((null !== $module) && isset($controllerDir[$module]));
+        return (is_string($module) && isset($controllerDir[$module]));
     }
 
     /**
