Index: demos/Zend/Gdata/Calendar.php
===================================================================
--- demos/Zend/Gdata/Calendar.php	(revision 6539)
+++ demos/Zend/Gdata/Calendar.php	(working copy)
@@ -569,7 +569,7 @@
         $reminder = $gc->newReminder();
         $reminder->setMinutes($minutes);
         $reminder->setMethod($method);
-        $when->reminder = array($reminder);
+        $when->reminders = array($reminder);
     }
     $eventNew = $event->save();
     return $eventNew;
