Index: Client.php
===================================================================
--- Client.php	(revision 7379)
+++ Client.php	(working copy)
@@ -117,7 +117,9 @@
          * Get the HTTP client and configure it for the endpoint URI.  Do this each time
          * because the Zend_Http_Client instance is shared among all Zend_Service_Abstract subclasses.
          */
-        self::getHttpClient()->setUri($this->_uri);
+        self::getHttpClient()
+            ->resetParameters()
+            ->setUri($this->_uri);
     }
 
     /**
