--- library/Zend/Service/Rackspace/Files/ObjectList.php	(revision 25005)
+++ library/Zend/Service/Rackspace/Files/ObjectList.php	(working copy)
@@ -68,7 +68,7 @@
             require_once 'Zend/Service/Rackspace/Files/Exception.php';
             throw new Zend_Service_Rackspace_Files_Exception("You must pass a Zend_Service_Rackspace_Files object");
         }
-        if (empty($list)) {
+        if (!is_array($list)) {
             require_once 'Zend/Service/Rackspace/Files/Exception.php';
             throw new Zend_Service_Rackspace_Files_Exception("You must pass an array of data objects");
         }
