--- library/Zend/View/Helper/FormErrors.php	(révision 9723)
+++ library/Zend/View/Helper/FormErrors.php	(copie de travail)
@@ -45,7 +45,7 @@
      * @var string Element block start/end tags and separator
      */
     protected $_htmlElementEnd       = '</li></ul>';
-    protected $_htmlElementStart     = '<ul class="errors"%s><li>';
+    protected $_htmlElementStart     = '<ul%s><li>';
     protected $_htmlElementSeparator = '</li><li>';
     /**#@-*/
 
@@ -63,6 +63,10 @@
             $escape = (bool) $options['escape'];
             unset($options['escape']);
         }
+        
+        if (empty($options['class'])) {
+            $options['class'] = 'errors';
+        }
 
         $start = $this->getElementStart();
         if (strstr($start, '%s')) {
