Index: tests/Zend/Db/Table/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/TestCommon.php	(revision 12791)
+++ tests/Zend/Db/Table/TestCommon.php	(working copy)
@@ -835,6 +835,29 @@
         $this->assertEquals(2, $count);
     }
 
+    public function testTableUpdateBindArray()
+    {
+        $bug_id = $this->_db->quoteIdentifier('bug_id', true);
+        $bug_description = $this->_db->foldCase('bug_description');
+        $bug_status      = $this->_db->foldCase('bug_status');
+        $data = array(
+            $bug_description => 'Implement Do What I Mean function',
+            $bug_status      => 'INCOMPLETE'
+        );
+        $table = $this->_table['bugs'];
+        $result = $table->update($data, "$bug_id = ?", array(2));
+        $this->assertEquals(1, $result);
+
+        // Query the row to see if we have the new values.
+        $rowset = $table->find(2);
+        $this->assertEquals(1, count($rowset), "Expecting rowset count to be 1");
+        $row = $rowset->current();
+        $bug_id = $this->_db->foldCase('bug_id');
+        $this->assertEquals(2, $row->$bug_id, "Expecting row->bug_id to be 2");
+        $this->assertEquals($data[$bug_description], $row->$bug_description);
+        $this->assertEquals($data[$bug_status], $row->$bug_status);
+    }
+
     public function testTableDelete()
     {
         $bug_id = $this->_db->quoteIdentifier('bug_id', true);
@@ -884,6 +907,20 @@
         $this->assertEquals(0, count($this->_table['bugs']->find(array(1, 3))));
     }
 
+    public function testTableDeleteBindArray()
+    {
+        $bug_id = $this->_db->quoteIdentifier('bug_id', true);
+
+        $table = $this->_table['bugs'];
+        $rowset = $table->find(array(1, 2));
+        $this->assertEquals(2, count($rowset));
+
+        $table->delete("$bug_id = ?", array(2));
+
+        $rowset = $table->find(array(1, 2));
+        $this->assertEquals(1, count($rowset));
+    }
+
     public function testTableCreateRow()
     {
         $table = $this->_table['bugs'];
Index: library/Zend/Db/Table/Abstract.php
===================================================================
--- library/Zend/Db/Table/Abstract.php	(revision 12791)
+++ library/Zend/Db/Table/Abstract.php	(working copy)
@@ -950,12 +950,13 @@
      *
      * @param  array        $data  Column-value pairs.
      * @param  array|string $where An SQL WHERE clause, or an array of SQL WHERE clauses.
+     * @param  mixed        $params  An array of data to bind to the placeholders.
      * @return int          The number of rows updated.
      */
-    public function update(array $data, $where)
+    public function update(array $data, $where, $bind = array())
     {
         $tableSpec = ($this->_schema ? $this->_schema . '.' : '') . $this->_name;
-        return $this->_db->update($tableSpec, $data, $where);
+        return $this->_db->update($tableSpec, $data, $where, $bind);
     }
 
     /**
@@ -1002,12 +1003,13 @@
      * Deletes existing rows.
      *
      * @param  array|string $where SQL WHERE clause(s).
+     * @param  mixed        $params  An array of data to bind to the placeholders.
      * @return int          The number of rows deleted.
      */
-    public function delete($where)
+    public function delete($where, $bind = array())
     {
         $tableSpec = ($this->_schema ? $this->_schema . '.' : '') . $this->_name;
-        return $this->_db->delete($tableSpec, $where);
+        return $this->_db->delete($tableSpec, $where, $bind);
     }
 
     /**

