Index: StripTags.php
===================================================================
--- StripTags.php	(revision 20339)
+++ StripTags.php	(working copy)
@@ -1,4 +1,5 @@
 <?php
-
 /**
  * Zend Framework
  *
@@ -35,6 +36,11 @@
 class Zend_Filter_StripTags implements Zend_Filter_Interface
 {
     /**
-     * Unique ID prefix used for allowing comments
-     */
-    const UNIQUE_ID_PREFIX = '__Zend_Filter_StripTags__';
-
-    /**
      * Whether comments are allowed
      *
      * If false (the default), then comments are removed from the input string.
@@ -232,7 +238,7 @@
     public function filter($value)
     {
         // Strip HTML comments first
+        $valueCopy = preg_replace('#<!--(?:[^<]+|<(?!\!--))*?(--\s*>)#s', '', (string) $value);
-        $valueCopy = preg_replace('#<!--(?:[^<]+|<(?!\!--))*?(--\s*>)#us', '', (string) $value);

         // Initialize accumulator for filtered data
         $dataFiltered = '';
Index: StripTagsTest.php
===================================================================
--- StripTagsTest.php	(revision 20329)
+++ StripTagsTest.php	(working copy)
@@ -433,8 +433,8 @@
     }

     /**
+     * Ensures expected behavior when comments are marked as allowed (in our
+     * case, this should have no effect) and a comment contains tags and
-     * Ensures expected behavior when comments are marked as allowed (in our
-     * case, this should have no effect) and a comment contains tags and
      * linebreaks
      *
      * @group ZF-8473
@@ -523,36 +523,6 @@
         $filtered = $this->_filter->filter($input);
         $this->assertNotContains('onclick', $filtered);
     }
+
+    /**
+     * @ZF-8828
+     */
+    public function testFilterIsoChars()
+    {
+        $input    = 'äöü<!-- a comment -->äöü';
+        $expected = 'äöüäöü';
+        $this->assertEquals($expected, $this->_filter->filter($input));
+
+        $input    = 'äöü<!-- a comment -->äöü';
+        $input    = iconv("UTF-8", "ISO-8859-1", $input);
+        $output   = $this->_filter->filter($input);
+        $this->assertFalse(empty($output));
+    }
+
+    /**
+     * @ZF-8828
+     */
+    public function testFilterIsoCharsInComment()
+    {
+        $input    = 'äöü<!--üßüßüß-->äöü';
+        $expected = 'äöüäöü';
+        $this->assertEquals($expected, $this->_filter->filter($input));
+
+        $input    = 'äöü<!-- a comment -->äöü';
+        $input    = iconv("UTF-8", "ISO-8859-1", $input);
+        $output   = $this->_filter->filter($input);
+        $this->assertFalse(empty($output));
+    }
 }

 // Call Zend_Filter_StripTagsTest::main() if this source file is executed directly.
