Index: ../MB_ZF/Zend/Db/Adapter/Mysqli.php
===================================================================
--- ../MB_ZF/Zend/Db/Adapter/Mysqli.php	(revision )
+++ ../MB_ZF/Zend/Db/Adapter/Mysqli.php	(revision )
@@ -279,7 +279,7 @@
      */
     protected function _connect()
     {
-        if ($this->_connection) {
+        if (mysqli_ping($this->_connection) !== FALSE) {
             return;
         }
 
@@ -345,7 +345,7 @@
      */
     public function isConnected()
     {
-        return ((bool) ($this->_connection instanceof mysqli));
+        return ((bool) (($this->_connection instanceof mysqli) && (mysqli_ping($this->_connection) !== FALSE)));
     }
 
     /**
