diff -B -u -r ZendFramework-1.10.6-minimal-clean/library/Zend/Feed.php ZendFramework-1.10.6-minimal/library/Zend/Feed.php
--- ZendFramework-1.10.6-minimal-clean/library/Zend/Feed.php	2010-01-06 03:05:09.000000000 +0100
+++ ZendFramework-1.10.6-minimal/library/Zend/Feed.php	2010-07-07 11:21:42.000000000 +0200
@@ -84,7 +84,8 @@
             /**
              * @see Zend_Http_Client
              */
-            require_once 'Zend/Http/Client.php';
+            require_once 'Zend/Loader.php';
+            Zend_Loader::loadClass('Zend_Http_Client');
             self::$_httpClient = new Zend_Http_Client();
         }
 
diff -B -u -r ZendFramework-1.10.6-minimal-clean/library/Zend/Http/Client.php ZendFramework-1.10.6-minimal/library/Zend/Http/Client.php
--- ZendFramework-1.10.6-minimal-clean/library/Zend/Http/Client.php	2010-04-19 20:44:26.000000000 +0200
+++ ZendFramework-1.10.6-minimal/library/Zend/Http/Client.php	2010-07-07 17:11:18.000000000 +0200
@@ -585,9 +586,7 @@
      */
     public function setCookieJar($cookiejar = true)
     {
-        if (! class_exists('Zend_Http_CookieJar')) {
-            require_once 'Zend/Http/CookieJar.php';
-        }
+        Zend_Loader::loadClass('Zend_Http_CookieJar');
 
         if ($cookiejar instanceof Zend_Http_CookieJar) {
             $this->cookiejar = $cookiejar;
@@ -625,9 +624,7 @@
      */
     public function setCookie($cookie, $value = null)
     {
-        if (! class_exists('Zend_Http_Cookie')) {
-            require_once 'Zend/Http/Cookie.php';
-        }
+        Zend_Loader::loadClass('Zend_Http_Cookie');
 
         if (is_array($cookie)) {
             foreach ($cookie as $c => $v) {
diff -B -u -r ZendFramework-1.10.6-minimal-clean/library/Zend/Uri.php ZendFramework-1.10.6-minimal/library/Zend/Uri.php
--- ZendFramework-1.10.6-minimal-clean/library/Zend/Uri.php	2010-05-03 20:49:28.000000000 +0200
+++ ZendFramework-1.10.6-minimal/library/Zend/Uri.php	2010-07-07 11:23:48.000000000 +0200
@@ -127,14 +127,12 @@
             }
         }
 
-        if (!class_exists($className)) {
-            require_once 'Zend/Loader.php';
-            try {
-                Zend_Loader::loadClass($className);
-            } catch (Exception $e) {
-                require_once 'Zend/Uri/Exception.php';
-                throw new Zend_Uri_Exception("\"$className\" not found");
-            }
+        require_once 'Zend/Loader.php';
+        try {
+            Zend_Loader::loadClass($className);
+        } catch (Exception $e) {
+            require_once 'Zend/Uri/Exception.php';
+            throw new Zend_Uri_Exception("\"$className\" not found");
         }
 
         $schemeHandler = new $className($scheme, $schemeSpecific);
