Index: library/Zend/Gdata/Gapps.php
===================================================================
--- library/Zend/Gdata/Gapps.php	(revision 21203)
+++ library/Zend/Gdata/Gapps.php	(working copy)
@@ -784,6 +784,9 @@
      */
     public function suspendUser($username) {
         $user = $this->retrieveUser($username);
+        if ($user === null) {
+            throw new Zend_Gdata_Gapps_ServiceException();
+        }
         $user->login->suspended = true;
         return $user->save();
     }
@@ -801,6 +804,9 @@
      */
     public function restoreUser($username) {
         $user = $this->retrieveUser($username);
+        if ($user === null) {
+            throw new Zend_Gdata_Gapps_ServiceException();
+        }
         $user->login->suspended = false;
         return $user->save();
     }
