Index: tests/Zend/Mail/MailTest.php
===================================================================
--- tests/Zend/Mail/MailTest.php	(revision 21568)
+++ tests/Zend/Mail/MailTest.php	(working copy)
@@ -132,6 +132,32 @@
     }
 
     /**
+     * @group ZF-9483 charset configuration
+     */
+    public function testCharset()
+    {
+        // default
+        $mailDefault = new Zend_Mail();
+        $this->assertEquals('iso-8859-1', $mailDefault->getCharset());
+
+        // to UTF
+        $mailDefault->setCharset('utf-8');
+        $this->assertEquals('utf-8', $mailDefault->getCharset());
+
+        // to ISO
+        $mailDefault->setCharset('iso-8859-1');
+        $this->assertEquals('iso-8859-1', $mailDefault->getCharset());
+
+        // init ISO
+        $mailIso = new Zend_Mail('iso-8859-1');
+        $this->assertEquals('iso-8859-1', $mailIso->getCharset());
+
+        // init UTF
+        $mailUtf = new Zend_Mail('utf-8');
+        $this->assertEquals('utf-8', $mailUtf->getCharset());
+    }
+
+    /**
      * Test case for a simple email text message with
      * multiple recipients.
      *

