Index: Int.php
===================================================================
--- Int.php	(Revision 3597)
+++ Int.php	(Arbeitskopie)
@@ -54,6 +54,11 @@
     {
         $this->_messages = array();
 
+        if (is_object($value)) {
+            $this->_messages[] = "object of type " . get_class($value) . " does not appear to be an integer";
+            return false;
+        }
+
         $locale = localeconv();
 
         $valueFiltered = str_replace($locale['decimal_point'], '.', $value);
