Index: tests/Zend/CurrencyTest.php
===================================================================
--- tests/Zend/CurrencyTest.php	(revision 18366)
+++ tests/Zend/CurrencyTest.php	(working copy)
@@ -619,4 +619,16 @@
         $this->assertEquals('$0.00', $currency->toCurrency(1.0E-4));
         $this->assertEquals('$0.00', $currency->toCurrency(1.0E-5));
     }
+    
+    /**
+     * @see ZF-7864
+     */
+    public function testDifferentCurrencyForLocale()
+    {
+    	$currency = new Zend_Currency("de_DE");
+    	$this->assertEquals('2,30 $', $currency->toCurrency(2.3, array('currency' => 'USD')));
+    	
+    	$currency = new Zend_Currency("USD", "de_DE");
+    	$this->assertEquals('2,30 $', $currency->toCurrency(2.3));
+    }
 }
