diff --git a/library/Zend/Mime/Decode.php b/library/Zend/Mime/Decode.php
index af413a9..3584052 100644
--- a/library/Zend/Mime/Decode.php
+++ b/library/Zend/Mime/Decode.php
@@ -140,6 +140,7 @@ class Zend_Mime_Decode
             @list($headers, $body) = @preg_split("%([\r\n]+)\\1%U", $message, 2);
         }
 
+        $headers = preg_replace("/=\n([^ ]+)/", "= \n\\1", $headers); // ZF-4450
         $headers = iconv_mime_decode_headers($headers, ICONV_MIME_DECODE_CONTINUE_ON_ERROR);
 
         if ($headers === false ) {
@@ -149,6 +150,7 @@ class Zend_Mime_Decode
 
         // normalize header names
         foreach ($headers as $name => $header) {
+
             $lower = strtolower($name);
             if ($lower == $name) {
                 continue;
diff --git a/tests/Zend/Mime/MessageTest.php b/tests/Zend/Mime/MessageTest.php
index d47c39f..94f97bd 100644
--- a/tests/Zend/Mime/MessageTest.php
+++ b/tests/Zend/Mime/MessageTest.php
@@ -136,4 +136,34 @@ EOD;
         $this->assertEquals('base64', $part2->encoding);
         $this->assertEquals('12', $part2->id);
     }
+
+    /**
+     * ZF-4450 - iconv_mime_decode_headers() fails to split some headers
+     */
+    public function testSplitHeaders()
+    {
+        $text = <<<EOT
+Delivered-To: someone@example.com
+Received: by 10.187.230.20 with SMTP id h20cs372027far;
+         Tue, 30 Sep 2008 12:38:11 -0700 (PDT)
+Received: by 10.100.125.12 with SMTP id x12mr6509213anc.159.1222803490118;
+         Tue, 30 Sep 2008 12:38:10 -0700 (PDT)
+DomainKey-Signature: a=rsa-sha1; c=nofws;
+         d=example.com; s=gamma;
+         h=message-id:date:reply-to:user-agent:mime-version:to:subject
+          :x-enigmail-version:content-type:content-transfer-encoding:from;
+         b=xV2gAKUIWaeAuYLxG60dH6fAjmQasppbdupoWxaIPD38549epbmlmdkgF8B3KGM341
+          eIHR+5LZdgDMyI+tGjJ6BiFnmO/j2IsstcF4PvRpcnZriOV1t/GsXkQzf0AVYDJ/i5RH
+          rtD/Rad3oPj/8gxQnbwApEueCiQW9lsr27A3s=
+Message-ID: <48E28018.70003@example.com>
+Date: Tue, 30 Sep 2008 09:38:00 -1000
+Reply-To: someone@example.com
+EOT;
+        $headers = array();
+        Zend_Mime_Decode::splitMessage($text, $headers, $null);
+        $this->assertArrayHasKey('date', $headers);
+        $this->assertArrayHasKey('reply-to', $headers);
+        $this->assertArrayHasKey('message-id', $headers);        
+        $this->assertEquals('<48E28018.70003@example.com>', $headers['message-id']);
+    }    
 }
