Index: library/Zend/Rest/Route.php
===================================================================
--- library/Zend/Rest/Route.php	(revision 19376)
+++ library/Zend/Rest/Route.php	(working copy)
@@ -255,6 +255,7 @@
             unset($params['index']);
             $url .= '/index';
             foreach ($params as $key => $value) {
+                if ($encode) $value = urlencode($value);
                 $url .= '/' . $key . '/' . $value;
             }
         } elseif (isset($params['id'])) {
