--- Router.php.orig	2009-06-21 20:50:06.000000000 +0200
+++ Router.php	2009-06-22 23:06:03.000000000 +0200
@@ -15,7 +15,7 @@
  * @category   Zend
  * @package    Zend_Application
  * @subpackage Resource
- * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  * @version    $Id$
  */
@@ -28,7 +28,7 @@
  * @package    Zend_Application
  * @subpackage Resource
  * @author     Dolf Schimmel
- * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 class Zend_Application_Resource_Router 
@@ -59,14 +59,19 @@
         if (null === $this->_router) {
             $bootstrap = $this->getBootstrap();
             $bootstrap->bootstrap('FrontController');
-            $front = $bootstrap->getContainer()->frontcontroller;
+            $this->_router = $bootstrap->getContainer()->frontcontroller->getRouter();
 
             $options = $this->getOptions();
             if(!isset($options['routes'])) {
                 $options['routes'] = array();
             }
+            
+
+            if(isset($options['chainNameSeparator'])) {
+                $this->_router->setChainNameSeparator($options['chainNameSeparator']);
+            }
+            
 
-            $this->_router = $front->getRouter();
             $this->_router->addConfig(new Zend_Config($options['routes']));
         }
         return $this->_router;
