Index: DbTable.php
===================================================================
--- DbTable.php	(revision 6034)
+++ DbTable.php	(working copy)
@@ -304,13 +304,13 @@
                 $this->_zendDb->quoteIdentifier($this->_credentialColumn)
                 . ' = ' . $this->_credentialTreatment, $this->_credential
                 )
-            . ' AS zend_auth_credential_match'
             );
 
         // get select
         $dbSelect = $this->_zendDb->select();
-        $dbSelect->from($this->_tableName, array('*', $credentialExpression))
-                 ->where($this->_zendDb->quoteIdentifier($this->_identityColumn) . ' = ?', $this->_identity);
+        $dbSelect->from($this->_tableName, '*')
+                 ->where($this->_zendDb->quoteIdentifier($this->_identityColumn) . ' = ?', $this->_identity)
+                 ->where($credentialExpression);
 
         // query for the identity
         try {
@@ -335,17 +335,8 @@
             return new Zend_Auth_Result($authResult['code'], $authResult['identity'], $authResult['messages']);
         }
 
-        $resultIdentity = $resultIdentities[0];
+        $this->_resultRow = $resultIdentities[0];
 
-        if ($resultIdentity['zend_auth_credential_match'] != '1') {
-            $authResult['code'] = Zend_Auth_Result::FAILURE_CREDENTIAL_INVALID;
-            $authResult['messages'][] = 'Supplied credential is invalid.';
-            return new Zend_Auth_Result($authResult['code'], $authResult['identity'], $authResult['messages']);
-        }
-
-        unset($resultIdentity['zend_auth_credential_match']);
-        $this->_resultRow = $resultIdentity;
-
         $authResult['code'] = Zend_Auth_Result::SUCCESS;
         $authResult['messages'][] = 'Authentication successful.';
         return new Zend_Auth_Result($authResult['code'], $authResult['identity'], $authResult['messages']);
