Index: HttpTest.php
===================================================================
--- HttpTest.php	(revision 12360)
+++ HttpTest.php	(working copy)
@@ -366,6 +366,27 @@
     {
         $this->assertSame('', $this->_request->getBaseUrl());
     }
+    
+    /*
+     * @descr Tests if an empty string gets returned when no basepath is set on the request.
+     * This is important on windows, where before this fix '\' was returned instead of an empty string.
+     * @group ZF-4810
+     */
+    public function testGetBasePathIsEmptyStringIfNoneSet()
+    {
+        $request = new Zend_Controller_Request_Http();
+        $this->assertEquals('', $request->getBasePath());    	
+    }
+    
+    /*
+     * @group ZF-4810
+     */
+    public function testBackslashIsAllowedInBasePath()
+    {
+        $request = new Zend_Controller_Request_Http();
+        $request->setBasePath('/home\\dir\\');
+        $this->assertEquals('/home\\dir', $request->getBasePath());    	
+    }
  
     public function testSetBaseUrl()
     {
