--- Zend/Form.php	Fri Jun 03 17:48:45 2011
+++ Zend/Form.php	Fri Jun 10 13:54:01 2011
@@ -2242,6 +2242,9 @@
         }
         $context = $data;
         foreach ($this->getElements() as $key => $element) {
+            if ($element->getIgnore()) {
+                continue;
+            }
             if (null !== $translator && $this->hasTranslator()
                     && !$element->hasTranslator()) {
                 $element->setTranslator($translator);
@@ -2300,6 +2303,9 @@
         $context    = $data;

         foreach ($this->getElements() as $key => $element) {
+            if ($element->getIgnore()) {
+                continue;
+            }
             $check = $data;
             if (($belongsTo = $element->getBelongsTo()) !== $eBelongTo) {
                 $check = $this->_dissolveArrayValue($data, $belongsTo);
