Index: trunk/library/Zend/Text/Figlet.php
===================================================================
--- trunk/library/Zend/Text/Figlet.php	(Revision 19098)
+++ trunk/library/Zend/Text/Figlet.php	(Arbeitskopie)
@@ -427,7 +427,8 @@
     public function render($text, $encoding = 'UTF-8')
     {
         if (!is_string($text)) {
-            throw new InvalidArgumentException('$text must be a string');
+            require_once 'Zend/Text/InvalidArgumentException.php';
+            throw new Zend_Text_InvalidArgumentException('$text must be a string');
         }
 
         if ($encoding !== 'UTF-8') {
Index: trunk/library/Zend/Text/InvalidArgumentException.php
===================================================================
--- trunk/library/Zend/Text/InvalidArgumentException.php	(Revision 0)
+++ trunk/library/Zend/Text/InvalidArgumentException.php	(Revision 0)
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Text
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id:$
+ */
+
+/**
+ * Zend_Text_Exception
+ */
+require_once 'Zend/Text/Exception.php';
+
+/**
+ * Text exceptions
+ *
+ * Class to represent InvalidArgumentExceptions that occur during Zend_Text operations.
+ *
+ * @category   Zend
+ * @package    Zend_Text
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_Text_InvalidArgumentException extends Zend_Text_Exception
+{
+}
