Index: library/Zend/CodeGenerator/Php/Docblock/Tag.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag.php	(revision 19685)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag.php	(working copy)
@@ -34,6 +34,30 @@
  * @see Zend_CodeGenerator_Php_Docblock_Tag_Return
  */
 require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Return.php';
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Uses
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Uses.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Var
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Var.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Subpackage
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Package
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Package.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Author
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Author.php';
 
 /**
  * @category   Zend
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Package.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Package.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Package.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id: $
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Package extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_package = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Package
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $packageTag = new self();
+        
+        $packageTag->setName('package');
+        $packageTag->setPackage($reflectionTagReturn->getPackage());
+        
+        return $packageTag;
+    }
+    
+    /**
+     * setPackage()
+     *
+     * @param string $package
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setPackage($package)
+    {
+        $this->_package = $package;
+        return $this;
+    }
+    
+    /**
+     * getPackage()
+     *
+     * @return string
+     */
+    public function getPackage()
+    {
+        return $this->_package;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@package ' . $this->_package;
+        return $output;
+    }
+    
+}
\ No newline at end of file
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id: $
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Subpackage extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_subpackage = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Subpackage
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $subpackageTag = new self();
+        
+        $subpackageTag->setName('subpackage');
+        $subpackageTag->setSubpackage($reflectionTagReturn->getSubpackage());
+        
+        return $subpackageTag;
+    }
+    
+    /**
+     * setSubpackage()
+     *
+     * @param string $subpackage
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setSubpackage($subpackage)
+    {
+        $this->_subpackage = $subpackage;
+        return $this;
+    }
+    
+    /**
+     * getSubpackage()
+     *
+     * @return string
+     */
+    public function getSubpackage()
+    {
+        return $this->_subpackage;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@subpackage ' . $this->_subpackage;
+        return $output;
+    }
+    
+}
\ No newline at end of file
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Var.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Var.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Var.php	(revision 0)
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id:  $
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Var extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_datatype = null;
+    
+    /**
+     * @var string
+     */
+    protected $_description = null;
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Var
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $returnTag = new self();
+        
+        $returnTag->setName('var');
+        $returnTag->setDatatype($reflectionTagReturn->getType()); // @todo rename
+        $returnTag->setDescription($reflectionTagReturn->getDescription());
+        
+        return $returnTag;
+    }
+    
+    /**
+     * setDatatype()
+     *
+     * @param string $datatype
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setDatatype($datatype)
+    {
+        $this->_datatype = $datatype;
+        return $this;
+    }
+    
+    /**
+     * getDatatype()
+     *
+     * @return string
+     */
+    public function getDatatype()
+    {
+        return $this->_datatype;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@var ' . $this->_datatype . ' ' . $this->_description;
+        return $output;
+    }
+    
+}
\ No newline at end of file
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Uses.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Uses.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Uses.php	(revision 0)
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @author     Chris Beck 
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id: $
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Uses extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_uses = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Uses
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $usesTag = new self();
+        
+        $usesTag->setName('uses');
+        $usesTag->setUses($reflectionTagReturn->getUses());
+        
+        return $usesTag;
+    }
+    
+    /**
+     * setUses()
+     *
+     * @param string $uses
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setUses($uses)
+    {
+        $this->_uses = $uses;
+        return $this;
+    }
+    
+    /**
+     * getUses()
+     *
+     * @return string
+     */
+    public function getUses()
+    {
+        return $this->_uses;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@uses ' . $this->_uses;
+        return $output;
+    }
+    
+}
\ No newline at end of file
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Author.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Author.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Author.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id: $
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Author extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_author = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Author
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $authorTag = new self();
+        
+        $authorTag->setName('author');
+        $authorTag->setAuthor($reflectionTagReturn->getAuthor());
+        
+        return $authorTag;
+    }
+    
+    /**
+     * setAuthor()
+     *
+     * @param string $author
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setAuthor($author)
+    {
+        $this->_author = $author;
+        return $this;
+    }
+    
+    /**
+     * getAuthor()
+     *
+     * @return string
+     */
+    public function getAuthor()
+    {
+        return $this->_author;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@author ' . $this->_author;
+        return $output;
+    }
+    
+}
\ No newline at end of file
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Version.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Version.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Version.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id:
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Version extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_version = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Version
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $versionTag = new self();
+        
+        $versionTag->setName('version');
+        $versionTag->setVersion($reflectionTagReturn->getVersion());
+        
+        return $versionTag;
+    }
+    
+    /**
+     * setVersion()
+     *
+     * @param string $version
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setVersion($version)
+    {
+        $this->_version = $version;
+        return $this;
+    }
+    
+    /**
+     * getVersion()
+     *
+     * @return string
+     */
+    public function getVersion()
+    {
+        return $this->_version;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@version ' . $this->_version;
+        return $output;
+    }
+    
+}
\ No newline at end of file
