Index: tests/Zend/View/Helper/FormCheckboxTest.php
===================================================================
--- tests/Zend/View/Helper/FormCheckboxTest.php	(revision 24746)
+++ tests/Zend/View/Helper/FormCheckboxTest.php	(working copy)
@@ -302,6 +302,24 @@
         $test = $this->helper->formCheckbox('foo', 0, array(), array(1,0));
         $this->assertNotContains('checked="checked"', $test);
     }
+
+    /**
+     * @group ZF-6624
+     */
+    public function testRenderingWithoutHiddenElement()
+    {
+        $html = $this->helper->formCheckbox(
+            'foo',
+            'bar',
+            array(
+                 'disableHidden' => true,
+            )
+        );
+        $this->assertSame(
+            '<input type="checkbox" name="foo" id="foo" value="bar">',
+            $html
+        );
+    }
 }
 
 // Call Zend_View_Helper_FormCheckboxTest::main() if this source file is executed directly.
