Index: incubator/Zend/Http/Response.php
===================================================================
--- incubator/Zend/Http/Response.php	(revision 878)
+++ incubator/Zend/Http/Response.php	(working copy)
@@ -485,15 +485,18 @@
                     }
 
                     $headers[$h_name][] = $h_value;
-                    end($headers);
-                    $last_header = key($headers);
                 } else {
                     $headers[$h_name] = $h_value;
-                    end($headers);
-                    $last_header = key($headers);
                 }
+                $last_header = $h_name;
             } elseif (preg_match("|^\s+(\S+)$|", $line, $m) && $last_header !== null) {
-                $headers[$last_header] .= $m[1];
+                if (is_array($headers[$last_header])) {
+                    end($headers[$last_header]);
+                    $last_header_key = key($headers[$last_header]);
+                    $headers[$last_header][$last_header_key] .= $m[1];
+                } else {
+                    $headers[$last_header] .= $m[1];
+                }
             }
         }
 
