Index: Unix.php
===================================================================
--- Unix.php	(revision 18749)
+++ Unix.php	(working copy)
@@ -126,6 +126,12 @@
     private $_ipcSemFile;
 
     /**
+     * Path to /temp directory instead of system one
+     * @var string
+     */
+    private $_tempDirPath = NULL;
+
+    /**
      * Constructor method
      *
      * Allocates a new pseudo-thread object. Optionally, set a PUID, a GUID and
@@ -135,13 +141,14 @@
      * @param  integer $puid
      * @param  integer $guid
      * @param  integer $umask
+     * @param  string  $tempDirPath
      * @throws ZendX_Console_Process_Exception When running on windows
      * @throws ZendX_Console_Process_Exception When running in web enviroment
      * @throws ZendX_Console_Process_Exception When shmop_* functions don't exist
      * @throws ZendX_Console_Process_Exception When pcntl_* functions don't exist
      * @throws ZendX_Console_Process_Exception When posix_* functions don't exist
      */
-    public function __construct($puid = null, $guid = null, $umask = null)
+    public function __construct($puid = null, $guid = null, $umask = null, $tempDirPath = null)
     {
         if (substr(PHP_OS, 0, 3) === 'WIN') {
             require_once 'ZendX/Console/Process/Exception.php';
@@ -166,6 +173,10 @@
         $this->_guid = $guid;
         $this->_puid = $puid;
 
+        if($tempDir = realpath($tempDirPath)) {
+            $this->_tempDirPath = $tempDir;
+        }
+
         if ($umask !== null) {
             umask($umask);
         }
@@ -610,7 +621,12 @@
      */
     private function _createIpcSegment()
     {
-        $this->_ipcSegFile = realpath(sys_get_temp_dir()) . '/' . rand() . $this->_name . '.shm';
+        if($this->_tempDirPath === null) {
+            $tempPath = realpath(sys_get_temp_dir());
+        } else {
+            $tempPath = $this->_tempDirPath;
+        }
+        $this->_ipcSegFile = $tempPath . '/' . rand() . $this->_name . '.shm';
         touch($this->_ipcSegFile);
 
         $shmKey = ftok($this->_ipcSegFile, 't');
@@ -637,7 +653,12 @@
      */
     private function _createIpcSemaphore()
     {
-        $this->_ipcSemFile = realpath(sys_get_temp_dir()) . '/' . rand() . $this->_name . '.sem';
+        if($this->_tempDirPath === null) {
+            $tempPath = realpath(sys_get_temp_dir());
+        } else {
+            $tempPath = $this->_tempDirPath;
+        }
+        $this->_ipcSemFile = $tempPath . '/' . rand() . $this->_name . '.sem';
         touch($this->_ipcSemFile);
 
         $semKey = ftok($this->_ipcSemFile, 't');
