Index: Zend/Controller/Request/Http.php
===================================================================
--- Zend/Controller/Request/Http.php	(revision 927)
+++ Zend/Controller/Request/Http.php	(working copy)
@@ -48,19 +48,19 @@
      * Base URL of request
      * @var string
      */
-    protected $_baseUrl = ''; 
+    protected $_baseUrl = null; 
 
     /**
      * Base path of request
      * @var string
      */
-    protected $_basePath = ''; 
+    protected $_basePath = null; 
 
     /**
      * PATH_INFO
      * @var string
      */
-    protected $_pathInfo = ''; 
+    protected $_pathInfo = null; 
 
     /**
      * Instance parameters
