Index: library/ZendX/JQuery/View/Helper/JQuery/Container.php
===================================================================
--- library/ZendX/JQuery/View/Helper/JQuery/Container.php	(revision 24462)
+++ library/ZendX/JQuery/View/Helper/JQuery/Container.php	(working copy)
@@ -112,6 +112,13 @@
      * @var Boolean
      */
     protected $_uiEnabled = false;
+	
+    /**
+     * jQuery UI Lang Library Enabled
+     *
+     * @var Boolean
+     */
+    protected $_uiLangEnabled = false;
 
     /**
      * Local jQuery UI Path. Use Google CDN if
@@ -309,6 +316,39 @@
     }
 
     /**
+     * Enable jQuery UI Library Rendering
+     *
+     * @return ZendX_JQuery_View_Helper_JQuery_Container
+     */
+    public function uiLangEnable()
+    {
+        $this->uiEnable();
+        $this->_uiLangEnabled = true;
+        return $this;
+    }
+
+    /**
+     * Disable jQuery UI Library Rendering
+     *
+     * @return ZendX_JQuery_View_Helper_JQuery_Container
+     */
+    public function uiLangDisable()
+    {
+        $this->_uiLangEnabled = false;
+        return $this;
+    }
+
+    /**
+     * Check wheater currently the jQuery UI library is enabled.
+     *
+     * @return boolean
+     */
+    public function uiLangIsEnabled()
+    {
+         return $this->_uiLangEnabled;
+    }
+
+    /**
      * Set jQuery UI version used.
      * 
      * @param  string $version
@@ -374,6 +414,16 @@
     {
     	return $this->_uiPath;
     }
+	
+    /**
+     * Return the local jQuery UI Path if set.
+     *
+     * @return string
+     */
+    public function getUiLangPath()
+    {
+    	return $this->_uiLangPath;
+    }
 
     /**
      * Proxies to getUiPath() for consistency in function naming.
@@ -384,6 +434,16 @@
     {
         return $this->getUiPath();
     }
+	
+    /**
+     * Proxies to getUiLangPath() for consistency in function naming.
+     *
+     * @return string
+     */
+    public function getUiLangLocalPath()
+    {
+        return $this->getUiLangPath();
+    }
 
     /**
      * Is the jQuery Ui loaded from local scope?
@@ -394,6 +454,16 @@
     {
     	return (null===$this->_uiPath ? false : true);
     }
+	
+    /**
+     * Is the jQuery Ui Language loaded from local scope?
+     *
+     * @return boolean
+     */
+    public function useUiLangLocal()
+    {
+    	return (null===$this->_uiLangPath ? false : true);
+    }
 
     /**
      * Is the jQuery Ui enabled and loaded from CDN?
@@ -715,6 +785,11 @@
                 $uiPath = $this->_getJQueryUiLibraryPath();
                 $scriptTags .= '<script type="text/javascript" src="'.$uiPath.'"></script>'.PHP_EOL;
             }
+			
+            if($this->uiLangIsEnabled()) {
+                $uiLangPath = $this->_getJQueryUiLangLibraryPath();
+                $scriptTags .= '<script type="text/javascript" src="'.$uiLangPath.'"></script>'.PHP_EOL;
+            }
 
             if(ZendX_JQuery_View_Helper_JQuery::getNoConflictMode() == true) {
                 $scriptTags .= '<script type="text/javascript">var $j = jQuery.noConflict();</script>'.PHP_EOL;
@@ -839,4 +914,26 @@
         }
         return $uiPath;
     }
-}
+
+    /**
+     * @return string
+     */
+    protected function _getJQueryUiLangLibraryPath()
+    {
+        if($this->useUiCdn()) {
+            $baseUri = $this->_getJQueryLibraryBaseCdnUri();
+            $uiLangPath = $baseUri.
+                ZendX_JQuery::CDN_SUBFOLDER_JQUERYUI .
+                $this->getUiCdnVersion() .
+                "/i18n/jquery-ui-i18n.min.js";
+        } else if($this->useUiLangLocal()) {
+            $uiLangPath = $this->getUiLangPath();
+        }
+        return $uiLangPath;
+    }
+	
+	public function setLanguageCode($strCode){
+		$this->uiLangEnable ();
+		$this->addOnLoad("$.datepicker.setDefaults( $.datepicker.regional[ '".$strCode."' ] );");
+	}
+}
\ No newline at end of file
