Index: Client.php
===================================================================
--- Client.php	(revision 19246)
+++ Client.php	(working copy)
@@ -111,6 +111,8 @@
         'strictredirects' => false,
         'useragent'       => 'Zend_Http_Client',
         'timeout'         => 10,
+		'timeoutRetry'    => 2,
+		'timeoutRetrySleep' => 1, 	// wait 1 second between each timeoutRetry
         'adapter'         => 'Zend_Http_Client_Adapter_Socket',
         'httpversion'     => self::HTTP_1,
         'keepalive'       => false,
Index: Client/Adapter/Socket.php
===================================================================
--- Client/Adapter/Socket.php	(revision 19246)
+++ Client/Adapter/Socket.php	(working copy)
@@ -220,13 +220,24 @@
 
             $flags = STREAM_CLIENT_CONNECT;
             if ($this->config['persistent']) $flags |= STREAM_CLIENT_PERSISTENT;
+			
+			$retry = 0;
+			while(!$this->socket) {
+				
+				$this->socket = @stream_socket_client($host . ':' . $port,
+													  $errno,
+													  $errstr,
+													  (int) $this->config['timeout'],
+													  $flags,
+													  $context);
 
-            $this->socket = @stream_socket_client($host . ':' . $port,
-                                                  $errno,
-                                                  $errstr,
-                                                  (int) $this->config['timeout'],
-                                                  $flags,
-                                                  $context);
+				// If not a timeout error or no timeout retry, break loop
+				if( $errno != 60 || !$this->config['timeoutretry'] || $retry >= $this->config['timeoutretry'] )
+					break;
+				
+				sleep($this->config['timeoutretrysleep']);
+				$retry++;
+			}
 
             if (! $this->socket) {
                 $this->close();
