Index: Zend/Service/Flickr/OnlineTest.php
===================================================================
--- Zend/Service/Flickr/OnlineTest.php	(revision 17021)
+++ Zend/Service/Flickr/OnlineTest.php	(working copy)
@@ -88,7 +88,7 @@
                          
         $resultSet = $this->_flickr->groupPoolGetPhotos('20083316@N00', $options);
         
-        $this->assertEquals(21770, $resultSet->totalResultsAvailable);
+        $this->assertGreaterThan(20000, $resultSet->totalResultsAvailable);
         $this->assertEquals(10, $resultSet->totalResults());
         $this->assertEquals(10, $resultSet->totalResultsReturned);
         $this->assertEquals(1, $resultSet->firstResultPosition);
@@ -133,9 +133,9 @@
                          'page'     => 1,
                          'extras'   => 'license, date_upload, date_taken, owner_name, icon_server');
 
-        $resultSet = $this->_flickr->userSearch('darby.felton@yahoo.com', $options);
+        $resultSet = $this->_flickr->userSearch('lulievron', $options);
 
-        $this->assertEquals(16, $resultSet->totalResultsAvailable);
+        $this->assertGreaterThan(10, $resultSet->totalResultsAvailable);
         $this->assertEquals(10, $resultSet->totalResults());
         $this->assertEquals(10, $resultSet->totalResultsReturned);
         $this->assertEquals(1, $resultSet->firstResultPosition);
@@ -203,9 +203,9 @@
         $this->assertEquals(1, $resultSet->firstResultPosition);
 
         foreach ($resultSet as $result) {
-            $this->assertTrue($result instanceof Zend_Service_Flickr_Result);
+            $this->assertType('Zend_Service_Flickr_Result', $result);
             if (isset($dateTakenPrevious)) {
-                $this->assertTrue(strcmp($result->datetaken, $dateTakenPrevious) > 0);
+                $this->assertTrue(strcmp($result->datetaken, $dateTakenPrevious) > 0, "Result sorting is not as expected");
             }
             $dateTakenPrevious = $result->datetaken;
         }
