--- Twitter.php	(Revision 17809)
+++ Twitter.php	(Arbeitskopie)
@@ -85,7 +85,9 @@
         'directMessage',
         'friendship',
         'account',
-        'favorite'
+        'favorite',
+		'friends',
+		'followers'
     );
 
     /**
@@ -777,4 +779,64 @@
         $response = $this->restPost($path);
         return new Zend_Rest_Client_Result($response->getBody());
     }
+
+    /**
+	 * Returns numeric IDs for every user the specified user is following.
+     *
+     * $params may contain one or more of the following:
+     * - 'id': Id or screen name of a user for whom to fetch the friends
+     * - 'page': Retrieve a different page of resuls
+     * @return Zend_Rest_Client_Result
+     */
+    public function friendsIds($params=array())
+    {
+        $this->_init();
+        $path = '/friends/ids';
+        $_params = array();
+        foreach ($params as $key => $value) {
+            switch (strtolower($key)) {
+                case 'id':
+                    $path .= '/' . $value;
+                    break;
+                case 'page':
+                    $_params['page'] = (int) $value;
+                    break;
+                default:
+                    break;
+            }
+        }
+        $path .= '.xml';
+        $response = $this->restGet($path, $_params);
+        return new Zend_Rest_Client_Result($response->getBody());
+    }
+
+    /**
+	 * Returns numeric IDs for every user following the specified user.
+     *
+     * $params may contain one or more of the following:
+     * - 'id': Id or screen name of a user for whom to fetch the followers
+     * - 'page': Retrieve a different page of resuls
+     * @return Zend_Rest_Client_Result
+     */
+    public function followersIds($params=array())
+    {
+        $this->_init();
+        $path = '/followers/ids';
+        $_params = array();
+        foreach ($params as $key => $value) {
+            switch (strtolower($key)) {
+                case 'id':
+                    $path .= '/' . $value;
+                    break;
+                case 'page':
+                    $_params['page'] = (int) $value;
+                    break;
+                default:
+                    break;
+            }
+        }
+        $path .= '.xml';
+        $response = $this->restGet($path, $_params);
+        return new Zend_Rest_Client_Result($response->getBody());
+    }
 }
