--- Adapter.php-v1.5.1	2008-05-08 15:38:13.000000000 +0200
+++ Adapter.php	2008-05-08 15:44:40.000000000 +0200
@@ -111,10 +111,11 @@ abstract class Zend_Translate_Adapter {
                      RecursiveIteratorIterator::SELF_FIRST) as $file => $info) {
                 if ($info->isDir()) {
 
-                    $directory = $info->getPath();
+                    // $basename = basename($info->getPath()); // php < 5.2.2
+                    $basename = $info->getBasename(); // php >= 5.2.2
                     // pathname as locale
-                    if (($this->_options['scan'] === self::LOCALE_DIRECTORY) and (Zend_Locale::isLocale((string) $info))) {
-                        $locale = (string) $info;
+                    if (($this->_options['scan'] === self::LOCALE_DIRECTORY) and (Zend_Locale::isLocale($basename))) {
+                        $locale = $basename;
                     }
 
                 } else if ($info->isFile()) {
