Index: Element.php
===================================================================
--- Element.php	(revision 165)
+++ Element.php	(working copy)
@@ -623,11 +623,6 @@
      */
     public function getLabel()
     {
-        $translator = $this->getTranslator();
-        if (null !== $translator) {
-            return $translator->translate($this->_label);
-        }
-
         return $this->_label;
     }
 
