Index: library/Zend/View/Abstract.php
===================================================================
--- library/Zend/View/Abstract.php	(revision 24045)
+++ library/Zend/View/Abstract.php	(working copy)
@@ -1181,10 +1181,10 @@
         if (!isset($store[$name])) {
             $class = $this->getPluginLoader($type)->load($name);
             $store[$name] = new $class();
-            if (method_exists($store[$name], 'setView')) {
-                $store[$name]->setView($this);
-            }
         }
+        if(method_exists($store[$name], 'setView') && $store[$name]->view !== $this) {
+            $store[$name]->setView($this);
+        }
 
         $this->$storeVar = $store;
         return $store[$name];
Index: tests/Zend/ViewTest.php
===================================================================
--- tests/Zend/ViewTest.php	(revision 24045)
+++ tests/Zend/ViewTest.php	(working copy)
@@ -1147,7 +1147,7 @@
         $paths = $view->getScriptPaths();
         $this->assertContains($path, $paths, var_export($paths, 1));
     }
-    
+
     /**
      * @group ZF-10042
      */
@@ -1156,6 +1156,22 @@
         $view = new Zend_View(array('assign' => array('foo' => 'bar')));
         $this->assertEquals('bar', $view->foo);
     }
+
+    /**
+     * @group ZF-11405
+     */
+    public function testViewHelperIsWithCurrentViewObject()
+    {
+        $view = new Zend_View();
+
+        $view->setHelperPath(dirname(__FILE__) . '/View/_stubs/HelperDir1');
+        $view->getHelper('setViewHelper');
+
+        $clonedView = clone $view;
+        $helper = $clonedView->getHelper('setViewHelper');
+        $this->assertSame($clonedView, $helper->view, 'View object registered with view helper'
+            . ' should be the same as view from which helper being called from');
+    }
 }
 
 /**
Index: tests/Zend/View/_stubs/HelperDir1/SetViewHelper.php
===================================================================
--- tests/Zend/View/_stubs/HelperDir1/SetViewHelper.php	(revision 0)
+++ tests/Zend/View/_stubs/HelperDir1/SetViewHelper.php	(revision 0)
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_View
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+
+
+/**
+ * @category   Zend
+ * @package    Zend_View
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_View_Helper_SetViewHelper
+{
+    public $view;
+
+    public function setViewHelper()
+    {
+        return $this;
+    }
+
+    public function setView(Zend_View $view)
+    {
+        $this->view = $view;
+        return $this;
+    }
+}

Property changes on: tests\Zend\View\_stubs\HelperDir1\SetViewHelper.php
___________________________________________________________________
Added: svn:keywords
   + Id

