Index: Zend/Date.php
===================================================================
--- Zend/Date.php       (Revision 3597)
+++ Zend/Date.php       (Arbeitskopie)
@@ -176,7 +176,7 @@
 
         if (is_null($date)) {
             $date = Zend_Date::now();
-            if (($part !== null && $part !== Zend_Date::TIMESTAMP) {
+            if (($part !== null) && ($part !== Zend_Date::TIMESTAMP)) {
                 $date = $date->get($part);
             }
         }

