Index: Element.php
===================================================================
--- Element.php	(revision 20381)
+++ Element.php	(working copy)
@@ -229,6 +229,7 @@
      * - Zend_Config: Zend_Config with options for configuring element
      *
      * @param  string|array|Zend_Config $spec
+     * @param  array|Zend_Config $options
      * @return void
      * @throws Zend_Form_Exception if no element name after initialization
      */
