Index: Zend/Tool/Project/Context/Zf/ControllerFile.php
===================================================================
--- Zend/Tool/Project/Context/Zf/ControllerFile.php	(revision 22500)
+++ Zend/Tool/Project/Context/Zf/ControllerFile.php	(working copy)
@@ -134,6 +134,11 @@
                                 'body' => <<<EOS
 \$errors = \$this->_getParam('error_handler');
 
+if (!\$errors instanceof ArrayObject) {
+    \$message = 'This action can not be accessed directly.';
+    throw new Zend_Controller_Action_Exception(\$message);
+}
+
 switch (\$errors->type) {
     case Zend_Controller_Plugin_ErrorHandler::EXCEPTION_NO_ROUTE:
     case Zend_Controller_Plugin_ErrorHandler::EXCEPTION_NO_CONTROLLER:
