--- RouterTest.php.orig	2009-06-21 22:34:55.000000000 +0200
+++ RouterTest.php	2009-06-22 23:23:01.000000000 +0200
@@ -61,9 +61,7 @@
 
         Zend_Loader_Autoloader::resetInstance();
         $this->autoloader = Zend_Loader_Autoloader::getInstance();
-
         $this->application = new Zend_Application('testing');
-
         $this->bootstrap = new Zend_Application_Bootstrap_Bootstrap($this->application);
 
         Zend_Controller_Front::getInstance()->resetInstance();
@@ -100,6 +98,14 @@
         $test = $resource->init();
         $this->assertTrue($test instanceof Zend_Controller_Router_Rewrite);
     }
+    
+    public function testChainNameSeparatorIsParsedOnToRouter()
+    {
+        $resource = new Zend_Application_Resource_Router(array('chainNameSeparator' => '_unitTestSep_'));
+        $resource->setBootstrap($this->bootstrap);
+        $router = $resource->init();
+        $this->assertEquals('_unitTestSep_', $router->getChainNameSeparator());
+    }
 
     public function testOptionsPassedToResourceAreUsedToCreateRoutes()
     {
