Index: Abstract.php
===================================================================
--- Abstract.php	(revision 15641)
+++ Abstract.php	(working copy)
@@ -1084,6 +1084,10 @@
             return null;
         }
 
+        if (null === $this->_translator) {
+            require_once 'Zend/Form.php';
+            return Zend_Form::getDefaultTranslator();
+        }
         return $this->_translator;
     }
 
