Index: library/Zend/Form/Element.php
===================================================================
--- library/Zend/Form/Element.php	(revision 19148)
+++ library/Zend/Form/Element.php	(working copy)
@@ -851,6 +851,18 @@
     }
 
     /**
+     * Removes an element attribute
+     *
+     * @param  string $name
+     * @return Zend_Form_Element
+     * @throws Zend_Form_Exception for invalid $name values
+     */
+    public function removeAttrib($name)
+    {
+        return $this->setAttrib($name, null);
+    }
+
+    /**
      * Retrieve element attribute
      *
      * @param  string $name
Index: tests/Zend/Form/ElementTest.php
===================================================================
--- tests/Zend/Form/ElementTest.php	(revision 19148)
+++ tests/Zend/Form/ElementTest.php	(working copy)
@@ -438,6 +438,14 @@
         $this->assertFalse(isset($this->element->foo));
     }
 
+    public function testRemoveAttribUnsetsAttrib()
+    {
+        $this->element->setAttrib('foo', 'bar');
+        $this->assertEquals('bar', $this->element->getAttrib('foo'));
+        $this->element->removeAttrib('foo');
+        $this->assertFalse(isset($this->element->foo));
+    }
+
     public function testSetAttribsSetsMultipleAttribs()
     {
         $this->element->setAttribs(array(
