Index: FormRadio.php
===================================================================
--- FormRadio.php	(revision 13748)
+++ FormRadio.php	(working copy)
@@ -156,7 +156,7 @@
             $optId = $id . '-' . $filter->filter($opt_value);
 
             // Wrap the radios in labels
-            $radio = '<label'
+            $radio = '<label for="' . $optId . '"'
                     . $this->_htmlAttribs($label_attribs) . '>'
                     . (('prepend' == $labelPlacement) ? $opt_label : '')
                     . '<input type="' . $this->_inputType . '"'
