--- /usr/local/share/ZendFramework/1.10.4/library/Zend/Pdf/StringParser.php	2010-01-06 03:05:09.000000000 +0100
+++ StringParser.php	2010-05-03 08:49:41.000000000 +0200
@@ -264,7 +264,15 @@
             }
         } else {
             $start = $this->offset;
-            $this->offset += strcspn($this->data, "()<>[]{}/%\x00\t\n\f\r ", $this->offset);
+
+            $compare = '';
+            if( version_compare( phpversion(), '5.2.5' ) >= 0) {
+                $compare = "()<>[]{}/%\x00\t\n\f\r ";
+            } else {
+                $compare = "()<>[]{}/%\x00\t\n\r ";
+            }
+
+            $this->offset += strcspn($this->data, $compare, $this->offset);
 
             return substr($this->data, $start, $this->offset - $start);
         }
