Index: /library/Zend/Application/Bootstrap/BootstrapAbstract.php
===================================================================
--- /library/Zend/Application/Bootstrap/BootstrapAbstract.php	(revision 15465)
+++ /library/Zend/Application/Bootstrap/BootstrapAbstract.php	(working copy)
@@ -226,7 +226,7 @@
      */
     public function registerPluginResource($resource, $options = null)
     {
-        if (is_string($resource) && class_exists($resource)) {
+        if (is_string($resource) && class_exists($resource, false)) {
             $options = (array) $options;
             $options['bootstrap'] = $this;
             $resource = new $resource($options);
