Index: tests/Zend/Validate/File/UploadTest.php
===================================================================
--- tests/Zend/Validate/File/UploadTest.php	(revision 23564)
+++ tests/Zend/Validate/File/UploadTest.php	(working copy)
@@ -252,6 +252,18 @@
         $validator->setFiles($files);
         $this->assertEquals($files, $validator->getFiles());
     }
+    
+    /**
+     * @group ZF-10738
+     */
+    public function testSetFilesNull()
+    {
+        $_FILES = NULL;
+        $validator = new Zend_Validate_File_Upload();
+        $validator->setFiles(NULL);
+        $this->assertEquals(array(), $validator->getFiles());
+    }
+
 }
 
 // Call Zend_Validate_File_UploadTest::main() if this source file is executed directly.
