diff -Naur library-1.8.2/Zend/Application/Resource/Session.php library-1.8.2.1/Zend/Application/Resource/Session.php
--- library-1.8.2/Zend/Application/Resource/Session.php	2009-04-17 14:20:40.000000000 +0200
+++ library-1.8.2.1/Zend/Application/Resource/Session.php	2009-05-29 22:08:33.000000000 +0200
@@ -40,14 +40,17 @@
     protected $_saveHandler = null;
 
     /**
-     * Set session save handler
+     * Get session save handler
      *
-     * @param  array|string|Zend_Session_SaveHandler_Interface $saveHandler
-     * @return Zend_Application_Resource_Session
+     * @return Zend_Session_SaveHandler_Interface
      * @throws Zend_Application_Resource_Exception When $saveHandler is no valid save handler
      */
-    public function setSaveHandler($saveHandler)
+    public function getSaveHandler()
     {
+        if (null === $this->_saveHandler)
+        {
+            $options = $this->getOptions();
+            $saveHandler = $options['saveHandler'];
         if (is_array($saveHandler)) {
             if (!array_key_exists('class', $saveHandler)) {
                 throw new Zend_Application_Resource_Exception('Session save handler class not provided in options');
@@ -66,8 +69,9 @@
         }
 
         $this->_saveHandler = $saveHandler;
+        }
 
-        return $this;
+        return $this->_saveHandler;
     }
 
     /**
@@ -87,7 +91,7 @@
         }
 
         if ($this->_saveHandler !== null) {
-            Zend_Session::setSaveHandler($this->_saveHandler);
+            Zend_Session::setSaveHandler($this->getSaveHandler());
         }
     }
 }
