--- tests/Zend/Http/UserAgent/AllTests.php  (revision 23864)
+++ tests/Zend/Http/UserAgent/AllTests.php  (working copy)
@@ -26,6 +26,7 @@
 
 require_once 'Zend/Http/UserAgentTest.php';
 require_once 'Zend/Http/UserAgent/AbstractDeviceTest.php';
+require_once 'Zend/Http/UserAgent/MobileTest.php';
 require_once 'Zend/Http/UserAgent/Features/Adapter/WurflApiTest.php';
 
 /**
@@ -47,8 +48,9 @@
     {
         $suite = new PHPUnit_Framework_TestSuite('Zend Framework - Zend_Http - UserAgent');
 
-        $suite->addTestSuite('Zend_Http_UserAgentTest');
+       $suite->addTestSuite('Zend_Http_UserAgentTest');
         $suite->addTestSuite('Zend_Http_UserAgent_AbstractDeviceTest');
+       $suite->addTestSuite('Zend_Http_UserAgent_MobileTest');
         $suite->addTestSuite('Zend_Http_UserAgent_Features_Adapter_WurflApiTest');
 
         return $suite;
