Index: PrepareElementsTest.php
===================================================================
--- PrepareElementsTest.php	(revision 23986)
+++ PrepareElementsTest.php	(working copy)
@@ -147,6 +147,19 @@
             }
         }
     }
+
+    public function testEachSubFormShouldBePrepared()
+    {
+        $subForm = new Zend_Form_SubForm();
+        $subSubForm = new Zend_Form_SubForm();
+
+        $subForm->addSubForm($subSubForm, "subSubForm");
+        $this->form->addSubForm($subForm, "subForm");
+
+        $this->form->render();
+
+        $this->assertEquals("subForm[subSubForm]", $this->form->getSubForm("subForm")->getSubForm("subSubForm")->getElementsBelongTo());
+    }
 }
 
 // Call Zend_Form_Decorator_PrepareElementsTest::main() if this source file is executed directly.
