--- library/Zend/Dojo/View/Helper/Form.php	(revision 23865)
+++ library/Zend/Dojo/View/Helper/Form.php	(working copy)
@@ -70,10 +70,6 @@
             $attribs['id'] = $id;
         }
 
-        if (false === $content) {
-            $content = '';
-        }
-
         $attribs = $this->_prepareDijit($attribs, array(), 'layout');
 
         return $this->getFormHelper()->form($id, $attribs, $content);
