--- library/Zend/Form.php	(Revision 17)
+++ library/Zend/Form.php	(Arbeitskopie)
@@ -1437,11 +1437,11 @@
 
         if (null === $name) {
             foreach ($this->getElements() as $element) {
-                $element->setBelongsTo($array);
+                $element->setBelongsToWithBase($array);
             }
         } else {
             if (null !== ($element = $this->getElement($name))) {
-                $element->setBelongsTo($array);
+                $element->setBelongsToWithBase($array);
             }
         }
     }
Index: library/Zend/Form/Element.php
===================================================================
--- library/Zend/Form/Element.php	(Revision 17)
+++ library/Zend/Form/Element.php	(Arbeitskopie)
@@ -71,6 +71,13 @@
     protected $_belongsTo;
 
     /**
+     * Base array to which element belongs 
+     * 
+     * @var string
+     */
+    protected $_belongsToBase;
+
+    /**
      * Element decorators
      * @var array
      */
@@ -798,6 +805,50 @@
     }
 
     /**
+     * Set array to which element belongs with Base array
+     *
+     * @param  string $array
+     * @return Zend_Form_Element
+     */
+    public function setBelongsToWithBase($array)
+    {
+        $array = $this->filterName($array, true);
+        if ('' !== $array) {
+            $belongsToBase = $this->_belongsToBase;
+            $belongsTo = $this->_belongsTo;
+            $belongsTo = str_replace($belongsToBase, '', $belongsTo);
+            if ('' !== $belongsTo && 0 !== strpos($belongsTo, '[')) {
+                $key = strtok($belongsTo, '[');
+                $belongsTo = str_replace($key, "[$key]", $belongsTo);
+            }
+            $this->_belongsToBase = $array;
+            $this->_belongsTo = $array . $belongsTo;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Return array name to which element belongs without Base array
+     *
+     * @return string
+     */
+    public function getBelongsToWithoutBase()
+    {
+        $belongsTo = str_replace($this->_belongsToBase, '', $this->_belongsTo);
+        if (0 === strpos($belongsTo, '[')) {
+            $belongsTo = ltrim($belongsTo, '[');
+            $key = strtok($belongsTo, ']');
+            $belongsTo = str_replace("$key]", $key, $belongsTo);
+        }
+        if ('' === $belongsTo) {
+            $belongsTo = null;
+        }
+        return $belongsTo;
+    }
+
+
+    /**
      * Return element type
      *
      * @return string
Index: library/Zend/Form/Decorator/PrepareElements.php
===================================================================
--- library/Zend/Form/Decorator/PrepareElements.php	(Revision 17)
+++ library/Zend/Form/Decorator/PrepareElements.php	(Arbeitskopie)
@@ -71,7 +71,7 @@
             $item->setView($view)
                  ->setTranslator($translator);
             if ($item instanceof Zend_Form_Element) {
-                $item->setBelongsTo($belongsTo);
+                $item->setBelongsToWithBase($belongsTo);
             } elseif (null !== $belongsTo && ($item instanceof Zend_Form)) {
                 if ($item->isArray()) {
                     $name = $this->mergeBelongsTo($belongsTo, $item->getElementsBelongTo());
@@ -82,7 +82,7 @@
                 $this->_recursivelyPrepareForm($item);
             } elseif (null !== $belongsTo && ($item instanceof Zend_Form_DisplayGroup)) {
                 foreach ($item as $element) {
-                    $element->setBelongsTo($belongsTo);
+                    $element->setBelongsToWithBase($belongsTo);
                 }
             }
         }
Index: library/Zend/Form/Decorator/FormElements.php
===================================================================
--- library/Zend/Form/Decorator/FormElements.php	(Revision 17)
+++ library/Zend/Form/Decorator/FormElements.php	(Arbeitskopie)
@@ -84,7 +84,7 @@
             $item->setView($view)
                  ->setTranslator($translator);
             if ($item instanceof Zend_Form_Element) {
-                $item->setBelongsTo($belongsTo);
+                $item->setBelongsToWithBase($belongsTo);
             } elseif (null !== $belongsTo && ($item instanceof Zend_Form)) {
                 if ($item->isArray()) {
                     $name = $this->mergeBelongsTo($belongsTo, $item->getElementsBelongTo());
@@ -94,7 +94,7 @@
                 }
             } elseif (null !== $belongsTo && ($item instanceof Zend_Form_DisplayGroup)) {
                 foreach ($item as $element) {
-                    $element->setBelongsTo($belongsTo);
+                    $element->setBelongsToWithBase($belongsTo);
                 }
             }
 
