Index: library/Zend/Db/Statement/Mysqli.php
===================================================================
--- library/Zend/Db/Statement/Mysqli.php	(Revision 7591)
+++ library/Zend/Db/Statement/Mysqli.php	(Arbeitskopie)
@@ -234,10 +234,14 @@
         }
         // send $params as input parameters to the statement
         if ($params) {
-            array_unshift($params, str_repeat('s', count($params)));
+            $stmtParams = array();
+            foreach ($params as $k => &$param) {
+                $stmtParams[$k] = &$param;
+            }
+            array_unshift($stmtParams, str_repeat('s', count($params)));
             call_user_func_array(
                 array($this->_stmt, 'bind_param'),
-                $params
+                $stmtParams
             );
         }
 
