Index: library/Zend/View/Helper/FormRadio.php
===================================================================
--- library/Zend/View/Helper/FormRadio.php	(revision 24786)
+++ library/Zend/View/Helper/FormRadio.php	(working copy)
@@ -157,7 +157,7 @@
 
             // Wrap the radios in labels
             $radio = '<label'
-                    . $this->_htmlAttribs($label_attribs) . ' for="' . $optId . '">'
+                    . $this->_htmlAttribs($label_attribs) . '>'
                     . (('prepend' == $labelPlacement) ? $opt_label : '')
                     . '<input type="' . $this->_inputType . '"'
                     . ' name="' . $name . '"'
