Index: library/Zend/Application/Bootstrap/BootstrapAbstract.php
===================================================================
--- library/Zend/Application/Bootstrap/BootstrapAbstract.php	(revision 21194)
+++ library/Zend/Application/Bootstrap/BootstrapAbstract.php	(working copy)
@@ -355,6 +355,9 @@
                 $spec = (array) $spec;
                 $spec['bootstrap'] = $this;
                 $instance = new $plugin($spec);
+                if (!$instance instanceof Zend_Application_Resource_Resource) {
+                    continue;
+                }
                 $pluginName = $this->_resolvePluginResourceName($instance);
                 unset($this->_pluginResources[$plugin]);
                 $this->_pluginResources[$pluginName] = $instance;
