Index: library/Zend/File/Transfer/Adapter/Http.php
===================================================================
--- library/Zend/File/Transfer/Adapter/Http.php	(revisione 374)
+++ library/Zend/File/Transfer/Adapter/Http.php	(copia locale)
@@ -19,6 +19,9 @@
  * @version   $Id: $
  */
 
+/** Zend_File_Transfer */
+require_once 'Zend/File/Transfer.php';
+
 require_once 'Zend/File/Transfer/Adapter/Abstract.php';
 
 /**
@@ -101,14 +104,18 @@
 
                 // Should never return false when it's tested by the upload validator
                 if (!move_uploaded_file($content['tmp_name'], ($directory . $content['name']))) {
-                    if ($content['options']['ignoreNoFile']) {
-                        $this->_files[$file]['received'] = true;
-                        $this->_files[$file]['filtered'] = true;
-                        continue;
+                    if (Zend_File_Transfer::$_unitTestEnabled) {
+                        rename($content['tmp_name'], ($directory . $content['name']));
+                    } else {
+                        if ($content['options']['ignoreNoFile']) {
+                            $this->_files[$file]['received'] = true;
+                            $this->_files[$file]['filtered'] = true;
+                            continue;
+                        }
+
+                        $this->_files[$file]['received'] = false;
+                        return false;
                     }
-
-                    $this->_files[$file]['received'] = false;
-                    return false;
                 }
 
                 $this->_files[$file]['received'] = true;
Index: library/Zend/File/Transfer.php
===================================================================
--- library/Zend/File/Transfer.php	(revisione 363)
+++ library/Zend/File/Transfer.php	(copia locale)
@@ -30,6 +30,14 @@
 class Zend_File_Transfer
 {
     /**
+     * Whether or not Zend_File_Transfer is being used with unit tests
+     *
+     * @internal
+     * @var bool
+     */
+    public static $_unitTestEnabled = false;
+
+    /**
      * Creates a file processing handler
      *
      * @param string $protocol Protocol to use
Index: library/Zend/Test/PHPUnit/ControllerTestCase.php
===================================================================
--- library/Zend/Test/PHPUnit/ControllerTestCase.php	(revisione 363)
+++ library/Zend/Test/PHPUnit/ControllerTestCase.php	(copia locale)
@@ -15,6 +15,9 @@
 /** Zend_Layout */
 require_once 'Zend/Layout.php';
 
+/** Zend_File_Transfer */
+require_once 'Zend/File/Transfer.php';
+
 /** Zend_Session */
 require_once 'Zend/Session.php';
 
@@ -190,6 +193,7 @@
         Zend_Layout::resetMvcInstance();
         Zend_Controller_Action_HelperBroker::resetHelpers();
         $this->frontController->resetInstance();
+        Zend_File_Transfer::$_unitTestEnabled = true;
         Zend_Session::$_unitTestEnabled = true;
     }
 
Index: library/Zend/Validate/File/Upload.php
===================================================================
--- library/Zend/Validate/File/Upload.php	(revisione 374)
+++ library/Zend/Validate/File/Upload.php	(copia locale)
@@ -167,7 +167,10 @@
             $this->_value = $file;
             switch($content['error']) {
                 case 0:
-                    if (!is_uploaded_file($content['tmp_name'])) {
+                    /** Zend_File_Transfer */
+                    require_once 'Zend/File/Transfer.php';
+
+                    if (!is_uploaded_file($content['tmp_name']) and !Zend_File_Transfer::$_unitTestEnabled) {
                         $this->_throw($file, self::ATTACK);
                     }
                     break;
