Index: Zend/Mail.php
===================================================================
--- Zend/Mail.php	(revision 3646)
+++ Zend/Mail.php	(working copy)
@@ -200,16 +200,17 @@
      *
      * @param string $txt
      * @param string $charset
+     * @param const $encoding Encoding type.  Zend_Mime::ENCODING_* constant
      * @return Zend_Mime_Part
     */
-    public function setBodyText($txt, $charset=null)
+    public function setBodyText($txt, $charset=null, $encoding = Zend_Mime::ENCODING_QUOTEDPRINTABLE)
     {
         if ($charset === null) {
             $charset = $this->_charset;
         }
 
         $mp = new Zend_Mime_Part($txt);
-        $mp->encoding = Zend_Mime::ENCODING_QUOTEDPRINTABLE;
+        $mp->encoding = $encoding;
         $mp->type = Zend_Mime::TYPE_TEXT;
         $mp->disposition = Zend_Mime::DISPOSITION_INLINE;
         $mp->charset = $charset;
