Index: library/Zend/Form/Element.php
===================================================================
--- library/Zend/Form/Element.php	(revision 24787)
+++ library/Zend/Form/Element.php	(working copy)
@@ -904,6 +904,7 @@
     public function getAttribs()
     {
         $attribs = get_object_vars($this);
+        unset($attribs['helper']);
         foreach ($attribs as $key => $value) {
             if ('_' == substr($key, 0, 1)) {
                 unset($attribs[$key]);
Index: library/Zend/Form/Element/Captcha.php
===================================================================
--- library/Zend/Form/Element/Captcha.php	(revision 24787)
+++ library/Zend/Form/Element/Captcha.php	(working copy)
@@ -127,24 +127,6 @@
     }
 
     /**
-     * Return all attributes
-     *
-     * @return array
-     */
-    public function getAttribs()
-    {
-        $attribs = get_object_vars($this);
-        unset($attribs['helper']);
-        foreach ($attribs as $key => $value) {
-            if ('_' == substr($key, 0, 1)) {
-                unset($attribs[$key]);
-            }
-        }
-
-        return $attribs;
-    }
-
-    /**
      * Set options
      *
      * Overrides to allow passing captcha options
