Index: tests/Zend/View/Helper/HeadLinkTest.php
===================================================================
--- tests/Zend/View/Helper/HeadLinkTest.php	(revision 24708)
+++ tests/Zend/View/Helper/HeadLinkTest.php	(working copy)
@@ -467,6 +467,25 @@
         $this->helper->appendStylesheet(array('href' => '/bar/baz', 'id' => 'foo'));
         $this->assertContains('id="foo"', $this->helper->toString());
     }
+
+    /**
+     * @group ZF-11643
+     */
+    public function testSizesAttributeIsSupported()
+    {
+        $this->helper->headLink(
+            array(
+                 'rel'   => 'icon',
+                 'href'  => 'favicon.png',
+                 'sizes' => '16x16',
+                 'type'  => 'image/png',
+            )
+        );
+
+        $expected = '<link href="favicon.png" rel="icon" type="image/png" sizes="16x16" >';
+
+        $this->assertEquals($expected, $this->helper->toString());
+    }
 }
 
 // Call Zend_View_Helper_HeadLinkTest::main() if this source file is executed directly.
