--- library/Zend/Feed/Entry/Abstract.php	(revision 7614)
+++ library/Zend/Feed/Entry/Abstract.php	(working copy)
@@ -17,7 +17,7 @@
  * @package    Zend_Feed
  * @copyright  Copyright (c) 2005-2007 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id:$
+ * @version    $Id$
  */
 
 
Index: library/Zend/Feed/Entry/Rss.php
===================================================================
--- library/Zend/Feed/Entry/Rss.php	(revision 7614)
+++ library/Zend/Feed/Entry/Rss.php	(working copy)
@@ -17,7 +17,7 @@
  * @package    Zend_Feed
  * @copyright  Copyright (c) 2005-2007 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id:$
+ * @version    $Id$
  */
 
 
@@ -44,4 +44,21 @@
      */
     protected $_rootElement = 'item';
 
+
+    /**
+     * Easy access to <content:encoded> elements.
+     *
+     * The namespace is stripped from elements meaning <content:encoded> is 
+     * supposed to be accessed with $item->encoded() however as we allow RSS 
+     * creation with $dataentry->content it is illogical to ignore this alias.
+     *
+     * @return string
+     */
+    public function content()
+    {
+    	$content = parent::__get('encoded');
+    	
+    	return $content->__toString();
+    }
+
 }
