Index: tests/Zend/Db/Table/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/TestCommon.php	(revision 20439)
+++ tests/Zend/Db/Table/TestCommon.php	(working copy)
@@ -1008,12 +1008,15 @@
         $table = $this->_table['bugs'];
         $select = $table->select()
             ->where("$bug_id = ?", 2);
-
+        
+        $oldSelect = clone $select;
+        
         $row = $table->fetchRow($select);
         $this->assertType('Zend_Db_Table_Row_Abstract', $row,
             'Expecting object of type Zend_Db_Table_Row_Abstract, got '.get_class($row));
         $bug_id = $this->_db->foldCase('bug_id');
         $this->assertEquals(2, $row->$bug_id);
+        $this->assertEquals((string)$oldSelect, (string)$select);
     }
 
     public function testTableFetchRowOrderAsc()
