--- tests/Zend/Http/CookieTest.php	(revision 23226)
+++ tests/Zend/Http/CookieTest.php	(working copy)
@@ -451,6 +451,14 @@
     }
 
     /**
+     * @group ZF-10506
+     */
+    public function testPregMatchIsQuoted()
+    {
+        $this->assertFalse(Zend_Http_Cookie::matchCookieDomain('foo.bar.com', 'www.foozbar.com'));
+    }
+
+    /**
      * Data Providers
      */
 
Index: library/Zend/Http/Cookie.php
===================================================================
--- library/Zend/Http/Cookie.php	(revision 23226)
+++ library/Zend/Http/Cookie.php	(working copy)
@@ -395,7 +395,7 @@
 
         // Check for either exact match or suffix match
         return ($cookieDomain == $host ||
-                preg_match("/\.$cookieDomain$/", $host));
+                preg_match('/\.' . preg_quote($cookieDomain) . '$/', $host));
     }
 
     /**
