Index: TestCommon.php
===================================================================
--- TestCommon.php	(revision 19044)
+++ TestCommon.php	(working copy)
@@ -2005,4 +2005,13 @@
         unset($stmt);
         $util->dropTable($tableName);
     }
+
+    /**
+     * @group ZF-7737
+     */
+    public function testQuoteIntoReplacesPlaceholderAtFirstCharacterWhenCountIsNotNull()
+    {
+        $quotedString = $this->_db->quoteInto( "? = bar", "foo", NULL, 1 );
+        $this->assertEquals( $quotedString, "'foo' = bar" );
+    }
 }
