Index: Controller/Action/Helper/ContextSwitch.php
===================================================================
--- Controller/Action/Helper/ContextSwitch.php	(revision 18907)
+++ Controller/Action/Helper/ContextSwitch.php	(working copy)
@@ -243,6 +243,10 @@
         if (empty($contexts)) {
             return;
         }
+        
+        if (null == $format && $this->_defaultContext) {
+            $format = $this->_defaultContext;
+        }
 
         // Return if no context parameter provided
         if (!$context = $request->getParam($this->getContextParam())) {
