Index: Amf/Parse/Amf3/Serializer.php
===================================================================
--- Amf/Parse/Amf3/Serializer.php	(revision 712)
+++ Amf/Parse/Amf3/Serializer.php	(working copy)
@@ -66,6 +66,8 @@
                     $this->_stream->writeDouble($data);
                     break;
                 case Zend_Amf_Constants::AMF3_STRING:
+				case Zend_Amf_Constants::AMF3_XMLSTRING;
+				case Zend_Amf_Constants::AMF3_XML;
                     $this->writeString($data);
                     break;
                 case Zend_Amf_Constants::AMF3_DATE:
@@ -119,6 +121,12 @@
                         $markerType = Zend_Amf_Constants::AMF3_DATE;
                     } else if ($data instanceof Zend_Amf_Value_ByteArray) {
                         $markerType = Zend_Amf_Constants::AMF3_BYTEARRAY;
+					} else if ($data instanceof SimpleXMLElement) {
+						$markerType = Zend_Amf_Constants::AMF3_XMLSTRING;
+						$data = $data->asXML();
+					} else if ($data instanceof DOMDocument) {
+						$markerType = Zend_Amf_Constants::AMF3_XMLSTRING;
+						$data = $data->saveXML();
                     } else {
                         $markerType = Zend_Amf_Constants::AMF3_OBJECT;
                     }
