diff --git a/library/Zend/Locale.php b/library/Zend/Locale.php
index 7431cd7..f8de545 100644
--- a/library/Zend/Locale.php
+++ b/library/Zend/Locale.php
@@ -780,10 +780,10 @@ class Zend_Locale
             if (($locale === 'auto') or (empty($locale))) {
                 $locale = self::$_auto;
             }
+        }
 
-            if (is_array($locale) === true) {
-                $locale = key($locale);
-            }
+        if (is_array($locale) === true) {
+            $locale = key($locale);
         }
 
         // This can only happen when someone extends Zend_Locale and erases the default
