diff --ignore-space-change -ruN Http-orig/Client.php Http/Client.php
--- Http-orig/Client.php	2006-07-11 05:01:05.000000000 -0500
+++ Http/Client.php	2006-08-15 11:31:10.000000000 -0500
@@ -67,6 +67,9 @@
          */
 
         // Follow HTTP redirections, up to $redirectMax of them
+        
+        $status = null;
+        
         for ($redirect = 0; $redirect <= $redirectMax; $redirect++) {
 
             // Build the HTTP request
@@ -79,15 +82,24 @@
             // Open a TCP connection
             $socket = $this->_openConnection();
 
+            
+            
             // Make the HTTP request
             fwrite($socket, implode("\r\n", $request) . "\r\n\r\n");
 
             // Fetch the HTTP response
             $response = $this->_read($socket);
 
+            $finalUri = $this->_uri;
+            $response->setFinalUri($finalUri);
+            $response->setRedirectStatus($status);
+
             // If the HTTP response was a redirect, and we are allowed to follow additional redirects
             if ($response->isRedirect() && $redirect < $redirectMax) {
 
+            	$status = $response->getStatus();
+            	
+
                 // Fetch the HTTP response headers
                 $headers = $response->getHeaders();
 
diff --ignore-space-change -ruN Http-orig/Response.php Http/Response.php
--- Http-orig/Response.php	2006-07-11 05:01:05.000000000 -0500
+++ Http/Response.php	2006-08-15 11:31:34.000000000 -0500
@@ -50,6 +50,21 @@
      */
     protected $_body;
 
+     /**
+     * Final url  uri ultimately used after all redirects have been followed
+     *
+     * @var null|Zend_Uri
+     */
+    protected $_finalUri;
+    
+    /**
+     * Redirect status  status returned by the (last) redirect issued before we hit 
+     * maxRedirects
+     *
+     * @var null|int
+     */
+    protected $_redirectStatus;
+
 
 	/**
 	 * Class Constructor, read and parse HTTP response
@@ -61,6 +76,8 @@
 		$this->_code    = $code;
 		$this->_headers = $headers;
 		$this->_body    = $body;
+		$this->_finalUri  = null;
+		$this->_redirectStatus  = null;
 	}
 
 
@@ -129,4 +146,46 @@
 	{
 		return $this->_headers;
 	}
+	
+	/**
+	 * Return finalUrlPath
+	 * 
+	 * @return Zend_Uri
+	 */
+	public function getFinalUri()
+	{
+		 return $this->_finalUri;
+	}
+	/**
+	 * Set finalUrlPath
+	 * @param string finalUrlPath urlPath of last followed redirect (if any)
+	 * @return array
+	 */
+	public function setFinalUri($finalUri)
+	{
+		 $this->_finalUri = $finalUri;
+	}
+	/**
+	 * Set redirect Status
+	 * @param int status status header of last followed redirect (if any)
+	 * @return array
+	 */
+	public function setRedirectStatus($status)
+	{
+		 $this->_redirectStatus = $status;
+		 
+	}
+	
+	/**
+	 * Return redirectStatus
+	 * 
+	 * @return int
+	 */
+	public function getRedirectStatus()
+	{
+		 return $this->_redirectStatus;
+	}
+	
+	
+	
 }
\ No newline at end of file
