Index: Front.php
===================================================================
--- Front.php	(revision 7891)
+++ Front.php	(working copy)
@@ -310,7 +310,11 @@
      */
     public function addModuleDirectory($path)
     {
-        $dir = new DirectoryIterator($path);
+        try{
+            $dir = new DirectoryIterator($path);
+        }catch(Exception $e){
+            throw new Zend_Controller_Exception("Directory $path not readable");
+        }
         foreach ($dir as $file) {
             if ($file->isDot() || !$file->isDir()) {
                 continue;
