Index: Zend/Controller/Action/Helper/Redirector.php
===================================================================
--- Zend/Controller/Action/Helper/Redirector.php	(revision 1096)
+++ Zend/Controller/Action/Helper/Redirector.php	(working copy)
@@ -325,9 +325,6 @@
         // prevent header injections
         $url = str_replace(array("\n", "\r"), '', $url);
 
-        $exit        = $this->getExit();
-        $prependBase = $this->getPrependBase();
-        $code        = $this->getCode();
         if (null !== $options) {
             if (isset($options['exit'])) {
                 $this->setExit(($options['exit']) ? true : false);
