Index: BootstrapAbstract.php
===================================================================
--- BootstrapAbstract.php	(revision 22206)
+++ BootstrapAbstract.php	(working copy)
@@ -352,7 +352,8 @@
                 continue;
             }
 
-            if (class_exists($plugin)) { //@SEE ZF-7550
+			if (class_exists($plugin) && //@SEE ZF-7550
+				($plugin instanceof Zend_Application_Resource_Resource))	{ //@SEE ZF-8299
                 $spec = (array) $spec;
                 $spec['bootstrap'] = $this;
                 $instance = new $plugin($spec);

