Index: Zend/XmlRpc/Value.php
===================================================================
--- Zend/XmlRpc/Value.php	(Revision 6673)
+++ Zend/XmlRpc/Value.php	(Arbeitskopie)
@@ -243,17 +243,9 @@
                 // If the PHP array is an assosiative array the native type will be 'struct'
                 $obj = 'Zend_XmlRpc_Value_Array';
 
-                // Go over the elements in the array, if the key is different than the index
-                //  it means this array has associative keys and it's a struct
-                if (is_array($value)) { // If the value is not array, it can't be an associated array
-                    $i = 0;
-                    foreach ($value as $key => $element) {
-                        if ($i !== $key) {
-                            $obj = 'Zend_XmlRpc_Value_Struct';
-                            break;
-                        }
-                        ++$i;
-                    }
+                // If the keys in the array are not ascending, it is a struct
+                if (is_array($value) and array_keys($value) !== range(0, count($value) - 1)) {
+                    $obj = 'Zend_XmlRpc_Value_Struct';
                 }
                 return new $obj($value);
 
