--- tests/Zend/Controller/Router/RouteTest.php	(revision 23864)
+++ tests/Zend/Controller/Router/RouteTest.php	(working copy)
@@ -313,6 +313,17 @@
         $this->assertTrue(array_key_exists('2000', $values));
     }
 
+    /** @group ZF-11284 */
+    public function testVariablesWithWildcardAndDefaultValueCanBeOverwritten()
+    {
+        $route = new Zend_Controller_Router_Route(':controller/:action/*', array('foo' => 'bar'));
+        $values = $route->match('c/a/foo/baz');
+
+        $this->assertEquals('c', $values['controller']);
+        $this->assertEquals('a', $values['action']);
+        $this->assertEquals('baz', $values['foo']);
+    }
+
     public function testRootRoute()
     {
         $route = new Zend_Controller_Router_Route('/');
