Index: Zend/Feed/Atom.php
===================================================================
--- Zend/Feed/Atom.php  (revision 4178)
+++ Zend/Feed/Atom.php  (working copy)
@@ -189,2 +189,2 @@
         $id = $this->_element->createElement('id', $array->link);
         $feed->appendChild($id);

-        $title = $this->_element->createElement('title', $array->title);
+        $title = $this->_element->createElement('title');
+        $title->appendChild($this->_element->createCDATASection($array->title));
         $feed->appendChild($title);

         if (isset($array->author)) {
@@ -221,1 +222,1 @@
         $feed->appendChild($link);

         if (isset($array->description)) {
-            $subtitle = $this->_element->createElement('subtitle', $array->description);
+            $subtitle = $this->_element->createElement('subtitle');
+            $subtitle->appendChild($this->_element->createCDATASection($array->description));
             $feed->appendChild($subtitle);
         }

@@ -267,2 +269,2 @@
             $id = $this->_element->createElement('id', isset($dataentry->guid) ? $dataentry->guid : $dataentry->link);
             $entry->appendChild($id);

-            $title = $this->_element->createElement('title', $dataentry->title);
+            $title = $this->_element->createElement('title');
+            $title->appendChild($this->_element->createCDATASection($dataentry->title));
             $entry->appendChild($title);

             $updated = isset($array->lastUpdate) ? $array->lastUpdate : time();
Index: Zend/Feed/Rss.php
===================================================================
--- Zend/Feed/Rss.php   (revision 4178)
+++ Zend/Feed/Rss.php   (working copy)
@@ -125,2 +125,2 @@
     {
         $channel = $this->_element->createElement('channel');

-        $title = $this->_element->createElement('title', $array->title);
+        $title = $this->_element->createElement('title');
+        $title->appendChild($this->_element->createCDATASection($array->title));
         $channel->appendChild($title);

         $link = $this->_element->createElement('link', $array->link);
         $channel->appendChild($link);

-        $description = isset($array->description) ? $array->description : '';
-        $description = $this->_element->createElement('description', $description);
+        $desc = isset($array->description) ? $array->description : '';
+        $description = $this->_element->createElement('description');
+        $description->appendChild($this->_element->createCDATASection($desc));
         $channel->appendChild($description);

         $pubdate = isset($array->lastUpdate) ? $array->lastUpdate : time();
@@ -374,2 +376,2 @@
         foreach ($array as $dataentry) {
             $item = $this->_element->createElement('item');

-            $title = $this->_element->createElement('title', $dataentry->title);
+            $title = $this->_element->createElement('title');
+            $title->appendChild($this->_element->createCDATASection($dataentry->title));
             $item->appendChild($title);

             $link = $this->_element->createElement('link', $dataentry->link);
