--- library/Zend/Mime/Part.php	(revision 22916)
+++ library/Zend/Mime/Part.php	(working copy)
@@ -167,6 +167,10 @@
                           . " boundary=\"" . $this->boundary . '"';
         }
 
+        if ($this->filename) {
+            $contentType .= '; name="' . $this->filename . '"';
+        }
+
         $headers[] = array('Content-Type', $contentType);
 
         if ($this->encoding) {
Index: tests/Zend/Mime/PartTest.php
===================================================================
--- tests/Zend/Mime/PartTest.php	(revision 22916)
+++ tests/Zend/Mime/PartTest.php	(working copy)
@@ -20,6 +20,8 @@
  * @version    $Id$
  */
 
+require_once dirname(dirname(dirname(__FILE__))) . DIRECTORY_SEPARATOR . 'TestHelper.php';
+
 /**
  * Zend_Mime_Part
  */
@@ -63,11 +65,9 @@
 
     public function testHeaders()
     {
-        $expectedHeaders = array('Content-Type: text/plain',
+        $expectedHeaders = array('Content-Type: text/plain; charset=iso8859-1; name="test.txt"',
                                  'Content-Transfer-Encoding: ' . Zend_Mime::ENCODING_BASE64,
-                                 'Content-Disposition: attachment',
-                                 'filename="test.txt"',
-                                 'charset=iso8859-1',
+                                 'Content-Disposition: attachment; filename="test.txt"',
                                  'Content-ID: <4711>');
 
         $actual = $this->part->getHeaders();
