diff --git a/library/Zend/Service/Amazon/SimpleDb/Page.php b/library/Zend/Service/Amazon/SimpleDb/Page.php
index 28e4f9b..38560e9 100644
--- a/library/Zend/Service/Amazon/SimpleDb/Page.php
+++ b/library/Zend/Service/Amazon/SimpleDb/Page.php
@@ -81,7 +81,7 @@ class Zend_Service_Amazon_SimpleDb_Page
      */
     public function isLast()
     {
-        return (null === $this->_token);
+        return (empty($this->_token));
     }
 
     /**
