Index: library/Zend/Soap/Server.php
===================================================================
--- library/Zend/Soap/Server.php	(revision 22135)
+++ library/Zend/Soap/Server.php	(working copy)
@@ -210,6 +210,8 @@
                     $this->setWsdl($value);
                     break;
                 case 'featues':
+                	trigger_error(__CLASS__ . '::' . __METHOD__ . ': the option featues is deprecated as of 1.10.x and will be removed with 2.0.0; use features instead', E_USER_NOTICE);
+                case 'features':
                     $this->setSoapFeatures($value);
                     break;
                 case 'cache_wsdl':
