--- library/Zend/Controller/Dispatcher.php	(revision 2498)
+++ library/Zend/Controller/Dispatcher.php	(working copy)
@@ -559,7 +559,7 @@
                 include_once $spec;
                 if (!class_exists($class)) {
                     while (strstr($class, '_')) {
-                        $class = substr($class, strpos($class, '_'));
+                        $class = substr($class, strpos($class, '_')+1);
                         if (class_exists($class)) {
                             return $class;
                         }
