Index: library/Zend/Gdata/Gapps.php
===================================================================
--- library/Zend/Gdata/Gapps.php	(revision 6247)
+++ library/Zend/Gdata/Gapps.php	(working copy)
@@ -495,7 +495,7 @@
      * Create a new user from a UserEntry.
      * 
      * @param Zend_Gdata_Gapps_UserEntry $user The user entry to insert.
-     * @param string $url (optional) The URI where the user should be 
+     * @param string $uri (optional) The URI where the user should be 
      *          uploaded to. If null, the default user creation URI for 
      *          this domain will be used.
      * @return Zend_Gdata_Gapps_UserEntry The inserted user entry as 
@@ -517,9 +517,9 @@
     /**
      * Create a new nickname from a NicknameEntry.
      * 
-     * @param Zend_Gdata_Gapps_NicknameEntry $user The nickname entry to 
+     * @param Zend_Gdata_Gapps_NicknameEntry $nickname The nickname entry to 
      *          insert.
-     * @param string $url (optional) The URI where the nickname should be 
+     * @param string $uri (optional) The URI where the nickname should be 
      *          uploaded to. If null, the default nickname creation URI for 
      *          this domain will be used.
      * @return Zend_Gdata_Gapps_NicknameEntry The inserted nickname entry as 
@@ -541,9 +541,9 @@
     /**
      * Create a new email list from an EmailListEntry.
      * 
-     * @param Zend_Gdata_Gapps_EmailListEntry $user The email list entry to 
-     *          insert.
-     * @param string $url (optional) The URI where the email list should be 
+     * @param Zend_Gdata_Gapps_EmailListEntry $emailList The email list entry
+     *          to insert.
+     * @param string $uri (optional) The URI where the email list should be 
      *          uploaded to. If null, the default email list creation URI for 
      *          this domain will be used.
      * @return Zend_Gdata_Gapps_EmailListEntry The inserted email list entry 
@@ -565,9 +565,9 @@
     /**
      * Create a new email list recipient from an EmailListRecipientEntry.
      * 
-     * @param Zend_Gdata_Gapps_EmailListRecipientEntry $user The recipient 
+     * @param Zend_Gdata_Gapps_EmailListRecipientEntry $recipient The recipient 
      *          entry to insert.
-     * @param string $url (optional) The URI where the recipient should be 
+     * @param string $uri (optional) The URI where the recipient should be 
      *          uploaded to. If null, the default recipient creation URI for 
      *          this domain will be used.
      * @return Zend_Gdata_Gapps_EmailListRecipientEntry The inserted 
@@ -577,7 +577,7 @@
      * @throws Zend_Gdata_Gapps_ServiceException
      * @return Zend_Gdata_App_Feed
      */
-    public function insertEmailListRecipient($emailList, $uri = null)
+    public function insertEmailListRecipient($recipient, $uri = null)
     {
         if ($uri === null) {
             require_once 'Zend/Gdata/App/InvalidArgumentException.php';
@@ -586,7 +586,7 @@
         } elseif ($uri instanceof Zend_Gdata_Gapps_EmailListEntry) {
             $uri = $uri->getLink('edit')->href;
         }
-        $newEntry = $this->insertEntry($emailList, $uri, 'Zend_Gdata_Gapps_EmailListRecipientEntry');
+        $newEntry = $this->insertEntry($recipient, $uri, 'Zend_Gdata_Gapps_EmailListRecipientEntry');
         return $newEntry;
     }
     
