Index: library/Zend/Http/Response.php
===================================================================
--- library/Zend/Http/Response.php	(revision 15570)
+++ library/Zend/Http/Response.php	(working copy)
@@ -253,7 +253,7 @@
         $body = '';
 
         // Decode the body if it was transfer-encoded
-        switch ($this->getHeader('transfer-encoding')) {
+        switch (strtolower($this->getHeader('transfer-encoding'))) {
 
             // Handle chunked body
             case 'chunked':
Index: library/Zend/Http/Client/Adapter/Socket.php
===================================================================
--- library/Zend/Http/Client/Adapter/Socket.php	(revision 15570)
+++ library/Zend/Http/Client/Adapter/Socket.php	(working copy)
@@ -244,7 +244,7 @@
 
         // If we got a 'transfer-encoding: chunked' header
         if (isset($headers['transfer-encoding'])) {
-            if ($headers['transfer-encoding'] == 'chunked') {
+            if (strtolower($headers['transfer-encoding']) == 'chunked') {
                 do {
                     $line  = @fgets($this->socket);
                     $chunk = $line;
