Index: library/Zend/View/Helper/HeadScript.php
===================================================================
--- library/Zend/View/Helper/HeadScript.php	(revision 23530)
+++ library/Zend/View/Helper/HeadScript.php	(working copy)
@@ -371,7 +371,6 @@
             throw $e;
         }
 
-        $this->_isValid($value);
         return $this->getContainer()->offsetSet($index, $value);
     }
 
