Index: Zend/Measure/Abstract.php
===================================================================
--- Zend/Measure/Abstract.php	(revision 10909)
+++ Zend/Measure/Abstract.php	(working copy)
@@ -19,8 +19,19 @@
  * @version   $Id$
  */
 
+/**
+ * @see Zend_Locale
+ */
 require_once 'Zend/Locale.php';
+
+/**
+ * @see Zend_Locale_Math
+ */
 require_once 'Zend/Locale/Math.php';
+
+/**
+ * @see Zend_Locale_Format
+ */
 require_once 'Zend/Locale/Format.php';
 
 /**
