Index: Standard.php
===================================================================
--- Standard.php	(revision 7891)
+++ Standard.php	(working copy)
@@ -327,6 +327,11 @@
 
         $controllerDirs      = $this->getControllerDirectory();
         $this->_curModule    = $this->_defaultModule;
+        if(!isset($controllerDirs[$this->_defaultModule])){
+            require_once 'Zend/Controller/Dispatcher/Exception.php';
+            throw new Zend_Controller_Dispatcher_Exception("Module $this->_defaultModule does not
+             exist in controllers directories");
+        }
         $this->_curDirectory = $controllerDirs[$this->_defaultModule];
         $module = $request->getModuleName();
         if ($this->isValidModule($module)) {
@@ -374,6 +379,11 @@
         $module              = $request->getModuleName();
         $controllerDirs      = $this->getControllerDirectory();
         $this->_curModule    = $this->_defaultModule;
+        if(!isset($controllerDirs[$this->_defaultModule])){
+            require_once 'Zend/Controller/Dispatcher/Exception.php';
+            throw new Zend_Controller_Dispatcher_Exception("Module $this->_defaultModule does not
+             exist in controllers directories");
+        }
         $this->_curDirectory = $controllerDirs[$this->_defaultModule];
         if ($this->isValidModule($module)) {
             $found = false;
