--- library/Zend/Service/Twitter.php	(revision 14708)
+++ library/Zend/Service/Twitter.php	(working copy)
@@ -357,10 +357,12 @@
      *
      * @param  string $status
      * @param  int $in_reply_to_status_id
+     * @param  string $source A custom source to display as 'sent via <source>' in tweets. 
+     *                        must be an official twitter source.
      * @return Zend_Rest_Client_Result
      * @throws Zend_Service_Twitter_Exception if message is too short or too long
      */
-    public function statusUpdate($status, $in_reply_to_status_id = null)
+    public function statusUpdate($status, $in_reply_to_status_id = null, $source = null)
     {
         $this->_init();
         $path = '/statuses/update.xml';
@@ -380,6 +382,10 @@
         if(is_numeric($in_reply_to_status_id) && !empty($in_reply_to_status_id)) {
             $data['in_reply_to_status_id'] = $in_reply_to_status_id;
         }
+        
+        if(!is_null($source) && is_string($source)) {
+          $data['source'] = $source;
+        }
 
         //$this->status = $status;
         $response = $this->restPost($path, $data);
