Index: ../trunk/library/Zend/Form/Element/Hash.php
===================================================================
--- ../trunk/library/Zend/Form/Element/Hash.php	(revision 19479)
+++ ../trunk/library/Zend/Form/Element/Hash.php	(working copy)
@@ -229,13 +229,11 @@
     /**
      * Render CSRF token in form
      *
-     * @param  Zend_View_Interface $view
-     * @return string
+     * @return void
      */
-    public function render(Zend_View_Interface $view = null)
+    protected function _preRender()
     {
         $this->initCsrfToken();
-        return parent::render($view);
     }
 
     /**
