Index: /net/marshall/home/web/zend/library/Zend/Db/Table/Rowset.php
===================================================================
--- /net/marshall/home/web/zend/library/Zend/Db/Table/Rowset.php	(revision 849)
+++ /net/marshall/home/web/zend/library/Zend/Db/Table/Rowset.php	(working copy)
@@ -82,6 +82,7 @@
         $this->_db    = $config['db'];
         $this->_table = $config['table'];
         $this->_data  = $config['data'];
+        $this->_rowClassName = $config['rowClassName'];
         
         // set the count of rows
         $this->_count = count($this->_data);
@@ -114,7 +115,8 @@
         // do we already have a row object for this position?
         if (empty($this->_rows[$this->_pointer])) {
             // create a row object
-            $this->_rows[$this->_pointer] = new Zend_Db_Table_Row(array(
+            $class = $this->_rowClassName;
+            $this->_rows[$this->_pointer] = new $class(array(
                 'db'    => $this->_db,
                 'table' => $this->_table,
                 'data'  => $this->_data[$this->_pointer]
