Index: library/Zend/View/Helper/HtmlList.php
===================================================================
--- library/Zend/View/Helper/HtmlList.php	(revision 7763)
+++ library/Zend/View/Helper/HtmlList.php	(working copy)
@@ -57,13 +57,13 @@
 
         foreach ($items as $item) {
             if (!is_array($item)) {
-                if ($escape) $item = $this->view->escape($item); 
+                if ($escape) $item = $this->view->escape($item);
                 $list .= '<li>' . $item . '</li>';
             } else {
                 if (5 < strlen($list)) {
-                    $list = substr($list, 0, strlen($list) - 5) . $this->htmlList($item, $ordered) . '</li>';
+                    $list = substr($list, 0, strlen($list) - 5) . $this->htmlList($item, $ordered, $attribs, $escape) . '</li>';
                 } else {
-                    $list .= '<li>' . $this->htmlList($item, $ordered) . '</li>';
+                    $list .= '<li>' . $this->htmlList($item, $ordered, $attribs, $escape) . '</li>';
                 }
             }
         }
