diff --git a/bin/zf.sh b/bin/zf.sh
index 877c78f..f6981db 100755
--- bin/zf.sh
+++ bin/zf.sh
@@ -32,13 +32,8 @@ fi
 if test "@php_dir@" != '@'php_dir'@'; then
     PHP_DIR="@php_dir@"
 else
-    SELF_LINK="$0"
-    SELF_LINK_TMP="$(readlink "$SELF_LINK")"
-    while test -n "$SELF_LINK_TMP"; do
-        SELF_LINK="$SELF_LINK_TMP"
-        SELF_LINK_TMP="$(readlink "$SELF_LINK")"
-    done
-    PHP_DIR="$(dirname "$SELF_LINK")"
+    SELF_LOCATION="$(which "$0")"
+    PHP_DIR="$(dirname "$SELF_LOCATION")"
 fi

 $PHP_BIN -d safe_mode=Off -f "$PHP_DIR/zf.php" -- $@
