Index: tests/Zend/Form/Element/HashTest.php
===================================================================
--- tests/Zend/Form/Element/HashTest.php	(revision 19078)
+++ tests/Zend/Form/Element/HashTest.php	(working copy)
@@ -199,6 +199,16 @@
     }
 
     /**
+     * @group ZF-7404
+     */
+    public function testShouldRenderHashTokenIfRenderedThroughMagicCall()
+    {
+        $this->element->setView($this->getView());
+        $html = $this->element->renderViewHelper();
+        $this->assertContains($this->element->getHash(), $html, 'Html is: ' . $html);
+    }
+
+    /**
      * Used by test methods susceptible to ZF-2794, marks a test as incomplete
      *
      * @link   http://framework.zend.com/issues/browse/ZF-2794
