Index: library/ZendX/JQuery/View/Helper/JQuery/Container.php
===================================================================
--- library/ZendX/JQuery/View/Helper/JQuery/Container.php	(revision 24462)
+++ library/ZendX/JQuery/View/Helper/JQuery/Container.php	(working copy)
@@ -112,6 +112,20 @@
      * @var Boolean
      */
     protected $_uiEnabled = false;
+	
+    /**
+     * jQuery UI Lang Library Enabled
+     *
+     * @var Boolean
+     */
+    protected $_uiLangEnabled = false;
+    
+    /**
+     * jQuery UI Theme loader Enabled
+     *
+     * @var Boolean
+     */
+    protected $_uiThemeEnabled = false;
 
     /**
      * Local jQuery UI Path. Use Google CDN if
@@ -127,6 +141,13 @@
      * @var String
      */
     protected $_uiVersion = ZendX_JQuery::DEFAULT_UI_VERSION;
+    
+    /**
+     * jQuery UI Theme
+     *
+     * @var String
+     */
+    protected $_uiTheme = "";
 
     /**
      * Load CDN Path from SSL or Non-SSL?
@@ -207,6 +228,16 @@
     {
         return $this->_version;
     }
+    
+    /**
+     * Get jQuery library theme
+     *
+     * @return string
+     */
+    public function getTheme()
+    {
+        return $this->_uiTheme;
+    }
 
     /**
      * Use CDN, using version specified. Currently supported
@@ -309,6 +340,61 @@
     }
 
     /**
+     * Enable jQuery UI Library Rendering
+     *
+     * @return ZendX_JQuery_View_Helper_JQuery_Container
+     */
+    public function uiLangEnable()
+    {
+        $this->uiEnable();
+        $this->_uiLangEnabled = true;
+        return $this;
+    }
+    
+    /**
+     * Enable jQuery UI Theme Rendering
+     *
+     * @return ZendX_JQuery_View_Helper_JQuery_Container
+     */
+    protected function uiThemeEnable()
+    {
+        $this->uiEnable();
+        $this->_uiThemeEnabled = true;
+        return $this;
+    }
+
+    /**
+     * Disable jQuery UI Library Rendering
+     *
+     * @return ZendX_JQuery_View_Helper_JQuery_Container
+     */
+    public function uiLangDisable()
+    {
+        $this->_uiLangEnabled = false;
+        return $this;
+    }
+
+    /**
+     * Check wheater currently the jQuery UI library is enabled.
+     *
+     * @return boolean
+     */
+    public function uiLangIsEnabled()
+    {
+         return $this->_uiLangEnabled;
+    }
+    
+    /**
+     * Check wheater currently the jQuery UI Theme is enabled.
+     *
+     * @return boolean
+     */
+    public function uiThemeIsEnabled()
+    {
+         return $this->_uiThemeEnabled;
+    }
+
+    /**
      * Set jQuery UI version used.
      * 
      * @param  string $version
@@ -374,6 +460,16 @@
     {
     	return $this->_uiPath;
     }
+	
+    /**
+     * Return the local jQuery UI Path if set.
+     *
+     * @return string
+     */
+    public function getUiLangPath()
+    {
+    	return $this->_uiLangPath;
+    }
 
     /**
      * Proxies to getUiPath() for consistency in function naming.
@@ -384,6 +480,16 @@
     {
         return $this->getUiPath();
     }
+	
+    /**
+     * Proxies to getUiLangPath() for consistency in function naming.
+     *
+     * @return string
+     */
+    public function getUiLangLocalPath()
+    {
+        return $this->getUiLangPath();
+    }
 
     /**
      * Is the jQuery Ui loaded from local scope?
@@ -394,6 +500,16 @@
     {
     	return (null===$this->_uiPath ? false : true);
     }
+	
+    /**
+     * Is the jQuery Ui Language loaded from local scope?
+     *
+     * @return boolean
+     */
+    public function useUiLangLocal()
+    {
+    	return (null===$this->_uiLangPath ? false : true);
+    }
 
     /**
      * Is the jQuery Ui enabled and loaded from CDN?
@@ -673,6 +789,10 @@
     	if( ($this->getRenderMode() & ZendX_JQuery::RENDER_STYLESHEETS) == 0) {
             return '';
     	}
+    
+        if($this->uiThemeIsEnabled()) {
+            $this->addStylesheet($this->_getJQueryUiThemePath());
+        }
 
         foreach ($this->getStylesheets() as $stylesheet) {
             $stylesheets[] = $stylesheet;
@@ -715,6 +835,11 @@
                 $uiPath = $this->_getJQueryUiLibraryPath();
                 $scriptTags .= '<script type="text/javascript" src="'.$uiPath.'"></script>'.PHP_EOL;
             }
+			
+            if($this->uiLangIsEnabled()) {
+                $uiLangPath = $this->_getJQueryUiLangLibraryPath();
+                $scriptTags .= '<script type="text/javascript" src="'.$uiLangPath.'"></script>'.PHP_EOL;
+            }
 
             if(ZendX_JQuery_View_Helper_JQuery::getNoConflictMode() == true) {
                 $scriptTags .= '<script type="text/javascript">var $j = jQuery.noConflict();</script>'.PHP_EOL;
@@ -816,7 +941,7 @@
             $baseUri = $this->_getJQueryLibraryBaseCdnUri();
             $source = $baseUri .
                 ZendX_JQuery::CDN_SUBFOLDER_JQUERY .
-                $this->getCdnVersion() .
+                $this->getVersion() .
             	ZendX_JQuery::CDN_JQUERY_PATH_GOOGLE;
         }
 
@@ -832,11 +957,55 @@
             $baseUri = $this->_getJQueryLibraryBaseCdnUri();
             $uiPath = $baseUri.
                 ZendX_JQuery::CDN_SUBFOLDER_JQUERYUI .
-                $this->getUiCdnVersion() .
+                $this->getUiVersion() .
                 "/jquery-ui.min.js";
         } else if($this->useUiLocal()) {
             $uiPath = $this->getUiPath();
         }
         return $uiPath;
     }
-}
+
+    /**
+     * @return string
+     */
+    protected function _getJQueryUiLangLibraryPath()
+    {
+        if($this->useUiCdn()) {
+            $baseUri = $this->_getJQueryLibraryBaseCdnUri();
+            $uiLangPath = $baseUri.
+                ZendX_JQuery::CDN_SUBFOLDER_JQUERYUI .
+                $this->getUiVersion() .
+                "/i18n/jquery-ui-i18n.min.js";
+        } else if($this->useUiLangLocal()) {
+            $uiLangPath = $this->getUiLangPath();
+        }
+        return $uiLangPath;
+    }
+    
+    /**
+     * @return string
+     */
+    protected function _getJQueryUiThemePath()
+    {
+        if($this->useUiCdn()) {
+            $baseUri = $this->_getJQueryLibraryBaseCdnUri();
+            $uiThemePath = $baseUri.
+                ZendX_JQuery::CDN_SUBFOLDER_JQUERYUI .
+                $this->getUiVersion() . '/themes/'. $this->getTheme().
+                "/jquery-ui.css";
+        } else if($this->useUiThemeLocal()) {
+            $uiThemePath = $this->getUiThemePath();
+        }
+        return $uiThemePath;
+    }
+
+	public function setLanguageCode($strCode){
+		$this->uiLangEnable ();
+		$this->addOnLoad("$.datepicker.setDefaults( $.datepicker.regional[ '".$strCode."' ] );");
+	}
+	
+	public function setTheme($strTheme){
+	    $this->_uiTheme = $strTheme;
+		$this->uiThemeEnable ();
+	}
+}
\ No newline at end of file
