Index: library/Zend/Controller/Action.php
===================================================================
--- library/Zend/Controller/Action.php	(revision 24159)
+++ library/Zend/Controller/Action.php	(working copy)
@@ -505,14 +505,18 @@
                 $this->_classMethods = get_class_methods($this);
             }
 
-            // preDispatch() didn't change the action, so we can continue
-            if ($this->getInvokeArg('useCaseSensitiveActions') || in_array($action, $this->_classMethods)) {
-                if ($this->getInvokeArg('useCaseSensitiveActions')) {
-                    trigger_error('Using case sensitive actions without word separators is deprecated; please do not rely on this "feature"');
+            // If pre-dispatch hooks introduced a redirect then stop dispatch
+            // @see ZF-7496
+            if (!($this->getResponse()->isRedirect())) {                
+                // preDispatch() didn't change the action, so we can continue
+                if ($this->getInvokeArg('useCaseSensitiveActions') || in_array($action, $this->_classMethods)) {
+                    if ($this->getInvokeArg('useCaseSensitiveActions')) {
+                        trigger_error('Using case sensitive actions without word separators is deprecated; please do not rely on this "feature"');
+                    }
+                    $this->$action();
+                } else {
+                    $this->__call($action, array());
                 }
-                $this->$action();
-            } else {
-                $this->__call($action, array());
             }
             $this->postDispatch();
         }
Index: tests/Zend/Test/PHPUnit/ControllerTestCaseTest.php
===================================================================
--- tests/Zend/Test/PHPUnit/ControllerTestCaseTest.php	(revision 24159)
+++ tests/Zend/Test/PHPUnit/ControllerTestCaseTest.php	(working copy)
@@ -781,6 +781,61 @@
                : gettype($boot);
         $this->assertTrue($boot === $this->testCase->bootstrap->getBootstrap(), $type);
     }
+    
+    /**
+     * @group ZF-7496
+     * @dataProvider providerRedirectWorksAsExpectedFromHookMethodsInActionController
+     */
+    public function testRedirectWorksAsExpectedFromHookMethodsInActionController($dispatchTo)
+    {
+        $this->testCase->getFrontController()->setControllerDirectory(dirname(__FILE__) . '/_files/application/controllers');
+        $this->testCase->dispatch($dispatchTo);
+        $this->testCase->assertRedirectTo('/login');
+        $this->assertNotEquals('action body', $this->testCase->getResponse()->getBody());
+    }
+    
+    /**
+     * Data provider for testRedirectWorksAsExpectedFromHookMethodsInActionController
+     * @return array
+     */
+    public function providerRedirectWorksAsExpectedFromHookMethodsInActionController()
+    {
+        return array(
+            array('/zend-test-redirect-from-init/baz'),
+            array('/zend-test-redirect-from-pre-dispatch/baz')
+        );
+    }
+    
+    /**
+     * @group ZF-7496
+     * @dataProvider providerRedirectWorksAsExpectedFromHookMethodsInFrontControllerPlugin
+     */
+    public function testRedirectWorksAsExpectedFromHookMethodsInFrontControllerPlugin($pluginName)
+    {
+        require_once dirname(__FILE__) . "/_files/application/plugins/RedirectFrom{$pluginName}.php";
+        $className = "Application_Plugin_RedirectFrom{$pluginName}";
+        
+        $fc = $this->testCase->getFrontController();
+        $fc->setControllerDirectory(dirname(__FILE__) . '/_files/application/controllers')
+           ->registerPlugin(new $className());
+        $this->testCase->dispatch('/');
+        $this->testCase->assertRedirectTo('/login');
+        $this->assertNotEquals('action body', $this->testCase->getResponse()->getBody());
+    }
+    
+    /**
+     * Data provider for testRedirectWorksAsExpectedFromHookMethodsInFrontControllerPlugin
+     * @return array
+     */
+    public function providerRedirectWorksAsExpectedFromHookMethodsInFrontControllerPlugin()
+    {
+        return array(
+            array('RouteStartup'),
+            array('RouteShutdown'),
+            array('DispatchLoopStartup'),
+            array('PreDispatch')
+        );
+    }
 }
 
 // Concrete test case class for testing purposes
Index: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromPreDispatch.php
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromPreDispatch.php	(revision 0)
+++ tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromPreDispatch.php	(revision 0)
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Application_Plugin_RedirectFromPreDispatch extends Zend_Controller_Plugin_Abstract
+{
+    public function preDispatch(Zend_Controller_Request_Abstract $r)
+    {
+        return $this->getResponse()->setRedirect('/login');
+    }
+}

Property changes on: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromPreDispatch.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteShutdown.php
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteShutdown.php	(revision 0)
+++ tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteShutdown.php	(revision 0)
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Application_Plugin_RedirectFromRouteShutdown extends Zend_Controller_Plugin_Abstract
+{
+    public function routeShutdown(Zend_Controller_Request_Abstract $r)
+    {
+        return $this->getResponse()->setRedirect('/login');
+    }
+}

Property changes on: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteShutdown.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromDispatchLoopStartup.php
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromDispatchLoopStartup.php	(revision 0)
+++ tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromDispatchLoopStartup.php	(revision 0)
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Application_Plugin_RedirectFromDispatchLoopStartup extends Zend_Controller_Plugin_Abstract
+{
+    public function dispatchLoopStartup(Zend_Controller_Request_Abstract $r)
+    {
+        return $this->getResponse()->setRedirect('/login');
+    }
+}

Property changes on: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromDispatchLoopStartup.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteStartup.php
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteStartup.php	(revision 0)
+++ tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteStartup.php	(revision 0)
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Application_Plugin_RedirectFromRouteStartup extends Zend_Controller_Plugin_Abstract
+{
+    public function routeStartup(Zend_Controller_Request_Abstract $r)
+    {
+        return $this->getResponse()->setRedirect('/login');
+    }
+}

Property changes on: tests/Zend/Test/PHPUnit/_files/application/plugins/RedirectFromRouteStartup.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromInitController.php
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromInitController.php	(revision 0)
+++ tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromInitController.php	(revision 0)
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class ZendTestRedirectFromInitController extends Zend_Controller_Action
+{
+    public function init()
+    {
+        return $this->_redirect('/login');
+    }
+
+    public function bazAction()
+    {
+        $this->_helper->viewRenderer->setNoRender(true);
+        echo 'action body';
+    }
+}

Property changes on: tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromInitController.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromPreDispatchController.php
===================================================================
--- tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromPreDispatchController.php	(revision 0)
+++ tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromPreDispatchController.php	(revision 0)
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @category   Zend
+ * @package    Zend_Test
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class ZendTestRedirectFromPreDispatchController extends Zend_Controller_Action
+{
+    public function preDispatch()
+    {
+        return $this->_redirect('/login');
+    }
+
+    public function bazAction()
+    {
+        $this->_helper->viewRenderer->setNoRender(true);
+        echo 'action body';
+    }
+}

Property changes on: tests/Zend/Test/PHPUnit/_files/application/controllers/ZendTestRedirectFromPreDispatchController.php
___________________________________________________________________
Added: svn:keywords
   + Id

