### Eclipse Workspace Patch 1.0
#P zf-standard
Index: library/Zend/Controller/Request/Http.php
===================================================================
--- library/Zend/Controller/Request/Http.php	(revision 22187)
+++ library/Zend/Controller/Request/Http.php	(working copy)
@@ -990,6 +990,12 @@
             if (!empty($headers[$header])) {
                 return $headers[$header];
             }
+            $header = strtolower($header);
+            foreach ($headers as $key => $value) {
+                if (strtolower($key) == $header) {
+                    return $value;
+                }
+            }
         }
 
         return false;
