Index: Abstract.php
===================================================================
--- Abstract.php	(revision 14118)
+++ Abstract.php	(working copy)
@@ -207,6 +207,28 @@
     }
 
     /**
+     * Unset row field value
+     *
+     * @param  string $columnName The column key.
+     * @return Zend_Db_Table_Row_Abstract
+     * @throws Zend_Db_Table_Row_Exception
+     */
+    public function __unset($columnName)
+    {
+        $columnName = $this->_transformColumn($columnName);
+        if (!array_key_exists($columnName, $this->_data)) {
+            require_once 'Zend/Db/Table/Row/Exception.php';
+            throw new Zend_Db_Table_Row_Exception("Specified column \"$columnName\" is not in the row");
+        }
+        if ($this->isConnected() && in_array($columnName, $this->_table->info('primary'))) {
+            require_once 'Zend/Db/Table/Row/Exception.php';
+            throw new Zend_Db_Table_Row_Exception("Specified column \"$columnName\" is a primary key and should not be unsetted");
+        }
+        unset($this->_data[$columnName]);
+        return $this;
+    }
+
+    /**
      * Test existence of row field
      *
      * @param  string  $columnName   The column key.
