*** ZendFramework-1.9.7/library/Zend/XmlRpc/Value/BigInteger.php	2009-09-30 15:35:47.000000000 +0200
--- ZendFramework-1.9.7_fixed/library/Zend/XmlRpc/Value/BigInteger.php	2010-01-20 14:19:40.000000000 +0100
***************
*** 48,54 ****
      {
          require_once 'Zend/Crypt/Math/BigInteger.php';
          $this->_integer = new Zend_Crypt_Math_BigInteger();
!         $this->_value = $this->_integer->init($this->_value);
  
          $this->_type = self::XMLRPC_TYPE_I8;
      }
--- 48,54 ----
      {
          require_once 'Zend/Crypt/Math/BigInteger.php';
          $this->_integer = new Zend_Crypt_Math_BigInteger();
!         $this->_value = $this->_integer->init($value);
  
          $this->_type = self::XMLRPC_TYPE_I8;
      }
***************
*** 60,65 ****
       */
      public function getValue()
      {
!         return $this->_integer;
      }
  }
--- 60,65 ----
       */
      public function getValue()
      {
!         return $this->_value;
      }
  }
