Index: incubator/library/Zend/View/Helper/FormErrors.php
===================================================================
--- incubator/library/Zend/View/Helper/FormErrors.php   (revision 7829)
+++ incubator/library/Zend/View/Helper/FormErrors.php   (working copy)
@@ -63,9 +63,14 @@
             $attribs = $this->_htmlAttribs($options);
             $start   = sprintf($start, $attribs);
         }
+        
+        $cleanErrors = array();
+        foreach ((array) $errors as $validator => $error) {
+            $cleanErrors[$validator] = $this->view->escape($error);
+        }
 
         $html  = $start
-               . implode($this->getElementSeparator(), (array) $errors)
+               . implode($this->getElementSeparator(), (array) $cleanErrors)
                . $this->getElementEnd();
 
         return $html;
