Index: library/Zend/Form/Decorator/ViewHelper.php
===================================================================
--- library/Zend/Form/Decorator/ViewHelper.php	(revision 25167)
+++ library/Zend/Form/Decorator/ViewHelper.php	(working copy)
@@ -196,7 +196,8 @@
             if ($element instanceof $type) {
                 if (stristr($type, 'button')) {
                     $element->content = $element->getLabel();
-                    return null;
+
+                    return $element->getValue();
                 }
                 return $element->getLabel();
             }
