Index: tests/ZendX/Console/Process/UnixTest.php
===================================================================
--- tests/ZendX/Console/Process/UnixTest.php	(revision 22855)
+++ tests/ZendX/Console/Process/UnixTest.php	(working copy)
@@ -157,6 +157,30 @@
         
         $process->stop();
     }
+
+    /**
+     * @group ZF-9991
+     */
+    public function testManagerShouldBeAbleToWaitAChild()
+    {
+        $process = new sleepingProcess();
+        $process->start();
+
+        $actual = null;
+        while (true) {
+            $pid = pcntl_wait($status);
+            if (-1 == $pid) {
+                break;
+            }
+
+            $actualPid = $pid;
+            $actualStatus = $status;
+            usleep(1000);
+        }
+
+        $this->assertGreaterThan(1, $actualPid);
+        $this->assertSame(0, $actualStatus);
+    }
 }
 
 class simpleProcess extends ZendX_Console_Process_Unix
Index: library/ZendX/Console/Process/Unix.php
===================================================================
--- library/ZendX/Console/Process/Unix.php	(revision 22855)
+++ library/ZendX/Console/Process/Unix.php	(working copy)
@@ -213,8 +213,6 @@
 
         // @see http://www.php.net/manual/en/function.pcntl-fork.php#41150
         @ob_end_flush();
-        
-        pcntl_signal(SIGCHLD, SIG_IGN);
 
         $pid = @pcntl_fork();
         if ($pid === -1) {
