Index: tests/Zend/View/Helper/FormTest.php
===================================================================
--- tests/Zend/View/Helper/FormTest.php (revision 24478)
+++ tests/Zend/View/Helper/FormTest.php (revision 24476)
@@ -167,18 +167,6 @@
         $form = $this->helper->form('FormName', array('action' => '/foo', 'method' => 'get'));
         $this->assertNotRegexp('/<form[^>]*(name="FormName")/', $form);
     }
-
-    /**
-     * @group ZF-11747
-     */
-    public function testClosingTagIsPrintedWhenContentIsOmitted()
-    {
-        $form = $this->helper->form('FormName');
-        // Check that opening tag was printed
-        $this->assertContains('<form', $form);
-        // Check that closing tag was printed
-        $this->assertContains('</form>', $form);
-    }
 }

 // Call Zend_View_Helper_FormTest::main() if this source file is executed directly.
Index: library/Zend/View/Helper/Form.php
===================================================================
--- library/Zend/View/Helper/Form.php   (revision 24478)
+++ library/Zend/View/Helper/Form.php   (revision 24476)
@@ -73,11 +73,10 @@
                . '>';

         if (false !== $content) {
-            $xhtml .= $content;
+            $xhtml .= $content
+                   .  '</form>';
         }

-        $xhtml .= '</form>';
-
         return $xhtml;
     }
 }