Index: Zend/Db/Adapter/Sqlsrv.php
===================================================================
--- Zend/Db/Adapter/Sqlsrv.php	(revision 22500)
+++ Zend/Db/Adapter/Sqlsrv.php	(working copy)
@@ -437,9 +437,9 @@
         $sql    = "exec sp_columns @table_name = " . $this->quoteIdentifier($tableName, true);
         $stmt   = $this->query($sql);
         $result = $stmt->fetchAll(Zend_Db::FETCH_NUM);
-		
-		// ZF-7698
-		$stmt->closeCursor();
+
+        // ZF-7698
+        $stmt->closeCursor();
         
         if (count($result) == 0) {
             return array();
Index: Zend/Db/Statement/Sqlsrv.php
===================================================================
--- Zend/Db/Statement/Sqlsrv.php	(revision 22500)
+++ Zend/Db/Statement/Sqlsrv.php	(working copy)
@@ -39,16 +39,22 @@
 
     /**
      * The connection_stmt object original string.
+     *
+     * @var string
      */
     protected $_originalSQL;
 
     /**
      * Column names.
+     *
+     * @var array
      */
-    protected $_keys;
+    protected $_keys = array();
 
     /**
      * Query executed
+     *
+     * @return boolean
      */
     protected $_executed = false;
 
@@ -376,11 +382,11 @@
             require_once 'Zend/Db/Statement/Sqlsrv/Exception.php';
             throw new Zend_Db_Statement_Sqlsrv_Exception(sqlsrv_errors());
         }
-		
-		// reset column keys
-		$this->_keys = null;
 
-		return true;
+        // reset column keys
+        $this->_keys = array();
+
+        return true;
     }
 
     /**
@@ -411,8 +417,8 @@
 
         return $num_rows;
     }
-	
-	/**
+
+    /**
      * Returns an array containing all of the result set rows.
      *
      * @param int $style OPTIONAL Fetch mode.
Index: Zend/Db/Table/Abstract.php
===================================================================
--- Zend/Db/Table/Abstract.php	(revision 22500)
+++ Zend/Db/Table/Abstract.php	(working copy)
@@ -90,7 +90,7 @@
     /**
      * Optional Zend_Db_Table_Definition object
      *
-     * @var unknown_type
+     * @var Zend_Db_Table_Definition
      */
     protected $_definition = null;
 
Index: Zend/Db/Statement.php
===================================================================
--- Zend/Db/Statement.php	(revision 22500)
+++ Zend/Db/Statement.php	(working copy)
@@ -476,7 +476,7 @@
     /**
      * Gets the resource or object setup by the
      * _parse
-     * @return unknown_type
+     * @return resource|object The driver level statement object/resource.
      */
     public function getDriverStatement()
     {
