Index: Element.php
===================================================================
--- Element.php	(revision 936)
+++ Element.php	(working copy)
@@ -141,8 +141,20 @@
      * Element label
      * @var string
      */
-    protected $_label;
-
+    protected $_label;
+    
+    /**
+     * Static plugin loaders (improves performance)
+     * @var array
+     */
+    static protected $_staticLoaders = array();
+
+    /**
+     * Option to disable static loaders if needed
+     * @var boolean
+     */
+    static protected $_useStaticLoaders = true;
+    
     /**
      * Plugin loaders for filter and validator chains
      * @var array
@@ -1000,15 +1012,28 @@
                     $prefixSegment = 'Form_Decorator';
                     $pathSegment   = 'Form/Decorator';
                 }
-                if (!isset($this->_loaders[$type])) {
-                    require_once 'Zend/Loader/PluginLoader.php';
-                    $this->_loaders[$type] = new Zend_Loader_PluginLoader(
-                        array('Zend_' . $prefixSegment . '_' => 'Zend/' . $pathSegment . '/')
-                    );
+                if (!isset($this->_loaders[$type])) {
+                    if (self::$_useStaticLoaders && isset(self::$_staticLoaders[$type]))
+                    {
+                      $this->_loaders[$type] = self::$_staticLoaders[$type];
+                    }
+                    else
+                    {
+                      require_once 'Zend/Loader/PluginLoader.php';
+                      $this->_loaders[$type] = new Zend_Loader_PluginLoader(
+                          array('Zend_' . $prefixSegment . '_' => 'Zend/' . $pathSegment . '/')
+                      );
+                      
+                      if (self::$_useStaticLoaders)
+                      {
+                        // this is our new static loader
+                        self::$_staticLoaders[$type] = $this->_loaders[$type];
+                      }
+                    }
                 }
                 return $this->_loaders[$type];
             default:
                 require_once 'Zend/Form/Exception.php';
                 throw new Zend_Form_Exception(sprintf('Invalid type "%s" provided to getPluginLoader()', $type));
         }
     }