Index: library/Zend/Mail/Transport/Abstract.php
===================================================================
--- library/Zend/Mail/Transport/Abstract.php	(revision 20423)
+++ library/Zend/Mail/Transport/Abstract.php	(working copy)
@@ -112,7 +112,7 @@
      * - {@link $header} - message header
      * - {@link $body} - message body
      */
-    abstract protected function _sendMail();
+    abstract public function sendMail();
 
     /**
      * Return all mail headers as an array
Index: library/Zend/Mail/Transport/Smtp.php
===================================================================
--- library/Zend/Mail/Transport/Smtp.php	(revision 20423)
+++ library/Zend/Mail/Transport/Smtp.php	(working copy)
@@ -111,7 +111,7 @@
      * @param  string $host OPTIONAL (Default: 127.0.0.1)
      * @param  array|null $config OPTIONAL (Default: null)
      * @return void
-     * 
+     *
      * @todo Someone please make this compatible
      *       with the SendMail transport class.
      */
@@ -155,11 +155,12 @@
      *
      * @param Zend_Mail_Protocol_Abstract $client
      *
-     * @return void
+     * @return Zend_Mail_Transport_Smtp Provides a fluent interface
      */
     public function setConnection(Zend_Mail_Protocol_Abstract $connection)
     {
         $this->_connection = $connection;
+        return $this;
     }
 
 
@@ -170,6 +171,9 @@
      */
     public function getConnection()
     {
+        if ($this->_connection === null) {
+            $this->_setDefaultConnection();
+        }
         return $this->_connection;
     }
 
@@ -180,22 +184,12 @@
      * developer to add a custom adapter if required before mail is sent.
      *
      * @return void
-     * @todo Rename this to sendMail, it's a public method...
      */
-    public function _sendMail()
+    public function sendMail()
     {
         // If sending multiple messages per session use existing adapter
         if (!($this->_connection instanceof Zend_Mail_Protocol_Smtp)) {
-            // Check if authentication is required and determine required class
-            $connectionClass = 'Zend_Mail_Protocol_Smtp';
-            if ($this->_auth) {
-                $connectionClass .= '_Auth_' . ucwords($this->_auth);
-            }
-            if (!class_exists($connectionClass)) {
-                require_once 'Zend/Loader.php';
-                Zend_Loader::loadClass($connectionClass);
-            }
-            $this->setConnection(new $connectionClass($this->_host, $this->_port, $this->_config));
+            $this->getConnection();
             $this->_connection->connect();
             $this->_connection->helo($this->_name);
         } else {
@@ -214,8 +208,39 @@
         // Issue DATA command to client
         $this->_connection->data($this->header . Zend_Mime::LINEEND . $this->body);
     }
+    
+    /**
+     * Send an email via the SMTP connection protocol
+     *
+     * The connection via the protocol adapter is made just-in-time to allow a
+     * developer to add a custom adapter if required before mail is sent.
+     *
+     * @return void
+     * @deprecated
+     */
+    public function _sendMail()
+    {
+        trigger_error('The method _sendMail is deprecated. Use sendMail', E_USER_NOTICE);
+        $this->sendMail();
+    }
 
     /**
+     * @return void
+     */
+    protected function _setDefaultConnection()
+    {
+        // Check if authentication is required and determine required class
+        $connectionClass = 'Zend_Mail_Protocol_Smtp';
+        if ($this->_auth) {
+            $connectionClass .= '_Auth_' . ucwords($this->_auth);
+        }
+        if (!class_exists($connectionClass)) {
+            require_once 'Zend/Loader.php';
+            Zend_Loader::loadClass($connectionClass);
+        }
+        $this->setConnection(new $connectionClass($this->_host, $this->_port, $this->_config));
+    }
+    /**
      * Format and fix headers
      *
      * Some SMTP servers do not strip BCC headers. Most clients do it themselves as do we.
@@ -240,4 +265,5 @@
         // Prepare headers
         parent::_prepareHeaders($headers);
     }
+
 }
Index: library/Zend/Mail/Transport/Sendmail.php
===================================================================
--- library/Zend/Mail/Transport/Sendmail.php	(revision 20423)
+++ library/Zend/Mail/Transport/Sendmail.php	(working copy)
@@ -77,7 +77,6 @@
         $this->parameters = $parameters;
     }
 
-
     /**
      * Send mail using PHP native mail()
      *
@@ -85,7 +84,7 @@
      * @return void
      * @throws Zend_Mail_Transport_Exception on mail() failure
      */
-    public function _sendMail()
+    public function sendMail()
     {
         set_error_handler(array($this, '_handleMailErrors'));
         if ($this->parameters === null) {
@@ -113,6 +112,20 @@
         }
     }
 
+    /**
+     * Send mail using PHP native mail()
+     *
+     * @access public
+     * @return void
+     * @throws Zend_Mail_Transport_Exception on mail() failure
+     * @deprecated
+     */
+    public function _sendMail()
+    {
+        trigger_error('The method _sendMail is deprecated. Use sendMail', E_USER_NOTICE);
+        $this->sendMail();
+    }
+
 
     /**
      * Format and fix headers
