--- library/Zend/Db/Statement/Pdo.php	(revision 14150)
+++ library/Zend/Db/Statement/Pdo.php	(working copy)
@@ -140,6 +140,9 @@
             $parameter = ":$parameter";
         }
         try {
+            // Ensure that the parameter and its value are recorded.
+            $this->_bindParam[$parameter] = $value;
+
             if ($type === null) {
                 return $this->_stmt->bindValue($parameter, $value);
             } else {
