Index: Server.php
===================================================================
--- Server.php	(revision 21231)
+++ Server.php	(working copy)
@@ -916,13 +916,14 @@
     public function fault($fault = null, $code = "Receiver")
     {
         if ($fault instanceof Exception) {
-            $class = get_class($fault);
-            if (in_array($class, $this->_faultExceptions)) {
-                $message = $fault->getMessage();
-                $eCode   = $fault->getCode();
-                $code    = empty($eCode) ? $code : $eCode;
-            } else {
-                $message = 'Unknown error';
+            $message = 'Unknown error';
+            foreach ($this->_faultExceptions as $registeredFaults) {
+                if ($fault instanceOf $registeredFaults) {
+                    $message = $fault->getMessage();
+                    $eCode   = $fault->getCode();
+                    $code    = empty($eCode) ? $code : $eCode;
+                    break;
+                }
             }
         } elseif(is_string($fault)) {
             $message = $fault;
