Index: Zend/AllTests.php
===================================================================
--- Zend/AllTests.php	(revision 6038)
+++ Zend/AllTests.php	(working copy)
@@ -71,7 +71,7 @@
 require_once 'Zend/VersionTest.php';
 require_once 'Zend/ViewTest.php';
 require_once 'Zend/XmlRpc/AllTests.php';
-require_once 'Zend/Json/AllTests.php';
+require_once 'Zend/Json/JsonXMLTest.php';
 
 
 class Zend_AllTests
