--- library/Zend/Controller/Router/Route.php    (revision 914)
+++ library/Zend/Controller/Router/Route.php    (working copy)
@@ -157,10 +157,13 @@
      * @param array An array of variable and value pairs used as parameters 
      * @return string Route path with user submitted parameters
      */
-    public function assemble($data = array())
+    public function assemble($data = array(), $build = true)
     {
 
         $url = array();
+        if ($build) {
+               $data = array_merge($this->_values, $data);
+        }
 
         foreach ($this->_parts as $key => $part) {
