Index: library/Zend/View/Helper/FormSelect.php
===================================================================
--- library/Zend/View/Helper/FormSelect.php	(revision 24628)
+++ library/Zend/View/Helper/FormSelect.php	(working copy)
@@ -123,8 +123,14 @@
                 if (null !== $translator) {
                     $opt_value = $translator->translate($opt_value);
                 }
+                
+                // Normalize label before using in element id
+                $opt_id_regex = $this->view->doctype()->isHtml5()
+                    ? '/\s/' : '/[^a-z0-9-_:.]/i';
+                $opt_id_value = preg_replace($opt_id_regex,'',$opt_value);
+                
                 $opt_id = ' id="' . $this->view->escape($id) . '-optgroup-'
-                        . $this->view->escape($opt_value) . '"';
+                        . $this->view->escape($opt_id_value) . '"';
                 $list[] = '<optgroup'
                         . $opt_disable
                         . $opt_id
Index: tests/Zend/View/Helper/FormSelectTest.php
===================================================================
--- tests/Zend/View/Helper/FormSelectTest.php	(revision 24628)
+++ tests/Zend/View/Helper/FormSelectTest.php	(working copy)
@@ -330,6 +330,40 @@
         $this->assertRegexp('/<optgroup[^>]*?id="baz-optgroup-bar"[^>]*?"bar"[^>]*?/', $html, $html);
     }
  
+    /**
+     * @group ZF-12050
+     */
+    public function testHtml4OptGroupLabelIsProperlySantiziedBeforeUseInId()
+    {
+        $html = $this->helper->formSelect(array(
+            'name'    => 'baz',
+            'options' => array(
+                'Foo && BAR' => array(
+                    '1' => 'one',
+                    '2' => 'two'
+                )
+            )
+        ));
+        $this->assertRegexp('/<optgroup[^>]*?id="baz-optgroup-FooBAR"[^>]*?/', $html, $html);
+    }
+ 
+    /**
+     * @group ZF-12050
+     */
+    public function testHtml5OptGroupLabelIsProperlySantiziedBeforeUseInId()
+    {
+        $this->helper->view->doctype('HTML5');
+        $html = $this->helper->formSelect(array(
+            'name'    => 'baz',
+            'options' => array(
+                'Foo && BAR' => array(
+                    '1' => 'one',
+                    '2' => 'two'
+                )
+            )
+        ));
+        $this->assertRegexp('/<optgroup[^>]*?id="baz-optgroup-Foo&amp;&amp;BAR"[^>]*?/', $html, $html);
+    }
 }
 
 // Call Zend_View_Helper_FormSelectTest::main() if this sourc