Index: library/Zend/Gdata/Calendar/Extension/WebContent.php
===================================================================
--- library/Zend/Gdata/Calendar/Extension/WebContent.php	(revision 6539)
+++ library/Zend/Gdata/Calendar/Extension/WebContent.php	(working copy)
@@ -32,7 +32,7 @@
  * @copyright  Copyright (c) 2005-2007 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Gdata_Calendar_Extension_WebContent extends Zend_Gdata_App_Extension
+class Zend_Gdata_Calendar_Extension_WebContent extends Zend_Gdata_Extension
 {
 
     protected $_rootNamespace = 'gCal';
