Index: Model.php
===================================================================
--- Model.php	(revision 22314)
+++ Model.php	(working copy)
@@ -67,6 +67,11 @@
         }
 
         $modelsDirectory = self::_getModelsDirectoryResource($profile, $moduleName);
+        
+        if (!$modelsDirectory instanceof Zend_Tool_Project_Profile_Resource) {
+            return false;
+        }
+        
         return (($modelsDirectory->search(array('modelFile' => array('modelName' => $modelName)))) instanceof Zend_Tool_Project_Profile_Resource);
     }
     
