*** Backend.php.orig	Tue Oct 24 12:04:21 2006
--- Backend.php	Tue Oct 24 12:04:39 2006
***************
*** 17,27 ****
   * @subpackage Backend
   * @copyright  Copyright (c) 2006 Zend Technologies USA Inc. (http://www.zend.com)
   * @license    http://framework.zend.com/license/new-bsd     New BSD License
   */
  
!  
  /**
   * @package    Zend_Cache
   * @subpackage Backend
   * @copyright  Copyright (c) 2006 Zend Technologies USA Inc. (http://www.zend.com)
   * @license    http://framework.zend.com/license/new-bsd     New BSD License
--- 17,33 ----
   * @subpackage Backend
   * @copyright  Copyright (c) 2006 Zend Technologies USA Inc. (http://www.zend.com)
   * @license    http://framework.zend.com/license/new-bsd     New BSD License
   */
  
! 
! /**
!  * Zend_Cache
!  */
! require_once 'Zend/Cache.php'; 
! 
! 
  /**
   * @package    Zend_Cache
   * @subpackage Backend
   * @copyright  Copyright (c) 2006 Zend Technologies USA Inc. (http://www.zend.com)
   * @license    http://framework.zend.com/license/new-bsd     New BSD License
***************
*** 88,97 ****
--- 94,108 ----
                  Zend_Cache::throwException("Incorrect option name : $name");
              }
              if (array_key_exists($name, $this->_directives)) {
                  $this->_directives[$name] = $value;
              }
+             if ($name=='logging') {
+                 if ((!class_exists('Zend_Log', false)) && ($value)) {
+                     Zend_Cache::throwException('logging feature is on but Zend_Log is not "loaded"');    
+                 }
+             }
          }
      } 
      
      /**
       * Set an option
