Index: library/Zend/Controller/Request/Http.php
===================================================================
--- library/Zend/Controller/Request/Http.php	(revision 23365)
+++ library/Zend/Controller/Request/Http.php	(revision )
@@ -1033,7 +1033,10 @@
         $name   = $this->getServer('SERVER_NAME');
         $port   = $this->getServer('SERVER_PORT');
 
-        if (($scheme == self::SCHEME_HTTP && $port == 80) || ($scheme == self::SCHEME_HTTPS && $port == 443)) {
+        if(null === $name) {
+            return '';
+        }
+        elseif (($scheme == self::SCHEME_HTTP && $port == 80) || ($scheme == self::SCHEME_HTTPS && $port == 443)) {
             return $name;
         } else {
             return $name . ':' . $port;
Index: tests/Zend/Controller/Request/HttpTest.php
===================================================================
--- tests/Zend/Controller/Request/HttpTest.php	(revision 23365)
+++ tests/Zend/Controller/Request/HttpTest.php	(revision )
@@ -949,9 +949,18 @@
 
         $this->assertEquals( '/module/controller/action', $pathInfo, $pathInfo);
     }
-    
+
+    /**
+     * @group ZF-9899
+     */
+    public function testHostNameShouldBeEmpty()
+    {
+        $request = new Zend_Controller_Request_Http();
+        $this->assertEquals('', $request->getHttpHost(), 'HttpHost should be :');
-}
-
+    }
+    
+}
+
 // Call Zend_Controller_Request_HttpTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == "Zend_Controller_Request_HttpTest::main") {
     Zend_Controller_Request_HttpTest::main();
