--- tests/Zend/Dojo/View/Helper/DojoTest.php	(revision 24401)
+++ tests/Zend/Dojo/View/Helper/DojoTest.php	(working copy)
@@ -909,6 +909,74 @@
         $this->assertRegexp('#(function\\(\\){alert\\(\'foo\'\\);})#', $output);
     }
 
+    /**
+     * @group ZF-11519
+     */
+    public function testCdnSslShouldBeDisabledByDefault()
+    {
+        $this->helper->enable();
+        
+        $this->assertFalse($this->helper->getCdnSsl());
+    }
+    
+    /**
+     * @group ZF-11519
+     */
+    public function testCdnSslCanBeEnabled()
+    {
+        $this->helper->enable();
+        $this->helper->setCdnSsl(true);
+        
+        $this->assertTrue($this->helper->getCdnSsl());
+    }
+    
+    /**
+     * @group ZF-11519
+     */
+    public function testEnablingCdnSslChangesBaseCdnIfDefaultGoogleCdn()
+    {
+        $this->helper->enable();
+        $this->helper->setCdnSsl(true);
+        
+        $this->assertSame(Zend_Dojo::CDN_BASE_GOOGLE_SSL, $this->helper->getCdnBase());
+    }
+    
+    /**
+     * @group ZF-11519
+     */
+    public function testEnableCdnSslForGoogleCdn()
+    {
+        $this->helper->enable();
+        $this->helper->setCdnSsl(true);
+        $this->helper->setCdnBase(Zend_Dojo::CDN_BASE_GOOGLE);
+        
+        $this->assertSame(Zend_Dojo::CDN_BASE_GOOGLE_SSL, $this->helper->getCdnBase());
+    }
+    
+    /**
+     * @group ZF-11519
+     */
+    public function testSettingGoogleSslCdnEnablesSsl()
+    {
+        $this->helper->enable();
+        $this->helper->setCdnBase(Zend_Dojo::CDN_BASE_GOOGLE_SSL);
+        
+        $this->assertSame(Zend_Dojo::CDN_BASE_GOOGLE_SSL, $this->helper->getCdnBase());
+        $this->assertTrue($this->helper->getCdnSsl());
+    }
+    
+    /**
+     * @group ZF-11519
+     * @expectedException Zend_Dojo_View_Exception
+     */
+    public function testEnableCdnSslForAolCdnThrowsException()
+    {
+        $this->helper->enable();
+        $this->helper->setCdnSsl(true);
+        $this->helper->setCdnBase(Zend_Dojo::CDN_BASE_AOL);
+        $this->helper->getCdnBase();
+    }
+    
     public function setupDojo()
     {
         $this->helper->requireModule('dijit.layout.ContentPane')
