Index: library/Zend/Cache/Backend/Static.php
===================================================================
--- library/Zend/Cache/Backend/Static.php	(revision 23621)
+++ library/Zend/Cache/Backend/Static.php	(working copy)
@@ -294,6 +294,10 @@
      */
     public function remove($id)
     {
+        if (ctype_xdigit($id)) {
+            $id = $this->_decodeId($id);
+        }
+
         if (!$this->_verifyPath($id)) {
             Zend_Cache::throwException('Invalid cache id: does not match expected public_dir path');
         }
@@ -329,6 +333,10 @@
      */
     public function removeRecursively($id)
     {
+        if (ctype_xdigit($id)) {
+            $id = $this->_decodeId($id);
+        }
+
         if (!$this->_verifyPath($id)) {
             Zend_Cache::throwException('Invalid cache id: does not match expected public_dir path');
         }
Index: library/Zend/Controller/Action/Helper/Cache.php
===================================================================
--- library/Zend/Controller/Action/Helper/Cache.php	(revision 23621)
+++ library/Zend/Controller/Action/Helper/Cache.php	(working copy)
@@ -129,6 +129,8 @@
      */
     public function removePage($relativeUrl, $recursive = false)
     {
+        $relativeUrl = $this->_encodeCacheId($relativeUrl);
+
         $cache = $this->getCache(Zend_Cache_Manager::PAGECACHE);
         if ($recursive) {
             $backend = $cache->getBackend();
Index: tests/Zend/Cache/StaticBackendTest.php
===================================================================
--- tests/Zend/Cache/StaticBackendTest.php	(revision 23621)
+++ tests/Zend/Cache/StaticBackendTest.php	(working copy)
@@ -87,6 +87,12 @@
         $this->assertFalse($this->_instance->test(bin2hex('/barbar')));
     }
 
+    public function testRemoveWithHexadecimalId()
+    {
+        $this->assertTrue($this->_instance->remove(bin2hex('/bar')));
+        $this->assertFalse($this->_instance->test(bin2hex('/bar')));
+    }
+
     public function testOptionsSetTagCache()
     {
         $test = new Zend_Cache_Backend_Static(array('tag_cache'=>$this->_innerCache));
Index: tests/Zend/Controller/Action/Helper/CacheTest.php
===================================================================
--- tests/Zend/Controller/Action/Helper/CacheTest.php	(revision 23621)
+++ tests/Zend/Controller/Action/Helper/CacheTest.php	(working copy)
@@ -178,14 +178,14 @@
 {
     public function remove($id)
     {
-        if ($id == '/foo') {return 'verified';}
+        if ($id == bin2hex('/foo')) {return 'verified';}
     }
 }
 class Mock_Zend_Cache_Page_2 extends Zend_Cache_Backend
 {
     public function removeRecursively($id)
     {
-        if ($id == '/foo') {return 'verified';}
+        if ($id == bin2hex('/foo')) {return 'verified';}
     }
 }
 class Mock_Zend_Cache_Page_3 extends Zend_Cache_Core
