--- FormSelect.php	2006-12-15 14:29:37.000000000 -0500
+++ FormSelect-eric.php	2006-12-15 14:29:34.000000000 -0500
@@ -112,30 +112,58 @@
                    . $this->_htmlAttribs($attribs)
                    . ">\n\t";
             
             // build the list of options
             $list = array();
             foreach ($options as $opt_value => $opt_label) {
-            
-                // option value and label
-                $opt = '<option'
-                     . ' value="' . htmlspecialchars($opt_value, ENT_COMPAT, 'UTF-8') . '"'
-                     . ' label="' . htmlspecialchars($opt_label, ENT_COMPAT, 'UTF-8') . '"';
-                     
-                // selected?
-                if (in_array($opt_value, $value)) {
-                    $opt .= ' selected="selected"';
+                
+                if (is_array($opt_label)) {
+                    $list[] = '<optgroup'
+                            . ' name="' . htmlspecialchars($opt_value, ENT_COMPAT, 'UTF-8') . '">';
+                            
+                    foreach($opt_label as $opt_value => $label) {
+                        $list[] = $this->_build($value, $opt_value, $label);
+                    }
+                    
+                    $list[] = '</optgroup>';
+                } else {
+                    $list[] = $this->_build($value, $opt_value, $opt_label);
                 }
-                
-                // close and add
-                $opt .= '>' . htmlspecialchars($opt_label, ENT_COMPAT, 'UTF-8') . "</option>";
-                $list[] = $opt;
             }
             
             // add the options to the xhtml and close the select
             $xhtml .= implode("\n\t", $list) . "\n</select>";
             
         }
         
         return $xhtml;
     }
+    
+    /**
+     *  Builds the actual <option> tag
+     * 
+     * @access private
+     * 
+     * @param mixed $value The option value to mark as 'selected'
+     * 
+     * @param string $opt_value Options Value
+     * 
+     * @param string $opt_label Options Label
+     * 
+     * @return string Option Tag XHTML
+     */
+    private function _build($selected, $opt_value, $opt_label)
+    {
+        $opt = '<option'
+             . ' value="' . htmlspecialchars($opt_value, ENT_COMPAT, 'UTF-8') . '"'
+             . ' label="' . htmlspecialchars($opt_label, ENT_COMPAT, 'UTF-8') . '"';
+             
+        // selected?
+        if (in_array($opt_value, $value)) {
+            $opt .= ' selected="selected"';
+        }
+        
+        $opt .= '>' . htmlspecialchars($opt_label, ENT_COMPAT, 'UTF-8') . "</option>";
+        
+        return $opt;
+    }
 }
