--- Abstract.old	2009-12-28 18:53:30.000000000 -0500
+++ Abstract.php	2009-12-28 22:11:32.000000000 -0500
@@ -1266,15 +1266,24 @@
             foreach ($whereList as $keyValueSets) {
                 $whereAndTerms = array();
                 foreach ($keyValueSets as $keyPosition => $keyValue) {
-                    $type = $this->_metadata[$keyNames[$keyPosition]]['DATA_TYPE'];
-                    $columnName = $this->_db->quoteIdentifier($keyNames[$keyPosition], true);
-                    $whereAndTerms[] = $this->_db->quoteInto(
-                        $tableName . '.' . $columnName . ' = ?',
-                        $keyValue, $type);
+                    // Check only for null or an empty array to avoid an empty where clause
+                    if ($keyValue !== null && ($keyValue !== array())) {
+                        $type = $this->_metadata[$keyNames[$keyPosition]]['DATA_TYPE'];
+                        $columnName = $this->_db->quoteIdentifier($keyNames[$keyPosition], true);
+                        $whereAndTerms[] = $this->_db->quoteInto(
+                            $tableName . '.' . $columnName . ' = ?',
+                            $keyValue, $type);
+                    }
                 }
-                $whereOrTerms[] = '(' . implode(' AND ', $whereAndTerms) . ')';
+
+                if (!empty($whereAndTerms)) {
+                    $whereOrTerms[] = '(' . implode(' AND ', $whereAndTerms) . ')';
+                }
+            }
+
+            if (!empty($whereOrTerms)) {
+                $whereClause = '(' . implode(' OR ', $whereOrTerms) . ')';
             }
-            $whereClause = '(' . implode(' OR ', $whereOrTerms) . ')';
         }
 
         // issue ZF-5775 (empty where clause should return empty rowset)
