Index: library/Zend/Dojo/View/Helper/Dojo/Container.php
===================================================================
--- library/Zend/Dojo/View/Helper/Dojo/Container.php	(revision 21858)
+++ library/Zend/Dojo/View/Helper/Dojo/Container.php	(working copy)
@@ -273,6 +273,15 @@
                     } else {
                         $this->disable();
                     }
+                    break;
+                case 'useprogrammatic':
+                    Zend_Dojo_View_Helper_Dojo::setUseProgrammatic(intval($value));
+                    break;
+                case 'usedeclarative':
+                    if(intval($value)) {
+                        Zend_Dojo_View_Helper_Dojo::setUseDeclarative();
+                    }
+                    break;
             }
         }
 
Index: tests/Zend/Application/Resource/DojoTest.php
===================================================================
--- tests/Zend/Application/Resource/DojoTest.php	(revision 21858)
+++ tests/Zend/Application/Resource/DojoTest.php	(working copy)
@@ -110,11 +110,7 @@
             'localPath'          => '/ofc/ZF/Rules/',
         );
 
-        $resource = new Zend_Application_Resource_Dojo($options);
-        $resource->setBootstrap($this->bootstrap);
-        $resource->init();
-        $resource->getBootstrap()->bootstrap('view');
-        $dojo = $resource->getBootstrap()->view->dojo();
+        $dojo = $this->_initDojoResource($options);
 
         $test = array(
             'requireModules' => $dojo->getModules(),
@@ -122,6 +118,44 @@
         );
         $this->assertEquals($options, $test);
     }
+
+    public function testUseProgrammaticOptionPassed()
+    {
+        $options = array('useProgrammatic' => 1);
+
+        $dojo = $this->_initDojoResource($options);
+
+        $this->assertEquals(true, Zend_Dojo_View_Helper_Dojo::useProgrammatic());
+    }
+
+    public function testUseProgrammaticNoScriptOptionPassed()
+    {
+        $options = array('useProgrammatic' => -1);
+
+        $dojo = $this->_initDojoResource($options);
+
+        $this->assertEquals(true, Zend_Dojo_View_Helper_Dojo::useProgrammaticNoScript());
+    }
+
+    public function testUseDeclarativeOptionPassed()
+    {
+        $options = array('useDeclarative' => 1);
+
+        $dojo = $this->_initDojoResource($options);
+
+        $this->assertEquals(true, Zend_Dojo_View_Helper_Dojo::useDeclarative());
+    }
+
+    protected function _initDojoResource($options)
+    {
+    	$resource = new Zend_Application_Resource_Dojo($options);
+        $resource->setBootstrap($this->bootstrap);
+        $resource->init();
+        $resource->getBootstrap()->bootstrap('view');
+        $dojo = $resource->getBootstrap()->view->dojo();
+
+        return $dojo;
+    }
 }
 
 if (PHPUnit_MAIN_METHOD == 'Zend_Application_Resource_DojoTest::main') {
