Index: external/Zend_Framework/Zend/Log/Adapter/File.php
===================================================================
--- external/Zend_Framework/Zend/Log/Adapter/File.php	(Revision 2796)
+++ external/Zend_Framework/Zend/Log/Adapter/File.php	(Arbeitskopie)
@@ -186,7 +186,7 @@
         }
         
         if (! $this->_fileResource = @fopen($this->_filename, $this->_accessMode, false)) {
-            throw new Zend_Log_Adapter_Exception("Log file \"$filename\" could not be opened");
+            throw new Zend_Log_Adapter_Exception("Log file \"{$this->_filename}\" could not be opened");
         }
 
         return true;
