--- library/Zend/Json/Server.php	2009-08-13 12:50:25.000000000 +0200
+++ Zend/Json/Server.php	2009-08-13 12:55:06.000000000 +0200
@@ -44,6 +44,12 @@
      * @var bool
      */
     protected $_autoEmitResponse = true;
+    
+    /**
+     * Flag: whether or not to auto handle exceptions
+     * @var bool
+     */
+    protected $_autoHandleExceptions = true;
 
     /**
      * @var bool Flag; allow overwriting existing methods when creating server definition
@@ -300,6 +306,28 @@
         return $this->_autoEmitResponse;
     }
 
+    /**
+     * Set flag indicating whether or not to auto handle exceptions
+     * 
+     * @param  bool $flag 
+     * @return Zend_Json_Server
+     */
+    public function setAutoHandleExceptions($flag)
+    {
+        $this->_autoHandleExceptions = (bool) $flag;
+        return $this;
+    }
+
+    /**
+     * Will we auto handle exceptions?
+     * 
+     * @return bool
+     */
+    public function autoHandleExceptions()
+    {
+        return $this->_autoHandleExceptions;
+    }
+    
     // overloading for SMD metadata
     /**
      * Overload to accessors of SMD object
@@ -529,7 +557,12 @@
         try {
             $result = $this->_dispatch($invocable, $params);
         } catch (Exception $e) {
-            return $this->fault($e->getMessage(), $e->getCode());
+            if ($this->autoHandleExceptions()) {
+        	    return $this->fault($e->getMessage(), $e->getCode());
+            } else {
+                throw $e;
+            }
+            
         }
 
         $this->getResponse()->setResult($result);
