--- library/Zend/Loader/Autoloader/Resource.php	(revision 18161)
+++ library/Zend/Loader/Autoloader/Resource.php	(working copy)
@@ -171,7 +171,7 @@
             return false;
         }
 
-        $final = substr($class, strlen($lastMatch));
+        $final = substr($class, strlen($lastMatch) + 1);
         $path = $this->_components[$lastMatch];
         return include $path . '/' . str_replace('_', '/', $final) . '.php';
     }
