Index: library/Zend/Captcha/Image.php
===================================================================
--- library/Zend/Captcha/Image.php	(revision 24672)
+++ library/Zend/Captcha/Image.php	(working copy)
@@ -580,7 +580,7 @@
         $suffixLength = strlen($this->_suffix);
         foreach (new DirectoryIterator($imgdir) as $file) {
             if (!$file->isDot() && !$file->isDir()) {
-                if ($file->getMTime() < $expire) {
+                if (file_exists($file->getPathname()) && $file->getMTime() < $expire) {
                     // only deletes files ending with $this->_suffix
                     if (substr($file->getFilename(), -($suffixLength)) == $this->_suffix) {
                         unlink($file->getPathname());
