Index: library/ZendX/JQuery/View/Helper/AjaxLink.php
===================================================================
--- library/ZendX/JQuery/View/Helper/AjaxLink.php	(revision 21529)
+++ library/ZendX/JQuery/View/Helper/AjaxLink.php	(working copy)
@@ -254,7 +254,7 @@
         $js = implode($js);
 
         if($inline == true) {
-            $attribs['onClick'] = $js;
+            $attribs['onclick'] = $js;
         } else {
             if(!isset($attribs['id'])) {
                 $clickClass = sprintf("ajaxLink%d", ZendX_JQuery_View_Helper_AjaxLink::$currentLinkCallbackId);
@@ -305,4 +305,4 @@
         }
         return $requestHandler;
     }
-}
\ No newline at end of file
+}
