Index: library/Zend/Controller/Dispatcher/Standard.php
===================================================================
--- library/Zend/Controller/Dispatcher/Standard.php	(Revision 7358)
+++ library/Zend/Controller/Dispatcher/Standard.php	(Arbeitskopie)
@@ -133,6 +133,18 @@
     }
 
     /**
+     * Format action class name
+     *
+     * @param string $moduleName Name of the current module
+     * @param string $className Name of the action class
+     * @return string Formatted class name
+     */
+    public function formatClassName($moduleName, $className)
+    {
+        return $this->formatModuleName($moduleName) . '_' . $className;
+    }
+
+    /**
      * Convert a class name to a filename
      *
      * @param string $class
@@ -277,7 +289,7 @@
     {
         $finalClass  = $className;
         if ($this->_defaultModule != $this->_curModule) {
-            $finalClass = $this->formatModuleName($this->_curModule) . '_' . $className;
+            $finalClass = $this->formatClassName($this->_curModule, $className);
         }
         if (class_exists($finalClass, false)) {
             return $finalClass;
