Index: Redirector.php
===================================================================
--- Redirector.php	(revision 14153)
+++ Redirector.php	(working copy)
@@ -65,6 +65,12 @@
     protected $_useAbsoluteUri = false;
 
     /**
+     * Whether or not to close the session before exiting
+     * @var boolean
+     */
+    protected $_closeSessionOnExit = true;
+
+    /**
      * Retrieve HTTP status code to emit on {@link _redirect()} call
      *
      * @return int
@@ -85,7 +91,7 @@
     {
         $code = (int)$code;
         if ((300 > $code) || (307 < $code) || (304 == $code) || (306 == $code)) {
-            require_once 'Zend/Controller/Exception.php';
+            require_once 'Zend/Controller/Action/Exception.php';
             throw new Zend_Controller_Action_Exception('Invalid redirect HTTP status code (' . $code  . ')');
         }
 
@@ -151,6 +157,29 @@
     }
 
     /**
+     * Retrieve flag for whether or not {@link redirectAndExit()} shall close the session before
+     * exiting.
+     *
+     * @return boolean
+     */
+    public function getCloseSessionOnExit()
+    {
+        return $this->_closeSessionOnExit;
+    }
+
+    /**
+     * Set flag for whether or not {@link redirectAndExit()} shall close the session before exiting.
+     *
+     * @param  boolean $flag
+     * @return Zend_Controller_Action_Helper_Redirector Provides a fluent interface
+     */
+    public function setCloseSessionOnExit($flag)
+    {
+        $this->_closeSessionOnExit = ($flag) ? true : false;
+        return $this;
+    }
+
+    /**
      * Return use absolute URI flag
      *
      * @return boolean
@@ -447,11 +476,13 @@
      */
     public function redirectAndExit()
     {
-        // Close session, if started
-        if (class_exists('Zend_Session', false) && Zend_Session::isStarted()) {
-            Zend_Session::writeClose();
-        } elseif (isset($_SESSION)) {
-            session_write_close();
+        if ($this->getCloseSessionOnExit()) {
+            // Close session, if started
+            if (class_exists('Zend_Session', false) && Zend_Session::isStarted()) {
+                Zend_Session::writeClose();
+            } elseif (isset($_SESSION)) {
+                session_write_close();
+            }
         }
 
         $this->getResponse()->sendHeaders();
