--- ZendFramework/Zend/Http/Client.php  (revision 2545)
+++ ZendFramework/Zend/Http/Client.php  (working copy)
@@ -687,13 +687,14 @@
         do {
             // Clone the URI and add the additional GET parameters to it
             $uri = clone $this->uri;
-            $uri_params = array();
-            parse_str($uri->getQuery(), $uri_params);
-            $uri->setQuery(array_merge($uri_params, $this->paramsGet));
+            if (!empty($this->paramsGet)) {
+                $uri_params = array();
+                parse_str($uri->getQuery(), $uri_params);
+                $uri->setQuery(array_merge($uri_params, $this->paramsGet));
+            }
