--- tests/Zend/Loader/_files/ZfTest/CacheTest2.php	(revision 0)
+++ tests/Zend/Loader/_files/ZfTest/CacheTest2.php	(revision 0)
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Loader
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2010 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id: CacheTest.php 23508 2010-12-15 18:17:16Z andries $
+ */
+
+
+
+
+/**
+ * @category   Zend
+ * @package    Zend_Loader
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2010 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class ZfTest_CacheTest2
+{
+}
Index: tests/Zend/Loader/PluginLoaderTest.php
===================================================================
--- tests/Zend/Loader/PluginLoaderTest.php	(revision 23554)
+++ tests/Zend/Loader/PluginLoaderTest.php	(working copy)
@@ -437,6 +437,22 @@
     }
 
     /**
+     * @group ZF-8429
+     * @depends testPluginLoaderShouldAppendIncludeCacheWhenClassIsFound
+     */
+    public function testPluginLoaderShouldAppendWithConsistentDirectorySeparator()
+    {
+        $cacheFile = $this->_includeCache;
+        Zend_Loader_PluginLoader::setIncludeFileCache($cacheFile);
+        $loader = new Zend_Loader_PluginLoader(array());
+        $loader->addPrefixPath('ZfTest', dirname(__FILE__) . '/_files/ZfTest');
+        $className = $loader->load('CacheTest2');
+        $this->assertTrue(file_exists($cacheFile));
+        $cache = file_get_contents($cacheFile);
+        $this->assertNotContains('\\', $cache);
+    }
+
+    /**
      * @group ZF-5208
      */
     public function testStaticRegistryNamePersistsInDifferentLoaderObjects()
Index: library/Zend/Loader/PluginLoader.php
===================================================================
--- library/Zend/Loader/PluginLoader.php	(revision 23554)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -476,6 +476,10 @@
         } else {
             $file = file_get_contents(self::$_includeFileCache);
         }
+
+        // ZF-8429
+        $incFile = str_replace('\\', '/', $incFile);
+
         if (!strstr($file, $incFile)) {
             $file .= "\ninclude_once '$incFile';";
             file_put_contents(self::$_includeFileCache, $file);
