Index: library/Zend/Validate/GreaterThan.php
===================================================================
--- library/Zend/Validate/GreaterThan.php	(revision 20167)
+++ library/Zend/Validate/GreaterThan.php	(working copy)
@@ -66,6 +66,8 @@
     {
         if ($min instanceof Zend_Config) {
             $min = $min->toArray();
+        }
+        if (is_array($min)) {
             if (array_key_exists('min', $min)) {
                 $min = $min['min'];
             } else {
