--- library/Zend/Tool/Project/Context/Zf/ControllerFile.php	(revision 18211)
+++ library/Zend/Tool/Project/Context/Zf/ControllerFile.php	(working copy)
@@ -26,6 +26,11 @@
 require_once 'Zend/Tool/Project/Context/Filesystem/File.php';
 
 /**
+ * @see Zend_Tool_Project_Context_Filesystem_File
+ */
+require_once 'Zend/Tool/Project/Context/Filesystem/File.php';
+
+/**
  * @see Zend_CodeGenerator_Php_File
  */
 require_once 'Zend/CodeGenerator/Php/File.php';
@@ -113,8 +118,16 @@
     {
 
         $filter = new Zend_Filter_Word_DashToCamelCase();
+        
+        $this->_controllerName = $this->_resource->getAttribute('controllerName');
+        $moduleName = $this->_resource->getAttribute('moduleName');
 
-        $className = $filter->filter($this->_controllerName) . 'Controller';
+        if (! is_null($moduleName)) {
+            $className = ucfirst($moduleName) . '_' . $filter->filter($this->_controllerName)
+                       . 'Controller';
+        } else {
+            $className = $filter->filter($this->_controllerName) . 'Controller';
+        }
 
         $codeGenFile = new Zend_CodeGenerator_Php_File(array(
             'fileName' => $this->getPath(),
Index: library/Zend/Tool/Project/Provider/Controller.php
===================================================================
--- library/Zend/Tool/Project/Provider/Controller.php	(revision 18211)
+++ library/Zend/Tool/Project/Provider/Controller.php	(working copy)
@@ -81,7 +81,9 @@
             throw new Zend_Tool_Project_Provider_Exception($exceptionMessage);
         }
 
-        $newController = $controllersDirectory->createResource('controllerFile', array('controllerName' => $controllerName));
+        $newController = $controllersDirectory->createResource('controllerFile', array(
+									'controllerName' => $controllerName,
+									'moduleName' => $moduleName));
 
         return $newController;
     }
