--- Zend/Cache/Frontend/File.php	(revision 19708)
+++ Zend/Cache/Frontend/File.php	(working copy)
@@ -174,7 +174,7 @@
      * Test if a cache is available for the given id
      *
      * @param  string $id Cache id
-     * @return boolean True is a cache is available, false else
+     * @return int|false Last modified time of cache entry if it is available, false otherwise
      */
     public function test($id)
     {
Index: Zend/Cache/Core.php
===================================================================
--- Zend/Cache/Core.php	(revision 19708)
+++ Zend/Cache/Core.php	(working copy)
@@ -307,7 +307,7 @@
      * Test if a cache is available for the given id
      *
      * @param  string $id Cache id
-     * @return boolean True is a cache is available, false else
+     * @return int|false Last modified time of cache entry if it is available, false otherwise
      */
     public function test($id)
     {
