Index: library/Zend/CodeGenerator/Php/Parameter.php
===================================================================
--- library/Zend/CodeGenerator/Php/Parameter.php	(revision 20197)
+++ library/Zend/CodeGenerator/Php/Parameter.php	(working copy)
@@ -237,7 +237,7 @@
             $output .= ' = ';
             if (is_string($this->_defaultValue)) {
                 $output .= '\'' . $this->_defaultValue . '\'';
-            } else if($this->_defaultValue instanceof Zend_CodeGenerator_Php_ParameterDefaultValue) {
+            } else if($this->_defaultValue instanceof Zend_CodeGenerator_Php_Parameter_DefaultValue) {
                 $output .= (string)$this->_defaultValue;
             } else {
                 $output .= $this->_defaultValue;
@@ -247,4 +247,4 @@
         return $output;
     }
 
-}
\ No newline at end of file
+}
