--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic.response	(revision 0)
@@ -0,0 +1,22 @@
+HTTP/1.1 200 OK
+Date: Wed, 21 Mar 2007 20:00:46 GMT
+Server: Apache/2.0.52
+Content-length: 3023
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<photos page="1" pages="429" perpage="10" total="4285">
+	<photo id="428222530" owner="36046768@N00" secret="6d2e56ed92" server="146" farm="1" title="Our mobile solution" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174407931" datetaken="2007-03-20 12:25:31" datetakengranularity="0" ownername="jrodgers" iconserver="8" iconfarm="1" />
+	<photo id="427883929" owner="87185455@N00" secret="e46c580947" server="163" farm="1" title="moreoptions" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377640" datetaken="2007-03-20 01:00:40" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427884403" owner="87185455@N00" secret="fb2ebbb275" server="183" farm="1" title="right4" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377693" datetaken="2007-03-20 01:01:33" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427887192" owner="87185455@N00" secret="a5a5781ea0" server="156" farm="1" title="left3" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377980" datetaken="2007-03-20 01:06:20" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427883923" owner="87185455@N00" secret="25902ff728" server="146" farm="1" title="left4" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377639" datetaken="2007-03-20 01:00:39" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427884394" owner="87185455@N00" secret="48678fb007" server="145" farm="1" title="rbutton" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377692" datetaken="2007-03-20 01:01:32" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427883930" owner="87185455@N00" secret="37e83d6bde" server="163" farm="1" title="right4" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377640" datetaken="2007-03-20 01:00:40" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427884398" owner="87185455@N00" secret="ab68b726ae" server="154" farm="1" title="right1" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377693" datetaken="2007-03-20 01:01:33" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427883924" owner="87185455@N00" secret="bbfdee1f96" server="162" farm="1" title="right3" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377639" datetaken="2007-03-20 01:00:39" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+	<photo id="427884401" owner="87185455@N00" secret="b561b6636a" server="160" farm="1" title="right3" ispublic="1" isfriend="0" isfamily="0" license="0" dateupload="1174377693" datetaken="2007-03-20 01:01:33" datetakengranularity="0" ownername="wcbzero2006" iconserver="0" iconfarm="0" />
+</photos>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884401.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884401.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884401.response	(revision 0)
@@ -0,0 +1,17 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:49 GMT
+Server: Apache/2.0.52
+Content-length: 1012
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/160/427884401_b561b6636a_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884401&amp;size=sq" />
+	<size label="Thumbnail" width="100" height="3" source="http://farm1.static.flickr.com/160/427884401_b561b6636a_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884401&amp;size=t" />
+	<size label="Small" width="240" height="8" source="http://farm1.static.flickr.com/160/427884401_b561b6636a_m.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884401&amp;size=s" />
+	<size label="Medium" width="500" height="16" source="http://farm1.static.flickr.com/160/427884401_b561b6636a.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884401&amp;size=m" />
+	<size label="Large" width="531" height="17" source="http://farm1.static.flickr.com/160/427884401_6356f5fae7_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427884401&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884403.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884403.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884403.response	(revision 0)
@@ -0,0 +1,15 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:46 GMT
+Server: Apache/2.0.52
+Content-length: 639
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/183/427884403_fb2ebbb275_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884403&amp;size=sq" />
+	<size label="Thumbnail" width="23" height="3" source="http://farm1.static.flickr.com/183/427884403_fb2ebbb275_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884403&amp;size=t" />
+	<size label="Large" width="23" height="3" source="http://farm1.static.flickr.com/183/427884403_bd4c170074_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427884403&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883930.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883930.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883930.response	(revision 0)
@@ -0,0 +1,15 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:48 GMT
+Server: Apache/2.0.52
+Content-length: 639
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/163/427883930_37e83d6bde_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883930&amp;size=sq" />
+	<size label="Thumbnail" width="23" height="3" source="http://farm1.static.flickr.com/163/427883930_37e83d6bde_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883930&amp;size=t" />
+	<size label="Large" width="23" height="3" source="http://farm1.static.flickr.com/163/427883930_edb91cc3a2_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427883930&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883923.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883923.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883923.response	(revision 0)
@@ -0,0 +1,16 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:47 GMT
+Server: Apache/2.0.52
+Content-length: 828
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/146/427883923_25902ff728_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883923&amp;size=sq" />
+	<size label="Thumbnail" width="100" height="8" source="http://farm1.static.flickr.com/146/427883923_25902ff728_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883923&amp;size=t" />
+	<size label="Small" width="221" height="17" source="http://farm1.static.flickr.com/146/427883923_25902ff728_m.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883923&amp;size=s" />
+	<size label="Large" width="221" height="17" source="http://farm1.static.flickr.com/146/427883923_398e28d146_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427883923&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883924.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883924.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883924.response	(revision 0)
@@ -0,0 +1,17 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:48 GMT
+Server: Apache/2.0.52
+Content-length: 1012
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/162/427883924_bbfdee1f96_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883924&amp;size=sq" />
+	<size label="Thumbnail" width="100" height="3" source="http://farm1.static.flickr.com/162/427883924_bbfdee1f96_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883924&amp;size=t" />
+	<size label="Small" width="240" height="8" source="http://farm1.static.flickr.com/162/427883924_bbfdee1f96_m.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883924&amp;size=s" />
+	<size label="Medium" width="500" height="16" source="http://farm1.static.flickr.com/162/427883924_bbfdee1f96.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883924&amp;size=m" />
+	<size label="Large" width="531" height="17" source="http://farm1.static.flickr.com/162/427883924_e27b0e1bf3_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427883924&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427887192.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427887192.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427887192.response	(revision 0)
@@ -0,0 +1,15 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:47 GMT
+Server: Apache/2.0.52
+Content-length: 639
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/156/427887192_a5a5781ea0_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427887192&amp;size=sq" />
+	<size label="Thumbnail" width="11" height="1" source="http://farm1.static.flickr.com/156/427887192_a5a5781ea0_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427887192&amp;size=t" />
+	<size label="Large" width="11" height="1" source="http://farm1.static.flickr.com/156/427887192_ebd45dbf31_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427887192&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884394.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884394.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884394.response	(revision 0)
@@ -0,0 +1,15 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:47 GMT
+Server: Apache/2.0.52
+Content-length: 641
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/145/427884394_48678fb007_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884394&amp;size=sq" />
+	<size label="Thumbnail" width="92" height="37" source="http://farm1.static.flickr.com/145/427884394_48678fb007_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884394&amp;size=t" />
+	<size label="Large" width="92" height="37" source="http://farm1.static.flickr.com/145/427884394_d1e788022b_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427884394&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosExceptionGroupIdInvalid.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosExceptionGroupIdInvalid.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosExceptionGroupIdInvalid.response	(revision 0)
@@ -0,0 +1,11 @@
+HTTP/1.1 200 OK
+Date: Thu, 29 Mar 2007 23:28:10 GMT
+Server: Apache/2.0.52
+Content-length: 105
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="fail">
+	<err code="1" msg="Group not found" />
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883929.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883929.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427883929.response	(revision 0)
@@ -0,0 +1,16 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:46 GMT
+Server: Apache/2.0.52
+Content-length: 829
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/163/427883929_e46c580947_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883929&amp;size=sq" />
+	<size label="Thumbnail" width="100" height="11" source="http://farm1.static.flickr.com/163/427883929_e46c580947_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883929&amp;size=t" />
+	<size label="Small" width="116" height="13" source="http://farm1.static.flickr.com/163/427883929_e46c580947_m.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427883929&amp;size=s" />
+	<size label="Large" width="116" height="13" source="http://farm1.static.flickr.com/163/427883929_f0230e0d5a_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427883929&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884398.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884398.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_427884398.response	(revision 0)
@@ -0,0 +1,17 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:48 GMT
+Server: Apache/2.0.52
+Content-length: 1012
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/154/427884398_ab68b726ae_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884398&amp;size=sq" />
+	<size label="Thumbnail" width="100" height="3" source="http://farm1.static.flickr.com/154/427884398_ab68b726ae_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884398&amp;size=t" />
+	<size label="Small" width="240" height="7" source="http://farm1.static.flickr.com/154/427884398_ab68b726ae_m.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884398&amp;size=s" />
+	<size label="Medium" width="500" height="14" source="http://farm1.static.flickr.com/154/427884398_ab68b726ae.jpg" url="http://www.flickr.com/photo_zoom.gne?id=427884398&amp;size=m" />
+	<size label="Large" width="531" height="15" source="http://farm1.static.flickr.com/154/427884398_c372035a18_o.gif" url="http://www.flickr.com/photo_zoom.gne?id=427884398&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_428222530.response
===================================================================
--- tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_428222530.response	(revision 0)
+++ tests/Zend/Service/Flickr/_files/testGroupPoolGetPhotosBasic-result_428222530.response	(revision 0)
@@ -0,0 +1,17 @@
+HTTP/1.1 200 OK
+Date: Thu, 22 Mar 2007 19:09:45 GMT
+Server: Apache/2.0.52
+Content-length: 1020
+Connection: close
+Content-type: text/xml; charset=utf-8
+
+<?xml version="1.0" encoding="utf-8" ?>
+<rsp stat="ok">
+<sizes>
+	<size label="Square" width="75" height="75" source="http://farm1.static.flickr.com/146/428222530_6d2e56ed92_s.jpg" url="http://www.flickr.com/photo_zoom.gne?id=428222530&amp;size=sq" />
+	<size label="Thumbnail" width="100" height="86" source="http://farm1.static.flickr.com/146/428222530_6d2e56ed92_t.jpg" url="http://www.flickr.com/photo_zoom.gne?id=428222530&amp;size=t" />
+	<size label="Small" width="240" height="208" source="http://farm1.static.flickr.com/146/428222530_6d2e56ed92_m.jpg" url="http://www.flickr.com/photo_zoom.gne?id=428222530&amp;size=s" />
+	<size label="Medium" width="500" height="432" source="http://farm1.static.flickr.com/146/428222530_6d2e56ed92.jpg" url="http://www.flickr.com/photo_zoom.gne?id=428222530&amp;size=m" />
+	<size label="Original" width="769" height="665" source="http://farm1.static.flickr.com/146/428222530_d1bfe0bc57_o.png" url="http://www.flickr.com/photo_zoom.gne?id=428222530&amp;size=o" />
+</sizes>
+</rsp>
Index: tests/Zend/Service/Flickr/OfflineTest.php
===================================================================
--- tests/Zend/Service/Flickr/OfflineTest.php	(revision 7796)
+++ tests/Zend/Service/Flickr/OfflineTest.php	(working copy)
@@ -352,8 +352,175 @@
             $this->assertContains('parameters are invalid', $e->getMessage());
         }
     }
+    
+    /**
+     * Basic testing to ensure that groupPoolGetPhotos() works as expected
+     *
+     * @return void
+     */
+    public function testGroupPoolGetPhotosBasic()
+    {
+        $this->_flickr->getRestClient()
+                      ->getHttpClient()
+                      ->setAdapter($this->_httpClientAdapterTest);
 
+        $this->_httpClientAdapterTest->setResponse($this->_loadResponse(__FUNCTION__));
+
+        $options = array(
+            'per_page' => 10,
+            'page'     => 1,
+            'extras'   => 'license, date_upload, date_taken, owner_name, icon_server'
+            );
+
+        $resultSet = $this->_flickr->groupPoolGetPhotos('20083316@N00', $options);
+
+        $this->assertEquals(4285, $resultSet->totalResultsAvailable);
+        $this->assertEquals(10, $resultSet->totalResults());
+        $this->assertEquals(10, $resultSet->totalResultsReturned);
+        $this->assertEquals(1, $resultSet->firstResultPosition);
+
+        $this->assertEquals(0, $resultSet->key());
+
+        try {
+            $resultSet->seek(-1);
+            $this->fail('Expected OutOfBoundsException not thrown');
+        } catch (OutOfBoundsException $e) {
+            $this->assertContains('Illegal index', $e->getMessage());
+        }
+
+        $resultSet->seek(9);
+
+        try {
+            $resultSet->seek(10);
+            $this->fail('Expected OutOfBoundsException not thrown');
+        } catch (OutOfBoundsException $e) {
+            $this->assertContains('Illegal index', $e->getMessage());
+        }
+
+        $resultSet->rewind();
+
+        $resultSetIds = array(
+            '428222530',
+            '427883929',
+            '427884403',
+            '427887192',
+            '427883923',
+            '427884394',
+            '427883930',
+            '427884398',
+            '427883924',
+            '427884401'
+            );
+
+        $this->assertTrue($resultSet->valid());
+
+        foreach ($resultSetIds as $resultSetId) {
+            $this->_httpClientAdapterTest->setResponse($this->_loadResponse(__FUNCTION__ . "-result_$resultSetId"));
+            $result = $resultSet->current();
+            $this->assertTrue($result instanceof Zend_Service_Flickr_Result);
+            $resultSet->next();
+        }
+
+        $this->assertFalse($resultSet->valid());
+    }
+
     /**
+     * Ensures that groupPoolGetPhotos() throws an exception when an option is invalid
+     *
+     * @return void
+     */
+    public function testGroupPoolGetPhotosExceptionOptionInvalid()
+    {
+      try {
+          $this->_flickr->groupPoolGetPhotos('irrelevant', array('unexpected' => null));
+          $this->fail('Expected Zend_Service_Exception not thrown');
+      } catch (Zend_Service_Exception $e) {
+          $this->assertContains('parameters are invalid', $e->getMessage());
+      }
+    }
+
+    /**
+     * Ensures that _validateGroupPoolGetPhotos() throws an exception when the per_page option is invalid
+     *
+     * @return void
+     */
+    public function testValidateGroupPoolGetPhotosExceptionPerPageInvalid()
+    {
+      try {
+          $this->_flickrProxy->proxyValidateGroupPoolGetPhotos(array('per_page' => -1));
+          $this->fail('Expected Zend_Service_Exception not thrown');
+      } catch (Zend_Service_Exception $e) {
+          $this->assertContains('"per_page" option', $e->getMessage());
+      }
+    }
+
+    /**
+     * Ensures that _validateGroupPoolGetPhotos() throws an exception when the page option is invalid
+     *
+     * @return void
+     */
+    public function testValidateGroupPoolGetPhotosExceptionPageInvalid()
+    {
+      try {
+          $this->_flickrProxy->proxyValidateGroupPoolGetPhotos(array('per_page' => 10, 'page' => 1.23));
+          $this->fail('Expected Zend_Service_Exception not thrown');
+      } catch (Zend_Service_Exception $e) {
+          $this->assertContains('"page" option', $e->getMessage());
+      }
+    }
+
+    /**
+     * Ensures that groupPoolGetPhotos() throws an exception when an invalid group_id is given
+     *
+     * @return void
+     */
+    public function testGroupPoolGetPhotosExceptionGroupIdInvalid()
+    {
+        $this->_flickr->getRestClient()
+                      ->getHttpClient()
+                      ->setAdapter($this->_httpClientAdapterTest);
+
+        $this->_httpClientAdapterTest->setResponse($this->_loadResponse(__FUNCTION__));
+
+        try {
+            $this->_flickr->groupPoolGetPhotos('2e38a9d9425d7e2c9d0788455e9ccc61');
+            $this->fail('Expected Zend_Service_Exception not thrown');
+        } catch (Zend_Service_Exception $e) {
+            $this->assertContains('Group not found', $e->getMessage());
+        }
+    }
+
+    /**
+     * Ensures that groupPoolGetPhotos() throws an exception when an invalid group_id is given
+     *
+     * @return void
+     */
+    public function testGroupPoolGetPhotosExceptionGroupIdEmpty()
+    {
+        try {
+            $this->_flickr->groupPoolGetPhotos('0');
+            $this->fail('Expected Zend_Service_Exception not thrown');
+        } catch (Zend_Service_Exception $e) {
+            $this->assertContains('supply a group id', $e->getMessage());
+        }
+    }
+     
+    /**
+     * Ensures that groupPoolGetPhotos() throws an exception when an array is given for group_id
+     *
+     * @return void
+     */
+    public function testGroupPoolGetPhotosExceptionGroupIdArray()
+    {
+        try {
+            $this->_flickr->groupPoolGetPhotos(array());
+            $this->fail('Expected Zend_Service_Exception not thrown');
+        } catch (Zend_Service_Exception $e) {
+            $this->assertContains('supply a group id', $e->getMessage());
+        }
+    }
+
+    /**
      * Utility method that saves an HTTP response to a file
      *
      * @param  string $name
@@ -389,6 +556,11 @@
     {
         $this->_validateTagSearch($options);
     }
+    
+    public function proxyValidateGroupPoolGetPhotos(array $options)
+    {
+        $this->_validateGroupPoolGetPhotos($options);
+    }
 
     public function proxyCompareOptions(array $options, array $validOptions)
     {
Index: tests/Zend/Service/Flickr/OnlineTest.php
===================================================================
--- tests/Zend/Service/Flickr/OnlineTest.php	(revision 7796)
+++ tests/Zend/Service/Flickr/OnlineTest.php	(working copy)
@@ -74,7 +74,54 @@
                       ->getHttpClient()
                       ->setAdapter($this->_httpClientAdapterSocket);
     }
+    
+    /**
+     * Basic testing to ensure that groupPoolGetPhotos works as expected
+     *
+     * @return void
+     */
+    public function testGroupPoolGetPhotosBasic()
+    {
+        $options = array('per_page' => 10,
+                         'page'     => 1,
+                         'extras'   => 'license, date_upload, date_taken, owner_name, icon_server');
+                         
+        $resultSet = $this->_flickr->groupPoolGetPhotos('20083316@N00', $options);
+        
+        $this->assertEquals(21770, $resultSet->totalResultsAvailable);
+        $this->assertEquals(10, $resultSet->totalResults());
+        $this->assertEquals(10, $resultSet->totalResultsReturned);
+        $this->assertEquals(1, $resultSet->firstResultPosition);
 
+        $this->assertEquals(0, $resultSet->key());
+        
+        try {
+            $resultSet->seek(-1);
+            $this->fail('Expected OutOfBoundsException not thrown');
+        } catch (OutOfBoundsException $e) {
+            $this->assertContains('Illegal index', $e->getMessage());
+        }
+
+        $resultSet->seek(9);
+
+        try {
+            $resultSet->seek(10);
+            $this->fail('Expected OutOfBoundsException not thrown');
+        } catch (OutOfBoundsException $e) {
+            $this->assertContains('Illegal index', $e->getMessage());
+        }
+
+        $resultSet->rewind();
+
+        $count = 0;
+        foreach ($resultSet as $result) {
+            $this->assertTrue($result instanceof Zend_Service_Flickr_Result);
+            $count++;
+        }
+
+        $this->assertEquals(10, $count);
+    }
+
     /**
      * Basic testing to ensure that userSearch() works as expected
      *
Index: library/Zend/Service/Flickr.php
===================================================================
--- library/Zend/Service/Flickr.php	(revision 7796)
+++ library/Zend/Service/Flickr.php	(working copy)
@@ -192,8 +192,64 @@
         require_once 'Zend/Service/Flickr/ResultSet.php';
         return new Zend_Service_Flickr_ResultSet($dom, $this);
     }
+    
+    /**
+     * Finds photos in a group's pool.
+     *
+     * @param  string $query   group id
+     * @param  array  $options Additional parameters to refine your query.
+     * @return Zend_Service_Flickr_ResultSet
+     * @throws Zend_Service_Exception
+     */
+    public function groupPoolGetPhotos($query, array $options = array())
+    {
+        static $method = 'flickr.groups.pools.getPhotos';
+        static $defaultOptions = array('per_page' => 10,
+                                       'page'     => 1,
+                                       'extras'   => 'license, date_upload, date_taken, owner_name, icon_server');
 
+        if (empty($query) || !is_string($query)) {
+            /**
+             * @see Zend_Service_Exception
+             */
+            require_once 'Zend/Service/Exception.php';
+            throw new Zend_Service_Exception('You must supply a group id');
+        }
 
+        $options['group_id'] = $query;
+
+        $options = $this->_prepareOptions($method, $options, $defaultOptions);
+
+        $this->_validateGroupPoolGetPhotos($options);
+
+        // now search for photos
+        $restClient = $this->getRestClient();
+        $restClient->getHttpClient()->resetParameters();
+        $response = $restClient->restGet('/services/rest/', $options);
+
+        if ($response->isError()) {
+            /**
+            * @see Zend_Service_Exception
+            */
+            require_once 'Zend/Service/Exception.php';
+            throw new Zend_Service_Exception('An error occurred sending request. Status code: '
+                                           . $response->getStatus());
+        }
+
+        $dom = new DOMDocument();
+        $dom->loadXML($response->getBody());
+
+        self::_checkErrors($dom);
+
+        /**
+        * @see Zend_Service_Flickr_ResultSet
+        */
+        require_once 'Zend/Service/Flickr/ResultSet.php';
+        return new Zend_Service_Flickr_ResultSet($dom, $this);
+    }
+
+
+
     /**
      * Utility function to find Flickr User IDs for usernames.
      *
@@ -453,8 +509,63 @@
         }
 
     }
+    
+    
+    /**
+    * Validate Group Search Options
+    *
+    * @param  array $options
+    * @throws Zend_Service_Exception
+    * @return void
+    */
+    protected function _validateGroupPoolGetPhotos(array $options)
+    {
+        $validOptions = array('api_key', 'tags', 'method', 'group_id', 'per_page', 'page', 'extras', 'min_upload_date',
+                              'min_taken_date', 'max_upload_date', 'max_taken_date');
 
+        $this->_compareOptions($options, $validOptions);
 
+        /**
+        * @see Zend_Validate_Between
+        */
+        require_once 'Zend/Validate/Between.php';
+        $between = new Zend_Validate_Between(1, 500, true);
+        if (!$between->isValid($options['per_page'])) {
+            /**
+            * @see Zend_Service_Exception
+            */
+            require_once 'Zend/Service/Exception.php';
+            throw new Zend_Service_Exception($options['per_page'] . ' is not valid for the "per_page" option');
+        }
+
+        /**
+        * @see Zend_Validate_Int
+        */
+        require_once 'Zend/Validate/Int.php';
+        $int = new Zend_Validate_Int();
+        
+        if (!$int->isValid($options['page'])) {
+            /**
+            * @see Zend_Service_Exception
+            */
+            require_once 'Zend/Service/Exception.php';
+            throw new Zend_Service_Exception($options['page'] . ' is not valid for the "page" option');
+        }
+
+        // validate extras, which are delivered in csv format
+        if (isset($options['extras'])) {
+            $extras = explode(',', $options['extras']);
+            $validExtras = array('license', 'date_upload', 'date_taken', 'owner_name', 'icon_server');
+            foreach($extras as $extra) {
+                /**
+                * @todo The following does not do anything [yet], so it is commented out.
+                */
+                //in_array(trim($extra), $validExtras);
+            }
+        }
+    }
+
+
     /**
      * Throws an exception if and only if the response status indicates a failure
      *
