Index: library/Zend/Tool/Framework/Client/Console.php
===================================================================
--- library/Zend/Tool/Framework/Client/Console.php	(revision 16050)
+++ library/Zend/Tool/Framework/Client/Console.php	(working copy)
@@ -116,6 +116,12 @@
             }
         }
 
+        // otherwise register the default autoloader: Zend_Loader_Autoloader
+        else {
+            require_once 'Zend/Loader/Autoloader.php';
+            $this->_registry->setLoader(Zend_Loader_Autoloader::getInstance());
+        } // END if-else
+
         return;
     }
 
