Index: library/Zend/View/Helper/FormSelect.php
===================================================================
--- library/Zend/View/Helper/FormSelect.php	(revision 24786)
+++ library/Zend/View/Helper/FormSelect.php	(working copy)
@@ -160,8 +160,7 @@
         }
 
         $opt = '<option'
-             . ' value="' . $this->view->escape($value) . '"'
-             . ' label="' . $this->view->escape($label) . '"';
+             . ' value="' . $this->view->escape($value) . '"';
 
         // selected?
         if (in_array((string) $value, $selected)) {
