Index: HtmlListTest.php
===================================================================
--- HtmlListTest.php	(revision 19365)
+++ HtmlListTest.php	(working copy)
@@ -228,6 +228,19 @@
     }
 
     /**
+     * @see ZF-8400
+     * Tests what happens if an empty array is given @see ZF-8400
+     */
+    public function testItemsPassedAsEmptyArray()
+    {
+        $items = array();
+
+        $list = $this->helper->htmlList($items);
+		
+        $this->assertContains('', $list);
+    }
+
+    /**
      * @see ZF-2870
      */
     public function testEscapeFlagShouldBePassedRecursively()
