Index: tests/Zend/Db/Table/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/TestCommon.php  (revision 24462)
+++ tests/Zend/Db/Table/TestCommon.php  (working copy)
@@ -1652,9 +1652,44 @@
         $this->assertEquals(1, count($rowset));
     }

+    /**
+     * @group ZF-3144
+     */
+    public function testCascadingOperationReferenceColumnFallsBackToPrimaryKeyWhenNotSpecified()
+    {
+        Zend_Db_Table::setDefaultAdapter($this->_db);
+
+        $this->_util->createTable('ZF3144_products', array(
+            'id'          => 'tinyint(4) NOT NULL',
+            'PRIMARY KEY' => 'id'
+        ));
+
+        $this->_util->createTable('ZF3144_files', array(
+            'id'          => 'tinyint(4) NOT NULL',
+            'productId'   => 'tinyint(4) NOT NULL',
+            'PRIMARY KEY' => 'id'
+        ));
+
+        require_once dirname(__FILE__) . "/_files/My/ZF3144/Products.php";
+        $tblProducts = new My_ZF3144_Products();
+        $tblProducts->createRow(array('id'=>1))->save();
+        $tblProducts->createRow(array('id'=>2))->save();

+        require_once dirname(__FILE__) . "/_files/My/ZF3144/Files.php";
+        $tblFiles = new My_ZF3144_Files();
+        $tblFiles->createRow(array('id'=>7, 'productId'=>2))->save();
+        $tblFiles->createRow(array('id'=>8, 'productId'=>2))->save();
+
+        $tblProducts->find(2)->current()->delete();
+
+        $this->_util->dropTable('ZF3144_products');
+        $this->_util->dropTable('ZF3144_files');
+
+        Zend_Db_Table::setDefaultAdapter();
+    }


+
     /**
      * Returns a clean Zend_Cache_Core with File backend
      *
Index: tests/Zend/Db/Table/_files/My/ZF3144/Files.php
===================================================================
--- tests/Zend/Db/Table/_files/My/ZF3144/Files.php      (revision 0)
+++ tests/Zend/Db/Table/_files/My/ZF3144/Files.php      (revision 0)
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Db
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see Zend_Db_Table_Abstract
+ */
+require_once 'Zend/Db/Table/Abstract.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_Db
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class My_ZF3144_Files extends Zend_Db_Table_Abstract
+{
+    protected $_name = 'ZF3144_files';
+    protected $_primary = 'id';
+    protected $_referenceMap = array(
+        'Product' => array(
+            'columns'           => 'productId',
+            'refTableClass'     => 'My_ZF3144_Products',
+           // 'refColumns purposely omitted
+            'onDelete'          => self::CASCADE,
+        ),
+    );
+}

Property changes on: tests/Zend/Db/Table/_files/My/ZF3144/Files.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Db/Table/_files/My/ZF3144/Products.php
===================================================================
--- tests/Zend/Db/Table/_files/My/ZF3144/Products.php   (revision 0)
+++ tests/Zend/Db/Table/_files/My/ZF3144/Products.php   (revision 0)
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Db
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see Zend_Db_Table_Abstract
+ */
+require_once 'Zend/Db/Table/Abstract.php';
+
+/**
+ * require other test files needed, this will
+ * ensure that Zend_Loader::loadClass is not called
+ */
+require_once dirname(__FILE__) . '/Files.php';
+
+
+
+
+
+/**
+ * @category   Zend
+ * @package    Zend_Db
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class My_ZF3144_Products extends Zend_Db_Table_Abstract
+{
+    protected $_name = 'ZF3144_products';
+    protected $_primary = 'id';
+    protected $_dependentTables = array('MY_ZF3144_Files');
+
+}

Property changes on: tests/Zend/Db/Table/_files/My/ZF3144/Products.php
___________________________________________________________________
Added: svn:keywords
   + Id
