From 8201803ecc50284a857b3a951bee29c5335223af Mon Sep 17 00:00:00 2001
From: Thinkscape <abodera@gmail.com>
Date: Fri, 2 Sep 2011 14:11:36 +0200
Subject: [PATCH] ZF2-58 Classmap generator broken on Win32

---
 bin/classmap_generator.php |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/classmap_generator.php b/bin/classmap_generator.php
index 91a8d19..3b9a30d 100644
--- a/bin/classmap_generator.php
+++ b/bin/classmap_generator.php
@@ -81,7 +81,7 @@ if (array_key_exists('PWD', $_SERVER)) {
 $libraryPath = '';
 if (isset($opts->l)) {
     $libraryPath = $opts->l;
-    $libraryPath = rtrim($libraryPath, '/\\') . '/';
+    $libraryPath = rtrim($libraryPath, '/\\') . DIRECTORY_SEPARATOR;
     if (!is_dir($libraryPath)) {
         echo "Invalid library directory provided" . PHP_EOL . PHP_EOL;
         echo $opts->getUsageMessage();
@@ -134,7 +134,7 @@ $strip .= DIRECTORY_SEPARATOR;
 iterator_apply($l, function() use ($l, $map, $strip, $libraryPath){
     $file      = $l->current();
     $namespace = empty($file->namespace) ? '' : $file->namespace . '\\';
-    $filename  = str_replace($strip, '', $file->getPath() . '/' . $file->getFilename());
+    $filename  = str_replace($strip, '', $file->getPath() . DIRECTORY_SEPARATOR . $file->getFilename());
 
     // Add in relative path to library
     $filename  = $libraryPath . $filename;
-- 
1.7.6.msysgit.0

