Index: library/Zend/Http/UserAgent/AbstractDevice.php
===================================================================
--- library/Zend/Http/UserAgent/AbstractDevice.php	(revision 23692)
+++ library/Zend/Http/UserAgent/AbstractDevice.php	(working copy)
@@ -648,7 +648,9 @@
                     if (isset($comment[4])) {
                         $result['browser_build'] = trim($comment[4]);
                     }
-                    $result['browser_language'] = trim($comment[3]);
+					if (isset($comment[3])) {
+						$result['browser_language'] = trim($comment[3]);
+					}
 
                     // Netscape
                     if ($result['browser_name'] == 'Navigator' || $result['browser_name'] == 'Netscape6') {
