Index: standard/trunk/library/Zend/Controller/Request/Http.php
===================================================================
--- standard/trunk/library/Zend/Controller/Request/Http.php	(revision 14630)
+++ standard/trunk/library/Zend/Controller/Request/Http.php	(working copy)
@@ -387,11 +387,10 @@
                 $requestUri = $_SERVER['HTTP_X_REWRITE_URL'];
             } elseif (isset($_SERVER['REQUEST_URI'])) {
                 $requestUri = $_SERVER['REQUEST_URI'];
-                if (isset($_SERVER['HTTP_HOST']) && strstr($requestUri, $_SERVER['HTTP_HOST'])) {
-                    $pathInfo    = parse_url($requestUri, PHP_URL_PATH);
-                    $queryString = parse_url($requestUri, PHP_URL_QUERY);
-                    $requestUri  = $pathInfo
-                                 . ((empty($queryString)) ? '' : '?' . $queryString);
+                // Http proxy reqs setup request uri with scheme and host [and port] + the url path, only use url path
+                $schemeAndHttpHost = $this->getScheme() . $this->getHttpHost();
+                if (strpos($requestUri, $schemeAndHttpHost) === 0) {
+                    $requestUri = substr($requestUri, strlen($schemeAndHttpHost));
                 }
             } elseif (isset($_SERVER['ORIG_PATH_INFO'])) { // IIS 5.0, PHP as CGI
                 $requestUri = $_SERVER['ORIG_PATH_INFO'];
