Index: Action.php
--- Action.php Base (BASE)
+++ Action.php Locally Modified (Based On LOCAL)
@@ -574,11 +574,15 @@
      *
      * @param string $paramName
      * @param mixed $default
+     * @param Zend_Validate_Interface $validator
      * @return mixed
      */
-    protected function _getParam($paramName, $default = null)
+    protected function _getParam($paramName, $default = null, Zend_Validate_Interface $validator = null)
     {
         $value = $this->getRequest()->getParam($paramName);
+        if (!(null === $validator) && !($validator->isValid($value))) {
+            $value = $default;
+        }
         if ((null == $value) && (null !== $default)) {
             $value = $default;
         }
