--- C:/Documents and Settings/n.joseph/Bureau/Statement-r8064.php	mar. mars  2 09:11:52 2010
+++ C:/Documents and Settings/n.joseph/Bureau/Statement.php	mar. mars  2 09:15:39 2010
@@ -445,4 +445,15 @@
         return true;
     }
 
+    /**
+     * Abstract Methods
+     */
+
+    /**
+     * Executes a prepared statement.
+     *
+     * @param array $params OPTIONAL Values to bind to parameter placeholders.
+     * @return bool
+     */
+    abstract protected function _execute(array $params = null);
 }
