--- Zend/Controller/Action/Helper/AllTests.php	(revision 23761)
+++ Zend/Controller/Action/Helper/AllTests.php	(working copy)
@@ -24,16 +24,16 @@
     define('PHPUnit_MAIN_METHOD', 'Zend_Controller_Action_Helper_AllTests::main');
 }
 
-require_once 'Zend/Controller/Action/Helper/ActionStackTest.php';
-require_once 'Zend/Controller/Action/Helper/AjaxContextTest.php';
-require_once 'Zend/Controller/Action/Helper/AutoCompleteTest.php';
-require_once 'Zend/Controller/Action/Helper/CacheTest.php';
-require_once 'Zend/Controller/Action/Helper/ContextSwitchTest.php';
-require_once 'Zend/Controller/Action/Helper/FlashMessengerTest.php';
-require_once 'Zend/Controller/Action/Helper/JsonTest.php';
-require_once 'Zend/Controller/Action/Helper/RedirectorTest.php';
-require_once 'Zend/Controller/Action/Helper/UrlTest.php';
-require_once 'Zend/Controller/Action/Helper/ViewRendererTest.php';
+require_once 'ActionStackTest.php';
+require_once 'AjaxContextTest.php';
+require_once 'AutoCompleteTest.php';
+require_once 'CacheTest.php';
+require_once 'ContextSwitchTest.php';
+require_once 'FlashMessengerTest.php';
+require_once 'JsonTest.php';
+require_once 'RedirectorTest.php';
+require_once 'UrlTest.php';
+require_once 'ViewRendererTest.php';
 
 /**
  * @category   Zend
@@ -68,7 +68,7 @@
         $suite->addTestSuite('Zend_Controller_Action_Helper_ViewRendererTest');
 
         if (version_compare(PHP_VERSION, '5.3', '>=')) {
-            require_once 'Zend/Controller/Action/Helper/NamespaceTest.php';
+            require_once 'NamespaceTest.php';
             $suite->addTestSuite('Zend_Controller_Action_Helper_NamespaceTest');
         }
 
