Index: tests/Zend/Filter/InputTest.php
===================================================================
--- tests/Zend/Filter/InputTest.php	(Revision 6880)
+++ tests/Zend/Filter/InputTest.php	(Arbeitskopie)
@@ -397,7 +397,7 @@
             )
         );
         $options = array(
-            Zend_Filter_Input::NAMESPACE => 'TestNamespace'
+            Zend_Filter_Input::INPUT_NAMESPACE => 'TestNamespace'
         );
 
         $ip = get_include_path();
@@ -1018,7 +1018,7 @@
             'field3' => 'digits'
         );
         $options = array(
-            Zend_Filter_Input::NAMESPACE => 'TestNamespace'
+            Zend_Filter_Input::INPUT_NAMESPACE => 'TestNamespace'
         );
 
         $ip = get_include_path();
@@ -1202,7 +1202,7 @@
             'field3' => 'digits'
         );
         $options = array(
-            Zend_Filter_Input::NAMESPACE => 'TestNamespace'
+            Zend_Filter_Input::INPUT_NAMESPACE => 'TestNamespace'
         );
 
         $ip = get_include_path();
Index: library/Zend/Filter/Input.php
===================================================================
--- library/Zend/Filter/Input.php	(Revision 7049)
+++ library/Zend/Filter/Input.php	(Arbeitskopie)
@@ -52,7 +52,7 @@
     const FIELDS            = 'fields';
     const FILTER_CHAIN      = 'filterChain';
     const MISSING_MESSAGE   = 'missingMessage';
-    const NAMESPACE         = 'namespace';
+    const INPUT_NAMESPACE   = 'namespace';
     const NOT_EMPTY_MESSAGE = 'notEmptyMessage';
     const PRESENCE          = 'presence';
     const PRESENCE_OPTIONAL = 'optional';
@@ -408,7 +408,7 @@
                 case self::ESCAPE_FILTER:
                     $this->setDefaultEscapeFilter($value);
                     break;
-                case self::NAMESPACE:
+                case self::INPUT_NAMESPACE:
                     $this->addNamespace($value);
                     break;
                 case self::ALLOW_EMPTY:
