--- Zend/Controller/Action.php	(revision 19658)
+++ Zend/Controller/Action.php	(working copy)
@@ -580,7 +580,7 @@
     protected function _getParam($paramName, $default = null)
     {
         $value = $this->getRequest()->getParam($paramName);
-        if ((null == $value) && (null !== $default)) {
+        if ((null === $value) && (null !== $default)) {
             $value = $default;
         }
 
