diff --git a/Zend/Application/Resource/View.php b/Zend/Application/Resource/View.php
index 2550a94..a039d70 100644
--- a/Zend/Application/Resource/View.php
+++ b/Zend/Application/Resource/View.php
@@ -52,9 +52,7 @@ class Zend_Application_Resource_View extends Zend_Application_Resource_ResourceA
     {
         $view = $this->getView();
 
-        $viewRenderer = new Zend_Controller_Action_Helper_ViewRenderer();
-        $viewRenderer->setView($view);
-        Zend_Controller_Action_HelperBroker::addHelper($viewRenderer);
+        Zend_Controller_Action_HelperBroker::getStaticHelper('viewRenderer')->setView($view);
         return $view;
     }
 
