Index: XmlRpc/Value.php
===================================================================
--- XmlRpc/Value.php	(revision 8537)
+++ XmlRpc/Value.php	(working copy)
@@ -247,6 +247,9 @@
     {
         switch (gettype($value)) {
             case 'object':
+                if ($value instanceof Zend_XmlRpc_Value) { // If value is instance of Zend_XmlRpc_Value we just return this object
+                    return $value;
+                }
                 // We convert the object into a struct
                 $value = get_object_vars($value);
                 // Break intentionally omitted
