Index: library/Zend/Service/Twitter.php
===================================================================
--- library/Zend/Service/Twitter.php	(revision 13984)
+++ library/Zend/Service/Twitter.php	(working copy)
@@ -350,7 +350,7 @@
     {
         $this->_init();
         $path = '/statuses/update.xml';
-        $len  = strlen($status);
+        $len  = iconv_strlen($status, 'UTF-8');
         if ($len > 140) {
             include_once 'Zend/Service/Twitter/Exception.php';
             throw new Zend_Service_Twitter_Exception('Status must be no more than 140 characters in length');
@@ -585,7 +585,7 @@
         $this->_init();
         $path = '/direct_messages/new.xml';
 
-        $len = strlen($text);
+        $len = iconv_strlen($text, 'UTF-8');
         if (0 == $len) {
             throw new Zend_Service_Twitter_Exception('Direct message must contain at least one character');
         } elseif (140 < $len) {
Index: tests/Zend/Service/TwitterTest.php
===================================================================
--- tests/Zend/Service/TwitterTest.php	(revision 13984)
+++ tests/Zend/Service/TwitterTest.php	(working copy)
@@ -431,6 +431,15 @@
         } catch (Exception $e) {
         }
     }
+	
+	public function testPostStatusUpdateUTF8ShouldNotThrowException()
+	{
+		try {
+			$response = $this->twitter->status->update( str_repeat('Mär', 46) . 'Mä' );
+		} catch (Exception $e) {
+			$this->fail('Trying to post a utf8 string of 140 chars should not throw exception');
+		}
+	}
 
     /**
      * $return void
