Index: Zend/Captcha/Word.php
===================================================================
--- Zend/Captcha/Word.php	(revision 11802)
+++ Zend/Captcha/Word.php	(working copy)
@@ -336,7 +336,7 @@
     	}
         
     	$this->_id = $context[$name]['id'];
-    	if ($value != $this->getWord()) {
+    	if ($value !== $this->getWord()) {
             $this->_error(self::BAD_CAPTCHA);
             return false;
         }
