--- Zend/Db/Adapter/Oracle.php.old	2006-07-11 09:22:55.753835000 +0200
+++ Zend/Db/Adapter/Oracle.php	2006-07-04 09:52:34.000000000 +0200
@@ -174,13 +174,16 @@
      */
     public function describeTable($table)
     {
-        $sql = "DESCRIBE $table";
-        $result = $this->fetchAll($sql);
-        foreach ($result as $key => $val) {
-            $descr[$val['Field']] = array(
-                'name'    => $val['Field'],
-                'notnull' => (bool) ($val['Null'] === ''), // not null is empty, null is yes
-                'type'    => $val['Type'],
+        $table = strtoupper($table);
+        $sql = "SELECT COLUMN_NAME, DATA_TYPE, DATA_LENGTH, NULLABLE, DATA_DEFAULT FROM USER_TAB_COLUMNS WHERE TABLE_NAME= '$table' ORDER BY COLUMN_NAME";
+        $result = $this->query($sql);
+        while ($val = $result->fetch() ) {
+			$descr[$val['COLUMN_NAME']] = array(
+                'name'    => $val['COLUMN_NAME'],
+                'notnull' => (bool) ($val['NULLABLE'] === 'N'), // NULLABLE is N when mandatory
+                'type'    => $val['DATA_TYPE'],
+                'default'    => $val['DATA_DEFAULT'],
+                'length'    => $val['DATA_LENGTH'],
             );
         }
         return $descr;
