Index: _files/modules/foo/controllers/FooController.php
===================================================================
--- _files/modules/foo/controllers/FooController.php    (revision 10181)
+++ _files/modules/foo/controllers/FooController.php    (working copy)
@@ -18,4 +18,9 @@
     {
         $this->render();
     }
+
+    public function doubleNestAction()
+    {
+        $this->render();
+    }
 }
Index: _files/modules/foo/views/scripts/foo/double-nest.phtml
===================================================================
--- _files/modules/foo/views/scripts/foo/double-nest.phtml      (revision 0)
+++ _files/modules/foo/views/scripts/foo/double-nest.phtml      (revision 0)
@@ -0,0 +1,3 @@
+<?php
+echo "Foo_DoubleNest";
+echo $this->action('nest', 'foo', 'foo');
\ No newline at end of file
Index: _files/modules/foo/views/scripts/foo/nested.phtml
===================================================================
--- _files/modules/foo/views/scripts/foo/nested.phtml   (revision 10181)
+++ _files/modules/foo/views/scripts/foo/nested.phtml   (working copy)
@@ -1,3 +1,4 @@
 <?
 $this->headTitle()->append('Nested Stuff');
 
+echo "Foo_Nested";
Index: _files/modules/foo/views/scripts/foo/nest.phtml
===================================================================
--- _files/modules/foo/views/scripts/foo/nest.phtml     (revision 10181)
+++ _files/modules/foo/views/scripts/foo/nest.phtml     (working copy)
@@ -1,4 +1,5 @@
 <?
 $this->headTitle('Foo Nest')
      ->headTitle()->setSeparator(' - ');
-$this->action('nested', 'foo', 'foo');
+echo "Foo_Nest";
+echo $this->action('nested', 'foo', 'foo');
Index: ActionTest.php
===================================================================
--- ActionTest.php      (revision 10181)
+++ ActionTest.php      (working copy)
@@ -286,8 +286,16 @@
         $this->assertSame($viewRenderer, $viewRendererPostAction);
     }
     
-    
-    
+    /**
+     * Test response body of double+ nested content is not prepended
+     *
+     * @issue ZF-3456
+     */
+    public function testDoubleNestingReturnsCorrectBodyAction()
+    {
+        $html = $this->helper->action('double-nest', 'foo', 'foo');
+        $this->assertEquals('Foo_DoubleNestFoo_NestFoo_Nested', $html);
+    }
 }
 
 // Call Zend_View_Helper_ActionTest::main() if this source file is executed directly.
 
