Index: tests/Zend/Rest/RouteTest.php
===================================================================
--- tests/Zend/Rest/RouteTest.php	(revision 23883)
+++ tests/Zend/Rest/RouteTest.php	(working copy)
@@ -933,7 +933,21 @@
         $url = $route->assemble($params, false, false);
         $this->assertEquals('users/index/1/extra/parameter/another/parameter', $url);
     }
+    /**
+     * @group ZF-9115
+     */
+    public function test_request_get_user_params()
+    {
+        $uri = Zend_Uri::factory('http://localhost.com/user/index?a=1&b=2');
+        $request = new Zend_Controller_Request_Http($uri);
+        $request->setParam('test', 5);
+        $config = array('mod'=>array('user'));
+        $this->_invokeRouteMatch($request, $config);
+        $this->assertEquals(array("test"=>5), $request->getUserParams());
+        $this->assertEquals(array("test"=>5,"a"=>1,"b"=>2), $request->getParams());
+    }
 
+
     private function _buildRequest($method, $uri)
     {
         $request = new Zend_Controller_Request_HttpTestCase();

