Index: library/Zend/Validate/Int.php
===================================================================
--- library/Zend/Validate/Int.php	(revision 17663)
+++ library/Zend/Validate/Int.php	(working copy)
@@ -110,8 +110,7 @@
 
         } else {
             try {
-                if (!Zend_Locale_Format::isInteger($value, array('locale' => 'en')) &&
-                    !Zend_Locale_Format::isInteger($value, array('locale' => $this->_locale))) {
+                if (!Zend_Locale_Format::isInteger($value, array('locale' => $this->_locale))) {
                     $this->_error(self::NOT_INT);
                     return false;
                 }
