Index: Input.php
===================================================================
--- Input.php   (revision 1890)
+++ Input.php   (working copy)
@@ -31,7 +31,7 @@
  * @copyright  Copyright (c) 2006 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Filter_Input
+class Zend_Filter_Input implements ArrayAccess, IteratorAggregate, Countable
 {
     protected $_source = NULL;

@@ -44,6 +44,33 @@
         }
     }

+    //IteratorAggregate
+    public function getIterator() {
+        return new ArrayIterator($this->_source);
+    }
+
+    //Countable
+    public function count() {
+        return count($this->_source);
+    }
+
+    //ArrayAccess
+    public function offsetExists($offset) {
+        return $this->keyExists($offset);
+    }
+
+    public function offsetGet($offset) {
+        return $this->getRaw($offset);
+    }
+
+    public function offsetSet($offset, $value) {
+        throw new Exception("Zend_Filter_Input is a read-only collection.");
+    }
+
+    public function offsetUnset($offset) {
+        throw new Exception("Zend_Filter_Input is a read-only collection.");
+    }
+
     /**
      * Returns only the alphabetic characters in value.
      *
