--- Zend/Navigation/Page/Mvc.php	Mon Sep 26 14:46:24 2011
+++ Zend/Navigation/Page/Mvc.php	Mon Sep 26 14:46:24 2011
@@ -143,6 +143,9 @@
             $reqParams = array();
             if ($request) {
                 $reqParams = $request->getParams();
+                $reqParams[$request->getModuleKey()] = $request->getModuleName();
+                $reqParams[$request->getControllerKey()] = $request->getControllerName();
+                $reqParams[$request->getActionKey()] = $request->getActionName();
                 if (!array_key_exists('module', $reqParams)) {
                     $reqParams['module'] = $front->getDefaultModule();
                 }
