Index: tests/Zend/View/Helper/FormErrorsTest.php
===================================================================
--- tests/Zend/View/Helper/FormErrorsTest.php	(revision 25051)
+++ tests/Zend/View/Helper/FormErrorsTest.php	(working copy)
@@ -167,9 +167,26 @@
     public function testCanSetClassAttribute()
     {
         $options = array('class' => 'custom-class');
-        $acutallHtml = $this->helper->formErrors(array(), $options);
-        $this->assertEquals('<ul class="custom-class"><li></li></ul>', $acutallHtml);
+        $actuallHtml = $this->helper->formErrors(array(), $options);
+        $this->assertEquals('<ul class="custom-class"><li></li></ul>', $actuallHtml);
     }
+
+    /**
+     * @group ZF-5962
+     */
+    public function testCanSetElementStringsPerOptions()
+    {
+        $actual = $this->helper->formErrors(
+            array('foo', 'bar', 'baz'),
+            array(
+                 'elementStart'     => '<p>',
+                 'elementEnd'       => '</p>',
+                 'elementSeparator' => '<br>',
+            )
+        );
+
+        $this->assertEquals('<p>foo<br>bar<br>baz</p>', $actual);
+    }
 }
 
 // Call Zend_View_Helper_FormErrorsTest::main() if this source file is executed directly.
