Index: Zend_Framework_Incubator/library/Zend/Amf/Parse/Amf3/Deserializer.php
===================================================================
--- Zend_Framework_Incubator/library/Zend/Amf/Parse/Amf3/Deserializer.php	(Revision 15779)
+++ Zend_Framework_Incubator/library/Zend/Amf/Parse/Amf3/Deserializer.php	(Arbeitskopie)
@@ -152,7 +152,7 @@
             // Check if the integer should be negative
             if (($result & 0x10000000) != 0) {
                 //and extend the sign bit
-                $result |= 0xe0000000;
+                $result |= ~0xFFFFFFF;
             }
         }
         return $result;
