Index: library/Zend/Session.php
===================================================================
--- library/Zend/Session.php	(revision 23890)
+++ library/Zend/Session.php	(working copy)
@@ -575,11 +575,11 @@
                         unset($_SESSION['__ZF'][$namespace]['ENVGH']);
                     }
                 }
+                
+                if (isset($namespace) && empty($_SESSION['__ZF'][$namespace])) {
+                    unset($_SESSION['__ZF'][$namespace]);
+                }
             }
-
-            if (isset($namespace) && empty($_SESSION['__ZF'][$namespace])) {
-                unset($_SESSION['__ZF'][$namespace]);
-            }
         }
 
         if (isset($_SESSION['__ZF']) && empty($_SESSION['__ZF'])) {
