Index: HeadMeta.php
===================================================================
--- HeadMeta.php	(revision 22314)
+++ HeadMeta.php	(working copy)
@@ -180,6 +180,24 @@
         $this->set($item);
         return $this;
     }
+    
+    /**
+     * Retrieve item from placeholder container
+     * 
+     * @param mixed $value
+     * @return mixed|NULL
+     */
+    public function getItem($value)
+    {
+        $container = $this->getContainer();
+        foreach ($container->getArrayCopy() as $index => $item) {
+            if ($item->type == $value->type && $item->{$item->type} == $value->{$value->type}) {
+                return $item;
+            }
+        }
+        
+        return null;
+    }
 
     /**
      * Determine if item is valid
@@ -220,6 +238,11 @@
             $e->setView($this->view);
             throw $e;
         }
+        
+        if (null !== ($item = $this->getItem($value))) {
+            $value->content.= ','.$item->content;
+            return $this->set($value);
+        }
 
         return $this->getContainer()->append($value);
     }
@@ -278,6 +301,11 @@
             $e->setView($this->view);
             throw $e;
         }
+        
+        if (null !== ($item = $this->getItem($value))) {
+            $value->content = $item->content.','.$value->content;
+            return $this->set($value);
+        }
 
         return $this->getContainer()->prepend($value);
     }
