Index: library/Zend/Queue/Adapter/Db.php
===================================================================
--- library/Zend/Queue/Adapter/Db.php	(revision 17310)
+++ library/Zend/Queue/Adapter/Db.php	(working copy)
@@ -349,7 +349,7 @@
             $db->beginTransaction();
 
             $query = $db->select();
-            if ($this->_config['options'][Zend_Db_Select::FOR_UPDATE]) {
+            if ($this->_options['options'][Zend_Db_Select::FOR_UPDATE]) {
                 // turn on forUpdate
                 $query->forUpdate();
             }
