Index: library/Zend/Search/Lucene/Analysis/TokenFilter/LowerCase.php
===================================================================
--- library/Zend/Search/Lucene/Analysis/TokenFilter/LowerCase.php	(revision 24527)
+++ library/Zend/Search/Lucene/Analysis/TokenFilter/LowerCase.php	(working copy)
@@ -45,14 +45,8 @@
      */
     public function normalize(Zend_Search_Lucene_Analysis_Token $srcToken)
     {
-        $newToken = new Zend_Search_Lucene_Analysis_Token(
-                                     strtolower( $srcToken->getTermText() ),
-                                     $srcToken->getStartOffset(),
-                                     $srcToken->getEndOffset());
-
-        $newToken->setPositionIncrement($srcToken->getPositionIncrement());
-
-        return $newToken;
+        $srcToken->setTermText(strtolower($srcToken->getTermText()));
+        return $srcToken;
     }
 }
 
Index: library/Zend/Search/Lucene/Analysis/TokenFilter/LowerCaseUtf8.php
===================================================================
--- library/Zend/Search/Lucene/Analysis/TokenFilter/LowerCaseUtf8.php	(revision 24527)
+++ library/Zend/Search/Lucene/Analysis/TokenFilter/LowerCaseUtf8.php	(working copy)
@@ -57,14 +57,8 @@
      */
     public function normalize(Zend_Search_Lucene_Analysis_Token $srcToken)
     {
-        $newToken = new Zend_Search_Lucene_Analysis_Token(
-                                     mb_strtolower($srcToken->getTermText(), 'UTF-8'),
-                                     $srcToken->getStartOffset(),
-                                     $srcToken->getEndOffset());
-
-        $newToken->setPositionIncrement($srcToken->getPositionIncrement());
-
-        return $newToken;
+        $srcToken->setTermText(mb_strtolower($srcToken->getTermText(), 'UTF-8'));
+        return $srcToken;
     }
 }
 
Index: library/Zend/Search/Lucene/Analysis/Token.php
===================================================================
--- library/Zend/Search/Lucene/Analysis/Token.php	(revision 24527)
+++ library/Zend/Search/Lucene/Analysis/Token.php	(working copy)
@@ -123,6 +123,18 @@
     {
         return $this->_termText;
     }
+    
+    /**
+     * Sets the Token's term text.
+     * 
+     * @param string $text
+     * @return this
+     */
+    public function setTermText($text)
+    {
+        $this->_termText = $text;
+        return $this;
+    }
 
     /**
      * Returns this Token's starting offset, the position of the first character