Index: Zend/View/Helper/FormLabel.php
===================================================================
--- Zend/View/Helper/FormLabel.php	(revision 6748)
+++ Zend/View/Helper/FormLabel.php	(working copy)
@@ -52,7 +52,7 @@
         } else {
             // enabled; display label
             $xhtml = '<label'
-                   . ' for="' . $this->view->escape($name) . '"'
+                   . ' for="' . $this->view->escape($id) . '"'
                    . $this->_htmlAttribs($attribs)
                    . '>' . $this->view->escape($value) . '</label>';
         }
