Index: .
===================================================================
--- .	(revision 21211)
+++ .	(working copy)
@@ -184,26 +184,26 @@
         return $this;
     }
 
-	/**
-	 * Clears the specified HTTP header
-	 *
-	 * @param  string $name
-	 * @return Zend_Controller_Response_Abstract
-	 */
-	public function clearHeader($name)
-	{
-		if (! count($this->_headers)) {
-			return $this;
-		}
+        /**
+         * Clears the specified HTTP header
+         *
+         * @param  string $name
+         * @return Zend_Controller_Response_Abstract
+         */
+        public function clearHeader($name)
+        {
+                if (! count($this->_headers)) {
+                        return $this;
+                }
 
-		foreach ($this->_headers as $index => $header) {
-			if ($name == $header['name']) {
-				unset($this->_headers[$index]);
-			}
-		}
+                foreach ($this->_headers as $index => $header) {
+                        if ($name == $header['name']) {
+                                unset($this->_headers[$index]);
+                        }
+                }
 
-		return $this;
-	}
+                return $this;
+        }
 
     /**
      * Set raw HTTP header
@@ -244,23 +244,23 @@
         return $this;
     }
 
-	/**
-	 * Clears the specified raw HTTP header
-	 *
-	 * @param  string $headerRaw
-	 * @return Zend_Controller_Response_Abstract
-	 */
-	public function clearRawHeader($headerRaw)
-	{
-		if (! count($this->_headersRaw)) {
-			return $this;
-		}
+        /**
+         * Clears the specified raw HTTP header
+         *
+         * @param  string $headerRaw
+         * @return Zend_Controller_Response_Abstract
+         */
+        public function clearRawHeader($headerRaw)
+        {
+                if (! count($this->_headersRaw)) {
+                        return $this;
+                }
 
-		$key = array_search($headerRaw, $this->_headersRaw);
-		unset($this->_headersRaw[$key]);
+                $key = array_search($headerRaw, $this->_headersRaw);
+                unset($this->_headersRaw[$key]);
 
-		return $this;
-	}
+                return $this;
+        }
 
     /**
      * Clear all headers, normal and raw
