Index: Word.php
===================================================================
--- Word.php	(revision 11372)
+++ Word.php	(working copy)
@@ -300,11 +300,12 @@
      */
     public function generate()
     {
-	    $this->_session = null;
-        $id             = $this->_generateRandomId();
+        $id   = $this->_generateRandomId();
+        $word = $this->_generateWord();
+        
         $this->_setId($id);
-        $word           = $this->_generateWord();
         $this->_setWord($word);
+        
 	    return $id;
     }
 
