Index: library/Zend/Tool/Framework/Loader/IncludePathLoader.php
===================================================================
--- library/Zend/Tool/Framework/Loader/IncludePathLoader.php	(revision 16)
+++ library/Zend/Tool/Framework/Loader/IncludePathLoader.php	(working copy)
@@ -90,6 +90,9 @@
             
             // iterate over the accepted items
             foreach ($iterator as $item) {
+                if ($item->isDir()) {
+                    continue;
+                }
                 
                 // ensure that the same named file from separate include_paths is not loaded
                 $relativeItem = preg_replace('#^' . preg_quote($realIncludePath . DIRECTORY_SEPARATOR, '#') . '#', '', $item->getRealPath());
