--- Xml.php     2008-07-21 18:31:27.000000000 +0200
+++ XmlNew.php  2008-08-31 23:28:35.000000000 +0200
@@ -143,26 +143,32 @@
         $config = array();
 
         // Search for parent node values
-        if (count($xmlObject->attributes()) > 0) {
-            foreach ($xmlObject->attributes() as $key => $value) {
-                if ($key === 'extends') {
-                    continue;
-                }
+        $aNamespaces = array_merge(array(''), array_keys($xmlObject->getNamespaces(true)));
+        foreach($aNamespaces as $sNamespace) {
+            $aAttributes = empty($sNamespace) ? $xmlObject->attributes() : $xmlObject->attributes($sNamespace, true);
+            if (count($aAttributes) > 0) {
+                foreach ($aAttributes as $key => $value) {
+                    if (!empty($sNamespace)) {
+                        $key = sprintf('%s:%s', $sNamespace, $key);
+                    }
+                    if ($key === 'extends') {
+                        continue;
+                    }
 
-                $value = (string) $value;
+                    $value = (string) $value;
 
-                if (array_key_exists($key, $config)) {
-                    if (!is_array($config[$key])) {
-                        $config[$key] = array($config[$key]);
-                    }
+                    if (array_key_exists($key, $config)) {
+                        if (!is_array($config[$key])) {
+                            $config[$key] = array($config[$key]);
+                        }
 
-                    $config[$key][] = $value;
-                } else {
-                    $config[$key] = $value;
+                        $config[$key][] = $value;
+                    } else {
+                        $config[$key] = $value;
+                    }
                 }
             }
         }
-
         // Search for children
         if (count($xmlObject->children()) > 0) {
             foreach ($xmlObject->children() as $key => $value) {
@@ -176,7 +182,19 @@
                         $value = $this->_toArray($value);
                     }
                 } else {
-                    $value = (string) $value;
+                    // Search also for namespaces attributes
+                    $bNamespacesAttributesExists = false;
+                    $aNamespaces = array_keys($xmlObject->getNamespaces(true));
+                    foreach($aNamespaces as $sNamespace) {
+                        if (count($value->attributes($sNamespace, true)) > 0) {
+                            $value = $this->_toArray($value);
+                            $bNamespacesAttributesExists = true;
+                            break;
+                        }
+                    }
+                    if (!$bNamespacesAttributesExists) {
+                        $value = (string) $value;
+                    }
                 }
 
                 if (array_key_exists($key, $config)) {