Index: Application/Bootstrap/BootstrapAbstract.php
===================================================================
--- Application/Bootstrap/BootstrapAbstract.php	(revision 19391)
+++ Application/Bootstrap/BootstrapAbstract.php	(working copy)
@@ -117,10 +117,7 @@
         $options = array_change_key_case($options, CASE_LOWER);
         $this->_optionKeys = array_merge($this->_optionKeys, array_keys($options));
 
-        $methods = get_class_methods($this);
-        foreach ($methods as $key => $method) {
-            $methods[$key] = strtolower($method);
-        }
+        $methods = array_map('strtolower', get_class_methods($this));
 
         if (array_key_exists('pluginpaths', $options)) {
             $pluginLoader = $this->getPluginLoader();
@@ -132,7 +129,14 @@
         }
 
         foreach ($options as $key => $value) {
-            $method = 'set' . strtolower($key);
+            $key = strtolower($key);
+            $blackListKey = array('application', 'container',
+                                  'options'    , 'pluginloader');
+            if (in_array($key, $blackListKey)) {
+                require_once 'Zend/Application/Bootstrap/Exception.php';
+                throw new Zend_Application_Bootstrap_Exception("Option \"{$key}\" invalid");
+            }
+            $method = 'set' . $key;
 
             if (in_array($method, $methods)) {
                 $this->$method($value);
