--- tests/Zend/Http/UserAgentTest.php	(revision 23226)
+++ tests/Zend/Http/UserAgentTest.php	(working copy)
@@ -551,4 +551,18 @@
         $this->assertEquals('.foobar', $storage->getNamespace());
         $this->assertEquals('data', $storage->getMember());
     }
+
+    /**
+     * @group ZF-10595
+     */
+    public function testAGroupDefinedAndSerialized()
+    {
+        $config    = $this->config;
+        $userAgent = new Zend_Http_UserAgent($config);
+        $device    = $userAgent->getDevice();
+
+        $userAgent = unserialize(serialize($userAgent));
+        $device    = $userAgent->getDevice();
+        $groups = $device->getAllGroups();
+    }
 }
Index: library/Zend/Http/UserAgent/AbstractDevice.php
===================================================================
--- library/Zend/Http/UserAgent/AbstractDevice.php	(revision 23226)
+++ library/Zend/Http/UserAgent/AbstractDevice.php	(working copy)
@@ -90,6 +90,13 @@
     protected $_aFeatures = array();
 
     /**
+     * Browser/Device features groups
+     *
+     * @var array
+     */
+    protected $_aGroup = array();
+
+    /**
      * Constructor
      *
      * @param  null|string|array $userAgent If array, restores from serialized version
@@ -121,6 +128,7 @@
     {
         $spec = array(
             '_aFeatures'      => $this->_aFeatures,
+            '_aGroup'         => $this->_aGroup,
             '_browser'        => $this->_browser,
             '_browserVersion' => $this->_browserVersion,
             '_userAgent'      => $this->_userAgent,
