Index: tests/Zend/Auth/Storage/SessionTest.php
===================================================================
--- tests/Zend/Auth/Storage/SessionTest.php	(revision 0)
+++ tests/Zend/Auth/Storage/SessionTest.php	(revision 0)
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Auth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * PHPUnit_Framework_TestCase
+ */
+
+
+/**
+ * @see Zend_Auth
+ */
+require_once 'Zend/Auth.php';
+
+/**
+ * @see Zend_Auth_Storage_Session
+ */
+require_once 'Zend/Auth/Storage/Session.php';
+
+
+/**
+ * @category   Zend
+ * @package    Zend_Auth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Auth_Storage
+ */
+class Zend_Auth_Storage_SessionTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Zend_Auth instance
+     * @var Zend_Auth
+     */
+    protected $auth;
+    
+    
+    public function setUp()
+    {
+        $this->auth = Zend_Auth::getInstance();
+        $this->auth->setStorage(new Zend_Auth_Storage_Session());
+    }
+    
+    /**
+     * @group ZF-11694
+     */
+    public function testClearIdentityWorksWithObjects()
+    {
+        // Authenticate the user and confirm
+        $adapter = new ZF11694_TestAuthAdapter('test','foobar');
+        $result = $this->auth->authenticate($adapter);
+        $this->assertTrue($result->isValid());
+        $this->assertEquals('ZF11694_User', get_class($result->getIdentity()));
+        
+        // Write identity to storage and confirm
+        $this->auth->getStorage()->write($result->getIdentity());
+        $this->assertTrue($this->auth->hasIdentity());
+        $this->assertEquals('ZF11694_User', get_class($this->auth->getIdentity()));
+        
+        // Now erase it and confirm
+        $this->auth->clearIdentity();
+        $this->assertFalse($this->auth->hasIdentity());
+        $this->assertNotEquals('ZF11694_User', get_class($this->auth->getIdentity()));
+    }
+    
+}
+
+class ZF11694_User
+{
+    public $username;
+    public $password;
+}
+
+class ZF11694_TestAuthAdapter implements Zend_Auth_Adapter_Interface
+{
+    protected $user;
+    
+    public function __construct($username, $password)
+    {
+        $this->user = new ZF11694_User;
+        $this->user->username = $username;
+        $this->user->password = $password;
+    }
+    
+    public function authenticate()
+    {
+        return new Zend_Auth_Result(
+            Zend_Auth_Result::SUCCESS,
+            $this->user
+        );
+    }
+    
+}
+

Property changes on: tests/Zend/Auth/Storage/SessionTest.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Auth/Storage/AllTests.php
===================================================================
--- tests/Zend/Auth/Storage/AllTests.php	(revision 0)
+++ tests/Zend/Auth/Storage/AllTests.php	(revision 0)
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Auth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+if (!defined('PHPUnit_MAIN_METHOD')) {
+    define('PHPUnit_MAIN_METHOD', 'Zend_Auth_Storage_AllTests::main');
+}
+
+require_once 'Zend/Auth/Storage/DigestTest.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_Auth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Auth_Storage
+ */
+class Zend_Auth_Storage_AllTests
+{
+    /**
+     * Runs this test suite
+     *
+     * @return void
+     */
+    public static function main()
+    {
+        PHPUnit_TextUI_TestRunner::run(self::suite());
+    }
+
+    /**
+     * Creates and returns this test suite
+     *
+     * @return PHPUnit_Framework_TestSuite
+     */
+    public static function suite()
+    {
+        $suite = new PHPUnit_Framework_TestSuite('Zend Framework - Zend_Auth_Storage');
+
+        $suite->addTestSuite('Zend_Auth_Storage_SessionTest');
+
+        return $suite;
+    }
+}
+
+if (PHPUnit_MAIN_METHOD == 'Zend_Auth_Storage_AllTests::main') {
+    Zend_Auth_Storage_AllTests::main();
+}

Property changes on: tests/Zend/Auth/Storage/AllTests.php
___________________________________________________________________
Added: svn:keywords
   + Id

Index: tests/Zend/Auth/AllTests.php
===================================================================
--- tests/Zend/Auth/AllTests.php	(revision 24509)
+++ tests/Zend/Auth/AllTests.php	(working copy)
@@ -25,6 +25,7 @@
 }
 
 require_once 'Zend/Auth/Adapter/AllTests.php';
+require_once 'Zend/Auth/Storage/AllTests.php';
 
 /**
  * @category   Zend
@@ -56,6 +57,7 @@
         $suite = new PHPUnit_Framework_TestSuite('Zend Framework - Zend_Auth');
 
         $suite->addTest(Zend_Auth_Adapter_AllTests::suite());
+        $suite->addTest(Zend_Auth_Storage_AllTests::suite());
 
         return $suite;
     }
