--- trunk/library/Zend/Mail/Protocol/Abstract.php	2008-09-16 17:23:07.000000000 +0200
+++ patch/library/Zend/Mail/Protocol/Abstract.php	2008-09-20 15:13:02.000000000 +0200
@@ -221,7 +221,7 @@
         $errorStr = '';
 
         // open connection
-        $this->_socket = stream_socket_client($remote, $errorNum, $errorStr, self::TIMEOUT_CONNECTION);
+        $this->_socket = @stream_socket_client($remote, $errorNum, $errorStr, self::TIMEOUT_CONNECTION);
 
         if ($this->_socket === false) {
             if ($errorNum == 0) {
