diff -ru Amf-v1.9.7/Parse/Amf3/Serializer.php Amf.perf/Parse/Amf3/Serializer.php
--- Amf-v1.9.7/Parse/Amf3/Serializer.php	2009-11-12 10:26:19.000000000 -0600
+++ Amf.perf/Parse/Amf3/Serializer.php	2010-01-17 22:30:46.000000000 -0600
@@ -220,9 +220,9 @@
             return $this;
         }
 
-        $ref = array_search($string, $this->_referenceStrings, true);
+        $ref = array_key_exists($string, $this->_referenceStrings) ? $this->_referenceStrings[$string] : false;
         if($ref === false){
-            $this->_referenceStrings[] = $string;
+            $this->_referenceStrings[$string] = count($this->_referenceStrings);
             $this->writeBinaryString($string);
         } else {
             $ref <<= 1;
