Index: library/Zend/Search/Lucene/Storage/Directory/Filesystem.php
===================================================================
--- library/Zend/Search/Lucene/Storage/Directory/Filesystem.php	(revision 13739)
+++ library/Zend/Search/Lucene/Storage/Directory/Filesystem.php	(working copy)
@@ -184,10 +184,10 @@
         unset($this->_fileHandlers[$filename]);
         $this->_fileHandlers[$filename] = new Zend_Search_Lucene_Storage_File_Filesystem($this->_dirPath . '/' . $filename, 'w+b');
 
-        // Set file permissions, but don't care about any possible failures, since file may be already
-        // created by anther user which has to care about right permissions
-        @chmod($this->_dirPath . '/' . $filename, self::$_defaultFilePermissions);
-
+        $pathAndFilename = $this->_dirPath . '/' . $filename;
+        if (fileowner($pathAndFilename) === getmyuid()) {
+            chmod($pathAndFilename, self::$_defaultFilePermissions);
+        }
         return $this->_fileHandlers[$filename];
     }
 
