--- library/Zend/Dojo/Form/Element/Editor.php	(revision 24415)
+++ library/Zend/Dojo/Form/Element/Editor.php	(working copy)
@@ -391,13 +391,15 @@
      */
     public function setHeight($height)
     {
-        if (!preg_match('/^\d+(em|px|%)?$/i', $height)) {
-            require_once 'Zend/Form/Element/Exception.php';
-            throw new Zend_Form_Element_Exception('Invalid height provided; must be integer or CSS measurement');
+        if ($height !== '') {
+            if (!preg_match('/^\d+(em|px|%)?$/i', $height)) {
+                require_once 'Zend/Form/Element/Exception.php';
+                throw new Zend_Form_Element_Exception('Invalid height provided; must be integer or CSS measurement');
+            }
+            if (!preg_match('/(em|px|%)$/', $height)) {
+                $height .= 'px';
+            }
         }
-        if (!preg_match('/(em|px|%)$/', $height)) {
-            $height .= 'px';
-        }
         return $this->setDijitParam('height', $height);
     }
 
