Index: tests/Zend/Application/_files/ZF-8412.ini
===================================================================
--- tests/Zend/Application/_files/ZF-8412.ini	(revision )
+++ tests/Zend/Application/_files/ZF-8412.ini	(revision )
@@ -0,0 +1,7 @@
+[production]
+foo = "bar"
+
+[testing : production]
+resources.frontcontroller.foo = 1
+resources.frontcontroller.controllerDirectory = "/controllers"
+resources.frontcontroller.bar = 1
\ No newline at end of file
Index: tests/Zend/Application/ApplicationTest.php
===================================================================
--- tests/Zend/Application/ApplicationTest.php	(revision 23162)
+++ tests/Zend/Application/ApplicationTest.php	(revision )
@@ -489,7 +489,32 @@
 
         $this->assertEquals('baz', $application->getOption('foo'));
     }
+
+    /**
+     * @group ZF-8412
+     * @expectedException Zend_Controller_Dispatcher_Exception
+     */
+    public function testFrontControllerResourceAcceptsCustomParams()
+    {
+        $application = new Zend_Application('testing', array(
+            'config' => array(
+                dirname(__FILE__) . '/_files/ZF-8412.ini'
+                )
+            )
+        );
+
+        $bootstrapper = $application->getBootstrap();
+        $bootstrapper->bootstrap('frontcontroller');
+        
+        $front = $bootstrapper->getResource('frontcontroller');
+        $this->assertTrue($front instanceof Zend_Controller_Front);
+
+        $controllerDirectory = $front->getControllerDirectory();
+        $this->assertTrue(is_array($controllerDirectory));
+        $this->assertEquals(count($controllerDirectory), 1);
+        $front->dispatch();
-}
+    }
+}
 
 if (PHPUnit_MAIN_METHOD == 'Zend_Application_ApplicationTest::main') {
     Zend_Application_ApplicationTest::main();
