Index: Zend/Service/Yahoo.php
===================================================================
--- Zend/Service/Yahoo.php      (révision 2613)
+++ Zend/Service/Yahoo.php      (copie de travail)
@@ -152,7 +152,7 @@

         $this->_rest = new Zend_Service_Rest;

-        $this->_rest->setUri('api.search.yahoo.com');
+        $this->_rest->setUri('http://api.search.yahoo.com');

         $response = $this->_rest->restGet('/ImageSearchService/V1/imageSearch', $options);