Index: library/Zend/Navigation/Page.php
===================================================================
--- library/Zend/Navigation/Page.php	(revision 25123)
+++ library/Zend/Navigation/Page.php	(working copy)
@@ -258,10 +258,10 @@
                   isset($options['module']) || isset($options['route']) ||
                   isset($options['params']);
 
-        if ($hasMvc && !$hasUri) {
+        if ($hasMvc) {
             require_once 'Zend/Navigation/Page/Mvc.php';
             return new Zend_Navigation_Page_Mvc($options);
-        } elseif ($hasUri && !$hasMvc) {
+        } elseif ($hasUri) {
             require_once 'Zend/Navigation/Page/Uri.php';
             return new Zend_Navigation_Page_Uri($options);
         } else {
