Index: library/Zend/Validate/Hostname.php
===================================================================
--- library/Zend/Validate/Hostname.php	(revision 7384)
+++ library/Zend/Validate/Hostname.php	(working copy)
@@ -391,7 +391,7 @@
         }
 
         // Check input against local network name schema; last chance to pass validation
-        $regexLocal = "/^(([a-zA-Z0-9\x2d]{1,63}\x2e)*[a-zA-Z0-9\x2d]{1,63}){1,254}$/";
+        $regexLocal = '/^(([a-zA-Z0-9\x2d]{1,63}\x2e)*[a-zA-Z0-9\x2d]{1,63}){1,254}$/';
         $status = @preg_match($regexLocal, $valueString);
         if (false === $status) {
             /**
