Index: library/Zend/Cache/Backend/File.php
===================================================================
--- library/Zend/Cache/Backend/File.php	(revision 22081)
+++ library/Zend/Cache/Backend/File.php	(working copy)
@@ -395,6 +395,29 @@
     }
 
     /**
+     * Public frontend to get an option value
+     *
+     * @param  string $name  Name of the option
+     * @throws Zend_Cache_Exception
+     * @return mixed option value
+     */
+    public function getOption($name)
+    {
+        if (is_string($name)) {
+            $name = strtolower($name);
+            if (array_key_exists($name, $this->_options)) {
+                // This is a Core option
+                return $this->_options[$name];
+            }
+            if (array_key_exists($name, $this->_specificOptions)) {
+                // This a specic option of this frontend
+                return $this->_specificOptions[$name];
+            }
+        }
+        Zend_Cache::throwException("Incorrect option name : $name");
+    }
+
+    /**
      * Give (if possible) an extra lifetime to the given cache id
      *
      * @param string $id cache id
Index: tests/Zend/Cache/FileBackendTest.php
===================================================================
--- tests/Zend/Cache/FileBackendTest.php	(revision 22081)
+++ tests/Zend/Cache/FileBackendTest.php	(working copy)
@@ -1,4 +1,4 @@
-<?php
+﻿<?php
 /**
  * Zend Framework
  *
@@ -126,6 +126,11 @@
         $this->assertFalse($res);
     }
 
+    public function testSettingOptionsWithZendConfig()
+    {
+        $test = new Zend_Cache_Backend_File(array('cache_dir' => '/tmp/cache'));
+        $this->assertEquals('/tmp/cache', $test->getOption('cache_dir'));
+    }
 }
 
 
