Index: tests/Zend/Gdata/CalendarOnlineTest.php
===================================================================
--- tests/Zend/Gdata/CalendarOnlineTest.php	(revision 23872)
+++ tests/Zend/Gdata/CalendarOnlineTest.php	(working copy)
@@ -62,20 +62,21 @@
     }
 
     /**
-     * @see ZF-1701
+     * @group ZF-1701
      */
-    /*
     public function testCalendarOnlineFeed()
     {
         $eventFeed = $this->gdata->getCalendarEventFeed();
-        foreach ($eventFeed as $event) {
-            $title = $event->title;
-            $times = $event->when;
-            $location = $event->where;
-            $recurrence = $event->recurrence;
+        $this->assertTrue(strpos($eventFeed->title->text, TESTS_ZEND_GDATA_CLIENTLOGIN_EMAIL)
+                !== false);
+        $eventCount = 0;
+        foreach ( $eventFeed as $event ) {
+            $this->assertType('Zend_Gdata_Calendar_EventEntry', $event);
+            $eventCount++;
         }
+        $this->assertTrue($eventCount > 0 );
+        $this->assertTrue(count($eventFeed) == $eventCount);
     }
-    */
 
     function getEvent($eventId)
     {
