--- ZendFramework-1.7.5/tests/Zend/Dom/QueryTest.php	2008-10-15 18:00:56.000000000 +0200
+++ ZendFramework-1.7.5-new/tests/Zend/Dom/QueryTest.php	2009-02-26 10:36:42.000000000 +0100
@@ -100,6 +100,18 @@
         $this->assertEquals(Zend_Dom_Query::DOC_HTML, $this->query->getDocumentType());
     }
 
+    public function testXmlPrologShouldBeIgnored()
+    {
+        $xml = <<<EOT
+<?xml version="1.0" encoding="UTF-8"?>
+<feed xmlns="http://www.w3.org/2005/Atom">
+</feed>
+EOT;
+        $this->query->setDocument($xml);
+        $result = $this->query->query('feed');
+        $this->assertEquals(1, count($result));
+    }
+
     public function testQueryingWithoutRegisteringDocumentShouldThrowException()
     {
         try {
