Index: library/Zend/View/Helper/FormFile.php
===================================================================
--- library/Zend/View/Helper/FormFile.php	(revision 24549)
+++ library/Zend/View/Helper/FormFile.php	(working copy)
@@ -62,19 +62,13 @@
             $disabled = ' disabled="disabled"';
         }
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         // build the element
         $xhtml = '<input type="file"'
                 . ' name="' . $this->view->escape($name) . '"'
                 . ' id="' . $this->view->escape($id) . '"'
                 . $disabled
                 . $this->_htmlAttribs($attribs)
-                . $endTag;
+                . $this->getClosingBracket();
 
         return $xhtml;
     }
Index: library/Zend/View/Helper/FormReset.php
===================================================================
--- library/Zend/View/Helper/FormReset.php	(revision 24549)
+++ library/Zend/View/Helper/FormReset.php	(working copy)
@@ -64,12 +64,6 @@
             $disabled = ' disabled="disabled"';
         }
 
-        // get closing tag
-        $endTag = '>';
-        if ($this->view->doctype()->isXhtml()) {
-            $endTag = ' />';
-        }
-
         // Render button
         $xhtml = '<input type="reset"'
                . ' name="' . $this->view->escape($name) . '"'
@@ -82,7 +76,7 @@
         }
 
         // add attributes, close, and return
-        $xhtml .= $this->_htmlAttribs($attribs) . $endTag;
+        $xhtml .= $this->_htmlAttribs($attribs) . $this->getClosingBracket();
         return $xhtml;
     }
 }
Index: library/Zend/View/Helper/FormPassword.php
===================================================================
--- library/Zend/View/Helper/FormPassword.php	(revision 24549)
+++ library/Zend/View/Helper/FormPassword.php	(working copy)
@@ -74,12 +74,6 @@
             unset($attribs['renderPassword']);
         }
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         // render the element
         $xhtml = '<input type="password"'
                 . ' name="' . $this->view->escape($name) . '"'
@@ -87,7 +81,7 @@
                 . $valueString
                 . $disabled
                 . $this->_htmlAttribs($attribs)
-                . $endTag;
+                . $this->getClosingBracket();
 
         return $xhtml;
     }
Index: library/Zend/View/Helper/FormImage.php
===================================================================
--- library/Zend/View/Helper/FormImage.php	(revision 24549)
+++ library/Zend/View/Helper/FormImage.php	(working copy)
@@ -80,12 +80,6 @@
             $disabled = ' disabled="disabled"';
         }
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         // build the element
         $xhtml = '<input type="image"'
                 . ' name="' . $this->view->escape($name) . '"'
@@ -94,7 +88,7 @@
                 . $value
                 . $disabled
                 . $this->_htmlAttribs($attribs)
-                . $endTag;
+                . $this->getClosingBracket();
 
         return $xhtml;
     }
Index: library/Zend/View/Helper/FormText.php
===================================================================
--- library/Zend/View/Helper/FormText.php	(revision 24549)
+++ library/Zend/View/Helper/FormText.php	(working copy)
@@ -65,19 +65,13 @@
             $disabled = ' disabled="disabled"';
         }
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         $xhtml = '<input type="text"'
                 . ' name="' . $this->view->escape($name) . '"'
                 . ' id="' . $this->view->escape($id) . '"'
                 . ' value="' . $this->view->escape($value) . '"'
                 . $disabled
                 . $this->_htmlAttribs($attribs)
-                . $endTag;
+                . $this->getClosingBracket();
 
         return $xhtml;
     }
Index: library/Zend/View/Helper/FormCheckbox.php
===================================================================
--- library/Zend/View/Helper/FormCheckbox.php	(revision 24549)
+++ library/Zend/View/Helper/FormCheckbox.php	(working copy)
@@ -81,12 +81,6 @@
             $disabled = ' disabled="disabled"';
         }
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         // build the element
         $xhtml = '';
         if (!$disable && !strstr($name, '[]')) {
@@ -99,7 +93,7 @@
                 . $checkedOptions['checkedString']
                 . $disabled
                 . $this->_htmlAttribs($attribs)
-                . $endTag;
+                . $this->getClosingBracket();
 
         return $xhtml;
     }
Index: library/Zend/View/Helper/FormRadio.php
===================================================================
--- library/Zend/View/Helper/FormRadio.php	(revision 24549)
+++ library/Zend/View/Helper/FormRadio.php	(working copy)
@@ -123,12 +123,6 @@
         // ensure value is an array to allow matching multiple times
         $value = (array) $value;
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         // Set up the filter - Alnum + hyphen + underscore
         require_once 'Zend/Filter/PregReplace.php';
         $pattern = @preg_match('/\pL/u', 'a') 
@@ -172,7 +166,7 @@
                     . $checked
                     . $disabled
                     . $this->_htmlAttribs($attribs)
-                    . $endTag
+                    . $this->getClosingBracket()
                     . (('append' == $labelPlacement) ? $opt_label : '')
                     . '</label>';
 
Index: library/Zend/View/Helper/FormSubmit.php
===================================================================
--- library/Zend/View/Helper/FormSubmit.php	(revision 24549)
+++ library/Zend/View/Helper/FormSubmit.php	(working copy)
@@ -67,12 +67,6 @@
             $id = ' id="' . $this->view->escape($id) . '"';
         }
 
-        // XHTML or HTML end tag?
-        $endTag = ' />';
-        if (($this->view instanceof Zend_View_Abstract) && !$this->view->doctype()->isXhtml()) {
-            $endTag= '>';
-        }
-
         // Render the button.
         $xhtml = '<input type="submit"'
                . ' name="' . $this->view->escape($name) . '"'
@@ -80,7 +74,7 @@
                . ' value="' . $this->view->escape($value) . '"'
                . $disabled
                . $this->_htmlAttribs($attribs)
-               . $endTag;
+               . $this->getClosingBracket();
 
         return $xhtml;
     }
Index: tests/Zend/View/Helper/FormImageTest.php
===================================================================
--- tests/Zend/View/Helper/FormImageTest.php	(revision 24549)
+++ tests/Zend/View/Helper/FormImageTest.php	(working copy)
@@ -62,6 +62,8 @@
     protected function setUp()
     {
         $this->view = new Zend_View();
+        $this->view->doctype('HTML4_LOOSE');  // Reset doctype to default
+        
         $this->helper = new Zend_View_Helper_FormImage();
         $this->helper->setView($this->view);
     }
@@ -92,6 +94,29 @@
         $this->assertRegexp('/<input[^>]*?name="foo"/', $button);
         $this->assertRegexp('/<input[^>]*?type="image"/', $button);
     }
+    
+    /**
+     * @group ZF-11477
+     */
+    public function testRendersAsHtmlByDefault()
+    {
+        $test = $this->helper->formImage(array(
+            'name' => 'foo',
+        ));
+        $this->assertNotContains(' />', $test);
+    }
+
+    /**
+     * @group ZF-11477
+     */
+    public function testCanRendersAsXHtml()
+    {
+        $this->view->doctype('XHTML1_STRICT');
+        $test = $this->helper->formImage(array(
+            'name' => 'foo',
+        ));
+        $this->assertContains(' />', $test);
+    }
 }
 
 // Call Zend_View_Helper_FormImageTest::main() if this source file is executed directly.
Index: tests/Zend/View/Helper/FormRadioTest.php
===================================================================
--- tests/Zend/View/Helper/FormRadioTest.php	(revision 24549)
+++ tests/Zend/View/Helper/FormRadioTest.php	(working copy)
@@ -59,6 +59,7 @@
     public function setUp()
     {
         $this->view   = new Zend_View();
+        $this->view->doctype('HTML4_LOOSE'); // Set default doctype
         $this->helper = new Zend_View_Helper_FormRadio();
         $this->helper->setView($this->view);
     }
@@ -429,6 +430,45 @@
         $this->assertRegExp('/<input([^>]*)(id="'.$name.'--1")([^>]*)(checked="checked")/', $html);
     }
     
+    /**
+     * @group ZF-11477
+     */
+    public function testRendersAsHtmlByDefault()
+    {
+        $options = array(
+            'foo' => 'Foo',
+            'bar' => 'Bar',
+            'baz' => 'Baz'
+        );
+        $html = $this->helper->formRadio(array(
+            'name'    => 'foo',
+            'options' => $options,
+        ));
+
+        $this->assertContains('value="foo">', $html);
+        $this->assertContains('value="bar">', $html);
+        $this->assertContains('value="baz">', $html);
+    }
+
+    /**
+     * @group ZF-11477
+     */
+    public function testCanRendersAsXHtml()
+    {
+        $this->view->doctype('XHTML1_STRICT');
+        $options = array(
+            'foo' => 'Foo',
+            'bar' => 'Bar',
+            'baz' => 'Baz'
+        );
+        $html = $this->helper->formRadio(array(
+            'name'    => 'foo',
+            'options' => $options,
+        ));
+        $this->assertContains('value="foo" />', $html);
+        $this->assertContains('value="bar" />', $html);
+        $this->assertContains('value="baz" />', $html);
+    }
 }
 
 // Call Zend_View_Helper_FormRadioTest::main() if this source file is executed directly.
