--- ZendFramework-1.7.5/tests/Zend/Dojo/View/Helper/EditorTest.php	2008-10-18 16:29:41.000000000 +0200
+++ ZendFramework-1.7.5-new/tests/Zend/Dojo/View/Helper/EditorTest.php	2009-02-26 11:52:08.000000000 +0100
@@ -176,6 +176,14 @@
         $this->helper->editor('foo');
         $this->assertFalse($this->view->dojo()->registerDojoStylesheet());
     }
+
+    public function testNonCorePluginsModulesShouldBeRequiredByDojoContainer()
+    {
+        $plugins = array('copy', 'cut', 'paste', 'createLink');
+        $html = $this->helper->editor('foo', '', array('plugins' => $plugins));
+        $modules = $this->view->dojo()->getModules();
+        $this->assertContains('dijit._editor.plugins.LinkDialog', $modules);
+    }
 }
 
 // Call Zend_Dojo_View_Helper_EditorTest::main() if this source file is executed directly.
