Index: tests/Zend/Navigation/PageTest.php
===================================================================
--- tests/Zend/Navigation/PageTest.php	(revision 24362)
+++ tests/Zend/Navigation/PageTest.php	(working copy)
@@ -684,6 +684,14 @@
 
         $page->setVisible(0);
         $this->assertFalse($page->isVisible());
+        
+        /**
+         * ZF-10146
+         * 
+         * @link http://framework.zend.com/issues/browse/ZF-10146
+         */
+        $page->setVisible('False');
+        $this->assertFalse($page->isVisible());
 
         $page->setVisible(array());
         $this->assertFalse($page->isVisible());
