Index: Route.php
===================================================================
--- Route.php	(revision 2914)
+++ Route.php	(working copy)
@@ -92,11 +92,14 @@
      * Matches a user submitted path with parts defined by a map. Assigns and 
      * returns an array of variables on a successful match.  
      *
-     * @param string Path used to match against this routing map 
+     * @param Zend_Controller_Request_Http Request used to match against this 
+     * routing map
      * @return array|false An array of assigned values or a false on a mismatch
      */
-    public function match($path)
+    public function match(Zend_Controller_Request_Http $request)
     {
+        // This matcher only needs the path info
+        $path = $request->getPathInfo();
 
         $pathStaticCount = 0;
         $defaults = $this->_defaults;
