Index: Db.php
===================================================================
--- Db.php	(revision 18383)
+++ Db.php	(working copy)
@@ -87,7 +87,33 @@
             throw new Zend_Queue_Exception('Options array item: Zend_Db_Select::FOR_UPDATE must be boolean');
         }
 
-        $options = &$this->_options['driverOptions'];
+        if ($this->_options['dbAdapter'] != null 
+                && $this->_options['dbAdapter'] instanceof Zend_Db_Adapter_Abstract) {
+        	$db = $this->_options['dbAdapter'];
+        } else {
+        	$db = $this->_initDbAdapter();
+        }
+
+        $this->_queueTable = new Zend_Queue_Adapter_Db_Queue(array(
+            'db' => $db,
+        ));
+
+        $this->_messageTable = new Zend_Queue_Adapter_Db_Message(array(
+            'db' => $db,
+        ));
+        
+    }
+    
+    /**
+     * Initialize Db adapter using 'driverOptions' section of the _options array 
+     *
+     * Throws an exception if the adapter cannot connect to DB.
+     *
+     * @return Zend_Db_Adapter_Abstract
+     * @throws Zend_Queue_Exception
+     */
+    private function _initDbAdapter() {
+    	$options = &$this->_options['driverOptions'];
         if (!array_key_exists('type', $options)) {
             require_once 'Zend/Queue/Exception.php';
             throw new Zend_Queue_Exception("Configuration array must have a key for 'type' for the database type to use");
@@ -118,18 +144,12 @@
 
         try {
             $db = Zend_Db::factory($type, $options);
-
-            $this->_queueTable = new Zend_Queue_Adapter_Db_Queue(array(
-                'db' => $db,
-            ));
-
-            $this->_messageTable = new Zend_Queue_Adapter_Db_Message(array(
-                'db' => $db,
-            ));
         } catch (Zend_Db_Exception $e) {
             require_once 'Zend/Queue/Exception.php';
             throw new Zend_Queue_Exception('Error connecting to database: ' . $e->getMessage(), $e->getCode());
         }
+        
+        return $db;
     }
 
     /********************************************************************
