Index: library/Zend/Server/Reflection/Function/Abstract.php
===================================================================
--- library/Zend/Server/Reflection/Function/Abstract.php	(revision 23180)
+++ library/Zend/Server/Reflection/Function/Abstract.php	(working copy)
@@ -339,7 +339,9 @@
             require_once 'Zend/Server/Reflection/Exception.php';
             throw new Zend_Server_Reflection_Exception(
                'Variable number of arguments is not supported for services (except optional parameters). '
-             . 'Number of function arguments must correspond to actual number of arguments described in a docblock.');
+             . 'Number of function arguments in ' . $function->getDeclaringClass()->getName() . '::'
+             . $function->getName() . '() must correspond to actual number of arguments described in the '
+             . 'docblock.');
         }
 
         $paramTypes = array();
