Index: Zend/Application/Resource/Mail.php
===================================================================
--- Zend/Application/Resource/Mail.php	(revision 23688)
+++ Zend/Application/Resource/Mail.php	(working copy)
@@ -124,6 +124,7 @@
         }
 
         unset($options['type']);
+        unset($options['register']);
 
         switch($transportName) {
             case 'Zend_Mail_Transport_Smtp':
