Index: Request/Http.php
===================================================================
--- Request/Http.php	(revision 4776)
+++ Request/Http.php	(working copy)
@@ -324,6 +324,8 @@
         if ($requestUri === null) { 
             if (isset($_SERVER['HTTP_X_REWRITE_URL'])) { // check this first so IIS will catch
                 $requestUri = $_SERVER['HTTP_X_REWRITE_URL']; 
+            } elseif (isset($_SERVER['REDIRECT_URL'])) { // Check is using mod_rewrite PT flag
+            	$requestUri = $_SERVER['REDIRECT_URL'];
             } elseif (isset($_SERVER['REQUEST_URI'])) { 
                 $requestUri = $_SERVER['REQUEST_URI']; 
             } elseif (isset($_SERVER['ORIG_PATH_INFO'])) { // IIS 5.0, PHP as CGI
