Index: library/Zend/Form/DisplayGroup.php
===================================================================
--- library/Zend/Form/DisplayGroup.php	(revision 25030)
+++ library/Zend/Form/DisplayGroup.php	(working copy)
@@ -977,6 +977,16 @@
     }
 
     /**
+     * Does this display group have its own specific translator?
+     *
+     * @return bool
+     */
+    public function hasTranslator()
+    {
+        return (bool) $this->getTranslator();
+    }
+
+    /**
      * Indicate whether or not translation should be disabled
      *
      * @param  bool $flag
