Index: tests/Zend/Application/Resource/LocaleTest.php
===================================================================
--- tests/Zend/Application/Resource/LocaleTest.php	(revision 22824)
+++ tests/Zend/Application/Resource/LocaleTest.php	(working copy)
@@ -117,9 +117,17 @@
         $this->assertTrue(Zend_Registry::isRegistered('Foo_Bar'));
         $this->assertSame(Zend_Registry::get('Foo_Bar'), $locale);
     }
-    
+
+    /** 
+     * @group ZF-10381
+     */
     public function testOptionsPassedToResourceAreUsedToSetLocaleState1()
     {
+        $locales = Zend_Locale::getBrowser() + Zend_Locale::getEnvironment();
+        if (empty($locales) || in_array('kok_IN', $locales)) {
+            $this->markTestSkipped(__METHOD__ . ': you need to configure a locale');
+        }
+
         $options = array(
             'default'      => 'kok_IN',
         );
@@ -128,8 +136,7 @@
         $resource->setBootstrap($this->bootstrap);
         $resource->init();
         $locale   = $resource->getLocale();
-        
-        // This test will fail if your configured locale is kok_IN
+
         $this->assertFalse('kok_IN' == $locale->__toString());
         $this->assertSame(Zend_Registry::get('Zend_Locale'), $locale);
     }
