Index: .
===================================================================
--- .	(revision 11713)
+++ .	(working copy)
@@ -72,8 +72,9 @@
     public function __construct($name, $uri, $extractComplexTypes = true)
     {
         if ($uri instanceof Zend_Uri_Http) {
-            $uri = $this->_uri = $uri->getUri();
+            $uri = $uri->getUri();
         }
+        $this->_uri = $uri;
 
         /**
          * @todo change DomDocument object creation from cparsing to construxting using API
