### Eclipse Workspace Patch 1.0
#P Clever
Index: library/Zend/Form/Decorator/Label.php
===================================================================
--- library/Zend/Form/Decorator/Label.php	(revision 17462)
+++ library/Zend/Form/Decorator/Label.php	(working copy)
@@ -314,7 +314,7 @@
             require_once 'Zend/Form/Decorator/HtmlTag.php';
             $decorator = new Zend_Form_Decorator_HtmlTag();
             $decorator->setOptions(array('tag' => $tag,
-                                         'id'  => $this->getElement()->getName() . '-label'));
+                                         'id'  => $this->getElement()->getId() . '-label'));
 
             $label = $decorator->render($label);
         }
