Index: Action.php
===================================================================
--- Action.php	(revision 10539)
+++ Action.php	(working copy)
@@ -373,6 +373,16 @@
     }
 
     /**
+     * Return the Helper Broker object
+     *
+     * @return Zend_Controller_Action_HelperBroker
+     */
+    public function getHelperBroker()
+    {
+        return $this->_helper;
+    }
+
+    /**
      * Get a helper by name
      *
      * @param  string $helperName
