Index: tests/Zend/Application/ApplicationTest.php
===================================================================
--- tests/Zend/Application/ApplicationTest.php	(revision 18232)
+++ tests/Zend/Application/ApplicationTest.php	(working copy)
@@ -414,6 +414,23 @@
         $actual     = $autoloader->getZfPath();
         $this->assertContains($latest, $actual);
     }
+
+    /**
+     * @group ZF-6618
+     */
+    public function testCanExecuteBoostrapResourceViaApplicationInstanceBootstrapMethod() {
+        $application = new Zend_Application('testing', array(
+            'bootstrap' => array(
+                'path' => dirname(__FILE__) . '/_files/ZfAppBootstrap.php',
+                'class' => 'ZfAppBootstrap'
+                )
+            )
+        );
+        $application->bootstrap('foo');
+
+        $this->assertEquals(1, $application->getBootstrap()->fooExecuted);
+        $this->assertEquals(0, $application->getBootstrap()->barExecuted);
+    }
 }
 
 if (PHPUnit_MAIN_METHOD == 'Zend_Application_ApplicationTest::main') {
Index: library/Zend/Application.php
===================================================================
--- library/Zend/Application.php	(revision 18232)
+++ library/Zend/Application.php	(working copy)
@@ -339,11 +339,12 @@
     /**
      * Bootstrap application
      *
+     * @param  null|string|array $resource
      * @return Zend_Application
      */
-    public function bootstrap()
+    public function bootstrap($resource = null)
     {
-        $this->getBootstrap()->bootstrap();
+        $this->getBootstrap()->bootstrap($resource);
         return $this;
     }
 
