Index: Element.php
===================================================================
--- Element.php	(revision 149)
+++ Element.php	(working copy)
@@ -1106,6 +1106,26 @@
                 'breakChainOnFailure' => $breakChainOnFailure,
                 'options'             => $options,
             );
+        } elseif (is_array($validator)) {
+            foreach ($validator as $name => $spec) {
+                break;
+            }
+            if (is_numeric($name)) {
+                require_once 'Zend/Form/Exception.php';
+                throw new Zend_Form_Exception('Invalid alias provided to addValidator; must be alphanumeric string');
+            }
+            if (is_string($spec)) {
+                $validator = array(
+                    'validator' => $spec,
+                    'breakChainOnFailure' => $breakChainOnFailure,
+                    'options'   => $options,
+                );
+            } elseif ($spec instanceof Zend_Validate_Interface) {
+                $validator = $spec;
+                if (!isset($validator->zfBreakChainOnFailure)) {
+                    $validator->zfBreakChainOnFailure = $breakChainOnFailure;
+                }
+            }
         } else {
             // require_once 'Zend/Form/Exception.php';
             throw new Zend_Form_Exception('Invalid validator provided to addValidator; must be string or Zend_Validate_Interface');
