--- Date.php    (revision 2710)
+++ Date.php    (working copy)
@@ -165,7 +165,7 @@
         }

         // set datepart
-        if ($part !== null or (!is_numeric($date))) {
+        if (($part !== null && $part !== Zend_Date::TIMESTAMP) or (!is_numeric($date))) {
             $this->_Date = new Zend_Date_DateObject(0);
             $this->set($date, $part, $gmt, $this->_Locale);
         } else {
@@ -212,7 +212,7 @@

         $return = $this->_calcdetail($calc, $stamp, Zend_Date::TIMESTAMP, false, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, false, $locale);
+            return $this;
         }
         return $return;
     }
@@ -2293,7 +2293,7 @@
         }
         $return = $this->_calcdetail($calc, $time, Zend_Date::TIME_MEDIUM, true, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -2431,7 +2431,7 @@
         }
         $return = $this->_calcdetail($calc, $date, Zend_Date::DATE_MEDIUM, true, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -2555,7 +2555,7 @@
         }
         $return = $this->_calcdetail($calc, $iso, Zend_Date::ISO_8601, true, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -2677,7 +2677,7 @@

         $return = $this->_calcdetail($calc, $arpa, Zend_Date::RFC_822, true, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, false, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3024,7 +3024,7 @@
         }
         $return = $this->_calcdetail($calc, $year, Zend_Date::YEAR, true, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3181,7 +3181,7 @@
         }
         $return = $this->_calcdetail($calc, $found, Zend_Date::MONTH_DIGIT, true, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3323,7 +3323,7 @@
         }
         $return = $this->_calcdetail($calc, $day, $type, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3470,7 +3470,7 @@
         }
         $return = $this->_calcdetail($calc, $weekday, $type, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3600,7 +3600,7 @@

         $return = $this->_calcdetail($calc, $day, Zend_Date::DAY_OF_YEAR, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3721,7 +3721,7 @@

         $return = $this->_calcdetail($calc, $hour, Zend_Date::HOUR_SHORT, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3842,7 +3842,7 @@

         $return = $this->_calcdetail($calc, $minute, Zend_Date::MINUTE_SHORT, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -3963,7 +3963,7 @@

         $return = $this->_calcdetail($calc, $second, Zend_Date::SECOND_SHORT, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, true, $locale);
+            return $this;
         }
         return $return;
     }
@@ -4265,7 +4265,7 @@

         $return = $this->_calcdetail($calc, $week, Zend_Date::WEEK, $gmt, $locale);
         if ($calc != 'cmp') {
-            return new Zend_Date($this->_Date->getTimestamp(), Zend_Date::TIMESTAMP, false, $locale);
+            return $this;
         }
         return $return;
     }
