Index: C:/Documents and Settings/MARCOS SANCHEZ/workspace/ZendFramework/library/Zend/Controller/Front.php
===================================================================
--- C:/Documents and Settings/MARCOS SANCHEZ/workspace/ZendFramework/library/Zend/Controller/Front.php	(revision 7256)
+++ C:/Documents and Settings/MARCOS SANCHEZ/workspace/ZendFramework/library/Zend/Controller/Front.php	(working copy)
@@ -230,7 +230,7 @@
      * to the directory specified.
      *
      * @param string $directory
-     * @param string $module Optional argument; module with which to associate directory. If none provided, assumes 'defualt'
+     * @param string $module Optional argument; module with which to associate directory. If none provided, assumes 'default'
      * @return Zend_Controller_Front
      * @throws Zend_Controller_Exception if directory not found or readable
      */
