--- library/Zend/Loader/PluginLoader.php	(revision 7528)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -302,7 +302,7 @@
                 $classFile = str_replace('_', DIRECTORY_SEPARATOR, $name) . '.php';
                 $className = $prefix . $name;
                                 
-                if (class_exists($className)) {
+                if (class_exists($className, false)) {
                     $found = true;
                     break 2;
                 }
