Index: library/Zend/Application/Resource/Modules.php
===================================================================
--- library/Zend/Application/Resource/Modules.php	(revision 24218)
+++ library/Zend/Application/Resource/Modules.php	(working copy)
@@ -103,13 +103,13 @@
                 // resource, don't re-execute.
                 continue;
             }
-            
+
             $bootstraps[$module] = $bootstrapClass;
         }
 
         return $this->_bootstraps = $this->bootstrapBootstraps($bootstraps);
     }
-    
+
     /*
      * Bootstraps the bootstraps found. Allows for easy extension.
      * @param array $bootstraps Array containing the bootstraps to instantiate
@@ -117,15 +117,14 @@
     protected function bootstrapBootstraps($bootstraps)
     {
         $bootstrap = $this->getBootstrap();
-        $out = array();
-        
+        $out = new ArrayObject(array(), ArrayObject::ARRAY_AS_PROPS);
+
         foreach($bootstraps as $module => $bootstrapClass) {
             $moduleBootstrap = new $bootstrapClass($bootstrap);
             $moduleBootstrap->bootstrap();
-            $this->_bootstraps[$module] = $moduleBootstrap;
             $out[$module] = $moduleBootstrap;
         }
-        
+
         return $out;
     }
 
