Index: library/Zend/Validate/NotEmpty.php
===================================================================
--- library/Zend/Validate/NotEmpty.php	(revision 20167)
+++ library/Zend/Validate/NotEmpty.php	(working copy)
@@ -164,7 +164,7 @@
         if ($type >= self::SPACE) {
             $type -= self::SPACE;
             if (is_string($value) && (preg_match('/^\s+$/s', $value))) {
-                $this->_error(self::INVALID);
+                $this->_error(self::IS_EMPTY);
                 return false;
             }
         }
@@ -182,7 +182,7 @@
         if ($type >= self::EMPTY_ARRAY) {
             $type -= self::EMPTY_ARRAY;
             if (is_array($value) && ($value == array())) {
-                $this->_error(self::INVALID);
+                $this->_error(self::IS_EMPTY);
                 return false;
             }
         }
@@ -191,7 +191,7 @@
         if ($type >= self::ZERO) {
             $type -= self::ZERO;
             if (is_string($value) && ($value == '0')) {
-                $this->_error(self::INVALID);
+                $this->_error(self::IS_EMPTY);
                 return false;
             }
         }
@@ -200,7 +200,7 @@
         if ($type >= self::STRING) {
             $type -= self::STRING;
             if (is_string($value) && ($value == '')) {
-                $this->_error(self::INVALID);
+                $this->_error(self::IS_EMPTY);
                 return false;
             }
         }
@@ -209,7 +209,7 @@
         if ($type >= self::FLOAT) {
             $type -= self::FLOAT;
             if (is_float($value) && ($value == 0.0)) {
-                $this->_error(self::INVALID);
+                $this->_error(self::IS_EMPTY);
                 return false;
             }
         }
@@ -218,7 +218,7 @@
         if ($type >= self::INTEGER) {
             $type -= self::INTEGER;
             if (is_int($value) && ($value == 0)) {
-                $this->_error(self::INVALID);
+                $this->_error(self::IS_EMPTY);
                 return false;
             }
         }
