Index: Form/Decorator/Label.php
===================================================================
--- Form/Decorator/Label.php	(Revision 22516)
+++ Form/Decorator/Label.php	(Arbeitskopie)
@@ -257,10 +257,6 @@
             return '';
         }
 
-        if (null !== ($translator = $element->getTranslator())) {
-            $label = $translator->translate($label);
-        }
-
         $optPrefix = $this->getOptPrefix();
         $optSuffix = $this->getOptSuffix();
         $reqPrefix = $this->getReqPrefix();
Index: Form/Element/Submit.php
===================================================================
--- Form/Element/Submit.php	(Revision 22516)
+++ Form/Element/Submit.php	(Arbeitskopie)
@@ -77,10 +77,6 @@
             $value = $this->getName();
         }
 
-        if (null !== ($translator = $this->getTranslator())) {
-            return $translator->translate($value);
-        }
-
         return $value;
     }
 
