--- tests/Zend/Dojo/Form/Element/EditorTest.php	(revision 23864)
+++ tests/Zend/Dojo/Form/Element/EditorTest.php	(working copy)
@@ -262,6 +262,28 @@
         $this->assertEquals($this->element->getDijitParam('updateInterval'), $this->element->getUpdateInterval());
         $this->assertEquals(300, $this->element->getUpdateInterval());
     }
+    
+    public function testMinHeightCanBeSetToPixels()
+    {
+        $this->element->setMinHeight('250px');
+        $this->assertEquals($this->element->getDijitParam('minHeight'), $this->element->getMinHeight());
+        $this->assertEquals('250px', $this->element->getMinHeight());
+    }
+    
+    public function testMinHeightCanBeSetToPercentage()
+    {
+        $this->element->setMinHeight('50%');
+        $this->assertEquals($this->element->getDijitParam('minHeight'), $this->element->getMinHeight());
+        $this->assertEquals('50%', $this->element->getMinHeight());
+    }
+    
+    public function testMinHeightDefaultMeasurementIsEm()
+    {
+        $this->element->setMinHeight('10');
+        $this->assertEquals($this->element->getDijitParam('minHeight'), $this->element->getMinHeight());
+        $this->assertEquals('10em', $this->element->getMinHeight());
+    }
+    
 }
 
 // Call Zend_Dojo_Form_Element_EditorTest::main() if this source file is executed directly.
