Index: Zend/Controller/Request/HttpTest.php
===================================================================
--- Zend/Controller/Request/HttpTest.php	(revision 12339)
+++ Zend/Controller/Request/HttpTest.php	(working copy)
@@ -366,6 +366,20 @@
     {
         $this->assertSame('', $this->_request->getBaseUrl());
     }
+    
+    /**
+     * @descr Tests if an empty string gets returned when no basepath is set on the request.
+     * This is important on windows, where before this fix '\' was returned instead of an empty string.
+     * The fix relies on $_SERVER['WINDIR'] being set on windows and not on other OS's. Another superglobal
+     * that might be useful to this fix is $_SERVER['SERVER_SOFTWARE'], which can be scanned for 'win' to 
+     * determine the os.
+     * @group ZF-4810
+     *
+     */
+    public function testGetBasePathIsEmptyStringIfNoneSet(){
+		$request = new Zend_Controller_Request_Http();
+		$this->assertEquals('', $request->getBasePath());    	
+    }
  
     public function testSetBaseUrl()
     {
