Index: tests/Zend/Form/Element/CaptchaTest.php
===================================================================
--- tests/Zend/Form/Element/CaptchaTest.php	(revision 25179)
+++ tests/Zend/Form/Element/CaptchaTest.php	(working copy)
@@ -112,17 +112,17 @@
         $element = new Zend_Form_Element_Captcha(
             'foo',
             array(
-                'prefixPath' => array(
-                    'prefix' => 'Zend_Form_Element_CaptchaTest',
-                    'path'   => dirname(__FILE__) . '/_files',
-                ),
-                'captcha' => 'Foo',
+                 'prefixPath' => array(
+                     'prefix' => 'Zend_Form_Element_CaptchaTest',
+                     'path'   => dirname(__FILE__) . '/_files',
+                 ),
+                 'captcha'    => 'Foo',
             )
         );
 
-        $this->assertType(
-            'Zend_Form_Element_CaptchaTest_Captcha_Foo',
-            $element->getCaptcha()
+        $this->assertTrue(
+            $element->getCaptcha() instanceof
+                Zend_Form_Element_CaptchaTest_Captcha_Foo
         );
     }
 
Index: tests/Zend/Form/Element/FileTest.php
===================================================================
--- tests/Zend/Form/Element/FileTest.php	(revision 25179)
+++ tests/Zend/Form/Element/FileTest.php	(working copy)
@@ -498,13 +498,21 @@
     public function testElementShouldAllowAdapterWithBackslahes()
     {
         if (version_compare(PHP_VERSION, '5.3.0', '<')) {
-            $this->markTestSkipped(__CLASS__ . '::' . __METHOD__ . ' requires PHP 5.3.0 or greater');
+            $this->markTestSkipped(
+                __CLASS__ . '::' . __METHOD__ . ' requires PHP 5.3.0 or greater'
+            );
             return;
         }
-        $this->element->addPrefixPath('Zend\Form\Element\FileTest\Adapter', dirname(__FILE__) . '/_files/TransferAdapter', 'transfer_adapter');
+        $this->element->addPrefixPath(
+            'Zend\Form\Element\FileTest\Adapter',
+            dirname(__FILE__) . '/_files/TransferAdapter',
+            'transfer_adapter'
+        );
         $this->element->setTransferAdapter('Bar');
         $test = $this->element->getTransferAdapter();
-        $this->assertType('\Zend\Form\Element\FileTest\Adapter\Bar', $test);
+        $this->assertTrue(
+            $test instanceof \Zend\Form\Element\FileTest\Adapter\Bar
+        );
     }
 }
 
Index: tests/Zend/Form/Element/SelectTest.php
===================================================================
--- tests/Zend/Form/Element/SelectTest.php	(revision 25179)
+++ tests/Zend/Form/Element/SelectTest.php	(working copy)
@@ -251,6 +251,26 @@
     }
 
     /**
+     * @group ZF-8452
+     */
+    public function testRenderingAsArray()
+    {
+        $this->element->addMultiOption('bar', 'Bar')
+                      ->setIsArray(true)
+                      ->setDecorators(array('ViewHelper'));
+
+        $actual   = $this->element->render($this->getView());
+        $expected = PHP_EOL
+                  . '<select name="foo[]" id="foo">'
+                  . PHP_EOL
+                  . '    <option value="bar" label="Bar">Bar</option>'
+                  . PHP_EOL
+                  . '</select>';
+
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
      * Used by test methods susceptible to ZF-2794, marks a test as incomplete
      *
      * @link   http://framework.zend.com/issues/browse/ZF-2794
Index: library/Zend/Form/Element/Select.php
===================================================================
--- library/Zend/Form/Element/Select.php	(revision 25179)
+++ library/Zend/Form/Element/Select.php	(working copy)
@@ -35,6 +35,12 @@
 class Zend_Form_Element_Select extends Zend_Form_Element_Multi
 {
     /**
+     * 'multiple' attribute
+     * @var string
+     */
+    public $multiple = false;
+
+    /**
      * Use formSelect view helper by default
      * @var string
      */
