Index: library/Zend/Service/Rackspace/Abstract.php
===================================================================
--- library/Zend/Service/Rackspace/Abstract.php	(revision 25298)
+++ library/Zend/Service/Rackspace/Abstract.php	(working copy)
@@ -329,6 +329,11 @@
     {
         $client = $this->getHttpClient();
         $client->resetParameters(true);
+        if ($method == 'PUT' && empty($body)) {
+            // if left at NULL a PUT request will always have 
+            // Content-Type: x-url-form-encoded, which breaks copyObject()
+            $client->setEncType(''); 
+        }
         if (empty($headers[self::AUTHUSER_HEADER])) {
             $headers[self::AUTHTOKEN]= $this->getToken();
         } 
