--- library/Zend/Form/Element/Submit.php	(revision 24057)
+++ library/Zend/Form/Element/Submit.php	(working copy)
@@ -71,16 +71,12 @@
      */
     public function getLabel()
     {
-        $value = parent::getLabel();
+        $value = parent::getLabel(); // note that label translated by the parent
 
         if (null === $value) {
             $value = $this->getName();
         }
 
-        if (null !== ($translator = $this->getTranslator())) {
-            return $translator->translate($value);
-        }
-
         return $value;
     }
 
Index: library/Zend/Form/Decorator/Label.php
===================================================================
--- library/Zend/Form/Decorator/Label.php	(revision 24057)
+++ library/Zend/Form/Decorator/Label.php	(working copy)
@@ -300,10 +300,6 @@
             return '';
         }
 
-        if (null !== ($translator = $element->getTranslator())) {
-            $label = $translator->translate($label);
-        }
-
         $optPrefix = $this->getOptPrefix();
         $optSuffix = $this->getOptSuffix();
         $reqPrefix = $this->getReqPrefix();
