Index: Abstract.php
===================================================================
--- Abstract.php	(revision 6818)
+++ Abstract.php	(working copy)
@@ -516,6 +516,15 @@
         return $this;
     }
 
+	/**
+     * Refreshes properties from the database.
+     *
+     * @return void
+     */
+	public function refresh() {
+		return $this->_refresh();
+	}
+
     /**
      * Retrieves an instance of the parent table.
      *
