Index: Zend/Db/Table/Row.php
===================================================================
--- Zend/Db/Table/Row.php	(revision 884)
+++ Zend/Db/Table/Row.php	(revision 885)
@@ -94,7 +94,11 @@
     {
         $under = array_search($camel, $this->_info['cols']);
         if ($under) {
-            return $this->_data[$under];
+            if (isset($this->_data[$under])) {
+                return $this->_data[$under];
+            } else {
+                return null;
+            }
         } else {
             throw new Zend_Db_Table_Row_Exception("column '$camel' not in row");
         }
@@ -188,7 +192,7 @@
     public function setFromArray($data)
     {
         foreach ($data as $key => $val) {
-            if (array_key_exists($key, $this->_data)) {
+            if (array_key_exists($key, $this->_info['cols'])) {
                 $this->_data[$key] = $val;
             }
         }
