Index: library/Zend/Validate/File/FilesSize.php
===================================================================
--- library/Zend/Validate/File/FilesSize.php	(revision 20430)
+++ library/Zend/Validate/File/FilesSize.php	(working copy)
@@ -135,6 +135,7 @@
                     $this->_throw($file, self::TOO_BIG);
                     $this->setMax($max);
                 } else {
+                    $this->_size = $size;
                     $this->_throw($file, self::TOO_BIG);
                 }
             }
@@ -147,6 +148,7 @@
                 $this->_throw($file, self::TOO_SMALL);
                 $this->setMin($min);
             } else {
+                $this->_size = $size;
                 $this->_throw($file, self::TOO_SMALL);
             }
         }
Index: library/Zend/Validate/File/Size.php
===================================================================
--- library/Zend/Validate/File/Size.php	(revision 20430)
+++ library/Zend/Validate/File/Size.php	(working copy)
@@ -293,6 +293,7 @@
                 $this->_min  = $min;
                 $this->_size = $size;
             } else {
+                $this->_size = $size;
                 $this->_throw($file, self::TOO_SMALL);
             }
         }
@@ -306,6 +307,7 @@
                 $this->_max  = $max;
                 $this->_size = $size;
             } else {
+                $this->_size = $size;
                 $this->_throw($file, self::TOO_BIG);
             }
         }
