Index: ../library/Zend/Soap/Client.php
===================================================================
--- ../library/Zend/Soap/Client.php	(revision 11076)
+++ ../library/Zend/Soap/Client.php	(working copy)
@@ -305,7 +305,7 @@
     {
         foreach ($classmap as $type => $class) {
             if (!class_exists($class)) {
-                throw new Zend_Soap_Class_Exception('Invalid class in class map');
+                throw new Zend_Soap_Client_Exception('Invalid class in class map');
             }
         }
 
