Index: Mail.php
===================================================================
--- Mail.php	(revision 12598)
+++ Mail.php	(working copy)
@@ -390,9 +390,7 @@
       if (Zend_Mime::isPrintable($value)) {
           return $value;
       } else {
-          $quotedValue = Zend_Mime::encodeQuotedPrintable($value);
-          $quotedValue = str_replace(array('?', ' ', '_'), array('=3F', '=20', '=5F'), $quotedValue);
-          return '=?' . $this->_charset . '?Q?' . $quotedValue . '?=';
+		  return '=?' . $this->_charset . '?B?' . Zend_Mime::encodeBase64($value) . '?=';
       }
     }
 
