Index: YouTubeOnlineTest.php
===================================================================
--- YouTubeOnlineTest.php	(revision 23703)
+++ YouTubeOnlineTest.php	(working copy)
@@ -144,13 +144,13 @@
         $service = Zend_Gdata_YouTube::AUTH_SERVICE_NAME;
         $authenticationURL= 'https://www.google.com/youtube/accounts/ClientLogin';
         $httpClient = Zend_Gdata_ClientLogin::getHttpClient(
-                                          $username = $this->user,
-                                          $password = $this->pass,
-                                          $service = $service,
-                                          $client = null,
-                                          $source = 'Google-UnitTests-1.0',
-                                          $loginToken = null,
-                                          $loginCaptcha = null,
+                                          $this->user,
+                                          $this->pass,
+                                          $service,
+                                          null, // Client
+                                          'Google-UnitTests-1.0',
+                                          null, // Login Token
+                                          null, // Login CAPTCHA
                                           $authenticationURL);
 
         $this->gdata = new Zend_Gdata_YouTube($httpClient,
@@ -311,13 +311,13 @@
         $authenticationURL =
             'https://www.google.com/youtube/accounts/ClientLogin';
         $httpClient = Zend_Gdata_ClientLogin::getHttpClient(
-                                          $username = $this->user,
-                                          $password = $this->pass,
-                                          $service = $service,
-                                          $client = null,
+                                          $this->user,
+                                          $this->pass,
+                                          $service,
+                                          null, // Client
                                           $source = 'Google-UnitTests-1.0',
-                                          $loginToken = null,
-                                          $loginCaptcha = null,
+                                          null, // Login Token
+                                          null, // Login CAPTCHA
                                           $authenticationURL);
 
         $yt = new Zend_Gdata_YouTube(
@@ -394,13 +394,13 @@
         $authenticationURL =
             'https://www.google.com/youtube/accounts/ClientLogin';
         $httpClient = Zend_Gdata_ClientLogin::getHttpClient(
-                                          $username = $this->user,
-                                          $password = $this->pass,
-                                          $service = $service,
-                                          $client = null,
+                                          $this->user,
+                                          $this->pass,
+                                          $service,
+                                          null, // Client
                                           $source = 'Google-UnitTests-1.0',
-                                          $loginToken = null,
-                                          $loginCaptcha = null,
+                                          null, // Login Token
+                                          null, // Login CAPTCHA
                                           $authenticationURL);
 
         $yt = new Zend_Gdata_YouTube(
@@ -489,13 +489,13 @@
         $authenticationURL =
             'https://www.google.com/youtube/accounts/ClientLogin';
         $httpClient = Zend_Gdata_ClientLogin::getHttpClient(
-                                          $username = $this->user,
-                                          $password = $this->pass,
-                                          $service = $service,
-                                          $client = null,
+                                          $this->user,
+                                          $this->pass,
+                                          $service,
+                                          null, // Client
                                           $source = 'Google-UnitTests-1.0',
-                                          $loginToken = null,
-                                          $loginCaptcha = null,
+                                          null, // Login Token
+                                          null, // Login CAPTCHA
                                           $authenticationURL);
 
         $yt = new Zend_Gdata_YouTube(
@@ -584,13 +584,13 @@
         $authenticationURL =
             'https://www.google.com/youtube/accounts/ClientLogin';
         $httpClient = Zend_Gdata_ClientLogin::getHttpClient(
-                                          $username = $this->user,
-                                          $password = $this->pass,
-                                          $service = $service,
-                                          $client = null,
+                                          $this->user,
+                                          $this->pass,
+                                          $service,
+                                          null, // Client
                                           $source = 'Google-UnitTests-1.0',
-                                          $loginToken = null,
-                                          $loginCaptcha = null,
+                                          null, // Login Token
+                                          null, // Login CAPTCHA
                                           $authenticationURL);
 
         $yt = new Zend_Gdata_YouTube(
@@ -682,13 +682,13 @@
         $authenticationURL =
             'https://www.google.com/youtube/accounts/ClientLogin';
         $httpClient = Zend_Gdata_ClientLogin::getHttpClient(
-                                          $username = $this->user,
-                                          $password = $this->pass,
-                                          $service = $service,
-                                          $client = null,
+                                          $this->user,
+                                          $this->pass,
+                                          $service,
+                                          null, // Client
                                           $source = 'Google-UnitTests-1.0',
-                                          $loginToken = null,
-                                          $loginCaptcha = null,
+                                          null, // Login Token
+                                          null, // Login CAPTCHA
                                           $authenticationURL);
 
         $yt = new Zend_Gdata_YouTube($httpClient, 'Google-UnitTests-1.0', $clientId, $developerKey);
