--- library/Zend/Application/Resource/View.php  (revision 23872)
+++ library/Zend/Application/Resource/View.php  (working copy)
@@ -77,7 +77,12 @@
             }
             if (isset($options['contentType'])) {
                 $this->_view->headMeta()->appendHttpEquiv('Content-Type', $options['contentType']);
-            }
+           }
+           if (isset($options['assign']) && is_array($options['assign'])) {
+               foreach ($options['assign'] as $key => $value) {
+                   $this->_view->assign($key, $value);
+               }
+           }
         }
         return $this->_view;
     }
