Index: Zend/Translate/Adapter.php
===================================================================
--- Zend/Translate/Adapter.php	(revision 19619)
+++ Zend/Translate/Adapter.php	(revision 19620)
@@ -564,7 +564,10 @@
 
             $rule = Zend_Translate_Plural::getPlural($number, $locale);
             if (isset($this->_translate[$locale][$plural[0]][$rule])) {
-                return $this->_translate[$locale][$plural[0]][$rule];
+                if (isset($this->_translate[$locale][$plural[0]]) && !is_array($this->_translate[$locale][$plural[0]])) 
+                    return $this->_translate[$locale][$plural[0]];
+                else 
+                    return $this->_translate[$locale][$plural[0]][$rule];
             }
         } else if (strlen($locale) != 2) {
             // faster than creating a new locale and separate the leading part
@@ -758,4 +761,4 @@
      * @return string
      */
     abstract public function toString();
-}
\ No newline at end of file
+}
