Index: Http.php
===================================================================
--- Http.php	(revision 2914)
+++ Http.php	(working copy)
@@ -319,10 +319,10 @@
     public function setRequestUri($requestUri = null) 
     { 
         if ($requestUri === null) { 
-            if (isset($_SERVER['REQUEST_URI'])) { 
+            if (isset($_SERVER['HTTP_X_REWRITE_URL'])) { 
+                $requestUri = $_SERVER['HTTP_X_REWRITE_URL']; 
+            } elseif (isset($_SERVER['REQUEST_URI'])) { 
                 $requestUri = $_SERVER['REQUEST_URI']; 
-            } elseif (isset($_SERVER['HTTP_X_REWRITE_URL'])) { 
-                $requestUri = $_SERVER['HTTP_X_REWRITE_URL']; 
             } elseif (isset($_SERVER['ORIG_PATH_INFO'])) { // IIS 5.0, PHP as CGI
                 $requestUri = $_SERVER['ORIG_PATH_INFO'];
                 if (!empty($_SERVER['QUERY_STRING'])) {
