Index: Zend_Framework/Zend/XmlRpc/Client.php
===================================================================
--- Zend_Framework/Zend/XmlRpc/Client.php	(Revision 2796)
+++ Zend_Framework/Zend/XmlRpc/Client.php	(Arbeitskopie)
@@ -347,7 +347,7 @@
         $http->setUri($this->_serverAddress);
         // Set the content-type header as text/xml
         // What if the given HTTP client already has headres ? it shouldn't override them
-        $http->setHeaders(array('Content-Type: text/xml; charset=iso-8859-1'));
+        $http->setHeaders(array('Content-Type: text/xml; charset=iso-8859-1', 'User-Agent: Zend_Xmlrpc_Client'));
 
         $response = $http->post($request_data);
         /* @var $response Zend_Http_Response */
