Index: library/Zend/OpenId/Consumer/Storage/File.php
===================================================================
--- library/Zend/OpenId/Consumer/Storage/File.php	(revision 24790)
+++ library/Zend/OpenId/Consumer/Storage/File.php	(working copy)
@@ -141,14 +141,6 @@
             }
             $data = serialize(array($url, $handle, $macFunc, $secret, $expires));
             fwrite($f, $data);
-            if (function_exists('symlink')) {
-                @unlink($name2);
-                if (symlink($name1, $name2)) {
-                    fclose($f);
-                    fclose($lock);
-                    return true;
-                }
-            }
             $f2 = @fopen($name2, 'w+');
             if ($f2) {
                 fwrite($f2, $data);
