Index: Log.php
===================================================================
--- Log.php	(revision 933)
+++ Log.php	(working copy)
@@ -396,6 +396,16 @@
             }
         }
 
+        /**
+         * If the supplied logName is actually an array of logNames, then
+         * call the function recursively to post to all the logs.
+         */
+        if (is_array($logName)) {
+            foreach ($logName as $l) {
+                self::log($message, $level, $fields, $l);
+            }
+            return true;
+        }
 
         /**
          * For any fields that were not specified, use the defaults.
@@ -408,18 +418,6 @@
             }
         }
 
-
-        /**
-         * If the supplied logName is actually an array of logNames, then
-         * call the function recursively to post to all the logs.
-         */
-        if (is_array($logName)) {
-            foreach ($logName as $l) {
-                self::log($message, $level, $fields, $l);
-            }
-            return true;
-        }
-
         // Write the message to the log if the current log level will accept it.
         /* @var $logger Zend_Log */
         $logger = self::_getInstance($logName);
