Index: FieldsetTest.php
--- FieldsetTest.php Ausgangspunkt (BASE)
+++ FieldsetTest.php Lokal geändert (ausgehend von LOCAL)
@@ -229,7 +229,22 @@
                                 ->render('content');
         $this->assertContains('<fieldset id="fieldset-id"', $html);
     }
+    
+    /**
+     * @group ZF-10803
+     */
+    public function testFormIdOverridesFieldsetId()
+    {
+        $form = new Zend_Form();
+        $form->setAttrib('id', 'form-id')
+             ->setView($this->getView());
+        
+        $html = $this->decorator->setElement($form)
+                                ->render('content');
+        
+        $this->assertContains('<fieldset id="fieldset-form-id"', $html);
 }
+}
 
 // Call Zend_Form_Decorator_FieldsetTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == "Zend_Form_Decorator_FieldsetTest::main") {
