Index: library/Zend/Gdata/App.php
===================================================================
--- library/Zend/Gdata/App.php	(revision 25179)
+++ library/Zend/Gdata/App.php	(working copy)
@@ -545,14 +545,20 @@
         }
 
         // Set an If-Match header if:
-        //   - This isn't a DELETE
+        //   - This is a DELETE - See ZF-10194
         //   - If this isn't a GET, the Etag isn't weak
         //   - A similar header (If-Match/If-None-Match) hasn't already been
         //     set.
-        if ($method != 'DELETE' && (
+        if ($method == 'DELETE') {
+            if ($rawData != null && isset($rawData->etag) && $rawData->etag != '') {
+                $headers['If-Match'] = $rawData->etag;
+            } else {
+                $headers['If-Match'] = '*';
+            }
+        }elseif (
                 !array_key_exists('If-Match', $headers) &&
                 !array_key_exists('If-None-Match', $headers)
-                ) ) {
+                ) {
             $allowWeak = $method == 'GET';
             if ($ifMatchHeader = $this->generateIfMatchHeaderData(
                     $data, $allowWeak)) {
