Index: library/ZendFramework-1.9.3PL1/Zend/Pdf/Resource/Image/Jpeg.php
===================================================================
--- library/ZendFramework-1.9.3PL1/Zend/Pdf/Resource/Image/Jpeg.php	(revision 5321)
+++ library/ZendFramework-1.9.3PL1/Zend/Pdf/Resource/Image/Jpeg.php	(working copy)
@@ -58,7 +58,7 @@
         }
 
         $gd_options = gd_info();
-        if (!$gd_options['JPG Support'] ) {
+        if (!array_key_exists('JPG Support', $gd_options) && !array_key_exists('JPEG Support', $gd_options)) {
             require_once 'Zend/Pdf/Exception.php';
             throw new Zend_Pdf_Exception('JPG support is not configured properly.');
         }
