--- C:/Users/Christian/AppData/Local/Temp/3/StringTrim.p-revBASE.svn002.tmp.php	Do Sep 24 10:29:46 2009
+++ E:/SVNROOT/ZendFramework-trunk/library/Zend/Filter/StringTrim.php	Do Sep 24 10:28:08 2009
@@ -113,8 +113,8 @@
     protected function _unicodeTrim($value, $charlist = '\\\\s')
     {
         $chars = preg_replace(
-            array( '/[\^\-\]\\\]/S', '/\\\{4}/S' ),
-            array( '\\\\\\0', '\\' ),
+            array( '/[\^\-\]\\\]/S', '/\\\{4}/S', '/\//' ),
+            array( '\\\\\\0', '\\', '\/' ),
             $charlist
         );
 
