Index: Zend/Form/ElementMultiCheckbox.php
--- Zend/Form/ElementMultiCheckbox.php Base (BASE)
+++ Zend/Form/ElementMultiCheckbox.php Locally Modified (Based On LOCAL)
@@ -49,4 +49,23 @@
      * @var bool
      */
     protected $_isArray = true;
+
+
+    /**
+     * Load default decorators
+     *
+     * Disables "for" attribute of label if label decorator enabled.
+     *
+     * @return Zend_Form_Element_Radio
+     */
+    public function loadDefaultDecorators()
+    {
+        if ($this->loadDefaultDecoratorsIsDisabled()) {
+            return $this;
+        }
+        parent::loadDefaultDecorators();
+        $this->addDecorator('Label', array('tag' => 'dt',
+                                           'disableFor' => true));
+        return $this;
+    }
