--- trunk/library/Zend/Form.php	(revision 11372)
+++ trunk/library/Zend/Form.php	(working copy)
@@ -303,7 +303,7 @@
 
         if (isset($options['displayGroupPrefixPath'])) {                          
             $this->addDisplayGroupPrefixPaths($options['displayGroupPrefixPath']);      
-            unset($options['elementPrefixPath']);                             
+            unset($options['displayGroupPrefixPath']);                             
         }
 
         if (isset($options['elements'])) {
