Index: Form.php
===================================================================
--- Form.php	(revision 9797)
+++ Form.php	(working copy)
@@ -979,7 +979,7 @@
      * Remove element
      * 
      * @param  string $name 
-     * @return boolean
+     * @return Zend_Form
      */
     public function removeElement($name)
     {
@@ -996,10 +996,10 @@
                     }
                 }
             }
-            return true;
-        }
-
-        return false;
+        } else {
+		    throw new Zend_Form_Exception("Zend_Form cannot remove non-existent element: '$name'");
+		}
+		return $this;
     }
 
     /**
