--- library/Zend/File/Transfer/Adapter/Abstract.php.orig	2009-03-24 23:06:13.000000000 +0100
+++ library/Zend/File/Transfer/Adapter/Abstract.php	2009-05-29 04:26:47.000000000 +0200
@@ -1377,26 +1377,27 @@
             foreach ($files as $find) {
                 $found = array();
                 foreach ($this->_files as $file => $content) {
-                    if (!isset($content['name'])) {
-                        continue;
-                    }
-
-                    if (($content['name'] === $find) && isset($content['multifiles'])) {
-                        foreach ($content['multifiles'] as $multifile) {
-                            $found[] = $multifile;
+                    
+                    if (isset($content['name']) && $content['name'] === $find) {
+                        if (isset($content['multifiles'])) {
+                            
+                            foreach ($content['multifiles'] as $multifile) {
+                                $found[] = $multifile;
+                            }
+                            break;
+                            
+                        } else {
+                            
+                            $found[] = $file;
+                            break;
+                            
                         }
-                        break;
                     }
 
                     if ($file === $find) {
                         $found[] = $file;
                         break;
                     }
-
-                    if ($content['name'] === $find) {
-                        $found[] = $file;
-                        break;
-                    }
                 }
 
                 if (empty($found)) {
