Index: Date.php
===================================================================
--- Date.php	(revision 21278)
+++ Date.php	(working copy)
@@ -2312,6 +2312,11 @@
                         $day   = 1;
                         $year  = 1970;
                     }
+
+                    if (!isset($parsed['second'])) {
+                        $parsed['second'] = 0;
+                    }
+
                     return $this->_assign($calc, $this->mktime($parsed['hour'], $parsed['minute'], $parsed['second'], $month, $day, $year, true),
                                                  $this->mktime($hour,           $minute,           $second,           $month, $day, $year, true), false);
                 } catch (Zend_Locale_Exception $e) {
@@ -2796,6 +2801,9 @@
                     throw new Zend_Date_Exception($e->getMessage(), 0, $e);
                 }
             }
+            if (!isset($parsed['second'])) {
+                $parsed['second'] = 0;
+            }
             $time  = str_pad($parsed['hour'], 2, '0', STR_PAD_LEFT) . ":";
             $time .= str_pad($parsed['minute'], 2, '0', STR_PAD_LEFT) . ":";
             $time .= str_pad($parsed['second'], 2, '0', STR_PAD_LEFT);
