--- library/Zend/Db/Adapter/Pdo/Mssql.php	(revision 4754)
+++ library/Zend/Db/Adapter/Pdo/Mssql.php	(working copy)
@@ -122,26 +122,23 @@
     {
         $sql = "exec sp_columns @table_name = " . $this->quoteIdentifier($tableName);
         $stmt = $this->query($sql);
-        // @todo: change to FETCH_NUM
-        $result = $stmt->fetchAll(Zend_Db::FETCH_BOTH);
+        $result = $stmt->fetchAll(Zend_Db::FETCH_NUM);
 
-        // @todo: change to integers according to keys returned in FETCH_NUM result set
-        $column_name = 'column_name';
-        $table_name  = 'table_name';
-        $column_def  = 'column_def';
-        $type_name   = 'type_name';
-        $nullable    = 'nullable';
-        $length      = 'length';
-        $scale       = 'scale';
-        $precision   = 'precision';
+        $table_name      = 2;
+        $column_name     = 3;
+        $type_name       = 5;
+        $precision       = 6;
+        $length          = 7;
+        $scale           = 8;
+        $nullable        = 10;
+        $column_def      = 12;
+        $column_position = 16;
 
         $sql = "exec sp_pkeys @table_name = " . $this->quoteIdentifier($tableName);
         $stmt = $this->query($sql);
-        // @todo: change to FETCH_NUM
-        $primaryKeysResult = $stmt->fetchAll(Zend_Db::FETCH_BOTH);
+        $primaryKeysResult = $stmt->fetchAll(Zend_Db::FETCH_NUM);
 
-        // @todo: change to integers according to keys returned in FETCH_NUM result set
-        $pkey_column_name = 'coumn_name';
+        $pkey_column_name = 3;
 
         foreach ($primaryKeysResult as $row) {
             $primaryKeyColumn[$row[$pkey_column_name]] = true;
@@ -149,17 +146,16 @@
 
         $desc = array();
         foreach ($result as $key => $row) {
-            list($type, $rest) = explode(' ', $row[$type_name], 2);
+            list($type) = explode(' ', $row[$type_name], 2);
 
-            if (array_key_exists($primaryKeyColumn, $row[$column_name])) {
-                $is_primary = true;
-            }
-
+            $is_primary = array_key_exists($row[$column_name], $primaryKeyColumn);
+            $primary_position = $is_primary ? (int) $row[$column_position] : null;
+            
             $desc[$row[$column_name]] = array(
                 'SCHEMA_NAME'      => null, // @todo
                 'TABLE_NAME'       => $row[$table_name],
                 'COLUMN_NAME'      => $row[$column_name],
-                'COLUMN_POSITION'  => null, // @todo
+                'COLUMN_POSITION'  => (int) $row[$column_position],
                 'DATA_TYPE'        => $type,
                 'DEFAULT'          => $row[$column_def],
                 'NULLABLE'         => (bool) $row[$nullable],
@@ -168,7 +164,7 @@
                 'PRECISION'        => $row[$precision],
                 'UNSIGNED'         => null, // @todo
                 'PRIMARY'          => (bool) $is_primary,
-                'PRIMARY_POSITION' => null // @todo
+                'PRIMARY_POSITION' => $primary_position
             );
         }
         return $desc;
