Index: tests/Zend/Validate/FloatTest.php
===================================================================
--- tests/Zend/Validate/FloatTest.php	(revision 23552)
+++ tests/Zend/Validate/FloatTest.php	(working copy)
@@ -215,4 +215,59 @@
         $this->assertFalse($valid->isValid('1000,3'));
         $this->assertFalse($valid->isValid('1.000,3'));
     }
+    
+    /**
+     * @group ZF-10533
+     */
+    public function testPhpLocaleEnStringTypeHandlesCommaSeparatorFollowedByTwoDigitsCorrectly()
+    {
+        $valid = new Zend_Validate_Float(array('locale'=>'en_US'));
+        $this->assertFalse($valid->isValid('0,22'));        
+    }
+    
+    /**
+     * @group ZF-10533
+     */
+    public function testPhpLocaleEnStringTypeHandlesCommaSeparatorFollowedByThreeDigitsCorrectly()
+    {
+        $valid = new Zend_Validate_Float(array('locale'=>'en_US'));
+        $this->assertFalse($valid->isValid('0,333'));        
+    }
+    
+    /**
+     * @group ZF-10533
+     */
+    public function testPhpLocaleEnStringTypeHandlesCommaSeparatorFollowedByFourDigitsCorrectly()
+    {
+        $valid = new Zend_Validate_Float(array('locale'=>'en_US'));
+        $this->assertFalse($valid->isValid('0,4444'));        
+    }
+    
+    /**
+     * @group ZF-10533
+     */
+    public function testPhpLocaleEnStringTypeHandlesNonZeroNumberFollowedByCommaSeparatorFollowedByTwoDigitsCorrectly()
+    {
+        $valid = new Zend_Validate_Float(array('locale'=>'en_US'));
+        $this->assertFalse($valid->isValid('8,22'));        
+    }
+    
+    /**
+     * @group ZF-10533
+     */
+    public function testPhpLocaleEnStringTypeHandlesNonZeroNumberFollowedByCommaSeparatorFollowedByThreeDigitsCorrectly()
+    {
+        $valid = new Zend_Validate_Float(array('locale'=>'en_US'));
+        $this->assertFalse($valid->isValid('8,333'));        
+    }
+    
+    /**
+     * @group ZF-10533
+     */
+    public function testPhpLocaleEnStringTypeHandlesNonZeroNumberFollowedByCommaSeparatorFollowedByFourDigitsCorrectly()
+    {
+        $valid = new Zend_Validate_Float(array('locale'=>'en_US'));
+        $this->assertFalse($valid->isValid('8,4444'));        
+    }
+    
 }
