Index: tests/Zend/Validate/AllTests.php
===================================================================
--- tests/Zend/Validate/AllTests.php	(revision 22484)
+++ tests/Zend/Validate/AllTests.php	(working copy)
@@ -31,7 +31,9 @@
 require_once 'Zend/Validate/AlphaTest.php';
 require_once 'Zend/Validate/BarcodeTest.php';
 require_once 'Zend/Validate/BetweenTest.php';
+require_once 'Zend/Validate/CallbackTest.php';
 require_once 'Zend/Validate/CcnumTest.php';
+require_once 'Zend/Validate/CreditCardTest.php';
 require_once 'Zend/Validate/DateTest.php';
 require_once 'Zend/Validate/Db/AllTests.php';
 require_once 'Zend/Validate/DigitsTest.php';
@@ -41,6 +43,7 @@
 require_once 'Zend/Validate/GreaterThanTest.php';
 require_once 'Zend/Validate/HexTest.php';
 require_once 'Zend/Validate/HostnameTest.php';
+require_once 'Zend/Validate/IbanTest.php';
 require_once 'Zend/Validate/IdenticalTest.php';
 require_once 'Zend/Validate/InArrayTest.php';
 require_once 'Zend/Validate/IntTest.php';
@@ -49,6 +52,7 @@
 require_once 'Zend/Validate/LessThanTest.php';
 require_once 'Zend/Validate/MessageTest.php';
 require_once 'Zend/Validate/NotEmptyTest.php';
+require_once 'Zend/Validate/PostCodeTest.php';
 require_once 'Zend/Validate/RegexTest.php';
 require_once 'Zend/Validate/Sitemap/AllTests.php';
 require_once 'Zend/Validate/StringLengthTest.php';
@@ -87,7 +91,9 @@
         $suite->addTestSuite('Zend_Validate_AlphaTest');
         $suite->addTestSuite('Zend_Validate_BarcodeTest');
         $suite->addTestSuite('Zend_Validate_BetweenTest');
+        $suite->addTestSuite('Zend_Validate_CallbackTest');
         $suite->addTestSuite('Zend_Validate_CcnumTest');
+        $suite->addTestSuite('Zend_Validate_CreditCardTest');
         $suite->addTestSuite('Zend_Validate_DateTest');
         $suite->addTest(Zend_Validate_Db_AllTests::suite());
         $suite->addTestSuite('Zend_Validate_DigitsTest');
@@ -97,6 +103,7 @@
         $suite->addTestSuite('Zend_Validate_GreaterThanTest');
         $suite->addTestSuite('Zend_Validate_HexTest');
         $suite->addTestSuite('Zend_Validate_HostnameTest');
+        $suite->addTestSuite('Zend_Validate_IbanTest');
         $suite->addTestSuite('Zend_Validate_IdenticalTest');
         $suite->addTestSuite('Zend_Validate_InArrayTest');
         $suite->addTestSuite('Zend_Validate_IntTest');
@@ -105,6 +112,7 @@
         $suite->addTestSuite('Zend_Validate_LessThanTest');
         $suite->addTestSuite('Zend_Validate_MessageTest');
         $suite->addTestSuite('Zend_Validate_NotEmptyTest');
+        $suite->addTestSuite('Zend_Validate_PostCodeTest');
         $suite->addTestSuite('Zend_Validate_RegexTest');
         $suite->addTest(Zend_Validate_Sitemap_AllTests::suite());
         $suite->addTestSuite('Zend_Validate_StringLengthTest');
