Index: library/Zend/Feed/Writer/Extension/ITunes/Feed.php
===================================================================
--- library/Zend/Feed/Writer/Extension/ITunes/Feed.php	(revision 25051)
+++ library/Zend/Feed/Writer/Extension/ITunes/Feed.php	(working copy)
@@ -205,7 +205,7 @@
     /**
      * Set "explicit" flag
      *
-     * @param  bool $value
+     * @param  string $value    Valid values: "yes", "no" or "clean"
      * @return Zend_Feed_Writer_Extension_ITunes_Feed
      */
     public function setItunesExplicit($value)
