Index: tests/Zend/Form/FormTest.php
===================================================================
--- tests/Zend/Form/FormTest.php	(revisione 14377)
+++ tests/Zend/Form/FormTest.php	(copia locale)
@@ -1050,6 +1050,13 @@
         $this->assertEquals('foobar', $this->form->getElementsBelongTo());
     }
 
+    public function testArrayToWhichElementsBelongCanBeZero()
+    {
+        $this->testElementsInitiallyBelongToNoArrays();
+        $this->form->setElementsBelongTo('0');
+        $this->assertEquals('0', $this->form->getElementsBelongTo());
+    }
+
     public function testSettingArrayToWhichElementsBelongEmptyClearsIt()
     {
         $this->testCanSetArrayToWhichElementsBelong();
