Index: Search/Lucene/Document/Html.php
===================================================================
--- Search/Lucene/Document/Html.php	(Revision 17925)
+++ Search/Lucene/Document/Html.php	(Arbeitskopie)
@@ -393,6 +393,10 @@
             $words = array($words);
         }
 
+        if (count($words) == 0) {
+            return $this->_doc->saveHTML();
+        }
+
         $wordsToHighlightList = array();
         $analyzer = Zend_Search_Lucene_Analysis_Analyzer::getDefault();
         foreach ($words as $wordString) {
