diff --git library/Zend/Log.php library/Zend/Log.php
index 30e744d..a321f2e 100644
--- library/Zend/Log.php
+++ library/Zend/Log.php
@@ -198,6 +198,16 @@ class Zend_Log
     }
 
     /**
+     * Get writers. Return list of attached writers
+     *
+     * @return array
+     */
+    public function getWriters()
+    {
+        return $this->_writers;
+    }
+
+    /**
      * Set an extra item to pass to the log writers.
      *
      * @param  $name    Name of the field
