Index: incubator/Zend/Http/Client.php
===================================================================
--- incubator/Zend/Http/Client.php	(revision 860)
+++ incubator/Zend/Http/Client.php	(working copy)
@@ -925,7 +925,7 @@
      */
     static public function encodeFormData($boundary, $name, $value, $filename = null, $headers = array()) {
         $ret = "--{$boundary}\r\n" .
-            "Content-Disposition: form-data; name={$name}";
+            "Content-Disposition: form-data; name=\"{$name}\"";
             
         if ($filename) $ret .= "; filename={$filename}";
         $ret .= "\r\n";
