Index: Tag.php
===================================================================
--- Tag.php	(revision 17949)
+++ Tag.php	(working copy)
@@ -58,7 +58,7 @@
     {
         $matches = array();
 
-        if (!preg_match('#^@(\w+)\s#', $tagDocblockLine, $matches)) {
+        if (!preg_match('#^@(\w+)(\s|$)#', $tagDocblockLine, $matches)) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception('No valid tag name found within provided docblock line.');
         }
@@ -109,13 +109,13 @@
         $matches = array();
 
         // find the line
-        if (!preg_match('#^@(\w+)\s(.*)?#', $tagDocblockLine, $matches)) {
+        if (!preg_match('#^@(\w+)(?:\s+([^\s].*)|$)?#', $tagDocblockLine, $matches)) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception('Provided docblock line does not contain a valid tag');
         }
 
         $this->_name = $matches[1];
-        if ($matches[2]) {
+        if (isset($matches[2]) && $matches[2]) {
             $this->_description = $matches[2];
         }
     }
Index: Tag/Return.php
===================================================================
--- Tag/Return.php	(revision 17949)
+++ Tag/Return.php	(working copy)
@@ -43,7 +43,7 @@
      */
     public function __construct($tagDocblockLine)
     {
-        if (!preg_match('#^@(\w+)\s(\w+)(?:\s(.*))?#', $tagDocblockLine, $matches)) {
+        if (!preg_match('#^@(\w+)\s+(\w+)(?:\s+(.*))?#', $tagDocblockLine, $matches)) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception('Provided docblock line is does not contain a valid tag');
         }
Index: Tag/Param.php
===================================================================
--- Tag/Param.php	(revision 17949)
+++ Tag/Param.php	(working copy)
@@ -49,7 +49,7 @@
     {
         $matches = array();
 
-        if (!preg_match('#^@(\w+)\s(\w+)(?:\s(\$\S+))?(?:\s(.*))?#s', $tagDocblockLine, $matches)) {
+        if (!preg_match('#^@(\w+)\s+(\w+)(?:\s+(\$\S+))?(?:\s+(.*))?#s', $tagDocblockLine, $matches)) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception('Provided docblock line is does not contain a valid tag');
         }
