Index: library/Zend/Http/UserAgent/Features/Adapter/WurflApi.php
===================================================================
--- library/Zend/Http/UserAgent/Features/Adapter/WurflApi.php	(revision 23481)
+++ library/Zend/Http/UserAgent/Features/Adapter/WurflApi.php	(working copy)
@@ -82,7 +82,8 @@
                     $wurflConfig  = new WURFL_Configuration_InMemoryConfig();
                     $wurflConfig->wurflFile($c['wurfl']['main-file'])
                                 ->wurflPatch($c['wurfl']['patches'])
-                                ->persistence($c['persistence']['provider'], $c['persistence']['dir']);
+                                ->persistence($c['persistence']['provider'], $c['persistence']['params'])
+                                ->cache($c['cache']['provider'], $c['cache']['params']);
                 }
                 
                 $wurflManagerFactory = new WURFL_WURFLManagerFactory($wurflConfig);
