--- tests/Zend/Form/FormTest.php	(revision 23443)
+++ tests/Zend/Form/FormTest.php	(working copy)
@@ -2497,11 +2497,17 @@
 
     public function testCanCheckIfErrorsAreRegistered()
     {
-        $this->assertFalse($this->form->isErrors());
+        $this->assertFalse($this->form->hasErrors());
         $this->testCanValidateFullFormContainingOnlyElements();
-        $this->assertTrue($this->form->isErrors());
+        $this->assertTrue($this->form->hasErrors());
     }
 
+    public function testDeprecatedIsErrorsProxiesToHasErrors()
+    {
+        $this->testCanValidateFullFormContainingOnlyElements();
+        $this->assertEquals($this->form->isErrors(), $this->form->hasErrors());
+    }
+
     public function testCanRetrieveErrorCodesFromAllElementsAfterFailedValidation()
     {
         $this->_checkZf2794();
@@ -2892,9 +2898,9 @@
     public function testShouldAllowMarkingFormAsInvalid()
     {
         $this->form->addErrorMessage('Invalid values entered');
-        $this->assertFalse($this->form->isErrors());
+        $this->assertFalse($this->form->hasErrors());
         $this->form->markAsError();
-        $this->assertTrue($this->form->isErrors());
+        $this->assertTrue($this->form->hasErrors());
         $messages = $this->form->getMessages();
         $this->assertEquals(1, count($messages));
         $this->assertEquals('Invalid values entered', array_shift($messages));
@@ -2902,11 +2908,11 @@
 
     public function testShouldAllowPushingErrorsOntoErrorStackWithErrorMessages()
     {
-        $this->assertFalse($this->form->isErrors());
+        $this->assertFalse($this->form->hasErrors());
         $this->form->setErrors(array('Error 1', 'Error 2'))
                    ->addError('Error 3')
                    ->addErrors(array('Error 4', 'Error 5'));
-        $this->assertTrue($this->form->isErrors());
+        $this->assertTrue($this->form->hasErrors());
         $messages = $this->form->getMessages();
         $this->assertEquals(5, count($messages));
         foreach (range(1, 5) as $id) {
Index: library/Zend/Form.php
===================================================================
--- library/Zend/Form.php	(revision 23443)
+++ library/Zend/Form.php	(working copy)
@@ -2457,10 +2457,21 @@
     /**
      * Are there errors in the form?
      *
+     * @deprecated since 1.11.1 - use hasErrors() instead
      * @return bool
      */
     public function isErrors()
     {
+        return $this->hasErrors();
+    }
+    
+    /**
+     * Are there errors in the form?
+     *
+     * @return bool
+     */
+    public function hasErrors()
+    {
         return $this->_errorsExist;
     }
 
