Index: Zend/Filter/Input.php
===================================================================
--- Zend/Filter/Input.php	(revision 5217)
+++ Zend/Filter/Input.php	(working copy)
@@ -317,6 +317,18 @@
         $this->_process();
         return !(empty($this->_validFields));
     }
+    
+    /**
+     * @return boolean
+     */
+    public function isValid()
+    {
+        if ($this->hasInvalid() || $this->hasMissing()) {
+            return false;
+        }
+        
+        return true;
+    }
 
     /**
      * @param string $fieldName
