Index: library/Zend/Validate/PostCode.php
===================================================================
--- library/Zend/Validate/PostCode.php	(revision 20427)
+++ library/Zend/Validate/PostCode.php	(working copy)
@@ -70,7 +70,7 @@
      *
      * @param string|Zend_Locale|array|Zend_Config $options
      */
-    public function __construct($options)
+    public function __construct($options = null)
     {
         if ($options instanceof Zend_Config) {
             $options = $options->toArray();
