Index: LowerCase.php
===================================================================
--- LowerCase.php	(revision 12589)
+++ LowerCase.php	(working copy)
@@ -30,7 +30,7 @@
  * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Filter_File_LowerCase implements Zend_Filter_Interface
+class Zend_Filter_File_LowerCase extends Zend_Filter_StringToLower
 {
     /**
      * Defined by Zend_Filter_Interface
@@ -59,7 +59,7 @@
             throw new Zend_Filter_Exception("Problem while reading file '$value'");
         }
 
-        $content = strtolower($content);
+        $content = parent::filter($content);
         $result  = file_put_contents($value, $content);
 
         if (!$result) {
Index: UpperCase.php
===================================================================
--- UpperCase.php	(revision 12589)
+++ UpperCase.php	(working copy)
@@ -30,7 +30,7 @@
  * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Filter_File_UpperCase implements Zend_Filter_Interface
+class Zend_Filter_File_UpperCase extends Zend_Filter_StringToUpper
 {
     /**
      * Defined by Zend_Filter_Interface
@@ -59,7 +59,7 @@
             throw new Zend_Filter_Exception("Problem while reading file '$value'");
         }
 
-        $content = strtoupper($content);
+        $content = parent::filter($content);
         $result  = file_put_contents($value, $content);
 
         if (!$result) {
