Index: trunk/library/Zend/Dojo/View/Helper/Dijit.php
===================================================================
--- trunk/library/Zend/Dojo/View/Helper/Dijit.php	(Revision 19079)
+++ trunk/library/Zend/Dojo/View/Helper/Dijit.php	(Arbeitskopie)
@@ -171,7 +171,7 @@
      * @param  string|null $dijit
      * @return string
      */
-    public function _createFormElement($id, $value, array $params, array $attribs, $dijit = null)
+    protected function _createFormElement($id, $value, array $params, array $attribs, $dijit = null)
     {
         if (!array_key_exists('id', $attribs)) {
             $attribs['id'] = $id;
