Index: library/Zend/View/Helper/HeadMeta.php
===================================================================
--- library/Zend/View/Helper/HeadMeta.php	(revision 24537)
+++ library/Zend/View/Helper/HeadMeta.php	(working copy)
@@ -375,6 +375,14 @@
             $this->_escape($item->content),
             $modifiersString
         );
+        
+        if (isset($item->modifiers['conditional'])
+            && !empty($item->modifiers['conditional'])
+            && is_string($item->modifiers['conditional']))
+        {
+            $meta = '<!--[if ' . $this->_escape($item->modifiers['conditional']) . ']>' . $meta . '<![endif]-->';
+        }
+        
         return $meta;
     }
 
Index: tests/Zend/View/Helper/HeadMetaTest.php
===================================================================
--- tests/Zend/View/Helper/HeadMetaTest.php	(revision 24537)
+++ tests/Zend/View/Helper/HeadMetaTest.php	(working copy)
@@ -521,7 +521,18 @@
 			'<meta charset="utf-8"/>',
 			$view->headMeta()->toString());
 	}
-
+    
+    /**
+     * @group ZF-11835
+     */
+    public function testConditional() 
+    {
+        $html = $this->helper->appendHttpEquiv('foo', 'bar', array('conditional' => 'lt IE 7'))->toString();
+        
+        $this->assertRegExp("|^<!--\[if lt IE 7\]>|", $html);
+        $this->assertRegExp("|<!\[endif\]-->$|", $html);
+    }
+    
 }
 
 // Call Zend_View_Helper_HeadMetaTest::main() if this source file is executed directly.
