--- Consumer_old.php	2008-04-11 16:10:43.000000000 -0700
+++ Consumer_new.php	2008-04-17 22:04:50.000000000 -0700
@@ -240,7 +240,7 @@
                 }
                 /* OpenID 2.0 (11.2) Verifying Discovered Information */
                 if (isset($params['openid_claimed_id'])) {
-                    $id = $params['openid_claimed_id'];
+                    list($id, $rest) = explode("#", $params['openid_claimed_id']);
                     if (!$this->_discovery($id, $discovered_server, $discovered_version) ||
                         (isset($params['openid_identity']) &&
                          $params["openid_identity"] != $id) ||
@@ -270,6 +270,7 @@
             }
 
             /* OpenID 2.0 (11.2) Verifying Discovered Information */
+            list($id, $rest) = explode("#", $id);
             if ((isset($params['openid_identity']) &&
                  $params["openid_identity"] != $id) ||
                 (isset($params['openid_op_endpoint']) &&
