Index: tests/Zend/Application/Resource/DbTest.php
===================================================================
--- tests/Zend/Application/Resource/DbTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/DbTest.php	(working copy)
@@ -162,7 +162,7 @@
         );
         $resource = new Zend_Application_Resource_Db($config);
         $resource->init();
-        $this->assertType('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
+        $this->assertInstanceOf('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
     }
 
     /**
@@ -196,7 +196,7 @@
         );
         $resource = new Zend_Application_Resource_Db($config);
         $resource->init();
-        $this->assertType('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
+        $this->assertInstanceOf('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
     }
 
     /**
Index: tests/Zend/Application/Resource/LogTest.php
===================================================================
--- tests/Zend/Application/Resource/LogTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/LogTest.php	(working copy)
@@ -174,7 +174,7 @@
         $resource->setBootstrap($this->bootstrap);
         $log = $resource->init();
 
-        $this->assertType('Zend_Log', $log);
+        $this->assertInstanceOf('Zend_Log', $log);
 
         $log->log($message, Zend_Log::INFO);
         rewind($stream);
Index: tests/Zend/Application/Resource/ModulesTest.php
===================================================================
--- tests/Zend/Application/Resource/ModulesTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/ModulesTest.php	(working copy)
@@ -169,7 +169,7 @@
         $this->assertArrayHasKey('foo',     (array)$bootstraps);
         $this->assertArrayHasKey('default', (array)$bootstraps);
     }
-    
+
     public function testBootstrapBootstrapsIsOwnMethod()
     {
         require_once 'Zend/Application/Resource/Modules.php';
@@ -197,13 +197,13 @@
         $resource->setBootstrap($this->bootstrap);
         $resource->init();
         $bootstraps = $resource->getExecutedBootstraps();
-        $this->assertType('ArrayObject', $bootstraps);
+        $this->assertInstanceOf('ArrayObject', $bootstraps);
     }
 }
 
 require_once 'Zend/Application/Resource/Modules.php';
 class ZendTest_Application_Resource_ModulesHalf
-    extends Zend_Application_Resource_Modules 
+    extends Zend_Application_Resource_Modules
 {
     protected function bootstrapBootstraps($bootstraps)
     {
Index: tests/Zend/Application/Resource/MultidbTest.php
===================================================================
--- tests/Zend/Application/Resource/MultidbTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/MultidbTest.php	(working copy)
@@ -211,7 +211,7 @@
         $options['defaultMetadataCache'] = $cache;
         $resource = new Zend_Application_Resource_Multidb($options);
         $resource->init();
-        $this->assertType('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
+        $this->assertInstanceOf('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
     }
 
     /**
@@ -240,7 +240,7 @@
         $resource = new Zend_Application_Resource_Multidb($options);
         $resource->setBootstrap($this->bootstrap);
         $resource->init();
-        $this->assertType('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
+        $this->assertInstanceOf('Zend_Cache_Core', Zend_Db_Table::getDefaultMetadataCache());
     }
 }
 
Index: tests/Zend/Application/Resource/SessionTest.php
===================================================================
--- tests/Zend/Application/Resource/SessionTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/SessionTest.php	(working copy)
@@ -58,7 +58,7 @@
 
         $this->resource->setSaveHandler($saveHandlerClassName);
 
-        $this->assertType($saveHandlerClassName, $this->resource->getSaveHandler());
+        $this->assertInstanceOf($saveHandlerClassName, $this->resource->getSaveHandler());
     }
 
     public function testSetSaveHandlerArray()
@@ -68,7 +68,7 @@
 
         $this->resource->setSaveHandler(array('class' => $saveHandlerClassName));
 
-        $this->assertType($saveHandlerClassName, $this->resource->getSaveHandler());
+        $this->assertInstanceOf($saveHandlerClassName, $this->resource->getSaveHandler());
     }
 
     public function testSetOptions()
Index: tests/Zend/Application/Resource/TranslateTest.php
===================================================================
--- tests/Zend/Application/Resource/TranslateTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/TranslateTest.php	(working copy)
@@ -120,7 +120,7 @@
             $resource->getTranslate();
             $this->fail('Expected Zend_Application_Resource_Exception');
         } catch (Zend_Application_Resource_Exception $e) {
-            $this->assertType('Zend_Application_Resource_Exception', $e);
+            $this->assertInstanceOf('Zend_Application_Resource_Exception', $e);
         }
     }
 
@@ -171,7 +171,7 @@
         $resource->setBootstrap($this->bootstrap);
         $resource->init();
 
-        $this->assertType('Zend_Cache_Core', Zend_Translate::getCache());
+        $this->assertInstanceOf('Zend_Cache_Core', Zend_Translate::getCache());
         Zend_Translate::removeCache();
     }
 
Index: tests/Zend/Application/Resource/UseragentTest.php
===================================================================
--- tests/Zend/Application/Resource/UseragentTest.php	(revision 24228)
+++ tests/Zend/Application/Resource/UseragentTest.php	(working copy)
@@ -101,7 +101,7 @@
         $resource->init();
         $ua      = $resource->getUserAgent();
         $storage = $ua->getStorage();
-        $this->assertType('Zend_Http_UserAgent_Storage_NonPersistent', $storage);
+        $this->assertInstanceOf('Zend_Http_UserAgent_Storage_NonPersistent', $storage);
     }
 
     public function testInjectsUserAgentIntoViewHelperWhenViewResourcePresent()
