Index: tests/Zend/Reflection/FileTest.php
===================================================================
--- tests/Zend/Reflection/FileTest.php	(revision 24802)
+++ tests/Zend/Reflection/FileTest.php	(working copy)
@@ -139,5 +139,20 @@
         $this->assertEquals('Zend_Reflection_TestSampleInterface', $class->getName());
         $this->assertTrue($class->isInterface());
     }
+
+    /**
+     * @group ZF-12155
+     */
+    public function testFileCanReflectFunctionsContainingVariablesEmbeddedInStringWithCurlyBraces()
+    {
+        $fileToRequire = dirname(__FILE__) . '/_files/FunctionWithEmbeddedVariableInString.php';
+        require_once $fileToRequire;
+        $reflectionFile = new Zend_Reflection_File($fileToRequire);
+        $functions = $reflectionFile->getFunctions();
+        $this->assertEquals(2, count($functions));
+        $this->assertContainsOnly('Zend_Reflection_Function', $functions);
+        $this->assertEquals('firstOne', $functions[0]->getName());
+        $this->assertEquals('secondOne', $functions[1]->getName());
+    }
 }
 
Index: tests/Zend/Reflection/_files/FunctionWithEmbeddedVariableInString.php
===================================================================
--- tests/Zend/Reflection/_files/FunctionWithEmbeddedVariableInString.php	(revision 0)
+++ tests/Zend/Reflection/_files/FunctionWithEmbeddedVariableInString.php	(revision 0)
@@ -0,0 +1,11 @@
+<?php
+
+function firstOne() {
+    $substitute = "Testing";
+    $varA = "${substitute} 123!";
+    $varB = "{$substitute} 123!";
+    $varC = "$substitute 123!";
+    $varD = "${substitute}";
+}
+
+function secondOne() {}
