Index: library/Zend/Db/Adapter/Pdo/Abstract.php
===================================================================
--- library/Zend/Db/Adapter/Pdo/Abstract.php	(revision 15638)
+++ library/Zend/Db/Adapter/Pdo/Abstract.php	(working copy)
@@ -68,6 +68,7 @@
         unset($dsn['options']);
         unset($dsn['charset']);
         unset($dsn['driver_options']);
+        unset($dsn['persistent']);
 
         // use all remaining parts in the DSN
         foreach ($dsn as $key => $val) {
@@ -114,6 +115,10 @@
         // create PDO connection
         $q = $this->_profiler->queryStart('connect', Zend_Db_Profiler::CONNECT);
 
+        if ($this->_config['persistent']) {
+            $this->_config['driver_options'][PDO::ATTR_PERSISTENT] = true;
+        }
+        
         try {
             $this->_connection = new PDO(
                 $dsn,
Index: library/Zend/Db/Adapter/Abstract.php
===================================================================
--- library/Zend/Db/Adapter/Abstract.php	(revision 15638)
+++ library/Zend/Db/Adapter/Abstract.php	(working copy)
@@ -209,6 +209,10 @@
         if (!isset($config['charset'])) {
             $config['charset'] = null;
         }
+        
+        if (!isset($config['persistent'])) {
+            $config['persistent'] = false;
+        }
 
         $this->_config = array_merge($this->_config, $config);
         $this->_config['options'] = $options;
Index: library/Zend/Db/Adapter/Oracle.php
===================================================================
--- library/Zend/Db/Adapter/Oracle.php	(revision 15638)
+++ library/Zend/Db/Adapter/Oracle.php	(working copy)
@@ -117,7 +117,9 @@
             throw new Zend_Db_Adapter_Oracle_Exception('The OCI8 extension is required for this adapter but the extension is not loaded');
         }
 
-        $this->_connection = @oci_connect(
+        $connectFunction = ($this->_config['persistent']) ? 'oci_pconnect' : 'oci_connect';
+        
+        $this->_connection = @$connectFunction(
                 $this->_config['username'],
                 $this->_config['password'],
                 $this->_config['dbname'],
