Index: zend/library/Zend/Db/Select.php
===================================================================
--- zend/library/Zend/Db/Select.php	(revision 18530)
+++ zend/library/Zend/Db/Select.php	(working copy)
@@ -712,14 +712,27 @@
      * Clear parts of the Select object, or an individual part.
      *
      * @param string $part OPTIONAL
+     * @param string $correlationName OPTIONAL
      * @return Zend_Db_Select
      */
-    public function reset($part = null)
+    public function reset($part = null, $correlationName = null)
     {
         if ($part == null) {
             $this->_parts = self::$_partsInit;
         } else if (array_key_exists($part, self::$_partsInit)) {
-            $this->_parts[$part] = self::$_partsInit[$part];
+            if ($correlationName == null) {
+                $this->_parts[$part] = self::$_partsInit[$part];
+            } else if (isset($this->_parts[$part][$correlationName])) {
+                /**
+                * First remove columns
+                */
+                foreach ($this->_parts[self::COLUMNS] as $key=>$column) {
+                    if ($column[0] == $correlationName) {
+                        unset($this->_parts[self::COLUMNS][$key]);
+                    }
+                }
+                unset($this->_parts[$part][$correlationName]);
+            }
         }
         return $this;
     }
