Index: View/Helper/Action.php
===================================================================
--- View/Helper/Action.php	(revision 15802)
+++ View/Helper/Action.php	(working copy)
@@ -127,13 +127,23 @@
                       ->setControllerName($controller) 
                       ->setActionName($action) 
                       ->setDispatched(true); 
+        
+        // if the action called throws an exception - we need to put the old view 
+        // renderer back in place and reset objects before throwing it.
+        try {
+            $this->dispatcher->dispatch($this->request, $this->response);           
+        } catch (Exception $e) {
+            $exception = $e;
+        }
  
-        $this->dispatcher->dispatch($this->request, $this->response); 
- 
         // reset the viewRenderer object to it's original state
         Zend_Controller_Action_HelperBroker::addHelper($viewRendererObj);
 
-        
+        // if we caught an exception earlier - now is the time to throw it.
+        if ($exception) {
+            $this->resetObjects();
+            throw $exception;
+        }
         if (!$this->request->isDispatched() 
             || $this->response->isRedirect()) 
         { 
