Index: ZendFramework/lastest/library/Zend/File/Transfer/Adapter/Http.php
===================================================================
--- ZendFramework/lastest/library/Zend/File/Transfer/Adapter/Http.php	(revision 11240)
+++ ZendFramework/lastest/library/Zend/File/Transfer/Adapter/Http.php	(working copy)
@@ -161,4 +161,13 @@
         }
         return $result;
     }
+    
+    
+    /**
+     * Remove errors
+     */
+    public function clearErrors(){
+        $this->_messages = array();
+    }
+    
 }
