Index: Zend/Config/Xml.php
===================================================================
--- Zend/Config/Xml.php	(revision 10131)
+++ Zend/Config/Xml.php	(working copy)
@@ -63,6 +63,16 @@
             throw new Zend_Config_Exception('Filename is not set');
         }
 
+        if (! file_exists($filename)) {
+            require_once 'Zend/Config/Exception.php';
+            throw new Zend_Config_Exception('File "'. $filename .'" does not exist');
+        }
+
+        if (! is_readable($filename)) {
+            require_once 'Zend/Config/Exception.php';
+            throw new Zend_Config_Exception('File "'. $filename .'" is not readable.');
+        }
+
         $config = simplexml_load_file($filename);
 
         if (null === $section) {
