--- Session.php.dist	2008-09-16 15:11:50.000000000 +0200
+++ Session.php	2008-09-16 15:10:38.000000000 +0200
@@ -96,6 +96,7 @@
         'cookie_path'               => null,
         'cookie_domain'             => null,
         'cookie_secure'             => null,
+        'cookie_httponly'           => null,
         'use_cookies'               => null,
         'use_only_cookies'          => 'on',
         'referer_check'             => null,
