--- Collection_typo.php	2007-09-10 12:33:54.000000000 +0200
+++ Collection.php	2007-09-10 12:34:02.000000000 +0200
@@ -67,7 +67,7 @@
             /* @var $value Zend_XmlRpc_Value */
 
             if (!$value instanceof parent) {
-                throw new Zend_Xml_Rpc_Value_Exception('Values of '. get_class($this) .' type must be Zend_XmlRpc_Value objects');
+                throw new Zend_XmlRpc_Value_Exception('Values of '. get_class($this) .' type must be Zend_XmlRpc_Value objects');
             }
             $values[$key] = $value->getValue();
         }
