Index: library/Zend/View/Abstract.php
===================================================================
--- library/Zend/View/Abstract.php	(Revision 11279)
+++ library/Zend/View/Abstract.php	(Arbeitskopie)
@@ -775,10 +775,24 @@
      * Processes a view script and returns the output.
      *
      * @param string $name The script script name to process.
+     * @param string $module The module where to search for the script.
      * @return string The script output.
      */
-    public function render($name)
+    public function render($name, $module = null)
     {
+        // If a module is given, save the current paths and add the new basePath
+        if ($module !== null) {
+            $originalPaths = $this->_path;
+        
+            $moduleDir = Zend_Controller_Front::getInstance()->getControllerDirectory($module);
+            if (null === $moduleDir) {
+                require_once 'Zend/View/Exception.php';
+                throw new Zend_View_Exception('Cannot render view; module does not exist');
+            }
+            $viewsDir = dirname($moduleDir) . '/views';
+            $this->addBasePath($viewsDir);
+        }
+        
         // find the script file name using the parent private method
         $this->_file = $this->_script($name);
         unset($name); // remove $name from local scope
@@ -786,6 +800,11 @@
         ob_start();
         $this->_run($this->_file);
 
+        // If a module is given, restore the old basePath
+        if ($module !== null) {
+            $this->_path = $originalPaths;
+        }
+        
         return $this->_filter(ob_get_clean()); // filter output
     }
 
