Index: library/Zend/Tool/Project/Context/Zf/ControllerFile.php
===================================================================
--- library/Zend/Tool/Project/Context/Zf/ControllerFile.php	(revision 15883)
+++ library/Zend/Tool/Project/Context/Zf/ControllerFile.php	(working copy)
@@ -53,6 +53,11 @@
      * @var string
      */
     protected $_controllerName = 'index';
+
+    /**
+     * @var string
+     */
+    protected $_moduleName = null;
     
     /**
      * @var string
@@ -67,6 +72,7 @@
     public function init()
     {
         $this->_controllerName = $this->_resource->getAttribute('controllerName');
+        $this->_moduleName = $this->_resource->getAttribute('moduleName');
         $this->_filesystemName = ucfirst($this->_controllerName) . 'Controller.php';
         parent::init();
         return $this;
@@ -114,7 +120,7 @@
 
         $filter = new Zend_Filter_Word_DashToCamelCase();
         
-        $className = $filter->filter($this->_controllerName) . 'Controller';
+        $className = ($this->_moduleName ? ucfirst($this->_moduleName) . '_' : '') . $filter->filter($this->_controllerName) . 'Controller';
         
         $codeGenFile = new Zend_CodeGenerator_Php_File(array(
             'fileName' => $this->getPath(),
Index: library/Zend/Tool/Project/Provider/Controller.php
===================================================================
--- library/Zend/Tool/Project/Provider/Controller.php	(revision 15883)
+++ library/Zend/Tool/Project/Provider/Controller.php	(working copy)
@@ -81,7 +81,7 @@
             throw new Zend_Tool_Project_Provider_Exception($exceptionMessage);
         }
         
-        $newController = $controllersDirectory->createResource('controllerFile', array('controllerName' => $controllerName));
+        $newController = $controllersDirectory->createResource('controllerFile', array('controllerName' => $controllerName, 'moduleName' => $moduleName));
 
         return $newController;
     }
