--- library/Zend/Controller/Router/Rewrite.php	2009-06-06 14:41:49.000000000 +0200
+++ library/Zend/Controller/Router/Rewrite.php	2009-05-14 14:43:34.000000000 +0200
@@ -364,19 +364,19 @@
         }
 
         // Find the matching route
-        foreach (array_reverse($this->_routes) as $name => $route) {
+        foreach (array_reverse($this->_routes, true) as $name => $route) {
             // TODO: Should be an interface method. Hack for 1.0 BC
             if (method_exists($route, 'isAbstract') && $route->isAbstract()) {
                 continue;
             }