Index: library/Zend/CodeGenerator/Php/Property/DefaultValue.php
===================================================================
--- library/Zend/CodeGenerator/Php/Property/DefaultValue.php	(Revision 21529)
+++ library/Zend/CodeGenerator/Php/Property/DefaultValue.php	(Arbeitskopie)
@@ -243,21 +243,20 @@
         if ($type == self::TYPE_AUTO) {
             $type = $this->_getAutoDeterminedType($value);
 
-            if ($type == self::TYPE_ARRAY) {
-                $rii = new RecursiveIteratorIterator(
-                    $it = new RecursiveArrayIterator($value),
-                    RecursiveIteratorIterator::SELF_FIRST
-                    );
-                foreach ($rii as $curKey => $curValue) {
-                    if (!$curValue instanceof Zend_CodeGenerator_Php_Property_DefaultValue) {
-                        $curValue = new self(array('value' => $curValue));
-                        $rii->getSubIterator()->offsetSet($curKey, $curValue);
-                    }
-                    $curValue->setArrayDepth($rii->getDepth());
+        }
+        if ($type == self::TYPE_ARRAY) {
+            $rii = new RecursiveIteratorIterator(
+                $it = new RecursiveArrayIterator($value),
+                RecursiveIteratorIterator::SELF_FIRST
+                );
+            foreach ($rii as $curKey => $curValue) {
+                if (!$curValue instanceof Zend_CodeGenerator_Php_Property_DefaultValue) {
+                    $curValue = new self(array('value' => $curValue));
+                    $rii->getSubIterator()->offsetSet($curKey, $curValue);
                 }
-                $value = $rii->getSubIterator()->getArrayCopy();
+                $curValue->setArrayDepth($rii->getDepth());
             }
-
+            $value = $rii->getSubIterator()->getArrayCopy();
         }
 
         $output = '';
