Index: tests/Zend/View/Helper/FormCheckboxTest.php
===================================================================
--- tests/Zend/View/Helper/FormCheckboxTest.php	(revision 23872)
+++ tests/Zend/View/Helper/FormCheckboxTest.php	(working copy)
@@ -142,8 +142,59 @@
         ));
         $this->assertRegexp('/<input[^>]*?(id="foo-bar-baz")/', $html);
     }
-
+    
     /**
+     * @group ZF-3611
+     */
+    public function testArrayOfCheckboxItemsDoNotAllHaveSameId()
+    {
+        $html  = $this->helper->formCheckbox(array(
+            'name'  => 'foo[]',
+            'value' => 'bar0'
+        ));
+        $html .= $this->helper->formCheckbox(array(
+            'name'  => 'foo[]',
+            'value' => 'bar1'
+        ));
+        $html .= $this->helper->formCheckbox(array(
+            'name'  => 'foo[]',
+            'value' => 'bar2'
+        ));
+        
+        $this->assertTrue(preg_match_all('/<input[^>]*?(id="foo[^"]*")/', $html, $results) == 3);
+        $this->assertArrayHasKey(1, $results);
+        $this->assertTrue(count($results[1]) == 3 );
+        $this->assertFalse($results[1][0] == $results[1][1]);
+        $this->assertFalse($results[1][1] == $results[1][2]);
+    }
+    
+    /**
+     * @group ZF-3611
+     */
+    public function testArrayOfCheckboxItemsHasProperIdFormat()
+    {
+        $html  = $this->helper->formCheckbox(array(
+            'name'  => 'foo[]',
+            'value' => 'bar0'
+        ));
+        $html .= $this->helper->formCheckbox(array(
+            'name'  => 'foo[]',
+            'value' => 'bar1'
+        ));
+        $html .= $this->helper->formCheckbox(array(
+            'name'  => 'foo[]',
+            'value' => 'bar2'
+        ));
+        
+        $this->assertTrue(preg_match_all('/<input[^>]*?(id="foo[^"]*")/', $html, $results) == 3);
+        $this->assertArrayHasKey(1, $results);
+        $this->assertTrue(count($results[1]) == 3 );
+        $this->assertEquals('id="foo-bar0"', $results[1][0]);
+        $this->assertEquals('id="foo-bar1"', $results[1][1]);
+        $this->assertEquals('id="foo-bar2"', $results[1][2]);
+    }
+    
+    /**
      * @group ZF-2230
      */
     public function testDoesNotRenderHiddenElementsForCheckboxArray()
