Index: Zend/Locale.php
===================================================================
--- Zend/Locale.php	(revision 16416)
+++ Zend/Locale.php	(working copy)
@@ -938,11 +938,12 @@
             throw new Zend_Locale_Exception('Autodetection of Locale has been failed!');
         }
 
+        $locale = (string) $locale;
         if (strpos($locale, '-') !== false) {
             $locale = strtr($locale, '-', '_');
         }
 
-        return (string) $locale;
+        return $locale;
     }
 
     /**
