Index: tests/Zend/Application/Resource/CacheManagerTest.php
===================================================================
--- tests/Zend/Application/Resource/CacheManagerTest.php	(version 23692)
+++ tests/Zend/Application/Resource/CacheManagerTest.php	(working copy)
@@ -237,6 +237,59 @@
         $this->assertTrue($cache->getBackend() instanceof Zend_Cache_Backend_Custom_Naming);
         $this->assertTrue($cache instanceof Zend_Cache_Frontend_CustomNaming);
     }
+
+    public function testShouldCreateInstanceOfZendLogFromVailidCongig()
+    {
+        $options = array(
+            'page' => array(
+                'frontend' => array(
+                    'name' => 'Core',
+                    'options' => array(
+                        'logging' => true,
+                        'logger' => array(
+                            array(
+                                'writerName' => 'Stream',
+                                'writerParams' => array(
+                                    'stream' => 'php://output'
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+        );
+        $resource = new Zend_Application_Resource_Cachemanager($options);
+        $manager = $resource->init();
+        try {
+            $cacheTemplate = $manager->getCache('page');
+        } catch (Zend_Cache_Exception $e) {
+            $this->fail('Logger object was not created from vailid config');
+        }
+    }
+
+    public function testShouldNotCreateInstanceOfZendLogFromInvailidCongig()
+    {
+        $options = array(
+            'page' => array(
+                'frontend' => array(
+                    'name' => 'Core',
+                    'options' => array(
+                        'logging' => true,
+                        'logger' => ''
+
+                    )
+                )
+            )
+        );
+        $resource = new Zend_Application_Resource_Cachemanager($options);
+        $manager = $resource->init();
+        try {
+            $cacheTemplate = $manager->getCache('page');
+            $this->fail('Logger object was created from invailid config');
+        } catch (Zend_Cache_Exception $e) {
+
+        }
+    }
 }
 
 
Index: library/Zend/Cache/Backend.php
===================================================================
--- library/Zend/Cache/Backend.php	(version 23692)
+++ library/Zend/Cache/Backend.php	(working copy)
@@ -231,6 +231,12 @@
             if ($this->_directives['logger'] instanceof Zend_Log) {
                 return;
             }
+            try {
+                Zend_Log::factory($this->_directives['logger']);
+                return;
+            } catch (Zend_Log_Exception $e) {
+                Zend_Cache::throwException('Logger config is not accepted by Zend_Log::factory()', $e);
+            }
             Zend_Cache::throwException('Logger object is not an instance of Zend_Log class.');
         }
 
Index: library/Zend/Cache/Core.php
===================================================================
--- library/Zend/Cache/Core.php	(version 23692)
+++ library/Zend/Cache/Core.php	(working copy)
@@ -708,8 +708,17 @@
             return;
         }
 
-        if (isset($this->_options['logger']) && $this->_options['logger'] instanceof Zend_Log) {
-            return;
+        if (isset($this->_options['logger'])) {
+            if ($this->_options['logger'] instanceof Zend_Log) {
+                return;
+            }
+            try {
+                $this->_options['logger'] = Zend_Log::factory($this->_options['logger']);
+                return;
+            } catch (Zend_Log_Exception $e) {
+                Zend_Cache::throwException('Logger config is not accepted by Zend_Log::factory()', $e);
+            }
+            Zend_Cache::throwException('Logger object is not an instance of Zend_Log class.');
         }
 
         // Create a default logger to the standard output stream
