Index: library/Zend/Form/Decorator/HtmlTag.php
===================================================================
--- library/Zend/Form/Decorator/HtmlTag.php	(revision 25022)
+++ library/Zend/Form/Decorator/HtmlTag.php	(working copy)
@@ -73,6 +73,13 @@
     protected $_tagFilter;
 
     /**
+     * Error class
+     *
+     * @var string
+     */
+    protected $_errorClass;
+
+    /**
      * Convert options to tag attributes
      *
      * @return string
@@ -181,6 +188,39 @@
     }
 
     /**
+     * Set error class
+     *
+     * @param $class
+     *
+     * @return Zend_Form_Decorator_HtmlTag
+     */
+    public function setErrorClass($class)
+    {
+        $this->_errorClass = $class;
+        return $this;
+    }
+
+    /**
+     * Get error class
+     *
+     * If no error class is registered, either via setErrorClass() or as an
+     * option, uses 'invalid'.
+     *
+     * @return string
+     */
+    public function getErrorClass()
+    {
+        if (null === $this->_errorClass) {
+            if (null !== ($class = $this->getOption('errorClass'))) {
+                $this->setErrorClass($class);
+                $this->removeOption('errorClass');
+            }
+        }
+
+        return $this->_errorClass;
+    }
+
+    /**
      * Render content wrapped in an HTML tag
      *
      * @param  string $content
@@ -188,11 +228,12 @@
      */
     public function render($content)
     {
-        $tag       = $this->getTag();
-        $placement = $this->getPlacement();
-        $noAttribs = $this->getOption('noAttribs');
-        $openOnly  = $this->getOption('openOnly');
-        $closeOnly = $this->getOption('closeOnly');
+        $tag        = $this->getTag();
+        $placement  = $this->getPlacement();
+        $noAttribs  = $this->getOption('noAttribs');
+        $openOnly   = $this->getOption('openOnly');
+        $closeOnly  = $this->getOption('closeOnly');
+        $errorClass = $this->getErrorClass();
         $this->removeOption('noAttribs');
         $this->removeOption('openOnly');
         $this->removeOption('closeOnly');
@@ -200,6 +241,14 @@
         $attribs = null;
         if (!$noAttribs) {
             $attribs = $this->getOptions();
+
+            if (null !== $errorClass && $this->getElement()->hasErrors()) {
+                if (array_key_exists('class', $attribs)) {
+                    $attribs['class'] .= ' ' . $errorClass;
+                } else {
+                    $attribs['class'] = $errorClass;
+                }
+            }
         }
 
         switch ($placement) {
