Index: library/Zend/View/Helper/HeadMeta.php
===================================================================
--- library/Zend/View/Helper/HeadMeta.php       (revision 24687)
+++ library/Zend/View/Helper/HeadMeta.php       (working copy)
@@ -202,14 +202,15 @@
             return false;
         }

+        $isHtml5 = is_null($this->view) ? false : $this->view->doctype()->isHtml5();
+
         if (!isset($item->content)
-        && (! $this->view->doctype()->isHtml5()
-        || (! $this->view->doctype()->isHtml5() && $item->type !== 'charset'))) {
+        && (! $isHtml5 || (! $isHtml5 && $item->type !== 'charset'))) {
             return false;
         }

         // <meta property= ... /> is only supported with doctype RDFa
-        if (!$this->view->doctype()->isRdfa()
+        if ( !is_null($this->view) && !$this->view->doctype()->isRdfa()
             && $item->type === 'property') {
             return false;
         }
@@ -341,7 +342,7 @@

         $modifiersString = '';
         foreach ($item->modifiers as $key => $value) {
-            if ($this->view->doctype()->isHtml5()
+            if (!is_null($this->view) && $this->view->doctype()->isHtml5()
             && $key == 'scheme') {
                 require_once 'Zend/View/Exception.php';
                 throw new Zend_View_Exception('Invalid modifier '
Index: tests/Zend/View/Helper/HeadMetaTest.php
===================================================================
--- tests/Zend/View/Helper/HeadMetaTest.php     (revision 24687)
+++ tests/Zend/View/Helper/HeadMetaTest.php     (working copy)
@@ -532,7 +532,18 @@
         $this->assertRegExp("|^<!--\[if lt IE 7\]>|", $html);
         $this->assertRegExp("|<!\[endif\]-->$|", $html);
     }
-
+
+    /**
+     * @group ZF-11910
+     */
+    public function testStandaloneInstantiationWithoutViewDoesNotCauseFatalError()
+    {
+        $expected = '<meta name="foo" content="bar" />';
+
+        $helper = new Zend_View_Helper_HeadMeta();
+        $result = (string)$helper->headMeta()->appendName('foo','bar');
+        $this->assertEquals($expected, $result);
+    }
 }

 // Call Zend_View_Helper_HeadMetaTest::main() if this source file is executed directly.