Index: tests/Zend/Db/Table/Rowset/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/Rowset/TestCommon.php   (revision 21090)
+++ tests/Zend/Db/Table/Rowset/TestCommon.php   (working copy)
@@ -263,5 +263,21 @@
         }
         $this->assertFalse($connected);
     }
+    
+    /**
+     * @group ZF-8486
+     */
+    public function testTableRowsetIteratesAndWillReturnLastRowAfter()
+    {
+        $table = $this->_table['bugs'];
+        $rowset = $table->fetchAll('bug_id IN (1,2,3,4)', 'bug_id ASC');
+        foreach ($rowset as $row) {
+            $lastRow = $row->toArray();
+        }
+        
+        $items = $rowset->count();
 
+        $getRowRow = $rowset->getRow($items-1);
+    }
+
 }