Index: library/Zend/Db/Adapter/Mysqli.php
===================================================================
--- library/Zend/Db/Adapter/Mysqli.php	(revision 13376)
+++ library/Zend/Db/Adapter/Mysqli.php	(working copy)
@@ -302,6 +302,12 @@
             $port = null;
         }
 
+        if (isset($this->_config['socket'])) {
+            $socket = $this->_config['socket'];
+        } else {
+            $socket = null;
+        }
+        
         $this->_connection = mysqli_init();
 
         if(!empty($this->_config['driver_options'])) {
@@ -325,7 +331,8 @@
             $this->_config['username'],
             $this->_config['password'],
             $this->_config['dbname'],
-            $port
+            $port,
+            $socket
         );
 
         if ($_isConnected === false || mysqli_connect_errno()) {
