Index: DojoTest.php
===================================================================
--- DojoTest.php	(revision 24426)
+++ DojoTest.php	(working copy)
@@ -200,6 +200,89 @@
         $this->assertTrue($this->helper->isEnabled());
     }
 
+    public function testShouldSetSchemeToHttpWhenServerHttpsIsDisabled()
+    {
+        $this->helper->enable();
+        $this->helper->registerDojoStylesheet(true);
+        $this->helper->addStylesheetModule('dijit.themes.claro');
+        $this->helper->addStylesheet('http://example.org/css/style.css');
+        $this->helper->addLayer('http://example.org/js/layer.js');
+        $this->helper->registerModulePath('module', 'http://example.org/js/module');
+        $html = $this->helper->__toString();
+        $this->assertRegexp('#@import "http://.+/dojo/[0-9.]+/dojo/resources/dojo.css"#', $html);
+        $this->assertRegexp('#@import "http://.+/dojo/[0-9.]+/dijit/themes/claro/claro.css"#', $html);
+        $this->assertContains('@import "http://example.org/css/style.css"', $html);
+        $this->assertRegexp('#src="http://.+/dojo/[0-9.]+/dojo/dojo.xd.js"#', $html);
+        $this->assertContains('src="http://example.org/js/layer.js', $html);
+        $this->assertContains('dojo.registerModulePath("module", "http://example.org/js/module")', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojo", "http://.+/dojo/[0-9.]+/dojo"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dijit", "http://.+/dojo/[0-9.]+/dijit"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojox", "http://.+/dojo/[0-9.]+/dojox"\)#', $html);
+    }
+
+    public function testShouldSetSchemeToHttpsWhenServerHttpsIsEnabled()
+    {
+        $_SERVER['HTTPS'] = 'on';
+        $this->helper->enable();
+        $this->helper->registerDojoStylesheet(true);
+        $this->helper->addStylesheetModule('dijit.themes.claro');
+        $this->helper->addStylesheet('http://example.org/css/style.css');
+        $this->helper->addLayer('http://example.org/js/layer.js');
+        $this->helper->registerModulePath('module', 'http://example.org/js/module');
+        $html = $this->helper->__toString();
+        $this->assertRegexp('#@import "https://.+/dojo/[0-9.]+/dojo/resources/dojo.css"#', $html);
+        $this->assertRegexp('#@import "https://.+/dojo/[0-9.]+/dijit/themes/claro/claro.css"#', $html);
+        $this->assertContains('@import "https://example.org/css/style.css"', $html);
+        $this->assertRegexp('#src="https://.+/dojo/[0-9.]+/dojo/dojo.xd.js"#', $html);
+        $this->assertContains('src="https://example.org/js/layer.js', $html);
+        $this->assertContains('dojo.registerModulePath("module", "https://example.org/js/module")', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojo", "https://.+/dojo/[0-9.]+/dojo"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dijit", "https://.+/dojo/[0-9.]+/dijit"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojox", "https://.+/dojo/[0-9.]+/dojox"\)#', $html);
+    }
+
+    public function testShouldSetSchemeToHttpsEvenWhenServerHttpsIsDisabled()
+    {
+        $this->helper->enable();
+        $this->helper->setSecure(true);
+        $this->helper->registerDojoStylesheet(true);
+        $this->helper->addStylesheetModule('dijit.themes.claro');
+        $this->helper->addStylesheet('http://example.org/css/style.css');
+        $this->helper->addLayer('http://example.org/js/layer.js');
+        $this->helper->registerModulePath('module', 'http://example.org/js/module');
+        $html = $this->helper->__toString();
+        $this->assertRegexp('#@import "https://.+/dojo/[0-9.]+/dojo/resources/dojo.css"#', $html);
+        $this->assertRegexp('#@import "https://.+/dojo/[0-9.]+/dijit/themes/claro/claro.css"#', $html);
+        $this->assertContains('@import "https://example.org/css/style.css"', $html);
+        $this->assertRegexp('#src="https://.+/dojo/[0-9.]+/dojo/dojo.xd.js"#', $html);
+        $this->assertContains('src="https://example.org/js/layer.js', $html);
+        $this->assertContains('dojo.registerModulePath("module", "https://example.org/js/module")', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojo", "https://.+/dojo/[0-9.]+/dojo"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dijit", "https://.+/dojo/[0-9.]+/dijit"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojox", "https://.+/dojo/[0-9.]+/dojox"\)#', $html);    }
+
+    public function testShouldSetSchemeToHttpEvenWhenServerHttpsIsEnabled()
+    {
+        $_SERVER['HTTPS'] = 'on';
+        $this->helper->enable();
+        $this->helper->setSecure(false);
+        $this->helper->registerDojoStylesheet(true);
+        $this->helper->addStylesheetModule('dijit.themes.claro');
+        $this->helper->addStylesheet('http://example.org/css/style.css');
+        $this->helper->addLayer('http://example.org/js/layer.js');
+        $this->helper->registerModulePath('module', 'http://example.org/js/module');
+        $html = $this->helper->__toString();
+        $this->assertRegexp('#@import "http://.+/dojo/[0-9.]+/dojo/resources/dojo.css"#', $html);
+        $this->assertRegexp('#@import "http://.+/dojo/[0-9.]+/dijit/themes/claro/claro.css"#', $html);
+        $this->assertContains('@import "http://example.org/css/style.css"', $html);
+        $this->assertRegexp('#src="http://.+/dojo/[0-9.]+/dojo/dojo.xd.js"#', $html);
+        $this->assertContains('src="http://example.org/js/layer.js', $html);
+        $this->assertContains('dojo.registerModulePath("module", "http://example.org/js/module")', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojo", "http://.+/dojo/[0-9.]+/dojo"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dijit", "http://.+/dojo/[0-9.]+/dijit"\)#', $html);
+        $this->assertRegexp('#dojo.registerModulePath\("dojox", "http://.+/dojo/[0-9.]+/dojox"\)#', $html);
+    }
+
     public function testShouldUtilizeCdnByDefault()
     {
         $this->helper->enable();
