Index: tests/Zend/Form/Element/SubmitTest.php
===================================================================
--- tests/Zend/Form/Element/SubmitTest.php	(revision 19070)
+++ tests/Zend/Form/Element/SubmitTest.php	(working copy)
@@ -179,6 +179,25 @@
         $this->assertTrue($this->element->isChecked());
     }
 
+    /**
+     * Tests that the isChecked method works as expected when using a translator.
+     * @group ZF-4073
+     */
+    public function testIsCheckedReturnsExpectedValueWhenUsingTranslator()
+    {
+        $translations = array('label' => 'translation');
+        $translate = new Zend_Translate('array', $translations);
+
+        $submit = new Zend_Form_Element_Submit('foo', 'label');
+        $submit->setTranslator($translate);
+        $submit->setValue($translations['label']);
+        
+        $this->assertTrue($submit->isChecked());
+
+        $submit->setValue('label');
+        $this->assertFalse($submit->isChecked());
+    }
+
     /*
      * Tests if title attribute (tooltip) is translated if the default decorators are loaded.
      * These decorators should load the Tooltip decorator as the first decorator.
