Index: library/Zend/Config/Json.php
===================================================================
--- library/Zend/Config/Json.php	(revision 24790)
+++ library/Zend/Config/Json.php	(working copy)
@@ -220,7 +220,8 @@
     {
         foreach ($this->_getConstants() as $constant) {
             if (strstr($value, $constant)) {
-                $value = str_replace($constant, constant($constant), $value);
+                $replacement = str_replace('\\', '\\\\', constant($constant));
+                $value = str_replace($constant, $replacement, $value);
             }
         }
         return $value;
