Index: Ini.php
===================================================================
--- Ini.php	(revision 13150)
+++ Ini.php	(working copy)
@@ -85,6 +85,10 @@
      */
     public function __construct($filename, $section = null, $options = false)
     {
+        if (!is_null(self::$_defaultCache)) {
+            $this->_cache = self::$_defaultCache;
+        }
+        
         if (empty($filename)) {
             /**
              * @see Zend_Config_Exception
@@ -104,7 +108,18 @@
                 $this->_nestSeparator = (string) $options['nestSeparator'];
             }
         }
-
+        
+        if ($this->_cacheEnabled()) {
+            $cache = $this->_cache->load(spl_object_hash($this));
+            if ($cache !== false) {
+                $vars = get_object_vars($this);
+                foreach ($vars as $var=>$val) {
+                	$this->$var = $cache->$var;
+                }
+                return;
+            }
+        }
+        
         set_error_handler(array($this, '_loadFileErrorHandler'));
         $iniArray = parse_ini_file($filename, true); // Warnings and errors are suppressed
         restore_error_handler();
@@ -177,6 +192,9 @@
         }
 
         $this->_loadedSection = $section;
+        if ($this->_cacheEnabled() && $this->_cache->load(spl_object_hash($this)) === false) {
+            $this->_cache->save($this, spl_object_hash($this));
+        }
     }
     
     /**
