Index: IncludePathLoader.php
===================================================================
--- IncludePathLoader.php	(Revision 18381)
+++ IncludePathLoader.php	(Arbeitskopie)
@@ -91,6 +91,9 @@
             // iterate over the accepted items
             foreach ($iterator as $item) {
                 
+                if (strpos($item->__toString(), 'PHPUnit' . DIRECTORY_SEPARATOR . 'Framework')) {
+                   continue;
+               }
                 // ensure that the same named file from separate include_paths is not loaded
                 $relativeItem = preg_replace('#^' . preg_quote($realIncludePath . DIRECTORY_SEPARATOR, '#') . '#', '', $item->getRealPath());
                 
