Index: library/Zend/Db/Adapter/Mysqli.php
===================================================================
--- library/Zend/Db/Adapter/Mysqli.php	(revision 11722)
+++ library/Zend/Db/Adapter/Mysqli.php	(working copy)
@@ -297,9 +297,47 @@
             $port = null;
         }
 
+        $this->_connection = mysqli::init();
+        
+        if(!empty($this->_config['driver_options'])) {
+            foreach($this->_config['driver_options'] as $option=>$value) {
+                switch($option) {
+                    case MYSQLI_INIT_COMMAND:
+                        // break intentionally omitted
+                    case 'initcommand':
+                        mysqli_options($this->_connection, MYSQLI_INIT_COMMAND, $value);
+                       break;
+                    case MYSQLI_OPT_CONNECT_TIMEOUT:
+                        // break intentionally omitted
+                    case 'connecttimeout':
+                        mysqli_options($this->_connection, MYSQLI_OPT_CONNECT_TIMEOUT, $value);
+                       break;
+                    case MYSQLI_OPT_LOCAL_INFILE:
+                        // break intentionally omitted
+                    case 'localinfile':
+                        mysqli_options($this->_connection, MYSQLI_OPT_LOCAL_INFILE, $value);
+                       break;
+                    case MYSQLI_READ_DEFAULT_FILE:
+                        // break intentionally omitted
+                    case 'readdefaultfile':
+                        mysqli_options($this->_connection, MYSQLI_READ_DEFAULT_FILE, $value);
+                       break;
+                    case MYSQLI_READ_DEFAULT_GROUP:
+                        // break intentionally omitted
+                    case 'readdefaultgroup':
+                       mysqli_options($this->_connection, MYSQLI_READ_DEFAULT_GROUP, $value);
+                       break;
+                    default:
+                        // don't do anything.
+                        break;
+                }
+            }
+        }
+        
         // Suppress connection warnings here.
         // Throw an exception instead.
-        @$this->_connection = new mysqli(
+        $_isConnected = @mysqli_real_connect(
+            $this->_connection,
             $this->_config['host'],
             $this->_config['username'],
             $this->_config['password'],
@@ -306,7 +344,8 @@
             $this->_config['dbname'],
             $port
         );
-        if ($this->_connection === false || mysqli_connect_errno()) {
+        
+        if ($_isConnected === false || mysqli_connect_errno()) {
             /**
              * @see Zend_Db_Adapter_Mysqli_Exception
              */
