Index: vendor/zend-framework/library/Zend/Search/Lucene/Storage/Directory/Filesystem.php
===================================================================
--- vendor/zend-framework/library/Zend/Search/Lucene/Storage/Directory/Filesystem.php	(revision 9489)
+++ vendor/zend-framework/library/Zend/Search/Lucene/Storage/Directory/Filesystem.php	(working copy)
@@ -176,6 +176,7 @@
      */
     public function createFile($filename)
     {
+    	$fileExists = file_exists($this->_dirPath . '/' . $filename);
         if (isset($this->_fileHandlers[$filename])) {
             $this->_fileHandlers[$filename]->close();
         }
@@ -184,7 +185,7 @@
 
         global $php_errormsg;
         $trackErrors = ini_get('track_errors'); ini_set('track_errors', '1');
-        if (!@chmod($this->_dirPath . '/' . $filename, self::$_defaultFilePermissions)) {
+        if (!$fileExists && !@chmod($this->_dirPath . '/' . $filename, self::$_defaultFilePermissions)) {
             ini_set('track_errors', $trackErrors);
             throw new Zend_Search_Lucene_Exception($php_errormsg);
         }
