--- library/Zend/Paginator/Adapter/Interface.php	(revision 23299)
+++ library/Zend/Paginator/Adapter/Interface.php	(working copy)
@@ -30,13 +30,6 @@
 interface Zend_Paginator_Adapter_Interface extends Countable
 {
     /**
-     * Returns the total number of rows in the collection.
-     *
-     * @return integer
-     */
-    //public function count();
-
-    /**
      * Returns an collection of items for a page.
      *
      * @param  integer $offset Page offset
@@ -44,4 +37,4 @@
      * @return array
      */
     public function getItems($offset, $itemCountPerPage);
-}
\ No newline at end of file
+}
