Index: library/Zend/Search/Lucene/Analysis/Analyzer.php
===================================================================
--- library/Zend/Search/Lucene/Analysis/Analyzer.php	(revision 7513)
+++ library/Zend/Search/Lucene/Analysis/Analyzer.php	(working copy)
@@ -23,24 +23,6 @@
 /** Zend_Search_Lucene_Analysis_Token */
 require_once 'Zend/Search/Lucene/Analysis/Token.php';
 
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8 */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Utf8.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8Num */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Utf8Num.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Text */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Text.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Text_CaseInsensitive */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Text/CaseInsensitive.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_TextNum */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/TextNum.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_TextNum_CaseInsensitive */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/TextNum/CaseInsensitive.php';
-
 /** Zend_Search_Lucene_Analysis_TokenFilter_StopWords */
 require_once 'Zend/Search/Lucene/Analysis/TokenFilter/StopWords.php';
 
