Index: library/Zend/Filter.php
===================================================================
--- library/Zend/Filter.php	(revision 917)
+++ library/Zend/Filter.php	(working copy)
@@ -229,16 +229,14 @@
     }
 
     /**
-     * Returns value if it is a valid email format, FALSE otherwise.
+     * Returns TRUE if it is a valid email format, FALSE otherwise.
      *
      * @param mixed $value
-     * @return mixed
+     * @return bool
      */
     public static function isEmail($value)
     {
-        /**
-         * @todo RFC 2822 (http://www.ietf.org/rfc/rfc2822.txt)
-         */
+        throw new Zend_Filter_Exception('isEmail() method is not currently implemented');
     }
 
     /**
Index: tests/Zend/FilterTest.php
===================================================================
--- tests/Zend/FilterTest.php	(revision 917)
+++ tests/Zend/FilterTest.php	(working copy)
@@ -109,4 +109,9 @@
         $this->assertTrue(Zend_Filter::isPhone('6134123456'), '"6134123456" is a US Phone number');
         $this->assertFalse(Zend_Filter::isPhone('1004123456'), '"100123456" is not a valid US Phone number');
     }
+
+    public function testisEmail()
+    {    	
+        $this->assertTrue(Zend_Filter::isEmail('name@hostname.com'), '"name@hostname.com" is a valid email address');
+    }
 }
