Index: Search/Lucene/Document/Html.php
===================================================================
--- Search/Lucene/Document/Html.php	(revision 9039)
+++ Search/Lucene/Document/Html.php	(working copy)
@@ -105,7 +105,7 @@
 
         $linkNodes = $this->_doc->getElementsByTagName('a');
         foreach ($linkNodes as $linkNode) {
-            if (($href = $linkNode->getAttribute('href')) != '') {
+            if (($href = $linkNode->getAttribute('href')) != '' && $linkNode->getAttribute('rel') != 'nofollow' ) {
                 $this->_links[] = $href;
             }
         }
