Index: tests/Zend/Form/ElementTest.php
===================================================================
--- tests/Zend/Form/ElementTest.php	(revision 23558)
+++ tests/Zend/Form/ElementTest.php	(working copy)
@@ -182,7 +182,7 @@
     }
 
     /**
-     * @see ZF-2851
+     * @group ZF-2851
      */
     public function testSetNameShouldNotAllowEmptyString()
     {
@@ -313,7 +313,7 @@
     }
 
     /**
-     * @see ZF-2862
+     * @group ZF-2862
      */
     public function testBreakChainOnFailureFlagsForExistingValidatorsRemainSetWhenNotEmptyValidatorAutoInserted()
     {
@@ -848,7 +848,7 @@
     }
 
     /**#@+
-     * @see ZF-2988
+     * @group ZF-2988
      */
     public function testSettingErrorMessageShouldOverrideValidationErrorMessages()
     {
@@ -1385,7 +1385,7 @@
     }
 
     /**
-     * @see ZF-3597
+     * @group ZF-3597
      */
     public function testAddingConcreteDecoratorShouldHonorOrder()
     {
@@ -1449,7 +1449,7 @@
     }
 
     /**
-     * @see ZF-3069
+     * @group ZF-3069
      */
     public function testRemovingNamedDecoratorsShouldWork()
     {
@@ -1516,7 +1516,7 @@
     }
 
     /**
-     * @see ZF-3376
+     * @group ZF-3376
      */
     public function testSetDecoratorsShouldAcceptReturnOfGetDecorators()
     {
@@ -2201,7 +2201,7 @@
     }
 
     /**
-     * @see ZF-6822
+     * @group ZF-6822
      */
     public function testValidatorByUsingStringNotation()
     {
Index: tests/Zend/Form/Decorator/LabelTest.php
===================================================================
--- tests/Zend/Form/Decorator/LabelTest.php	(revision 23558)
+++ tests/Zend/Form/Decorator/LabelTest.php	(working copy)
@@ -211,7 +211,7 @@
     }
 
     /**
-     * @see ZF-3538
+     * @group ZF-3538
      */
     public function testRenderShouldNotUtilizeElementClass()
     {
Index: tests/Zend/Form/Decorator/FormTest.php
===================================================================
--- tests/Zend/Form/Decorator/FormTest.php	(revision 23558)
+++ tests/Zend/Form/Decorator/FormTest.php	(working copy)
@@ -109,7 +109,7 @@
     }
 
     /**
-     * @see ZF-3643
+     * @group ZF-3643
      */
     public function testShouldPreferFormIdAttributeOverFormName()
     {
Index: tests/Zend/Form/Decorator/ImageTest.php
===================================================================
--- tests/Zend/Form/Decorator/ImageTest.php	(revision 23558)
+++ tests/Zend/Form/Decorator/ImageTest.php	(working copy)
@@ -155,7 +155,7 @@
     }
 
     /**
-     * @see ZF-2714
+     * @group ZF-2714
      */
     public function testImageElementAttributesPassedWithDecoratorOptionsToViewHelper()
     {
Index: tests/Zend/Form/Decorator/FieldsetTest.php
===================================================================
--- tests/Zend/Form/Decorator/FieldsetTest.php	(revision 23558)
+++ tests/Zend/Form/Decorator/FieldsetTest.php	(working copy)
@@ -127,7 +127,7 @@
     }
 
     /**
-     * @see ZF-7054
+     * @group ZF-7054
      */
     public function testCustomIdSupersedesElementId()
     {
@@ -141,7 +141,7 @@
     }
 
     /**
-     * @see ZF-2981
+     * @group ZF-2981
      */
     public function testActionAndMethodAttributesShouldNotBePresentInFieldsetTag()
     {
@@ -157,7 +157,7 @@
     }
 
     /**#@+
-     * @see ZF-3731
+     * @group ZF-3731
      */
     public function testIdShouldBePrefixedWithFieldset()
     {
@@ -184,7 +184,7 @@
     /**#@-*/
 
     /**
-     * @see ZF-3728
+     * @group ZF-3728
      */
     public function testEnctypeAttributeShouldNotBePresentInFieldsetTag()
     {
@@ -200,7 +200,7 @@
     }
 
     /**
-     * @see ZF-3499
+     * @group ZF-3499
      */
     public function testHelperAttributeShouldNotBePresentInFieldsetTag()
     {
Index: tests/Zend/Form/SubFormTest.php
===================================================================
--- tests/Zend/Form/SubFormTest.php	(revision 23558)
+++ tests/Zend/Form/SubFormTest.php	(working copy)
@@ -95,7 +95,7 @@
     // Bugfixes
 
     /**
-     * @see ZF-2883
+     * @group ZF-2883
      */
     public function testDisplayGroupsShouldInheritSubFormNamespace()
     {
@@ -112,7 +112,7 @@
     }
 
     /**
-     * @see ZF-3272
+     * @group ZF-3272
      */
     public function testRenderedSubFormDtShouldContainNoBreakSpace()
     {
Index: tests/Zend/Form/Element/SelectTest.php
===================================================================
--- tests/Zend/Form/Element/SelectTest.php	(revision 23558)
+++ tests/Zend/Form/Element/SelectTest.php	(working copy)
@@ -137,7 +137,7 @@
     /**
      * No explicit assertions; just checking for error conditions
      *
-     * @see ZF-2847
+     * @group ZF-2847
      */
     public function testTranslationShouldNotRaiseWarningsWithNestedGroups()
     {
@@ -162,7 +162,6 @@
     }
 
     /**
-     * @see   ZF-3953
      * @group ZF-3953
      */
     public function testUsingZeroAsValueShouldSelectAppropriateOption()
@@ -202,7 +201,7 @@
      * Test isValid() on select elements without optgroups. This
      * ensures fixing ZF-3985 doesn't break existing functionality.
      *
-     * @see ZF-3985
+     * @group ZF-3985
      */
     public function testIsValidWithPlainOptions()
     {
Index: tests/Zend/Form/Element/MultiCheckboxTest.php
===================================================================
--- tests/Zend/Form/Element/MultiCheckboxTest.php	(revision 23558)
+++ tests/Zend/Form/Element/MultiCheckboxTest.php	(working copy)
@@ -157,7 +157,7 @@
     }
 
     /**
-     * @see ZF-2830
+     * @group ZF-2830
      */
     public function testRenderingMulticheckboxCreatesCorrectArrayNotation()
     {
@@ -170,7 +170,7 @@
     }
 
     /**
-     * @see ZF-2828
+     * @group ZF-2828
      */
     public function testCanPopulateCheckboxOptionsFromPostedData()
     {
@@ -219,7 +219,7 @@
     }
 
     /**#+
-     * @see ZF-3286
+     * @group ZF-3286
      */
     public function testShouldRegisterInArrayValidatorByDefault()
     {
Index: tests/Zend/Form/Element/MultiselectTest.php
===================================================================
--- tests/Zend/Form/Element/MultiselectTest.php	(revision 23558)
+++ tests/Zend/Form/Element/MultiselectTest.php	(working copy)
@@ -151,7 +151,7 @@
     }
 
     /**
-     * @see ZF-2824
+     * @group ZF-2824
      */
     public function testCanSetMultiOptionsUsingAssocArraysWithKeyValueKeys()
     {
@@ -176,7 +176,7 @@
     }
 
     /**
-     * @see ZF-2824
+     * @group ZF-2824
      */
     public function testCanSetMultiOptionsUsingConfigWithKeyValueKeys()
     {
Index: tests/Zend/Form/DisplayGroupTest.php
===================================================================
--- tests/Zend/Form/DisplayGroupTest.php	(revision 23558)
+++ tests/Zend/Form/DisplayGroupTest.php	(working copy)
@@ -312,7 +312,7 @@
     }
 
     /**
-     * @see ZF-3069
+     * @group ZF-3069
      */
     public function testRemovingNamedDecoratorsShouldWork()
     {
@@ -360,7 +360,7 @@
     }
 
     /**
-     * @see ZF-3494
+     * @group ZF-3494
      */
     public function testGetViewShouldNotReturnNullWhenViewRendererIsActive()
     {
Index: tests/Zend/Acl/AclTest.php
===================================================================
--- tests/Zend/Acl/AclTest.php	(revision 23558)
+++ tests/Zend/Acl/AclTest.php	(working copy)
@@ -1052,7 +1052,7 @@
      * Ensures that the $onlyParents argument to inheritsRole() works
      *
      * @return void
-     * @see    http://framework.zend.com/issues/browse/ZF-2502
+     * @group ZF-2502
      */
     public function testRoleInheritanceSupportsCheckingOnlyParents()
     {
@@ -1066,7 +1066,7 @@
      * Ensures that the solution for ZF-2234 works as expected
      *
      * @return void
-     * @see    http://framework.zend.com/issues/browse/ZF-2234
+     * @group ZF-2234
      */
     public function testAclInternalDFSMethodsBehaveProperly()
     {
Index: tests/Zend/Validate/EmailAddressTest.php
===================================================================
--- tests/Zend/Validate/EmailAddressTest.php	(revision 23558)
+++ tests/Zend/Validate/EmailAddressTest.php	(working copy)
@@ -382,7 +382,7 @@
     }
 
     /**
-     * @see ZF-2861
+     * @group ZF-2861
      */
     public function testHostnameValidatorMessagesShouldBeTranslated()
     {
@@ -416,7 +416,7 @@
     }
 
     /**
-     * @see ZF-4888
+     * @group ZF-4888
      */
     public function testEmailsExceedingLength()
     {
@@ -430,7 +430,7 @@
     }
 
     /**
-     * @see ZF-4352
+     * @group ZF-4352
      */
     public function testNonStringValidation()
     {
@@ -438,7 +438,7 @@
     }
 
     /**
-     * @see ZF-7490
+     * @group ZF-7490
      */
     public function testSettingHostnameMessagesThroughEmailValidator()
     {
Index: tests/Zend/Validate/AbstractTest.php
===================================================================
--- tests/Zend/Validate/AbstractTest.php	(revision 23558)
+++ tests/Zend/Validate/AbstractTest.php	(working copy)
@@ -201,7 +201,7 @@
     }
 
     /**
-     * @see ZF-4463
+     * @group ZF-4463
      */
     public function testDoesNotFailOnObjectInput()
     {
Index: tests/Zend/Validate/DateTest.php
===================================================================
--- tests/Zend/Validate/DateTest.php	(revision 23558)
+++ tests/Zend/Validate/DateTest.php	(working copy)
@@ -89,7 +89,7 @@
     /**
      * Ensures that characters trailing an otherwise valid date cause the input to be invalid
      *
-     * @see    http://framework.zend.com/issues/browse/ZF-1804
+     * @group ZF-1804
      * @return void
      */
     public function testCharactersTrailingInvalid()
@@ -103,7 +103,7 @@
     /**
      * Ensures that characters leading an otherwise valid date cause the input to be invalid
      *
-     * @see    http://framework.zend.com/issues/browse/ZF-1804
+     * @group ZF-1804
      * @return void
      */
     public function testCharactersLeadingInvalid()
@@ -127,7 +127,7 @@
     /**
      * Ensures that the validator can handle different manual dateformats
      *
-     * @see    http://framework.zend.com/issues/browse/ZF-2003
+     * @group ZF-2003
      * @return void
      */
     public function testUseManualFormat()
@@ -153,7 +153,7 @@
     /**
      * Ensures that the validator can handle different dateformats from locale
      *
-     * @see    http://framework.zend.com/issues/browse/ZF-2003
+     * @group ZF-2003
      * @return void
      */
     public function testUseLocaleFormat()
@@ -191,7 +191,7 @@
     /**
      * Ensures that the validator can handle different dateformats from locale
      *
-     * @see    http://framework.zend.com/issues/browse/ZF-2003
+     * @group ZF-2003
      * @return void
      */
     public function testLocaleContructor()
@@ -249,7 +249,7 @@
      * @param  integer $errline
      * @param  array   $errcontext
      * @return void
-     * @see    http://framework.zend.com/issues/browse/ZF-2789
+     * @group ZF-2789
      */
     public function errorHandlerIgnore($errno, $errstr, $errfile, $errline, array $errcontext)
     {
Index: tests/Zend/Validate/NotEmptyTest.php
===================================================================
--- tests/Zend/Validate/NotEmptyTest.php	(revision 23558)
+++ tests/Zend/Validate/NotEmptyTest.php	(working copy)
@@ -530,7 +530,7 @@
     }
 
     /**
-     * @see ZF-3236
+     * @group ZF-3236
      */
     public function testStringWithZeroShouldNotBeTreatedAsEmpty()
     {
Index: tests/Zend/Validate/IpTest.php
===================================================================
--- tests/Zend/Validate/IpTest.php	(revision 23558)
+++ tests/Zend/Validate/IpTest.php	(working copy)
@@ -117,8 +117,8 @@
     }
 
     /**
-     * @see ZF-2694
-     * @see ZF-8253
+     * @group ZF-2694
+     * @group ZF-8253
      */
     public function testIPv6addresses()
     {
Index: tests/Zend/Validate/IdenticalTest.php
===================================================================
--- tests/Zend/Validate/IdenticalTest.php	(revision 23558)
+++ tests/Zend/Validate/IdenticalTest.php	(working copy)
@@ -102,7 +102,7 @@
     }
 
     /**
-     * @see ZF-6953
+     * @group ZF-6953
      */
     public function testValidatingAgainstEmptyToken()
     {
@@ -111,7 +111,7 @@
     }
 
     /**
-     * @see ZF-7128
+     * @group ZF-7128
      */
     public function testValidatingAgainstNonStrings()
     {
Index: tests/Zend/Validate/HostnameTest.php
===================================================================
--- tests/Zend/Validate/HostnameTest.php	(revision 23558)
+++ tests/Zend/Validate/HostnameTest.php	(working copy)
@@ -274,8 +274,8 @@
     /**
      * Test changed with ZF-6676, as IP check is only involved when IP patterns match
      *
-     * @see ZF-2861
-     * @see ZF-6676
+     * @group ZF-2861
+     * @group ZF-6676
      */
     public function testValidatorMessagesShouldBeTranslated()
     {
@@ -301,7 +301,7 @@
     }
 
     /**
-     * @see ZF-6033
+     * @group ZF-6033
      */
     public function testNumberNames()
     {
@@ -320,7 +320,7 @@
     }
 
     /**
-     * @see ZF-6133
+     * @group ZF-6133
      */
     public function testPunycodeDecoding()
     {
@@ -357,7 +357,7 @@
     }
 
     /**
-     * @see ZF-7277
+     * @group ZF-7277
      */
     public function testDifferentIconvEncoding()
     {
Index: tests/Zend/Uri/HttpTest.php
===================================================================
--- tests/Zend/Uri/HttpTest.php	(revision 23558)
+++ tests/Zend/Uri/HttpTest.php	(working copy)
@@ -79,7 +79,7 @@
      * Make sure an exception is thrown when trying to use fromString() with a
      * non-HTTP scheme
      *
-     * @see http://framework.zend.com/issues/browse/ZF-4395
+     * @group ZF-4395
      *
      * @expectedException Zend_Uri_Exception
      */
Index: tests/Zend/View/Helper/HeadTitleTest.php
===================================================================
--- tests/Zend/View/Helper/HeadTitleTest.php	(revision 23558)
+++ tests/Zend/View/Helper/HeadTitleTest.php	(working copy)
@@ -221,7 +221,7 @@
     }
 
    /**
-    * @see ZF-8036
+    * @group ZF-8036
     */
     public function testHeadTitleZero()
     {
Index: tests/Zend/View/Helper/FormResetTest.php
===================================================================
--- tests/Zend/View/Helper/FormResetTest.php	(revision 23558)
+++ tests/Zend/View/Helper/FormResetTest.php	(working copy)
@@ -91,7 +91,7 @@
     }
 
     /**
-     * @see ZF-2845
+     * @group ZF-2845
      */
     public function testShouldAllowDisabling()
     {
Index: tests/Zend/View/Helper/FieldsetTest.php
===================================================================
--- tests/Zend/View/Helper/FieldsetTest.php	(revision 23558)
+++ tests/Zend/View/Helper/FieldsetTest.php	(working copy)
@@ -93,7 +93,7 @@
     }
 
     /**
-     * @see ZF-2913
+     * @group ZF-2913
      */
     public function testEmptyLegendShouldNotRenderLegendTag()
     {
Index: tests/Zend/View/Helper/FormRadioTest.php
===================================================================
--- tests/Zend/View/Helper/FormRadioTest.php	(revision 23558)
+++ tests/Zend/View/Helper/FormRadioTest.php	(working copy)
@@ -110,7 +110,7 @@
     }
 
     /**
-     * @see ZF-3206
+     * @group ZF-3206
      */
     public function testSpecifyingLabelPlacementShouldNotOverwriteValue()
     {
Index: tests/Zend/View/Helper/FormCheckboxTest.php
===================================================================
--- tests/Zend/View/Helper/FormCheckboxTest.php	(revision 23558)
+++ tests/Zend/View/Helper/FormCheckboxTest.php	(working copy)
@@ -144,7 +144,7 @@
     }
 
     /**
-     * @see ZF-2230
+     * @group ZF-2230
      */
     public function testDoesNotRenderHiddenElementsForCheckboxArray()
     {
@@ -156,7 +156,7 @@
     }
 
     /**
-     * @see ZF-3149
+     * @group ZF-3149
      */
     public function testShouldRenderHiddenElementShowingUncheckedOptionForNonArrayNames()
     {
@@ -213,7 +213,7 @@
     }
 
     /**
-     * @see ZF-3149
+     * @group ZF-3149
      */
     public function testCheckedAttributeNotRenderedIfItEvaluatesToFalse()
     {
@@ -228,7 +228,7 @@
     }
 
     /**
-     * @see ZF-3149
+     * @group ZF-3149
      */
     public function testShouldCheckValueIfValueMatchesCheckedOption()
     {
@@ -242,7 +242,7 @@
     }
 
     /**
-     * @see ZF-3149
+     * @group ZF-3149
      */
     public function testShouldOnlySetValueIfValueMatchesCheckedOption()
     {
@@ -251,7 +251,7 @@
     }
 
     /**
-     * @see ZF-3149
+     * @group ZF-3149
      */
     public function testShouldNotCheckValueIfValueDoesNotMatchCheckedOption()
     {
@@ -274,7 +274,7 @@
     }
 
    /**
-    * @see ZF-6467
+    * @group ZF-6467
     */
    public function testShouldNotShowHiddenFieldIfDisableIsTrue()
    {
Index: tests/Zend/View/Helper/HtmlListTest.php
===================================================================
--- tests/Zend/View/Helper/HtmlListTest.php	(revision 23558)
+++ tests/Zend/View/Helper/HtmlListTest.php	(working copy)
@@ -134,7 +134,7 @@
     }
 
     /*
-     * @see ZF-5018
+     * @group ZF-5018
      */
     public function testMakeNestedUnorderedList()
     {
@@ -150,7 +150,7 @@
     }
 
     /*
-     * @see ZF-5018
+     * @group ZF-5018
      */
     public function testMakeNestedDeepUnorderedList()
     {
@@ -193,7 +193,7 @@
     }
 
     /**
-     * @see ZF-2527
+     * @group ZF-2527
      */
     public function testEscapeFlagHonoredForMultidimensionalLists()
     {
@@ -207,7 +207,7 @@
     }
 
     /**
-     * @see ZF-2527
+     * @group ZF-2527
      * Added the s modifier to match newlines after @see ZF-5018
      */
     public function testAttribsPassedIntoMultidimensionalLists()
@@ -223,7 +223,7 @@
     }
 
     /**
-     * @see ZF-2870
+     * @group ZF-2870
      */
     public function testEscapeFlagShouldBePassedRecursively()
     {
Index: tests/Zend/View/Helper/FormPasswordTest.php
===================================================================
--- tests/Zend/View/Helper/FormPasswordTest.php	(revision 23558)
+++ tests/Zend/View/Helper/FormPasswordTest.php	(working copy)
@@ -74,7 +74,7 @@
     }
 
     /**
-     * @see ZF-1666
+     * @group ZF-1666
      */
     public function testCanDisableElement()
     {
@@ -88,7 +88,7 @@
     }
 
     /**
-     * @see ZF-1666
+     * @group ZF-1666
      */
     public function testDisablingElementDoesNotRenderHiddenElements()
     {
@@ -121,7 +121,7 @@
     }
 
     /**
-     * @see ZF-2860
+     * @group ZF-2860
      */
     public function testShouldRenderValueWhenRenderPasswordFlagPresentAndTrue()
     {
@@ -130,7 +130,7 @@
     }
 
     /**
-     * @see ZF-2860
+     * @group ZF-2860
      */
     public function testRenderPasswordAttribShouldNeverBeRendered()
     {
Index: tests/Zend/View/Helper/PartialLoopTest.php
===================================================================
--- tests/Zend/View/Helper/PartialLoopTest.php	(revision 23558)
+++ tests/Zend/View/Helper/PartialLoopTest.php	(working copy)
@@ -269,8 +269,8 @@
     }
 
     /**
-     * @see ZF-3350
-     * @see ZF-3352
+     * @group ZF-3350
+     * @group ZF-3352
      */
     public function testShouldNotCastToArrayIfObjectIsTraversable()
     {
@@ -296,7 +296,7 @@
     }
 
     /**
-     * @see ZF-3083
+     * @group ZF-3083
      */
     public function testEmptyArrayPassedToPartialLoopShouldNotThrowException()
     {
@@ -319,7 +319,7 @@
     }
 
     /**
-     * @see ZF-2737
+     * @group ZF-2737
      * @link http://framework.zend.com/issues/browse/ZF-2737
      */
     public function testPartialLoopIncramentsPartialCounter()
@@ -344,7 +344,7 @@
     }
 
     /**
-     * @see ZF-5174
+     * @group ZF-5174
      * @link http://framework.zend.com/issues/browse/ZF-5174
      */
     public function testPartialLoopPartialCounterResets()
Index: tests/Zend/View/Helper/FormTest.php
===================================================================
--- tests/Zend/View/Helper/FormTest.php	(revision 23558)
+++ tests/Zend/View/Helper/FormTest.php	(working copy)
@@ -99,7 +99,7 @@
     }
 
     /**
-     * @see ZF-3832
+     * @group ZF-3832
      */
     public function testEmptyIdShouldNotRenderIdAttribute()
     {
Index: tests/Zend/Db/Adapter/Pdo/MysqlTest.php
===================================================================
--- tests/Zend/Db/Adapter/Pdo/MysqlTest.php	(revision 23558)
+++ tests/Zend/Db/Adapter/Pdo/MysqlTest.php	(working copy)
@@ -105,7 +105,7 @@
     /**
      * Ensures that driver_options are properly passed along to PDO
      *
-     * @see    http://framework.zend.com/issues/browse/ZF-285
+     * @group ZF-285
      * @return void
      */
     public function testAdapterDriverOptions()
Index: tests/Zend/Db/Adapter/Pdo/SqliteTest.php
===================================================================
--- tests/Zend/Db/Adapter/Pdo/SqliteTest.php	(revision 23558)
+++ tests/Zend/Db/Adapter/Pdo/SqliteTest.php	(working copy)
@@ -186,8 +186,7 @@
     }
 
     /**
-     * @return void
-     * @see    http://framework.zend.com/issues/browse/ZF-2293
+     * @group ZF-2293
      */
     public function testAdapterSupportsLengthInTableMetadataForVarcharFields()
     {
Index: tests/Zend/Dojo/View/Helper/DojoTest.php
===================================================================
--- tests/Zend/Dojo/View/Helper/DojoTest.php	(revision 23558)
+++ tests/Zend/Dojo/View/Helper/DojoTest.php	(working copy)
@@ -799,7 +799,6 @@
     }
 
     /**
-     * @see   ZF-3962
      * @group ZF-3962
      */
     public function testHelperShouldAllowDisablingParseOnLoadWithDeclarativeStyle()
Index: tests/Zend/Dojo/View/Helper/CheckBoxTest.php
===================================================================
--- tests/Zend/Dojo/View/Helper/CheckBoxTest.php	(revision 23558)
+++ tests/Zend/Dojo/View/Helper/CheckBoxTest.php	(working copy)
@@ -142,7 +142,6 @@
     }
 
     /**
-     * @see ZF-4006
      * @group ZF-4006
      */
     public function testElementShouldUseCheckedValueForCheckboxInput()
Index: tests/Zend/Dojo/Form/Element/RadioButtonTest.php
===================================================================
--- tests/Zend/Dojo/Form/Element/RadioButtonTest.php	(revision 23558)
+++ tests/Zend/Dojo/Form/Element/RadioButtonTest.php	(working copy)
@@ -182,7 +182,7 @@
     }
 
     /**#+
-     * @see ZF-3286
+     * @group ZF-3286
      */
     public function testShouldRegisterInArrayValidatorByDefault()
     {
Index: tests/Zend/Dojo/Form/Element/ComboBoxTest.php
===================================================================
--- tests/Zend/Dojo/Form/Element/ComboBoxTest.php	(revision 23558)
+++ tests/Zend/Dojo/Form/Element/ComboBoxTest.php	(working copy)
@@ -146,7 +146,7 @@
     }
 
     /**#@+
-     * @see ZF-3286
+     * @group ZF-3286
      */
     public function testShouldNeverRegisterInArrayValidatorAutomatically()
     {
Index: tests/Zend/Dojo/Form/Element/FilteringSelectTest.php
===================================================================
--- tests/Zend/Dojo/Form/Element/FilteringSelectTest.php	(revision 23558)
+++ tests/Zend/Dojo/Form/Element/FilteringSelectTest.php	(working copy)
@@ -116,7 +116,7 @@
     }
 
     /**#@+
-     * @see ZF-3286
+     * @group ZF-3286
      */
     public function testShouldRegisterInArrayValidatorWhenNoStoreProvided()
     {
Index: tests/Zend/XmlRpc/ClientTest.php
===================================================================
--- tests/Zend/XmlRpc/ClientTest.php	(revision 23558)
+++ tests/Zend/XmlRpc/ClientTest.php	(working copy)
@@ -153,7 +153,7 @@
     }
 
     /**
-     * @see ZF-2090
+     * @group ZF-2090
      */
     public function testSuccessfullyDetectsEmptyArrayParameterAsArray()
     {
@@ -174,7 +174,7 @@
     }
 
     /**
-     * @see ZF-1412
+     * @group ZF-1412
      *
      * @return void
      */
Index: tests/Zend/XmlRpc/ServerTest.php
===================================================================
--- tests/Zend/XmlRpc/ServerTest.php	(revision 23558)
+++ tests/Zend/XmlRpc/ServerTest.php	(working copy)
@@ -601,7 +601,7 @@
     }
 
     /**
-     * @see ZF-2872
+     * @group ZF-2872
      */
     public function testCanMarshalBase64Requests()
     {
Index: tests/Zend/Captcha/FigletTest.php
===================================================================
--- tests/Zend/Captcha/FigletTest.php	(revision 23558)
+++ tests/Zend/Captcha/FigletTest.php	(working copy)
@@ -113,7 +113,6 @@
     }
 
     /*
-     * @see ZF-8268
      * @group ZF-8268
      */
     public function testLabelIdIsCorrect()
