Index: Fieldset.php
--- Fieldset.php Ausgangspunkt (BASE)
+++ Fieldset.php Lokal geändert (ausgehend von LOCAL)
@@ -131,7 +131,9 @@
         $name    = $element->getFullyQualifiedName();
         $id      = (string)$element->getId();
 
-        if (!array_key_exists('id', $attribs) && '' !== $id) {
+        if ((!array_key_exists('id', $attribs) || $attribs['id'] == $id)
+            && '' !== $id)
+        {
             $attribs['id'] = 'fieldset-' . $id;
         }
 
