Index: tests/Zend/Form/Element/MultiCheckboxTest.php
===================================================================
--- tests/Zend/Form/Element/MultiCheckboxTest.php	(revision 25022)
+++ tests/Zend/Form/Element/MultiCheckboxTest.php	(working copy)
@@ -326,9 +326,9 @@
         $this->element->setLabel('Foo');
 
         $expected = '<dt id="foo-label"><label class="optional">Foo</label></dt>'
-                  . "\n"
+                  . PHP_EOL
                   . '<dd id="foo-element">'
-                  . "\n"
+                  . PHP_EOL
                   . '</dd>';
         $this->assertSame($expected, $this->element->render($this->getView()));
     }
@@ -341,7 +341,7 @@
         $this->element->setLabel('Foo')->removeDecorator('label');
 
         $expected = '<dd id="foo-element">'
-                  . "\n"
+                  . PHP_EOL
                   . '</dd>';
         $this->assertSame($expected, $this->element->render($this->getView()));
     }