Index: tests/Zend/Navigation/ContainerTest.php
===================================================================
--- tests/Zend/Navigation/ContainerTest.php	(revision 24447)
+++ tests/Zend/Navigation/ContainerTest.php	(working copy)
@@ -366,6 +366,20 @@
         $this->assertEquals(3, count($nav),
                             'Expected 3 pages, found ' . count($nav));
     }
+    
+    /**
+     * @group ZF-9815
+     */
+    public function testAddPagesShouldWorkWithNavigationContainer()
+    {        
+        $nav = new Zend_Navigation();
+        $nav->addPages($this->_getFindByNavigation());
+        
+        $this->assertEquals(3, count($nav),
+                            'Expected 3 pages, found ' . count($nav));
+        
+        $this->assertEquals($nav->toArray(), $this->_getFindByNavigation()->toArray());
+    }
 
     public function testAddPagesShouldThrowExceptionWhenGivenString()
     {
