--- Image.php	2010-06-17 13:25:20.000000000 -0400
+++ Image.php	2010-06-17 13:25:16.000000000 -0400
@@ -580,7 +580,11 @@
         foreach (new DirectoryIterator($imgdir) as $file) {
             if (!$file->isDot() && !$file->isDir()) {
                 if ($file->getMTime() < $expire) {
-                    unlink($file->getPathname());
+                    $len = strlen($this->_suffix);
+                    // only deletes files ending with $this->_suffix
+                    if (substr($file->getFilename(), -($len), $len) == $this->_suffix) {
+                        unlink($file->getPathname());
+                    }
                 }
             }
         }
