Index: library/Zend/Db/Table/Abstract.php
===================================================================
--- library/Zend/Db/Table/Abstract.php	(revision 24628)
+++ library/Zend/Db/Table/Abstract.php	(working copy)
@@ -1178,7 +1178,7 @@
      * @param  array|string $where SQL WHERE clause(s).
      * @return int          The number of rows deleted.
      */
-    public function delete($where)
+    public function delete($where='')
     {
         $tableSpec = ($this->_schema ? $this->_schema . '.' : '') . $this->_name;
         return $this->_db->delete($tableSpec, $where);
Index: tests/Zend/Db/Table/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/TestCommon.php	(revision 24628)
+++ tests/Zend/Db/Table/TestCommon.php	(working copy)
@@ -1006,6 +1006,17 @@
 
         $this->assertEquals(0, count($this->_table['bugs']->find(array(1, 3))));
     }
+    
+    /**
+     * @group ZF-12052
+     */
+    public function testTableDeleteEverything()
+    {
+        $beforeCount = count($this->_table['bugs']->fetchAll());
+        $this->assertEquals($beforeCount, $this->_table['bugs']->delete());
+        $this->assertEquals(0, count($this->_table['bugs']->fetchAll()));
+        
+    }
 
     public function testTableCreateRow()
     {
