Index: Adapter/Activemq.php
===================================================================
--- Adapter/Activemq.php	(revision 22080)
+++ Adapter/Activemq.php	(working copy)
@@ -80,7 +80,7 @@
         if (array_key_exists('stompClient', $options)) {
             $this->_client = $options['stompClient'];
         } else {
-            $this->_client = new Zend_Queue_Stomp_Client($options['scheme'], $options['host'], $options['port']);
+            $this->_client = new Zend_Queue_Stomp_Client($options['scheme'], $options['host'], $options['port'], 'Zend_Queue_Stomp_Client_Connection', 'Zend_Queue_Stomp_Frame', $options);
         }
 
         $connect = $this->_client->createFrame();
Index: Stomp/Client.php
===================================================================
--- Stomp/Client.php	(revision 22080)
+++ Stomp/Client.php	(working copy)
@@ -37,6 +37,12 @@
      * @var array
      */
     protected $_connection;
+    
+    /**
+     * Connection options
+     * @var array
+     */
+    protected $_options;
 
     /**
      * Add a server to connections
@@ -48,12 +54,14 @@
     public function __construct(
         $scheme = null, $host = null, $port = null,
         $connectionClass = 'Zend_Queue_Stomp_Client_Connection',
-        $frameClass = 'Zend_Queue_Stomp_Frame'
+        $frameClass = 'Zend_Queue_Stomp_Frame',
+        $options = array()
     ) {
         if (($scheme !== null)
             && ($host !== null)
             && ($port !== null)
         ) {
+        	$this->_options = $options;        	
             $this->addConnection($scheme, $host, $port, $connectionClass);
             $this->getConnection()->setFrameClass($frameClass);
         }
@@ -94,7 +102,7 @@
 
         $connection = new $class();
 
-        if ($connection->open($scheme, $host, $port)) {
+        if ($connection->open($scheme, $host, $port, $this->_options)) {
             $this->setConnection($connection);
             return true;
         }
Index: Stomp/Client/Connection.php
===================================================================
--- Stomp/Client/Connection.php	(revision 22080)
+++ Stomp/Client/Connection.php	(working copy)
@@ -76,7 +76,7 @@
             throw new Zend_Queue_Exception("Unable to connect to $str; error = $errstr ( errno = $errno )");
         }
 
-        stream_set_blocking($this->_socket, 0); // non blocking
+//        stream_set_blocking($this->_socket, 0); // non blocking
 
         if (!isset($options['timeout_sec'])) {
             $options['timeout_sec'] = self::READ_TIMEOUT_DEFAULT_SEC;
