Index: Zend/Controller/Plugin/Broker.php
===================================================================
--- Zend/Controller/Plugin/Broker.php	(revision 19920)
+++ Zend/Controller/Plugin/Broker.php	(working copy)
@@ -339,6 +339,7 @@
                     throw new Zend_Controller_Exception($e->getMessage(), $e->getCode(), $e);
                 } else {
                     $this->getResponse()->setException($e);
+                    $this->getRequest()->setDispatched(false);
                 }
             }
         }
