Index: Response.php
===================================================================
--- Response.php	(revision 24211)
+++ Response.php	(working copy)
@@ -173,7 +173,6 @@
     {
         if ($this->isError()) {
             $response = array(
-                'result' => null,
                 'error'  => $this->getError()->toArray(),
                 'id'     => $this->getId(),
             );
@@ -181,7 +180,6 @@
             $response = array(
                 'result' => $this->getResult(),
                 'id'     => $this->getId(),
-                'error'  => null,
             );
         }
 
