Index: library/Zend/File/Transfer/Adapter/Http.php
===================================================================
--- library/Zend/File/Transfer/Adapter/Http.php	(revision 25051)
+++ library/Zend/File/Transfer/Adapter/Http.php	(working copy)
@@ -128,6 +128,10 @@
         // Workaround for a PHP error returning empty $_FILES when form data exceeds php settings
         if (empty($this->_files) && ($content > 0)) {
             if (is_array($files)) {
+                if (0 === count($files)) {
+                    return false;
+                }
+
                 $files = current($files);
             }
 
