--- library/Zend/View/Helper/HeadScript.php	(revision 14266)
+++ library/Zend/View/Helper/HeadScript.php	(working copy)
@@ -448,11 +448,12 @@
         $items = array();
         $this->getContainer()->ksort();
         foreach ($this as $item) {
-            if (!$this->_isValid($item)) {
+            if (!$this->_isValid($item) || isset($item->passed)) {
                 continue;
             }
 
-            $items[] = $this->itemToString($item, $indent, $escapeStart, $escapeEnd);
+			$items[] = $this->itemToString($item, $indent, $escapeStart, $escapeEnd);
+			$item->passed = true;
         }
 
         $return = implode($this->getSeparator(), $items);
