Index: tests/Zend/Crypt/Rsa/RsaTest.php
===================================================================
--- tests/Zend/Crypt/Rsa/RsaTest.php	(revision 24790)
+++ tests/Zend/Crypt/Rsa/RsaTest.php	(working copy)
@@ -40,14 +40,8 @@
 
     public function setUp()
     {
-        try {
-            $rsaObject = new Zend_Crypt_Rsa();
-        } catch (Zend_Crypt_Rsa_Exception $e) {
-            if (strpos($e->getMessage(), 'requires openssl extension') !== false) {
-                $this->markTestSkipped($e->getMessage());
-            } else {
-                throw $e;
-            }
+        if (!extension_loaded('openssl')) {
+            $this->markTestSkipped('Zend_Crypt_Rsa requires openssl extension to be loaded.');
         }
 
         $this->_testPemString = <<<RSAKEY
