Index: incubator/tools/http_server/src/Request.php
===================================================================
--- incubator/tools/http_server/src/Request.php	(revision 1546)
+++ incubator/tools/http_server/src/Request.php	(working copy)
@@ -19,6 +19,7 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 
+require_once 'Zend/Uri/Http.php';
 
 /**
  * @category   Zend
@@ -133,6 +134,15 @@
 
 		list( $this->method, $uri, $this->protocol_version ) = explode( " ", array_shift( $lines ) );
 
+		// If we got a full URI, extract the path out of it
+		if (Zend_Uri_Http::check($uri)) {
+			$uriobj = Zend_Uri::factory($uri);
+			$uri = $uriobj->getPath();
+			if ($uriobj->getFragment()) $uri .= '#' . $uriobj->getFragment();
+			if ($uriobj->getQuery()) $uri .= '?' . $uriobj->getQuery();
+			unset($uriobj);
+		}
+		
 		$this->setURI( $uri );
 
 		$headers = array();
