*** Page.php	2009-04-28 17:43:16.000000000 +0200
--- Page.php.new	2009-04-28 17:43:24.000000000 +0200
***************
*** 436,442 ****
          if ($this->_contents == '') {
              return;
          }
! 
          if ($this->_pageDictionary->Contents->getType() != Zend_Pdf_Element::TYPE_ARRAY) {
              /**
               * It's a stream object.
--- 436,446 ----
          if ($this->_contents == '') {
              return;
          }
!         
!         if (is_null($this->_pageDictionary->Contents)){
!             $this->_pageDictionary->Contents = new Zend_Pdf_Element_Array();
!         }
!         
          if ($this->_pageDictionary->Contents->getType() != Zend_Pdf_Element::TYPE_ARRAY) {
              /**
               * It's a stream object.
