Index: Zend/Application/Bootstrap/Bootstrap.php
===================================================================
--- Zend/Application/Bootstrap/Bootstrap.php	(revision 22347)
+++ Zend/Application/Bootstrap/Bootstrap.php	(working copy)
@@ -59,7 +59,7 @@
     {
         parent::__construct($application);
 
-        if ($application->hasOption('resourceloader')) {
+        if (!$application->hasOption('resourceloader')) {
             $this->setOptions(array(
                 'resourceloader' => $application->getOption('resourceloader')
             ));
