Index: Abstract.php
===================================================================
--- Abstract.php	(revision 2998)
+++ Abstract.php	(working copy)
@@ -241,6 +241,10 @@
                     header($header['name'] . ': ' . $header['value'], false);
                 }
             }
+            if (!$httpCodeSent) {
+                header('HTTP/1.1 ' . $this->_httpResponseCode);
+                $httpCodeSent = true;
+            }
         }
 
         return $this;
