Index: Zend/Loader.php
===================================================================
--- Zend/Loader.php	(revision 11)
+++ Zend/Loader.php	(working copy)
@@ -80,7 +80,9 @@
             self::loadFile($file, $dirs, true);
         } else {
             self::_securityCheck($file);
-            include_once $file;
+            if (self::isReadable($file)) {
+                include_once $file;
+            }
         }
 
         if (!class_exists($class, false) && !interface_exists($class, false)) {
@@ -131,10 +133,12 @@
         /**
          * Try finding for the plain filename in the include_path.
          */
-        if ($once) {
-            include_once $filename;
-        } else {
-            include $filename;
+        if (self::isReadable($filename)) {
+            if ($once) {
+                include_once $filename;
+            } else {
+                include $filename;
+            }
         }
 
         /**

