--- tests/Zend/Dojo/Form/Element/SubmitButtonTest.php	(revision 23940)
+++ tests/Zend/Dojo/Form/Element/SubmitButtonTest.php	(working copy)
@@ -40,6 +40,9 @@
 /** Zend_Dojo_View_Helper_Dojo */
 require_once 'Zend/Dojo/View/Helper/Dojo.php';
 
+/** Zend_Dojo_Form */
+require_once 'Zend/Dojo/Form.php';
+
 /**
  * Test class for Zend_Dojo_Form_Element_SubmitButton.
  *
@@ -192,6 +195,55 @@
         $html = $this->element->render();
         $this->assertRegexp('/<input[^>]*(value="Label!")/', $html, $html);
     }
+    
+     /**
+      * @group ZF-5017
+      */
+     public function testIsCheckedReturnsTrueForFormWithoutNameAttribute()
+     {
+         $view = $this->getView();
+         $form = new Zend_Dojo_Form();
+         $form->setView($view);
+         
+         $form->addElement($this->element);
+         
+         $this->assertEquals(null, $form->getName());
+         
+         $this->assertFalse($form->getElement('foo')->isChecked());
+         
+         $form->getElement('foo')->setValue('foo');
+         
+         $this->assertTrue($form->getElement('foo')->isChecked());
+         
+         $html = $form->render();
+         
+         $this->assertRegexp('/<input[^>]*(value="foo")/', $html, $html);
+     }
+     
+     /**
+      * @group ZF-5017
+      */
+     public function testIsCheckedReturnsTrueForFormWithDifferentNameAttribute()
+     {
+         $view = $this->getView();
+         $form = new Zend_Dojo_Form();
+         $form->setName('testform');
+         $form->setView($view);
+         
+         $form->addElement($this->element);
+         
+         $this->assertEquals('testform', $form->getName());
+         
+         $this->assertFalse($form->getElement('foo')->isChecked());
+         
+         $form->getElement('foo')->setValue('foo');
+         
+         $this->assertTrue($form->getElement('foo')->isChecked());
+         
+         $html = $form->render();
+         
+         $this->assertRegexp('/<input[^>]*(value="foo")/', $html, $html);
+     }
 }
 
 // Call Zend_Dojo_Form_Element_SubmitButtonTest::main() if this source file is executed directly.
