Index: ../../library/Zend/View/Abstract.php
===================================================================
--- ../../library/Zend/View/Abstract.php	(revision 18731)
+++ ../../library/Zend/View/Abstract.php	(working copy)
@@ -557,6 +557,23 @@
     {
         return $this->getPluginLoader('helper')->getPaths();
     }
+    
+    /**
+     * Registers a helper object, bypassing plugin loader
+     *
+     * @param Zend_View_Helper_Abstract|object $helper
+     * @param string $name
+     * @return Zend_View_Abstract
+     */
+    public function registerHelper($helper, $name)
+    {
+        if (!is_object($helper)) {
+            throw new Zend_View_Exception('View helper must be an object.', $this);
+        }
+        $name = ucfirst($name);
+        $this->_helper[$name] = $helper;
+        return $this;
+    }
 
     /**
      * Get a helper by name
