Index: Abstract.php
===================================================================
--- Abstract.php	(revision 22396)
+++ Abstract.php	(working copy)
@@ -62,8 +62,9 @@
 
     /**
      * Maximum of the transaction log
+     * @var integer
      */
-    const MAXIMUM_LOG = 64;
+    protected $_maximumLog = 64;
 
 
     /**
@@ -159,7 +160,18 @@
         $this->_disconnect();
     }
 
+    /**
+     * Set the maximum log size 
+     * 
+     * @param integer $maximumLog Maximum log size
+     * @return void
+     */
+    public function setMaximumLog($maximumLog)
+    {
+        $this->_maximumLog = $maximumLog;
+    }
 
+
     /**
      * Create a connection to the remote host
      *
@@ -219,7 +231,7 @@
      */
     protected function _addLog($value)
     {
-        if (count($this->_log) >= self::MAXIMUM_LOG) {
+        if ($this->_maximumLog >= 0 && count($this->_log) >= $this->_maximumLog) {
             array_shift($this->_log);
         }
 
