diff --git a/includes/Zend/Log/Writer/Syslog.php b/includes/Zend/Log/Writer/Syslog.php
index 3409ae5..c5752b5 100644
--- a/includes/Zend/Log/Writer/Syslog.php
+++ b/includes/Zend/Log/Writer/Syslog.php
@@ -170,6 +170,12 @@ class Zend_Log_Writer_Syslog extends Zend_Log_Writer_Abstract
             $this->_initializeSyslog();
         }
 
-        syslog($priority, $event['message']);
+		$message = $event['message'];
+
+		if ($this->_formatter instanceof Zend_Log_Formatter_Interface) {
+			$message = $this->_formatter->format($event);
+		}
+
+        syslog($priority, $message);
     }
 }
