Index: tests/Zend/Filter/StripTagsTest.php
===================================================================
--- tests/Zend/Filter/StripTagsTest.php	(revision 22549)
+++ tests/Zend/Filter/StripTagsTest.php	(revision )
@@ -602,7 +602,22 @@
         $expected = '<img width="10" height="10" src=\'wont_be_matched.jpg\'>';
         $this->assertEquals($expected, $filter->filter($input));
     }
+
+    /**
+     * Ensures that a comment at the end of input string, which is not closed
+     * is correctly stripped
+     *
+     * @see    ZF-10256
+     * @link   http://framework.zend.com/issues/browse/ZF-10256
+     * @return void
+     */
+    public function testNotClosedHtmlCommentAtEndOfString()
+    {
+        $input    = 'text<!-- not closed comment at the end';
+        $expected = 'text';
+        $this->assertEquals($expected, $this->_filter->filter($input));
-}
+    }
+}
 
 // Call Zend_Filter_StripTagsTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD === 'Zend_Filter_StripTagsTest::main') {
Index: library/Zend/Filter/StripTags.php
===================================================================
--- library/Zend/Filter/StripTags.php	(revision 22547)
+++ library/Zend/Filter/StripTags.php	(revision )
@@ -244,7 +244,12 @@
             $pos   = strrpos($value, '<!--');
             $start = substr($value, 0, $pos);
             $value = substr($value, $pos);
+            // If there is no comment closing tag, strip whole text
+            if (!preg_match('/--\s*>/s', $value)) {
+                $value = '';
+            } else {
-            $value = preg_replace('/<(?:!(?:--[\s\S]*?--\s*)?(>))/s', '',  $value);
+                $value = preg_replace('/<(?:!(?:--[\s\S]*?--\s*)?(>))/s', '',  $value);
+            }
             $value = $start . $value;
         }
 
