--- library/Zend/Controller/Action/Helper/Redirector.php	(revision 23864)
+++ library/Zend/Controller/Action/Helper/Redirector.php	(working copy)
@@ -294,9 +294,9 @@
             }
         }
 
-        $params['module']     = $module;
-        $params['controller'] = $controller;
-        $params['action']     = $action;
+        $params[$request->getModuleKey()]     = $module;
+        $params[$request->getControllerKey()] = $controller;
+        $params[$request->getActionKey()]     = $action;
 
         $router = $this->getFrontController()->getRouter();
         $url    = $router->assemble($params, 'default', true);
