Index: tests/Zend/Soap/ClientTest.php
===================================================================
--- tests/Zend/Soap/ClientTest.php	(Revision 24556)
+++ tests/Zend/Soap/ClientTest.php	(Arbeitskopie)
@@ -95,6 +95,8 @@
                                 'cache_wsdl'     => 8,
                                 'features'       => 4,
 
+                                'typemap'        => array(array()),
+
                                 'compression'    => SOAP_COMPRESSION_ACCEPT | SOAP_COMPRESSION_GZIP | 5);
 
         $client->setOptions($nonWsdlOptions);
@@ -127,6 +129,8 @@
 
                              'stream_context' => $ctx,
 
+                             'typemap'        => array(array()),
+
                              'compression'    => SOAP_COMPRESSION_ACCEPT | SOAP_COMPRESSION_GZIP | 5);
 
         $client1->setOptions($wsdlOptions);
Index: library/Zend/Soap/Client.php
===================================================================
--- library/Zend/Soap/Client.php	(Revision 24556)
+++ library/Zend/Soap/Client.php	(Arbeitskopie)
@@ -85,6 +85,7 @@
     protected $_passphrase          = null;
     protected $_compression         = null;
     protected $_connection_timeout  = null;
+    protected $_typemap             = null;
     protected $_stream_context      = null;
     protected $_features            = null;
     protected $_cache_wsdl          = null;
@@ -260,6 +261,9 @@
                 case 'features':
                     $this->setSoapFeatures($value);
                     break;
+                case 'typemap':
+                    $this->setTypemap($value);
+                    break;
                 case 'cache_wsdl':
                     $this->setWsdlCache($value);
                     break;
@@ -314,6 +318,7 @@
         $options['stream_context'] = $this->getStreamContext();
         $options['cache_wsdl']     = $this->getWsdlCache();
         $options['features']       = $this->getSoapFeatures();
+        $options['typemap']        = $this->getTypemap();
         $options['user_agent']     = $this->getUserAgent();
 
         foreach ($options as $key => $value) {
@@ -857,6 +862,31 @@
     }
 
     /**
+     * Set the SOAP Typemap option.
+     *
+     * @param  array $typemap
+     * @return Zend_Soap_Client
+     */
+    public function setTypemap($typemap)
+    {
+        $this->_typemap = $typemap;
+
+        $this->_soapClient = null;
+
+        return $this;
+    }
+
+    /**
+     * Return current SOAP Typemap option
+     *
+     * @return array
+     */
+    public function getTypemap()
+    {
+        return $this->_typemap;
+    }
+
+    /**
      * Set the SOAP Wsdl Caching Options
      *
      * @param string|int|boolean|null $caching
