Index: library/Zend/Db/Table/Abstract.php
===================================================================
--- library/Zend/Db/Table/Abstract.php	(revision 20439)
+++ library/Zend/Db/Table/Abstract.php	(working copy)
@@ -1365,7 +1365,8 @@
             $select->limit(1);
 
         } else {
-            $select = $where->limit(1);
+            $select = clone $where;
+            $select->limit(1);
         }
 
         $rows = $this->_fetch($select);
