Index: tests/Zend/Db/Profiler/StaticTest.php
===================================================================
--- tests/Zend/Db/Profiler/StaticTest.php	(revision 20096)
+++ tests/Zend/Db/Profiler/StaticTest.php	(revision )
@@ -496,6 +496,8 @@
         $this->assertEquals(count($queries), count($queryProfiles));
 
         foreach ($queryProfiles as $queryId => $queryProfile) {
+            $this->assertNotNull($queryProfile->getStartedMicrotime());
+
             $this->assertTrue(isset($queries[$queryId]));
             $this->assertEquals($queries[$queryId]['sql'], $queryProfile->getQuery());
             $this->assertEquals($queries[$queryId]['typeExpected'], $queryProfile->getQueryType());
Index: library/Zend/Db/Profiler/Query.php
===================================================================
--- library/Zend/Db/Profiler/Query.php	(revision 20096)
+++ library/Zend/Db/Profiler/Query.php	(revision )
@@ -195,5 +195,19 @@
 
         return $this->_endedMicrotime - $this->_startedMicrotime;
     }
+
+    /**
+     * Get the time (in seconds) when the profiler started running.
+     *
+     * @return bool|float
+     */
+    public function getStartedMicrotime()
+    {
+        if(null === $this->_startedMicrotime) {
+            return false;
-}
+        }
 
+        return $this->_startedMicrotime;
+    }
+}
+
