Index: library/Zend/Controller/Router/Route/Chain.php
===================================================================
--- library/Zend/Controller/Router/Route/Chain.php	(revision 24596)
+++ library/Zend/Controller/Router/Route/Chain.php	(working copy)
@@ -72,9 +72,10 @@
      */
     public function match($request, $partial = null)
     {
-        $path    = trim($request->getPathInfo(), self::URI_DELIMITER);
-        $subPath = $path;
-        $values  = array();
+        $path      = trim($request->getPathInfo(), self::URI_DELIMITER);
+        $subPath   = $path;
+        $values    = array();
+        $numRoutes = count($this->_routes);
 
         foreach ($this->_routes as $key => $route) {
             if ($key > 0
@@ -99,7 +100,7 @@
                 $match = $request;
             }
 
-            $res = $route->match($match, true);
+            $res = $route->match($match, true, ($key == $numRoutes - 1));
             if ($res === false) {
                 return false;
             }
