Index: Abstract.php
===================================================================
--- Abstract.php	(revision 4462)
+++ Abstract.php	(working copy)
@@ -657,7 +657,8 @@
 
         $select = $db->select()
             ->from(array('i' => $interName), array())
-            ->join(array('m' => $matchName), $joinCond, '*');
+            ->join(array('m' => $matchName), $joinCond, '*')
+            ->distinct();
 
         $callerMap = $this->_prepareReference($intersectionTable, $this->_getTable(), $callerRefRule);
 
