Index: Mysqli.php
===================================================================
--- Mysqli.php	(revision 19505)
+++ Mysqli.php	(working copy)
@@ -93,6 +93,16 @@
      * @var string
      */
     protected $_defaultStmtClass = 'Zend_Db_Statement_Mysqli';
+    
+    /**
+     * Keeps track of transaction nest level, to emulate mysql support, -1 meaning no transaction
+     * has begun, 0 meaning there is no nesting, 1 meaning there are 2 transactions, ad infintum
+     * 
+     * @var integer
+     */
+    protected $_transaction_depth = -1;
+    
+    protected $_should_emulate_nesting = false;
 
     /**
      * Quote a raw string.
@@ -417,6 +427,7 @@
      */
     protected function _beginTransaction()
     {
+        $this->_transaction_depth++;
         $this->_connect();
         $this->_connection->autocommit(false);
     }
@@ -428,6 +439,11 @@
      */
     protected function _commit()
     {
+        $this->_transaction_depth--;
+        if( $this->shouldEmulateNesting() )
+        {
+            return;
+        }
         $this->_connect();
         $this->_connection->commit();
         $this->_connection->autocommit(true);
@@ -440,6 +456,11 @@
      */
     protected function _rollBack()
     {
+        $this->_transaction_depth--;
+        if( $this->shouldEmulateNesting() )
+        {
+            return;
+        }
         $this->_connect();
         $this->_connection->rollback();
         $this->_connection->autocommit(true);
@@ -546,4 +567,14 @@
         $revision = (int) ($version % 100);
         return $major . '.' . $minor . '.' . $revision;
     }
+    
+    protected function shouldEmulateNesting()
+    {
+        return $this->_should_emulate_nesting && $this->isNested();   
+    }
+    
+    protected function isNested()
+    {
+        return $this->_transaction_depth != 0;
+    }
 }
