--- Zend/Paginator/Adapter/DbSelect.php	Sun Apr 10 14:03:02 2011
+++ Zend/Paginator/Adapter/DbSelect.php	Sun Apr 10 14:03:02 2011
@@ -237,7 +237,17 @@
                     $groupPart = $column;
                 }
             } else if (!empty($groupParts)) {
-                $groupPart = $db->quoteIdentifier($groupParts[0], true);
+                $column = $groupParts[0];
+                foreach($columnParts as $col) {
+                    if ($col[2] == $column) {
+                        array_pop($col);        // Remove alias
+                        $groupPart = $db->quoteIdentifier($col, true);
+                        break;
+                    }
+                }
+                if (!$groupPart) {
+                    $groupPart = $db->quoteIdentifier($groupParts[0], true);
+                }
             }

             /**
