Index: Zend/Db/Adapter/Pdo/Abstract.php
===================================================================
--- Zend/Db/Adapter/Pdo/Abstract.php	(revision 688)
+++ Zend/Db/Adapter/Pdo/Abstract.php	(working copy)
@@ -82,11 +82,19 @@
 
         // create PDO connection
         $q = $this->_profiler->queryStart('connect', Zend_Db_Profiler::CONNECT);
-        $this->_connection = new PDO(
-            $this->_dsn(),
-            $this->_config['username'],
-            $this->_config['password']
-        );
+
+        try {
+            $this->_connection = new PDO(
+                $this->_dsn(),
+                $this->_config['username'],
+                $this->_config['password']
+            );
+        }
+        catch (PDOException $e) {
+            echo 'Connection to the datbase failed';
+            exit;
+        }
+
         $this->_profiler->queryEnd($q);
 
         // force names to lower case
