--- library/ZendX/JQuery/View/Helper/AjaxLink.php	(revision 23775)
+++ library/ZendX/JQuery/View/Helper/AjaxLink.php	(working copy)
@@ -242,11 +242,11 @@
 
         switch($requestHandler) {
             case 'GET':
-                $js[] = sprintf("%s.get('%s', %s, function(data, textStatus) { %s }, '%s');return false;",
+                $js[] = sprintf('%s.get("%s", %s, function(data, textStatus) { %s }, "%s");return false;',
                     $jqHandler, $url, $params, implode(" ", $callbackCompleteJs), $options['dataType']);
                 break;
             case 'POST':
-                $js[] = sprintf("%s.post('%s', %s, function(data, textStatus) { %s }, '%s');return false;",
+                $js[] = sprintf('%s.post("%s", %s, function(data, textStatus) { %s }, "%s");return false;',
                     $jqHandler, $url, $params, implode(" ", $callbackCompleteJs), $options['dataType']);
                 break;
         }
@@ -305,4 +305,4 @@
         }
         return $requestHandler;
     }
-}
\ No newline at end of file
+}
