Index: library/Zend/Navigation/Page.php
===================================================================
--- library/Zend/Navigation/Page.php	(revision 24362)
+++ library/Zend/Navigation/Page.php	(working copy)
@@ -740,6 +740,10 @@
      */
     public function setVisible($visible = true)
     {
+        if (is_string($visible) && 'false' == strtolower($visible)) {
+            $visible = false;
+        }
+        
         $this->_visible = (bool) $visible;
         return $this;
     }
