Index: Mvc.php
--- Mvc.php Ausgangspunkt (BASE)
+++ Mvc.php Lokal geändert (ausgehend von LOCAL)
@@ -94,7 +94,6 @@
      */
     protected $_resetParams = true;
 
-        
     /**
      * Whether href should be encoded when assembling URL
      *
@@ -104,6 +103,13 @@
     protected $_encodeUrl = true;
 
     /**
+     * Whether this page should be considered active
+     *
+     * @var bool
+     */
+    protected $_active = null;
+
+    /**
      * Cached href
      *
      * The use of this variable minimizes execution time when getHref() is
@@ -137,7 +143,7 @@
      */
     public function isActive($recursive = false)
     {
-        if (!$this->_active) {
+        if (null === $this->_active) {
             $front     = Zend_Controller_Front::getInstance();
             $request   = $front->getRequest();
             $reqParams = array();
@@ -186,6 +192,8 @@
                 $this->_active = true;
                 return true;
             }
+            
+            $this->_active = false;
         }
 
         return parent::isActive($recursive);
