Index: Core.php
===================================================================
--- Core.php	(revision 4580)
+++ Core.php	(working copy)
@@ -146,7 +146,7 @@
             Zend_Cache::throwException('Options parameter must be an array');
         }  
         while (list($name, $value) = each($options)) {
-            $this->_setOption($name, $value);
+            $this->setOption($name, $value);
         }
         $this->_loggerSanity();
     }