Index: FormSelect.php
===================================================================
--- FormSelect.php	(revision 21318)
+++ FormSelect.php	(working copy)
@@ -97,13 +97,19 @@
         }
 
         // now start building the XHTML.
-        $disabled = '';
+        $xhtml = '';
+        $disabled = '';        
         if (true === $disable) {
             $disabled = ' disabled="disabled"';
+            if(!empty($value) && empty($multiple)) {
+            	$_value = array_shift($value);
+                $xhtml = $this->_hidden($name,$_value);
+                unset($_value);
+            }
         }
 
         // Build the surrounding select element first.
-        $xhtml = '<select'
+        $xhtml .= '<select'
                 . ' name="' . $this->view->escape($name) . '"'
                 . ' id="' . $this->view->escape($id) . '"'
                 . $multiple
