--- library/Zend/Soap/Wsdl/Strategy/DefaultComplexType.php	(revision 15010)
+++ library/Zend/Soap/Wsdl/Strategy/DefaultComplexType.php	(working copy)
@@ -49,7 +49,7 @@
         $all = $dom->createElement('xsd:all');
 
         foreach ($class->getProperties() as $property) {
-            if ($property->isPublic() && preg_match_all('/@var\s+([^\s]+)/m', $property->getDocComment(), $matches)) {
+            if ($property->isPublic() && preg_match('/@var\s+([^\s]+)/m', $property->getDocComment(), $matches)) {
 
                 /**
                  * @todo check if 'xsd:element' must be used here (it may not be compatible with using 'complexType'
@@ -57,7 +57,20 @@
                  */
                 $element = $dom->createElement('xsd:element');
                 $element->setAttribute('name', $property->getName());
-                $element->setAttribute('type', $this->getContext()->getType(trim($matches[1][0])));
+                
+                $propTypes = explode('|', $matches[1]);
+                $propType = null;
+                foreach ($propTypes as $tmpType) {
+                    if ($tmpType == 'null') {
+                        $element->setAttribute('nillable', 'true');
+                    } else if ($propType === null) {
+                        $propType = $tmpType;
+                    }
+                }
+                if ($propType !== null) {
+                    $element->setAttribute('type', $this->getContext()->getType($propType));
+                }
+                
                 $all->appendChild($element);
             }
         }
