Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/PackageTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/PackageTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/PackageTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Package.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_PackageTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Package
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Package();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testPackageGetterAndSetterPersistValue()
+    {
+        $this->_tag->setPackage('Foo');
+        $this->assertEquals('Foo', $this->_tag->getPackage());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/GroupTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/GroupTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/GroupTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Group.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_GroupTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Group
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Group();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testGroupGetterAndSetterPersistValue()
+    {
+        $this->_tag->setGroup('Foo');
+        $this->assertEquals('Foo', $this->_tag->getGroup());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/SeeTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/SeeTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/SeeTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/See.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_SeeTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_See
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_See();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testSeeGetterAndSetterPersistValue()
+    {
+        $this->_tag->setSee('Foo');
+        $this->assertEquals('Foo', $this->_tag->getSee());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/CategoryTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/CategoryTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/CategoryTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Category.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_CategoryTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Category
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Category();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testCategoryGetterAndSetterPersistValue()
+    {
+        $this->_tag->setCategory('Foo');
+        $this->assertEquals('Foo', $this->_tag->getCategory());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/ReturnTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/ReturnTest.php	(revision 20016)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/ReturnTest.php	(working copy)
@@ -17,7 +17,7 @@
  * @subpackage UnitTests
  * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
+ * @version    $Id$
  */
 
 
@@ -45,7 +45,7 @@
 {
 
     /**
-     * @var Zend_CodeGenerator_Php_Docblock_Tag
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Return
      */
     protected $_tag = null;
 
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/UsesTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/UsesTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/UsesTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Uses.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_UsesTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Uses
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Uses();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testUsesGetterAndSetterPersistValue()
+    {
+        $this->_tag->setUses('Foo');
+        $this->assertEquals('Foo', $this->_tag->getUses());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/ParamTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/ParamTest.php	(revision 20016)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/ParamTest.php	(working copy)
@@ -17,7 +17,7 @@
  * @subpackage UnitTests
  * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
+ * @version    $Id$
  */
 
 
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/AuthorTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/AuthorTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/AuthorTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Author.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_AuthorTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Author
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Author();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testAuthorGetterAndSetterPersistValue()
+    {
+        $this->_tag->setAuthor('Foo');
+        $this->assertEquals('Foo', $this->_tag->getAuthor());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/LicenseTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/LicenseTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/LicenseTest.php	(revision 0)
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/License.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_LicenseTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_License
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_License();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testUrlGetterAndSetterPersistValue()
+    {
+        $this->_tag->setUrl('http://Foo/');
+        $this->assertEquals('http://Foo/', $this->_tag->getUrl());
+    }
+
+    public function testLicenseProducesCorrectDocBlockLine()
+    {
+        $this->_tag->setUrl('http://foo/');
+        $this->_tag->setDescription('bar bar bar');
+        $this->assertEquals("@license http://foo/ bar bar bar\n", $this->_tag->generate());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/SubpackageTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/SubpackageTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/SubpackageTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_SubpackageTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Subpackage
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Subpackage();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testSubpackageGetterAndSetterPersistValue()
+    {
+        $this->_tag->setSubpackage('Foo');
+        $this->assertEquals('Foo', $this->_tag->getSubpackage());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/VersionTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/VersionTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/VersionTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Version.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_VersionTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Version
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Version();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testVersionGetterAndSetterPersistValue()
+    {
+        $this->_tag->setVersion('Foo');
+        $this->assertEquals('Foo', $this->_tag->getVersion());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/CopyrightTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/CopyrightTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/CopyrightTest.php	(revision 0)
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Copyright.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_CopyrightTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Copyright
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Copyright();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testCopyrightGetterAndSetterPersistValue()
+    {
+        $this->_tag->setCopyright('Foo');
+        $this->assertEquals('Foo', $this->_tag->getCopyright());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/Docblock/Tag/VarTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/Docblock/Tag/VarTest.php	(revision 0)
+++ tests/Zend/CodeGenerator/Php/Docblock/Tag/VarTest.php	(revision 0)
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+
+/**
+ * @see TestHelper
+ */
+require_once dirname(__FILE__) . '/../../../../../TestHelper.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Class
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Var.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ *
+ * @group Zend_CodeGenerator
+ * @group Zend_CodeGenerator_Php
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_VarTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Zend_CodeGenerator_Php_Docblock_Tag_Var
+     */
+    protected $_tag = null;
+
+    public function setup()
+    {
+        $this->_tag = new Zend_CodeGenerator_Php_Docblock_Tag_Var();
+    }
+
+    public function teardown()
+    {
+        $this->_tag = null;
+    }
+
+    public function testDatatypeGetterAndSetterPersistValue()
+    {
+        $this->_tag->setDatatype('Foo');
+        $this->assertEquals('Foo', $this->_tag->getDatatype());
+    }
+
+    public function testVarProducesCorrectDocBlockLine()
+    {
+        $this->_tag->setDatatype('string');
+        $this->_tag->setDescription('bar bar bar');
+        $this->assertEquals('@var string bar bar bar', $this->_tag->generate());
+    }
+
+}
Index: tests/Zend/CodeGenerator/Php/AllTests.php
===================================================================
--- tests/Zend/CodeGenerator/Php/AllTests.php	(revision 20016)
+++ tests/Zend/CodeGenerator/Php/AllTests.php	(working copy)
@@ -29,8 +29,19 @@
 require_once 'Zend/CodeGenerator/Php/ClassTest.php';
 require_once 'Zend/CodeGenerator/Php/DocblockTest.php';
 require_once 'Zend/CodeGenerator/Php/Docblock/TagTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/AuthorTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/CategoryTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/CopyrightTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/GroupTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/LicenseTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/PackageTest.php';
 require_once 'Zend/CodeGenerator/Php/Docblock/Tag/ParamTest.php';
 require_once 'Zend/CodeGenerator/Php/Docblock/Tag/ReturnTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/SeeTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/SubpackageTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/UsesTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/VarTest.php';
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/VersionTest.php';
 require_once 'Zend/CodeGenerator/Php/FileTest.php';
 require_once 'Zend/CodeGenerator/Php/MethodTest.php';
 require_once 'Zend/CodeGenerator/Php/ParameterTest.php';
@@ -59,8 +70,19 @@
         $suite->addTestSuite('Zend_CodeGenerator_Php_ClassTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_DocblockTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_TagTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_AuthorTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_CategoryTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_CopyrightTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_GroupTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_LicenseTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_PackageTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_ParamTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_ReturnTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_SeeTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_SubpackageTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_UsesTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_VarTest');
+        $suite->addTestSuite('Zend_CodeGenerator_Php_Docblock_Tag_VersionTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_FileTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_MethodTest');
         $suite->addTestSuite('Zend_CodeGenerator_Php_ParameterTest');
Index: tests/Zend/CodeGenerator/Php/DocblockTest.php
===================================================================
--- tests/Zend/CodeGenerator/Php/DocblockTest.php	(revision 20016)
+++ tests/Zend/CodeGenerator/Php/DocblockTest.php	(working copy)
@@ -17,7 +17,7 @@
  * @subpackage UnitTests
  * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
+ * @version    $Id$
  */
 
 /**
@@ -73,15 +73,40 @@
     public function testTagGettersAndSetters()
     {
         $this->_docblock->setTag(array('name' => 'blah'));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Author(array('author' => 'Some Developer')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Category(array('category'=>'Some Category')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Copyright(array('copyright' => '(c) Foo')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Group(array('group' => 'Some_Group')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_License(array('url' => 'http://foo/', 
+                                                                                'description' => 'bar bar bar')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Package(array('package' => 'Baz')));
         $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Param(array('datatype' => 'string')));
         $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Return(array('datatype' => 'int')));
-        $this->assertEquals(3, count($this->_docblock->getTags()));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_See(array('see' => 'Something')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Subpackage(array('subpackage' => 'Child')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Uses(array('uses' => 'Stuff')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Var(array('datatype' => 'string', 
+                                                                             'description' => 'bar bar bar')));
+        $this->_docblock->setTag(new Zend_CodeGenerator_Php_Docblock_Tag_Version(array('version' => '1234')));
+        $this->assertEquals(14, count($this->_docblock->getTags()));
 
         $target = <<<EOS
 /**
  * @blah 
+ * @author Some Developer
+ * @category Some Category
+ * @copyright (c) Foo
+ * @group Some_Group
+ * @license http://foo/ bar bar bar
+ * 
+ * @package Baz
  * @param string
- * @return int
+ * @return int 
+ * @see Something
+ * @subpackage Child
+ * @uses Stuff
+ * @var string bar bar bar
+ * @version 1234
  */
 
 EOS;
Index: library/Zend/CodeGenerator/Php/Docblock/Tag.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag.php	(revision 20016)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag.php	(working copy)
@@ -26,6 +26,36 @@
 require_once 'Zend/CodeGenerator/Php/Abstract.php';
 
 /**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Author
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Author.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Category
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Category.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Copyright
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Copyright.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Group
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Group.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_License
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/License.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Package
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Package.php';
+
+/**
  * @see Zend_CodeGenerator_Php_Docblock_Tag_Param
  */
 require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Param.php';
@@ -36,6 +66,32 @@
 require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Return.php';
 
 /**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_See
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/See.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Subpackage
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Uses
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Uses.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Var
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Var.php';
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag_Version
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag/Version.php';
+
+
+/**
  * @category   Zend
  * @package    Zend_CodeGenerator
  * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
@@ -181,4 +237,4 @@
         return '@' . $this->_name . ' ' . $this->_description;
     }
 
-}
\ No newline at end of file
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Package.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Package.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Package.php	(revision 0)
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Package extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_package = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Package
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $packageTag = new self();
+        
+        $packageTag->setName('package');
+        $packageTag->setPackage($reflectionTagReturn->getPackage());
+        
+        return $packageTag;
+    }
+    
+    /**
+     * setPackage()
+     *
+     * @param string $package
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setPackage($package)
+    {
+        $this->_package = $package;
+        return $this;
+    }
+    
+    /**
+     * getPackage()
+     *
+     * @return string
+     */
+    public function getPackage()
+    {
+        return $this->_package;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@package ' . $this->_package;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Group.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Group.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Group.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Group extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_group = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Group
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $groupTag = new self();
+        
+        $groupTag->setName('group');
+        $groupTag->setGroup($reflectionTagReturn->getGroup());
+        
+        return $groupTag;
+    }
+    
+    /**
+     * setGroup()
+     *
+     * @param string $group
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Group
+     */
+    public function setGroup($group)
+    {
+        $this->_group = $group;
+        return $this;
+    }
+    
+    /**
+     * getGroup()
+     *
+     * @return string
+     */
+    public function getGroup()
+    {
+        return $this->_group;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@group ' . $this->_group;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/See.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/See.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/See.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_See extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_see = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_See
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $seeTag = new self();
+        
+        $seeTag->setName('see');
+        $seeTag->setSee($reflectionTagReturn->getSee());
+        
+        return $seeTag;
+    }
+    
+    /**
+     * setSee()
+     *
+     * @param string $see
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_See
+     */
+    public function setSee($see)
+    {
+        $this->_see = $see;
+        return $this;
+    }
+    
+    /**
+     * getSee()
+     *
+     * @return string
+     */
+    public function getSee()
+    {
+        return $this->_see;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@see ' . $this->_see;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Category.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Category.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Category.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Category extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_category = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Category
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $categoryTag = new self();
+        
+        $categoryTag->setName('category');
+        $categoryTag->setCategory($reflectionTagReturn->getCategory());
+        
+        return $categoryTag;
+    }
+    
+    /**
+     * setCategory()
+     *
+     * @param string $category
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Category
+     */
+    public function setCategory($category)
+    {
+        $this->_category = $category;
+        return $this;
+    }
+    
+    /**
+     * getCategory()
+     *
+     * @return string
+     */
+    public function getCategory()
+    {
+        return $this->_category;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@category ' . $this->_category;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Uses.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Uses.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Uses.php	(revision 0)
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Uses extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_uses = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Uses
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $usesTag = new self();
+        
+        $usesTag->setName('uses');
+        $usesTag->setUses($reflectionTagReturn->getUses());
+        
+        return $usesTag;
+    }
+    
+    /**
+     * setUses()
+     *
+     * @param string $uses
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setUses($uses)
+    {
+        $this->_uses = $uses;
+        return $this;
+    }
+    
+    /**
+     * getUses()
+     *
+     * @return string
+     */
+    public function getUses()
+    {
+        return $this->_uses;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@uses ' . $this->_uses;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/License.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/License.php	(revision 20016)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/License.php	(working copy)
@@ -95,4 +95,4 @@
         return $output;
     }
 
-}
\ No newline at end of file
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Author.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Author.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Author.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Author extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_author = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Author
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $authorTag = new self();
+        
+        $authorTag->setName('author');
+        $authorTag->setAuthor($reflectionTagReturn->getAuthor());
+        
+        return $authorTag;
+    }
+    
+    /**
+     * setAuthor()
+     *
+     * @param string $author
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setAuthor($author)
+    {
+        $this->_author = $author;
+        return $this;
+    }
+    
+    /**
+     * getAuthor()
+     *
+     * @return string
+     */
+    public function getAuthor()
+    {
+        return $this->_author;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@author ' . $this->_author;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Subpackage.php	(revision 0)
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Subpackage extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_subpackage = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Subpackage
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $subpackageTag = new self();
+        
+        $subpackageTag->setName('subpackage');
+        $subpackageTag->setSubpackage($reflectionTagReturn->getSubpackage());
+        
+        return $subpackageTag;
+    }
+    
+    /**
+     * setSubpackage()
+     *
+     * @param string $subpackage
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setSubpackage($subpackage)
+    {
+        $this->_subpackage = $subpackage;
+        return $this;
+    }
+    
+    /**
+     * getSubpackage()
+     *
+     * @return string
+     */
+    public function getSubpackage()
+    {
+        return $this->_subpackage;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@subpackage ' . $this->_subpackage;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Version.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Version.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Version.php	(revision 0)
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Version extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_version = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Version
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $versionTag = new self();
+        
+        $versionTag->setName('version');
+        $versionTag->setVersion($reflectionTagReturn->getVersion());
+        
+        return $versionTag;
+    }
+    
+    /**
+     * setVersion()
+     *
+     * @param string $version
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setVersion($version)
+    {
+        $this->_version = $version;
+        return $this;
+    }
+    
+    /**
+     * getVersion()
+     *
+     * @return string
+     */
+    public function getVersion()
+    {
+        return $this->_version;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@version ' . $this->_version;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Copyright.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Copyright.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Copyright.php	(revision 0)
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Copyright extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_copyright = null;
+    
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Copyright
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $copyrightTag = new self();
+        
+        $copyrightTag->setName('copyright');
+        $copyrightTag->setCopyright($reflectionTagReturn->getCopyright());
+        
+        return $copyrightTag;
+    }
+    
+    /**
+     * setCopyright()
+     *
+     * @param string $copyright
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Copyright
+     */
+    public function setCopyright($copyright)
+    {
+        $this->_copyright = $copyright;
+        return $this;
+    }
+    
+    /**
+     * getCopyright()
+     *
+     * @return string
+     */
+    public function getCopyright()
+    {
+        return $this->_copyright;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@copyright ' . $this->_copyright;
+        return $output;
+    }
+    
+}
Index: library/Zend/CodeGenerator/Php/Docblock/Tag/Var.php
===================================================================
--- library/Zend/CodeGenerator/Php/Docblock/Tag/Var.php	(revision 0)
+++ library/Zend/CodeGenerator/Php/Docblock/Tag/Var.php	(revision 0)
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @subpackage PHP
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * @see Zend_CodeGenerator_Php_Docblock_Tag
+ */
+require_once 'Zend/CodeGenerator/Php/Docblock/Tag.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_CodeGenerator
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_CodeGenerator_Php_Docblock_Tag_Var extends Zend_CodeGenerator_Php_Docblock_Tag 
+{
+    
+    /**
+     * @var string
+     */
+    protected $_datatype = null;
+    
+    /**
+     * @var string
+     */
+    protected $_description = null;
+    
+    /**
+     * fromReflection()
+     *
+     * @param Zend_Reflection_Docblock_Tag $reflectionTagReturn
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Var
+     */
+    public static function fromReflection(Zend_Reflection_Docblock_Tag $reflectionTagReturn)
+    {
+        $returnTag = new self();
+        
+        $returnTag->setName('var');
+        $returnTag->setDatatype($reflectionTagReturn->getType()); // @todo rename
+        $returnTag->setDescription($reflectionTagReturn->getDescription());
+        
+        return $returnTag;
+    }
+    
+    /**
+     * setDatatype()
+     *
+     * @param string $datatype
+     * @return Zend_CodeGenerator_Php_Docblock_Tag_Return
+     */
+    public function setDatatype($datatype)
+    {
+        $this->_datatype = $datatype;
+        return $this;
+    }
+    
+    /**
+     * getDatatype()
+     *
+     * @return string
+     */
+    public function getDatatype()
+    {
+        return $this->_datatype;
+    }
+
+
+    /**
+     * generate()
+     *
+     * @return string
+     */
+    public function generate()
+    {
+        $output = '@var ' . $this->_datatype . ' ' . $this->_description;
+        return $output;
+    }
+    
+}
