Index: FormFile.php
===================================================================
--- FormFile.php	(revision 22206)
+++ FormFile.php	(working copy)
@@ -91,7 +91,7 @@
             'fileName' => $this->getPath(),
             'classes' => array(
                 new Zend_CodeGenerator_Php_Class(array(
-                    'name' => $className,
+                    'name' => ucfirst($className),
                     'extendedClass' => 'Zend_Form',
                     'methods' => array(
                         new Zend_CodeGenerator_Php_Method(array(
