Index: Zend/Db/Table/Row.php
===================================================================
--- Zend/Db/Table/Row.php	(revision 796)
+++ Zend/Db/Table/Row.php	(revision 797)
@@ -74,7 +74,8 @@
         
         if ($config['data'] === false) {
             // empty row, use blanks
-            $cols = array_keys($this->_info['cols']);
+            $cols = array_keys(array_diff_key($this->_info['cols'],
+                                              $this->_info['defaults']));
             $data = array_fill(0, count($cols), null);
             $this->_data = array_combine ($cols, $data);
         } else {
