Index: Mime.php
===================================================================
--- Mime.php	(revision 6198)
+++ Mime.php	(working copy)
@@ -39,7 +39,7 @@
     const DISPOSITION_ATTACHMENT = 'attachment';
     const DISPOSITION_INLINE = 'inline';
     const LINELENGTH = 74;
-    const LINEEND = "\n";
+    const LINEEND = "\r\n";
     const MULTIPART_ALTERNATIVE = 'multipart/alternative';
     const MULTIPART_MIXED = 'multipart/mixed';
     const MULTIPART_RELATED = 'multipart/related';
