--- FormCheckbox.php.orig	2010-01-26 19:06:55.000000000 +0000
+++ FormCheckbox.php	2010-01-26 19:05:34.000000000 +0000
@@ -89,7 +89,7 @@
 
         // build the element
         $xhtml = '';
-        if (!$disable && !strstr($name, '[]')) {
+        if (!$disable && !preg_match('/\[[\w]*\]/', $name)) {
             $xhtml = $this->_hidden($name, $checkedOptions['uncheckedValue']);
         }
         $xhtml .= '<input type="checkbox"'
