Index: incubator/tools/http_server/src/Server.php
===================================================================
--- incubator/tools/http_server/src/Server.php	(revision 1546)
+++ incubator/tools/http_server/src/Server.php	(working copy)
@@ -46,6 +46,11 @@
 		{
 			throw new Zend_Http_Server_Exception( "Socket extension not found" );
 		}
+		
+		// Make sure pcntl functions are available
+		if (! function_exists('pcntl_fork')) {
+			throw new Zend_Http_Server_Exception('PCNTL exntension not found');
+		}
 
 		if( ( $this->socket = socket_create( AF_INET, SOCK_STREAM, SOL_TCP ) ) < 0 )
 		{
