Index: ViewRenderer.php
===================================================================
--- ViewRenderer.php    (revision 5811)
+++ ViewRenderer.php    (working copy)
@@ -802,7 +802,7 @@
             return;
         }

-        if (!$this->_noRender
+        if (!$this->_neverRender && !$this->_noRender
             && (null !== $this->_actionController)
             && $this->getRequest()->isDispatched()
             && !$this->getResponse()->isRedirect())