Index: library/Zend/Search/Lucene/Analysis/Analyzer.php
===================================================================
--- library/Zend/Search/Lucene/Analysis/Analyzer.php	(revision 13469)
+++ library/Zend/Search/Lucene/Analysis/Analyzer.php	(working copy)
@@ -19,41 +19,6 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 
-
-/** Zend_Search_Lucene_Analysis_Token */
-require_once 'Zend/Search/Lucene/Analysis/Token.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8 */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Utf8.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8_CaseInsensitive */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Utf8/CaseInsensitive.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8Num */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Utf8Num.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Utf8Num_CaseInsensitive */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Utf8Num/CaseInsensitive.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Text */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Text.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_Text_CaseInsensitive */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Text/CaseInsensitive.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_TextNum */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/TextNum.php';
-
-/** Zend_Search_Lucene_Analysis_Analyzer_Common_TextNum_CaseInsensitive */
-require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/TextNum/CaseInsensitive.php';
-
-/** Zend_Search_Lucene_Analysis_TokenFilter_StopWords */
-require_once 'Zend/Search/Lucene/Analysis/TokenFilter/StopWords.php';
-
-/** Zend_Search_Lucene_Analysis_TokenFilter_ShortWords */
-require_once 'Zend/Search/Lucene/Analysis/TokenFilter/ShortWords.php';
-
-
 /**
  * An Analyzer is used to analyze text.
  * It thus represents a policy for extracting index terms from text.
@@ -167,6 +132,8 @@
     public static function getDefault()
     {
         if (!self::$_defaultImpl instanceof Zend_Search_Lucene_Analysis_Analyzer) {
+            /** Zend_Search_Lucene_Analysis_Analyzer_Common_Text_CaseInsensitive */
+            require_once 'Zend/Search/Lucene/Analysis/Analyzer/Common/Text/CaseInsensitive.php';
             self::$_defaultImpl = new Zend_Search_Lucene_Analysis_Analyzer_Common_Text_CaseInsensitive();
         }
 
