Index: Table.php
===================================================================
--- Table.php	(revision 921)
+++ Table.php	(working copy)
@@ -371,7 +371,7 @@
         $select = $this->_db->select();
 
         // the FROM clause
-        $select->from($this->_name, array_keys($this->_cols));
+        $select->from($this->_db->quoteIdentifier($this->_name), array_keys($this->_cols));
 
         // the WHERE clause
         $where = (array) $where;
