Index: library/Zend/Form/Decorator/File.php
===================================================================
--- library/Zend/Form/Decorator/File.php	(revision 25022)
+++ library/Zend/Form/Decorator/File.php	(working copy)
@@ -96,11 +96,10 @@
             return $content;
         }
 
-        $name      = $element->getName();
-        $attribs   = $this->getAttribs();
-        if (!array_key_exists('id', $attribs)) {
-            $attribs['id'] = $name;
-        }
+        $name          = $element->getFullyQualifiedName();
+        $attribs       = $element->getAttribs();
+        $id            = $element->getId();
+        $attribs['id'] = $id;
 
         $separator = $this->getSeparator();
         $placement = $this->getPlacement();
@@ -112,13 +111,20 @@
         }
 
         if (Zend_File_Transfer_Adapter_Http::isApcAvailable()) {
-            $markup[] = $view->formHidden(ini_get('apc.rfc1867_name'), uniqid(), array('id' => 'progress_key'));
+            $markup[] = $view->formHidden(
+                ini_get('apc.rfc1867_name'),
+                uniqid(),
+                array('id' => 'progress_key')
+            );
         } else if (Zend_File_Transfer_Adapter_Http::isUploadProgressAvailable()) {
-            $markup[] = $view->formHidden('UPLOAD_IDENTIFIER', uniqid(), array('id' => 'progress_key'));
+            $markup[] = $view->formHidden(
+                'UPLOAD_IDENTIFIER',
+                uniqid(),
+                array('id' => 'progress_key')
+            );
         }
 
         if ($element->isArray()) {
-            $name .= "[]";
             $count = $element->getMultiFile();
             for ($i = 0; $i < $count; ++$i) {
                 $htmlAttribs        = $attribs;
