Index: library/Zend/Translate/Adapter.php
===================================================================
--- library/Zend/Translate/Adapter.php	(Revision 24266)
+++ library/Zend/Translate/Adapter.php	(Arbeitskopie)
@@ -249,7 +249,8 @@
                 new RecursiveRegexIterator(
                     new RecursiveDirectoryIterator($options['content'], RecursiveDirectoryIterator::KEY_AS_PATHNAME),
                     '/^(?!.*(\.svn|\.cvs)).*$/', RecursiveRegexIterator::MATCH
-                )
+                ),
+                RecursiveIteratorIterator::SELF_FIRST
             );
             
             foreach ($iterator as $directory => $info) {
