Index: C:/Basil/NetBeansProjects/trunk/library/Zend/Form/Element.php
===================================================================
--- C:/Basil/NetBeansProjects/trunk/library/Zend/Form/Element.php	(revision 23520)
+++ C:/Basil/NetBeansProjects/trunk/library/Zend/Form/Element.php	(working copy)
@@ -854,6 +854,31 @@
     }
 
     /**
+     * Remove element attribute
+     * @param string $name
+     * @return Zend_Form_Element
+     * @throws Zend_Form_Exception for invalid $name values
+     */
+    public function removeAttrib($name)
+    {
+        $name = (string) $name;
+
+        if (empty($name)) {
+            require_once 'Zend/Form/Exception.php';
+            throw new Zend_Form_Exception(sprintf('Attribute cannot be empty'));
+        } elseif (!isset($this->$name) || null === $this->$name) {
+            require_once 'Zend/Form/Exception.php';
+            throw new Zend_Form_Exception(sprintf('Attribute "%s" does not exist', $name));
+        } else {
+            $this->setAttrib($name, null);
+        }
+
+        return $this;
+
+    }
+
+
+    /**
      * Set multiple attributes at once
      *
      * @param  array $attribs
