Index: library/Zend/Tool/Framework/Client/Console/ArgumentParser.php
===================================================================
--- library/Zend/Tool/Framework/Client/Console/ArgumentParser.php	(revision 18303)
+++ library/Zend/Tool/Framework/Client/Console/ArgumentParser.php	(working copy)
@@ -414,6 +414,7 @@
             // process ParameterInfo into array for command line WORD (argument) matching
             $wordArguments[$parameterInfo['position']]['parameterName'] = $parameterInfo['name'];
             $wordArguments[$parameterInfo['position']]['optional']      = $parameterInfo['optional'];
+            $wordArguments[$parameterInfo['position']]['default']       = $parameterInfo['default'];
             $wordArguments[$parameterInfo['position']]['type']          = $parameterInfo['type'];
 
             // keep a translation of console to canonical names
@@ -450,7 +451,17 @@
                 if (!array_key_exists($wordIndex-1, $wordStack) || !array_key_exists($wordIndex, $wordArguments)) {
                     break;
                 }
-                $this->_request->setProviderParameter($wordArguments[$wordIndex]['parameterName'], $wordStack[$wordIndex-1]);
+
+                // if default value for argument is set, fill parameter value with string after '='
+                if (! is_null($wordArguments[$wordIndex]['default'])) {
+                    $this->_request->setProviderParameter($wordArguments[$wordIndex]['parameterName'],
+                                                          substr($wordStack[$wordIndex - 1],
+                                                               strpos($wordStack[$wordIndex - 1], '=') + 1));
+                } else {
+                    $this->_request->setProviderParameter($wordArguments[$wordIndex]['parameterName'],
+                                                          $wordStack[$wordIndex - 1]);
+                }
+
                 unset($wordStack[$wordIndex-1]);
             }
         }
