Index: Filter.php
===================================================================
--- Filter.php	(revision 8884)
+++ Filter.php	(working copy)
@@ -91,7 +91,7 @@
     public static function get($value, $classBaseName, array $args = array(), $namespaces = array())
     {
         require_once 'Zend/Loader.php';
-        $namespaces = array_merge(array('Zend_Filter'), (array) $namespaces);
+        $namespaces = array_merge((array) $namespaces, array('Zend_Filter'));
         foreach ($namespaces as $namespace) {
             $className = $namespace . '_' . ucfirst($classBaseName);
             try {
