Index: Atom.php
===================================================================
--- Atom.php	(revision 18872)
+++ Atom.php	(working copy)
@@ -295,7 +295,7 @@
             return $this->_data['commentcount'];
         }
 
-        $commentcount = $this->getExtension('Thread')>getCommentCount();
+        $commentcount = $this->getExtension('Thread')->getCommentCount();
 
         if (!$commentcount) {
             $commentcount = $this->getExtension('Atom')->getCommentCount();
