diff --git a/tests/Zend/View/Helper/PartialLoopTest.php b/tests/Zend/View/Helper/PartialLoopTest.php
old mode 100644
new mode 100755
index d936ac5..e3da49b
--- a/tests/Zend/View/Helper/PartialLoopTest.php
+++ b/tests/Zend/View/Helper/PartialLoopTest.php
@@ -323,6 +323,33 @@ class Zend_View_Helper_PartialLoopTest extends PHPUnit_Framework_TestCase
         }
     }
 
+
+    /**
+     * @see ZF-7157
+     * @link http://framework.zend.com/issues/browse/ZF-7151
+     */
+    public function testPartialLoopSetsTotalCount()
+    {
+        $data = array(
+            array('message' => 'foo'),
+            array('message' => 'bar'),
+            array('message' => 'baz'),
+            array('message' => 'bat')
+        );
+
+        $view = new Zend_View(array(
+            'scriptPath' => $this->basePath . '/default/views/scripts'
+        ));
+        $this->helper->setView($view);
+
+        $result = $this->helper->partialLoop('partialLoopCouter.phtml', $data);
+        foreach ($data as $key=>$item) {
+            $string = 'Total count: ' . count($data);
+            $this->assertContains($string, $result);
+        }
+    }
+
+
     /**
      * @see ZF-5174
      * @link http://framework.zend.com/issues/browse/ZF-5174
diff --git a/tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml b/tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml
old mode 100644
new mode 100755
index 13439e6..6f7a9bf
--- a/tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml
+++ b/tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml
@@ -1,2 +1,3 @@
 This is an iteration: <?php echo $this->message ?>, pointer at <?php echo $this->partialCounter ?>
+Total count: <?php echo $this->partialTotalCount ?>
 
