Index: tests/Zend/Service/TwitterTest.php
===================================================================
--- tests/Zend/Service/TwitterTest.php	(revision 20197)
+++ tests/Zend/Service/TwitterTest.php	(working copy)
@@ -450,6 +450,23 @@
     }
 
     /**
+     * @group ZF-8891
+     * @return void
+     */
+    public function testPostStatusRetweetReturnsResponse()
+    {
+        /* @var $response Zend_Rest_Client_Result */
+        $response = $this->twitter->status->retweet(5466220198);
+        $this->assertTrue($response instanceof Zend_Rest_Client_Result);
+
+        $httpClient = $this->twitter->getLocalHttpClient();
+        $httpRequest = $httpClient->getLastRequest();
+        $httpResponse = $httpClient->getLastResponse();
+        $this->assertTrue($httpResponse->isSuccessful(), $httpResponse->getStatus() . ': ' . var_export($httpRequest, 1) . '\n' . $httpResponse->getHeadersAsString());
+        $this->assertTrue(isset($response->status));
+    }
+
+    /**
      * $return void
      */
     public function testPostStatusUpdateToLongShouldThrowException()
Index: library/Zend/Service/Twitter.php
===================================================================
--- library/Zend/Service/Twitter.php	(revision 20197)
+++ library/Zend/Service/Twitter.php	(working copy)
@@ -464,6 +464,22 @@
     }
 
     /**
+     * Retweet a status message
+     *
+     * @param  int $id ID of status to retweet
+     * @throws Zend_Http_Client_Exception if HTTP request fails or times out
+     * @return Zend_Rest_Client_Result
+     */
+    public function statusRetweet($id)
+    {
+        $this->_init();
+        $path = '/statuses/retweet/' . $this->_validInteger($id) . '.xml';
+        $response = $this->_post($path);
+        return new Zend_Rest_Client_Result($response->getBody());
+    }
+
+
+    /**
      * User friends
      *
      * @param  int|string $id Id or username of user for whom to fetch friends
