Index: library/Zend/Db/Table/Select.php
===================================================================
--- library/Zend/Db/Table/Select.php	(revision 10150)
+++ library/Zend/Db/Table/Select.php	(working copy)
@@ -80,6 +80,11 @@
     {
         $this->_adapter = $table->getAdapter();
         $this->_info    = $table->info();
+        
+        $primary = $this->_info[Zend_Db_Table_Abstract::NAME];
+        $schema  = $this->_info[Zend_Db_Table_Abstract::SCHEMA];
+        $this->from($primary, self::SQL_WILDCARD, $schema);
+        
         return $this;
     }
 
