Index: ViewTest.php
===================================================================
--- ViewTest.php	(revision 18731)
+++ ViewTest.php	(working copy)
@@ -1074,6 +1074,21 @@
         $paths = $view->getFilterPaths();
         $this->assertTrue(array_key_exists('My_View_', $paths), var_export($paths, 1));
     }
+    
+    /**
+     * @group ZF-8177
+     */
+    public function testRegisterHelper()
+    {
+    	require_once 'View/_stubs/HelperDir1/Stub1.php';
+    	
+    	$view = new Zend_View();
+    	$helper = new Foo_View_Helper_Stub1();
+    	$view->registerHelper($helper, 'stub1');
+    	
+    	$this->assertEquals($view->getHelper('stub1'), $helper);
+    	$this->assertEquals($view->stub1(), 'foo');
+    }
 }
 
 /**
