--- library/Zend/Db/Table/Abstract.php	(revision 24462)
+++ library/Zend/Db/Table/Abstract.php	(working copy)
@@ -885,7 +885,7 @@
             // then throw an exception.
             if (empty($this->_primary)) {
                 require_once 'Zend/Db/Table/Exception.php';
-                throw new Zend_Db_Table_Exception('A table must have a primary key, but none was found');
+                throw new Zend_Db_Table_Exception('A table must have a primary key, but none was found for table "' . $this->_name . '"');
             }
         } else if (!is_array($this->_primary)) {
             $this->_primary = array(1 => $this->_primary);
