--- library/Zend/CodeGenerator/Php/Property/DefaultValue.php	(Revision 18385)
+++ library/Zend/CodeGenerator/Php/Property/DefaultValue.php	(Arbeitskopie)
@@ -263,15 +263,20 @@
         $output = '';
 
         switch ($type) {
+            case self::TYPE_BOOLEAN:
+                $output .= ( $value ? 'true' : 'false' );
+                break;
             case self::TYPE_STRING:
-                $output .= "'" . $value . "'";
+                $output .= "'" . addcslashes($value, "'") . "'";
                 break;
+            case self::TYPE_NULL:
+                $output .= 'null';
+                break;
             case self::TYPE_NUMBER:
             case self::TYPE_INTEGER:
             case self::TYPE_INT:
             case self::TYPE_FLOAT:
             case self::TYPE_DOUBLE:
-            case self::TYPE_NULL:
             case self::TYPE_CONSTANT:
                 $output .= $value;
                 break;
@@ -292,7 +297,7 @@
                         $outputParts[] = $partV;
                         $noKeyIndex++;
                     } else {
-                        $outputParts[] = (is_int($n) ? $n : "'" . $n . "'") . ' => ' . $partV;
+                        $outputParts[] = (is_int($n) ? $n : "'" . addcslashes($n, "'") . "'") . ' => ' . $partV;
                     }
 
                 }
