Index: tests/Zend/Form/ElementTest.php
===================================================================
--- tests/Zend/Form/ElementTest.php	(revision 24787)
+++ tests/Zend/Form/ElementTest.php	(working copy)
@@ -458,12 +458,25 @@
         );
         $this->element->setAttribs($attribs);
 
-        $attribs['helper'] = 'formText';
-
         $received = $this->element->getAttribs();
         $this->assertEquals($attribs, $received);
     }
 
+    /**
+     * @group ZF-6061
+     */
+    public function testHelperDoesNotShowUpInAttribs()
+    {
+        $attribs = array(
+            'foo' => 'bar',
+            'bar' => 'baz',
+            'baz' => 'bat'
+        );
+        $this->element->setAttribs($attribs);
+
+        $this->assertFalse(array_key_exists('helper', $this->element->getAttribs()));
+    }
+
     public function testPassingNullValuesToSetAttribsUnsetsAttribs()
     {
         $this->testSetAttribsSetsMultipleAttribs();
Index: tests/Zend/Form/_files/views/withouthelperinattribs.phtml
===================================================================
--- tests/Zend/Form/_files/views/withouthelperinattribs.phtml	(revision 0)
+++ tests/Zend/Form/_files/views/withouthelperinattribs.phtml	(revision 0)
@@ -0,0 +1,7 @@
+<?php
+
+echo $this->{$this->element->helper}(
+    $this->element->getName(),
+    null,
+    $this->element->getAttribs()
+);
\ No newline at end of file
Index: tests/Zend/Form/Decorator/ViewScriptTest.php
===================================================================
--- tests/Zend/Form/Decorator/ViewScriptTest.php	(revision 24787)
+++ tests/Zend/Form/Decorator/ViewScriptTest.php	(working copy)
@@ -222,6 +222,21 @@
         $this->assertContains('This text prefixes the content', $test);
         $this->assertContains('This text appends the content', $test);
     }
+
+    /**
+     * @group ZF-6061
+     */
+    public function testRenderingWithoutHelperInAttribs()
+    {
+        $this->decorator->setViewScript('withouthelperinattribs.phtml')
+            ->setElement($this->getElement());
+
+        $expected = 'Foo:'
+                  . "\n"
+                  . '<input type="text" name="foo" id="foo" value="">';
+
+        $this->assertSame($expected, $this->decorator->render('Foo:'));
+    }
 }
 
 // Call Zend_Form_Decorator_ViewScriptTest::main() if this source file is executed directly.