Index: Loader.php
===================================================================
--- Loader.php	(revision 14250)
+++ Loader.php	(working copy)
@@ -80,6 +80,11 @@
             self::loadFile($file, $dirs, true);
         } else {
             self::_securityCheck($file);
+            if (!file_exists($filename)) {
+                require_once 'Zend/Exception.php';
+                throw new Zend_Exception("File \"$filename\" does not exist");
+            }
+            
             include $file;
         }
 
