Index: tests/Zend/Loader/Autoloader/ResourceTest.php
===================================================================
--- tests/Zend/Loader/Autoloader/ResourceTest.php	(revision 23564)
+++ tests/Zend/Loader/Autoloader/ResourceTest.php	(working copy)
@@ -419,6 +419,49 @@
         $path = $this->loader->autoload('Something_Totally_Wrong');
         $this->assertFalse($path);
     }
+
+    /**
+     * @group ZF-10836
+     */
+    public function testAutoloaderResourceConstructorDataArgumentNamespaceKeyMustPreceedResourceTypesKey()
+    {
+        // Set up autoloader where namespace key is in the "wrong" place
+        $data = array(
+            'basePath'      => 'path/to/some/directory',
+            'resourceTypes' => array(
+                'acl' => array(
+                    'path'      => 'acls/',
+                    'namespace' => 'Acl',
+                ),
+                'form' => array(
+                    'path'      => 'forms/',
+                    'namespace' => 'Form',
+                )
+            ),
+            'namespace'     => 'My'
+        );
+        $zlar1 = new Zend_Loader_Autoloader_Resource($data);
+        
+        // Set up autoloader where namespace is in the "right" place
+        $data = array(
+            'basePath'      => 'path/to/some/directory',
+        	'namespace'     => 'My',
+            'resourceTypes' => array(
+                'acl' => array(
+                    'path'      => 'acls/',
+                    'namespace' => 'Acl',
+                ),
+                'form' => array(
+                    'path'      => 'forms/',
+                    'namespace' => 'Form',
+                )
+            )
+        );
+        $zlar2 = new Zend_Loader_Autoloader_Resource($data);
+
+        // Check that the results are the same
+        $this->assertEquals($zlar1, $zlar2);
+    }
 }
 
 if (PHPUnit_MAIN_METHOD == 'Zend_Loader_Autoloader_ResourceTest::main') {
Index: library/Zend/Loader/Autoloader/Resource.php
===================================================================
--- library/Zend/Loader/Autoloader/Resource.php	(revision 23564)
+++ library/Zend/Loader/Autoloader/Resource.php	(working copy)
@@ -206,6 +206,14 @@
      */
     public function setOptions(array $options)
     {
+        // Process namespace argument first
+        // namespace cannot be null, so we can use isset to test
+        // see ZF-10836
+        if (isset($options['namespace'])) {
+            $this->setNamespace($options['namespace']);
+            unset($options['namespace']);
+        }
+        
         $methods = get_class_methods($this);
         foreach ($options as $key => $value) {
             $method = 'set' . ucfirst($key);
