Index: Ini.php
===================================================================
--- Ini.php	(revision 23445)
+++ Ini.php	(working copy)
@@ -54,6 +54,15 @@
      * @var boolean
      */
     protected $_skipExtends = false;
+    /**
+     * The full path of current ini file
+     * @var string
+     */
+    protected $_fileName;
+    /**
+     * Root section(s) to be loaded
+     */
+    protected $_rootSections = null;
 
     /**
      * Loads the section $section from the config file $filename for
@@ -124,6 +133,9 @@
 
         $iniArray = $this->_loadIniFile($filename);
 
+        // stores the filename for further reference (@include)
+        $this->_fileName = realpath($filename);
+
         if (null === $section) {
             // Load entire file
             $dataArray = array();
@@ -140,6 +152,9 @@
             if (!is_array($section)) {
                 $section = array($section);
             }
+
+            $this->_rootSections = $section;
+            
             $dataArray = array();
             foreach ($section as $sectionName) {
                 if (!isset($iniArray[$sectionName])) {
@@ -301,7 +316,20 @@
                 throw new Zend_Config_Exception("Invalid key '$key'");
             }
         } else {
-            $config[$key] = $value;
+            // checks for inclusion commands (@include special key)
+            if ($key === "@include") {
+                $includedFilesPath = dirname($this->_fileName) . DIRECTORY_SEPARATOR . $value;
+                $includedFiles = glob($includedFilesPath);
+                print_r($includedFiles);
+                $data = array();
+                foreach ($includedFiles as $file) {
+                    $subConfig = new self($file, $this->_rootSections);
+                    $data = array_merge($data, $subConfig->toArray());
+                }
+                $config = array_merge($config, $data);
+            } else {
+                $config[$key] = $value;
+            }
         }
         return $config;
     }
