--- library/Zend/Locale/Data.php	(revision 24585)
+++ library/Zend/Locale/Data.php	(working copy)
@@ -907,7 +907,7 @@
                 break;
         }
 
-        if (isset(self::$_cache)) {
+        if (isset(self::$_cache) && !self::$_cacheDisabled) {
             if (self::$_cacheTags) {
                 self::$_cache->save( serialize($temp), $id, array('Zend_Locale'));
             } else {
@@ -1418,7 +1418,7 @@
         if (is_array($temp)) {
             $temp = current($temp);
         }
-        if (isset(self::$_cache)) {
+        if (isset(self::$_cache) && !self::$_cacheDisabled) {
             if (self::$_cacheTags) {
                 self::$_cache->save( serialize($temp), $id, array('Zend_Locale'));
             } else {
