Index: library/Zend/Search/Lucene/Search/QueryHit.php
===================================================================
--- library/Zend/Search/Lucene/Search/QueryHit.php	(revision 24746)
+++ library/Zend/Search/Lucene/Search/QueryHit.php	(working copy)
@@ -83,6 +83,19 @@
 
 
     /**
+     * Convenience function for checking existence of fields from the document
+     * associated with this hit.
+     *
+     * @param string $offset
+     * @return string
+     */
+    public function __isset($offset)
+    {
+        return $this->getDocument()->fieldExists($offset);
+    }
+
+
+    /**
      * Return the document object for this hit
      *
      * @return Zend_Search_Lucene_Document
