Index: Doctype.php
===================================================================
--- Doctype.php	(revision 21981)
+++ Doctype.php	(working copy)
@@ -186,14 +186,14 @@
     {
         return (stristr($this->getDoctype(), 'xhtml') ? true : false);
     }
-	
+
 	/**
 	 * Is doctype HTML5? (HeadMeta uses this for validation)
 	 *
 	 * @return booleean
 	 */
 	public function isHtml5() {
-		return (stristr($this->doctype(), '<!DOCTYPE html>') ? true : false);
+		return (stristr($this->getDoctype(), '<!DOCTYPE html>') ? true : false);
 	}
 
     /**
