Index: Firebird.php
===================================================================
--- Firebird.php	(revision 15895)
+++ Firebird.php	(working copy)
@@ -291,7 +291,9 @@
 
         // Suppress connection warnings here.
         // Throw an exception instead.
-        $this->_connection = @ibase_connect(
+        $connectFunction = ($this->_config['persistent']) ? 'ibase_pconnect' : 'ibase_connect';
+        
+        $this->_connection = @$connectFunction(
                                 $this->_formatDbConnString($this->_config['host'],$this->_config['port'] ,$this->_config['dbname']),
                                 $this->_config['username'],
                                 $this->_config['password'],
