Index: Filter.php
===================================================================
--- Filter.php	(revision 8884)
+++ Filter.php	(working copy)
@@ -36,6 +36,14 @@
 class Zend_Filter implements Zend_Filter_Interface
 {
     /**
+     * Default namespaces used in Zend_Filter::get()
+     *
+     * @var array
+     * @static
+     */
+    protected static $_defaultNamespaces = array();
+
+    /**
      * Filter chain
      *
      * @var array
@@ -72,6 +80,25 @@
     }
 
     /**
+     * Sets the default namespaces for Zend_Filter::get()
+     *
+     * @param   array $namespaces
+     * @return  void
+     */
+    public static function setDefaultNamespaces(array $namespaces = array()) {
+        self::$_defaultNamespaces = $namespaces;
+    }
+
+    /**
+     * Gets the default namespaces for Zend_Filter::get()
+     *
+     * @param  array $namespaces
+     */
+    public static function getDefaultNamespaces() {
+        return self::$_defaultNamespaces;
+    }
+
+    /**
      * Returns a value filtered through a specified filter class, without requiring separate
      * instantiation of the filter object.
      *
@@ -91,7 +118,7 @@
     public static function get($value, $classBaseName, array $args = array(), $namespaces = array())
     {
         require_once 'Zend/Loader.php';
-        $namespaces = array_merge(array('Zend_Filter'), (array) $namespaces);
+        $namespaces = array_merge(array('Zend_Filter'), self::$_defaultNamespaces, (array) $namespaces);
         foreach ($namespaces as $namespace) {
             $className = $namespace . '_' . ucfirst($classBaseName);
             try {
