Index: library/Zend/Oauth/Http/Utility.php
===================================================================
--- library/Zend/Oauth/Http/Utility.php	(revision 21216)
+++ library/Zend/Oauth/Http/Utility.php	(working copy)
@@ -52,10 +52,13 @@
             'oauth_nonce'            => $this->generateNonce(),
             'oauth_signature_method' => $config->getSignatureMethod(),
             'oauth_timestamp'        => $this->generateTimestamp(),
-            'oauth_token'            => $config->getToken()->getToken(),
             'oauth_version'          => $config->getVersion(),
         );
 
+        if ($config->getToken()->getToken() != null) {
+            $params['oauth_token'] = $config->getToken()->getToken();
+        }
+
         if (!is_null($serviceProviderParams)) {
             $params = array_merge($params, $serviceProviderParams);
         }
