Index: ItemList.php
===================================================================
--- ItemList.php	(revision 15443)
+++ ItemList.php	(working copy)
@@ -197,7 +197,7 @@
      * Get the value of an offset
      *
      * @param  mixed $offset
-     * @return Zend_Tag_IItem
+     * @return Zend_Tag_Taggable
      */
     public function offsetGet($offset) {
         return $this->_items[$offset];
@@ -207,7 +207,7 @@
      * Append a new item
      *
      * @param  mixed          $offset
-     * @param  Zend_Tag_IItem $item
+     * @param  Zend_Tag_Taggable $item
      * @throws OutOfBoundsException When item does not implement Zend_Tag_Taggable
      * @return void
      */
Index: Item.php
===================================================================
--- Item.php	(revision 15443)
+++ Item.php	(working copy)
@@ -121,7 +121,7 @@
     }
     
     /**
-     * Defined by Zend_Tag_IItem
+     * Defined by Zend_Tag_Taggable
      *
      * @return string
      */
@@ -149,7 +149,7 @@
     }
     
     /**
-     * Defined by Zend_Tag_IItem
+     * Defined by Zend_Tag_Taggable
      *
      * @return float
      */
@@ -192,7 +192,7 @@
     }
     
     /**
-     * Defined by Zend_Tag_IItem
+     * Defined by Zend_Tag_Taggable
      *
      * @param  string $name
      * @param  mixed  $value
@@ -205,7 +205,7 @@
     }
     
     /**
-     * Defined by Zend_Tag_IItem
+     * Defined by Zend_Tag_Taggable
      *
      * @param  string $name
      * @return mixed
Index: Cloud.php
===================================================================
--- Cloud.php	(revision 15443)
+++ Cloud.php	(working copy)
@@ -161,7 +161,7 @@
     /**
      * Append a single tag to the cloud
      *
-     * @param  Zend_Tag_IItem|array $tag
+     * @param  Zend_Tag_Taggable|array $tag
      * @return Zend_Tag_Cloud
      */
     public function appendTag($tag)
