Index: library/Zend/Validate/PostCode.php
===================================================================
--- library/Zend/Validate/PostCode.php	(revision 20427)
+++ library/Zend/Validate/PostCode.php	(working copy)
@@ -156,6 +156,7 @@
      * Sets a self defined postal format as regex
      *
      * @param string $format
+     * @return Zend_Validate_Postcode
      */
     public function setFormat($format)
     {
@@ -173,6 +174,7 @@
         }
 
         $this->_format = $format;
+        return $this;
     }
 
     /**
