Index: library/Zend/View/Helper/FormSelect.php
===================================================================
--- library/Zend/View/Helper/FormSelect.php	(revision 21318)
+++ library/Zend/View/Helper/FormSelect.php	(working copy)
@@ -97,13 +97,17 @@
         }
 
         // now start building the XHTML.
-        $disabled = '';
+        $xhtml = '';
+        $disabled = '';        
         if (true === $disable) {
             $disabled = ' disabled="disabled"';
+            if(null !== $value && empty($multiple)) {
+                $xhtml = $this->_hidden($name,array_shift(array_values($value)));
+            }
         }
 
         // Build the surrounding select element first.
-        $xhtml = '<select'
+        $xhtml .= '<select'
                 . ' name="' . $this->view->escape($name) . '"'
                 . ' id="' . $this->view->escape($id) . '"'
                 . $multiple
