Index: library/Zend/Loader/PluginLoader.php
===================================================================
--- library/Zend/Loader/PluginLoader.php	(revision 21641)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -471,10 +471,9 @@
      */
     protected static function _appendIncFile($incFile)
     {
-        if (!file_exists(self::$_includeFileCache)) {
+        $file = file_get_contents(self::$_includeFileCache);
+        if (!$file) {
             $file = '<?php';
-        } else {
-            $file = file_get_contents(self::$_includeFileCache);
         }
         if (!strstr($file, $incFile)) {
             $file .= "\ninclude_once '$incFile';";
