Index: tests/Zend/Filter/PregReplaceTest.php
===================================================================
--- tests/Zend/Filter/PregReplaceTest.php	(revision 21072)
+++ tests/Zend/Filter/PregReplaceTest.php	(working copy)
@@ -130,9 +130,28 @@
         } catch (Exception $e) {
         }
     }
+    
+    /**
+ 	 * @group ZF-9202
+ 	 */
+    public function testExtendsPregReplace()
+    {
+    	$startMatchPattern = '~(&gt;){3,}~i';
+    	$filter = new XPregReplace();
+    	$this->assertEquals($startMatchPattern, $filter->getMatchPattern());
+    }
 }
 
 // Call Zend_Filter_PregReplaceTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == 'Zend_Filter_PregReplaceTest::main') {
     Zend_Filter_PregReplaceTest::main();
 }
+
+
+/**
+ * @group ZF-9202
+ */
+class XPregReplace extends Zend_Filter_PregReplace
+{
+	protected $_matchPattern = '~(&gt;){3,}~i';
+}
