Index: tests/Zend/Form/Element/FileTest.php
===================================================================
--- tests/Zend/Form/Element/FileTest.php	(revision 20215)
+++ tests/Zend/Form/Element/FileTest.php	(working copy)
@@ -401,6 +401,17 @@
         $this->assertEquals(1172, $element->getFileSize('baz.text'));
     }
 
+    public function testFileSizeWithDetectFileSize()
+    {
+        $element = new Zend_Form_Element_File('baz2');
+        $adapter = new Zend_Form_Element_FileTest_MockAdapter();
+        $element->setTransferAdapter($adapter);
+
+        $this->assertEquals('8B', $element->getFileSize('baz.text'));
+        $adapter->setOptions(array('useByteString' => false));
+        $this->assertEquals(8, $element->getFileSize('baz.text'));
+    }
+
     public function testMimeType()
     {
         $element = new Zend_Form_Element_File('baz');
@@ -520,6 +531,17 @@
                 'filtered'   => false,
                 'validators' => array(),
             ),
+            'baz2' => array(
+                'name'       => 'baz.text',
+                'type'       => 'text/plain',
+                'size'       => 1172,
+                'tmp_name'   => $testfile,
+                'options'   => array('ignoreNoFile' => false, 'useByteString' => true),
+                'validated'  => false,
+                'received'   => false,
+                'filtered'   => false,
+                'validators' => array(),
+            ),
             'file_1_' => array(
                 'name'       => 'baz.text',
                 'type'       => 'text/plain',
@@ -543,6 +565,8 @@
                 'validators' => array(),
             ),
             );
+
+        $this->_files['baz2']['size'] = $this->_detectFileSize($this->_files['baz2']);
     }
 
     public function send($options = null)

