--- tests/Zend/Dojo/Form/Element/EditorTest.php	(revision 24415)
+++ tests/Zend/Dojo/Form/Element/EditorTest.php	(working copy)
@@ -312,6 +312,25 @@
         $extraPlugins = $this->element->getDijitParam('extraPlugins');
         $this->assertNotContains('bold', $extraPlugins, var_export($extraPlugins, 1));
     }
+    
+    /**
+     * @group ZF-6371
+     */
+    public function testEditorHeightCanBeSetToEmptyString()
+    {
+        $this->element->setHeight('');
+        
+        $this->assertSame('', $this->element->getHeight());
+    }
+    
+    /**
+     * @group ZF-6371
+     * @expectedException Zend_Form_Element_Exception
+     */
+    public function testEditorHeightCanNotBeSetToNull()
+    {
+        $this->element->setHeight(null);
+    }
 }
 
 // Call Zend_Dojo_Form_Element_EditorTest::main() if this source file is executed directly.
