Index: Oci.php
===================================================================
--- Oci.php	(revision 4609)
+++ Oci.php	(working copy)
@@ -57,16 +57,21 @@
     {
         // baseline of DSN parts
         $dsn = $this->_config;
-
-        $tns = 'dbname=//' . $dsn['host'];
-        if (isset($dsn['port'])) {
-            $tns .= ':' . $dsn['port'];
+        
+        $tns = 'dbname=';
+        
+        // If 'host' parameter is set, add with //<hostname>
+        if (isset($dsn['host'])) {
+            $tns .= '//' . $dsn['host'];
+            if (isset($dsn['port'])) {
+                $tns .= ':' . $dsn['port'];
+            }
+            $tns .= '/';
         }
-        $tns .= '/' . $dsn['dbname'];
+        $tns .= $dsn['dbname'];
 
         return $this->_pdoType . ':' . $tns;
     }
-
     /**
      * Quote a raw string.
      * Most PDO drivers have an implementation for the quote() method,
