Index: ZendFramework/library/Zend/Db/Table/Abstract.php
===================================================================
--- ZendFramework/library/Zend/Db/Table/Abstract.php	(revision 6231)
+++ ZendFramework/library/Zend/Db/Table/Abstract.php	(working copy)
@@ -265,6 +265,14 @@
     }
 
     /**
+     * @return integer
+     */
+    public function getIdentity()
+    {
+        return $this->_identity;
+    }
+
+    /**
      * @param  string $classname
      * @return Zend_Db_Table_Abstract Provides a fluent interface
      */
Index: ZendFramework/library/Zend/Db/Table/Row/Abstract.php
===================================================================
--- ZendFramework/library/Zend/Db/Table/Row/Abstract.php	(revision 6231)
+++ ZendFramework/library/Zend/Db/Table/Row/Abstract.php	(working copy)
@@ -323,7 +323,7 @@
         if (is_array($primaryKey)) {
             $newPrimaryKey = $primaryKey;
         } else {
-            $newPrimaryKey = array(current((array)$this->_primary) => $primaryKey);
+            $newPrimaryKey = array($this->_primary[$this->_table->getIdentity()] => $primaryKey);
         }
 
         /**
