### Eclipse Workspace Patch 1.0
#P trunk
Index: library/Zend/Loader.php
===================================================================
--- library/Zend/Loader.php	(revision 23572)
+++ library/Zend/Loader.php	(working copy)
@@ -179,26 +179,40 @@
             return true;
         }
 
-        if (strtoupper(substr(PHP_OS, 0, 3)) == 'WIN'
-            && preg_match('/^[a-z]:/i', $filename)
-        ) {
-            // If on windows, and path provided is clearly an absolute path,
-            // return false immediately
+        // If a path is defined — whether absolute (starting with a drive letter or \ on Windows,
+        // or / on Unix/Linux systems) or relative to the current directory (starting with . or ..)
+        // — the include_path will be ignored altogether.
+
+        // test absolute path
+        if ( $filename[0] == DIRECTORY_SEPARATOR
+          || (strtoupper(substr(PHP_OS, 0, 3)) == 'WIN' && preg_match('/^[a-z]:/i', $filename))) {
             return false;
         }
 
+        // test current working directory
+        $substr = substr($filename, 0, 2);
+        if ($substr == './' || $substr == '.' . DIRECTORY_SEPARATOR) {
+            return false;
+        }
+
+        // test previous working directory
+        $substr = substr($filename, 0, 3);
+        if ($substr == '../' || $substr == '..' . DIRECTORY_SEPARATOR) {
+            return false;
+        }
+
         foreach (self::explodeIncludePath() as $path) {
             if ($path == '.') {
-                if (is_readable($filename)) {
-                    return true;
-                }
+                // already tested above
                 continue;
             }
-            $file = $path . '/' . $filename;
+
+            $file = $path . DIRECTORY_SEPARATOR . $filename;
             if (is_readable($file)) {
                 return true;
             }
         }
+
         return false;
     }
 
