Index: library/Zend/Captcha/Image.php
===================================================================
--- library/Zend/Captcha/Image.php	(révision 15859)
+++ library/Zend/Captcha/Image.php	(copie de travail)
@@ -589,6 +589,6 @@
      */
     public function render(Zend_View_Interface $view, $element = null)
     {
-        return '<img alt="'.$this->getImgAlt().'" src="' . $this->getImgUrl() . $this->getId() . $this->getSuffix() . '"/><br/>';
+        return '<img alt="'.$this->getImgAlt().'" src="' . $this->getImgUrl() . $this->getId() . $this->getSuffix() . '" width="'.$this->getWidth().'" height="'.$this->getHeight().'"/><br/>';
     }
 }
