Index: tests/Zend/View/Helper/Navigation/TestAbstract.php
===================================================================
--- tests/Zend/View/Helper/Navigation/TestAbstract.php	(revision 25009)
+++ tests/Zend/View/Helper/Navigation/TestAbstract.php	(working copy)
@@ -140,13 +140,17 @@
     }
 
     /**
-     * Returns the contens of the exepcted $file
+     * Returns the contens of the expected $file
      * @param  string $file
      * @return string
      */
     protected function _getExpected($file)
     {
-        return file_get_contents($this->_files . '/expected/' . $file);
+        return str_replace(
+            "\n",
+            PHP_EOL,
+            file_get_contents($this->_files . '/expected/' . $file)
+        );
     }
 
     /**
