Index: library/Zend/Log/Writer/Firebug.php
===================================================================
--- library/Zend/Log/Writer/Firebug.php	(revision 20197)
+++ library/Zend/Log/Writer/Firebug.php	(working copy)
@@ -191,10 +191,10 @@
         $message = $this->_formatter->format($event);
 
         $label = isset($event['firebugLabel'])?$event['firebugLabel']:null;
-
+        $traceOffset = count(debug_backtrace());
         Zend_Wildfire_Plugin_FirePhp::getInstance()->send($message,
                                                           $label,
                                                           $type,
-                                                          array('traceOffset'=>6));
+                                                          array('traceOffset'=>$traceOffset));
     }
 }
