--- library/Zend/Form/Element.php	(Revision 23151)
+++ library/Zend/Form/Element.php	(Arbeitskopie)
@@ -475,11 +475,12 @@
      * Set element name
      *
      * @param  string $name
+     * @param  boolean $allowBrackets
      * @return Zend_Form_Element
      */
-    public function setName($name)
+    public function setName($name, $allowBrackets = false)
     {
-        $name = $this->filterName($name);
+        $name = $this->filterName($name, $allowBrackets);
         if ('' === $name) {
             require_once 'Zend/Form/Exception.php';
             throw new Zend_Form_Exception('Invalid name provided; must contain only valid variable characters and be non-empty');
