Index: View.php
===================================================================
--- View.php	(revision 11060)
+++ View.php	(working copy)
@@ -58,12 +58,6 @@
     public function __construct($config = array())
     {
         $this->_useViewStream = (bool) ini_get('short_open_tag') ? false : true;
-        if ($this->_useViewStream) {
-            if (!in_array('zend.view', stream_get_wrappers())) {
-                require_once 'Zend/View/Stream.php';
-                stream_wrapper_register('zend.view', 'Zend_View_Stream');
-            }
-        }
 
         if (array_key_exists('useStreamWrapper', $config)) {
             $this->setUseStreamWrapper($config['useStreamWrapper']);
@@ -81,6 +75,14 @@
     public function setUseStreamWrapper($flag)
     {
         $this->_useStreamWrapper = (bool) $flag;
+
+        if ($this->_useStreamWrapper) {
+            if (!in_array('zend.view', stream_get_wrappers())) {
+                require_once 'Zend/View/Stream.php';
+                stream_wrapper_register('zend.view', 'Zend_View_Stream');
+            }
+        }
+
         return $this;
     }
 
