Index: Zend/View/Helper/FormErrors.php
===================================================================
--- Zend/View/Helper/FormErrors.php	(revision 14186)
+++ Zend/View/Helper/FormErrors.php	(working copy)
@@ -58,6 +58,16 @@
      */
     public function formErrors($errors, array $options = null)
     {
+        if (isset($options['elementStart'])) {
+            $this->setElementStart($options['elementStart']);
+        }
+        if (isset($options['elementEnd'])) {
+            $this->setElementEnd($options['elementEnd']);
+        }
+        if (isset($options['elementSeparator'])) {
+            $this->setElementSeparator($options['elementSeparator']);
+        }
+
         $escape = true;
         if (isset($options['escape'])) {
             $escape = (bool) $options['escape'];
