--- FormErrors.php.orig	2010-07-13 16:26:10.171875000 +0100
+++ FormErrors.php	2010-07-13 16:25:42.890625000 +0100
@@ -351,7 +351,7 @@
     public function getShowCustomFormErrors()
     {
         if (null === $this->_showCustomFormErrors) {
-            if (null === ($how =  $this->getOption('showCustomFormErrors'))) {
+            if (null === ($show =  $this->getOption('showCustomFormErrors'))) {
                 $this->setShowCustomFormErrors($this->_defaults['showCustomFormErrors']);
             } else {
                 $this->setShowCustomFormErrors($show);
