Index: library/Zend/XmlRpc/Generator/XmlWriter.php
===================================================================
--- library/Zend/XmlRpc/Generator/XmlWriter.php	(Revision 24046)
+++ library/Zend/XmlRpc/Generator/XmlWriter.php	(Arbeitskopie)
@@ -87,6 +87,6 @@
 
     public function saveXml()
     {
-        return $this->_xmlWriter->flush(false);
+        return $this->_xmlWriter->flush(true);
     }
 }
\ No newline at end of file
