Index: library/Zend/Search/Lucene.php
===================================================================
--- library/Zend/Search/Lucene.php	(revision 13711)
+++ library/Zend/Search/Lucene.php	(working copy)
@@ -413,7 +413,7 @@
             if ($this->_formatVersion == self::FORMAT_2_3) {
                 $docStoreOffset = $segmentsFile->readInt();
 
-                if ($docStoreOffset != -1) {
+                if ($docStoreOffset != (int)0xFFFFFFFF) {
                     $docStoreSegment        = $segmentsFile->readString();
                     $docStoreIsCompoundFile = $segmentsFile->readByte();
 
Index: library/Zend/Search/Lucene/Index/Writer.php
===================================================================
--- library/Zend/Search/Lucene/Index/Writer.php	(revision 13711)
+++ library/Zend/Search/Lucene/Index/Writer.php	(working copy)
@@ -496,7 +496,7 @@
                     if ($srcFormat == Zend_Search_Lucene::FORMAT_2_3) {
                         $docStoreOffset = $segmentsFile->readInt();
 
-                        if ($docStoreOffset != -1) {
+                        if ($docStoreOffset != (int)0xFFFFFFFF) {
                             $docStoreSegment        = $segmentsFile->readString();
                             $docStoreIsCompoundFile = $segmentsFile->readByte();
 
