diff --git a/library/Zend/Translator/Adapter.php b/library/Zend/Translator/Adapter.php
index 2c60269..13ca480 100644
--- a/library/Zend/Translator/Adapter.php
+++ b/library/Zend/Translator/Adapter.php
@@ -257,12 +257,12 @@ abstract class Adapter
 
                         $ignore .= $match . '|';
                     } else {
-                        $ignore .= $separator . $match . '|';
+                        $ignore .= preg_quote($separator . $match, '/') . '|';
                     }
                 }
                 $ignore = substr($ignore, 0, -1) . '/u';
             } else {
-                $ignore = '/' . $separator . $options['ignore'] . '/u';
+                $ignore = '/' . preg_quote($separator . $options['ignore'], '/') . '/u';
             }
 
             foreach (new RecursiveIteratorIterator(
