Index: Zend/Mail.php
===================================================================
--- Zend/Mail.php	(revision 3646)
+++ Zend/Mail.php	(working copy)
@@ -198,12 +198,17 @@
     /**
      * Sets the text body for the message.
      *
-     * @param string $txt
+     * @param string|Zend_Mime_Part $txt String to set body as, or explicitly set Zend_Mime_Part with text
      * @param string $charset
      * @return Zend_Mime_Part
     */
     public function setBodyText($txt, $charset=null)
     {
+        if ($txt instanceof Zend_Mime_Part) {
+            $this->_bodyText = $txt;
+            return $txt;
+        }
+        
         if ($charset === null) {
             $charset = $this->_charset;
         }
