Index: ../trunk/library/Zend/Form/Element.php
===================================================================
--- ../trunk/library/Zend/Form/Element.php	(revision 19479)
+++ ../trunk/library/Zend/Form/Element.php	(working copy)
@@ -214,14 +214,11 @@
     protected $_view;
 
     /**
-     * Is a specific decorator being rendered via the magic renderDecorator()?
-     * 
-     * This is to allow execution of logic inside the render() methods of child
-     * elements during the magic call while skipping the parent render() method.
-     * 
+     * Has the preRender hook been called already?
+     *
      * @var bool
      */
-    protected $_isPartialRendering = false;
+    protected $_readyToRender = false;
 
     /**
      * Constructor
@@ -930,9 +927,7 @@
     public function __call($method, $args)
     {
         if ('render' == substr($method, 0, 6)) {
-            $this->_isPartialRendering = true;
-            $this->render();
-            $this->_isPartialRendering = false;
+            $this->preRender();
             $decoratorName = substr($method, 6);
             if (false !== ($decorator = $this->getDecorator($decoratorName))) {
                 $decorator->setElement($this);
@@ -1957,6 +1952,33 @@
     }
 
     /**
+     * Wrap the real pre-render hook to ensure it is called only once.
+     *
+     * @return Zend_Form_Element
+     */
+    public function preRender()
+    {
+        if (!$this->_readyToRender) {
+            $this->_readyToRender = true;
+            $this->_preRender();
+        }
+        return $this;
+    }
+
+    /**
+     * Prepare the element for rendering.
+     *
+     * Pre-rendering logic should be placed here. If this needs to be called
+     * every time render is called, set the $_readyToRender flag to false at the
+     * end of the method.
+     *
+     * @return void
+     */
+    protected function _preRender()
+    {
+    }
+
+    /**
      * Render form element
      *
      * @param  Zend_View_Interface $view
@@ -1964,9 +1986,7 @@
      */
     public function render(Zend_View_Interface $view = null)
     {
-        if ($this->_isPartialRendering) {
-            return '';
-        }
+        $this->preRender();
 
         if (null !== $view) {
             $this->setView($view);
