Index: Partial.php
===================================================================
--- Partial.php	(revision 7365)
+++ Partial.php	(working copy)
@@ -81,7 +81,7 @@
                 if (method_exists($model, 'toArray')) {
                     $view->assign($model->toArray());
                 } else {
-                    $view->assign(get_object_vars($model));
+                    $view->assign('object', $model);
                 }
             }
         }
