--- DateTime_typo.php	2007-09-10 12:32:18.000000000 +0200
+++ DateTime.php	2007-09-10 12:32:24.000000000 +0200
@@ -55,7 +55,7 @@
         } else {
             $value = strtotime($value);
             if ($value === false || $value == -1) { // cannot convert the value to a timestamp
-                throw new Zend_Xml_Rpc_Value_Exception('Cannot convert given value \''. $value .'\' to a timestamp');
+                throw new Zend_XmlRpc_Value_Exception('Cannot convert given value \''. $value .'\' to a timestamp');
             }
         }
         $value = date('c', $value); // Convert the timestamp to iso8601 format
