Index: library/Zend/View/Helper/HeadStyle.php
===================================================================
--- library/Zend/View/Helper/HeadStyle.php	(revision 13182)
+++ library/Zend/View/Helper/HeadStyle.php	(working copy)
@@ -339,6 +339,13 @@
               . $indent . '<!--' . PHP_EOL . $indent . $item->content . PHP_EOL . $indent . '-->' . PHP_EOL
               . '</style>';
 
+        if (isset($item->attributes['conditional'])
+            && !empty($item->attributes['conditional'])
+            && is_string($item->attributes['conditional']))
+        {
+            $html = '<!--[if ' . $item->attributes['conditional'] . ']> ' . $html . '<![endif]-->';
+        }
+
         return $html;
     }
 
Index: library/Zend/View/Helper/HeadScript.php
===================================================================
--- library/Zend/View/Helper/HeadScript.php	(revision 13181)
+++ library/Zend/View/Helper/HeadScript.php	(working copy)
@@ -415,6 +415,13 @@
         }
         $html .= '</script>';
 
+        if (isset($item->attributes['conditional'])
+            && !empty($item->attributes['conditional'])
+            && is_string($item->attributes['conditional']))
+        {
+            $html = '<!--[if ' . $item->attributes['conditional'] . ']> ' . $html . '<![endif]-->';
+        }
+
         return $html;
     }
 
Index: tests/Zend/View/Helper/HeadStyleTest.php
===================================================================
--- tests/Zend/View/Helper/HeadStyleTest.php	(revision 13182)
+++ tests/Zend/View/Helper/HeadStyleTest.php	(working copy)
@@ -375,6 +375,16 @@
         $this->assertContains(' media="screen,projection"', $string);
 
     }
+
+    public function testConditionalScript()
+    {
+        $this->helper->appendStyle('
+a {
+    display: none;
+}', array('media' => 'screen,projection', 'conditional' => 'lt IE 7'));
+        $test = $this->helper->toString();
+        $this->assertContains('<!--[if lt IE 7]>', $test);
+    }
 }
 
 // Call Zend_View_Helper_HeadStyleTest::main() if this source file is executed directly.
Index: tests/Zend/View/Helper/HeadScriptTest.php
===================================================================
--- tests/Zend/View/Helper/HeadScriptTest.php	(revision 13181)
+++ tests/Zend/View/Helper/HeadScriptTest.php	(working copy)
@@ -391,6 +391,13 @@
         $this->helper->setAutoEscape(false)->appendFile('test.js?id=123&foo=bar');
         $this->assertEquals('<script type="text/javascript" src="test.js?id=123&foo=bar"></script>', $this->helper->toString());
     }
+
+    public function testConditionalScript()
+    {
+        $this->helper->headScript()->appendFile('/js/foo.js', 'text/javascript', array('conditional' => 'lt IE 7'));
+        $test = $this->helper->headScript()->toString();
+        $this->assertContains('<!--[if lt IE 7]>', $test);
+    }
 }
 
 // Call Zend_View_Helper_HeadScriptTest::main() if this source file is executed directly.
