--- Abstract.php.ORIG	2010-10-07 20:31:14.000000000 +0200
+++ Abstract.php	2010-10-07 20:32:14.000000000 +0200
@@ -83,7 +83,7 @@
     public function getModuleName()
     {
         if (null === $this->_module) {
-            $this->_module = $this->getParam($this->getModuleKey());
+            $this->_module = strtolower($this->getParam($this->getModuleKey()));
         }
 
         return $this->_module;
@@ -97,7 +97,7 @@
      */
     public function setModuleName($value)
     {
-        $this->_module = $value;
+        $this->_module = strtolower($value);
         return $this;
     }
 
