Index: library/Zend/Cache/Frontend/Page.php
===================================================================
--- library/Zend/Cache/Frontend/Page.php	(revision 10891)
+++ library/Zend/Cache/Frontend/Page.php	(working copy)
@@ -296,16 +296,14 @@
         }
         $contentType = null;
         $storedHeaders = array();
-        if (headers_sent()) {
-            $headersList = headers_list();
-            foreach($this->_specificOptions['memorize_headers'] as $key=>$headerName) {
-                foreach ($headersList as $headerSent) {
-                    $tmp = split(':', $headerSent);
-                    $headerSentName = trim(array_shift($tmp));
-                    if (strolower($headerName) == strtolower($headerSentName)) {
-                        $headerSentValue = trim(implode(':', $tmp));
-                        $storedHeaders[] = array($headerSentName, $headerSentValue);
-                    }
+        $headersList = headers_list();
+        foreach($this->_specificOptions['memorize_headers'] as $key=>$headerName) {
+            foreach ($headersList as $headerSent) {
+                $tmp = split(':', $headerSent);
+                $headerSentName = trim(array_shift($tmp));
+                if (strolower($headerName) == strtolower($headerSentName)) {
+                    $headerSentValue = trim(implode(':', $tmp));
+                    $storedHeaders[] = array($headerSentName, $headerSentValue);
                 }
             }
         }
