Index: tests/Zend/Oauth/Oauth/Http/RequestTokenTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Http/RequestTokenTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Http/RequestTokenTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Http/RequestToken.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Http
+ */
 class Zend_Oauth_Http_RequestTokenTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Http/UserAuthorizationTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Http/UserAuthorizationTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Http/UserAuthorizationTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Http/UserAuthorization.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Http
+ */
 class Zend_Oauth_Http_UserAuthorizationTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Http/AccessTokenTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Http/AccessTokenTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Http/AccessTokenTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Http/AccessToken.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Http
+ */
 class Zend_Oauth_Http_AccessTokenTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Http/UtilityTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Http/UtilityTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Http/UtilityTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Http/Utility.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Http
+ */
 class Zend_Oauth_Http_UtilityTest extends PHPUnit_Framework_TestCase
 {
     // see: http://wiki.oauth.net/TestCases (Parameter Encoding Tests)
Index: tests/Zend/Oauth/Oauth/Token/AccessTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Token/AccessTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Token/AccessTest.php	(working copy)
@@ -1,8 +1,37 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Token/Access.php';
 require_once 'Zend/Oauth/Config.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Token
+ */
 class Zend_Oauth_Token_AccessTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Token/AuthorizedRequestTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Token/AuthorizedRequestTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Token/AuthorizedRequestTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Token/AuthorizedRequest.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Token
+ */
 class Zend_Oauth_Token_AuthorizedRequestTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Token/RequestTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Token/RequestTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Token/RequestTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Token/Request.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Token
+ */
 class Zend_Oauth_Token_RequestTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Signature/AbstractTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Signature/AbstractTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Signature/AbstractTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Signature/Plaintext.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Signature
+ */
 class Zend_Oauth_Signature_AbstractTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Signature/RsaTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Signature/RsaTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Signature/RsaTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 //require_once 'Zend/Oauth/Signature/Rsa.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Signature
+ */
 class Zend_Oauth_Signature_RsaTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Signature/HmacTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Signature/HmacTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Signature/HmacTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Signature/Hmac.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Signature
+ */
 class Zend_Oauth_Signature_HmacTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/Signature/PlaintextTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/Signature/PlaintextTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/Signature/PlaintextTest.php	(working copy)
@@ -1,7 +1,36 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Signature/Plaintext.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ * @group      Zend_Oauth_Signature
+ */
 class Zend_Oauth_Signature_PlaintextTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/Oauth/ConsumerTest.php
===================================================================
--- tests/Zend/Oauth/Oauth/ConsumerTest.php	(revision 23882)
+++ tests/Zend/Oauth/Oauth/ConsumerTest.php	(working copy)
@@ -1,7 +1,35 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth/Consumer.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ */
 class Zend_Oauth_ConsumerTest extends PHPUnit_Framework_TestCase
 {
 
Index: tests/Zend/Oauth/AllTests.php
===================================================================
--- tests/Zend/Oauth/AllTests.php	(revision 23882)
+++ tests/Zend/Oauth/AllTests.php	(working copy)
@@ -1,4 +1,24 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 if (!defined('PHPUnit_MAIN_METHOD')) {
     define('PHPUnit_MAIN_METHOD', 'Zend_Oauth_AllTests::main');
@@ -18,6 +38,14 @@
 require_once 'Oauth/Token/AuthorizedRequestTest.php';
 require_once 'Oauth/Token/AccessTest.php';
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ */
 class Zend_Oauth_AllTests
 {
     public static function main()
Index: tests/Zend/Oauth/OauthTest.php
===================================================================
--- tests/Zend/Oauth/OauthTest.php	(revision 23882)
+++ tests/Zend/Oauth/OauthTest.php	(working copy)
@@ -1,9 +1,37 @@
 <?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
 
 require_once 'Zend/Oauth.php';
 
 class Test_Http_Client_19485876 extends Zend_Http_Client {}
 
+/**
+ * @category   Zend
+ * @package    Zend_Oauth
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @group      Zend_Oauth
+ */
 class Zend_OauthTest extends PHPUnit_Framework_TestCase
 {
 
