Index: Zend/Controller/Router/Route/Module.php
===================================================================
--- Zend/Controller/Router/Route/Module.php	(revision 10182)
+++ Zend/Controller/Router/Route/Module.php	(working copy)
@@ -221,20 +221,20 @@
         unset($params[$this->_actionKey]);
 
         foreach ($params as $key => $value) {
-            $url .= '/' . $key;
-            $url .= '/' . $value;
+            $url .= self::URI_DELIMITER . $key;
+            $url .= self::URI_DELIMITER . $value;
         }
 
         if (!empty($url) || $action !== $this->_defaults[$this->_actionKey]) {
-            $url = '/' . $action . $url;
+            $url = self::URI_DELIMITER . $action . $url;
         }
 
         if (!empty($url) || $controller !== $this->_defaults[$this->_controllerKey]) {
-            $url = '/' . $controller . $url;
+            $url = self::URI_DELIMITER . $controller . $url;
         }
 
         if (isset($module)) {
-            $url = '/' . $module . $url;
+            $url = self::URI_DELIMITER . $module . $url;
         }
 
         return ltrim($url, self::URI_DELIMITER);
