Index: tests/Zend/Reflection/_files/TestZF6961Class.php
===================================================================
--- tests/Zend/Reflection/_files/TestZF6961Class.php	(revision 0)
+++ tests/Zend/Reflection/_files/TestZF6961Class.php	(revision 0)
@@ -0,0 +1,11 @@
+<?php
+class Zend_Reflection_TestZF6961Class
+{
+    /**
+     * @param    string $one
+     * @param   string $two
+     */
+    public function doSomething($one, $two)
+    {
+    }
+}
\ No newline at end of file
Index: tests/Zend/Reflection/DocblockTest.php
===================================================================
--- tests/Zend/Reflection/DocblockTest.php	(revision 15950)
+++ tests/Zend/Reflection/DocblockTest.php	(working copy)
@@ -25,6 +25,7 @@
 require_once dirname(__FILE__) . '/../../TestHelper.php';
 
 /** requires */
+require_once dirname(__FILE__) . '/_files/TestZF6961Class.php';
 require_once 'Zend/Reflection/File.php';
 
 /**
@@ -89,6 +90,16 @@
         $this->assertEquals($returnTag->getType(), 'mixed');
     }
     
+    public function testZF6961()
+    {
+        /**
+         * This will trhow an exception unless ZF-6961 is fixed
+         */
+        $classReflection = new Zend_Reflection_Class('Zend_Reflection_TestZF6961Class');
+        
+        $classReflection->getMethod('doSomething')->getDocblock()->getTags();
+    }
+    
     public function testDocblockLines()
     {
         $classReflection = new Zend_Reflection_Class('Zend_Reflection_TestSampleClass5');
Index: library/Zend/Reflection/Docblock/Tag/Param.php
===================================================================
--- library/Zend/Reflection/Docblock/Tag/Param.php	(revision 15950)
+++ library/Zend/Reflection/Docblock/Tag/Param.php	(working copy)
@@ -49,7 +49,7 @@
     {
         $matches = array();
         
-        if (!preg_match('#^@(\w+)\s(\w+)(?:\s(\$\S+))?(?:\s(.*))?#s', $tagDocblockLine, $matches)) {
+        if (!preg_match('#^@(\w+)\s+(\w+)(?:\s(\$\S+))?(?:\s(.*))?#s', $tagDocblockLine, $matches)) {
             require_once 'Zend/Reflection/Exception.php';
             throw new Zend_Reflection_Exception('Provided docblock line is does not contain a valid tag');
         }

