--- library/Zend/Pdf/Page.php	(revision 21561)
+++ library/Zend/Pdf/Page.php	(working copy)
@@ -888,8 +888,8 @@
      * @param float $x
      * @param float $y
      * @param float $radius
-     * @param float $startAngle
-     * @param float $endAngle
+     * @param float $startAngle - in radians
+     * @param float $endAngle - in radians
      * @return Zend_Pdf_Page
      */
     public function clipCircle($x, $y, $radius, $startAngle = null, $endAngle = null)
@@ -914,8 +914,8 @@
      * @param float $y1
      * @param float $x2
      * @param float $y2
-     * @param float $startAngle
-     * @param float $endAngle
+     * @param float $startAngle - in radians
+     * @param float $endAngle - in radians
      * @return Zend_Pdf_Page
      */
     public function clipEllipse($x1, $y1, $x2, $y2, $startAngle = null, $endAngle = null)
@@ -1690,7 +1690,7 @@
      *
      * @param float $x  - the X co-ordinate of rotation point
      * @param float $y  - the Y co-ordinate of rotation point
-     * @param float $angle - rotation angle
+     * @param float $angle - rotation angle (in radians)
      * @return Zend_Pdf_Page
      */
     public function rotate($x, $y, $angle)
