Index: incubator/Zend/Http/Client.php
===================================================================
--- incubator/Zend/Http/Client.php	(revision 846)
+++ incubator/Zend/Http/Client.php	(revision 847)
@@ -759,7 +759,11 @@
         
         // Set the host header
         if (! isset($this->headers['host'])) {
-            $host = $this->uri->getHost() . ($this->uri->getPort() == 80 ? '' : ':' . $this->uri->getPort());
+            $host = $this->uri->getHost();
+            if (!(($this->uri->getScheme() == 'http' && $this->uri->getPort() == 80)
+                  || ($this->uri->getScheme() == 'https' && $this->uri->getPort() == 443))) {
+                $host .= ':' . $this->uri->getPort();
+            }
             $headers[] = "Host: {$host}";
         }
         
