Index: RewriteRouter.php
===================================================================
--- RewriteRouter.php	(revision 914)
+++ RewriteRouter.php	(working copy)
@@ -139,12 +139,12 @@
             }
         }
 
-        $actionObj = new Zend_Controller_Dispatcher_Token($controller, $action, $params);
+        $token = new Zend_Controller_Dispatcher_Token($controller, $action, $params);
 
-        if (!$dispatcher->isDispatchable($actionObj)) {
+        if (!$dispatcher->isDispatchable($token)) {
             throw new Zend_Controller_Router_Exception('Request could not be mapped to a route.');
         } else {
-            return $actionObj;
+            return $token;
         }
 
     }
