--- trunk/library/Zend/Validate/File/ImageSize.php	(revision 11372)
+++ trunk/library/Zend/Validate/File/ImageSize.php	(working copy)
@@ -52,7 +52,7 @@
         self::WIDTH_TOO_SMALL  => "Width of the image '%value%' is smaller than allowed",
         self::HEIGHT_TOO_BIG   => "Height of the image '%value%' is bigger than allowed",
         self::HEIGHT_TOO_SMALL => "Height of the image '%value%' is smaller than allowed",
-        self::NOT_DETECTED     => "Size if the image '%value%' could not be detected",
+        self::NOT_DETECTED     => "Size of the image '%value%' could not be detected",
         self::NOT_READABLE     => "The image '%value%' can not be read"
     );
 
