--- library/Zend/Service/Rackspace/Servers.php	(revision 25033)
+++ library/Zend/Service/Rackspace/Servers.php	(working copy)
@@ -1006,7 +1006,7 @@
             case '200' : 
             case '203' : // break intentionally omitted   
                 $backup = json_decode($result->getBody(),true);
-                return $image['backupSchedule'];
+                return $backup['backupSchedule'];
             case '503' :
                 $this->errorMsg= self::ERROR_SERVICE_UNAVAILABLE;
                 break;
@@ -1278,4 +1278,4 @@
         $this->errorCode= $status;
         return false;
     }
-}
\ No newline at end of file
+}
