Index: Data.php
===================================================================
--- Data.php	(revision 7119)
+++ Data.php	(working copy)

@@ -263,9 +263,14 @@
                 $val = implode('_' , $value);
             }
             $id = strtr('Zend_Locale_' . $locale . '_' . $path . '_' . $val, array('-' => '_', '/' => '_'));
-            if ($result = self::$_cache->load($id)) {
-                return unserialize($result);
+            if(Zend_Registry::isRegistered($id)){
+                return Zend_Registry::get($id);
             }
+            elseif ($result = self::$_cache->load($id)) {
+                $result = unserialize($result);
+                Zend_Registry::set($id, $result);
+                return $result;
+            }
         }