Index: Zend/Feed/Atom.php
===================================================================
--- Zend/Feed/Atom.php	(révision 4296)
+++ Zend/Feed/Atom.php	(copie de travail)
@@ -270,7 +270,7 @@
             $title = $this->_element->createElement('title', $dataentry->title);
             $entry->appendChild($title);
 
-            $updated = isset($array->lastUpdate) ? $array->lastUpdate : time();
+            $updated = isset($dataentry->lastUpdate) ? $dataentry->lastUpdate : time();
             $updated = $this->_element->createElement('updated', date(DATE_ATOM, $updated));
             $entry->appendChild($updated);
 
