Index: Ini.php
===================================================================
--- Ini.php	(revision 13608)
+++ Ini.php	(working copy)
@@ -109,14 +109,17 @@
             $iniString .= '[' . $sectionName . ']' . "\n"
                        .  $this->_addBranch($this->_config)
                        .  "\n";
-        } else {       
-            foreach ($this->_config as $sectionName => $data) {
+        } else {   
+            foreach ($this->_config as $key => $data) {
                 if (isset($extends[$sectionName])) {
                     $sectionName .= ' : ' . $extends[$sectionName];
                 }
                 
-                $iniString .= '[' . $sectionName . ']' . "\n"
-                           .  $this->_addBranch($data)
+                if (isset($sectionName)) {
+                	$iniString .= '[' . $sectionName . ']' . "\n";
+                }
+                
+                $iniString .= $this->_addBranch($this->_config)
                            .  "\n";
             }
         }
