Index: Zend/Cache/Core.php
===================================================================
--- Zend/Cache/Core.php	(revision 17344)
+++ Zend/Cache/Core.php	(working copy)
@@ -124,12 +124,16 @@
     /**
      * Constructor
      *
-     * @param  array $options Associative array of options
+     * @param  array|Zend_Config $options Associative array of options
      * @throws Zend_Cache_Exception
      * @return void
      */
-    public function __construct(array $options = array())
+    public function __construct($options = array())
     {
+        if ($options instanceof Zend_Config) {
+            $options = $options->toArray();
+        }
+
         while (list($name, $value) = each($options)) {
             $this->setOption($name, $value);
         }
@@ -520,7 +524,7 @@
         }
         return $this->_backend->getFillingPercentage();
     }
-
+
     /**
      * Return an array of metadatas for the given cache id
      *
