Index: DtDdWrapper.php
===================================================================
--- DtDdWrapper.php	(revision 23872)
+++ DtDdWrapper.php	(working copy)
@@ -64,7 +64,10 @@
             $dtLabel = '&#160;';
         }
 
-        return '<dt id="' . $elementName . '-label">' . $dtLabel . '</dt>' .
-               '<dd id="' . $elementName . '-element">' . $content . '</dd>';
+        $dtClass = $this->getOption('dtClass') ? ' class="' . $this->getOption('dtClass') . '"' : '';
+        $ddClass = $this->getOption('ddClass') ? ' class="' . $this->getOption('ddClass') . '"' : '';
+
+        return '<dt id="' . $elementName . '-label"' . $dtClass . '>' . $dtLabel . '</dt>' .
+               '<dd id="' . $elementName . '-element"' . $ddClass . '>' . $content . '</dd>';
     }
 }
