Index: Zend/Application/Resource/Mail.php
===================================================================
--- Zend/Application/Resource/Mail.php	(revision 22101)
+++ Zend/Application/Resource/Mail.php	(working copy)
@@ -107,20 +107,16 @@
     	}
     	
         $transportName = $options['type'];
-        if(!Zend_Loader_Autoloader::autoload($transportName))
-        {
-            $transportName = ucfirst(strtolower($transportName));
 
-            if(!Zend_Loader_Autoloader::autoload($transportName))
-            {
-                $transportName = 'Zend_Mail_Transport_' . $transportName;
-                if(!Zend_Loader_Autoloader::autoload($transportName)) {
-                    throw new Zend_Application_Resource_Exception(
-                        "Specified Mail Transport '{$transportName}'"
-                        . 'could not be found'
-                    );
-                }
-            }
+        if(substr($transportName, 0, 20) != 'Zend_Mail_Transport_') {
+        	$transportName = 'Zend_Mail_Transport_' . ucfirst(strtolower($transportName));
+        }
+
+        if(!Zend_Loader_Autoloader::autoload($transportName)) {
+            throw new Zend_Application_Resource_Exception(
+                "Specified Mail Transport '{$transportName}'"
+                . 'could not be found'
+            );
         }
         
         unset($options['type']);
