--- trunk/library/Zend/Validate/File/IsImage.php	(revision 19871)
+++ trunk/library/Zend/Validate/File/IsImage.php	(working copy)
@@ -35,13 +35,6 @@
 class Zend_Validate_File_IsImage extends Zend_Validate_File_MimeType
 {
     /**
-     * @const string Error constants
-     */
-    const FALSE_TYPE   = 'fileIsImageFalseType';
-    const NOT_DETECTED = 'fileIsImageNotDetected';
-    const NOT_READABLE = 'fileIsImageNotReadable';
-
-    /**
      * @var array Error message templates
      */
     protected $_messageTemplates = array(
