--- /Zend/View/Helper/Navigation/Menu.php~	2010-01-25 15:18:56.448507644 +0100
+++ /Zend/View/Helper/Navigation/Menu.php	2010-01-25 15:03:26.792506746 +0100
@@ -442,8 +442,8 @@
             }
 
             // render li tag and page
-            $liClass = $isActive ? ' class="active"' : '';
-            $html .= $myIndent . '    <li' . $liClass . '>' . self::EOL
+            $liClass = $isActive ? ' active' : '';
+            $html .= $myIndent . '    <li class="' . $page->getClass() . $liClass . '">' . self::EOL
                    . $myIndent . '        ' . $this->htmlify($page) . self::EOL;
 
             // store as previous depth for next iteration
