Index: library/Zend/View/Helper/FormSelect.php
===================================================================
--- library/Zend/View/Helper/FormSelect.php	(revision 6278)
+++ library/Zend/View/Helper/FormSelect.php	(working copy)
@@ -91,7 +91,7 @@
             // show the label, not the value, of the option.
             $list = array();
             foreach ($options as $opt_value => $opt_label) {
-                if (in_array($opt_value, $value)) {
+                if (in_array($opt_value, $value, $opt_value === 0)) {
                     // add the hidden value
                     $opt = $this->_hidden($name, $opt_value);
                     // add the display label
@@ -151,7 +151,7 @@
              . ' label="' . $this->view->escape($label) . '"';
 
         // selected?
-        if (in_array($value, $selected)) {
+        if (in_array($value, $selected, $value === 0)) {
             $opt .= ' selected="selected"';
         }
 
