### Eclipse Workspace Patch 1.0
#P ZF-STD-TRUNK
Index: library/Zend/CodeGenerator/Php/Property/DefaultValue.php
===================================================================
--- library/Zend/CodeGenerator/Php/Property/DefaultValue.php	(revision 20434)
+++ library/Zend/CodeGenerator/Php/Property/DefaultValue.php	(working copy)
@@ -286,7 +286,7 @@
                 $curArrayMultiblock = false;
                 if (count($value) > 1) {
                     $curArrayMultiblock = true;
-                    $output .= PHP_EOL . str_repeat($this->_indentation, $this->_arrayDepth+1);
+                    $output .= self::LINE_FEED . str_repeat($this->_indentation, $this->_arrayDepth+1);
                 }
                 $outputParts = array();
                 $noKeyIndex = 0;
@@ -302,9 +302,9 @@
                     }
 
                 }
-                $output .= implode(',' . PHP_EOL . str_repeat($this->_indentation, $this->_arrayDepth+1), $outputParts);
+                $output .= implode(',' . self::LINE_FEED . str_repeat($this->_indentation, $this->_arrayDepth+1), $outputParts);
                 if ($curArrayMultiblock == true) {
-                    $output .= PHP_EOL . str_repeat($this->_indentation, $this->_arrayDepth+1);
+                    $output .= self::LINE_FEED . str_repeat($this->_indentation, $this->_arrayDepth+1);
                 }
                 $output .= ')';
                 break;
