diff --git a/library/Zend/Service/Amazon/SimpleDb.php b/library/Zend/Service/Amazon/SimpleDb.php
index 5365192..d290381 100644
--- a/library/Zend/Service/Amazon/SimpleDb.php
+++ b/library/Zend/Service/Amazon/SimpleDb.php
@@ -402,7 +402,7 @@ class Zend_Service_Amazon_SimpleDb extends Zend_Service_Amazon_Abstract
             }
         }
 
-        $nextToken = (string)$xml->NextToken;
+        $nextToken = (string)$xml->SelectResult->NextToken;
 
         return new Zend_Service_Amazon_SimpleDb_Page($attributes, $nextToken);
     }
