Index: Server/Reflection/Function/Abstract.php
===================================================================
--- Server/Reflection/Function/Abstract.php	(revision 4880)
+++ Server/Reflection/Function/Abstract.php	(working copy)
@@ -293,7 +293,7 @@
             // Get param types and description
             if (preg_match_all('/@param\s+([^\s]+)/m', $docBlock, $matches)) {
                 $paramTypesTmp = $matches[1];
-                if (preg_match_all('/@param\s+\S+\s+(\$^\S+)\s+(.*?)(@|\*\/)/s', $docBlock, $matches))
+                if (preg_match_all(':@param\s+\S+\s+(\$\S+)\s+(.*?)(?=@|\*/):s', $docBlock, $matches))
                 {
                     $paramDesc = $matches[2];
                     foreach ($paramDesc as $key => $value) {
