Index: Gdata/YoutubeHttpClient.php
===================================================================
--- Gdata/YoutubeHttpClient.php	(revision 0)
+++ Gdata/YoutubeHttpClient.php	(revision 0)
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Gdata
+ * @subpackage Gdata
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+/**
+ * Zend_Gdata_HttpClient
+ */
+require_once 'Zend/Gdata/HttpClient.php';
+
+/**
+ * Gdata Http Client object.
+ *
+ * Class to extend the generic Zend Youtube Http Client with the ability to perform
+ * secure AuthSub requests and to retrieve additional response params
+ *
+ * @category   Zend
+ * @package    Zend_Gdata
+ * @subpackage Gdata
+ * @copyright  Copyright (c) 2005-2009 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_Gdata_YoutubeHttpClient extends Zend_Gdata_HttpClient
+{
+    /**
+     * @var string Canonized internal Youtube username
+     */
+    private $_actualYoutubeUsername = null;
+    /**
+     * @param $goog_resp array
+     * Template method
+     */
+    public function handleAdditionalResponseParams ($goog_resp){
+        if(!isset($goog_resp['YouTubeUser'])){
+            /**
+             * This means you specified not youtube client login url but more unified google one. 
+             * In this case Google.com does not provide youtube internal username
+             * @todo throw custom exception
+             */
+            throw new RuntimeException('Please specify CLIENTLOGIN_URL=`https://www.google.com/youtube/accounts/ClientLogin` instead of `https://www.google.com/accounts/ClientLogin`');
+        }
+        $this->_actualYoutubeUsername = $goog_resp['YouTubeUser'];
+    } 
+    
+    public function getActualYoutubeUsername(){
+        return $this->_actualYoutubeUsername;
+    }
+}
\ No newline at end of file

Property changes on: Gdata/YoutubeHttpClient.php
___________________________________________________________________
Added: svn:executable
   + *
Added: svn:keywords
   + LastChangedDate LastChangedRevision LastChangedBy Date Revision Author HeadURL Id

Index: Gdata/ClientLogin.php
===================================================================
--- Gdata/ClientLogin.php	(revision 18493)
+++ Gdata/ClientLogin.php	(working copy)
@@ -94,7 +94,12 @@
         }
 
         if ($client == null) {
-            $client = new Zend_Gdata_HttpClient();
+            if ($service == 'youtube'){
+                $client = new Zend_Gdata_YoutubeHttpClient();
+            }
+            else{
+                $client = new Zend_Gdata_HttpClient();
+            }
         }
         if (!$client instanceof Zend_Http_Client) {
             require_once 'Zend/Gdata/App/HttpException.php';
@@ -154,6 +159,7 @@
 
         if ($response->getStatus() == 200) {
             $client->setClientLoginToken($goog_resp['Auth']);
+            $client->handleAdditionalResponseParams($goog_resp);
             $useragent = $source . ' Zend_Framework_Gdata/' . Zend_Version::VERSION;
             $client->setConfig(array(
                     'strictredirects' => true,
Index: Gdata/HttpClient.php
===================================================================
--- Gdata/HttpClient.php	(revision 18493)
+++ Gdata/HttpClient.php	(working copy)
@@ -344,4 +344,12 @@
         return $this->setRawData($data, $enctype);
     }
 
+    /**
+     * @param $goog_resp array
+     * Template method
+     * Default action - to do nothing
+     */
+    public function handleAdditionalResponseParams ($goog_resp){
+        
+    }
 }
