Index: FormCheckbox.php
===================================================================
--- FormCheckbox.php	(revision 21015)
+++ FormCheckbox.php	(working copy)
@@ -72,6 +72,10 @@
             $checked = false;
             unset($attribs['checked']);
         }
+        if(isset($attribs['disabled']) && $attribs['disabled']) {
+            $disable = true;
+            unset($attribs['disabled']);
+        }
 
         $checkedOptions = self::determineCheckboxInfo($value, $checked, $checkedOptions);
 
