Index: Chain.php
===================================================================
--- Chain.php	(revision 19198)
+++ Chain.php	(working copy)
@@ -81,6 +81,7 @@
                 $separator = substr($subPath, 0, strlen($this->_separators[$key]));
 
                 if ($separator !== $this->_separators[$key]) {
+                    $request->setPathInfo($path);
                     return false;
                 }
 
@@ -97,6 +98,7 @@
 
             $res = $route->match($match, true);
             if ($res === false) {
+                $request->setPathInfo($path);
                 return false;
             }
 
