Index: PrepareElements.php
===================================================================
--- PrepareElements.php	(revision 23986)
+++ PrepareElements.php	(working copy)
@@ -72,12 +72,14 @@
                  ->setTranslator($translator);
             if ($item instanceof Zend_Form_Element) {
                 $item->setBelongsTo($belongsTo);
-            } elseif (!empty($belongsTo) && ($item instanceof Zend_Form)) {
-                if ($item->isArray()) {
-                    $name = $this->mergeBelongsTo($belongsTo, $item->getElementsBelongTo());
-                    $item->setElementsBelongTo($name, true);
-                } else {
-                    $item->setElementsBelongTo($belongsTo, true);
+            } elseif ($item instanceof Zend_Form) {
+                if (!empty($belongsTo)) {
+                    if ($item->isArray()) {
+                        $name = $this->mergeBelongsTo($belongsTo, $item->getElementsBelongTo());
+                        $item->setElementsBelongTo($name, true);
+                    } else {
+                        $item->setElementsBelongTo($belongsTo, true);
+                    }
                 }
                 $this->_recursivelyPrepareForm($item);
             } elseif (!empty($belongsTo) && ($item instanceof Zend_Form_DisplayGroup)) {
