Index: Response/Http.php
===================================================================
--- Response/Http.php	(revision 18436)
+++ Response/Http.php	(working copy)
@@ -44,6 +44,7 @@
             header('Cache-Control: cache, must-revalidate');
             header('Pragma: public');
             header('Content-Type: application/x-amf');
+            header('Content-Length: '.$this->_outputStream->getStreamLength());
         }
         return parent::getResponse();
     }
Index: Util/BinaryStream.php
===================================================================
--- Util/BinaryStream.php	(revision 18436)
+++ Util/BinaryStream.php	(working copy)
@@ -85,6 +85,16 @@
     }
 
     /**
+     * Returns the current stream length
+     *
+     * @return int
+     */
+    public function getStreamLength()
+    {
+        return strlen($this->_stream);
+    }
+
+    /**
      * Read the number of bytes in a row for the length supplied.
      *
      * @todo   Should check that there are enough bytes left in the stream we are about to read.
