Index: Zend/Http/Client.php
===================================================================
--- Zend/Http/Client.php	(revision 20156)
+++ Zend/Http/Client.php	(working copy)
@@ -1349,7 +1354,7 @@
         $ret = "--{$boundary}\r\n" .
             'Content-Disposition: form-data; name="' . $name .'"';
 
-        if ($filename) {
+        if ($filename !== null) {
             $ret .= '; filename="' . $filename . '"';
         }
         $ret .= "\r\n";
