Index: tests/Zend/Db/Table/Relationships/TestCommon.php
===================================================================
--- tests/Zend/Db/Table/Relationships/TestCommon.php	(revision 21084)
+++ tests/Zend/Db/Table/Relationships/TestCommon.php	(working copy)
@@ -194,12 +194,16 @@
 
         $originRows = $table->find(1);
         $originRow1 = $originRows->current();
+        
+        $productColumns = $this->_table['products']->info(Zend_Db_Table_Abstract::COLS);
 
         $destRows = $originRow1->findManyToManyRowset('My_ZendDbTable_TableProducts', 'My_ZendDbTable_TableBugsProducts');
         $this->assertType('Zend_Db_Table_Rowset_Abstract', $destRows,
             'Expecting object of type Zend_Db_Table_Rowset_Abstract, got '.get_class($destRows));
 
         $this->assertEquals(3, $destRows->count());
+        $firstRow = $destRows->current();
+        $this->assertEquals($productColumns, array_keys($firstRow->toArray()), "Columns in results don't match table's columns");
     }
 
     public function testTableRelationshipFindManyToManyRowsetSelect()
Index: library/Zend/Db/Table/Row/Abstract.php
===================================================================
--- library/Zend/Db/Table/Row/Abstract.php	(revision 21084)
+++ library/Zend/Db/Table/Row/Abstract.php	(working copy)
@@ -1051,7 +1051,7 @@
         }
         $joinCond = implode(' AND ', $joinCond);
 
-        $select->from(array('i' => $interName), Zend_Db_Select::SQL_WILDCARD, $interSchema)
+        $select->from(array('i' => $interName), array(), $interSchema)
                ->joinInner(array('m' => $matchName), $joinCond, Zend_Db_Select::SQL_WILDCARD, $matchSchema)
                ->setIntegrityCheck(false);
 
