Index: library/Zend/Form/Decorator/FormElements.php
===================================================================
--- library/Zend/Form/Decorator/FormElements.php	(revision 25030)
+++ library/Zend/Form/Decorator/FormElements.php	(working copy)
@@ -82,8 +82,13 @@
         $items          = array();
         $view           = $form->getView();
         foreach ($form as $item) {
-            $item->setView($view)
-                 ->setTranslator($translator);
+            $item->setView($view);
+
+            // Set translator
+            if (!$item->hasTranslator()) {
+                $item->setTranslator($translator);
+            }
+
             if ($item instanceof Zend_Form_Element) {
                 foreach ($displayGroups as $group) {
                     $elementName = $item->getName();
