Index: library/Zend/Search/Lucene/Document/Html.php
===================================================================
--- library/Zend/Search/Lucene/Document/Html.php	(revision 20191)
+++ library/Zend/Search/Lucene/Document/Html.php	(working copy)
@@ -198,7 +198,6 @@
     {
         if ($node->nodeType == XML_TEXT_NODE) {
             $text .= $node->nodeValue ;
-            $text .= ' ';
         } else if ($node->nodeType == XML_ELEMENT_NODE  &&  $node->nodeName != 'script') {
             foreach ($node->childNodes as $childNode) {
                 $this->_retrieveNodeText($childNode, $text);
