Index: Zend/Search/Lucene/LuceneTest.php
===================================================================
--- Zend/Search/Lucene/LuceneTest.php	(révision 12429)
+++ Zend/Search/Lucene/LuceneTest.php	(copie de travail)
@@ -143,6 +143,28 @@
         $this->assertEquals(count($hits), 3);
     }
 
+    /**
+     * @group ZF-2982
+     */
+    public function testUppercaseKeywordFind()
+    {
+        $index = Zend_Search_Lucene::create(dirname(__FILE__) . '/_index/_files');
+
+        // Zero terms
+        $doc = new Zend_Search_Lucene_Document();
+        $doc->addField(Zend_Search_Lucene_Field::Keyword('keyword', 'Value'));
+        $index->addDocument($doc);
+
+        unset($index);
+
+        $index = Zend_Search_Lucene::open(dirname(__FILE__) . '/_index/_files');
+
+        $this->assertEquals(1, count($index->find(Zend_Search_Lucene_Search_QueryParser::parse('Value'))));
+        $this->assertEquals(1, count($index->find(Zend_Search_Lucene_Search_QueryParser::parse('value'))));
+
+        $this->_clearDirectory(dirname(__FILE__) . '/_index/_files');
+    }
+
     public function testGetFieldNames()
     {
         $index = Zend_Search_Lucene::open(dirname(__FILE__) . '/_indexSample/_files');
