--- Application.php.orig	Tue May 12 10:45:23 2009
+++ Application.php	Sun May 24 21:27:35 2009
@@ -204,16 +204,29 @@
      * @param  mixed $array2 
      * @return array
      */
-    public function mergeOptions(array $array1, $array2 = null)
+    public static function mergeOptions(array $array1, $array2 = null)
     {
         if (is_array($array2)) {
-            foreach ($array2 as $key => $val) {
-                if (is_array($array2[$key])) {
-                    $array1[$key] = (array_key_exists($key, $array1) && is_array($array1[$key]))
-                                  ? $this->mergeOptions($array1[$key], $array2[$key]) 
-                                  : $array2[$key];
-                } else {
-                    $array1[$key] = $val;
+            $lcArray2 = array_change_key_case($array2, CASE_LOWER);
+            foreach ($array1 as $key => $value) {
+                $lcKey = strtolower($key);
+                if (array_key_exists($lcKey, $lcArray2)) {
+                    if (is_array($lcArray2[$lcKey])) {
+                        $array1[$key] = is_array($value) ? Zend_Application::mergeOptions($value, $lcArray2[$lcKey])
+                                                         : $lcArray2[$lcKey];
+                    }
+                    else {
+                        $array1[$key] = $lcArray2[$lcKey];
+                    }
+                    
+                    unset($lcArray2[$lcKey]);
+                }
+            }
+            foreach ($array2 as $key => $value) {
+                $lcKey = strtolower($key);
+                if (array_key_exists($lcKey, $lcArray2)) {
+                    assert(!array_key_exists($key, $array1));
+                    $array1[$key] = $value;
                 }
             }
         }
