Index: Zend/Http/Cookie.php
===================================================================
--- Zend/Http/Cookie.php	(revision 4516)
+++ Zend/Http/Cookie.php	(revision 4517)
@@ -295,6 +295,11 @@
                 switch (strtolower($k))    {
                     case 'expires':
                         $expires = strtotime($v);
+                        if ($expires === false) {
+                            // Unable to parse expiration timestamp, so don't
+                            // let it expire
+                            $expires = null;
+                        }
                         break;
                     case 'path':
                         $path = $v;
