Index: library/Zend/Controller/Action/Helper/ActionStack.php
===================================================================
--- library/Zend/Controller/Action/Helper/ActionStack.php	(revision 10552)
+++ library/Zend/Controller/Action/Helper/ActionStack.php	(working copy)
@@ -91,8 +91,8 @@
      * @throws Zend_Controller_Action_Exception 
      * @return Zend_Controller_Action_Helper_ActionStack
      */
-    public function actionToStack($action, $controller = null, $module = null, array $params = array())
-    {
+    public function actionToStack($action, $controller = null, $module = null, array $params = null )
+    {   
         if ($action instanceof Zend_Controller_Request_Abstract) {
             return $this->pushStack($action);
         } elseif (!is_string($action)) {
@@ -115,6 +115,7 @@
         
         $controller = (null === $controller) ? $request->getControllerName() : $controller;
         $module = (null === $module) ? $request->getModuleName() : $module;
+        $params = ( is_null($params) ? $request->getParams() : $params );
 
         /**
          * @see Zend_Controller_Request_Simple
