Index: Zend/Wildfire/Plugin/FirePhp/Message.php
===================================================================
--- Zend/Wildfire/Plugin/FirePhp/Message.php	(revision 10909)
+++ Zend/Wildfire/Plugin/FirePhp/Message.php	(working copy)
@@ -64,10 +64,10 @@
 
     /**
      * Creates a new message with the given style and message
-     * 
-     * @return 
+     *
      * @param string $style Style of the message.
      * @param mixed $message The message
+     * @return void
      */
     function __construct($style, $message)
     {
