Index: library/Zend/Db/Adapter/Mysqli.php
===================================================================
--- library/Zend/Db/Adapter/Mysqli.php	(revision 12711)
+++ library/Zend/Db/Adapter/Mysqli.php	(working copy)
@@ -298,9 +298,21 @@
             $port = null;
         }
 
+        $this->_connection = mysqli_init();
+        
+        if(!empty($this->_config['driver_options'])) {
+            foreach($this->_config['driver_options'] as $option=>$value) {
+                if(is_string($option)) {
+                    $option = constant($option);
+                }
+                mysqli_options($this->_connection, $option, $value);
+            }
+        }
+        
         // Suppress connection warnings here.
         // Throw an exception instead.
-        @$this->_connection = new mysqli(
+        $_isConnected = @mysqli_real_connect(
+            $this->_connection,
             $this->_config['host'],
             $this->_config['username'],
             $this->_config['password'],
@@ -307,7 +319,8 @@
             $this->_config['dbname'],
             $port
         );
-        if ($this->_connection === false || mysqli_connect_errno()) {
+        
+        if ($_isConnected === false || mysqli_connect_errno()) {
             /**
              * @see Zend_Db_Adapter_Mysqli_Exception
              */
