Index: tests/TestConfiguration.php.dist
===================================================================
--- tests/TestConfiguration.php.dist	(revision 6234)
+++ tests/TestConfiguration.php.dist	(working copy)
@@ -206,6 +206,12 @@
 define('TESTS_ZEND_GDATA_SPREADSHEETS_WORKSHEETID', 'default');
 
 /*
+ * This indicates that online tests for the Google Calendar API should
+ * be performed. The default calendar will be used.
+ */
+define('TESTS_ZEND_GDATA_CALENDAR_ONLINE_ENABLED', false);
+
+/*
  * This is the fully-qualified domain name for a domiain hosted using 
  * Google Apps. This domain must be registered with Google Apps and 
  * have API access enabled.
Index: tests/Zend/Gdata/AllTests.php
===================================================================
--- tests/Zend/Gdata/AllTests.php	(revision 6234)
+++ tests/Zend/Gdata/AllTests.php	(working copy)
@@ -247,7 +247,10 @@
                 $suite->addTestSuite('Zend_Gdata_GdataOnlineTest');
             }
             
-            $suite->addTestSuite('Zend_Gdata_CalendarOnlineTest');
+            if (defined('TESTS_ZEND_GDATA_CALENDAR_ONLINE_ENABLED') &&
+            constant('TESTS_ZEND_GDATA_CALENDAR_ONLINE_ENABLED') == true) {
+                $suite->addTestSuite('Zend_Gdata_CalendarOnlineTest');
+            }
             
             if (defined('TESTS_ZEND_GDATA_SPREADSHEETS_ONLINE_ENABLED') &&
             constant('TESTS_ZEND_GDATA_SPREADSHEETS_ONLINE_ENABLED') == true) {
