Index: library/Zend/Dojo.php
===================================================================
--- library/Zend/Dojo.php	(revision 24426)
+++ library/Zend/Dojo.php	(working copy)
@@ -49,6 +49,16 @@
     const CDN_DOJO_PATH_GOOGLE = '/dojo/dojo.xd.js';
 
     /**
+     * Base path to Yandex CDN
+     */
+    const CDN_BASE_YANDEX = 'http://yandex.st/dojo/';
+
+    /**
+     * Path to dojo on Yandex CDN (following version string)
+     */
+    const CDN_DOJO_PATH_YANDEX = '/dojo/dojo.xd.js';
+
+    /**
      * Dojo-enable a form instance
      *
      * @param  Zend_Form $form
