Index: Zend/Soap/Wsdl.php
===================================================================
--- Zend/Soap/Wsdl.php	(revision 5286)
+++ Zend/Soap/Wsdl.php	(working copy)
@@ -325,11 +325,11 @@
 
     public function addTypes($types)
     {
-        if ($types instanceof DomDocument) {
-            $dom = $this->wsdl->importNode($types->documentElement);
-            $this->wsdl->appendChild($types->documentElement);
+    	if ($types instanceof DomDocument) {
+            $dom = $this->dom->importNode($types->documentElement, true);
+            $this->wsdl->appendChild($dom);
         } elseif ($types instanceof DomNode || $types instanceof DomElement || $types instanceof DomDocumentFragment ) {
-            $dom = $this->wsdl->importNode($types);
+            $dom = $this->dom->importNode($types, true);
             $this->wsdl->appendChild($dom);
         }
     }
