Index: HeadTitle.php
===================================================================
--- HeadTitle.php	(revision 23703)
+++ HeadTitle.php	(working copy)
@@ -69,10 +69,10 @@
      */
     public function headTitle($title = null, $setType = null)
     {
-        if ($setType === null && $this->getDefaultAttachOrder() === null) {
-            $setType = Zend_View_Helper_Placeholder_Container_Abstract::APPEND;
-        } elseif ($setType === null && $this->getDefaultAttachOrder() !== null) {
-            $setType = $this->getDefaultAttachOrder();
+        if (null === $setType) {
+            $setType = (null === $this->getDefaultAttachOrder()) 
+                     ? Zend_View_Helper_Placeholder_Container_Abstract::APPEND 
+                     : $this->getDefaultAttachOrder();
         }
         $title = (string) $title;
         if ($title !== '') {
