Index: library/Zend/EventManager/Exception.php
===================================================================
--- library/Zend/EventManager/Exception.php	(revision 24746)
+++ library/Zend/EventManager/Exception.php	(working copy)
@@ -19,11 +19,6 @@
  */
 
 /**
- * @namespace
- */
-namespace Zend\EventManager;
-
-/**
  * Base exception interface
  *
  * @category   Zend
@@ -31,6 +26,6 @@
  * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-interface Exception
+interface Zend_EventManager_Exception
 {
 }
Index: library/Zend/EventManager/Exception/InvalidArgumentException.php
===================================================================
--- library/Zend/EventManager/Exception/InvalidArgumentException.php	(revision 24746)
+++ library/Zend/EventManager/Exception/InvalidArgumentException.php	(working copy)
@@ -19,11 +19,14 @@
  */
 
 /**
- * @namespace
+ * @see Zend_Exception
  */
-namespace Zend\EventManager\Exception;
+require_once 'Zend/Exception.php';
 
-use Zend\EventManager\Exception;
+/**
+ * @see Zend_EventManager_Exception
+ */
+require_once 'Zend/EventManager/Exception.php';
 
 /**
  * Invalid argument exception
@@ -33,7 +36,7 @@
  * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class InvalidArgumentException 
-    extends \InvalidArgumentException implements Exception
+class Zend_EventManager_Exception_InvalidArgumentException
+    extends Zend_Exception implements Zend_EventManager_Exception
 {
 }
Index: tests/Zend/EventManager/EventManagerTest.php
===================================================================
--- tests/Zend/EventManager/EventManagerTest.php	(revision 24746)
+++ tests/Zend/EventManager/EventManagerTest.php	(working copy)
@@ -551,6 +551,16 @@
         }
     }
 
+    /**
+     * @group ZF-12185
+     * @expectedException Zend_EventManager_Exception_InvalidArgumentException
+     */
+    public function testInvalidArgumentExceptionCanBeThrown()
+    {
+        require_once "Zend/EventManager/Exception/InvalidArgumentException.php";
+        throw new Zend_EventManager_Exception_InvalidArgumentException();
+    }
+
     /*
      * Listeners used in tests
      */
