Index: Abstract.php
===================================================================
--- Abstract.php	(revision 12073)
+++ Abstract.php	(working copy)
@@ -57,6 +57,21 @@
     protected $_config = array();
 
     /**
+     * Weither or not that object can get serialized
+     *
+     * @var bool
+     */
+    protected $_allowSerialization = true;
+
+    /**
+     * Weither or not the database should be reconnected
+     * to that adapter when waking up
+     *
+     * @var bool
+     */
+    protected $_autoReconnectOnUnserialize = false;
+
+    /**
      * Fetch mode
      *
      * @var integer
@@ -1022,6 +1037,33 @@
     }
 
     /**
+     * called when object is getting serialized
+     * 
+     * @return array
+     *
+     */
+    public function __sleep()
+    {
+        if ($this->_allowSerialization == false) {
+            require_once 'Zend/Db/Adapter/Exception.php';
+            throw new Zend_Db_Adapter_Exception(get_class($this) ." is not allowed to be serialized");
+        }
+        return array_keys(array_diff_key(get_object_vars($this), array('_connection'=>false)));
+    }
+
+    /**
+     * called when object is getting unserialized
+     * 
+     * @return void
+     */
+    public function __wakeup()
+    {
+        if ($this->_autoReconnectOnUnserialize == true) {
+            $this->getConnection();
+        }
+    }
+
+    /**
      * Abstract Methods
      */
 
