--- tests/Zend/Dojo/Form/Element/EditorTest.php	(revision 23864)
+++ tests/Zend/Dojo/Form/Element/EditorTest.php	(working copy)
@@ -262,6 +262,15 @@
         $this->assertEquals($this->element->getDijitParam('updateInterval'), $this->element->getUpdateInterval());
         $this->assertEquals(300, $this->element->getUpdateInterval());
     }
+    
+    public function testCanAddMultipleSeparatorsToEditor()
+    {
+        $this->element->setPlugins(array('undo', '|', 'bold', '|', 'italic'));
+        
+        $plugins = $this->element->getPlugins();
+        $this->assertEquals(5, count($plugins));
+    }
+    
 }
 
 // Call Zend_Dojo_Form_Element_EditorTest::main() if this source file is executed directly.
