Index: Abstract.php
===================================================================
--- Abstract.php        (revision 501)
+++ Abstract.php        (working copy)
@@ -476,7 +476,7 @@
         // only look for "$Name.php"
         $file = ucfirst($name) . '.php';
         foreach ($this->_path[$type] as $dir) {
-            if (is_readable($dir . $file)) {
+            if (Zend::isReadable($dir . $file)) {
                 include $dir . $file;
                 
                 if (! class_exists($class, false)) {
