Index: library/Zend/Feed/Reader.php
===================================================================
--- library/Zend/Feed/Reader.php	(revision 25022)
+++ library/Zend/Feed/Reader.php	(working copy)
@@ -240,7 +240,7 @@
                     $etag = $cache->load($cacheId.'_etag');
                 }
                 if ($lastModified === null) {
-                    $lastModified = $cache->load($cacheId.'_lastmodified');;
+                    $lastModified = $cache->load($cacheId.'_lastmodified');
                 }
                 if ($etag) {
                     $client->setHeaders('If-None-Match', $etag);
