Index: Session.php
===================================================================
--- Session.php	(revision 18355)
+++ Session.php	(working copy)
@@ -263,20 +263,30 @@
      */
     public static function setSaveHandler(Zend_Session_SaveHandler_Interface $saveHandler)
     {
+        if (self::isStarted()) {
+            require_once 'Zend/Session/Exception.php';
+            throw new Zend_Session_Exception('Can not change session.save_handler, because session already started');
+        }
+
         self::$_saveHandler = $saveHandler;
 
         if (self::$_unitTestEnabled) {
             return;
         }
 
-        session_set_save_handler(
-            array(&$saveHandler, 'open'),
-            array(&$saveHandler, 'close'),
-            array(&$saveHandler, 'read'),
-            array(&$saveHandler, 'write'),
-            array(&$saveHandler, 'destroy'),
-            array(&$saveHandler, 'gc')
+        $flag = session_set_save_handler(
+            array($saveHandler, 'open'),
+            array($saveHandler, 'close'),
+            array($saveHandler, 'read'),
+            array($saveHandler, 'write'),
+            array($saveHandler, 'destroy'),
+            array($saveHandler, 'gc')
             );
+
+        if (!$flag) {
+            require_once 'Zend/Session/Exception.php';
+            throw new Zend_Session_Exception('session_set_save_handler() failed');
+        }
     }
 
 
