Index: Translate.php
===================================================================
--- Translate.php	(revision 9303)
+++ Translate.php	(working copy)
@@ -38,6 +38,17 @@
      * @var Zend_Translate_Adapter
      */
     protected $_translator;
+    
+    /**
+     * Whether or not to escape output according to the view object
+     * @var boolean
+     */
+    protected $_escape = false;
+    
+    /**
+     * @var Zend_View_Interface
+     */
+    public $view;
 
     /**
      * Constructor for manually handling
@@ -85,7 +96,7 @@
             $options = $options[0];
         }
         $message = $translate->translate($messageid, $locale);
-        return vsprintf($message, $options);
+        return $this->_escape ? $this->view->escape(vsprintf($message, $options)) : vsprintf($message, $options);
     }
 
     /**
@@ -155,4 +166,37 @@
         }
         return $translate->getLocale();
     }
+    
+    /**
+     * Set escaping
+     *
+     * @param boolean $enable
+     * @return Zend_View_Helper_Translate
+     */
+    public function setEscape($enable = true)
+    {
+        $this->_escape = !!$enable;
+        return $this;
+    }
+    
+    /**
+     * Get escaping
+     *
+     * @return boolean
+     */
+    public function getEscape()
+    {
+        return $this->_escape;
+    }
+
+    /**
+     * Set view
+     *
+     * @param  Zend_View_Interface $view
+     * @return void
+     */
+    public function setView(Zend_View_Interface $view)
+    {
+        $this->view = $view;
+    }
 }
