Index: library/Zend/Service/Amazon/Ec2/Abstract.php
===================================================================
--- library/Zend/Service/Amazon/Ec2/Abstract.php	(revision 19975)
+++ library/Zend/Service/Amazon/Ec2/Abstract.php	(working copy)
@@ -240,7 +240,7 @@
 
         $arrData = array();
         foreach($paramaters as $key => $value) {
-            $arrData[] = $key . '=' . str_replace("%7E", "~", urlencode($value));
+            $arrData[] = $key . '=' . str_replace("%7E", "~", rawurlencode($value));
         }
 
         $data .= implode('&', $arrData);
