diff --git a/library/Zend/Cache/Core.php b/library/Zend/Cache/Core.php
index 5e8aa41..f8d0d79 100644
--- library/Zend/Cache/Core.php
+++ library/Zend/Cache/Core.php
@@ -376,7 +376,7 @@ class Zend_Cache_Core
         if (substr($string, 0, 9) == 'internal-') {
             Zend_Cache::throwException('"internal-*" ids or tags are reserved');
         }
-        if (!preg_match('~^[\w]+$~', $string)) {
+        if (!preg_match('~^[\w]+$~D', $string)) {
             Zend_Cache::throwException("Invalid id or tag '$string' : must use only [a-zA-Z0-9_]");
         }
     }
