--- /home/trashofmasters/Downloads/ZendFramework-1.10.2/tests/Zend/Loader/Autoloader/ResourceTest.php	2010-01-06 03:05:09.000000000 +0100
+++ Zend/Loader/Autoloader/ResourceTest.php	2010-03-07 07:02:19.000000000 +0100
@@ -424,6 +424,18 @@
         $path = $this->loader->autoload('Something_Totally_Wrong');
         $this->assertFalse($path);
     }
+
+    /**
+     * @group ZF-8205
+     */
+    public function testAutoloaderShouldLoadNamespacedClassesWhenHeadingNamespaceSeparatorIsPresentInNamespaceField()
+    {
+        $this->loader->addResourceTypes(array(
+            'model' => array('path' => 'models', 'namespace' => '\Model'),
+        ));
+        $test = $this->loader->getModel('ZendLoaderAutoloaderResourceNamespaceModelTest');
+        $this->assertTrue($test instanceof \FooBar\Model\ZendLoaderAutoloaderResourceNamespaceModelTest);
+    }
 }
 
 if (PHPUnit_MAIN_METHOD == 'Zend_Loader_Autoloader_ResourceTest::main') {
