Index: library/Zend/Application/Resource/Mail.php
===================================================================
--- library/Zend/Application/Resource/Mail.php	(revision 21012)
+++ library/Zend/Application/Resource/Mail.php	(working copy)
@@ -106,11 +106,12 @@
     		$options['type'] = 'sendmail';
     	}
     	
-        $transportName = ucfirst(strtolower($options['type']));
+        $transportName = $options['type'];
         unset($options['type']);
 
         if(!Zend_Loader_Autoloader::autoload($transportName)) {
-            $transportName = 'Zend_Mail_Transport_' . $transportName;
+            $transportName = 'Zend_Mail_Transport_' . ucfirst(strtolower($transportName));
+    	
 
             if(!Zend_Loader_Autoloader::autoload($transportName)) {
                 throw new Zend_Application_Resource_Exception(
