Index: library/Zend/View/Helper/FormLabel.php
===================================================================
--- library/Zend/View/Helper/FormLabel.php	(revision 19078)
+++ library/Zend/View/Helper/FormLabel.php	(working copy)
@@ -57,6 +57,9 @@
         $for   = (empty($attribs['disableFor']) || !$attribs['disableFor'])
                ? ' for="' . $this->view->escape($id) . '"'
                : '';
+        if (array_key_exists('disableFor', $attribs)) {
+            unset($attribs['disableFor']);
+        }
 
         // enabled; display label
         $xhtml = '<label'
