Index: Zend/Gdata/Calendar/EventQueryExceptionTest.php
===================================================================
--- Zend/Gdata/Calendar/EventQueryExceptionTest.php	(Revision 6880)
+++ Zend/Gdata/Calendar/EventQueryExceptionTest.php	(Arbeitskopie)
@@ -22,13 +22,12 @@
 require_once 'Zend/Gdata/Calendar.php';
 require_once 'Zend/Gdata/Calendar/EventQuery.php';
 require_once 'Zend/Http/Client.php';
-require_once 'PHPUnit/Extensions/ExceptionTestCase.php';
 
 /**
  * @package Zend_Gdata
  * @subpackage UnitTests
  */
-class Zend_Gdata_Calendar_EventQueryExceptionTest extends PHPUnit_Extensions_ExceptionTestCase
+class Zend_Gdata_Calendar_EventQueryExceptionTest extends PHPUnit_Framework_TestCase
 {
     
     const GOOGLE_DEVELOPER_CALENDAR = 'developer-calendar@google.com';
Index: Zend/Gdata/Gapps/ServiceExceptionTest.php
===================================================================
--- Zend/Gdata/Gapps/ServiceExceptionTest.php	(Revision 6880)
+++ Zend/Gdata/Gapps/ServiceExceptionTest.php	(Arbeitskopie)
@@ -20,7 +20,6 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 
-require_once 'PHPUnit/Extensions/ExceptionTestCase.php';
 require_once 'Zend/Gdata/Gapps/ServiceException.php';
 require_once 'Zend/Gdata/Gapps/Error.php';
 require_once 'Zend/Gdata/Gapps.php';
@@ -29,7 +28,7 @@
  * @package Zend_Gdata
  * @subpackage UnitTests
  */
-class Zend_Gdata_Gapps_ServiceExceptionTest extends PHPUnit_Extensions_ExceptionTestCase
+class Zend_Gdata_Gapps_ServiceExceptionTest extends PHPUnit_Framework_TestCase
 {
     protected $fixture;
     protected $data;
