--- orig-Yahoo.php      Wed Jul 12 16:59:53 2006
+++ Yahoo.php   Wed Jul 12 16:59:03 2006
@@ -152,7 +152,7 @@

         $this->_rest = new Zend_Service_Rest;

-        $this->_rest->setUri('api.search.yahoo.com');
+        $this->_rest->setUri('http://api.search.yahoo.com');

         $response = $this->_rest->restGet('/ImageSearchService/V1/imageSearch', $options);

@@ -207,7 +207,9 @@

         $this->_validateLocalSearch($options);

-        $this->_uri->setHost('api.local.yahoo.com');
+        $this->_rest = new Zend_Service_Rest;
+
+        $this->_rest->setUri('http://api.local.yahoo.com');

         $response = $this->_rest->restGet('/LocalSearchService/V1/localSearch', $options);

@@ -250,7 +252,9 @@

         $this->_validateNewsSearch($options);

-        $this->_uri->setHost('api.search.yahoo.com');
+        $this->_rest = new Zend_Service_Rest;
+
+        $this->_rest->setUri('http://api.search.yahoo.com');

         $response = $this->_rest->restGet('/NewsSearchService/V1/newsSearch', $options);

@@ -299,7 +303,9 @@
         $options = $this->_prepareOptions($query, $options, $default_options);
         $this->_validateWebSearch($options);

-        $this->_uri->setHost('api.search.yahoo.com');
+        $this->_rest = new Zend_Service_Rest;
+
+        $this->_rest->setUri('http://api.search.yahoo.com');

         $response = $this->_rest->restGet('/WebSearchService/V1/webSearch', $options);

