--- zend-framework.svn/library/Zend/Crypt/Math/BigInteger/Exception.php~	2009-05-01 23:34:03.303452679 +0200
+++ zend-framework.svn/library/Zend/Crypt/Math/BigInteger/Exception.php	2009-05-02 11:01:57.255575067 +0200
@@ -14,7 +14,7 @@
  * to license@zend.com so we can send you a copy immediately.
  *
  * @category   Zend
- * @package    Zend_Math
+ * @package    Zend_Crypt
  * @copyright  Copyright (c) 2007 Pádraic Brady
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  * @version    $Id: Exception.php 102 2007-07-13 15:16:01Z padraic $
@@ -22,16 +22,16 @@
 
 
 /**
- * Zend_Math_Exception
+ * Zend_Crypt_Math_Exception
  */
-require_once 'Zend/Math/Exception.php';
+require_once 'Zend/Crypt/Math/Exception.php';
 
 
 /**
  * @category   Zend
- * @package    Zend_Math
+ * @package    Zend_Crypt
  * @copyright  Copyright (c) 2007 Pádraic Brady
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Math_BigInteger_Exception extends Zend_Math_Exception
+class Zend_Crypt_Math_BigInteger_Exception extends Zend_Crypt_Math_Exception
 {}
