Index: library/Zend/Db/Statement/Oracle.php
===================================================================
--- library/Zend/Db/Statement/Oracle.php	(revision 24746)
+++ library/Zend/Db/Statement/Oracle.php	(working copy)
@@ -240,7 +240,7 @@
             }
             $error = false;
             foreach (array_keys($params) as $name) {
-                if (!@oci_bind_by_name($this->_stmt, $name, $params[$name], -1)) {
+                if (!$this->bindParam($name, $params[$name], null, -1)) {
                     $error = true;
                     break;
                 }
