Index: Twitter.php
===================================================================
--- Twitter.php	(revision 23292)
+++ Twitter.php	(working copy)
@@ -416,6 +416,11 @@
                 case 'max_id':
                     $_params['max_id'] = $this->_validInteger($value);
                     break;
+                case 'trim_user':
+                case 'include_entities':
+                case 'include_rts':
+                    $_params[strtolower($key)] = $value ? '1':'0';
+                    break;
                 default:
                     break;
             }
