Index: Form.php
===================================================================
--- Form.php	
+++ Form.php	
@@ -1013,6 +1013,7 @@
      * @param  string|Zend_Form_Element $element
      * @param  string $name
      * @param  array|Zend_Config $options
+     * @throws Zend_Form_Exception on invalid element
      * @return Zend_Form
      */
     public function addElement($element, $name = null, $options = null)
@@ -1050,6 +1051,9 @@
 
             $this->_elements[$name] = $element;
             $this->_elements[$name]->addPrefixPaths($prefixPaths);
+        } else {
+            require_once 'Zend/Form/Exception.php';
+            throw new Zend_Form_Exception('Invalid element provided');
         }
 
         $this->_order[$name] = $this->_elements[$name]->getOrder();
