--- tests/Zend/Dojo/Form/Element/DateTextBoxTest.php	(revision 24415)
+++ tests/Zend/Dojo/Form/Element/DateTextBoxTest.php	(working copy)
@@ -196,6 +196,18 @@
         $html = $this->element->render();
         $this->assertContains('dojoType="dijit.form.DateTextBox"', $html);
     }
+    
+    /**
+     * @group ZF-7813
+     */
+    public function testCanSetValue()
+    {
+        $this->element->setValue('2011-05-10');
+        $html = $this->element->render();
+        
+        $this->assertSame('2011-05-10', $this->element->getValue());
+        $this->assertContains('value="2011-05-10"', $html);
+    }
 }
 
 // Call Zend_Dojo_Form_Element_DateTextBoxTest::main() if this source file is executed directly.
Index: tests/Zend/Dojo/Form/Element/TimeTextBoxTest.php
===================================================================
--- tests/Zend/Dojo/Form/Element/TimeTextBoxTest.php	(revision 24415)
+++ tests/Zend/Dojo/Form/Element/TimeTextBoxTest.php	(working copy)
@@ -183,6 +183,18 @@
         $html = $this->element->render();
         $this->assertContains('dojoType="dijit.form.TimeTextBox"', $html);
     }
+    
+    /**
+     * @group ZF-7813
+     */
+    public function testCanSetValue()
+    {
+        $this->element->setValue('T08:00');
+        $html = $this->element->render();
+        
+        $this->assertSame('T08:00', $this->element->getValue());
+        $this->assertContains('value="T08:00"', $html);
+    }
 }
 
 // Call Zend_Dojo_Form_Element_TimeTextBoxTest::main() if this source file is executed directly.
