Index: tests/Zend/Controller/ActionTest.php
===================================================================
--- tests/Zend/Controller/ActionTest.php	(revision 23898)
+++ tests/Zend/Controller/ActionTest.php	(working copy)
@@ -478,6 +478,24 @@
         $controller->scriptAction();
         $this->assertContains('Inside custom/renderScript.php', $response->getBody());
     }
+    
+    /**
+     * @group ZF-11204
+     */
+    public function testThatCharactersStrippedFromControllerNameAlsoGetStrippedFromViewNameWhenUsingViewRenderer()
+    {
+        $request = new Zend_Controller_Request_Http();
+        $request->setControllerName('-view')
+                ->setActionName('index');
+        $response = new Zend_Controller_Response_Cli();
+        Zend_Controller_Front::getInstance()->setControllerDirectory(dirname(__FILE__) . DIRECTORY_SEPARATOR . '_files');
+        require_once dirname(__FILE__) . DIRECTORY_SEPARATOR . '_files' . DIRECTORY_SEPARATOR . 'ViewController.php';
+        $controller = new ViewController($request, $response);
+        
+        $controller->indexAction();
+        $this->assertSame('view', $request->getControllerName());
+        $this->assertSame('view/index.phtml', $controller->getViewScript());
+    }
 
     public function testMissingActionExceptionsDifferFromMissingMethods()
     {
