Index: library/Zend/View/Helper/FormLabel.php
===================================================================
--- library/Zend/View/Helper/FormLabel.php	(revision 7833)
+++ library/Zend/View/Helper/FormLabel.php	(working copy)
@@ -51,10 +51,18 @@
             // disabled; do nothing
         } else {
             // enabled; display label
+
+            if (isset($attribs['escape'])) {
+                if ($attribs['escape'] !== false) {
+                    $value = $this->view->escape($value);
+                }
+                unset($attribs['escape']);
+            }
+            
             $xhtml = '<label'
                    . ' for="' . $this->view->escape($id) . '"'
                    . $this->_htmlAttribs($attribs)
-                   . '>' . $this->view->escape($value) . '</label>';
+                   . '>' . $value . '</label>';
         }
 
         return $xhtml;
