Index: Filter/Input.php
===================================================================
--- Filter/Input.php	(revision 7216)
+++ Filter/Input.php	(working copy)
@@ -718,6 +718,17 @@
          * Evaluate the inputs against the validator chain.
          */
         if (count((array) $validatorRule[self::FIELDS]) > 1) {
+            if ($validatorRule[self::ALLOW_EMPTY] == true) {
+                $empty = true;
+                foreach ($data as $field) {
+                    if (!empty($field)) {
+                        $empty = false;
+                    }
+                }
+                if ($empty) {
+                    return;
+                }
+            }
             if (!$validatorRule[self::VALIDATOR_CHAIN]->isValid($data)) {
                 $this->_invalidMessages[$validatorRule[self::RULE]] = $validatorRule[self::VALIDATOR_CHAIN]->getMessages();
                 $this->_invalidErrors[$validatorRule[self::RULE]] = $validatorRule[self::VALIDATOR_CHAIN]->getErrors();
