Index: library/Zend/View/Helper/Navigation.php
===================================================================
--- library/Zend/View/Helper/Navigation.php	(revision 24555)
+++ library/Zend/View/Helper/Navigation.php	(working copy)
@@ -157,9 +157,17 @@
         }
 
         if (!$this->view->getPluginLoader('helper')->getPaths(self::NS)) {
+            // Add navigation helper path at the beginning
+            $paths = $this->view->getHelperPaths();
+            $this->view->setHelperPath(null);
+            
             $this->view->addHelperPath(
                     str_replace('_', '/', self::NS),
                     self::NS);
+            
+            foreach ($paths as $ns => $path) {
+                $this->view->addHelperPath($path, $ns);
+            }
         }
 
         if ($strict) {
