Index: FormErrorsTest.php
===================================================================
--- FormErrorsTest.php	(revision 17645)
+++ FormErrorsTest.php	(working copy)
@@ -81,8 +81,11 @@
         $view = new Zend_View();
         return $view;
     }
-
-    public function setupForm()
+	
+	/**
+	 * Sets up form with invalid data by default.
+	 */
+    public function setupForm($valid=false)
     {
         $form1 = new Zend_Form_SubForm;
         $form1->addElement('text', 'foo', array(
@@ -118,14 +121,32 @@
                     ),
                 ))
               ->addSubForm($form1, 'sub');
-        $form2->isValid(array(
-            'foo' => '',
-            'bar' => 'foo 2 u 2',
-            'sub' => array(
-                'foo' => '',
-                'bar' => 'foo 2 u 2',
-            ),
-        ));
+
+        switch ($valid)
+        {
+            case true:
+                $data = array(
+                	'foo' => 'value',
+					'bar' => 'foo',
+					'sub' => array(
+						'foo' => 'value',
+						'bar' => 'foo',
+                	),
+                );
+                break;
+            default:
+                $data = array(
+					'foo' => '',
+					'bar' => 'foo 2 u 2',
+					'sub' => array(
+						'foo' => '',
+						'bar' => 'foo 2 u 2',
+					),
+				);
+                break;
+        }
+        
+        $form2->isValid($data);
         $form2->setView($this->getView());
         $this->decorator->setElement($form2);
         $this->form = $form2;
@@ -158,6 +179,13 @@
             }
         }
     }
+    
+    public function testRenderReturnsInitialContentIfNoErrors()
+    {
+    	$this->setupForm(true);
+		$content = 'test content';
+        $this->assertSame($content, $this->decorator->render($content));
+    }
 
     public function testRenderAppendsMessagesToContentByDefault()
     {
