--- /C/wamp/apache/htdocs/zendframework/library/Zend/Application/Bootstrap/BootstrapAbstract.php.orig	Tue Jul  7 08:02:08 2009
+++ /C/wamp/apache/htdocs/zendframework/library/Zend/Application/Bootstrap/BootstrapAbstract.php	Thu Aug  6 16:40:32 2009
@@ -349,7 +349,7 @@
                 }
             }
 
-            if (class_exists($plugin)) {
+            if (class_exists($plugin, false)) {
                 $spec = (array) $spec;
                 $spec['bootstrap'] = $this;
                 $instance = new $plugin($spec);
@@ -363,7 +363,7 @@
             }
         }
 
-        return null;            
+        return null;
     }
 
     /**
