Index: library/Zend/ProgressBar/Adapter/Console.php
===================================================================
--- library/Zend/ProgressBar/Adapter/Console.php	(Revision 14763)
+++ library/Zend/ProgressBar/Adapter/Console.php	(Arbeitskopie)
@@ -45,6 +45,11 @@
     const ELEMENT_PERCENT = 'ELEMENT_PERCENT';
 
     /**
+     * absolute values of the progress
+     */
+    const ELEMENT_ABSOLUTE = 'ELEMENT_ABSOLUTE';
+
+    /**
      * Visual value of the progress
      */
     const ELEMENT_BAR = 'ELEMENT_BAR';
@@ -264,6 +269,7 @@
     public function setElements(array $elements)
     {
         $allowedElements = array(self::ELEMENT_PERCENT,
+                                 self::ELEMENT_ABSOLUTE,
                                  self::ELEMENT_BAR,
                                  self::ELEMENT_ETA,
                                  self::ELEMENT_TEXT);
@@ -432,6 +438,10 @@
                     $renderedElements[] = str_pad(round($percent * 100), 3, ' ', STR_PAD_LEFT) . '%';
                     break;
 
+                case self::ELEMENT_ABSOLUTE:
+                    $renderedElements[] = str_pad($current, strlen($max), ' ', STR_PAD_LEFT) . ' / ' . $max;
+                    break;
+
                 case self::ELEMENT_ETA:
                     // In the first 5 seconds we don't get accurate results,
                     // this skipping technique is found in many progressbar
@@ -507,6 +517,10 @@
                 $barWidth -= 4;
             }
 
+            if (in_array(self::ELEMENT_ABSOLUTE, $this->_elements)) {
+                $barWidth -= 11;
+            }
+
             if (in_array(self::ELEMENT_ETA, $this->_elements)) {
                 $barWidth -= 12;
             }
@@ -531,4 +545,4 @@
     {
         fwrite($this->getOutputStream(), $data);
     }
-}
\ No newline at end of file
+}
