Index: tests/Zend/ProgressBar/Adapter/ConsoleTest.php
===================================================================
--- tests/Zend/ProgressBar/Adapter/ConsoleTest.php	(Revision 14768)
+++ tests/Zend/ProgressBar/Adapter/ConsoleTest.php	(Arbeitskopie)
@@ -162,6 +162,14 @@
         $this->assertEquals('  0%', $adapter->getLastOutput());
     }
 
+    public function testAbsoluteOnly()
+    {
+        $adapter = new Zend_ProgressBar_Adapter_Console_Stub(array('width' => 20, 'elements' => array(Zend_ProgressBar_Adapter_Console::ELEMENT_ABSOLUTE)));
+        $adapter->notify(0, 100, 0, 0, null, null);
+
+        $this->assertEquals('  0 / 100', $adapter->getLastOutput());
+    }
+
     public function testEtaOnly()
     {
         $adapter = new Zend_ProgressBar_Adapter_Console_Stub(array('width' => 20, 'elements' => array(Zend_ProgressBar_Adapter_Console::ELEMENT_ETA)));
