Index: Interface.php
===================================================================
--- Interface.php	(revision 2914)
+++ Interface.php	(working copy)
@@ -31,7 +31,7 @@
  * @license    http://www.zend.com/license/framework/1_0.txt Zend Framework License version 1.0
  */
 interface Zend_Controller_Router_Route_Interface {
-    public function match($path);
+    public function match(Zend_Controller_Request_Http $request);
     public function assemble($data = array());
 }
 
