Index: Zend/Http/Client.php
===================================================================
--- Zend/Http/Client.php	(revision 4954)
+++ Zend/Http/Client.php	(working copy)
@@ -264,7 +264,10 @@
      */
     public function setUri($uri)
     {
-        if (is_string($uri)) {
+        if ($uri instanceof Zend_Uri_Http) {
+            // clone the URI in order to keep the passed parameter constant
+            $uri = clone $uri;
+        } elseif (is_string($uri)) {
             $uri = Zend_Uri::factory($uri);
         }
 
