Index: tests/Zend/View/Helper/Navigation/_files/helpers/Menu.php
===================================================================
--- tests/Zend/View/Helper/Navigation/_files/helpers/Menu.php	(revision 0)
+++ tests/Zend/View/Helper/Navigation/_files/helpers/Menu.php	(revision 0)
@@ -0,0 +1,39 @@
+<?php
+
+class My_View_Helper_Navigation_Menu
+    extends Zend_View_Helper_Navigation_HelperAbstract
+{
+    /**
+     * View helper entry point:
+     * Retrieves helper and optionally sets container to operate on
+     *
+     * @param  Zend_Navigation_Container $container  [optional] container to
+     *                                               operate on
+     * @return My_View_Helper_Navigation_Menu        fluent interface,
+     *                                               returns self
+     */
+    public function menu(Zend_Navigation_Container $container = null)
+    {
+        if (null !== $container) {
+            $this->setContainer($container);
+        }
+
+        return $this;
+    }
+    
+    /**
+     * Renders menu
+     *
+     * Implements {@link Zend_View_Helper_Navigation_Helper::render()}.
+     *
+     * @param  Zend_Navigation_Container $container  [optional] container to
+     *                                               render. Default is to
+     *                                               render the container
+     *                                               registered in the helper.
+     * @return string                                helper output
+     */
+    public function render(Zend_Navigation_Container $container = null)
+    {
+        return '<menu/>';
+    }
+}
\ No newline at end of file
Index: tests/Zend/View/Helper/Navigation/NavigationTest.php
===================================================================
--- tests/Zend/View/Helper/Navigation/NavigationTest.php	(revision 24555)
+++ tests/Zend/View/Helper/Navigation/NavigationTest.php	(working copy)
@@ -409,4 +409,55 @@
 
         $this->assertTrue(strpos($render, 'p2') !== false);
     }
+    
+    /**
+     * @group ZF-10458
+     */
+    public function testFindCustomHelper()
+    {
+        $this->_helper->view->addHelperPath(
+            $this->_files . '/helpers',
+            'My_View_Helper_Navigation'
+        );                
+        
+        $this->assertTrue(
+            $this->_helper->findHelper('menu') instanceof
+                My_View_Helper_Navigation_Menu
+        );
+    }
+    
+    /**
+     * @group ZF-10458
+     */
+    public function testAddHelperPath()
+    {
+        $this->_helper->view->addHelperPath(
+            $this->_files . '/helpers',
+            'My_View_Helper_Navigation'
+        );
+        
+        $expected = array(
+            'Zend_View_Helper_' => array(
+                'Zend/View/Helper/',
+            ),
+            'My_View_Helper_Navigation_' => array(
+                $this->_files . '/helpers/',
+            ),
+        );
+        
+        $this->assertSame($expected, $this->_helper->view->getHelperPaths());
+    }
+    
+    /**
+     * @group ZF-10458
+     */
+    public function testRenderCustomHelper()
+    {
+        $this->_helper->view->addHelperPath(
+            $this->_files . '/helpers',
+            'My_View_Helper_Navigation'
+        );
+        
+        $this->assertSame('<menu/>', (string) $this->_helper->menu());
+    }
 }