--- ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/Abstract.php~	2009-05-01 23:51:12.793349399 +0200
+++ ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/Abstract.php	2009-05-02 10:39:48.389040625 +0200
@@ -20,7 +20,7 @@
  * @version    $Id$
  */
 
-require_once "Interface.php";
+require_once "Zend/Soap/Wsdl/Strategy/Interface.php";
 
 /**
  * Abstract class for Zend_Soap_Wsdl_Strategy.
--- ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/AnyType.php~	2009-05-01 23:51:12.793349399 +0200
+++ ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/AnyType.php	2009-05-02 10:39:38.509042144 +0200
@@ -20,7 +20,7 @@
  * @version    $Id$
  */
 
-require_once "Interface.php";
+require_once "Zend/Soap/Wsdl/Strategy/Interface.php";
 
 class Zend_Soap_Wsdl_Strategy_AnyType implements Zend_Soap_Wsdl_Strategy_Interface
 {
@@ -44,4 +44,4 @@
     {
         return 'xsd:anyType';
     }
-}
\ Brak znaku nowej linii na końcu pliku
+}
--- ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/ArrayOfTypeSequence.php~	2009-05-01 23:51:12.793349399 +0200
+++ ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/ArrayOfTypeSequence.php	2009-05-02 10:39:13.049044140 +0200
@@ -19,7 +19,7 @@
  * @version    $Id$
  */
 
-require_once "Abstract.php";
+require_once "Zend/Soap/Wsdl/Strategy/Abstract.php";
 
 class Zend_Soap_Wsdl_Strategy_ArrayOfTypeSequence extends Zend_Soap_Wsdl_Strategy_Abstract
 {
@@ -147,4 +147,4 @@
             $this->getContext()->addType($arrayType);
         }
     }
-}
\ Brak znaku nowej linii na końcu pliku
+}
--- ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/DefaultComplexType.php~	2009-05-01 23:51:12.793349399 +0200
+++ ZendFramework-1.8.0/library/Zend/Soap/Wsdl/Strategy/DefaultComplexType.php	2009-05-02 10:39:21.962376624 +0200
@@ -20,7 +20,7 @@
  * @version    $Id$
  */
 
-require_once "Abstract.php";
+require_once "Zend/Soap/Wsdl/Strategy/Abstract.php";
 
 class Zend_Soap_Wsdl_Strategy_DefaultComplexType extends Zend_Soap_Wsdl_Strategy_Abstract
 {
