Index: PluginLoader.php
===================================================================
--- PluginLoader.php	(revision 13952)
+++ PluginLoader.php	(working copy)
@@ -455,6 +455,9 @@
             $file = '<?php';
         } else {
             $file = file_get_contents(self::$_includeFileCache);
+            if (strpos($file, '<?php') === false) {
+                $file = "<?php\n" . $file;
+            }
         }
         if (!strstr($file, $incFile)) {
             $file .= "\ninclude_once '$incFile';";
