Index: Pdf.php
===================================================================
--- Pdf.php	(revision 16406)
+++ Pdf.php	(working copy)
@@ -731,7 +731,7 @@
                     case 'Producer':
                         if (extension_loaded('mbstring') === true) {
                             $detected = mb_detect_encoding($value);
-                            if ($detected !== 'ASCII') {
+                            if ($detected !== false && $detected !== 'ASCII') {
                                 $value = chr(254) . chr(255) . mb_convert_encoding($value, 'UTF-16', $detected);
                             }
                         }
