Index: library/Zend/Form/Decorator/FormErrors.php
===================================================================
--- library/Zend/Form/Decorator/FormErrors.php	(revision 16059)
+++ library/Zend/Form/Decorator/FormErrors.php	(working copy)
@@ -351,6 +351,10 @@
             $label = $element->getName();
         }
 
+        if (null !== ($translator = $element->getTranslator())) { 
+            $label = $translator->translate($label);
+        }
+
         return $this->getMarkupElementLabelStart()
              . $view->escape($label)
              . $this->getMarkupElementLabelEnd();
