Index: library/Zend/Db/Adapter/Mysqli.php
===================================================================
--- library/Zend/Db/Adapter/Mysqli.php	(revision 11722)
+++ library/Zend/Db/Adapter/Mysqli.php	(working copy)
@@ -299,7 +299,39 @@
 
         // Suppress connection warnings here.
         // Throw an exception instead.
-        @$this->_connection = new mysqli(
+        /*@$this->_connection = new mysqli(
+            $this->_config['host'],
+            $this->_config['username'],
+            $this->_config['password'],
+            $this->_config['dbname'],
+            $port
+        );*/
+        
+        $this->_connection = mysqli::init();
+        
+        if(!empty($this->_config['driver_options'])) {
+            foreach($this->_config['driver_options'] as $option=>$value) {
+                switch($option) {
+                    case MYSQLI_INIT_COMMAND:
+                        // break intentionally omitted
+                    case MYSQLI_OPT_CONNECT_TIMEOUT:
+                        // break intentionally omitted
+                    case MYSQLI_OPT_LOCAL_INFILE:
+                        // break intentionally omitted
+                    case MYSQLI_READ_DEFAULT_FILE:
+                        // break intentionally omitted
+                    case MYSQLI_READ_DEFAULT_GROUP:
+                       mysqli_options($this->_connection, $option, $value);
+                       break;
+                    default:
+                        // don't do anything.
+                        break;
+                }
+            }
+        }
+        
+        mysqli_real_connect(
+            $this->_connection,
             $this->_config['host'],
             $this->_config['username'],
             $this->_config['password'],
@@ -306,6 +338,7 @@
             $this->_config['dbname'],
             $port
         );
+        
         if ($this->_connection === false || mysqli_connect_errno()) {
             /**
              * @see Zend_Db_Adapter_Mysqli_Exception
