--- tests/Zend/Form/FormTest.php	(revision 24129)
+++ tests/Zend/Form/FormTest.php	(working copy)
@@ -4461,6 +4461,24 @@
         $form = new Zend_Form();
         $form->addElement(NULL);
     }
+
+    /** @group ZF-10056 */
+    public function testIgnoredElementsDoNotGenerateErrorMessages()
+    {
+        $element = new Zend_Form_Element_Text('foo', array(
+            'validators' => array('alpha')
+        ));
+        $element2 = new Zend_Form_Element_Text('bar', array(
+            'validators' => array('alpha'),
+            'ignore' => true
+        ));
+        $this->form->addElements(array($element, $element2));
+        $this->form->isValid(array('foo' => 1, 'bar' => 'baz'));
+
+        $result = $this->form->getErrors();
+        $this->assertSame(array('alphaInvalid'), $result['foo']);
+        $this->assertSame(array(), $result['bar']);
+    }
 }
 
 class Zend_Form_FormTest_DisplayGroup extends Zend_Form_DisplayGroup
