Index: library/Zend/Dom/Query/Css2Xpath.php
===================================================================
--- library/Zend/Dom/Query/Css2Xpath.php	(revision 14875)
+++ library/Zend/Dom/Query/Css2Xpath.php	(working copy)
@@ -33,7 +33,7 @@
      * Transform CSS expression to XPath
      * 
      * @param  string $path 
-     * @return string
+     * @return string|array
      */
     public static function transform($path)
     {
Index: library/Zend/Dom/Query.php
===================================================================
--- library/Zend/Dom/Query.php	(revision 14875)
+++ library/Zend/Dom/Query.php	(working copy)
@@ -161,7 +161,7 @@
     /**
      * Perform an XPath query
      * 
-     * @param  string $xpathQuery
+     * @param  string|array $xpathQuery
      * @param  string $query CSS selector query
      * @return Zend_Dom_Query_Result
      */
