Index: tests/Zend/Controller/Response/HttpTest.php
===================================================================
--- tests/Zend/Controller/Response/HttpTest.php	(revision 23659)
+++ tests/Zend/Controller/Response/HttpTest.php	(working copy)
@@ -565,6 +565,15 @@
         $this->assertFalse($this->_response->isRedirect());
     }
 
+    /**
+     * @group 	ZF-10981
+     */
+    public function testIsNotRedirectWhen300ResponseCodeAndSetIgnoreRedirect()
+    {
+    	$this->_response->setHttpResponseCode(300)->setIgnoreRedirect();
+        $this->assertFalse($this->_response->isRedirect());
+    }
+    
     public function testHasExceptionOfType()
     {
         $this->assertFalse($this->_response->hasExceptionOfType('Zend_Controller_Response_Exception'));
Index: library/Zend/Controller/Response/Abstract.php
===================================================================
--- library/Zend/Controller/Response/Abstract.php	(revision 23659)
+++ library/Zend/Controller/Response/Abstract.php	(working copy)
@@ -297,6 +297,22 @@
     }
 
     /**
+     * Set redirect to false
+     * 
+     * Allows response code 300 to return an entity list of resource 
+     * characteristics and location(s)
+     * 
+     * @return Zend_Controller_Response_Abstract
+     */
+    public function setIgnoreRedirect()
+    {
+    	if ($this->_httpResponseCode == 300) {
+    		$this->_isRedirect = false;
+    	}
+    	return $this;
+    }
+    
+    /**
      * Retrieve HTTP response code
      *
      * @return int
