From 3c9d1a11b2ee94272ba21d458470975eb4c47811 Mon Sep 17 00:00:00 2001
From: Andy Fowler <andy@andyfowler.com>
Date: Tue, 26 Jan 2010 15:14:11 -0500
Subject: [PATCH] Additional bugfixes for numerically-named forms

---
 library/Zend/Form.php |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/library/Zend/Form.php b/library/Zend/Form.php
index 8cfea48..fd4acae 100644
--- a/library/Zend/Form.php
+++ b/library/Zend/Form.php
@@ -2293,7 +2293,7 @@ class Zend_Form implements Iterator, Countable, Zend_Validate_Interface
         $arrayKeys = array();
         foreach ($this->getSubForms() as $key => $subForm) {
             $array = $this->_getArrayName($subForm->getElementsBelongTo());
-            if (!empty($array)) {
+            if (!empty($array) || ('0' === $name)) {
                 if ($name == $array) {
                     return $subForm->getMessages(null, true);
                 }
@@ -2320,7 +2320,7 @@ class Zend_Form implements Iterator, Countable, Zend_Validate_Interface
             if (!empty($fMessages)) {
                 if (array_key_exists($key, $arrayKeys)) {
                     $fMessages = $this->_attachToArray($fMessages, $arrayKeys[$key]);
-                    $messages = array_merge($messages, $fMessages);
+                    $messages += $fMessages; // array union instead of array_merge to prevent shuffling numeric form names
                 } else {
                     $messages[$key] = $fMessages;
                 }
-- 
1.6.6.1

