Index: library/Zend/Mime/Decode.php
===================================================================
--- library/Zend/Mime/Decode.php	(revision 9578)
+++ library/Zend/Mime/Decode.php	(working copy)
@@ -193,7 +193,7 @@
         }
         
         $field = $firstName . '=' . $field;
-        if (!preg_match_all('%([^=\s]+)\s*=("[^"]+"|[^;]+)(;\s*|$)%', $field, $matches)) {
+        if (!preg_match_all('%([^=\s]+)\s*=\s*("[^"]+"|[^;]+)(;\s*|$)%', $field, $matches)) {
             throw new Zend_Exception('not a valid header field');
         }
 
