Index: Zend/Search/Lucene.php
===================================================================
--- Zend/Search/Lucene.php	(revision 16)
+++ Zend/Search/Lucene.php	(working copy)
@@ -1448,7 +1448,7 @@
     /**
      * Terms stream priority queue object
      *
-     * @var Zend_Search_Lucene_termStreamsPriorityQueue
+     * @var Zend_Search_Lucene_TermStreamsPriorityQueue
      */
     private $_termsStream = null;
 
@@ -1458,7 +1458,7 @@
     public function resetTermsStream()
     {
     	if ($this->_termsStream === null) {
-            $this->_termsStream = new Zend_Search_Lucene_termStreamsPriorityQueue($this->_segmentInfos);
+            $this->_termsStream = new Zend_Search_Lucene_TermStreamsPriorityQueue($this->_segmentInfos);
     	} else {
     		$this->_termsStream->resetTermsStream();
     	}
Index: Zend/Search/Lucene/MultiSearcher.php
===================================================================
--- Zend/Search/Lucene/MultiSearcher.php	(revision 16)
+++ Zend/Search/Lucene/MultiSearcher.php	(working copy)
@@ -867,7 +867,7 @@
     /**
      * Terms stream priority queue object
      *
-     * @var Zend_Search_Lucene_termStreamsPriorityQueue
+     * @var Zend_Search_Lucene_TermStreamsPriorityQueue
      */
     private $_termsStream = null;
 
@@ -877,7 +877,7 @@
     public function resetTermsStream()
     {
         if ($this->_termsStream === null) {
-            $this->_termsStream = new Zend_Search_Lucene_termStreamsPriorityQueue($this->_indices);
+            $this->_termsStream = new Zend_Search_Lucene_TermStreamsPriorityQueue($this->_indices);
         } else {
             $this->_termsStream->resetTermsStream();
         }
Index: Zend/Search/Lucene/TermStreamsPriorityQueue.php
===================================================================
--- Zend/Search/Lucene/TermStreamsPriorityQueue.php	(revision 16)
+++ Zend/Search/Lucene/TermStreamsPriorityQueue.php	(working copy)
@@ -33,7 +33,7 @@
  * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
-class Zend_Search_Lucene_termStreamsPriorityQueue implements Zend_Search_Lucene_Index_TermsStream_Interface
+class Zend_Search_Lucene_TermStreamsPriorityQueue implements Zend_Search_Lucene_Index_TermsStream_Interface
 {
 	/**
 	 * Array of term streams (Zend_Search_Lucene_Index_TermsStream_Interface objects)
