Index: StandardAutoloaderTest.php
===================================================================
--- StandardAutoloaderTest.php	(revision 24717)
+++ StandardAutoloaderTest.php	(working copy)
@@ -103,10 +103,10 @@
     {
         $options = array(
             'namespaces' => array(
-                'Zend\\'   => dirname(dirname(__FILE__)) . '/',
+                'Zend\\'   => dirname(dirname(__FILE__)) . DIRECTORY_SEPARATOR,
             ),
             'prefixes'   => array(
-                'Zend_'  => dirname(dirname(__FILE__)) . '/',
+                'Zend_'  => dirname(dirname(__FILE__)) . DIRECTORY_SEPARATOR,
             ),
             'fallback_autoloader' => true,
         );
@@ -120,10 +120,10 @@
     public function testPassingTraversableOptionsPopulatesProperties()
     {
         $namespaces = new ArrayObject(array(
-            'Zend\\' => dirname(dirname(__FILE__)) . '/',
+            'Zend\\' => dirname(dirname(__FILE__)) . DIRECTORY_SEPARATOR,
         ));
         $prefixes = new ArrayObject(array(
-            'Zend_' => dirname(dirname(__FILE__)) . '/',
+            'Zend_' => dirname(dirname(__FILE__)) . DIRECTORY_SEPARATOR,
         ));
         $options = new ArrayObject(array(
             'namespaces' => $namespaces,
