Index: Mysql.php
===================================================================
--- Mysql.php	(revision 165)
+++ Mysql.php	(working copy)
@@ -187,8 +187,7 @@
                 $scale = $matches[2];
             } else if (preg_match('/^((?:big|medium|small|tiny)?int)\((\d+)\)/', $row[$type], $matches)) {
                 $row[$type] = $matches[1];
-                // The optional argument of a MySQL int type is not precision
-                // or length; it is only a hint for display width.
+                $length = $matches[2];
             }
             if (strtoupper($row[$key]) == 'PRI') {
                 $primary = true;
