--- Frontcontroller.php.orig	2009-03-22 17:08:46.000000000 +0100
+++ Frontcontroller.php	2009-03-22 17:08:58.000000000 +0100
@@ -90,6 +90,10 @@ class Zend_Application_Resource_Frontcon
                     $front->setParams($value);
                     break;
 
+                case 'throwexceptions':
+                    $front->throwExceptions((bool)$value);
+                    break;
+
                 case 'plugins':
                     foreach ((array) $value as $pluginClass) {
                         $plugin = new $pluginClass();
