--- library/Zend/View/Helper/FormRadio.php      (revision 24600)
+++ library/Zend/View/Helper/FormRadio.php      (working copy)
@@ -153,8 +153,10 @@
             }
 
             // is it checked?
+            // $opt_value needs to be converted to string because we're doing strict in_array comparison
+            // and $value elements are strings
             $checked = '';
-            if (in_array($opt_value, $value)) {
+            if (in_array(strval($opt_value), $value, true)) {
                 $checked = ' checked="checked"';
             }
