Index: Filter.php
===================================================================
--- Filter.php	(revision 1328)
+++ Filter.php	(working copy)
@@ -524,6 +524,17 @@
         return (bool) preg_match('/(^\d{5}$)|(^\d{5}-\d{4}$)/', $value);
     }
 
+	/**
+     * Returnes values if it is not empty, FALSE otherwise.
+     *
+     * @param mixed $value
+     * @return mixed
+     */
+    public static function isEmpty($value)
+    {
+        return (bool) empty($value);
+    }
+
     /**
      * Returns value with all tags removed.
      *
Index: Filter/Input.php
===================================================================
--- Filter/Input.php	(revision 1328)
+++ Filter/Input.php	(working copy)
@@ -551,6 +551,25 @@
     }
 
 
+	/**
+     * Returnes value if it is not empty, FALSE otherwise
+     *
+     * @param mixed $key
+     * @return mixed
+     */
+    public function testEmpty($key)
+    {
+        if (!$this->keyExists($key)) {
+            return false;
+        }
+        if (Zend_Filter::isEmpty($this->_source[$key])) {
+            return $this->_source[$key];
+        }
+        
+        return false;
+    }
+
+
     /**
      * Returns value with all tags removed.
      *
