Index: library/Zend/Translate/Adapter/Array.php
===================================================================
--- library/Zend/Translate/Adapter/Array.php	(revision 22483)
+++ library/Zend/Translate/Adapter/Array.php	(working copy)
@@ -38,6 +38,36 @@
     private $_data = array();
 
     /**
+     * Generates the adapter
+     *
+     * @param  string|array|Zend_Config $options Translation content
+     */
+    public function __construct($options = array())
+    {
+        if ($options instanceof Zend_Config) {
+            $options = $options->toArray();
+        } else if (func_num_args() > 1) {
+            $args               = func_get_args();
+            $options            = array();
+            $options['content'] = array_shift($args);
+
+            if (!empty($args)) {
+                $options['locale'] = array_shift($args);
+            }
+
+            if (!empty($args)) {
+                $opt     = array_shift($args);
+                $options = array_merge($opt, $options);
+            }
+        } else if ((is_array($options) && !array_key_exists('content', $options))
+                   || is_string($options)) {
+            $options = array('content' => $options);
+        }
+
+        parent::__construct($options);
+    }
+
+    /**
      * Load translation data
      *
      * @param  string|array  $data
Index: tests/Zend/Translate/Adapter/_files/zf10020.php
===================================================================
--- tests/Zend/Translate/Adapter/_files/zf10020.php	(revision 0)
+++ tests/Zend/Translate/Adapter/_files/zf10020.php	(revision 0)
@@ -0,0 +1 @@
+<?php return array('message' => 'foo'); ?>
\ No newline at end of file
Index: tests/Zend/Translate/Adapter/ArrayTest.php
===================================================================
--- tests/Zend/Translate/Adapter/ArrayTest.php	(revision 22483)
+++ tests/Zend/Translate/Adapter/ArrayTest.php	(working copy)
@@ -334,6 +334,28 @@
     }
 
     /**
+     * @group ZF-10020
+     */
+    public function testInitializeMessage()
+    {
+        $messages = array('message' => 'foo');
+        $adapter = new Zend_Translate_Adapter_Array($messages);
+        $this->assertTrue($adapter->isTranslated('message'));
+
+        $messages = array(
+            'content' =>
+                array(
+                    'message' => 'foo'
+                )
+        );
+        $adapter = new Zend_Translate_Adapter_Array($messages);
+        $this->assertTrue($adapter->isTranslated('message'));
+
+        $adapter = new Zend_Translate_Adapter_Array(dirname(__FILE__) . '/_files/zf10020.php');
+        $this->assertTrue($adapter->isTranslated('message'));
+    }
+
+    /**
      * Ignores a raised PHP error when in effect, but throws a flag to indicate an error occurred
      *
      * @param  integer $errno
