Index: Application.php
===================================================================
--- Application.php	(revision 52)
+++ Application.php	(working copy)
@@ -124,7 +124,7 @@
     public function setOptions(array $options)
     {
         if (!empty($options['config'])) {
-            $options = $this->mergeOptions($options, $this->_loadConfig($options['config']));
+            $options = $this->mergeOptions($this->_loadConfig($options['config']), $options);
         }
         
         $this->_options = $options;
