Index: library/Zend/Date.php
===================================================================
--- library/Zend/Date.php	(revision 23872)
+++ library/Zend/Date.php	(working copy)
@@ -2260,7 +2260,7 @@
                     }
                     $parsed = Zend_Locale_Format::getTime($date, array('locale' => $locale, 'format_type' => 'iso', 'fix_date' => true));
                     return $this->_assign($calc, $this->mktime($parsed['hour'], $parsed['minute'], $parsed['second'], $month, $day, $year, true),
-                                                 $this->mktime($hour,           $minute,           $second,           $month, $day, $year, true), false);
+                                                 $this->mktime($hour,           $minute,           $second,           $month, $day, $year, true), $parsed['hour']);
                 } catch (Zend_Locale_Exception $e) {
                     require_once 'Zend/Date/Exception.php';
                     throw new Zend_Date_Exception($e->getMessage(), 0, $e, $date);
Index: tests/Zend/DateTest.php
===================================================================
--- tests/Zend/DateTest.php	(revision 23872)
+++ tests/Zend/DateTest.php	(working copy)
@@ -5669,6 +5669,17 @@
         $date = new Zend_Date(array('year' => 2008, 'month' => 10, 'day' => 12));
         $this->assertEquals('2008年10月12日', $date->get(Zend_Date::DATE_LONG, 'zh'));
     }
+	
+    /**
+     * @ZF-11230
+     */
+    public function testSetDateWithTimesPartOnDSTChange()
+    {
+        date_default_timezone_set('Europe/Kiev');
+        $date  = new Zend_Date(strtotime('2011-03-27T23:04:32+03:00'));
+        $date->set('00:00:00', Zend_Date::TIMES);
+        $this->assertEquals('2011-03-27T00:00:00+02:00', $date->getIso());
+    }
 }
 
 class Zend_Date_TestHelper extends Zend_Date
