--- library/Zend/Db/Adapter/Mysqli.php	(revision 23443)
+++ library/Zend/Db/Adapter/Mysqli.php	(working copy)
@@ -297,6 +297,12 @@
             $port = null;
         }
 
+        if (isset($this->_config['socket'])) {
+            $socket = $this->_config['socket'];
+        } else {
+            $socket = null;
+        }
+
         $this->_connection = mysqli_init();
 
         if(!empty($this->_config['driver_options'])) {
@@ -320,7 +326,8 @@
             $this->_config['username'],
             $this->_config['password'],
             $this->_config['dbname'],
-            $port
+            $port,
+            $socket
         );
 
         if ($_isConnected === false || mysqli_connect_errno()) {
Index: library/Zend/Db/Adapter/Abstract.php
===================================================================
--- library/Zend/Db/Adapter/Abstract.php	(revision 23443)
+++ library/Zend/Db/Adapter/Abstract.php	(working copy)
@@ -156,6 +156,7 @@
      * persistent     => (boolean) Whether to use a persistent connection or not, defaults to false
      * protocol       => (string) The network protocol, defaults to TCPIP
      * caseFolding    => (int) style of case-alteration used for identifiers
+     * socket         => (string) The socket or named pipe that should be used
      *
      * @param  array|Zend_Config $config An array or instance of Zend_Config having configuration data
      * @throws Zend_Db_Adapter_Exception
Index: documentation/manual/en/module_specs/Zend_Db_Adapter.xml
===================================================================
--- documentation/manual/en/module_specs/Zend_Db_Adapter.xml	(revision 23443)
+++ documentation/manual/en/module_specs/Zend_Db_Adapter.xml	(working copy)
@@ -380,6 +380,14 @@
                         load a non-Zend database adapter class.
                     </para>
                 </listitem>
+
+                <listitem>
+                    <para>
+                        <emphasis>socket</emphasis>:
+                        allows you to specify the socket or named pipe to use.
+                        Currently supported only by mysqli adapter.
+                    </para>
+                </listitem>
             </itemizedlist>
 
             <example id="zend.db.adapter.connecting.parameters.example1">
