Index: FormCheckboxTest.php
===================================================================
--- FormCheckboxTest.php	(revision 21015)
+++ FormCheckboxTest.php	(working copy)
@@ -95,6 +95,21 @@
         ));
         $this->assertRegexp('/<input[^>]*?(disabled="disabled")/', $html);
     }
+    public function testDoesNotRenderHiddenForDisableCheckbox()
+    {
+        $html = $this->helper->formCheckbox(array(
+            'name'    => 'foo',
+            'value'   => 'bar',
+            'attribs' => array('disabled' => 'disabled')
+        ));
+        $this->assertNotRegexp('/<input[^>]*?(type="hidden")/', $html);
+        $html = $this->helper->formCheckbox(array(
+            'name'    => 'foo',
+            'value'   => 'bar',
+            'attribs' => array('disable' => true)
+        ));
+        $this->assertNotRegexp('/<input[^>]*?(type="hidden")/', $html);
+    }
 
     /**
      * ZF-3505
