--- .svn/text-base/Debug.php.svn-base	2010-06-22 22:09:56.109492566 +0200
+++ Debug.php	2010-08-19 21:19:24.955611121 +0200
@@ -16,7 +16,7 @@
  * @package    Zend_Debug
  * @copyright  Copyright (c) 2005-2010 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id$
+ * @version    $Id: Debug.php 20096 2010-01-06 02:05:09Z bkarwin $
  */
 
 /**
@@ -105,4 +105,45 @@
         return $output;
     }
 
+	/**
+     * Debug helper function.  This is a wrapper for debug_print_backtrace()
+     * that adds the <pre /> tags, cleans up newlines and indents, and runs
+     * htmlentities() before output.
+     *
+     * @param  string $label OPTIONAL Label to prepend to output.
+     * @param  bool   $echo  OPTIONAL Echo output if true.
+     * @return string
+     */
+    public static function dump_backtrace($label=null, $echo=true)
+    {
+		// format the label
+        $label = ($label===null) ? '' : rtrim($label) . ' ';
+
+		// debug_print_backtrace into a buffer and keep the output
+        ob_start();
+        debug_print_backtrace();
+        $output = ob_get_clean();
+
+		// no need to trim spaces or breaks
+		if (self::getSapi() == 'cli') {
+			$output = PHP_EOL . $label
+                    . PHP_EOL . $output
+                    . PHP_EOL;
+		} else {
+            if(!extension_loaded('xdebug')) {
+                $output = htmlspecialchars($output, ENT_QUOTES);
+            }
+
+			$output = '<pre>'
+                    . $label
+                    . $output
+                    . '</pre>';
+		}
+
+		if ($echo) {
+            echo($output);
+        }
+        return $output;
+	}
+
 }
