Index: library/Zend/Config/Yaml.php
===================================================================
--- library/Zend/Config/Yaml.php	(revision 23879)
+++ library/Zend/Config/Yaml.php	(working copy)
@@ -315,7 +315,7 @@
                 // key: value
                 if (strlen($m[2])) {
                     // simple key: value
-                    $value = $m[2];
+                    $value = rtrim(preg_replace("/#.*$/", "", $m[2]));
                     // Check for booleans and constants
                     if (preg_match('/^(t(rue)?|on|y(es)?)$/i', $value)) {
                         $value = true;
Index: tests/Zend/Config/_files/inlinecomments.yaml
===================================================================
--- tests/Zend/Config/_files/inlinecomments.yaml	(revision 0)
+++ tests/Zend/Config/_files/inlinecomments.yaml	(revision 0)
@@ -0,0 +1,3 @@
+resources:
+    frontController:
+        controllerDirectory: APPLICATION_PATH/controllers #heynow!
\ No newline at end of file
Index: tests/Zend/Config/YamlTest.php
===================================================================
--- tests/Zend/Config/YamlTest.php	(revision 23879)
+++ tests/Zend/Config/YamlTest.php	(working copy)
@@ -46,6 +46,7 @@
         $this->_badIndentationConfig      = dirname(__FILE__) . '/_files/badindentation.yaml';
         $this->_booleansConfig            = dirname(__FILE__) . '/_files/booleans.yaml';
         $this->_constantsConfig           = dirname(__FILE__) . '/_files/constants.yaml';
+        $this->_yamlInlineCommentsConfig  = dirname(__FILE__) . '/_files/inlinecomments.yaml';
     }
 
     public function testLoadSingleSection()
@@ -315,4 +316,22 @@
         $this->assertEquals('ZEND_CONFIG_YAML_ENV', $config->env);
         $this->assertEquals('ZEND_CONFIG_YAML_ENV_PATH/test/this', $config->path);
     }
+    
+    /**
+     * @group ZF-11329
+     */
+    public function testAllowsInlineCommentsInValuesUsingHash()
+    {
+        $config = new Zend_Config_Yaml($this->_yamlInlineCommentsConfig, null);
+        $this->assertInstanceOf('Zend_Config', $config->resources);
+        $this->assertInstanceOf('Zend_Config', $config->resources->frontController);
+        $this->assertInternalType(
+            'string', 
+            $config->resources->frontController->controllerDirectory
+        );
+        $this->assertSame(
+            'APPLICATION_PATH/controllers',
+            $config->resources->frontController->controllerDirectory
+        );
+    }
 }
