Index: PluginLoaderTest.php
===================================================================
--- PluginLoaderTest.php	(revision 14488)
+++ PluginLoaderTest.php	(working copy)
@@ -85,6 +85,15 @@
         $this->assertEquals(2, count($paths['Zend_Loader_']));
     }
 
+    public function testAddPrefixPathMultipleTimes()
+    {
+        $loader = new Zend_Loader_PluginLoader();
+        $loader->addPrefixPath('Zend_Loader', $this->libPath . '/Zend/Loader')
+               ->addPrefixPath('Zend_Loader', $this->libPath . '/Zend/Loader');
+        $paths = $loader->getPaths();
+        $this->assertEquals(1, count($paths['Zend_Loader']));
+    }
+
     public function testAddPrefixPathStatically()
     {
         $this->key = 'foobar';
