--- Sqlite.php	2007-10-31 16:32:38.000000000 +0100
+++ Sqlite.php.patched	2007-10-31 16:33:56.000000000 +0100
@@ -93,6 +93,9 @@
      */
     public function __construct($options = array())
     {
+        if (!extension_loaded('sqlite')) {
+            Zend_Cache::throwException('The sqlite extension must be loaded for using this backend !');
+        }
         parent::__construct($options);
         if (is_null($this->_options['cache_db_complete_path'])) {
             Zend_Cache::throwException('cache_db_complete_path option has to set');
