Index: tests/runalltests.php
===================================================================
--- tests/runalltests.php	(revision 24859)
+++ tests/runalltests.php	(working copy)
@@ -50,7 +50,7 @@
 // run through phpunit
 while(list(, $file)=each($files)) {
     echo "Executing {$file}" . PHP_EOL;
-    shell_exec($PHPUNIT . ' --stderr -d memory_limit=-1 -d error_reporting=E_ALL\&E_STRICT -d display_errors=1 ' . escapeshellarg($file));
+    shell_exec($PHPUNIT . ' --verbose --debug --stderr -d memory_limit=-1 -d error_reporting=E_ALL\&E_STRICT -d display_errors=1 ' . escapeshellarg($file));
     echo PHP_EOL;
 }
 exit(0);
Index: tests/Zend/Reflection/MethodTest.php
===================================================================
--- tests/Zend/Reflection/MethodTest.php	(revision 24860)
+++ tests/Zend/Reflection/MethodTest.php	(working copy)
@@ -121,5 +121,15 @@
         $this->assertEquals("    {\n\n        return 'mixedValue';\n\n    }\n", $reflectionMethod->getContents(false));
     }
 
+    /**
+     * @group ZF-10870
+     */
+    public function testGetBodyReturnsCorrectBodyWhenMethodSignatureIsMultiline()
+    {
+        $body = '        // FUNKY SIGNATURE';
+        $reflectionMethod = new Zend_Reflection_Method('Zend_Reflection_TestSampleClass7', 'bigMethodSignature');
+        $this->assertEquals($body, $reflectionMethod->getBody());
+    }
+
 }
 
Index: tests/Zend/Reflection/_files/TestSampleClass.php
===================================================================
--- tests/Zend/Reflection/_files/TestSampleClass.php	(revision 24859)
+++ tests/Zend/Reflection/_files/TestSampleClass.php	(working copy)
@@ -182,3 +182,12 @@
 
     }
 }
+
+class Zend_Reflection_TestSampleClass7
+{
+    public function bigMethodSignature($arg1, $arg2, $arg3, 
+        $arg4, $arg5, $arg6)
+    {
+        // FUNKY SIGNATURE
+    }
+}
Index: tests/Zend/Reflection/FileTest.php
===================================================================
--- tests/Zend/Reflection/FileTest.php	(revision 24859)
+++ tests/Zend/Reflection/FileTest.php	(working copy)
@@ -60,7 +60,7 @@
         require_once $fileToRequire;
         $reflectionFile = new Zend_Reflection_File($fileToRequire);
         $this->assertEquals(get_class($reflectionFile), 'Zend_Reflection_File');
-        $this->assertEquals(count($reflectionFile->getClasses()), 8);
+        $this->assertEquals(count($reflectionFile->getClasses()), 9);
         $this->assertEquals(get_class($reflectionFile->getClass('Zend_Reflection_TestSampleClass2')), 'Zend_Reflection_Class');
     }
 
@@ -110,7 +110,7 @@
         require_once $fileToRequire;
         $reflectionFile = new Zend_Reflection_File($fileToRequire);
         $this->assertEquals(9, $reflectionFile->getStartLine());
-        $this->assertEquals(185, $reflectionFile->getEndLine());
+        $this->assertEquals(194, $reflectionFile->getEndLine());
     }
 
     public function testFileGetDocblockReturnsFileDocblock()
