Index: HeadStyle.php
===================================================================
--- HeadStyle.php	(revision 21981)
+++ HeadStyle.php	(working copy)
@@ -355,14 +355,19 @@
             }
         }
 
+        $conditional = (
+            isset($item->attributes['conditional'])
+            && !empty($item->attributes['conditional'])
+            && is_string($item->attributes['conditional'])
+        );
+
         $html = '<style type="text/css"' . $attrString . '>' . PHP_EOL
-              . $indent . '<!--' . PHP_EOL . $indent . $item->content . PHP_EOL . $indent . '-->' . PHP_EOL
+              . ($conditional ? '' : $indent . '<!--' . PHP_EOL)
+              . $indent . $item->content . PHP_EOL
+              . ($conditional ? '' : $indent . '-->' . PHP_EOL)
               . '</style>';
 
-        if (isset($item->attributes['conditional'])
-            && !empty($item->attributes['conditional'])
-            && is_string($item->attributes['conditional']))
-        {
+        if ($conditional) {
             $html = '<!--[if ' . $item->attributes['conditional'] . ']> ' . $html . '<![endif]-->';
         }
 
