Index: Route.php
===================================================================
--- Route.php	(revision 18355)
+++ Route.php	(working copy)
@@ -371,8 +371,9 @@
                 }
             } else {
                 if (!$reset) $data += $this->_wildcardData;
+                $defaults = $this->getDefaults();
                 foreach ($data as $var => $value) {
-                    if ($value !== null) {
+                    if ($value !== null && (!isset($defaults[$var]) || $value !== $defaults[$var])) {
                         $url[$key++] = $var;
                         $url[$key++] = $value;
                         $flag = true;
