--- Oracle.php.old	2006-06-30 00:01:04.000000000 +0200
+++ Oracle.php	2006-07-03 17:35:28.000000000 +0200
@@ -309,6 +309,9 @@
                 break;
             case Zend_Db::FETCH_OBJ:
                 break;
+            case Zend_Db::FETCH_COLUMN:
+				$flags |= OCI_ASSOC;
+                break;
             default:
                 throw new Zend_Db_Statement_Exception("invalid fetch mode specified");
                 break;
