*** Zend/Test/PHPUnit/Db/Metadata/Generic.php  2011-10-27 17:59:40.000000000 -0500
--- p1  2011-10-27 17:59:20.000000000 -0500
***************
*** 164,166 ****
--- 164,184 ----
      {
          return false;
      }
+     
+     /**
+      * Disables primary keys if rdbms does not allow setting them otherwise
+      *
+      * @param string $tableName
+      */
+     public function disablePrimaryKeys($tableName)
+     {
+     }
+     
+     /**
+      * Reenables primary keys after they have been disabled
+      *
+      * @param string $tableName
+      */
+     public function enablePrimaryKeys($tableName)
+     {
+     }
