Index: PluginLoader.php
--- PluginLoader.php Base (BASE)
+++ PluginLoader.php Locally Modified (Based On LOCAL)
@@ -256,7 +256,7 @@
 
         if ($path != null) {
             $pos = array_search($path, $registry[$prefix]);
-            if ($pos === null) {
+            if ($pos === null || $pos === false) {
                 require_once 'Zend/Loader/PluginLoader/Exception.php';
                 throw new Zend_Loader_PluginLoader_Exception('Prefix ' . $prefix . ' / Path ' . $path . ' was not found in the PluginLoader.');
             }
