Index: library/Zend/Http/Client.php
===================================================================
--- library/Zend/Http/Client.php	(revision 2920)
+++ library/Zend/Http/Client.php	(working copy)
@@ -180,6 +180,13 @@
      * @var string
      */
     protected $last_request = null;
+    
+    /**
+     * The last HTTP response object
+     *
+     * @var string
+     */
+    protected $last_response = null;
 
     /**
      * Redirection counter
@@ -665,6 +672,19 @@
     }
 
     /**
+     * Get the last HTTP response object
+     * 
+     * @return Zend_Http_Response
+     */
+    public function getLastResponse() {
+        if ($this->last_response instanceof Zend_Http_Response) {
+            return $this->last_response;
+        } else {
+            throw new Zend_Http_Exception('Response is empty');
+        }
+    }
+
+    /**
      * Send the HTTP request and return an HTTP response object
      *
      * @param string $method
@@ -705,6 +725,7 @@
                 $uri, $this->config['httpversion'], $headers, $body);
                 
             $response = $this->adapter->read();
+
             if (! $response)
                 throw new Zend_Http_Client_Exception('Unable to read response, or response is empty');
                 
@@ -761,6 +782,8 @@
             
         } while ($this->redirectCounter < $this->config['maxredirects']);
 
+        $this->last_response = $response;
+
         return $response;
     }
     
