Index: Form/Element/Radio.php
===================================================================
--- Form/Element/Radio.php	(revision 21857)
+++ Form/Element/Radio.php	(working copy)
@@ -49,9 +49,18 @@
      */
     public function loadDefaultDecorators()
     {
-        parent::loadDefaultDecorators();
-        if (false !== $decorator = $this->getDecorator('Label')) {
-            $decorator->setOption('disableFor', true);
+        if ($this->loadDefaultDecoratorsIsDisabled()) {
+            return;
         }
+
+        $decorators = $this->getDecorators();
+        if (empty($decorators)) {
+            $this->addDecorator('ViewHelper')
+                ->addDecorator('Errors')
+                ->addDecorator('Description', array('tag' => 'p', 'class' => 'description'))
+                ->addDecorator('HtmlTag', array('tag' => 'dd',
+                                                'id'  => $this->getName() . '-element'))
+                ->addDecorator('Label', array('tag' => 'dt', 'disableFor' => true));
+        }
     }
 }
