Index: Redirector.php
===================================================================
--- Redirector.php	(revision 9106)
+++ Redirector.php	(working copy)
@@ -19,7 +19,7 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 
-/** 
+/**
  * @see Zend_Controller_Action_Exception
  */
 require_once 'Zend/Controller/Action/Exception.php';
@@ -380,7 +380,7 @@
      * @param  array  $params
      * @return void
      */
-    public function goto($action, $controller = null, $module = null, array $params = array())
+    public function gotoAction($action, $controller = null, $module = null, array $params = array())
     {
         $this->setGoto($action, $controller, $module, $params);
 
@@ -502,4 +502,20 @@
     {
         $this->goto($action, $controller, $module, $params);
     }
+
+    /**
+     * Handles backwards compatabiliy for goto
+     *
+     * @param   string    $method
+     * @param   array     $args
+     * @return  void
+     */
+    public function __call($method, array $args)
+    {
+        if ($method == 'goto') {
+            return call_user_func_array(array($this, 'gotoAction'), $args);
+        }
+        require_once 'Zend/Controller/Action/Exception.php';
+        throw new Zend_Controller_Action_Exception("Unrecognized method '$method()'");
+    }
 }
