--- library/Zend/Service/Yahoo.php	(revision 7361)
+++ library/Zend/Service/Yahoo.php	(working copy)
@@ -253,8 +253,7 @@
     {
         static $defaultOptions = array('type'     => 'all',
                                        'start'    => 1,
-                                       'sort'     => 'rank',
-                                       'language' => 'en');
+                                       'sort'     => 'rank');
 
         $options = $this->_prepareOptions($query, $options, $defaultOptions);
 
@@ -411,7 +410,6 @@
     {
         static $defaultOptions = array('type'     => 'all',
                                        'start'    => 1,
-                                       'language' => 'en',
                                        'license'  => 'any',
                                        'results'  => 10,
                                        'format'   => 'any');
@@ -704,7 +702,9 @@
             throw new Zend_Service_Exception("Invalid value for option 'start': {$options['start']}");
         }
 
-        $this->_validateLanguage($options['language']);
+        if (isset($options['language'])) {
+            $this->_validateLanguage($options['language']);
+        }
 
         $this->_validateInArray('sort', $options['sort'], array('rank', 'date'));
         $this->_validateInArray('type', $options['type'], array('all', 'any', 'phrase'));
@@ -829,7 +829,9 @@
             throw new Zend_Service_Exception("Invalid value for option 'start': {$options['start']}");
         }
 
-        $this->_validateLanguage($options['language']);
+        if (isset($options['language'])) {
+            $this->_validateLanguage($options['language']);
+        }
 
         $this->_validateInArray('type', $options['type'], array('all', 'any', 'phrase'));
         $this->_validateInArray('format', $options['format'], array('any', 'html', 'msword', 'pdf', 'ppt', 'rss',
