--- Zend.15243/Application/Resource/Frontcontroller.php	2009-04-17 11:52:17.000000000 -0400
+++ Zend/Application/Resource/Frontcontroller.php	2009-04-28 21:57:21.000000000 -0400
@@ -48,13 +48,7 @@
         foreach ($this->getOptions() as $key => $value) {
             switch (strtolower($key)) {
                 case 'controllerdirectory':
-                    if (is_string($value)) {
-                        $front->setControllerDirectory($value);
-                    } elseif (is_array($value)) {
-                        foreach ($value as $module => $directory) {
-                            $front->setControllerDirectory($directory, $module);
-                        }
-                    }
+                    $front->setControllerDirectory($value);
                     break;
                     
                 case 'modulecontrollerdirectoryname':
