--- library/Zend/Db/Adapter/Abstract.php	(revision 17689)
+++ library/Zend/Db/Adapter/Abstract.php	(working copy)
@@ -738,7 +738,7 @@
      *
      * @param string|Zend_Db_Select $sql An SQL SELECT statement.
      * @param mixed $bind Data to bind into SELECT placeholders.
-     * @return string
+     * @return array 
      */
     public function fetchAssoc($sql, $bind = array())
     {
@@ -775,7 +775,7 @@
      *
      * @param string|Zend_Db_Select $sql An SQL SELECT statement.
      * @param mixed $bind Data to bind into SELECT placeholders.
-     * @return string
+     * @return array 
      */
     public function fetchPairs($sql, $bind = array())
     {
