Index: .
===================================================================
--- .	(revision 11496)
+++ .	(working copy)
@@ -399,7 +399,7 @@
      */
     public function getType($type)
     {
-        switch ($type) {
+        switch (strtolower($type)) {
             case 'string':
             case 'str':
                 return 'xsd:string';
@@ -427,12 +427,11 @@
                 break;
             case 'void':
                 return '';
-            default:
-                if (class_exists($type) && $this->_extractComplexTypes)
-                    return $this->addComplexType($type);
-                else
-                    return 'xsd:anyType';
-            }
+        }
+        if (class_exists($type) && $this->_extractComplexTypes)
+            return $this->addComplexType($type);
+        else
+            return 'xsd:anyType';
     }
 
     /**
