Index: library/Zend/Log/Filter/Abstract.php
===================================================================
--- library/Zend/Log/Filter/Abstract.php	(revision 21863)
+++ library/Zend/Log/Filter/Abstract.php	(working copy)
@@ -14,7 +14,7 @@
  *
  * @category   Zend
  * @package    Zend_Log
- * @subpackage Writer
+ * @subpackage Filter
  * @copyright  Copyright (c) 2005-2010 Zend Technologies USA Inc. (http://www.zend.com)
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  * @version    $Id$
