Index: library/Zend/Loader/PluginLoader.php
===================================================================
--- library/Zend/Loader/PluginLoader.php	(revision 24770)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -372,7 +372,11 @@
 
         $registry  = array_reverse($registry, true);
         $found     = false;
-        $classFile = str_replace('_', DIRECTORY_SEPARATOR, $name) . '.php';
+        if (false !== strpos($name, '\\')) {
+            $classFile = str_replace('\\', DIRECTORY_SEPARATOR, $name) . '.php';
+        } else {
+            $classFile = str_replace('_', DIRECTORY_SEPARATOR, $name) . '.php';
+        }
         $incFile   = self::getIncludeFileCache();
         foreach ($registry as $prefix => $paths) {
             $className = $prefix . $name;
