--- library/Zend/Json/Server.php	Sun Sep 11 14:25:08 2011
+++ library/Zend/Json/Server.php	Fri Dec 23 16:17:19 2011
@@ -464,6 +464,7 @@
         foreach ($method->getPrototypes() as $prototype) {
             $return[] = $prototype->getReturnType();
         }
+        $return = array_values(array_unique($return));
         if (1 == count($return)) {
             return $return[0];
         }
