--- tests/Zend/View/Helper/DoctypeTest.php	(revision 23474)
+++ tests/Zend/View/Helper/DoctypeTest.php	(working copy)
@@ -122,7 +122,7 @@
 
     public function testIsXhtmlReturnsTrueForXhtmlDoctypes()
     {
-        foreach (array('XHTML1_STRICT', 'XHTML1_TRANSITIONAL', 'XHTML1_FRAMESET', 'XHTML5') as $type) {
+        foreach (array('XHTML1_STRICT', 'XHTML1_TRANSITIONAL', 'XHTML1_FRAMESET', 'XHTML1_RDFA', 'XHTML5') as $type) {
             $doctype = $this->helper->doctype($type);
             $this->assertEquals($type, $doctype->getDoctype());
             $this->assertTrue($doctype->isXhtml());
@@ -159,7 +159,23 @@
             $this->assertFalse($doctype->isHtml5());
 		}
 	}
-	
+
+    public function testIsRdfa1()
+    {
+        $doctype = $this->helper->doctype('XHTML1_RDFA');
+        $this->assertTrue($doctype->isRdfa1());
+
+        // built-in doctypes
+        foreach (array('HTML4_STRICT', 'HTML4_LOOSE', 'HTML4_FRAMESET', 'XHTML1_STRICT', 'XHTML1_TRANSITIONAL', 'XHTML1_FRAMESET') as $type) {
+            $doctype = $this->helper->doctype($type);
+            $this->assertFalse($doctype->isRdfa1());
+        }
+
+        // custom doctype
+        $doctype = $this->helper->doctype('<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 10.0 Strict//EN" "http://framework.zend.com/foo/DTD/html10-custom.dtd">');
+        $this->assertFalse($doctype->isRdfa1());
+    }
+
 	public function testCanRegisterCustomHtml5Doctype() {
 		$doctype = $this->helper->doctype('<!DOCTYPE html>');
         $this->assertEquals('CUSTOM', $doctype->getDoctype());
Index: library/Zend/View/Helper/Doctype.php
===================================================================
--- library/Zend/View/Helper/Doctype.php	(revision 23474)
+++ library/Zend/View/Helper/Doctype.php	(working copy)
@@ -43,6 +43,7 @@
     const XHTML1_STRICT       = 'XHTML1_STRICT';
     const XHTML1_TRANSITIONAL = 'XHTML1_TRANSITIONAL';
     const XHTML1_FRAMESET     = 'XHTML1_FRAMESET';
+    const XHTML1_RDFA         = 'XHTML1_RDFA';
     const XHTML_BASIC1        = 'XHTML_BASIC1';
     const XHTML5              = 'XHTML5';
     const HTML4_STRICT        = 'HTML4_STRICT';
@@ -87,6 +88,7 @@
                     self::XHTML1_STRICT       => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">',
                     self::XHTML1_TRANSITIONAL => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">',
                     self::XHTML1_FRAMESET     => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Frameset//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-frameset.dtd">',
+                    self::XHTML1_RDFA         => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN" "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-1.dtd">',
                     self::XHTML_BASIC1        => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML Basic 1.0//EN" "http://www.w3.org/TR/xhtml-basic/xhtml-basic10.dtd">',
                     self::XHTML5              => '<!DOCTYPE html>',
                     self::HTML4_STRICT        => '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">',
@@ -116,6 +118,7 @@
                 case self::XHTML1_STRICT:
                 case self::XHTML1_TRANSITIONAL:
                 case self::XHTML1_FRAMESET:
+                case self::XHTML1_RDFA:
                 case self::XHTML_BASIC1:
                 case self::XHTML5:
                 case self::HTML4_STRICT:
@@ -197,6 +200,16 @@
 	}
 
     /**
+     * Is doctype RDFa 1.0?
+     * 
+     * @return boolean
+     */
+    public function isRdfa1()
+    {
+        return (stristr($this->doctype(), 'xhtml-rdfa-1.dtd') ? true : false);
+    }
+
+    /**
      * String representation of doctype
      *
      * @return string
Index: documentation/manual/en/module_specs/Zend_View-Helpers-Doctype.xml
===================================================================
--- documentation/manual/en/module_specs/Zend_View-Helpers-Doctype.xml	(revision 23474)
+++ documentation/manual/en/module_specs/Zend_View-Helpers-Doctype.xml	(working copy)
@@ -21,6 +21,7 @@
         <listitem><para><constant>XHTML1_STRICT</constant></para></listitem>
         <listitem><para><constant>XHTML1_TRANSITIONAL</constant></para></listitem>
         <listitem><para><constant>XHTML1_FRAMESET</constant></para></listitem>
+        <listitem><para><constant>XHTML1_RDFA</constant></para></listitem>
         <listitem><para><constant>XHTML_BASIC1</constant></para></listitem>
         <listitem><para><constant>HTML4_STRICT</constant></para></listitem>
         <listitem><para><constant>HTML4_LOOSE</constant></para></listitem>
@@ -94,6 +95,17 @@
     // do something differently
 }
 ]]></programlisting>
+
+        <para>
+            And also check if the doctype implements the <acronym>RDFa 1.0</acronym> standard
+        </para>
+
+        <programlisting language="php"><![CDATA[
+if ($view->doctype()->isRdfa1()) {
+    // do something differently
+}
+]]></programlisting>
+
     </example>
 </sect3>
 <!--
