Index: library/Zend/Db/Adapter/Abstract.php
===================================================================
--- library/Zend/Db/Adapter/Abstract.php        (revision 9063)
+++ library/Zend/Db/Adapter/Abstract.php        (working copy)
@@ -494,7 +494,7 @@
      * @param  mixed        $where UPDATE WHERE clause(s).
      * @return int          The number of affected rows.
      */
-    public function update($table, array $bind, $where = '')
+    public function update($table, array $bind, $where = '', array $params = null)
     {
         /**
          * Build "col = ?" pairs for the statement,
@@ -524,7 +524,8 @@
         /**
          * Execute the statement and return the number of affected rows
          */
-        $stmt = $this->query($sql, array_values($bind));
+        $params = array_merge(array_values($bind), $params);
+        $stmt = $this->query($sql, $params);
         $result = $stmt->rowCount();
         return $result;
     }
@@ -536,7 +537,7 @@
      * @param  mixed        $where DELETE WHERE clause(s).
      * @return int          The number of affected rows.
      */
-    public function delete($table, $where = '')
+    public function delete($table, $where = '', array $params = null)
     {
         $where = $this->_whereExpr($where);

@@ -550,7 +551,7 @@
         /**
          * Execute the statement and return the number of affected rows
          */
-        $stmt = $this->query($sql);
+        $stmt = $this->query($sql, $params);
         $result = $stmt->rowCount();
         return $result;
     }