Index: library/Zend/Form/Element/Button.php
===================================================================
--- library/Zend/Form/Element/Button.php	(revision 25167)
+++ library/Zend/Form/Element/Button.php	(working copy)
@@ -39,4 +39,18 @@
      * @var string
      */
     public $helper = 'formButton';
+
+    /**
+     * Validate element value (pseudo)
+     *
+     * There is no need to reset the value
+     *
+     * @param  mixed $value Is always ignored
+     * @param  mixed $context Is always ignored
+     * @return boolean Returns always TRUE
+     */
+    public function isValid($value, $context = null)
+    {
+        return true;
+    }
 }
