Index: ResponseTest.php
===================================================================
--- ResponseTest.php	(revision 12794)
+++ ResponseTest.php	(working copy)
@@ -170,6 +170,9 @@
      */
     public function testPhpDateTimeSerializedToAmf3Date()
     {
+        if (!class_exists('DateTime')) {
+            $this->markTestSkipped('The DateTime class is not available.');
+        }
         // Create php object to serialize
         date_default_timezone_set('America/Chicago');
         $dateSrc = '1978-10-23 4:20 America/Chicago';
@@ -671,6 +674,9 @@
      */
     public function testPhpDateTimeSerializedToAmf0Date()
     {
+        if (!class_exists('DateTime')) {
+            $this->markTestSkipped('The DateTime class is not available.');
+        }
         date_default_timezone_set('America/Chicago');
         $dateSrc = '1978-10-23 4:20 America/Chicago';
         $date = new DateTime($dateSrc, new DateTimeZone('America/Chicago'));
