Index: Zend/View/Abstract.php
===================================================================
--- Zend/View/Abstract.php	(revision 1349)
+++ Zend/View/Abstract.php	(working copy)
@@ -153,7 +153,11 @@
          * @todo exception?
          */
         if ($key[0] != '_') {
-            return isset($this->_vars[$key]) ? $this->_vars[$key] : null;
+        	if (isset($this->_vars[$key])) {
+        		return $this->_vars[$key];
+        	} else {
+        		return null;
+        	}
         }
     }
 
