Index: library/Zend/Json/Encoder.php
===================================================================
--- library/Zend/Json/Encoder.php       (revision 24101)
+++ library/Zend/Json/Encoder.php       (working copy)
@@ -151,7 +151,8 @@
             }
         }

-        return '{"__className":"' . get_class($value) . '"'
+        $className = get_class($value);
+        return '{"__className":' . $this->_encodeString($className)
                 . $props . '}';
     }

Index: tests/Zend/JsonTest.php
===================================================================
--- tests/Zend/JsonTest.php     (revision 24101)
+++ tests/Zend/JsonTest.php     (working copy)
@@ -762,6 +762,25 @@
         $expectedDecoding = '{"__className":"ArrayIterator","0":"foo"}';
         $this->assertEquals($encoded, $expectedDecoding);
     }
+
+    /**
+     * @group ZF-11356
+     */
+    public function testEncoderEscapesNamespacedClassNamesProperly()
+    {
+        if (version_compare(PHP_VERSION, '5.3.0') === -1) {
+            $this->markTestSkipped('Namespaces not available in PHP < 5.3.0');
+        }
+
+        require_once dirname(__FILE__ ) . "/Json/_files/Zf11356-NamespacedClass.php";
+        $inputValue = new \Zend\JsonTest\Zf11356\NamespacedClass(array('foo'));
+
+        $encoded = Zend_Json_Encoder::encode($inputValue);
+        $this->assertEquals(
+            '{"__className":"Zend\\\\JsonTest\\\\Zf11356\\\\NamespacedClass","0":"foo"}',
+            $encoded
+        );
+    }
 }

 /**
Index: tests/Zend/Json/_files/Zf11356-NamespacedClass.php
===================================================================
--- tests/Zend/Json/_files/Zf11356-NamespacedClass.php  (revision 0)
+++ tests/Zend/Json/_files/Zf11356-NamespacedClass.php  (revision 0)
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Json
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+namespace Zend\JsonTest\Zf11356;
+
+/**
+ * Namespaced Class for use in Zend_JsonTest::testEncoderEscapesNamespacedClassNamesProperly
+ *
+ * @category   Zend
+ * @package    Zend_Json
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class NamespacedClass extends \ArrayIterator
+{
+}
\ No newline at end of file

Property changes on: tests/Zend/Json/_files/Zf11356-NamespacedClass.php
___________________________________________________________________
Added: svn:keywords
   + Id

