Index: library/Zend/Cache/Manager.php
===================================================================
--- library/Zend/Cache/Manager.php	(revision 438)
+++ library/Zend/Cache/Manager.php	(working copy)
@@ -293,6 +293,15 @@
                 $current['backend']['options'][$key] = $value;
             }
         }
+
+        // ZF-11039
+        if (isset($options['frontend']['customFrontendNaming'])) {
+            $current['frontend']['customFrontendNaming'] = $options['frontend']['customFrontendNaming'];
+        }
+        if (isset($options['backend']['customBackendNaming'])) {
+            $current['backend']['customBackendNaming'] = $options['backend']['customBackendNaming'];
+        }
+
         return $current;
     }
 }

