Index: ProjectProvider.php
===================================================================
--- ProjectProvider.php	(revision 24246)
+++ ProjectProvider.php	(working copy)
@@ -29,6 +29,7 @@
  * @license    http://framework.zend.com/license/new-bsd     New BSD License
  */
 class Zend_Tool_Project_Provider_ProjectProvider extends Zend_Tool_Project_Provider_Abstract
+    implements Zend_Tool_Framework_Provider_Pretendable
 {
 
     /**
