Index: library/Zend/Cache/Frontend/Page.php
===================================================================
--- library/Zend/Cache/Frontend/Page.php	(revision 9496)
+++ library/Zend/Cache/Frontend/Page.php	(working copy)
@@ -175,7 +175,7 @@
     {
         $found = null;
         foreach ($this->_specificOptions['memorize_headers'] as $key => $value) {
-            if (strolower($value) == 'content-type') {
+            if (strtolower($value) == 'content-type') {
                 $found = $key;
             }
         }
@@ -300,10 +300,11 @@
             $headersList = headers_list();
             foreach($this->_specificOptions['memorize_headers'] as $key=>$headerName) {
                 foreach ($headersList as $headerSent) {
-                    $tmp = split(':', $header);
-                    $headerSentName = strolower(trim($tmp[0]));
-                    if (strolower($headerName) == $headerSentName) {
-                        $storedHeaders[] = array(trim($tmp[0]), trim($tmp[1]));
+                    $tmp = split(':', $headerSent);
+                    $headerSentName = trim(array_shift($tmp));
+					$headerSentValue = trim(implode(':', $tmp));
+                    if (strtolower($headerName) == strtolower($headerSentName)) {
+                        $storedHeaders[] = array($headerSentName, $headerSentValue);
                     }
                 }
             }
