Index: /home/shahar.e/Framework/trunk/library/Zend/Controller/Dispatcher/Standard.php
===================================================================
--- /home/shahar.e/Framework/trunk/library/Zend/Controller/Dispatcher/Standard.php	(revision 5223)
+++ /home/shahar.e/Framework/trunk/library/Zend/Controller/Dispatcher/Standard.php	(working copy)
@@ -259,6 +259,7 @@
         try {
             Zend_Loader::loadFile($file, $dir, true);
         } catch (Zend_Exception $e) {
+            require_once 'Zend/Controller/Dispatcher/Exception.php';
             throw new Zend_Controller_Dispatcher_Exception('Cannot load controller class "' . $className . '" from file "' . $file . '" in directory "' . $dir . '"');
         }
 
