--- D:/Downloads/Web/Date.php	Fri Apr 16 16:31:04 2010
+++ D:/5/Date.php	Fri Apr 16 16:31:58 2010
@@ -2816,7 +2816,7 @@
             $time .= str_pad($parsed['second'], 2, '0', STR_PAD_LEFT);
         }
 
-        $return = $this->_calcdetail($calc, $time, self::TIMES, 'de');
+        $return = $this->_calcdetail($calc, $time, self::TIMES, $locale);
         if ($calc != 'cmp') {
             return $this;
         }
@@ -2979,7 +2979,7 @@
             $date  = $parsed['day'] . "." . $parsed['month'] . "." . $parsed['year'];
         }
 
-        $return = $this->_calcdetail($calc, $date, self::DATE_MEDIUM, 'de');
+        $return = $this->_calcdetail($calc, $date, self::DATE_MEDIUM, $locale);
         if ($calc != 'cmp') {
             return $this;
         }
