Index: library/Zend/Soap/Wsdl.php
===================================================================
--- library/Zend/Soap/Wsdl.php	(revision 23986)
+++ library/Zend/Soap/Wsdl.php	(working copy)
@@ -560,7 +560,7 @@
                 return 'soap-enc:Array';
                 break;
             case 'object':
-                return 'xsd:struct';
+                return 'soap-enc:Struct';
                 break;
             case 'mixed':
                 return 'xsd:anyType';
