Index: tests/Zend/View/Helper/FormRadioTest.php
===================================================================
--- tests/Zend/View/Helper/FormRadioTest.php	(revision 19794)
+++ tests/Zend/View/Helper/FormRadioTest.php	(working copy)
@@ -309,11 +309,12 @@
         $options = array(
             'foo' => 'Foo',
             'bar' => 'Bar',
-            'baz' => 'Baz'
+            'baz' => 'Baz',
+            'Eggs'
         );
         $html = $this->helper->formRadio(array(
             'name'    => 'foo',
-            'value'   => array('foo', 'baz'),
+            'value'   => array('foo', 'baz', null),
             'options' => $options,
         ));
 
