Index: Zend/Log/Formatter/Firebug.php
===================================================================
--- Zend/Log/Formatter/Firebug.php	(revision 0)
+++ Zend/Log/Formatter/Firebug.php	(revision 0)
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Log
+ * @subpackage Formatter
+ * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+
+/** Zend_Log_Formatter_Interface */
+require_once 'Zend/Log/Formatter/Interface.php';
+
+/** Zend_Log_Exception */
+require_once 'Zend/Log/Exception.php';
+
+/**
+ * @category   Zend
+ * @package    Zend_Log
+ * @subpackage Formatter
+ * @copyright  Copyright (c) 2005-2008 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ */
+class Zend_Log_Formatter_Firebug implements Zend_Log_Formatter_Interface 
+{
+    /**
+     * This method formats the event for the firebug writer.
+     * 
+     * The default is to just send the message parameter, but through 
+     * extension of this class and calling the
+     * {@see Zend_Log_Writer_Firebug::setFormatter()} method you can
+     * pass as much of the event data as you are interested in.
+     *
+     * @param  array    $event    event data
+     * @return mixed              event message
+     */
+    public function format($event)
+    {
+        return $event['message'];
+    }
+}
Index: Zend/Log/Writer/Firebug.php
===================================================================
--- Zend/Log/Writer/Firebug.php	(revision 12682)
+++ Zend/Log/Writer/Firebug.php	(working copy)
@@ -73,6 +73,8 @@
         if (php_sapi_name()=='cli') {
             $this->setEnabled(false);
         }
+        
+        $this->_formatter = new Zend_Log_Formatter_Firebug();
     }
     
     /**
@@ -153,18 +155,6 @@
     }
 
     /**
-     * Formatting is not possible on this writer
-     *
-     * @return void
-     */
-    public function setFormatter($formatter)
-    {
-        /** @see Zend_Log_Exception */
-        require_once 'Zend/Log/Exception.php';
-        throw new Zend_Log_Exception(get_class() . ' does not support formatting');
-    }
-
-    /**
      * Log a message to the Firebug Console.
      *
      * @param array $event The event data
@@ -182,6 +172,12 @@
             $type = $this->_defaultPriorityStyle;
         }
         
-        Zend_Wildfire_Plugin_FirePhp::getInstance()->send($event['message'], null, $type);
+        $message = $this->_formatter->format($event);
+        
+        if(array_key_exists('firebugLabel',$event)) {
+            Zend_Wildfire_Plugin_FirePhp::getInstance()->send($message, $event['firebugLabel'], $type);
+        } else {        
+            Zend_Wildfire_Plugin_FirePhp::getInstance()->send($message, null, $type);
+        }
     }
 }

