Index: /include/Zend/Controller/Dispatcher/Standard.php
===================================================================
--- /include/Zend/Controller/Dispatcher/Standard.php	(revision 5133)
+++ /include/Zend/Controller/Dispatcher/Standard.php	(working copy)
@@ -125,6 +125,9 @@
      */
     public function formatModuleName($unformatted)
     {
+        if ($this->_defaultModule == $unformatted) {
+            return $unformatted;
+        }
         return ucfirst($this->_formatName($unformatted));
     }
 
