Index: Zend/Http/Client/Adapter/Socket.php
===================================================================
--- Zend/Http/Client/Adapter/Socket.php	(revision 11917)
+++ Zend/Http/Client/Adapter/Socket.php	(working copy)
@@ -290,14 +290,14 @@
                 $response .= $chunk;
                 
                 // Break if the connection ended prematurely
-                if (feof($this->socket)) break;
+                if (strlen($chunk) == 0 || feof($this->socket)) break;
             }
                     
         // Fallback: just read the response until EOF
         } else {
             while (($buff = @fread($this->socket, 8192)) !== false) {
                 $response .= $buff;
-                if (feof($this->socket)) break;
+                if (strlen($buff) == 0 || feof($this->socket)) break;
             }
 
             $this->close();
