Index: Abstract.php
===================================================================
--- Abstract.php	(revision 23808)
+++ Abstract.php	(working copy)
@@ -63,7 +63,11 @@
     {
         $validatorName = get_class($this);
 
-        return $_SESSION['__ZF']['VALID'][$validatorName];
+        if (isset($_SESSION['__ZF']['VALID'][$validatorName])) {
+            return $_SESSION['__ZF']['VALID'][$validatorName];
+        }
+
+        return null;
     }
 
 }
