Index: /library/Zend/Paginator.php
===================================================================
--- /library/Zend/Paginator.php	(revision 11235)
+++ /library/Zend/Paginator.php	(working copy)
@@ -518,7 +518,7 @@
         
         $items = $this->_adapter->getItems($offset, $this->_itemCountPerPage);
         
-        if (!$items instanceof Iterator) {
+        if (!$items instanceof Traversable) {
             $items = new ArrayIterator($items);
         }
         
