Index: Application/Bootstrap/BootstrapAbstract.php
===================================================================
--- Application/Bootstrap/BootstrapAbstract.php	(revision 19391)
+++ Application/Bootstrap/BootstrapAbstract.php	(working copy)
@@ -117,9 +117,14 @@
         $options = array_change_key_case($options, CASE_LOWER);
         $this->_optionKeys = array_merge($this->_optionKeys, array_keys($options));
 
-        $methods = get_class_methods($this);
-        foreach ($methods as $key => $method) {
-            $methods[$key] = strtolower($method);
+        $reflection        = new ReflectionClass($this);
+        $reflectionMethods = $reflection->getMethods();
+        $methods           = array();
+
+        foreach ($reflectionMethods as $key => $method) {
+            if ($method->class == $reflection->getName()) {
+                $methods[$key] = strtolower($method->name);
+            }
         }
 
         if (array_key_exists('pluginpaths', $options)) {
