Index: library/Zend/Controller/Action/Helper/FlashMessenger.php
===================================================================
--- library/Zend/Controller/Action/Helper/FlashMessenger.php	(revision 24607)
+++ library/Zend/Controller/Action/Helper/FlashMessenger.php	(working copy)
@@ -141,6 +141,8 @@
         }
 
         self::$_session->{$this->_namespace}[] = $message;
+		
+		self::$_messageAdded = true;
 
         return $this;
     }
