Index: ActionStack.php
===================================================================
--- ActionStack.php	(revision 7835)
+++ ActionStack.php	(working copy)
@@ -92,6 +92,11 @@
 
         $request = $this->getRequest();
 
+        if ($request instanceof Zend_Controller_Request_Abstract === false){
+            require_once 'Zend/Controller/Action/Exception.php';
+            throw new Zend_Controller_Action_Exception('Request object not set yet');
+        }
+        
         $controller = (null === $controller) ? $request->getControllerName() : $controller;
         $module = (null === $module) ? $request->getModuleName() : $module;
 
