Index: Mysqli.php
===================================================================
--- Mysqli.php	(Revision 4474)
+++ Mysqli.php	(Arbeitskopie)
@@ -264,7 +264,7 @@
     protected function _beginTransaction()
     {
         $this->_connect();
-        $this->_connection->beginTransaction();
+        $this->_connection->autocommit(false);
     }
 
     /**
@@ -276,6 +276,7 @@
     {
         $this->_connect();
         $this->_connection->commit();
+        $this->_connection->autocommit(true);
     }
 
     /**
@@ -287,6 +288,7 @@
     {
         $this->_connect();
         $this->_connection->rollBack();
+        $this->_connection->autocommit(true);
     }
 
     /**
