--- /cygdrive/c/WebServers/home/zend.lo/library/Zend/Rest/Route.php	2009-09-01 22:59:57.218750000 +0400
+++ /cygdrive/c/WebServers/home/zend.lo/public/Route.php	2009-09-03 00:22:28.970003200 +0400
@@ -67,7 +67,13 @@
      * @var Zend_Controller_Front
      */
     protected $_front;
-
+    
+    /**
+     * Adds action names for excluding from assembling
+     * @var array
+     */
+	protected $_excludedActions = null;
+	
     /**
      * Constructor
      *
@@ -88,6 +94,29 @@
         $this->_front      = $front;
         $this->_dispatcher = $front->getDispatcher();
     }
+    
+    /**
+     * Sets a map with action names to exclude from assembling.
+     *
+     * @param array $actions Array with action names to exclude
+     */
+    public function setExcludedActions(array $actions)
+    {
+    	$this->_excludedActions = $actions;
+    }
+    
+    /**
+     * Gets a map with actions names to exclude from assembling.
+     *
+     * @return array
+     */
+    public function getExcludedActions()
+    {
+    	if(null === $this->_excludedActions) {
+    		return array();
+    	}
+    	return $this->_excludedActions;
+    }
 
     /**
      * Matches a user submitted request. Assigns and returns an array of variables
@@ -214,8 +243,8 @@
      * @return string Route path with user submitted parameters
      */
     public function assemble($data = array(), $reset = false, $encode = true)
-    {
-        if (!$this->_keysSet) {
+	{
+		if (!$this->_keysSet) {
             if (null === $this->_request) {
                 $this->_request = $this->_front->getRequest();
             }
@@ -245,6 +274,17 @@
 
         $controller = $params[$this->_controllerKey];
         unset($params[$this->_controllerKey]);
+        
+        if(isset($params[$this->_actionKey])) {
+        	$action = $params[$this->_actionKey];
+        	if($action === $this->_defaults[$this->_actionKey] || in_array($action, $this->getExcludedActions())) {
+	        	$action = '';
+	        } else {
+	        	$action = '/' . $params[$this->_actionKey];
+	        }
+        } else {
+        	$action = '';
+        }
 
         unset($params[$this->_actionKey]);
 
@@ -258,16 +298,14 @@
             $url .= '/' . $params['id'];
         }
 
-        if (!empty($url) || $controller !== $this->_defaults[$this->_controllerKey]) {
-            $url = '/' . $controller . $url;
-        }
-
+       	$url = '/' . $controller . $action . $url;
+        
         if (isset($module)) {
             $url = '/' . $module . $url;
         }
-
+        
         return ltrim($url, self::URI_DELIMITER);
-    }
+	}
 
     /**
      * Tells Rewrite Router which version this Route is
