Index: library/Zend/Form/Element/Radio.php
===================================================================
--- library/Zend/Form/Element/Radio.php	(revision 24746)
+++ library/Zend/Form/Element/Radio.php	(working copy)
@@ -53,8 +53,14 @@
             return $this;
         }
         parent::loadDefaultDecorators();
-        $this->addDecorator('Label', array('tag' => 'dt',
-                                           'disableFor' => true));
+
+        // Disable 'for' attribute
+        if (isset($this->_decorators['Label'])
+            && !isset($this->_decorators['Label']['options']['disableFor']))
+        {
+             $this->_decorators['Label']['options']['disableFor'] = true;
+        }
+
         return $this;
     }
 }
