Index: Standard.php
===================================================================
--- Standard.php	(revision 7891)
+++ Standard.php	(working copy)
@@ -430,4 +430,14 @@
 
         return $this->formatActionName($action);
     }
+    
+    public function getDefaultModule()
+    {
+        $defaultModule = parent::getDefaultModule();
+        if (!$this->isValidModule($defaultModule)){
+            require_once 'Zend/Controller/Dispatcher/Exception.php';
+            throw new Zend_Controller_Dispatcher_Exception("Default module '$defaultModule' does not exist");
+        }
+        return $defaultModule;
+    }
 }
