Index: Zend/Http/Client/Adapter/Socket.php
===================================================================
--- Zend/Http/Client/Adapter/Socket.php (révision 2764)
+++ Zend/Http/Client/Adapter/Socket.php (copie de travail)
@@ -110,4 +110,9 @@
                 throw new Zend_Http_Client_Adapter_Exception(
                     'Unable to Connect to ' . $host . ':' . $port . '. Error #' . $errno . ': ' . $errstr);
             }
-
+
+            // Set the connection timeout
+            if (!stream_set_timeout($this->socket, (int) $this->config['timeout'])) {
+                throw new Zend_Http_Client_Adapter_Exception('Unable to set the connection timeout');
+            }
+
