Index: FormSelectTest.php
===================================================================
--- FormSelectTest.php	(revision 21318)
+++ FormSelectTest.php	(working copy)
@@ -144,6 +144,24 @@
         $this->assertRegexp('/<select[^>]*?disabled/', $html, $html);
         $this->assertNotRegexp('/<option[^>]*?disabled="disabled"/', $html, $html);
     }
+    /**
+     * ZF-9329
+     */
+    public function testHaveValueInDisabledSelect()
+    {
+    	$html = $this->helper->formSelect(array(
+            'name'    => 'baz',
+            'options' => array(
+                'foo' => 'Foo',
+                'bar' => 'Bar'
+            ),
+            'attribs' => array(
+               'disable' => true
+            ),
+            'value' => 'foo',
+        ));
+        $this->assertRegexp('/<input[^>]*?type="hidden"[^>]*?value="foo"/', $html, $html);
+    }
 
     /**
      * ZF-2513
