Index: documentation/manual/en/module_specs/Zend_Version.xml
===================================================================
--- documentation/manual/en/module_specs/Zend_Version.xml	(revision 22824)
+++ documentation/manual/en/module_specs/Zend_Version.xml	(working copy)
@@ -28,6 +28,19 @@
 $cmp = Zend_Version::compareVersion('2.0.0');
 ]]></programlisting>
     </example>
+
+    <para>
+        The static method <methodname>Zend_Version::getLatest()</methodname> provides the version 
+        number of the last stable release available for download on the site 
+        <ulink linked="http://framework.zend.com/download/latest">Zend Framework</ulink>.
+    </para>
+
+    <example id="zend.version.latest.example">
+        <programlisting language="php"><![CDATA[
+// returns 1.11.0 (or a later version)
+echo Zend_Version::getLatest();
+]]></programlisting>
+    </example>
 </sect1>
 <!--
 vim:se ts=4 sw=4 et:
Index: library/Zend/Version.php
===================================================================
--- library/Zend/Version.php	(revision 22797)
+++ library/Zend/Version.php	(working copy)
@@ -35,6 +35,13 @@
     const VERSION = '1.11.0dev';
 
     /**
+     * The latest stable version Zend Framework available
+     * 
+     * @var string
+     */
+    protected static $_lastestVersion;
+
+    /**
      * Compare the specified Zend Framework version string $version
      * with the current Zend_Version::VERSION of Zend Framework.
      *
@@ -50,4 +57,25 @@
         $version = preg_replace('/(\d)pr(\d?)/', '$1a$2', $version);
         return version_compare($version, strtolower(self::VERSION));
     }
-}
+
+    /**
+     * Fetches the version of the latest stable release
+     * 
+     * @link http://framework.zend.com/download/latest
+     * @return string
+     */
+    public static function getLatest()
+    {
+        if (null === self::$_lastestVersion) {
+            self::$_lastestVersion = 'not available';
+
+            $handle = fopen('http://framework.zend.com/api/zf-version', 'r');        
+            if (false !== $handle) {
+                self::$_lastestVersion = stream_get_contents($handle);
+                fclose($handle);
+            }
+        }
+
+        return self::$_lastestVersion;
+    }
+}
\ No newline at end of file
Index: tests/Zend/VersionTest.php
===================================================================
--- tests/Zend/VersionTest.php	(revision 22824)
+++ tests/Zend/VersionTest.php	(working copy)
@@ -89,6 +89,18 @@
         }
     }
 
+    /**
+     * @group ZF-10363
+     */
+    public function testFetchLatestVersion()
+    {
+        $actual = Zend_Version::getLatest();
+        if ('not available' === $actual) {
+            $this->markIncomplete('http://framework.zend.com/ may be down');
+        }
+
+        $this->assertRegExp('/^[1-2](\.[0-9]+){2}/', $actual);
+    }
 }
 
 if (PHPUnit_MAIN_METHOD == "Zend_VersionTest::main") {
