Index: tests/Zend/Form/_files/locale/zf-9422.php
===================================================================
--- tests/Zend/Form/_files/locale/zf-9422.php	(revision 0)
+++ tests/Zend/Form/_files/locale/zf-9422.php	(revision 0)
@@ -0,0 +1,19 @@
+<?php
+
+class Test_Adapter_ZF9422 extends Zend_Translate_Adapter_Array
+{
+    public function __construct()
+    {
+        parent::__construct(array(), 'en');
+    }
+
+    public function getLocale()
+    {
+        return 'en';
+    }
+
+    public function translate($messageId, $locale=null)
+    {
+        return "** $messageId **";
+    }
+}
\ No newline at end of file
Index: tests/Zend/Form/Decorator/LabelTest.php
===================================================================
--- tests/Zend/Form/Decorator/LabelTest.php	(revision 21533)
+++ tests/Zend/Form/Decorator/LabelTest.php	(working copy)
@@ -311,6 +311,23 @@
         $tag = $this->decorator->getTag();
         $this->assertTrue( NULL === $tag, $tag );
     }
+
+    /**
+     * @group ZF-9422
+     */
+    public function testRetrievingLabelRetrievesLabelTranslatedOnlyOnce()
+    {
+        require_once 'Zend/Translate/Adapter/Array.php';
+        require_once dirname(__FILE__) . '/../_files/locale/zf-9422.php';
+
+        $element = new Zend_Form_Element('foo');
+        $element->setView($this->getView())
+                ->setLabel('My Label')
+                ->setTranslator(new Test_Adapter_ZF9422);
+        $this->decorator->setElement($element);
+        $label = $this->decorator->getLabel();
+        $this->assertEquals('** My Label **', $label);
+    }
 }
 
 // Call Zend_Form_Decorator_LabelTest::main() if this source file is executed directly.
