Index: library/Zend/Controller/Response/Abstract.php
===================================================================
--- library/Zend/Controller/Response/Abstract.php	(revision 21781)
+++ library/Zend/Controller/Response/Abstract.php	(working copy)
@@ -342,30 +342,15 @@
             return $this;
         }
 
-        $httpCodeSent = false;
-
         foreach ($this->_headersRaw as $header) {
-            if (!$httpCodeSent && $this->_httpResponseCode) {
-                header($header, true, $this->_httpResponseCode);
-                $httpCodeSent = true;
-            } else {
-                header($header);
-            }
+            header($header);
         }
 
         foreach ($this->_headers as $header) {
-            if (!$httpCodeSent && $this->_httpResponseCode) {
-                header($header['name'] . ': ' . $header['value'], $header['replace'], $this->_httpResponseCode);
-                $httpCodeSent = true;
-            } else {
-                header($header['name'] . ': ' . $header['value'], $header['replace']);
-            }
+            header($header['name'] . ': ' . $header['value'], $header['replace']);
         }
 
-        if (!$httpCodeSent) {
-            header('HTTP/1.1 ' . $this->_httpResponseCode);
-            $httpCodeSent = true;
-        }
+        header('HTTP/1.1 ' . $this->_httpResponseCode);
 
         return $this;
     }
