Index: library/Zend/View/Helper/Navigation/HelperAbstract.php
===================================================================
--- library/Zend/View/Helper/Navigation/HelperAbstract.php	(revision 24695)
+++ library/Zend/View/Helper/Navigation/HelperAbstract.php	(working copy)
@@ -669,12 +669,15 @@
         }
 
         // get attribs for anchor element
-        $attribs = array(
-            'id'     => $page->getId(),
-            'title'  => $title,
-            'class'  => $page->getClass(),
-            'href'   => $page->getHref(),
-            'target' => $page->getTarget()
+        $attribs = array_merge(
+            array(
+                'id'     => $page->getId(),
+                'title'  => $title,
+                'class'  => $page->getClass(),
+                'href'   => $page->getHref(),
+                'target' => $page->getTarget()
+            ),
+            $page->getCustomHtmlAttribs()
         );
 
         return '<a' . $this->_htmlAttribs($attribs) . '>'
Index: library/Zend/View/Helper/Navigation/Menu.php
===================================================================
--- library/Zend/View/Helper/Navigation/Menu.php	(revision 24695)
+++ library/Zend/View/Helper/Navigation/Menu.php	(working copy)
@@ -231,14 +231,17 @@
 
         // does page have a href?
         if ($href = $page->getHref()) {
-            $element = 'a';
-            $attribs['href'] = $href;
-            $attribs['target'] = $page->getTarget();
+            $element              = 'a';
+            $attribs['href']      = $href;
+            $attribs['target']    = $page->getTarget();
             $attribs['accesskey'] = $page->getAccessKey();
         } else {
             $element = 'span';
         }
 
+        // Add custom HTML attributes
+        $attribs = array_merge($attribs, $page->getCustomHtmlAttribs());
+
         return '<' . $element . $this->_htmlAttribs($attribs) . '>'
              . $this->view->escape($label)
              . '</' . $element . '>';
