Index: incubator/Zend/Http/Cookie.php
===================================================================
--- incubator/Zend/Http/Cookie.php	(revision 849)
+++ incubator/Zend/Http/Cookie.php	(revision 850)
@@ -280,8 +280,9 @@
                 continue;
             }
             
-            list($k, $v) = explode('=', $part);
-            if (isset($k) && isset($v))    {
+            $keyValue = explode('=', $part, 2);
+            if (count($keyValue) == 2) {
+                list($k, $v) = $keyValue;
                 switch ($k)    {
                     case 'expires':
                         $expires = strtotime($v);
