Index: Zend/XmlRpc/Value/Scalar.php
===================================================================
--- Zend/XmlRpc/Value/Scalar.php	(Revision 2901)
+++ Zend/XmlRpc/Value/Scalar.php	(Arbeitskopie)
@@ -44,10 +44,10 @@
     public function getAsXML()
     {
         if (!$this->_as_xml) {   // The XML code was not calculated yet
-            $dom   = new DOMDocument('1.0', 'ISO-8859-1');
+            $dom   = new DOMDocument('1.0', 'UTF-8');
             $value = $dom->appendChild($dom->createElement('value'));
             $type  = $value->appendChild($dom->createElement($this->_type));
-            $type->appendChild($dom->createTextNode($this->_value));
+            $type->appendChild($dom->createTextNode($this->getValue()));
 
             $this->_as_dom = $value;
             $this->_as_xml = $this->_stripXmlDeclaration($dom);
