Index: Element.php
===================================================================
--- Element.php	(revision 21858)
+++ Element.php	(working copy)
@@ -1333,6 +1333,9 @@
             $this->setValidators($validators);
         }
 
+        /** @see Zend_Validate_Abstract */
+        require_once 'Zend/Validate/Abstract.php';
+
         // Find the correct translator. Zend_Validate_Abstract::getDefaultTranslator()
         // will get either the static translator attached to Zend_Validate_Abstract
         // or the 'Zend_Translate' from Zend_Registry.
