Index: library/Zend/Db/Statement/Mysqli.php
===================================================================
--- library/Zend/Db/Statement/Mysqli.php	(Revision 7696)
+++ library/Zend/Db/Statement/Mysqli.php	(Arbeitskopie)
@@ -235,9 +235,13 @@
         // send $params as input parameters to the statement
         if ($params) {
             array_unshift($params, str_repeat('s', count($params)));
+            $stmt_params = array();
+            foreach ($params as $k => &$value) {
+                $stmt_params[$k] = &$value;
+            }
             call_user_func_array(
                 array($this->_stmt, 'bind_param'),
-                $params
+                $stmt_params
             );
         }
 
