--- standard/trunk/library/Zend/Translate/Adapter.php	2008-08-26 10:31:55.000000000 +0200
+++ patch/trunk/library/Zend/Translate/Adapter.php	2008-08-26 10:32:28.000000000 +0200
@@ -133,6 +133,7 @@
             foreach (new RecursiveIteratorIterator(
                      new RecursiveDirectoryIterator($data, RecursiveDirectoryIterator::KEY_AS_PATHNAME), 
                      RecursiveIteratorIterator::SELF_FIRST) as $file => $info) {
+                if (preg_match('~/\.svn/~', $info->getPathname())) { continue; }
                 if ($info->isDir()) {
 
                     $directory = $info->getPath();
