Index: Zend/Http/Client/Adapter/Curl.php
===================================================================
--- Zend/Http/Client/Adapter/Curl.php (revision 576)
+++ Zend/Http/Client/Adapter/Curl.php (revision 873)
@@ -195,7 +195,4 @@
             curl_setopt($this->_curl, CURLOPT_PORT, intval($port));
         }
-
-        // Set timeout
-        curl_setopt($this->_curl, CURLOPT_CONNECTTIMEOUT, $this->_config['timeout']);
 
         // Set Max redirects
@@ -354,4 +351,7 @@
         }
 
+        // Set timeout
+        curl_setopt($this->_curl, CURLOPT_CONNECTTIMEOUT, $this->_config['timeout']);
+
         // send the request
         $this->_response = curl_exec($this->_curl);
