Index: Zend/Soap/AutoDiscover.php
===================================================================
--- Zend/Soap/AutoDiscover.php	(Revision 16086)
+++ Zend/Soap/AutoDiscover.php	(Arbeitskopie)
@@ -371,6 +371,10 @@
         if($prototype->getReturnType() == "void") {
             $isOneWayMessage = true;
         }
+        $docBlock = $function->getDocComment();
+        if(strpos($docBlock, '@throw')) {
+            $isOneWayMessage = false;
+        }
 
         if($isOneWayMessage == false) {
             // Add the output message (return value)
