Index: library/Zend/Tool/Framework/Client/Console/HelpSystem.php
===================================================================
--- library/Zend/Tool/Framework/Client/Console/HelpSystem.php	(revision 16050)
+++ library/Zend/Tool/Framework/Client/Console/HelpSystem.php	(working copy)
@@ -161,6 +161,8 @@
     protected function _respondWithHeader()
     {
         $this->_response->appendContent('Zend Framework', array('color' => array('hiWhite'), 'separator' => false));
+        require_once 'Zend/Loader/Autoloader.php'; Zend_Loader_Autoloader::getInstance();
+        require_once 'Zend/Version.php';
         $this->_response->appendContent(' Command Line Console Tool v' . Zend_Version::VERSION . '');
         return $this;
     }
