Index: library/Zend/Validate/Date.php
===================================================================
--- library/Zend/Validate/Date.php	(revision 17846)
+++ library/Zend/Validate/Date.php	(working copy)
@@ -46,7 +46,7 @@
         self::INVALID        => "Invalid type given, value should be string, integer, array or Zend_Date",
         self::NOT_YYYY_MM_DD => "'%value%' is not of the format YYYY-MM-DD",
         self::INVALID_DATE   => "'%value%' does not appear to be a valid date",
-        self::FALSEFORMAT    => "'%value%' does not fit given date format"
+        self::FALSEFORMAT    => "'%value%' does not fit given date format %format%"
     );
 
     /**
@@ -64,6 +64,13 @@
     protected $_locale;
 
     /**
+     * @var array
+     */
+    protected $_messageVariables = array(
+        'format'  => '_format'
+    );
+
+    /**
      * Sets validator options
      *
      * @param  string             $format OPTIONAL
