Index: tests/Zend/Soap/ClientTest.php
===================================================================
--- tests/Zend/Soap/ClientTest.php	(revision 24554)
+++ tests/Zend/Soap/ClientTest.php	(working copy)
@@ -517,6 +517,22 @@
 
         $this->assertSame($clientMock, $soap->getSoapClient());
     }
+    
+    /**
+     * @group ZF-11618
+     */
+    public function testReturnsBooleanValuesCorrectlyWhenPartOfAComplexClass()
+    {
+        $client = new Zend_Soap_Client(TESTS_ZEND_SOAP_AUTODISCOVER_ONLINE_SERVER_BASEURI . "/ZF11618.php?wsdl");
+
+        $bar = $client->isFoo('bar');
+        $this->assertEquals('stdClass', get_class($bar));
+        $this->assertFalse($bar->isFoo);
+        
+        $foo = $client->isFoo('foo');
+        $this->assertEquals('stdClass', get_class($foo));
+        $this->assertTrue($foo->isFoo);
+    }
 }
 
 
Index: tests/Zend/Soap/_files/fulltests/ZF11618.php
===================================================================
--- tests/Zend/Soap/_files/fulltests/ZF11618.php	(revision 0)
+++ tests/Zend/Soap/_files/fulltests/ZF11618.php	(revision 0)
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Zend Framework
+ *
+ * LICENSE
+ *
+ * This source file is subject to the new BSD license that is bundled
+ * with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://framework.zend.com/license/new-bsd
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@zend.com so we can send you a copy immediately.
+ *
+ * @category   Zend
+ * @package    Zend_Soap
+ * @subpackage UnitTests
+ * @copyright  Copyright (c) 2005-2011 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license    http://framework.zend.com/license/new-bsd     New BSD License
+ * @version    $Id$
+ */
+
+require_once "Zend/Soap/AutoDiscover.php";
+require_once "Zend/Soap/Server.php";
+require_once "Zend/Soap/Wsdl/Strategy/ArrayOfTypeSequence.php";
+
+
+/**
+ * Boolean Return Test Class Container for ZF-11411
+ * @see ZF-11618
+ */
+class ZF11618_BooleanTestClass 
+{
+    /**
+     * Is Foo?
+     * @param string $string
+     * @return ZF11618_BooleanTestClass_ReturnTest
+     */
+    public function isFoo($string)
+    {
+        $return = new ZF11618_BooleanTestClass_ReturnTest();
+        $return->isFoo = ( $string == 'foo' );
+        return $return;
+    }
+}
+
+/**
+ * Boolean Return Test Class for ZF-11411
+ * @see ZF-11618
+ */
+class ZF11618_BooleanTestClass_ReturnTest
+{
+    /**
+     * @var boolean
+     */
+    public $isFoo = false;
+}
+
+if(isset($_GET['wsdl'])) {
+    $server = new Zend_Soap_AutoDiscover('Zend_Soap_Wsdl_Strategy_ArrayOfTypeSequence');
+    $server->setOperationBodyStyle(array('use' => 'literal'));
+    $server->setBindingStyle(array('style' => 'rpc'));
+} else {
+    $uri = "http://".$_SERVER['HTTP_HOST']."/".$_SERVER['PHP_SELF']."?wsdl";
+    $server = new Zend_Soap_Server($uri);
+}
+$server->setClass('ZF11618_BooleanTestClass');
+$server->handle();
+

Property changes on: tests/Zend/Soap/_files/fulltests/ZF11618.php
___________________________________________________________________
Added: svn:keywords
   + Id

