--- tests/Zend/Config/IniTest.php   (revision 23872)
+++ tests/Zend/Config/IniTest.php   (working copy)
@@ -308,5 +308,12 @@
 
     }
 
-
+   /** @group ZF-9786 */
+   public function testAllowWhiteSpaceBetweenPieces()
+   {
+       $filename = dirname(__FILE__) . '/_files/zf9786.ini';
+       $config = new Zend_Config_Ini($filename);
+       $this->assertEquals('barbapapa', $config->elements->foo->type);
+       $this->assertEquals('barbazoo', $config->elements->bar->type);
+   }
 }
