Index: library/Zend/Loader/Autoloader/Interface.php
===================================================================
--- library/Zend/Loader/Autoloader/Interface.php	(revision 22206)
+++ library/Zend/Loader/Autoloader/Interface.php	(working copy)
@@ -30,5 +30,11 @@
  */
 interface Zend_Loader_Autoloader_Interface
 {
+    /**
+     * Autoload a class
+     *
+     * @param string $class
+     * @return bool
+     */
     public function autoload($class);
 }
