Index: trunk/library/Zend/Feed/Reader.php
===================================================================
--- trunk/library/Zend/Feed/Reader.php	(revision 1359)
+++ trunk/library/Zend/Feed/Reader.php	(working copy)
@@ -427,7 +427,7 @@
         } elseif(is_string($feed) && !empty($feed)) {
             @ini_set('track_errors', 1);
             $dom = new DOMDocument;
-            $status = @$doc->loadXML($string);
+            $status = $dom->loadXML($feed);
             @ini_restore('track_errors');
             if (!$status) {
                 if (!isset($php_errormsg)) {
