Index: tests/Zend/Measure/TemperatureTest.php
===================================================================
--- tests/Zend/Measure/TemperatureTest.php	(revision 6038)
+++ tests/Zend/Measure/TemperatureTest.php	(working copy)
@@ -395,5 +395,16 @@
         $unit  = $value->getConversionList();
         $this->assertTrue(is_array($unit), 'Array expected');
     }
+    
+    /**
+     * test KELVIN to FAHRENHEIT conversion value
+     * expected specific value
+     */
+	public function testKelvinToFahrenheitEquation()
+	{
+		$value = new Zend_Measure_Temperature(10, Zend_Measure_Temperature::KELVIN);
+		$value->setType(Zend_Measure_Temperature::FAHRENHEIT);
+		$this->assertEquals(-441.67, $value->getValue());
+	}
 
 }
Index: library/Zend/Measure/Temperature.php
===================================================================
--- library/Zend/Measure/Temperature.php	(revision 6038)
+++ library/Zend/Measure/Temperature.php	(working copy)
@@ -48,7 +48,7 @@
 
     protected $_UNITS = array(
         'CELSIUS'    => array(array('' => 1, '+' => 273.15),'°C'),
-        'FAHRENHEIT' => array(array('' => 1, '+' => 459.67, '/' => 1.8),'°F'),
+        'FAHRENHEIT' => array(array('' => 1, '/' => 1.8, '+' => 459.67),'°F'),
         'RANKINE'    => array(array('' => 1, '/' => 1.8),'°R'),
         'REAUMUR'    => array(array('' => 1, '*' => 1.25, '+' => 273.15),'°r'),
         'KELVIN'     => array(1,'°K'),
