Index: tests/Zend/Form/Decorator/ViewScriptTest.php
===================================================================
--- tests/Zend/Form/Decorator/ViewScriptTest.php	(revision 24878)
+++ tests/Zend/Form/Decorator/ViewScriptTest.php	(working copy)
@@ -232,7 +232,7 @@
             ->setElement($this->getElement());
 
         $expected = 'Foo:'
-                  . "\n"
+                  . PHP_EOL
                   . '<input type="text" name="foo" id="foo" value="">';
 
         $this->assertSame($expected, $this->decorator->render('Foo:'));