Index: tests/Zend/Db/Statement/MysqliTest.php
===================================================================
--- tests/Zend/Db/Statement/MysqliTest.php	(revision 13376)
+++ tests/Zend/Db/Statement/MysqliTest.php	(working copy)
@@ -65,7 +65,7 @@
         } catch (Zend_Exception $e) {
             $this->assertType('Zend_Db_Statement_Exception', $e,
                 'Expecting object of type Zend_Db_Statement_Exception, got '.get_class($e));
-            $this->assertEquals("Invalid bind-variable position ':id'", $e->getMessage());
+            $this->assertEquals("Invalid bind-variable name ':id'", $e->getMessage());
         }
     }
 
@@ -88,7 +88,7 @@
         } catch (Zend_Exception $e) {
             $this->assertType('Zend_Db_Statement_Exception', $e,
                 'Expecting object of type Zend_Db_Statement_Exception, got '.get_class($e));
-            $this->assertEquals("Invalid bind-variable position ':id'", $e->getMessage());
+            $this->assertEquals("Invalid bind-variable name ':id'", $e->getMessage());
         }
     }
 
