Index: tests/Zend/Form/Element/MultiselectTest.php
===================================================================
--- tests/Zend/Form/Element/MultiselectTest.php (revision 15725)
+++ tests/Zend/Form/Element/MultiselectTest.php (working copy)
@@ -290,6 +290,20 @@
     }

     /**
+     * @group ZF-5937
+     */
+    public function testAddMultiOptionShouldWorkAfterTranslatorIsDisabled()
+    {
+        $options = array(
+            'foovalue' => 'Foo',
+        );
+        $this->element->setDisableTranslator(true)
+                      ->addMultiOptions($options);
+        $test = $this->element->getMultiOption('foovalue');
+        $this->assertEquals($options['foovalue'], $test);
+    }
+
+    /**
      * Used by test methods susceptible to ZF-2794, marks a test as incomplete
      *
      * @link   http://framework.zend.com/issues/browse/ZF-2794
Index: library/Zend/Form/Element/Multi.php
===================================================================
--- library/Zend/Form/Element/Multi.php (revision 15725)
+++ library/Zend/Form/Element/Multi.php (working copy)
@@ -278,7 +278,7 @@
     protected function _translateOption($option, $value)
     {
         if ($this->translatorIsDisabled()) {
-            return true;
+            return false;
         }

         if (!isset($this->_translated[$option]) && !empty($value)) {

