Index: library/Zend/Test/PHPUnit/Constraint/ResponseHeader.php
===================================================================
--- library/Zend/Test/PHPUnit/Constraint/ResponseHeader.php	(version 23692)
+++ library/Zend/Test/PHPUnit/Constraint/ResponseHeader.php	(working copy)
@@ -83,6 +83,12 @@
     protected $_negate            = false;
 
     /**
+     *
+     * @var int  against which to compare response code
+     */
+    protected $_testedCode = null;
+	
+    /**
      * Constructor; setup constraint state
      *
      * @return void
@@ -192,11 +198,11 @@
         require_once 'Zend/Test/PHPUnit/Constraint/Exception.php';
         switch ($this->_assertType) {
             case self::ASSERT_RESPONSE_CODE:
-                $failure = 'Failed asserting response code "%s"';
+                $failure = 'Failed asserting that response code "%s" IS "%s"';
                 if ($this->_negate) {
-                    $failure = 'Failed asserting response code IS NOT "%s"';
+                    $failure = 'Failed asserting that response code "%s" IS NOT "%s"';
                 }
-                $failure = sprintf($failure, $this->_code);
+                $failure = sprintf($failure, $this->_testedCode, $this->_code);
                 break;
             case self::ASSERT_HEADER:
                 $failure = 'Failed asserting response header "%s" found';
@@ -249,8 +255,8 @@
      */
     protected function _code(Zend_Controller_Response_Abstract $response, $code)
     {
-        $test = $this->_getCode($response);
-        return ($test == $code);
+        $this->_testedCode = $this->_getCode($response);
+        return ($this->_testedCode == $code);
     }
 
     /**
@@ -262,8 +268,8 @@
      */
     protected function _notCode(Zend_Controller_Response_Abstract $response, $code)
     {
-        $test = $this->_getCode($response);
-        return ($test != $code);
+        $this->_testedCode = $this->_getCode($response);
+        return ($this->_testedCode != $code);
     }
 
     /**
