Index: tests/Zend/Db/Profiler/FirebugTest.php
===================================================================
--- tests/Zend/Db/Profiler/FirebugTest.php	(revision 19898)
+++ tests/Zend/Db/Profiler/FirebugTest.php	(working copy)
@@ -204,6 +204,29 @@
         $this->assertFalse($messages);
     }
 
+    /**
+     * @group ZF-8563
+     */
+    public function testCustomLabelAfterConstructor()
+    {
+        $channel = Zend_Wildfire_Channel_HttpHeaders::getInstance();
+        $protocol = $channel->getProtocol(Zend_Wildfire_Plugin_FirePhp::PROTOCOL_URI);
+
+        $this->_profiler = new Zend_Db_Profiler_Firebug('Label 1');
+        $this->_profiler->setEnabled(true);
+        $this->_profiler->setLabel('Label 2');
+        $this->_db->setProfiler($this->_profiler);
+        $this->_db->insert('foo', array('id'=>1,'col1'=>'original'));
+
+        Zend_Wildfire_Channel_HttpHeaders::getInstance()->flush();
+
+        $messages = $protocol->getMessages();
+
+        $this->assertEquals(substr($messages[Zend_Wildfire_Plugin_FirePhp::STRUCTURE_URI_FIREBUGCONSOLE]
+                                            [Zend_Wildfire_Plugin_FirePhp::PLUGIN_URI][0],0,38),
+                            '[{"Type":"TABLE","Label":"Label 2 (1 @');
+    }
+
 }
 
 
Index: library/Zend/Db/Profiler/Firebug.php
===================================================================
--- library/Zend/Db/Profiler/Firebug.php	(revision 19898)
+++ library/Zend/Db/Profiler/Firebug.php	(working copy)
@@ -158,4 +158,26 @@
                                                     (string)round($this->_totalElapsedTime,5)),
                                               $this->_label_template));
     }
+
+    /**
+     * Set the label
+     *
+     * @param string $label
+     * @return void
+     */
+    public function setLabel($label)
+    {
+        $this->_label = $label;
+    }
+
+    /**
+     * Get the label
+     *
+     * @return string
+     */
+    public function getLabel()
+    {
+        return $this->_label;
+    }
+    
 }
