Index: Partial.php
===================================================================
--- Partial.php	(revision 23772)
+++ Partial.php	(working copy)
@@ -40,6 +40,20 @@
     protected $_objectKey;
 
     /**
+     * Register view base path
+     *
+     * @param  Zend_View_Interface $view
+     * @param  string $module The module in which the partial resides
+     * @param  string $moduleDir
+     * @return Zend_View_Helper_Partial
+     */
+    protected function _viewAddBasePath(Zend_View_Interface $view, $module, $moduleDir)
+    {
+        $view->addBasePath($moduleDir . '/views');
+        return $this;
+    }
+
+    /**
      * Renders a template fragment within a variable scope distinct from the
      * calling View object.
      *
@@ -80,8 +94,7 @@
                 $e->setView($this->view);
                 throw $e;
             }
-            $viewsDir = dirname($moduleDir) . '/views';
-            $view->addBasePath($viewsDir);
+            $this->_viewAddBasePath($view, $module, $moduleDir);
         } elseif ((null == $model) && (null !== $module)
             && (is_array($module) || is_object($module)))
         {
