Index: Zend/Db/Select.php
===================================================================
--- Zend/Db/Select.php	(revision 847)
+++ Zend/Db/Select.php	(working copy)
@@ -288,9 +288,7 @@
      *
      * If a value is passed as the second param, it will be quoted
      * and replaced into the condition wherever a question-mark
-     * appears.
-     *
-     * Array values are quoted and comma-separated.
+     * appears. Array values are quoted and comma-separated.
      *
      * <code>
      * // simplest but non-secure
@@ -296,12 +294,20 @@
      * // simplest but non-secure
      * $select->where("id = $id");
      *
-     * // secure
+     * // secure (ID is quoted but matched anyway)
      * $select->where('id = ?', $id);
      *
-     * // equivalent security with named binding
+     * // alternatively, with named binding
      * $select->where('id = :id');
-     * $select->bind('id', $id);
+     * </code>
+     * 
+     * Note that it is more correct to use named bindings in your
+     * queries for values other than strings. When you use named
+     * bindings, don't forget to pass the values when actually
+     * making a query:
+     * 
+     * <code>
+     * $db->fetchAll($select, array('id' => 5));
      * </code>
      *
      * @param string $cond The WHERE condition.
@@ -380,21 +386,7 @@
      *
      * If a value is passed as the second param, it will be quoted
      * and replaced into the condition wherever a question-mark
-     * appears.
-     *
-     * Array values are quoted and comma-separated.
-     *
-     * <code>
-     * // simplest but non-secure
-     * $select->having("COUNT(id) = $count");
-     *
-     * // secure
-     * $select->having('COUNT(id) = ?', $count);
-     *
-     * // equivalent security with named binding
-     * $select->having('COUNT(id) = :count');
-     * $select->bind('count', $count);
-     * </code>
+     * appears. See {@link where()} for an example
      *
      * @param string $cond The HAVING condition.
      * @param string $val A single value to quote into the condition.
