diff --git a/ZendX/Application/Resource/Jquery.php b/ZendX/Application/Resource/Jquery.php
index d4ebc23..2b52988 100644
--- a/ZendX/Application/Resource/Jquery.php
+++ b/ZendX/Application/Resource/Jquery.php
@@ -155,13 +155,14 @@ class ZendX_Application_Resource_Jquery
             }
         }
 
-        if ((isset($key['uienable']) && (bool) $key['uienable']) 
-            || (isset($key['ui_enable']) && (bool) $key['ui_enable']) 
-            || (!isset($key['ui_enable']) && !isset($key['uienable'])))
-        {
+        if ((isset($options['uienable']) && (bool) $options['uienable'])
+            || (isset($options['ui_enable']) && (bool) $options['ui_enable'])
+            || (isset($options['ui-enable']) && (bool) $options['ui-enable'])) {
             $this->_view->JQuery()->uiEnable();
+        } elseif (isset($options['enable']) && (bool) $options['enable']) {
+            $this->_view->JQuery()->enable();
         } else {
-            $this->_view->JQuery()->uiDisable();
+            $this->_view->jQuery()->disable();
         }
     }
 }
