Index: PluginLoaderTest.php
===================================================================
--- PluginLoaderTest.php	(revision 13952)
+++ PluginLoaderTest.php	(working copy)
@@ -402,6 +402,29 @@
         $cache = file_get_contents($cacheFile);
         $this->assertContains('CacheTest.php', $cache);
     }
+    
+     /**
+     * @group ZF-5625
+     */
+    public function testPluginLoaderShouldAppendPhpOpeningTagToTheIncludeCacheIfNotPresentCauseFileEmptied()
+    {
+        $cacheFile = $this->_includeCache;
+        Zend_Loader_PluginLoader::setIncludeFileCache($cacheFile);
+        $loader = new Zend_Loader_PluginLoader(array());
+        $loader->addPrefixPath('Zend_View_Helper', $this->libPath . '/Zend/View/Helper');
+        $loader->addPrefixPath('ZfTest', dirname(__FILE__) . '/_files/ZfTest');
+        try {
+            $className = $loader->load('CacheTest');
+        } catch (Exception $e) {
+            $paths = $loader->getPaths();
+            $this->fail(sprintf("Failed loading helper; paths: %s", var_export($paths, 1)));
+        }
+        $this->assertTrue(file_exists($cacheFile));
+        file_put_contents($cacheFile,'');
+        $className = $loader->load('CacheTest');
+        $cache = file_get_contents($cacheFile);
+        $this->assertContains('<?php', $cache);
+    }
 
     /**
      * @group ZF-5208
