Index: ../library/Zend/Application.php
===================================================================
--- ../library/Zend/Application.php	(revision 16923)
+++ ../library/Zend/Application.php	(working copy)
@@ -116,13 +116,13 @@
      */
     public function setOptions(array $options)
     {
-        $options = array_change_key_case($options, CASE_LOWER);
-
         if (!empty($options['config'])) {
             $options = $this->mergeOptions($options, $this->_loadConfig($options['config']));
         }
         
         $this->_options = $options;
+
+		$options = array_change_key_case($options, CASE_LOWER);
         
         if (!empty($options['phpsettings'])) {
             $this->setPhpSettings($options['phpsettings']);
