Index: tests/Zend/Locale/FormatTest.php
===================================================================
--- tests/Zend/Locale/FormatTest.php	(revision 24425)
+++ tests/Zend/Locale/FormatTest.php	(working copy)
@@ -1093,4 +1093,22 @@
         $this->assertEquals('4,00', Zend_Locale_Format::toNumber(3.999, $options));
         $this->assertEquals('4,00', Zend_Locale_Format::toNumber(4, $options));
     }
+    
+    /**
+     * @ZF-10358
+     */
+    public function testYearAfterTimeZone()
+    {
+        $dateInfos = Zend_Locale_Format::getDate(
+        	'Tue, Sep 06 21:23:00 +0000 2011',
+            array('date_format' => 'EEE, MMM d HH:mm:ss ZZZZ yyyy', 'format_type' => 'iso')
+        );
+        
+        $this->assertEquals('06', $dateInfos['day']);
+        $this->assertEquals('09', $dateInfos['month']);
+        $this->assertEquals('2011', $dateInfos['year']);
+        $this->assertEquals('21', $dateInfos['hour']);
+        $this->assertEquals('23', $dateInfos['minute']);
+        $this->assertEquals('00', $dateInfos['second']);
+    }
 }
Index: library/Zend/Locale/Format.php
===================================================================
--- library/Zend/Locale/Format.php	(revision 24425)
+++ library/Zend/Locale/Format.php	(working copy)
@@ -786,6 +786,7 @@
         $hour  = iconv_strpos($format, 'H');
         $min   = iconv_strpos($format, 'm');
         $sec   = iconv_strpos($format, 's');
+        $gmtDiff = iconv_strpos($format, 'Z');
         $am    = null;
         if ($hour === false) {
             $hour = iconv_strpos($format, 'h');
@@ -848,6 +849,11 @@
             require_once 'Zend/Locale/Exception.php';
             throw new Zend_Locale_Exception("Unknown date format, neither date nor time in '" . $format . "' found");
         }
+        
+        if ($gmtDiff !== false) {
+            $parse[$gmtDiff] = 'Z';
+        }
+        
         ksort($parse);
 
         // get daytime
@@ -948,6 +954,9 @@
                     }
                     ++$cnt;
                     break;
+                case 'Z':
+                    ++$cnt;
+                    break;
             }
         }
 
