diff --git a/library/Zend/Loader/PluginLoader.php b/library/Zend/Loader/PluginLoader.php
index f74ad9e..4013ca6 100644
--- a/library/Zend/Loader/PluginLoader.php
+++ b/library/Zend/Loader/PluginLoader.php
@@ -372,7 +372,11 @@ class Zend_Loader_PluginLoader implements Zend_Loader_PluginLoader_Interface
 
         $registry  = array_reverse($registry, true);
         $found     = false;
-        $classFile = str_replace('_', DIRECTORY_SEPARATOR, $name) . '.php';
+        if (false !== strpos($name, '\\')) {
+            $classFile = str_replace('\\', DIRECTORY_SEPARATOR, $name) . '.php';
+        } else {
+            $classFile = str_replace('_', DIRECTORY_SEPARATOR, $name) . '.php';
+        }
         $incFile   = self::getIncludeFileCache();
         foreach ($registry as $prefix => $paths) {
             $className = $prefix . $name;
