Index: tests/Zend/View/Helper/_files/modules/default/views/scripts/grandParentPartialLoop.phtml
===================================================================
--- tests/Zend/View/Helper/_files/modules/default/views/scripts/grandParentPartialLoop.phtml	(wersja 0)
+++ tests/Zend/View/Helper/_files/modules/default/views/scripts/grandParentPartialLoop.phtml	(wersja 0)
@@ -0,0 +1,4 @@
+<?= "Grandparent pointer at {$this->partialCounter}" ?>
+
+<?= $this->partialLoop('parentPartialLoop.phtml', $this->children) ?>
+
Index: tests/Zend/View/Helper/_files/modules/default/views/scripts/parentPartialLoop.phtml
===================================================================
--- tests/Zend/View/Helper/_files/modules/default/views/scripts/parentPartialLoop.phtml	(wersja 0)
+++ tests/Zend/View/Helper/_files/modules/default/views/scripts/parentPartialLoop.phtml	(wersja 0)
@@ -0,0 +1,4 @@
+<?= "Parent pointer at {$this->partialCounter}" ?>
+
+<?= $this->partialLoop('partialLoopCounter.phtml', $this->children) ?>
+
Index: tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCounter.phtml
===================================================================
--- tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCounter.phtml	(wersja 0)
+++ tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCounter.phtml	(wersja 0)
@@ -0,0 +1,2 @@
+This is an iteration: <?php echo $this->message ?>, pointer at <?php echo $this->partialCounter ?>
+
Index: tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml
===================================================================
--- tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml	(wersja 15802)
+++ tests/Zend/View/Helper/_files/modules/default/views/scripts/partialLoopCouter.phtml	(kopia robocza)
@@ -1,2 +0,0 @@
-This is an iteration: <?php echo $this->message ?>, pointer at <?php echo $this->partialCounter ?>
-
Index: tests/Zend/View/Helper/PartialLoopTest.php
===================================================================
--- tests/Zend/View/Helper/PartialLoopTest.php	(wersja 15802)
+++ tests/Zend/View/Helper/PartialLoopTest.php	(kopia robocza)
@@ -316,7 +316,7 @@
         ));
         $this->helper->setView($view);
 
-        $result = $this->helper->partialLoop('partialLoopCouter.phtml', $data);
+        $result = $this->helper->partialLoop('partialLoopCounter.phtml', $data);
         foreach ($data as $key=>$item) {
             $string = 'This is an iteration: ' . $item['message'] . ', pointer at ' . ($key+1);
             $this->assertContains($string, $result);
@@ -341,18 +341,52 @@
         ));
         $this->helper->setView($view);
 
-        $result = $this->helper->partialLoop('partialLoopCouter.phtml', $data);
+        $result = $this->helper->partialLoop('partialLoopCounter.phtml', $data);
         foreach ($data as $key=>$item) {
             $string = 'This is an iteration: ' . $item['message'] . ', pointer at ' . ($key+1);
             $this->assertContains($string, $result);
         }
 
-        $result = $this->helper->partialLoop('partialLoopCouter.phtml', $data);
+        $result = $this->helper->partialLoop('partialLoopCounter.phtml', $data);
         foreach ($data as $key=>$item) {
             $string = 'This is an iteration: ' . $item['message'] . ', pointer at ' . ($key+1);
             $this->assertContains($string, $result);
         }
     }
+
+    public function testKeepParentPartialCounterWhenRenderingNestedPartialLoop()
+    {
+        $child = array('message' => 'foo');
+        $children = array($child, $child);
+
+        $parent = array('children' => $children);
+        $parents = array($parent, $parent);
+
+        $grandParent = array('children' => $parents);
+        $grandParents = array($grandParent, $grandParent);
+
+        $view = new Zend_View(array(
+            'scriptPath' => $this->basePath . '/default/views/scripts'
+        ));
+        $this->helper->setView($view);
+
+        $result = $this->helper->partialLoop('grandParentPartialLoop.phtml', $grandParents);
+
+        foreach ($grandParents as $grandParentKey => $grandParent) {
+            $string = 'Grandparent pointer at ' . ($grandParentKey + 1);
+            $this->assertContains($string, $result);
+
+            foreach ($grandParent['children'] as $parentKey => $parent) {
+                $string = 'Parent pointer at ' . ($parentKey + 1);
+                $this->assertContains($string, $result);
+
+                foreach ($parent['children'] as $childKey => $child) {
+                    $string = 'This is an iteration: ' . $child['message'] . ', pointer at ' . ($childKey + 1);
+                    $this->assertContains($string, $result);
+                }
+            }
+        }
+    }
 }
 
 class Zend_View_Helper_PartialLoop_IteratorTest implements Iterator
@@ -515,3 +549,4 @@
 if (PHPUnit_MAIN_METHOD == "Zend_View_Helper_PartialLoopTest::main") {
     Zend_View_Helper_PartialLoopTest::main();
 }
+
Index: library/Zend/View/Helper/PartialLoop.php
===================================================================
--- library/Zend/View/Helper/PartialLoop.php	(wersja 15802)
+++ library/Zend/View/Helper/PartialLoop.php	(kopia robocza)
@@ -82,6 +82,7 @@
         }
 
         $content = '';
+        $oldPartialCounter = $this->partialCounter; // for nested partial loops
         // reset the counter if it's call again
         $this->partialCounter = 0;
         foreach ($model as $item) {
@@ -91,6 +92,9 @@
             $content .= $this->partial($name, $module, $item);
         }
 
+        $this->partialCounter = $oldPartialCounter;
+
         return $content;
     }
 }
+
