Index: Paginator.php
===================================================================
--- Paginator.php	(revision 10296)
+++ Paginator.php	(working copy)
@@ -732,6 +732,7 @@
         // Item numbers
         if ($this->getCurrentItems() !== null) {
             $pages->currentItemCount = $this->getCurrentItemCount();
+			$pages->itemCountPerPage = $this->getItemCountPerPage();
             $pages->totalItemCount   = $this->_adapter->count();
             $pages->firstItemNumber  = (($this->_currentPageNumber - 1) * $this->_itemCountPerPage) + 1;
             $pages->lastItemNumber   = $pages->firstItemNumber + $pages->currentItemCount - 1;
