Index: library/Zend/Db/Adapter/Abstract.php
===================================================================
--- library/Zend/Db/Adapter/Abstract.php	(revision 21463)
+++ library/Zend/Db/Adapter/Abstract.php	(working copy)
@@ -521,7 +521,7 @@
      * @param array $bind Column-value pairs.
      * @return int The number of affected rows.
      */
-    public function insert($table, array $bind)
+    public function insert($table, array $bind, $delayed = false )
     {
         // extract and quote col names from the array keys
         $cols = array();
@@ -551,7 +551,7 @@
         }
 
         // build the statement
-        $sql = "INSERT INTO "
+        $sql = "INSERT " . ( $delayed ? '' : ' DELAYED ' ) . " INTO "
              . $this->quoteIdentifier($table, true)
              . ' (' . implode(', ', $cols) . ') '
              . 'VALUES (' . implode(', ', $vals) . ')';
