Index: Zend/Http/Client.php
===================================================================
--- Zend/Http/Client.php	(revision 2596)
+++ Zend/Http/Client.php	(working copy)
@@ -688,9 +688,15 @@
             // Clone the URI and add the additional GET parameters to it
             $uri = clone $this->uri;
             $uri_params = array();
-            parse_str($uri->getQuery(), $uri_params);
-            $uri->setQuery(array_merge($uri_params, $this->paramsGet));
-            
+            $query = $uri->getQuery();
+            if (!empty($this->paramsGet)) {
+                if (!empty($query)) {
+                    $query .= '&';
+                }
+                $query .= implode('&', $this->paramsGet);
+            }
+            $uri->setQuery($query);
+
             $body = $this->prepare_body();
             $headers = $this->prepare_headers();
             
