diff --git a/library/Zend/EventManager/Event.php b/library/Zend/EventManager/Event.php
index a35e480..954be21 100644
--- a/library/Zend/EventManager/Event.php
+++ b/library/Zend/EventManager/Event.php
@@ -114,8 +114,7 @@ class Event implements EventDescription
     {
         if (!is_array($params) && !is_object($params)) {
             throw new Exception\InvalidArgumentException(sprintf(
-                'Event parameters must be an array or object; received "%s"',
-                (is_object($params) ? get_class($params) : gettype($params))
+                'Event parameters must be an array or object; received "%s"', gettype($params)
             ));
         }
 
