Index: ZendMonitor.php
===================================================================
--- ZendMonitor.php	(revision 55)
+++ ZendMonitor.php	(working copy)
@@ -44,7 +44,7 @@
      */
     public function __construct()
     {
-        if (!function_exists('monitor_custom_event')) {
+        if (!function_exists('zend_monitor_custom_event')) {
             $this->_isEnabled = false;
         }
     }
@@ -103,9 +103,9 @@
         unset($event['priority'], $event['message']);
 
         if (!empty($event)) {
-            monitor_custom_event($priority, $message, false, $event);
+            zend_monitor_custom_event($priority, $message, $event);
         } else {
-            monitor_custom_event($priority, $message);
+            zend_monitor_custom_event($priority, $message);
         }
     }
 }
