Index: Abstract.php
===================================================================
--- Abstract.php	(revision 4192)
+++ Abstract.php	(working copy)
@@ -612,7 +612,8 @@
                 JOIN INFORMATION_SCHEMA.TABLE_CONSTRAINTS tc
                 ON (k.table_schema = tc.table_schema
                   AND k.table_name = tc.table_name
-                  AND tc.constraint_type = 'PRIMARY KEY'))
+                  AND tc.constraint_type = 'PRIMARY KEY'
+                  AND k.CONSTRAINT_NAME = tc.CONSTRAINT_NAME))
               ON (c.table_schema = k.table_schema
                 AND c.table_name = k.table_name
                 AND c.column_name = k.column_name)
