Index: tests/TestConfiguration.php.dist
===================================================================
--- tests/TestConfiguration.php.dist	(revision 24746)
+++ tests/TestConfiguration.php.dist	(working copy)
@@ -263,11 +263,6 @@
 defined('TESTS_ZEND_GDATA_GAPPS_PASSWORD') || define('TESTS_ZEND_GDATA_GAPPS_PASSWORD', 'password');
 
 /*
- * This is the ONLINE_ENABLED property for Google Base.
- */
-defined('TESTS_ZEND_GDATA_GBASE_ONLINE_ENABLED') || define('TESTS_ZEND_GDATA_GBASE_ONLINE_ENABLED', false);
-
-/*
  * This indicates that online tests for the Books Search data API
  * should be performed.
  */
Index: tests/Zend/Gdata/Gbase/_files/TestDataGbaseItemFeedSample1.xml
===================================================================
--- tests/Zend/Gdata/Gbase/_files/TestDataGbaseItemFeedSample1.xml	(revision 24746)
+++ tests/Zend/Gdata/Gbase/_files/TestDataGbaseItemFeedSample1.xml	(working copy)
@@ -1,91 +0,0 @@
-<?xml version='1.0' encoding='UTF-8'?>
-  <feed xmlns='http://www.w3.org/2005/Atom' xmlns:openSearch='http://a9.com/-/spec/opensearchrss/1.0/' xmlns:gm='http://base.google.com/ns-metadata/1.0' xmlns:g='http://base.google.com/ns/1.0' xmlns:batch='http://schemas.google.com/gdata/batch'>
-    <id>http://www.google.com/base/feeds/items</id>
-    <updated>2007-09-15T00:08:42.580Z</updated>
-    <title type='text'>Items matching query: [customer id(int):1385111]</title>
-    <link rel='alternate' type='text/html' href='http://base.google.com'/>
-    <link rel='http://schemas.google.com/g/2005#feed' type='application/atom+xml' href='http://www.google.com/base/feeds/items'/>
-    <link rel='http://schemas.google.com/g/2005#post' type='application/atom+xml' href='http://www.google.com/base/feeds/items'/>
-    <link rel='http://schemas.google.com/g/2005#batch' type='application/atom+xml' href='http://www.google.com/base/feeds/items/batch'/>
-    <link rel='self' type='application/atom+xml' href='http://www.google.com/base/feeds/items?start-index=1&amp;max-results=2'/>
-    <link rel='next' type='application/atom+xml' href='http://www.google.com/base/feeds/items?start-index=3&amp;max-results=2'/>
-    <author>
-      <name>Google Inc.</name>
-      <email>base@google.com</email>
-    </author>
-    <generator version='1.0' uri='http://base.google.com'>GoogleBase</generator>
-    <openSearch:totalResults>9</openSearch:totalResults>
-    <openSearch:startIndex>1</openSearch:startIndex>
-    <openSearch:itemsPerPage>2</openSearch:itemsPerPage>
-    <g:customer_id type='int'>1385111</g:customer_id>
-    <entry>
-      <id>http://www.google.com/base/feeds/items/1717711776806040891</id>
-      <published>2007-09-14T00:26:01.000Z</published>
-      <updated>2007-09-14T00:26:01.000Z</updated>
-      <category scheme='http://base.google.com/categories/itemtypes' term='Products'/>
-      <title type='text'>PHP Developer Handbook Second Edition</title>
-      <content type='html'>Essential handbook for PHP developers. This is a test item.</content>
-      <link rel='alternate' type='text/html' href='http://base.google.com/base/a/1385111/D1717711776806040891'/>
-      <link rel='self' type='application/atom+xml' href='http://www.google.com/base/feeds/items/1717711776806040891'/>
-      <link rel='edit' type='application/atom+xml' href='http://www.google.com/base/feeds/items/1717711776806040891'/>
-      <author>
-        <name>jacobm@google.com</name>
-        <email>anon-d2dp8lsofybp@base.google.com</email>
-      </author>
-      <g:item_type type='text'>Products</g:item_type>
-      <g:publisher type='text'>My Press</g:publisher>
-      <g:isbn type='text'>ISBN12345</g:isbn>
-      <g:condition type='text'>New</g:condition>
-      <g:published_date type='text'>Sep 2007</g:published_date>
-      <g:payment type='text'>Google Checkout</g:payment>
-      <g:upc type='text'>UPC12345</g:upc>
-      <g:brand type='text'>Oxford</g:brand>
-      <g:product_type type='text'>book</g:product_type>
-      <g:edition type='text'>First Edition</g:edition>
-      <g:year type='float'>2006.0</g:year>
-      <g:item_language type='text'>en</g:item_language>
-      <g:pages type='float'>253.0</g:pages>
-      <g:price type='floatUnit'>16.99 usd</g:price>
-      <g:target_country type='text'>US</g:target_country>
-      <g:customer_id type='int'>1385111</g:customer_id>
-      <g:customer_id type='int'>1385111</g:customer_id>
-      <g:quantity type='int'>10</g:quantity>
-      <g:author type='text'>John Doe</g:author>
-      <g:weight type='floatUnit'>2.2 lbs</g:weight>
-      <g:expiration_date type='dateTime'>2007-10-14T00:26:01.000Z</g:expiration_date>
-    </entry>
-    <entry>
-      <id>http://www.google.com/base/feeds/items/5366131833179111427</id>
-      <published>2007-09-14T00:32:03.000Z</published>
-      <updated>2007-09-14T00:32:04.000Z</updated>
-      <category scheme='http://base.google.com/categories/itemtypes' term='Products'/>
-      <title type='text'>PHP Developer Handbook Second Edition</title>
-      <content type='html'>Essential handbook for PHP developers. This is a test item.</content>
-      <link rel='alternate' type='text/html' href='http://base.google.com/base/a/1385111/D5366131833179111427'/>
-      <link rel='self' type='application/atom+xml' href='http://www.google.com/base/feeds/items/5366131833179111427'/>
-      <link rel='edit' type='application/atom+xml' href='http://www.google.com/base/feeds/items/5366131833179111427'/>
-      <author>
-        <name>jacobm@google.com</name>
-        <email>anon-d2dp8lsofybp@base.google.com</email>
-      </author>
-      <g:item_type type='text'>Products</g:item_type>
-      <g:isbn type='text'>ISBN12345</g:isbn>
-      <g:publisher type='text'>My Press</g:publisher>
-      <g:condition type='text'>New</g:condition>
-      <g:payment type='text'>Google Checkout</g:payment>
-      <g:upc type='text'>UPC12345</g:upc>
-      <g:edition type='text'>First Edition</g:edition>
-      <g:product_type type='text'>book</g:product_type>
-      <g:year type='float'>2006.0</g:year>
-      <g:item_language type='text'>en</g:item_language>
-      <g:pages type='float'>253.0</g:pages>
-      <g:price type='floatUnit'>12.99 usd</g:price>
-      <g:target_country type='text'>US</g:target_country>
-      <g:customer_id type='int'>1385111</g:customer_id>
-      <g:customer_id type='int'>1385111</g:customer_id>
-      <g:quantity type='int'>10</g:quantity>
-      <g:author type='text'>John Doe</g:author>
-      <g:weight type='floatUnit'>2.2 lbs</g:weight>
-      <g:expiration_date type='dateTime'>2007-10-14T00:32:03.000Z</g:expiration_date>
-    </entry>
-  </feed>
\ No newline at end of file
Index: tests/Zend/Gdata/Gbase/_files/TestDataGbaseSnippetFeedSample1.xml
===================================================================
--- tests/Zend/Gdata/Gbase/_files/TestDataGbaseSnippetFeedSample1.xml	(revision 24746)
+++ tests/Zend/Gdata/Gbase/_files/TestDataGbaseSnippetFeedSample1.xml	(working copy)
@@ -1,73 +0,0 @@
-<?xml version='1.0' encoding='UTF-8'?>
-  <feed xmlns='http://www.w3.org/2005/Atom' xmlns:openSearch='http://a9.com/-/spec/opensearchrss/1.0/' xmlns:gm='http://base.google.com/ns-metadata/1.0' xmlns:g='http://base.google.com/ns/1.0' xmlns:batch='http://schemas.google.com/gdata/batch'>
-    <id>http://www.google.com/base/feeds/snippets</id>
-    <updated>2007-09-15T00:09:38.999Z</updated>
-    <title type='text'>Items matching query: [title:PHP book]</title>
-    <link rel='alternate' type='text/html' href='http://base.google.com'/>
-    <link rel='http://schemas.google.com/g/2005#feed' type='application/atom+xml' href='http://www.google.com/base/feeds/snippets'/>
-    <link rel='http://schemas.google.com/g/2005#batch' type='application/atom+xml' href='http://www.google.com/base/feeds/snippets/batch'/>
-    <link rel='self' type='application/atom+xml' href='http://www.google.com/base/feeds/snippets?start-index=1&amp;max-results=2&amp;bq=%5Btitle%3APHP+book%5D'/>
-    <link rel='next' type='application/atom+xml' href='http://www.google.com/base/feeds/snippets?start-index=3&amp;max-results=2&amp;bq=%5Btitle%3APHP+book%5D'/>
-    <author>
-      <name>Google Inc.</name>
-      <email>base@google.com</email>
-    </author>
-    <generator version='1.0' uri='http://base.google.com'>GoogleBase</generator>
-    <openSearch:totalResults>191</openSearch:totalResults>
-    <openSearch:startIndex>1</openSearch:startIndex>
-    <openSearch:itemsPerPage>2</openSearch:itemsPerPage>
-    <entry>
-      <id>http://www.google.com/base/feeds/snippets/12314974507194001735</id>
-      <published>2007-06-06T07:02:31.000Z</published>
-      <updated>2007-09-14T07:48:53.000Z</updated>
-      <category scheme='http://base.google.com/categories/itemtypes' term='Products'/>
-      <title type='text'>The Zend PHP Certification Practice Test Book: Practice Questions for the Zend Certified Engineer Exam</title>
-      <content type='html'>John Coggeshall, Marco Tabini,Paperback, English-language edition,Pages:152,Pub BY Marco Tabini &amp; Associates,</content>
-      <link rel='alternate' type='text/html' href='http://search.barnesandnoble.com/booksearch/isbninquiry.asp?sourceid=Q000000630&amp;ean=9780973589887'/>
-      <link rel='self' type='application/atom+xml' href='http://www.google.com/base/feeds/snippets/12314974507194001735'/>
-      <author>
-        <name>Barnes &amp; Noble.com</name>
-        <email>anon-h0z4ch9cz5h5@base.google.com</email>
-      </author>
-      <g:item_type type='text'>Products</g:item_type>
-      <g:image_link type='url'>http://images.barnesandnoble.com.edgesuite.net/images/8950000/8957260.gif</g:image_link>
-      <g:publisher type='text'>Marco Tabini &amp; Associates,</g:publisher>
-      <g:isbn type='text'>0973589884</g:isbn>
-      <g:condition type='text'>new</g:condition>
-      <g:id type='text'>9780973589887</g:id>
-      <g:product_type type='text'>books</g:product_type>
-      <g:item_language type='text'>EN</g:item_language>
-      <g:pages type='int'>152</g:pages>
-      <g:price type='floatUnit'>21.99 usd</g:price>
-      <g:target_country type='text'>US</g:target_country>
-      <g:customer_id type='int'>1165992</g:customer_id>
-      <g:genre type='text'>Computers</g:genre>
-      <g:binding type='text'>paperback</g:binding>
-      <g:author type='text'>Marco Tabini</g:author>
-      <g:author type='text'>John Coggeshall</g:author>
-      <g:expiration_date type='dateTime'>2007-10-14T07:48:53.000Z</g:expiration_date>
-    </entry>
-    <entry>
-      <id>http://www.google.com/base/feeds/snippets/17983316950426332176</id>
-      <published>2007-09-07T04:21:51.000Z</published>
-      <updated>2007-09-14T04:09:34.000Z</updated>
-      <category scheme='http://base.google.com/categories/itemtypes' term='Products'/>
-      <title type='text'>PHP 5 / MYSQL PROGRAMMING FOR THE ABSOLUTE - BOOK *NEW</title>
-      <content type='html'>Sysqsystems includes more than two million titles in its online database, with an inventory of 1.3 Million titles available for immediate delivery.</content>
-      <link rel='alternate' type='text/html' href='http://adfarm.mediaplex.com/ad/ck/711-5256-8196-2?loc=http%3A%2F%2Fcgi.ebay.com%2FPHP-5-MYSQL-PROGRAMMING-FOR-THE-ABSOLUTE-BOOK-NEW_W0QQitemZ200149840989QQcmdZViewItem'/>
-      <link rel='self' type='application/atom+xml' href='http://www.google.com/base/feeds/snippets/17983316950426332176'/>
-      <author>
-        <name>eBay</name>
-        <email>rcross@ebay.com</email>
-      </author>
-      <g:item_type type='text'>Products</g:item_type>
-      <g:item_language type='text'>EN</g:item_language>
-      <g:price type='floatUnit'>18.42 usd</g:price>
-      <g:target_country type='text'>US</g:target_country>
-      <g:image_link type='url'>http://thumbs.ebaystatic.com/pict/200149840989_1.jpg</g:image_link>
-      <g:category type='text'>Books&gt;Nonfiction Books</g:category>
-      <g:customer_id type='int'>11729</g:customer_id>
-      <g:id type='text'>200149840989</g:id>
-      <g:expiration_date type='dateTime'>2007-10-14T04:09:34.000Z</g:expiration_date>
-    </entry>
-  </feed>
\ No newline at end of file
Index: tests/Zend/Gdata/Gbase/BaseAttributeTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/BaseAttributeTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/BaseAttributeTest.php	(working copy)
@@ -1,64 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_BaseAttributeTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->baseAttribute = new Zend_Gdata_Gbase_Extension_BaseAttribute();
-    }
-
-    public function testToAndFromString()
-    {
-
-        $this->baseAttribute->setName('price');
-        $this->baseAttribute->setText('10.99 USD');
-        $this->baseAttribute->setType('floatUnit');
-
-        $this->assertTrue($this->baseAttribute->getName() == 'price');
-        $this->assertTrue($this->baseAttribute->getText() == '10.99 USD');
-        $this->assertTrue($this->baseAttribute->getType() == 'floatUnit');
-
-        $newBaseAttribute = new Zend_Gdata_Gbase_Extension_BaseAttribute();
-        $doc = new DOMDocument();
-        $doc->loadXML($this->baseAttribute->saveXML());
-        $newBaseAttribute->transferFromDom($doc->documentElement);
-
-        $this->assertTrue($this->baseAttribute->getName() == $newBaseAttribute->getName());
-        $this->assertTrue($this->baseAttribute->getText() == $newBaseAttribute->getText());
-        $this->assertTrue($this->baseAttribute->getType() == $newBaseAttribute->getType());
-    }
-
-}
Index: tests/Zend/Gdata/Gbase/ItemEntryTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/ItemEntryTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/ItemEntryTest.php	(working copy)
@@ -1,68 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_ItemEntryTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->itemEntry = new Zend_Gdata_Gbase_ItemEntry();
-    }
-
-    public function testToAndFromString()
-    {
-        $this->itemEntry->setItemType('products');
-        $this->assertEquals($this->itemEntry->getItemType()->getText(), 'products');
-
-        $this->itemEntry->addGbaseAttribute('price', '10.99 USD', 'floatUnit');
-        $baseAttribute = $this->itemEntry->getGbaseAttribute('price');
-        $this->assertEquals(count($baseAttribute), 1);
-        $this->assertEquals($baseAttribute[0]->getName(), 'price');
-        $this->assertEquals($baseAttribute[0]->getText(), '10.99 USD');
-        $this->assertEquals($baseAttribute[0]->getType(), 'floatUnit');
-
-        $newItemEntry = new Zend_Gdata_Gbase_ItemEntry();
-        $doc = new DOMDocument();
-        $doc->loadXML($this->itemEntry->saveXML());
-        $newItemEntry->transferFromDom($doc->documentElement);
-        $rowDataFromXML = $newItemEntry->getGbaseAttribute('price');
-
-        $this->assertEquals($this->itemEntry->getItemType()->getText(), $newItemEntry->getItemType()->getText());
-        $this->assertEquals(count($rowDataFromXML), 1);
-        $this->assertEquals($rowDataFromXML[0]->getName(), 'price');
-        $this->assertEquals($rowDataFromXML[0]->getText(), '10.99 USD');
-        $this->assertEquals($rowDataFromXML[0]->getType(), 'floatUnit');
-    }
-
-}
Index: tests/Zend/Gdata/Gbase/ItemFeedTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/ItemFeedTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/ItemFeedTest.php	(working copy)
@@ -1,67 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_ItemFeedTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->itemFeed = new Zend_Gdata_Gbase_ItemFeed(
-                file_get_contents(dirname(__FILE__) . '/_files/TestDataGbaseItemFeedSample1.xml'),
-                true);
-    }
-
-    public function testToAndFromString()
-    {
-        $this->assertEquals(2, count($this->itemFeed->entries));
-        $this->assertEquals(2, $this->itemFeed->entries->count());
-        foreach($this->itemFeed->entries as $entry)
-        {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_ItemEntry);
-        }
-
-        $newItemFeed = new Zend_Gdata_Gbase_ItemFeed();
-        $doc = new DOMDocument();
-        $doc->loadXML($this->itemFeed->saveXML());
-        $newItemFeed->transferFromDom($doc->documentElement);
-
-        $this->assertEquals(2, count($newItemFeed->entries));
-        $this->assertEquals(2, $newItemFeed->entries->count());
-        foreach($newItemFeed->entries as $entry)
-        {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_ItemEntry);
-        }
-    }
-
-}
Index: tests/Zend/Gdata/Gbase/ItemQueryTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/ItemQueryTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/ItemQueryTest.php	(working copy)
@@ -1,92 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Gdata/Gbase/ItemQuery.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_ItemQueryTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->itemQuery = new Zend_Gdata_Gbase_ItemQuery();
-    }
-
-    public function testBq()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setBq('[title:PHP]');
-        $this->assertEquals($this->itemQuery->getBq(), '[title:PHP]');
-    }
-
-    public function testRefine()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setRefine('true');
-        $this->assertEquals($this->itemQuery->getRefine(), 'true');
-    }
-
-    public function testContent()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setContent('stats');
-        $this->assertEquals($this->itemQuery->getContent(), 'stats');
-    }
-
-    public function testOrderBy()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setOrderBy('relevancy');
-        $this->assertEquals($this->itemQuery->getOrderBy(), 'relevancy');
-    }
-
-    public function testSortOrder()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setOrderBy('descending');
-        $this->assertEquals($this->itemQuery->getOrderBy(), 'descending');
-    }
-
-    public function testCrowdBy()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setCrowdBy('attribute:5,content:2,url');
-        $this->assertEquals($this->itemQuery->getCrowdBy(), 'attribute:5,content:2,url');
-    }
-
-    public function testAdjust()
-    {
-        $this->itemQuery->resetParameters();
-        $this->itemQuery->setAdjust('true');
-        $this->assertEquals($this->itemQuery->getAdjust(), 'true');
-    }
-}
Index: tests/Zend/Gdata/Gbase/QueryTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/QueryTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/QueryTest.php	(working copy)
@@ -1,50 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Gdata/Gbase/Query.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_QueryTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->query = new Zend_Gdata_Gbase_Query();
-    }
-
-    public function testKey()
-    {
-        $this->query->setKey('xyz');
-        $this->assertEquals($this->query->getKey(), 'xyz');
-    }
-
-}
Index: tests/Zend/Gdata/Gbase/SnippetFeedTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/SnippetFeedTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/SnippetFeedTest.php	(working copy)
@@ -1,67 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_SnippetFeedTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->snippetFeed = new Zend_Gdata_Gbase_SnippetFeed(
-                file_get_contents(dirname(__FILE__) . '/_files/TestDataGbaseSnippetFeedSample1.xml'),
-                true);
-    }
-
-    public function testToAndFromString()
-    {
-        $this->assertEquals(2, count($this->snippetFeed->entries));
-        $this->assertEquals(2, $this->snippetFeed->entries->count());
-        foreach($this->snippetFeed->entries as $entry)
-        {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_SnippetEntry);
-        }
-
-        $newSnippetFeed = new Zend_Gdata_Gbase_SnippetFeed();
-        $doc = new DOMDocument();
-        $doc->loadXML($this->snippetFeed->saveXML());
-        $newSnippetFeed->transferFromDom($doc->documentElement);
-
-        $this->assertEquals(2, count($newSnippetFeed->entries));
-        $this->assertEquals(2, $newSnippetFeed->entries->count());
-        foreach($newSnippetFeed->entries as $entry)
-        {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_SnippetEntry);
-        }
-    }
-
-}
Index: tests/Zend/Gdata/Gbase/SnippetQueryTest.php
===================================================================
--- tests/Zend/Gdata/Gbase/SnippetQueryTest.php	(revision 24746)
+++ tests/Zend/Gdata/Gbase/SnippetQueryTest.php	(working copy)
@@ -1,93 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Gdata/Gbase/SnippetQuery.php';
-require_once 'Zend/Http/Client.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_Gbase_SnippetQueryTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $this->snippetQuery = new Zend_Gdata_Gbase_SnippetQuery();
-    }
-
-    public function testBq()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setBq('[title:PHP]');
-        $this->assertEquals($this->snippetQuery->getBq(), '[title:PHP]');
-    }
-
-    public function testRefine()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setRefine('true');
-        $this->assertEquals($this->snippetQuery->getRefine(), 'true');
-    }
-
-    public function testContent()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setContent('stats');
-        $this->assertEquals($this->snippetQuery->getContent(), 'stats');
-    }
-
-    public function testOrderBy()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setOrderBy('relevancy');
-        $this->assertEquals($this->snippetQuery->getOrderBy(), 'relevancy');
-    }
-
-    public function testSortOrder()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setOrderBy('descending');
-        $this->assertEquals($this->snippetQuery->getOrderBy(), 'descending');
-    }
-
-    public function testCrowdBy()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setCrowdBy('attribute:5,content:2,url');
-        $this->assertEquals($this->snippetQuery->getCrowdBy(), 'attribute:5,content:2,url');
-    }
-
-    public function testAdjust()
-    {
-        $this->snippetQuery->resetParameters();
-        $this->snippetQuery->setAdjust('true');
-        $this->assertEquals($this->snippetQuery->getAdjust(), 'true');
-    }
-
-}
Index: tests/Zend/Gdata/AllTests.php
===================================================================
--- tests/Zend/Gdata/AllTests.php	(revision 24746)
+++ tests/Zend/Gdata/AllTests.php	(working copy)
@@ -73,14 +73,6 @@
 require_once 'Zend/Gdata/WhereTest.php';
 require_once 'Zend/Gdata/WhoTest.php';
 
-require_once 'Zend/Gdata/Gbase/ItemEntryTest.php';
-require_once 'Zend/Gdata/Gbase/ItemFeedTest.php';
-require_once 'Zend/Gdata/Gbase/ItemQueryTest.php';
-require_once 'Zend/Gdata/Gbase/SnippetFeedTest.php';
-require_once 'Zend/Gdata/Gbase/SnippetQueryTest.php';
-require_once 'Zend/Gdata/Gbase/QueryTest.php';
-require_once 'Zend/Gdata/Gbase/BaseAttributeTest.php';
-
 require_once 'Zend/Gdata/CalendarTest.php';
 require_once 'Zend/Gdata/CalendarFeedTest.php';
 require_once 'Zend/Gdata/CalendarEventTest.php';
@@ -192,7 +184,6 @@
  * and authentication credentials
  */
 require_once 'Zend/Gdata/GdataOnlineTest.php';
-require_once 'Zend/Gdata/GbaseOnlineTest.php';
 require_once 'Zend/Gdata/CalendarOnlineTest.php';
 require_once 'Zend/Gdata/HealthOnlineTest.php';
 require_once 'Zend/Gdata/SpreadsheetsOnlineTest.php';
@@ -267,14 +258,6 @@
         $suite->addTestSuite('Zend_Gdata_WhereTest');
         $suite->addTestSuite('Zend_Gdata_WhoTest');
 
-        $suite->addTestSuite('Zend_Gdata_Gbase_ItemEntryTest');
-        $suite->addTestSuite('Zend_Gdata_Gbase_ItemFeedTest');
-        $suite->addTestSuite('Zend_Gdata_Gbase_ItemQueryTest');
-        $suite->addTestSuite('Zend_Gdata_Gbase_SnippetFeedTest');
-        $suite->addTestSuite('Zend_Gdata_Gbase_SnippetQueryTest');
-        $suite->addTestSuite('Zend_Gdata_Gbase_QueryTest');
-        $suite->addTestSuite('Zend_Gdata_Gbase_BaseAttributeTest');
-
         $suite->addTestSuite('Zend_Gdata_CalendarTest');
         $suite->addTestSuite('Zend_Gdata_CalendarFeedTest');
         $suite->addTestSuite('Zend_Gdata_CalendarEventTest');
@@ -395,11 +378,6 @@
                 $suite->addTestSuite('Zend_Gdata_GdataOnlineTest');
             }
 
-            if (defined('TESTS_ZEND_GDATA_GBASE_ONLINE_ENABLED') &&
-            constant('TESTS_ZEND_GDATA_GBASE_ONLINE_ENABLED') == true) {
-                $suite->addTestSuite('Zend_Gdata_GbaseOnlineTest');
-            }
-
             if (defined('TESTS_ZEND_GDATA_CALENDAR_ONLINE_ENABLED') &&
             constant('TESTS_ZEND_GDATA_CALENDAR_ONLINE_ENABLED') == true) {
                 $suite->addTestSuite('Zend_Gdata_CalendarOnlineTest');
Index: tests/Zend/Gdata/GbaseOnlineTest.php
===================================================================
--- tests/Zend/Gdata/GbaseOnlineTest.php	(revision 24746)
+++ tests/Zend/Gdata/GbaseOnlineTest.php	(working copy)
@@ -1,167 +0,0 @@
-<?php
-/**
- * Zend Framework
- *
- * LICENSE
- *
- * This source file is subject to the new BSD license that is bundled
- * with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * http://framework.zend.com/license/new-bsd
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@zend.com so we can send you a copy immediately.
- *
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @version    $Id $
- */
-
-require_once 'Zend/Gdata/Gbase.php';
-require_once 'Zend/Http/Client.php';
-require_once 'Zend/Gdata/ClientLogin.php';
-
-/**
- * @category   Zend
- * @package    Zend_Gdata_Gbase
- * @subpackage UnitTests
- * @copyright  Copyright (c) 2005-2012 Zend Technologies USA Inc. (http://www.zend.com)
- * @license    http://framework.zend.com/license/new-bsd     New BSD License
- * @group      Zend_Gdata
- * @group      Zend_Gdata_Gbase
- */
-class Zend_Gdata_GbaseOnlineTest extends PHPUnit_Framework_TestCase
-{
-
-    public function setUp()
-    {
-        $user = constant('TESTS_ZEND_GDATA_CLIENTLOGIN_EMAIL');
-        $pass = constant('TESTS_ZEND_GDATA_CLIENTLOGIN_PASSWORD');
-        $service = Zend_Gdata_Gbase::AUTH_SERVICE_NAME;
-        $client = Zend_Gdata_ClientLogin::getHttpClient($user, $pass, $service);
-        $this->gdata = new Zend_Gdata_Gbase($client);
-    }
-
-    public function testGetGbaseItemFeed()
-    {
-        $feed = $this->gdata->getGbaseItemFeed();
-        $this->assertTrue($feed instanceof Zend_Gdata_Gbase_ItemFeed);
-        foreach ($feed->entries as $entry) {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_ItemEntry);
-            $this->assertEquals($entry->getHttpClient(), $feed->getHttpClient());
-        }
-
-        $query = new Zend_Gdata_Gbase_ItemQuery();
-        $feed = $this->gdata->getGbaseItemFeed($query);
-        $this->assertTrue($feed instanceof Zend_Gdata_Gbase_ItemFeed);
-        foreach ($feed->entries as $entry) {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_ItemEntry);
-            $this->assertEquals($entry->getHttpClient(), $feed->getHttpClient());
-        }
-
-        $uri = $query->getQueryUrl();
-        $feed = $this->gdata->getGbaseItemFeed($uri);
-        $this->assertTrue($feed instanceof Zend_Gdata_Gbase_ItemFeed);
-        foreach ($feed->entries as $entry) {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_ItemEntry);
-            $this->assertEquals($entry->getHttpClient(), $feed->getHttpClient());
-        }
-    }
-
-    public function testGetGbaseItemEntry()
-    {
-        $newEntry = $this->gdata->newItemEntry();
-
-        $title = 'PHP Developer Handbook';
-        $newEntry->title = $this->gdata->newTitle(trim($title));
-
-        $desc = 'This is a test item';
-        $newEntry->content = $this->gdata->newContent($desc);
-        $newEntry->content->type = 'text';
-
-        $itemType = 'Products';
-        $newEntry->itemType = $itemType;
-        $newEntry->itemType->type = 'text';
-
-        $newEntry->addGbaseAttribute('product_type', 'book', 'text');
-        $newEntry->addGbaseAttribute('price', '12.99 usd', 'floatUnit');
-        $newEntry->addGbaseAttribute('quantity', '10', 'int');
-
-        $createdEntry = $this->gdata->insertGbaseItem($newEntry, false);
-        $itemId = $createdEntry->id->text;
-
-        $entry = $this->gdata->getGbaseItemEntry($itemId);
-        $this->assertTrue($entry instanceof Zend_Gdata_Gbase_ItemEntry);
-    }
-
-    public function testInsertGbaseItem()
-    {
-        $newEntry = $this->gdata->newItemEntry();
-
-        $title = 'PHP Developer Handbook';
-        $newEntry->title = $this->gdata->newTitle(trim($title));
-
-        $desc = 'Essential handbook for PHP developers.';
-        $newEntry->content = $this->gdata->newContent($desc);
-        $newEntry->content->type = 'text';
-
-        $itemType = 'Products';
-        $newEntry->itemType = $itemType;
-        $newEntry->itemType->type = 'text';
-
-        $newEntry->addGbaseAttribute('product_type', 'book', 'text');
-        $newEntry->addGbaseAttribute('price', '12.99 usd', 'floatUnit');
-        $newEntry->addGbaseAttribute('quantity', '10', 'int');
-
-        $createdEntry = $this->gdata->insertGbaseItem($newEntry, true);
-
-        $this->assertEquals($title, $createdEntry->title->text);
-        $this->assertEquals($desc, $createdEntry->content->text);
-        $this->assertEquals($itemType, $createdEntry->itemType->text);
-
-        $baseAttribute = $createdEntry->getGbaseAttribute('product_type');
-        $this->assertEquals('product_type', $baseAttribute[0]->name);
-        $this->assertEquals('book', $baseAttribute[0]->text);
-        $this->assertEquals('text', $baseAttribute[0]->type);
-
-        $baseAttribute = $createdEntry->getGbaseAttribute('price');
-        $this->assertEquals('price', $baseAttribute[0]->name);
-        $this->assertEquals('12.99 usd', $baseAttribute[0]->text);
-        $this->assertEquals('floatUnit', $baseAttribute[0]->type);
-
-        $baseAttribute = $createdEntry->getGbaseAttribute('quantity');
-        $this->assertEquals('quantity', $baseAttribute[0]->name);
-        $this->assertEquals('10', $baseAttribute[0]->text);
-        $this->assertEquals('int', $baseAttribute[0]->type);
-    }
-
-    public function testGetGbaseSnippetFeed()
-    {
-        $feed = $this->gdata->getGbaseSnippetFeed();
-        $this->assertTrue($feed instanceof Zend_Gdata_Gbase_SnippetFeed);
-        foreach ($feed->entries as $entry) {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_SnippetEntry);
-            $this->assertEquals($entry->getHttpClient(), $feed->getHttpClient());
-        }
-
-        $query = new Zend_Gdata_Gbase_SnippetQuery();
-        $feed = $this->gdata->getGbaseSnippetFeed($query);
-        $this->assertTrue($feed instanceof Zend_Gdata_Gbase_SnippetFeed);
-        foreach ($feed->entries as $entry) {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_SnippetEntry);
-            $this->assertEquals($entry->getHttpClient(), $feed->getHttpClient());
-        }
-
-        $uri = $query->getQueryUrl();
-        $feed = $this->gdata->getGbaseSnippetFeed($uri);
-        $this->assertTrue($feed instanceof Zend_Gdata_Gbase_SnippetFeed);
-        foreach ($feed->entries as $entry) {
-            $this->assertTrue($entry instanceof Zend_Gdata_Gbase_SnippetEntry);
-            $this->assertEquals($entry->getHttpClient(), $feed->getHttpClient());
-        }
-    }
-
-}
Index: library/Zend/Gdata/Gbase/ItemEntry.php
===================================================================
--- library/Zend/Gdata/Gbase/ItemEntry.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/ItemEntry.php	(working copy)
@@ -39,123 +39,4 @@
  */
 class Zend_Gdata_Gbase_ItemEntry extends Zend_Gdata_Gbase_Entry
 {
-    /**
-     * The classname for individual item entry elements.
-     *
-     * @var string
-     */
-    protected $_entryClassName = 'Zend_Gdata_Gbase_ItemEntry';
-
-    /**
-     * Set the value of the itme_type
-     *
-     * @param Zend_Gdata_Gbase_Extension_ItemType $value The desired value for the item_type
-     * @return Zend_Gdata_Gbase_ItemEntry Provides a fluent interface
-     */
-    public function setItemType($value)
-    {
-        $this->addGbaseAttribute('item_type', $value, 'text');
-        return $this;
-    }
-
-    /**
-     * Adds a custom attribute to the entry in the following format:
-     * &lt;g:[$name] type='[$type]'&gt;[$value]&lt;/g:[$name]&gt;
-     *
-     * @param string $name The name of the attribute
-     * @param string $value The text value of the attribute
-     * @param string $type (optional) The type of the attribute.
-     *          e.g.: 'text', 'number', 'floatUnit'
-     * @return Zend_Gdata_Gbase_ItemEntry Provides a fluent interface
-     */
-    public function addGbaseAttribute($name, $text, $type = null) {
-        $newBaseAttribute =  new Zend_Gdata_Gbase_Extension_BaseAttribute($name, $text, $type);
-        $this->_baseAttributes[] = $newBaseAttribute;
-        return $this;
-    }
-
-    /**
-     * Removes a Base attribute from the current list of Base attributes
-     *
-     * @param Zend_Gdata_Gbase_Extension_BaseAttribute $baseAttribute The attribute to be removed
-     * @return Zend_Gdata_Gbase_ItemEntry Provides a fluent interface
-     */
-    public function removeGbaseAttribute($baseAttribute) {
-        $baseAttributes = $this->_baseAttributes;
-        for ($i = 0; $i < count($this->_baseAttributes); $i++) {
-            if ($this->_baseAttributes[$i] == $baseAttribute) {
-                array_splice($baseAttributes, $i, 1);
-                break;
-            }
-        }
-        $this->_baseAttributes = $baseAttributes;
-        return $this;
-    }
-
-    /**
-     * Uploads changes in this entry to the server using Zend_Gdata_App
-     *
-     * @param boolean $dryRun Whether the transaction is dry run or not.
-     * @param string|null $uri The URI to send requests to, or null if $data
-     *        contains the URI.
-     * @param string|null $className The name of the class that should we
-     *        deserializing the server response. If null, then
-     *        'Zend_Gdata_App_Entry' will be used.
-     * @param array $extraHeaders Extra headers to add to the request, as an
-     *        array of string-based key/value pairs.
-     * @return Zend_Gdata_App_Entry The updated entry
-     * @throws Zend_Gdata_App_Exception
-     */
-    public function save($dryRun = false,
-                         $uri = null,
-                         $className = null,
-                         $extraHeaders = array())
-    {
-        if ($dryRun == true) {
-            $editLink = $this->getEditLink();
-            if ($uri == null && $editLink !== null) {
-                $uri = $editLink->getHref() . '?dry-run=true';
-            }
-            if ($uri === null) {
-                require_once 'Zend/Gdata/App/InvalidArgumentException.php';
-                throw new Zend_Gdata_App_InvalidArgumentException('You must specify an URI which needs deleted.');
-            }
-            $service = new Zend_Gdata_App($this->getHttpClient());
-            return $service->updateEntry($this,
-                                         $uri,
-                                         $className,
-                                         $extraHeaders);
-        } else {
-            parent::save($uri, $className, $extraHeaders);
-        }
-    }
-
-    /**
-     * Deletes this entry to the server using the referenced
-     * Zend_Http_Client to do a HTTP DELETE to the edit link stored in this
-     * entry's link collection.
-     *
-     * @param boolean $dyrRun Whether the transaction is dry run or not
-     * @return void
-     * @throws Zend_Gdata_App_Exception
-     */
-    public function delete($dryRun = false)
-    {
-        $uri = null;
-
-        if ($dryRun == true) {
-            $editLink = $this->getEditLink();
-            if ($editLink !== null) {
-                $uri = $editLink->getHref() . '?dry-run=true';
-            }
-            if ($uri === null) {
-                require_once 'Zend/Gdata/App/InvalidArgumentException.php';
-                throw new Zend_Gdata_App_InvalidArgumentException('You must specify an URI which needs deleted.');
-            }
-            parent::delete($uri);
-        } else {
-            parent::delete();
-        }
-    }
-
 }
Index: library/Zend/Gdata/Gbase/Entry.php
===================================================================
--- library/Zend/Gdata/Gbase/Entry.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/Entry.php	(working copy)
@@ -22,14 +22,14 @@
  */
 
 /**
- * @see Zend_Gdata_Entry
+ * @see Zend_Exception
  */
-require_once 'Zend/Gdata/Entry.php';
+require_once 'Zend/Exception.php';
 
 /**
- * @see Zend_Gdata_Gbase_Extension_BaseAttribute
+ * @see Zend_Gdata_Entry
  */
-require_once 'Zend/Gdata/Gbase/Extension/BaseAttribute.php';
+require_once 'Zend/Gdata/Entry.php';
 
 /**
  * Base class for working with Google Base entries.
@@ -44,108 +44,16 @@
  */
 class Zend_Gdata_Gbase_Entry extends Zend_Gdata_Entry
 {
-
     /**
-     * Name of the base class for Google Base entries
-     *
-     * var @string
-     */
-    protected $_entryClassName = 'Zend_Gdata_Gbase_Entry';
-
-    /**
-     * Google Base attribute elements in the 'g' namespace
-     *
-     * @var array
-     */
-    protected $_baseAttributes = array();
-
-    /**
      * Constructs a new Zend_Gdata_Gbase_ItemEntry object.
      * @param DOMElement $element (optional) The DOMElement on which to base this object.
      */
     public function __construct($element = null)
     {
-        $this->registerAllNamespaces(Zend_Gdata_Gbase::$namespaces);
-        parent::__construct($element);
+        throw new Zend_Exception(
+            'Google Base API has been discontinued by Google and was removed'
+            . ' from Zend Framework in 1.12.0.  For more information see: '
+            . 'http://googlemerchantblog.blogspot.ca/2010/12/new-shopping-apis-and-deprecation-of.html'
+        );    
     }
-
-    /**
-     * Retrieves a DOMElement which corresponds to this element and all
-     * child properties.  This is used to build an entry back into a DOM
-     * and eventually XML text for application storage/persistence.
-     *
-     * @param DOMDocument $doc The DOMDocument used to construct DOMElements
-     * @return DOMElement The DOMElement representing this element and all
-     *          child properties.
-     */
-    public function getDOM($doc = null, $majorVersion = 1, $minorVersion = null)
-    {
-        $element = parent::getDOM($doc, $majorVersion, $minorVersion);
-        foreach ($this->_baseAttributes as $baseAttribute) {
-            $element->appendChild($baseAttribute->getDOM($element->ownerDocument));
-        }
-        return $element;
-    }
-
-    /**
-     * Creates individual Entry objects of the appropriate type and
-     * stores them as members of this entry based upon DOM data.
-     *
-     * @param DOMNode $child The DOMNode to process
-     */
-    protected function takeChildFromDOM($child)
-    {
-        $absoluteNodeName = $child->namespaceURI . ':' . $child->localName;
-
-        if (strstr($absoluteNodeName, $this->lookupNamespace('g') . ':')) {
-            $baseAttribute = new Zend_Gdata_Gbase_Extension_BaseAttribute();
-            $baseAttribute->transferFromDOM($child);
-            $this->_baseAttributes[] = $baseAttribute;
-        } else {
-            parent::takeChildFromDOM($child);
-        }
-    }
-
-    /**
-     * Get the value of the itme_type
-     *
-     * @return Zend_Gdata_Gbase_Extension_ItemType The requested object.
-     */
-    public function getItemType()
-    {
-        $itemType = $this->getGbaseAttribute('item_type');
-        if (is_object($itemType[0])) {
-          return $itemType[0];
-        } else {
-          return null;
-        }
-    }
-
-    /**
-     * Return all the Base attributes
-     * @return Zend_Gdata_Gbase_Extension_BaseAttribute
-     */
-    public function getGbaseAttributes() {
-        return $this->_baseAttributes;
-    }
-
-    /**
-     * Return an array of Base attributes that match the given attribute name
-     *
-     * @param string $name The name of the Base attribute to look for
-     * @return array $matches Array that contains the matching list of Base attributes
-     */
-    public function getGbaseAttribute($name)
-    {
-        $matches = array();
-        for ($i = 0; $i < count($this->_baseAttributes); $i++) {
-            $baseAttribute = $this->_baseAttributes[$i];
-            if ($baseAttribute->rootElement == $name &&
-                $baseAttribute->rootNamespaceURI == $this->lookupNamespace('g')) {
-                $matches[] = &$this->_baseAttributes[$i];
-            }
-        }
-        return $matches;
-    }
-
 }
Index: library/Zend/Gdata/Gbase/ItemFeed.php
===================================================================
--- library/Zend/Gdata/Gbase/ItemFeed.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/ItemFeed.php	(working copy)
@@ -39,10 +39,4 @@
  */
 class Zend_Gdata_Gbase_ItemFeed extends Zend_Gdata_Feed
 {
-    /**
-     * The classname for individual item feed elements.
-     *
-     * @var string
-     */
-    protected $_entryClassName = 'Zend_Gdata_Gbase_ItemEntry';
 }
Index: library/Zend/Gdata/Gbase/Feed.php
===================================================================
--- library/Zend/Gdata/Gbase/Feed.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/Feed.php	(working copy)
@@ -22,6 +22,11 @@
  */
 
 /**
+ * @see Zend_Exception
+ */
+require_once 'Zend/Exception.php';
+
+/**
  * @see Zend_Gdata_Feed
  */
 require_once 'Zend/Gdata/Feed.php';
@@ -40,13 +45,6 @@
 class Zend_Gdata_Gbase_Feed extends Zend_Gdata_Feed
 {
     /**
-     * The classname for the feed.
-     *
-     * @var string
-     */
-    protected $_feedClassName = 'Zend_Gdata_Gbase_Feed';
-
-    /**
      * Create a new instance.
      *
      * @param DOMElement $element (optional) DOMElement from which this
@@ -54,7 +52,10 @@
      */
     public function __construct($element = null)
     {
-        $this->registerAllNamespaces(Zend_Gdata_Gbase::$namespaces);
-        parent::__construct($element);
+        throw new Zend_Exception(
+            'Google Base API has been discontinued by Google and was removed'
+            . ' from Zend Framework in 1.12.0.  For more information see: '
+            . 'http://googlemerchantblog.blogspot.ca/2010/12/new-shopping-apis-and-deprecation-of.html'
+        );    
     }
 }
Index: library/Zend/Gdata/Gbase/ItemQuery.php
===================================================================
--- library/Zend/Gdata/Gbase/ItemQuery.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/ItemQuery.php	(working copy)
@@ -22,14 +22,14 @@
  */
 
 /**
- * @see Zend_Gdata_Query
+ * @see Zend_Exception
  */
-require_once('Zend/Gdata/Query.php');
+require_once 'Zend/Exception.php';
 
 /**
  * @see Zend_Gdata_Gbase_Query
  */
-require_once('Zend/Gdata/Gbase/Query.php');
+require_once 'Zend/Gdata/Gbase/Query.php';
 
 
 /**
@@ -49,53 +49,4 @@
      * Path to the customer items feeds on the Google Base server.
      */
     const GBASE_ITEM_FEED_URI = 'https://www.google.com/base/feeds/items';
-
-    /**
-     * The default URI for POST methods
-     *
-     * @var string
-     */
-    protected $_defaultFeedUri = self::GBASE_ITEM_FEED_URI;
-
-    /**
-     * The id of an item
-     *
-     * @var string
-     */
-    protected $_id = null;
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setId($value)
-    {
-        $this->_id = $value;
-        return $this;
-    }
-
-    /*
-     * @return string id
-     */
-    public function getId()
-    {
-        return $this->_id;
-    }
-
-    /**
-     * Returns the query URL generated by this query instance.
-     *
-     * @return string The query URL for this instance.
-     */
-    public function getQueryUrl()
-    {
-        $uri = $this->_defaultFeedUri;
-        if ($this->getId() !== null) {
-            $uri .= '/' . $this->getId();
-        } else {
-            $uri .= $this->getQueryString();
-        }
-        return $uri;
-    }
-
 }
Index: library/Zend/Gdata/Gbase/SnippetEntry.php
===================================================================
--- library/Zend/Gdata/Gbase/SnippetEntry.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/SnippetEntry.php	(working copy)
@@ -39,10 +39,4 @@
  */
 class Zend_Gdata_Gbase_SnippetEntry extends Zend_Gdata_Gbase_Entry
 {
-    /**
-     * The classname for individual snippet entry elements.
-     *
-     * @var string
-     */
-    protected $_entryClassName = 'Zend_Gdata_Gbase_SnippetEntry';
 }
Index: library/Zend/Gdata/Gbase/Query.php
===================================================================
--- library/Zend/Gdata/Gbase/Query.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/Query.php	(working copy)
@@ -22,9 +22,14 @@
  */
 
 /**
+ * @see Zend_Exception
+ */
+require_once 'Zend/Exception.php';
+
+/**
  * @see Zend_Gdata_Query
  */
-require_once('Zend/Gdata/Query.php');
+require_once 'Zend/Gdata/Query.php';
 
 /**
  * Assists in constructing queries for Google Base
@@ -51,218 +56,14 @@
     const GBASE_SNIPPET_FEED_URI = 'https://www.google.com/base/feeds/snippets';
 
     /**
-     * The default URI for POST methods
-     *
-     * @var string
+     * Create Gdata_Query object
      */
-    protected $_defaultFeedUri = self::GBASE_ITEM_FEED_URI;
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_Query Provides a fluent interface
-     */
-    public function setKey($value)
+    public function __construct($url = null)
     {
-        if ($value !== null) {
-            $this->_params['key'] = $value;
-        } else {
-            unset($this->_params['key']);
-        }
-        return $this;
+        throw new Zend_Exception(
+            'Google Base API has been discontinued by Google and was removed'
+            . ' from Zend Framework in 1.12.0.  For more information see: '
+            . 'http://googlemerchantblog.blogspot.ca/2010/12/new-shopping-apis-and-deprecation-of.html'
+        );    
     }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setBq($value)
-    {
-        if ($value !== null) {
-            $this->_params['bq'] = $value;
-        } else {
-            unset($this->_params['bq']);
-        }
-        return $this;
-    }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setRefine($value)
-    {
-        if ($value !== null) {
-            $this->_params['refine'] = $value;
-        } else {
-            unset($this->_params['refine']);
-        }
-        return $this;
-    }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setContent($value)
-    {
-        if ($value !== null) {
-            $this->_params['content'] = $value;
-        } else {
-            unset($this->_params['content']);
-        }
-        return $this;
-    }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setOrderBy($value)
-    {
-        if ($value !== null) {
-            $this->_params['orderby'] = $value;
-        } else {
-            unset($this->_params['orderby']);
-        }
-        return $this;
-    }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setSortOrder($value)
-    {
-        if ($value !== null) {
-            $this->_params['sortorder'] = $value;
-        } else {
-            unset($this->_params['sortorder']);
-        }
-        return $this;
-    }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setCrowdBy($value)
-    {
-        if ($value !== null) {
-            $this->_params['crowdby'] = $value;
-        } else {
-            unset($this->_params['crowdby']);
-        }
-        return $this;
-    }
-
-    /**
-     * @param string $value
-     * @return Zend_Gdata_Gbase_ItemQuery Provides a fluent interface
-     */
-    public function setAdjust($value)
-    {
-        if ($value !== null) {
-            $this->_params['adjust'] = $value;
-        } else {
-            unset($this->_params['adjust']);
-        }
-        return $this;
-    }
-
-    /**
-     * @return string key
-     */
-    public function getKey()
-    {
-        if (array_key_exists('key', $this->_params)) {
-            return $this->_params['key'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string bq
-     */
-    public function getBq()
-    {
-        if (array_key_exists('bq', $this->_params)) {
-            return $this->_params['bq'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string refine
-     */
-    public function getRefine()
-    {
-        if (array_key_exists('refine', $this->_params)) {
-            return $this->_params['refine'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string content
-     */
-    public function getContent()
-    {
-        if (array_key_exists('content', $this->_params)) {
-            return $this->_params['content'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string orderby
-     */
-    public function getOrderBy()
-    {
-        if (array_key_exists('orderby', $this->_params)) {
-            return $this->_params['orderby'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string sortorder
-     */
-    public function getSortOrder()
-    {
-        if (array_key_exists('sortorder', $this->_params)) {
-            return $this->_params['sortorder'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string crowdby
-     */
-    public function getCrowdBy()
-    {
-        if (array_key_exists('crowdby', $this->_params)) {
-            return $this->_params['crowdby'];
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * @return string adjust
-     */
-    public function getAdjust()
-    {
-        if (array_key_exists('adjust', $this->_params)) {
-            return $this->_params['adjust'];
-        } else {
-            return null;
-        }
-    }
-
 }
Index: library/Zend/Gdata/Gbase/Extension/BaseAttribute.php
===================================================================
--- library/Zend/Gdata/Gbase/Extension/BaseAttribute.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/Extension/BaseAttribute.php	(working copy)
@@ -36,15 +36,7 @@
  */
 class Zend_Gdata_Gbase_Extension_BaseAttribute extends Zend_Gdata_App_Extension_Element
 {
-
     /**
-     * Namespace for Google Base elements
-     *
-     * var @string
-     */
-    protected $_rootNamespace = 'g';
-
-    /**
      * Create a new instance.
      *
      * @param string $name (optional) The name of the Base attribute
@@ -53,63 +45,10 @@
      */
     public function __construct($name = null, $text = null, $type = null)
     {
-        $this->registerAllNamespaces(Zend_Gdata_Gbase::$namespaces);
-        if ($type !== null) {
-          $attr = array('name' => 'type', 'value' => $type);
-          $typeAttr = array('type' => $attr);
-          $this->setExtensionAttributes($typeAttr);
-        }
-        parent::__construct($name,
-                            $this->_rootNamespace,
-                            $this->lookupNamespace($this->_rootNamespace),
-                            $text);
+        throw new Zend_Exception(
+            'Google Base API has been discontinued by Google and was removed'
+            . ' from Zend Framework in 1.12.0.  For more information see: '
+            . 'http://googlemerchantblog.blogspot.ca/2010/12/new-shopping-apis-and-deprecation-of.html'
+        );    
     }
-
-    /**
-     * Get the name of the attribute
-     *
-     * @return attribute name The requested object.
-     */
-    public function getName() {
-      return $this->_rootElement;
-    }
-
-    /**
-     * Get the type of the attribute
-     *
-     * @return attribute type The requested object.
-     */
-    public function getType() {
-      $typeAttr = $this->getExtensionAttributes();
-      return $typeAttr['type']['value'];
-    }
-
-    /**
-     * Set the 'name' of the Base attribute object:
-     *     &lt;g:[$name] type='[$type]'&gt;[$value]&lt;/g:[$name]&gt;
-     *
-     * @param Zend_Gdata_App_Extension_Element $attribute The attribute object
-     * @param string $name The name of the Base attribute
-     * @return Zend_Gdata_Extension_ItemEntry Provides a fluent interface
-     */
-    public function setName($name) {
-      $this->_rootElement = $name;
-      return $this;
-    }
-
-    /**
-     * Set the 'type' of the Base attribute object:
-     *     &lt;g:[$name] type='[$type]'&gt;[$value]&lt;/g:[$name]&gt;
-     *
-     * @param Zend_Gdata_App_Extension_Element $attribute The attribute object
-     * @param string $type The type of the Base attribute
-     * @return Zend_Gdata_Extension_ItemEntry Provides a fluent interface
-     */
-    public function setType($type) {
-      $attr = array('name' => 'type', 'value' => $type);
-      $typeAttr = array('type' => $attr);
-      $this->setExtensionAttributes($typeAttr);
-      return $this;
-    }
-
 }
Index: library/Zend/Gdata/Gbase/SnippetFeed.php
===================================================================
--- library/Zend/Gdata/Gbase/SnippetFeed.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/SnippetFeed.php	(working copy)
@@ -22,11 +22,16 @@
  */
 
 /**
- * @see Zend_Gdata_Gbase_Feed
+ * @see Zend_Exception
  */
-require_once 'Zend/Gdata/Gbase/Feed.php';
+require_once 'Zend/Exception.php';
 
 /**
+ * @see Zend_Gdata_Feed
+ */
+require_once 'Zend/Gdata/Feed.php';
+
+/**
  * Represents the Google Base Snippets Feed
  *
  * @link http://code.google.com/apis/base/
@@ -39,10 +44,4 @@
  */
 class Zend_Gdata_Gbase_SnippetFeed extends Zend_Gdata_Feed
 {
-    /**
-     * The classname for individual snippet feed elements.
-     *
-     * @var string
-     */
-    protected $_entryClassName = 'Zend_Gdata_Gbase_SnippetEntry';
 }
Index: library/Zend/Gdata/Gbase/SnippetQuery.php
===================================================================
--- library/Zend/Gdata/Gbase/SnippetQuery.php	(revision 24746)
+++ library/Zend/Gdata/Gbase/SnippetQuery.php	(working copy)
@@ -22,16 +22,11 @@
  */
 
 /**
- * Zend_Gdata_Query
+ * @see Zend_Gdata_Gbase_Query
  */
-require_once('Zend/Gdata/Query.php');
+require_once 'Zend/Gdata/Gbase/Query.php';
 
 /**
- * Zend_Gdata_Gbase_Query
- */
-require_once('Zend/Gdata/Gbase/Query.php');
-
-/**
  * Assists in constructing queries for Google Base Snippets Feed
  *
  * @link http://code.google.com/apis/base/
@@ -48,27 +43,4 @@
      * Path to the snippets feeds on the Google Base server.
      */
     const BASE_SNIPPET_FEED_URI = 'https://www.google.com/base/feeds/snippets';
-
-    /**
-     * The default URI for POST methods
-     *
-     * @var string
-     */
-    protected $_defaultFeedUri = self::BASE_SNIPPET_FEED_URI;
-
-    /**
-     * Returns the query URL generated by this query instance.
-     *
-     * @return string The query URL for this instance.
-     */
-    public function getQueryUrl()
-    {
-        $uri = $this->_defaultFeedUri;
-        if ($this->getCategory() !== null) {
-            $uri .= '/-/' . $this->getCategory();
-        }
-        $uri .= $this->getQueryString();
-        return $uri;
-    }
-
 }
Index: library/Zend/Gdata/Gbase.php
===================================================================
--- library/Zend/Gdata/Gbase.php	(revision 24746)
+++ library/Zend/Gdata/Gbase.php	(working copy)
@@ -22,31 +22,16 @@
  */
 
 /**
- * @see Zend_Gdata
+ * @see Zend_Exception
  */
-require_once 'Zend/Gdata.php';
+require_once 'Zend/Exception.php';
 
 /**
- * @see Zend_Gdata_Gbase_ItemFeed
+ * @see Zend_Gdata
  */
-require_once 'Zend/Gdata/Gbase/ItemFeed.php';
+require_once 'Zend/Gdata.php';
 
 /**
- * @see Zend_Gdata_Gbase_ItemEntry
- */
-require_once 'Zend/Gdata/Gbase/ItemEntry.php';
-
-/**
- * @see Zend_Gdata_Gbase_SnippetEntry
- */
-require_once 'Zend/Gdata/Gbase/SnippetEntry.php';
-
-/**
- * @see Zend_Gdata_Gbase_SnippetFeed
- */
-require_once 'Zend/Gdata/Gbase/SnippetFeed.php';
-
-/**
  * Service class for interacting with the Google Base data API
  *
  * @link http://code.google.com/apis/base
@@ -76,23 +61,6 @@
     const AUTH_SERVICE_NAME = 'gbase';
 
     /**
-     * The default URI for POST methods
-     *
-     * @var string
-     */
-    protected $_defaultPostUri = self::GBASE_ITEM_FEED_URI;
-
-    /**
-     * Namespaces used for Zend_Gdata_Gbase
-     *
-     * @var array
-     */
-    public static $namespaces = array(
-        array('g', 'http://base.google.com/ns/1.0', 1, 0),
-        array('batch', 'http://schemas.google.com/gdata/batch', 1, 0)
-    );
-
-    /**
      * Create Zend_Gdata_Gbase object
      *
      * @param Zend_Http_Client $client (optional) The HTTP client to use when
@@ -101,109 +69,10 @@
      */
     public function __construct($client = null, $applicationId = 'MyCompany-MyApp-1.0')
     {
-        $this->registerPackage('Zend_Gdata_Gbase');
-        $this->registerPackage('Zend_Gdata_Gbase_Extension');
-        parent::__construct($client, $applicationId);
-        $this->_httpClient->setParameterPost('service', self::AUTH_SERVICE_NAME);
+        throw new Zend_Exception(
+            'Google Base API has been discontinued by Google and was removed'
+            . ' from Zend Framework in 1.12.0.  For more information see: '
+            . 'http://googlemerchantblog.blogspot.ca/2010/12/new-shopping-apis-and-deprecation-of.html'
+        );
     }
-
-    /**
-     * Retreive feed object
-     *
-     * @param mixed $location The location for the feed, as a URL or Query
-     * @return Zend_Gdata_Gbase_ItemFeed
-     */
-    public function getGbaseItemFeed($location = null)
-    {
-        if ($location === null) {
-            $uri = self::GBASE_ITEM_FEED_URI;
-        } else if ($location instanceof Zend_Gdata_Query) {
-            $uri = $location->getQueryUrl();
-        } else {
-            $uri = $location;
-        }
-        return parent::getFeed($uri, 'Zend_Gdata_Gbase_ItemFeed');
-    }
-
-    /**
-     * Retreive entry object
-     *
-     * @param mixed $location The location for the feed, as a URL or Query
-     * @return Zend_Gdata_Gbase_ItemEntry
-     */
-    public function getGbaseItemEntry($location = null)
-    {
-        if ($location === null) {
-            require_once 'Zend/Gdata/App/InvalidArgumentException.php';
-            throw new Zend_Gdata_App_InvalidArgumentException(
-                    'Location must not be null');
-        } else if ($location instanceof Zend_Gdata_Query) {
-            $uri = $location->getQueryUrl();
-        } else {
-            $uri = $location;
-        }
-        return parent::getEntry($uri, 'Zend_Gdata_Gbase_ItemEntry');
-    }
-
-    /**
-     * Insert an entry
-     *
-     * @param Zend_Gdata_Gbase_ItemEntry $entry The Base entry to upload
-     * @param boolean $dryRun Flag for the 'dry-run' parameter
-     * @return Zend_Gdata_Gbase_ItemFeed
-     */
-    public function insertGbaseItem($entry, $dryRun = false)
-    {
-        if ($dryRun == false) {
-            $uri = $this->_defaultPostUri;
-        } else {
-            $uri = $this->_defaultPostUri . '?dry-run=true';
-        }
-        $newitem = $this->insertEntry($entry, $uri, 'Zend_Gdata_Gbase_ItemEntry');
-        return $newitem;
-    }
-
-    /**
-     * Update an entry
-     *
-     * @param Zend_Gdata_Gbase_ItemEntry $entry The Base entry to be updated
-     * @param boolean $dryRun Flag for the 'dry-run' parameter
-     * @return Zend_Gdata_Gbase_ItemEntry
-     */
-    public function updateGbaseItem($entry, $dryRun = false)
-    {
-        $returnedEntry = $entry->save($dryRun);
-        return $returnedEntry;
-    }
-
-    /**
-     * Delete an entry
-     *
-     * @param Zend_Gdata_Gbase_ItemEntry $entry The Base entry to remove
-     * @param boolean $dryRun Flag for the 'dry-run' parameter
-     * @return Zend_Gdata_Gbase_ItemFeed
-     */
-    public function deleteGbaseItem($entry, $dryRun = false)
-    {
-        $entry->delete($dryRun);
-        return $this;
-    }
-
-    /**
-     * Retrieve feed object
-     *
-     * @param mixed $location The location for the feed, as a URL or Query
-     * @return Zend_Gdata_Gbase_SnippetFeed
-     */
-    public function getGbaseSnippetFeed($location = null)
-    {
-        if ($location === null) {
-            $uri = self::GBASE_SNIPPET_FEED_URI;
-        } else if ($location instanceof Zend_Gdata_Query) {
-            $uri = $location->getQueryUrl();
-        } else {
-            $uri = $location;
-        }
-        return parent::getFeed($uri, 'Zend_Gdata_Gbase_SnippetFeed');
-    }
 }
Index: documentation/manual/de/module_specs/Zend_Gdata-Introduction.xml
===================================================================
--- documentation/manual/de/module_specs/Zend_Gdata-Introduction.xml	(revision 24746)
+++ documentation/manual/de/module_specs/Zend_Gdata-Introduction.xml	(working copy)
@@ -61,13 +61,6 @@
 
             <listitem>
                 <para>
-                    <link linkend="zend.gdata.gbase">Google Base</link> bietet die Möglichkeit
-                    Elemente in Google Base zu empfangen, senden, aktualisieren und zu löschen.
-                </para>
-            </listitem>
-
-            <listitem>
-                <para>
                     <link linkend="zend.gdata.youtube">YouTube</link> bietet die Möglichkeit Videos,
                     Kommentare, Favoriten, Einschreibungen, Benutzerprofile und vieles mehr zu
                     Suchen und zu Empfangen.
Index: documentation/manual/ja/module_specs/Zend_Gdata-Introduction.xml
===================================================================
--- documentation/manual/ja/module_specs/Zend_Gdata-Introduction.xml	(revision 24746)
+++ documentation/manual/ja/module_specs/Zend_Gdata-Introduction.xml	(working copy)
@@ -59,13 +59,6 @@
             </listitem>
             <listitem>
                 <para>
-                    <link linkend="zend.gdata.gbase">Google Base</link>
-                    は、Google Base のアイテムを取得したり
-                    アイテムを投稿、更新、あるいは削除したりできます。
-                </para>
-            </listitem>
-            <listitem>
-                <para>
                     <link linkend="zend.gdata.youtube">YouTube</link>
                     は、動画やコメント、お気に入り、登録チャンネル、
                     ユーザのプロファイルといった情報を検索して取得できます。
Index: documentation/manual/en/module_specs/Zend_Gdata-Introduction.xml
===================================================================
--- documentation/manual/en/module_specs/Zend_Gdata-Introduction.xml	(revision 24746)
+++ documentation/manual/en/module_specs/Zend_Gdata-Introduction.xml	(working copy)
@@ -60,14 +60,6 @@
 
             <listitem>
                 <para>
-                    <link linkend="zend.gdata.gbase">Google Base</link>
-                    provides the ability to retrieve, post, update, and
-                    delete items in Google Base.
-                </para>
-            </listitem>
-
-            <listitem>
-                <para>
                     <link linkend="zend.gdata.youtube">YouTube</link>
                     provides the ability to search and retrieve videos,
                     comments, favorites, subscriptions, user profiles
Index: documentation/manual/en/manual-print1.xml.in
===================================================================
--- documentation/manual/en/manual-print1.xml.in	(revision 24746)
+++ documentation/manual/en/manual-print1.xml.in	(working copy)
@@ -240,10 +240,8 @@
         <xi:include href="module_specs/Zend_Gdata_ClientLogin.xml" />
         <xi:include href="module_specs/Zend_Gdata_Calendar.xml" />
         <xi:include href="module_specs/Zend_Gdata_Docs.xml" />
-        <xi:include href="module_specs/Zend_Gdata_Health.xml" />
         <xi:include href="module_specs/Zend_Gdata_Spreadsheets.xml" />
         <xi:include href="module_specs/Zend_Gdata_Gapps.xml" />
-        <xi:include href="module_specs/Zend_Gdata_Gbase.xml" />
         <xi:include href="module_specs/Zend_Gdata_Photos.xml" />
         <xi:include href="module_specs/Zend_Gdata_YouTube.xml" />
         <xi:include href="module_specs/Zend_Gdata_Exception.xml" />
