--- File.php	2007-10-17 13:38:42.000000000 +0200
+++ File.php.new	2007-10-17 13:40:03.000000000 +0200
@@ -221,6 +221,7 @@
     {
         if ((!is_dir($this->_options['cache_dir'])) or (!is_writable($this->_options['cache_dir']))) {
             $this->_log("Zend_Cache_Backend_File::save() : cache_dir doesn't exist or is not writable");
+            return false;
         }
         $this->remove($id); // to avoid multiple files with the same cache id
         $lifetime = $this->getLifetime($specificLifetime);
