Index: DefaultValueTest.php
===================================================================
--- DefaultValueTest.php	(revision 24717)
+++ DefaultValueTest.php	(working copy)
@@ -95,24 +95,21 @@
                 ),
             new Zend_CodeGenerator_Php_Property_DefaultValue(array('value' => 'PHP_EOL', 'type' => 'constant'))
             );
+        $expectedSource = "array(" . PHP_EOL . 
+"        5," . PHP_EOL . 
+"        'one' => 1," . PHP_EOL . 
+"        'two' => '2'," . PHP_EOL . 
+"        array(" . PHP_EOL . 
+"            'foo'," . PHP_EOL . 
+"            'bar'," . PHP_EOL . 
+"            array(" . PHP_EOL . 
+"                'baz1'," . PHP_EOL . 
+"                'baz2'" . PHP_EOL . 
+"                )" . PHP_EOL . 
+"            )," . PHP_EOL . 
+"        PHP_EOL" . PHP_EOL . 
+"        );";
 
-        $expectedSource = <<<EOS
-array(
-        5,
-        'one' => 1,
-        'two' => '2',
-        array(
-            'foo',
-            'bar',
-            array(
-                'baz1',
-                'baz2'
-                )
-            ),
-        PHP_EOL
-        );
-EOS;
-
         $propDefaultValue = new Zend_CodeGenerator_Php_Property_DefaultValue();
         $propDefaultValue->setValue($targetValue);
         $generatedTargetSource = $propDefaultValue->generate();
