Index: IsCompressedTest.php
===================================================================
--- IsCompressedTest.php	(revision 24717)
+++ IsCompressedTest.php	(working copy)
@@ -69,6 +69,11 @@
                 );
         }
 
+        // Prevent error in the next check
+        if (!function_exists('mime_content_type')) {
+            $this->markTestSkipped('mime_content_type function is not available.');
+        }
+        
         // Sometimes mime_content_type() gives application/zip and sometimes 
         // application/x-zip ...
         $expectedMimeType = mime_content_type(dirname(__FILE__) . '/_files/test.zip');
