Index: Zend/Db/Table/Row.php
===================================================================
--- Zend/Db/Table/Row.php	(revision 3365)
+++ Zend/Db/Table/Row.php	(working copy)
@@ -43,7 +43,7 @@
     /**
      * Zend_Db_Adapter object from the table interface.
      *
-     * @var Zend_Db_Adapter
+     * @var Zend_Db_Adapter_Abstract
      */
     protected $_db;
 
@@ -139,7 +139,8 @@
             $this->_data[$primary] = null;
 
             // attempt the insert.
-            $result = $this->_table->insert($this->_data);
+            $this->_table->insert($this->_data);
+            $result = $this->_db->lastInsertId();
             if (is_numeric($result)) {
                 // insert worked, refresh with data from the table
                 $this->_data[$primary] = $result;
