Index: View/Helper/FormCheckbox.php
===================================================================
--- View/Helper/FormCheckbox.php	(revision 19162)
+++ View/Helper/FormCheckbox.php	(working copy)
@@ -64,15 +64,13 @@
         $info = $this->_getInfo($name, $value, $attribs);
         extract($info); // name, id, value, attribs, options, listsep, disable
 
-        $checked = false;
-        if (isset($attribs['checked']) && $attribs['checked']) {
-            $checked = true;
-            unset($attribs['checked']);
-        } elseif (isset($attribs['checked'])) {
-            $checked = false;
-            unset($attribs['checked']);
+        if (!isset($attribs['checked'])) {
+            $attribs['checked'] = false;
         }
 
+        $checked = (bool)$attribs['checked'];
+        unset($attribs['checked']);
+
         $checkedOptions = self::determineCheckboxInfo($value, $checked, $checkedOptions);
 
         // is the element disabled?
