diff --git a/tests/Zend/Controller/Plugin/ActionStackTest.php b/tests/Zend/Controller/Plugin/ActionStackTest.php
old mode 100644
new mode 100755
index aa05127..a98b32c
--- a/tests/Zend/Controller/Plugin/ActionStackTest.php
+++ b/tests/Zend/Controller/Plugin/ActionStackTest.php
@@ -15,7 +15,7 @@ require_once 'Zend/Registry.php';
 /**
  * Test class for Zend_Controller_Plugin_ActionStack.
  */
-class Zend_Controller_Plugin_ActionStackTest extends PHPUnit_Framework_TestCase 
+class Zend_Controller_Plugin_ActionStackTest extends PHPUnit_Framework_TestCase
 {
     public $key       = 'Zend_Controller_Plugin_ActionStack';
     public $registry;
@@ -253,6 +253,30 @@ class Zend_Controller_Plugin_ActionStackTest extends PHPUnit_Framework_TestCase
         $this->assertFalse($request->isDispatched());
     }
 
+
+    public function testForwardResetsRequestParamsIfFlagSet()
+    {
+        $plugin   = new Zend_Controller_Plugin_ActionStack();
+        $request  = $this->getNewRequest();
+        $params = array('foo' => 'bar','baz'=>'bat');
+        $request->setParams($params);
+        $plugin->setRequest($request);
+
+        $this->assertEquals($params,$plugin->getRequest()->getParams());
+
+        $next  = $this->getNewRequest();
+        $plugin->forward($next);
+
+        $this->assertEquals($params,$plugin->getRequest()->getParams());
+
+        $plugin->setClearRequestParams(true);
+
+        $next  = $this->getNewRequest();
+        $plugin->forward($next);
+
+        $this->assertEquals(array(),$plugin->getRequest()->getParams());
+    }
+
     /**
      * @return void
      */
diff --git a/tests/Zend/Controller/Request/HttpTest.php b/tests/Zend/Controller/Request/HttpTest.php
old mode 100644
new mode 100755
index bc6639a..7646e05
--- a/tests/Zend/Controller/Request/HttpTest.php
+++ b/tests/Zend/Controller/Request/HttpTest.php
@@ -173,6 +173,14 @@ class Zend_Controller_Request_HttpTest extends PHPUnit_Framework_TestCase
         $this->assertEquals('bar', $this->_request->getParam('foo'));
     }
 
+
+    public function testClearParams()
+    {
+        $this->_request->setParam('foo', 'bar');
+        $this->_request->clearParams();
+        $this->assertNull($this->_request->getParam('foo'));
+    }
+
     public function testSetGetParams()
     {
         $params = array(
