Index: library/Zend/Feed/Writer/Feed/FeedAbstract.php
===================================================================
--- library/Zend/Feed/Writer/Feed/FeedAbstract.php	(revision 25051)
+++ library/Zend/Feed/Writer/Feed/FeedAbstract.php	(working copy)
@@ -117,7 +117,7 @@
                 $author['uri'] = $name['uri'];
             }
         } else {
-            if (empty($name['name']) || !is_string($name['name'])) {
+            if (empty($name) || !is_string($name)) {
                 require_once 'Zend/Feed/Exception.php';
                 throw new Zend_Feed_Exception('Invalid parameter: "name" must be a non-empty string value');
             }
