Index: Client.php
===================================================================
--- Client.php	(revision 9440)
+++ Client.php	(working copy)
@@ -199,10 +199,10 @@
 
     /**
      * Contructor method. Will create a new HTTP client. Accepts the target
-     * URL and optionally and array of headers.
+     * URL and optionally configuration array.
      *
      * @param Zend_Uri_Http|string $uri
-     * @param array $headers Optional request headers to set
+     * @param array $config Configuration key-value pairs.
      */
     public function __construct($uri = null, $config = null)
     {
@@ -258,6 +258,7 @@
      *
      * @param array $config
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setConfig($config = array())
     {
@@ -281,6 +282,7 @@
      *
      * @param string $method
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setMethod($method = self::GET)
     {
@@ -314,6 +316,7 @@
      *     or an array of headers
      * @param mixed $value Header value or null
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setHeaders($name, $value = null)
     {
@@ -360,7 +363,7 @@
      * Note that if the header has more than one value, an array
      * will be returned.
      *
-     * @param unknown_type $key
+     * @param string $key
      * @return string|array|null The header value or null if it is not set
      */
     public function getHeader($key)
@@ -417,6 +420,7 @@
      * @param string $type GET or POST
      * @param string $name
      * @param string $value
+     * @return null
      */
     protected function _setParameter($type, $name, $value)
     {
@@ -469,6 +473,7 @@
      * @param string $password Password
      * @param string $type Authentication type
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setAuth($user, $password = '', $type = self::AUTH_BASIC)
     {
@@ -502,6 +507,7 @@
      *
      * @param Zend_Http_CookieJar|boolean $cookiejar Existing cookiejar object, true to create a new one, false to disable
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setCookieJar($cookiejar = true)
     {
@@ -539,6 +545,7 @@
      * @param Zend_Http_Cookie|string $cookie
      * @param string|null $value If "cookie" is a string, this is the cookie value.
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setCookie($cookie, $value = null)
     {
@@ -605,6 +612,7 @@
      * @param string $ctype Content type to use (if $data is set and $ctype is
      *     null, will be application/octet-stream)
      * @return Zend_Http_Client
+     * @throws Zend_Http_Client_Exception
      */
     public function setFileUpload($filename, $formname, $data = null, $ctype = null)
     {
@@ -712,6 +720,8 @@
      * seperated from ->request() to preserve logic and readability
      *
      * @param Zend_Http_Client_Adapter_Interface|string $adapter
+     * @return null
+     * @throws Zend_Http_Client_Exception
      */
     public function setAdapter($adapter)
     {
@@ -742,6 +752,7 @@
      *
      * @param string $method
      * @return Zend_Http_Response
+     * @throws Zend_Http_Client_Exception
      */
     public function request($method = null)
     {
@@ -922,6 +933,7 @@
      * Prepare the request body (for POST and PUT requests)
      *
      * @return string
+     * @throws Zend_Http_Client_Exception
      */
     protected function prepare_body()
     {
@@ -1054,6 +1066,7 @@
      * @param string $password
      * @param string $type
      * @return string
+     * @throws Zend_Http_Client_Exception
      */
     public static function encodeAuthHeader($user, $password, $type = self::AUTH_BASIC)
     {
