Index: ArrayOfTypeComplex.php
===================================================================
--- ArrayOfTypeComplex.php	(revision 20686)
+++ ArrayOfTypeComplex.php	(working copy)
@@ -47,8 +47,7 @@
     public function addComplexType($type)
     {
         if(in_array($type, $this->_inProcess)) {
-            require_once "Zend/Soap/Wsdl/Exception.php";
-            throw new Zend_Soap_Wsdl_Exception("Infinite recursion, cannot nest '".$type."' into itsself.");
+            return "tns:$type";
         }
         $this->_inProcess[$type] = $type;
 
