Index: tests/Zend/Gdata/Calendar/LinkTest.php
===================================================================
--- tests/Zend/Gdata/Calendar/LinkTest.php	(revision 5372)
+++ tests/Zend/Gdata/Calendar/LinkTest.php	(working copy)
@@ -66,26 +66,26 @@
         $newLink->extensionElements = array(
                 new Zend_Gdata_App_Extension_Element('foo', 'atom', null, 'bar'));
         $this->assertEquals(count($newLink->extensionElements), 1);
-        $this->assertEquals($this->link->rel, "http://nowhere.invalid/");
-        $this->assertEquals($this->link->title, "Somewhere");
-        $this->assertEquals($this->link->href, "http://somewhere.invalid/");
-        $this->assertEquals($this->link->type, "text/plain");
-        $this->assertEquals($this->link->webcontent->url, "a");
-        $this->assertEquals($this->link->webcontent->height, "1");
-        $this->assertEquals($this->link->webcontent->width, "2");
+        $this->assertEquals($newLink->rel, "http://nowhere.invalid/");
+        $this->assertEquals($newLink->title, "Somewhere");
+        $this->assertEquals($newLink->href, "http://somewhere.invalid/");
+        $this->assertEquals($newLink->type, "text/plain");
+        $this->assertEquals($newLink->webcontent->url, "a");
+        $this->assertEquals($newLink->webcontent->height, "1");
+        $this->assertEquals($newLink->webcontent->width, "2");
 
         /* try constructing using magic factory */
         $cal = new Zend_Gdata_Calendar();
         $newLink2 = $cal->newLink();
         $newLink2->transferFromXML($newLink->saveXML());
         $this->assertEquals(count($newLink2->extensionElements), 1);
-        $this->assertEquals($this->link->rel, "http://nowhere.invalid/");
-        $this->assertEquals($this->link->title, "Somewhere");
-        $this->assertEquals($this->link->href, "http://somewhere.invalid/");
-        $this->assertEquals($this->link->type, "text/plain");
-        $this->assertEquals($this->link->webcontent->url, "a");
-        $this->assertEquals($this->link->webcontent->height, "1");
-        $this->assertEquals($this->link->webcontent->width, "2");
+        $this->assertEquals($newLink2->rel, "http://nowhere.invalid/");
+        $this->assertEquals($newLink2->title, "Somewhere");
+        $this->assertEquals($newLink2->href, "http://somewhere.invalid/");
+        $this->assertEquals($newLink2->type, "text/plain");
+        $this->assertEquals($newLink2->webcontent->url, "a");
+        $this->assertEquals($newLink2->webcontent->height, "1");
+        $this->assertEquals($newLink2->webcontent->width, "2");
     }
 
     public function testEmptyLinkToAndFromStringShouldMatch() {
Index: tests/Zend/Gdata/Calendar/WebContentTest.php
===================================================================
--- tests/Zend/Gdata/Calendar/WebContentTest.php	(revision 5372)
+++ tests/Zend/Gdata/Calendar/WebContentTest.php	(working copy)
@@ -20,7 +20,6 @@
  */
 
 require_once 'Zend/Gdata/Calendar/Extension/WebContent.php';
-require_once 'Zend/Gdata/Calendar/Extension/WebContent.php';
 require_once 'Zend/Gdata/Calendar.php';
 
 /**
@@ -60,18 +59,18 @@
         $newWebContent->extensionElements = array(
                 new Zend_Gdata_App_Extension_Element('foo', 'atom', null, 'bar'));
         $this->assertEquals(count($newWebContent->extensionElements), 1);
-        $this->assertEquals($this->webContent->url, "http://nowhere.invalid/");
-        $this->assertEquals($this->webContent->height, "100");
-        $this->assertEquals($this->webContent->width, "200");
+        $this->assertEquals($newWebContent->url, "http://nowhere.invalid/");
+        $this->assertEquals($newWebContent->height, "100");
+        $this->assertEquals($newWebContent->width, "200");
 
         /* try constructing using magic factory */
         $cal = new Zend_Gdata_Calendar();
         $newWebContent2 = $cal->newWebContent();
         $newWebContent2->transferFromXML($newWebContent->saveXML());
         $this->assertEquals(count($newWebContent2->extensionElements), 1);
-        $this->assertEquals($this->webContent->url, "http://nowhere.invalid/");
-        $this->assertEquals($this->webContent->height, "100");
-        $this->assertEquals($this->webContent->width, "200");
+        $this->assertEquals($newWebContent2->url, "http://nowhere.invalid/");
+        $this->assertEquals($newWebContent2->height, "100");
+        $this->assertEquals($newWebContent2->width, "200");
     }
 
     public function testEmptyWebContentToAndFromStringShouldMatch() {
