Index: File.php
===================================================================
--- File.php	(revision 6822)
+++ File.php	(working copy)
@@ -213,17 +213,17 @@
         $file = $this->_file($id, $expire);
         $path = $this->_path($id);
         $firstTry = true;
-        $result = false;      
+        $result = false;
         if ($this->_options['hashed_directory_level'] > 0) {
             if (!is_writable($path)) {
                 // maybe, we just have to build the directory structure
                 @mkdir($this->_path($id), $this->_options['hashed_directory_umask'], true);
-                @chmod($this->_path($id), $this->_options['hashed_directory_umask']); // see #ZF-320 (this line is required in some configurations)      
+                @chmod($this->_path($id), $this->_options['hashed_directory_umask']); // see #ZF-320 (this line is required in some configurations)
             }
             if (!is_writable($path)) {
-                return false;    
+                return false;
             }
-        }        
+        }
         $fp = @fopen($file, "wb");
         if (!($fp)) {
             return false;
@@ -452,6 +452,9 @@
     {
         $filesToRemove = @glob($this->_file($this->_tagCacheId($id, '*'), '*'));
         $result = true;
+        if (!is_array($filesToRemove) || count($filesToRemove) < 1) {
+            return $result;
+        }
         foreach ($filesToRemove as $file) {
             $result = $result && ($this->_remove($file));
         }
