Index: tests/Zend/Form/Element/SubmitTest.php
===================================================================
--- tests/Zend/Form/Element/SubmitTest.php	(revision 20216)
+++ tests/Zend/Form/Element/SubmitTest.php	(working copy)
@@ -66,6 +66,7 @@
      */
     public function setUp()
     {
+        Zend_Form::setDefaultTranslator(null);
         $this->element = new Zend_Form_Element_Submit('foo');
     }
 

