Index: Client.php
===================================================================
--- Client.php	(revision 3512)
+++ Client.php	(working copy)
@@ -771,7 +771,7 @@
                         // Else, assume we have a relative path
                     } else {
                         // Get the current path directory, removing any trailing slashes
-                        $path = rtrim(dirname($this->uri->getPath()), "/");
+                        $path = rtrim(str_replace('\\', '/', dirname($this->uri->getPath())), "/");
                         $this->uri->setPath($path . '/' . $location);
                     }
                 }
