Index: library/Zend/Application/Application.php
===================================================================
--- library/Zend/Application/Application.php	(date 1314008990000)
+++ library/Zend/Application/Application.php	(revision )
@@ -174,7 +174,7 @@
             $this->setAutoloaderNamespaces($options['autoloadernamespaces']);
         }
 
-        if (!empty($option['autoloaderprefixes'])) {
+        if (!empty($options['autoloaderprefixes'])) {
             $this->setAutoloaderPrefixes($options['autoloaderprefixes']);
         }
 
