--- ResourceAbstract.php.orig	Tue May 12 10:45:23 2009
+++ ResourceAbstract.php	Sun May 24 18:31:41 2009
@@ -21,6 +21,11 @@
  */
 
 /**
+ * @see Zend_Application
+ */
+require_once 'Zend/Application.php';
+
+/**
  * @see Zend_Application_Resource_Resource
  */
 require_once 'Zend/Application/Resource/Resource.php';
@@ -97,7 +102,7 @@
             }
         }
         
-        $this->_options = $this->mergeOptions($this->_options, $options);
+        $this->_options = Zend_Application::mergeOptions($this->_options, $options);
 
         return $this;
     }
@@ -121,18 +126,7 @@
      */
     public function mergeOptions(array $array1, $array2 = null)
     {
-        if (is_array($array2)) {
-            foreach ($array2 as $key => $val) {
-                if (is_array($array2[$key])) {
-                    $array1[$key] = (array_key_exists($key, $array1) && is_array($array1[$key]))
-                                  ? $this->mergeOptions($array1[$key], $array2[$key]) 
-                                  : $array2[$key];
-                } else {
-                    $array1[$key] = $val;
-                }
-            }
-        }
-        return $array1;
+        return Zend_Application::mergeOptions($array1, $array2);
     }
 
     /**
