Index: tests/Zend/Rest/RouteTest.php
===================================================================
--- tests/Zend/Rest/RouteTest.php	(revision 17337)
+++ tests/Zend/Rest/RouteTest.php	(working copy)
@@ -121,6 +121,18 @@
         $this->assertEquals('index', $values['action']);
     }
 
+    public function test_RESTfulApp_GET_user_defaults()
+    {
+        $request = $this->_buildRequest('GET', '/user');
+        $values = $this->_invokeRouteMatch($request);
+         
+        $this->assertType('array', $values);
+        $this->assertTrue(isset($values['module']));
+        $this->assertEquals('default', $values['module']);
+        $this->assertEquals('user', $values['controller']);
+        $this->assertEquals('index', $values['action']);
+    }
+
     public function test_RESTfulApp_GET_user_index_withParams()
     {
         $request = $this->_buildRequest('GET', '/user/index/changedSince/123456789/status/active');
Index: library/Zend/Rest/Route.php
===================================================================
--- library/Zend/Rest/Route.php	(revision 17337)
+++ library/Zend/Rest/Route.php	(working copy)
@@ -141,7 +141,7 @@
             } elseif ($pathElementCount == 1) {
                  $params['id'] = array_shift($path);
             } elseif ($pathElementCount == 0 || $pathElementCount > 1) {
-                $specialGetTarget = 'list';
+                $specialGetTarget = 'index';
             }
  
             // Digest URI params 
