*** Zend/ProgressBar/Adapter/JsPull.php	2009-04-27 14:42:21.000000000 +0200
--- _Zend/ProgressBar/Adapter/JsPull.php	2009-04-27 14:30:01.000000000 +0200
***************
*** 70,79 ****
       */
      public function notify($current, $max, $percent, $timeTaken, $timeRemaining, $text)
      {
          $arguments = array(
              'current'       => $current,
              'max'           => $max,
!             'percent'       => ($percent * 100),
              'timeTaken'     => $timeTaken,
              'timeRemaining' => $timeRemaining,
              'text'          => $text,
--- 70,80 ----
       */
      public function notify($current, $max, $percent, $timeTaken, $timeRemaining, $text)
      {
+         $convert = localeconv();
          $arguments = array(
              'current'       => $current,
              'max'           => $max,
!             'percent'       => strtr(($percent * 100), $convert['decimal_point'], '.'),
              'timeTaken'     => $timeTaken,
              'timeRemaining' => $timeRemaining,
              'text'          => $text,
