--- library/Zend/Controller/Router/Route.php	(revision 23864)
+++ library/Zend/Controller/Router/Route.php	(working copy)
@@ -236,7 +236,7 @@
                     $count = count($path);
                     for($i = $pos; $i < $count; $i+=2) {
                         $var = urldecode($path[$i]);
-                        if (!isset($this->_wildcardData[$var]) && !isset($this->_defaults[$var]) && !isset($values[$var])) {
+                        if (!isset($this->_wildcardData[$var]) && !isset($values[$var])) {
                             $this->_wildcardData[$var] = (isset($path[$i+1])) ? urldecode($path[$i+1]) : null;
                         }
                     }
