Index: tests/Zend/Date/DateObjectTest.php
===================================================================
--- tests/Zend/Date/DateObjectTest.php	(revision 22533)
+++ tests/Zend/Date/DateObjectTest.php	(working copy)
@@ -544,6 +544,16 @@
         $this->assertTrue(($diff < 2), "Zend_Date_DateObject->_getTime() returned a significantly "
             . "different timestamp than expected: $diff seconds");
     }
+    
+    /**
+     * @group ZF-10107
+     */
+    public function testAddSixthMonthsToJulyIncrementsYear()
+    {
+        $date = new Zend_Date('2011-01-07', null, 'en_US');
+        $date->addMonth(6);
+        $this->assertEquals('2011-01-01', $date->toString('Y-MM-dd'));
+    }
 }
 
 class Zend_Date_DateObjectTestHelper extends Zend_Date
