diff --git a/library/Zend/Controller/Front.php b/library/Zend/Controller/Front.php
index fe50ff9..ded3c3a 100644
--- a/library/Zend/Controller/Front.php
+++ b/library/Zend/Controller/Front.php
@@ -141,7 +141,7 @@ class Zend_Controller_Front
      *
      * @return void
      */
-    private function __construct()
+    protected function __construct()
     {
         $this->_plugins = new Zend_Controller_Plugin_Broker();
     }
