--- library/Zend/Form.php	(revision 24057)
+++ library/Zend/Form.php	(working copy)
@@ -2674,6 +2674,9 @@
             throw new Zend_Form_Exception('Invalid decorator provided to addDecorator; must be string or Zend_Form_Decorator_Interface');
         }
 
+        if (empty($this->_decorators)) {
+            $this->loadDefaultDecorators();
+        }
         $this->_decorators[$name] = $decorator;
 
         return $this;
@@ -3290,9 +3293,11 @@
 
         $decorators = $this->getDecorators();
         if (empty($decorators)) {
+            $this->_decorators['phantom'] = null; // protect against infinite loop of adding decorators
             $this->addDecorator('FormElements')
                  ->addDecorator('HtmlTag', array('tag' => 'dl', 'class' => 'zend_form'))
                  ->addDecorator('Form');
+            unset($this->_decorators['phantom']);
         }
         return $this;
     }
