Index: tests/Zend/Validate/EmailAddressTest.php
===================================================================
--- tests/Zend/Validate/EmailAddressTest.php	(revision 21072)
+++ tests/Zend/Validate/EmailAddressTest.php	(working copy)
@@ -319,12 +319,13 @@
 
         // Are MX checks supported by this system?
         if (!$validator->validateMxSupported()) {
-            return true;
+            $this->markTestSkipped('Testing MX records is not supported with this configuration');
+            return;
         }
 
         $valuesExpected = array(
-            array(true, array('Bob.Jones@zend.com', 'Bob.Jones@studio24.net')),
-            array(false, array('Bob.Jones@madeupdomain242424a.com', 'Bob.Jones@madeupdomain242424b.net'))
+            array(true, array('Bob.Jones@zend.com', 'Bob.Jones@php.net')),
+            array(false, array('Bob.Jones@bad.example.com', 'Bob.Jones@anotherbad.example.net'))
             );
         foreach ($valuesExpected as $element) {
             foreach ($element[1] as $input) {
Index: library/Zend/Validate/EmailAddress.php
===================================================================
--- library/Zend/Validate/EmailAddress.php	(revision 21072)
+++ library/Zend/Validate/EmailAddress.php	(working copy)
@@ -443,8 +443,8 @@
     {
         $result = true;
         $mxHosts = array();
-        getmxrr($this->_hostname, $mxHosts);
-        if ($this->_options['deep'] && function_exists('checkdnsrr')) {
+        $result = getmxrr($this->_hostname, $mxHosts);
+        if ($result && $this->_options['deep'] && function_exists('checkdnsrr')) {
             $validAddress = false;
             $reserved     = true;
             foreach ($mxHosts as $hostname) {
