--- tests/Zend/Controller/Router/Route/ChainTest.php	(revision 23064)
+++ tests/Zend/Controller/Router/Route/ChainTest.php	(working copy)
@@ -32,6 +32,9 @@
 /** Zend_Controller_Router_Rewrite */
 require_once 'Zend/Controller/Router/Rewrite.php';
 
+/** Zend_Controller_Dispatcher_Standard */
+require_once 'Zend/Controller/Dispatcher/Standard.php';
+
 /** Zend_Controller_Router_Route_Chain */
 require_once 'Zend/Controller/Router/Route/Chain.php';
 
@@ -426,8 +429,8 @@
         $router = new Zend_Controller_Router_Rewrite();
         $front = Zend_Controller_Front::getInstance();
         $front->resetInstance();
-        $front->setDispatcher(new Zend_Controller_Router_RewriteTest_Dispatcher());
-        $front->setRequest(new Zend_Controller_Router_RewriteTest_Request());
+        $front->setDispatcher(new Zend_Controller_Router_ChainTest_Dispatcher());
+        $front->setRequest(new Zend_Controller_Router_ChainTest_Request());
         $router->setFrontController($front);
 
         $router->addConfig(new Zend_Config($routes));
@@ -683,6 +686,22 @@
     }
 }
 
+/**
+ * Zend_Controller_Router_ChainTest_Dispatcher - dispatcher object for router testing
+ */
+class Zend_Controller_Router_ChainTest_Dispatcher extends Zend_Controller_Dispatcher_Standard
+{
+    public function getDefaultControllerName()
+    {
+        return 'defctrl';
+    }
+
+    public function getDefaultAction()
+    {
+        return 'defact';
+    }
+}
+
 if (PHPUnit_MAIN_METHOD == "Zend_Controller_Router_Route_ChainTest::main") {
     Zend_Controller_Router_Route_ChainTest::main();
 }
