Index: library/Zend/Service/Amazon/Ec2/Ebs.php
===================================================================
--- library/Zend/Service/Amazon/Ec2/Ebs.php	(revision 21970)
+++ library/Zend/Service/Amazon/Ec2/Ebs.php	(working copy)
@@ -242,7 +242,7 @@
     {
         $params = array();
         $params['Action']       = 'DeleteVolume';
-        $params['volumeId']     = $volumeId;
+        $params['VolumeId']     = $volumeId;
 
         $response = $this->sendRequest($params);
         $xpath = $response->getXPath();
