Index: library/Zend/Db/Adapter/Pdo/Abstract.php
===================================================================
--- library/Zend/Db/Adapter/Pdo/Abstract.php	(revision 670)
+++ library/Zend/Db/Adapter/Pdo/Abstract.php	(working copy)
@@ -75,6 +75,11 @@
             throw new Zend_DB_Adapter_Exception('The PDO extension is required for this adapter but not loaded');
         }
 
+        // check for PDO extension
+        if (!in_array($this->_pdoType, PDO::getAvailableDrivers())) {
+            throw new Zend_DB_Adapter_Exception('The ' . $this->_pdoType . ' driver is not currently installed');
+        }
+
         // create PDO connection
         $q = $this->_profiler->queryStart('connect', Zend_Db_Profiler::CONNECT);
         $this->_connection = new PDO(
