Index: library/Zend/Loader/PluginLoader.php
===================================================================
--- library/Zend/Loader/PluginLoader.php	(revision 12107)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -285,7 +285,7 @@
     public function getClassName($name)
     {
         $name = $this->_formatName($name);
-        if ($this->_useStaticRegistry 
+        if ($this->_useStaticRegistry
             && isset(self::$_staticLoadedPlugins[$this->_useStaticRegistry][$name])
         ) {
             return self::$_staticLoadedPlugins[$this->_useStaticRegistry][$name];
@@ -298,14 +298,14 @@
 
     /**
      * Get path to plugin class
-     * 
-     * @param  mixed $name 
+     *
+     * @param  mixed $name
      * @return string|false False if not found
      */
     public function getClassPath($name)
     {
         $name = $this->_formatName($name);
-        if ($this->_useStaticRegistry 
+        if ($this->_useStaticRegistry
             && isset(self::$_staticLoadedPluginPaths[$this->_useStaticRegistry][$name])
         ) {
             return self::$_staticLoadedPluginPaths[$this->_useStaticRegistry][$name];
@@ -343,11 +343,6 @@
         foreach ($registry as $prefix => $paths) {
             $className = $prefix . $name;
 
-            if (class_exists($className, false)) {
-                $found = true;
-                break;
-            }
-
             $paths     = array_reverse($paths, true);
 
             foreach ($paths as $path) {
@@ -387,10 +382,10 @@
     /**
      * Set path to class file cache
      *
-     * Specify a path to a file that will add include_once statements for each 
+     * Specify a path to a file that will add include_once statements for each
      * plugin class loaded. This is an opt-in feature for performance purposes.
-     * 
-     * @param  string $file 
+     *
+     * @param  string $file
      * @return void
      * @throws Zend_Loader_PluginLoader_Exception if file is not writeable or path does not exist
      */
@@ -419,7 +414,7 @@
 
     /**
      * Retrieve class file cache path
-     * 
+     *
      * @return string|null
      */
     public static function getIncludeFileCache()
@@ -429,8 +424,8 @@
 
     /**
      * Append an include_once statement to the class file cache
-     * 
-     * @param  string $incFile 
+     *
+     * @param  string $incFile
      * @return void
      */
     protected static function _appendIncFile($incFile)
