--- Label.php (24593)
+++ Label.php (working copy)
@@ -29,8 +29,9 @@
  * - separator: separator to use between label and content (defaults to PHP_EOL)
  * - placement: whether to append or prepend label to content (defaults to prepend)
  * - tag: if set, used to wrap the label in an additional HTML tag
+ * - tagClass: if tag option is set, used to add a class to the label wrapper
  * - opt(ional)Prefix: a prefix to the label to use when the element is optional
- * - opt(iona)lSuffix: a suffix to the label to use when the element is optional
+ * - opt(ional)Suffix: a suffix to the label to use when the element is optional
  * - req(uired)Prefix: a prefix to the label to use when the element is required
  * - req(uired)Suffix: a suffix to the label to use when the element is required
  *
