Index: Zend/Controller/Request/Http.php
===================================================================
--- Zend/Controller/Request/Http.php	(revision 950)
+++ Zend/Controller/Request/Http.php	(revision 954)
@@ -416,18 +416,17 @@
 
             // Does the baseUrl have anything in common with the request_uri?
             $requestUri = $this->getRequestUri();
-            if (!strpos($requestUri, basename($baseUrl))) {
+            if (strpos($requestUri, dirname($baseUrl)) !== 0) {
                 $this->_baseUrl = '';
                 return $this;
             }
              
             // If using mod_rewrite or ISAPI_Rewrite strip the script filename 
-            // out of baseUrl. $pos !== 0 makes sure it is not matching a value 
+            // out of baseUrl. $pos === 0 makes sure it is not matching a value 
             // from PATH_INFO or QUERY_STRING 
-            if ((strlen($requestUri) >= strlen($baseUrl))
-                && ((false !== ($pos = strpos($requestUri, $baseUrl))) && ($pos !== 0))) 
+            if (0 === ($pos = strpos($requestUri, dirname($baseUrl)))) 
             { 
-                $baseUrl = substr($requestUri, 0, $pos + strlen($baseUrl));
+                $baseUrl = substr($requestUri, 0, $pos + strlen(dirname($baseUrl)));
             } 
         } 
          
