Index: tests/Zend/View/Helper/HeadLinkTest.php
===================================================================
--- tests/Zend/View/Helper/HeadLinkTest.php	(revision 12040)
+++ tests/Zend/View/Helper/HeadLinkTest.php	(working copy)
@@ -375,7 +375,17 @@
         $test = $this->helper->toString();
         $this->assertNotContains('[if false]', $test);
     }
+
+    /**
+     * @issue ZF-3928
+     * @link http://framework.zend.com/issues/browse/ZF-3928
+     */
+    public function testTurnOffAutoEscapeDoesNotEncodeAmpersand()
+    {
+        $this->helper->setAutoEscape(true)->appendStylesheet('/css/rules.css?id=123&foo=bar');
+        $this->assertContains('id=123&foo=bar', $this->helper->toString());
 }
+}
 
 // Call Zend_View_Helper_HeadLinkTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == "Zend_View_Helper_HeadLinkTest::main") {
Index: tests/Zend/View/Helper/HeadScriptTest.php
===================================================================
--- tests/Zend/View/Helper/HeadScriptTest.php	(revision 12040)
+++ tests/Zend/View/Helper/HeadScriptTest.php	(working copy)
@@ -381,7 +381,17 @@
         }
         $this->helper->headScript()->captureEnd();
     }
+
+    /**
+     * @issue ZF-3928
+     * @link http://framework.zend.com/issues/browse/ZF-3928
+     */
+    public function testTurnOffAutoEscapeDoesNotEncodeAmpersand()
+    {
+        $this->helper->setAutoEscape(false)->appendFile('test.js?id=123&foo=bar');
+        $this->assertEquals('<script type="text/javascript" src="test.js?id=123&foo=bar"></script>', $this->helper->toString());
 }
+}
 
 // Call Zend_View_Helper_HeadScriptTest::main() if this source file is executed directly.
 if (PHPUnit_MAIN_METHOD == "Zend_View_Helper_HeadScriptTest::main") {
Index: library/Zend/View/Helper/HeadLink.php
===================================================================
--- library/Zend/View/Helper/HeadLink.php	(revision 12040)
+++ library/Zend/View/Helper/HeadLink.php	(working copy)
@@ -264,7 +264,7 @@
         
         foreach ($this->_itemKeys as $itemKey) {
             if (isset($attributes[$itemKey])) {
-                $link .= $itemKey . '="' . $this->_escape($attributes[$itemKey]) . '" ';
+                $link .= sprintf('%s="%s" ', $itemKey, ($this->_autoEscape) ? $this->_escape($attributes[$itemKey]) : $attributes[$itemKey]);
             }
         }
         
Index: library/Zend/View/Helper/HeadScript.php
===================================================================
--- library/Zend/View/Helper/HeadScript.php	(revision 12040)
+++ library/Zend/View/Helper/HeadScript.php	(working copy)
@@ -404,11 +404,12 @@
                 if ('defer' == $key) {
                     $value = 'defer';
                 }
-                $attrString .= sprintf(' %s="%s"', $key, htmlspecialchars($value));
+                $attrString .= sprintf(' %s="%s"', $key, ($this->_autoEscape) ? $this->_escape($value) : $value);
             }
         }
 
-        $html  = $indent . '<script type="' . htmlspecialchars($item->type) . '"' . $attrString . '>';
+        $type = ($this->_autoEscape) ? $this->_escape($item->type) : $item->type;
+        $html  = $indent . '<script type="' . $type . '"' . $attrString . '>';
         if (!empty($item->source)) {
               $html .= PHP_EOL . $indent . '    ' . $escapeStart . PHP_EOL . $item->source . $indent . '    ' . $escapeEnd . PHP_EOL . $indent;
         }
