--- library/Zend/Mail/Storage/Imap.php	(revision 21593)
+++ library/Zend/Mail/Storage/Imap.php	(working copy)
@@ -121,7 +121,11 @@
         }
 
         if ($flags === null) {
-            return count($this->_protocol->search(array('ALL')));
+            if ($result = $this->_protocol->select($this->_currentFolder)) {
+                return $result['exists'];
+            } else {
+                return 0;
+            }
         }
 
         $params = array();
