Index: library/Zend/Pdf.php
===================================================================
--- library/Zend/Pdf.php	(revision 22206)
+++ library/Zend/Pdf.php	(working copy)
@@ -286,7 +286,7 @@
      * from a file.
 
      * $revision used to roll back document to specified version
-     * (0 - currtent version, 1 - previous version, 2 - ...)
+     * (0 - current version, 1 - previous version, 2 - ...)
      *
      * @param string  $source - PDF file to load
      * @param integer $revision
