Index: Response.php
===================================================================
--- Response.php	(revision 4225)
+++ Response.php	(working copy)
@@ -334,6 +334,7 @@
     /**
      * A convenience function that returns a text representation of
      * HTTP response codes. Returns 'Unknown' for unknown codes.
+     * Returns array of all codes, if $code is not specified.
      *
      * Conforms to HTTP/1.1 as defined in RFC 2616 (except for 'Unknown')
      * See http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10 for reference
@@ -342,7 +343,7 @@
      * @param boolean $http11 Use HTTP version 1.1 
      * @return string
      */ 
-    public static function responseCodeAsText($code, $http11 = true)
+    public static function responseCodeAsText($code = null, $http11 = true)
     {
         $messages = array (
             // Informational 1xx
@@ -396,7 +397,9 @@
 
         if (! $http11) $messages[302] = 'Moved Temporarily';
 
-        if (isset($messages[$code])) {
+        if ($code === null) {
+            return $messages;
+        } elseif (isset($messages[$code])) {
             return $messages[$code];
         } else {
             return 'Unknown';
