Index: /usr/share/php/Zend/Mime/Message.php
===================================================================
--- /usr/share/php/Zend/Mime/Message.php	(revision 13494)
+++ /usr/share/php/Zend/Mime/Message.php	(working copy)
@@ -247,7 +247,7 @@
         $res = new self();
         foreach ($parts as $part) {
             // now we build a new MimePart for the current Message Part:
-            $newPart = new Zend_Mime_Part($part);
+            $newPart = new Zend_Mime_Part($part['body']);
             foreach ($part['header'] as $key => $value) {
                 /**
                  * @todo check for characterset and filename
