Index: Element.php
===================================================================
--- Element.php	(revision 17686)
+++ Element.php	(working copy)
@@ -2099,7 +2099,7 @@
             if (null !== $translator) {
                 $message = $translator->translate($message);
             }
-            if ($this->isArray() || is_array($value)) {
+            if ($this->isArray() && is_array($value)) {
                 $aggregateMessages = array();
                 foreach ($value as $val) {
                     $aggregateMessages[] = str_replace('%value%', $val, $message);
