Index: library/Zend/Loader/StandardAutoloader.php
===================================================================
--- library/Zend/Loader/StandardAutoloader.php	(revision 25005)
+++ library/Zend/Loader/StandardAutoloader.php	(working copy)
@@ -314,7 +314,7 @@
         if ($type === self::ACT_AS_FALLBACK) {
             // create filename
             $filename     = $this->transformClassNameToFilename($class, '');
-            if (version_compare(PHP_VERSION, '5.3.0', '>=')) {
+            if (version_compare(PHP_VERSION, '5.3.2', '>=')) {
                 $resolvedName = stream_resolve_include_path($filename);
                 if ($resolvedName !== false) {
                     return include $resolvedName;