Index: Zend/Paginator.php
===================================================================
--- Zend/Paginator.php	(Revision 20305)
+++ Zend/Paginator.php	(Arbeitskopie)
@@ -709,7 +709,7 @@
         if ($this->_itemCountPerPage < 1) {
             $this->_itemCountPerPage = $this->getItemCountPerPage();
         }
-        $this->_pageCount        = $this->_calculatePageCount();
+        $this->_pageCount        = null;
         $this->_currentItems     = null;
         $this->_currentItemCount = null;

