Index: global/library/Zend/Db/Select.php
===================================================================
--- global/library/Zend/Db/Select.php	(revision 4086)
+++ global/library/Zend/Db/Select.php	(working copy)
@@ -682,7 +682,7 @@
     public function assemble()
     {
         $sql = self::SQL_SELECT;
-        foreach (array_keys(self::$_partsInit) as $part) {
+        foreach (array_keys($this->_parts) as $part) {
             $method = '_render' . ucfirst($part);
             if (method_exists($this, $method)) {
                 $sql = $this->$method($sql);
