Index: Test/PHPUnit/ControllerTestCase.php
===================================================================
--- Test/PHPUnit/ControllerTestCase.php	(revision 24042)
+++ Test/PHPUnit/ControllerTestCase.php	(working copy)
@@ -1120,7 +1120,7 @@
     /**
      * Retrieve test case request object
      *
-     * @return Zend_Controller_Request_Abstract
+     * @return Zend_Controller_Request_HttpTestCase
      */
     public function getRequest()
     {
@@ -1134,7 +1134,7 @@
     /**
      * Retrieve test case response object
      *
-     * @return Zend_Controller_Response_Abstract
+     * @return Zend_Controller_Response_HttpTestCase
      */
     public function getResponse()
     {
