Index: Zend/Gdata/AllTests.php
===================================================================
--- Zend/Gdata/AllTests.php	(revision 11250)
+++ Zend/Gdata/AllTests.php	(working copy)
@@ -46,6 +46,7 @@
 require_once 'Zend/Gdata/App/CategoryTest.php';
 require_once 'Zend/Gdata/App/ContentTest.php';
 require_once 'Zend/Gdata/App/ControlTest.php';
+require_once 'Zend/Gdata/App/EntryTest.php';
 require_once 'Zend/Gdata/App/FeedTest.php';
 require_once 'Zend/Gdata/App/GeneratorTest.php';
 require_once 'Zend/Gdata/App/CaptchaRequiredExceptionTest.php';
@@ -204,6 +205,7 @@
         $suite->addTestSuite('Zend_Gdata_App_CategoryTest');
         $suite->addTestSuite('Zend_Gdata_App_ContentTest');
         $suite->addTestSuite('Zend_Gdata_App_ControlTest');
+        $suite->addTestSuite('Zend_Gdata_App_EntryTest');
         $suite->addTestSuite('Zend_Gdata_App_FeedTest');
         $suite->addTestSuite('Zend_Gdata_App_GeneratorTest');
         $suite->addTestSuite('Zend_Gdata_App_CaptchaRequiredExceptionTest');