--- a/library/Zend/Mail/Part/Interface.php
+++ b/library/Zend/Mail/Part/Interface.php
@@ -50,6 +50,17 @@
     public function getContent();
 
     /**
+     * Decoded body of part
+     *
+     * If part is multipart or the transfer encoding is unknown,
+     * an exception is thrown.
+     *
+     * @return string decoded body
+     * @throws Zend_Mail_Exception
+     */
+    public function getDecodedContent();
+
+    /**
      * Return size of part
      *
      * @return int size
--- a/library/Zend/Mail/Part.php
+++ b/library/Zend/Mail/Part.php
@@ -181,7 +181,34 @@
         }
     }
 
+
     /**
+     * Decoded body of part
+     *
+     * If the transfer encoding is unknown, an exception is thrown.
+     *
+     * @return string decoded body
+     * @throws Zend_Mail_Exception
+     */
+    public function getDecodedContent()
+    {
+        $encoding = $this->getHeader('content-transfer-encoding');
+        switch ($encoding) {
+            case Zend_Mime::ENCODING_7BIT:
+            case Zend_Mime::ENCODING_8BIT:
+                return $this->getContent();
+            case Zend_Mime::ENCODING_QUOTEDPRINTABLE:
+                return quoted_printable_decode($this->getContent());
+            case Zend_Mime::ENCODING_BASE64:
+                return base64_decode($this->getContent());
+            default:
+                require_once 'Zend/Mail/Exception.php';
+                throw new Zend_Mail_Exception('unknown content transfer encoding: '.$encoding);
+        }
+    }
+
+
+    /**
      * Return size of part
      *
      * Quite simple implemented currently (not decoding). Handle with care.
