Index: JQuery/View/Helper/AutoComplete.php
===================================================================
--- JQuery/View/Helper/AutoComplete.php	(révision 18158)
+++ JQuery/View/Helper/AutoComplete.php	(copie de travail)
@@ -62,12 +62,14 @@
             require_once "ZendX/JQuery/Exception.php";
             throw new ZendX_JQuery_Exception("Cannot construct AutoComplete field without specifying Parameters Data and Url");
         }
-
+        $data   = $params['url'];
+        unset $params['url'];
         $params = ZendX_JQuery::encodeJson($params);
 
-        $js = sprintf('%s("#%s").autocomplete(%s);',
+        $js = sprintf('%s("#%s").autocomplete(%s,%s);',
             ZendX_JQuery_View_Helper_JQuery::getJQueryHandler(),
             $attribs['id'],
+            $data,
             $params
         );
 
@@ -75,4 +77,4 @@
 
         return $this->view->formText($id, $value, $attribs);
 	}
-}
\ No newline at end of file
+}
