Index: tests/Zend/Controller/Dispatcher/StandardTest.php
===================================================================
--- tests/Zend/Controller/Dispatcher/StandardTest.php	(revision 23552)
+++ tests/Zend/Controller/Dispatcher/StandardTest.php	(working copy)
@@ -657,6 +657,19 @@
             $this->assertContains('No default module', $e->getMessage());
         }
     }
+    
+    /**
+     * @group ZF-10373
+     */
+    public function testGetControllerClassIgnoresCapitalizationInModuleName()
+    {
+        $this->_dispatcher->setControllerDirectory(array('test'=>'/some/dir'));
+        
+        $request = new Zend_Controller_Request_Simple();
+        $request->setModuleName('Test')->setControllerName('test')->setActionName('test');
+        $class = $this->_dispatcher->getControllerClass($request);
+        $this->assertEquals('TestController', $class);
+    }
 }
 
 // Call Zend_Controller_Dispatcher_StandardTest::main() if this source file is executed directly.
Index: library/Zend/Controller/Dispatcher/Standard.php
===================================================================
--- library/Zend/Controller/Dispatcher/Standard.php	(revision 23552)
+++ library/Zend/Controller/Dispatcher/Standard.php	(working copy)
@@ -378,7 +378,7 @@
         $className = $this->formatControllerName($controllerName);
 
         $controllerDirs      = $this->getControllerDirectory();
-        $module = $request->getModuleName();
+        $module = strtolower($request->getModuleName());
         if ($this->isValidModule($module)) {
             $this->_curModule    = $module;
             $this->_curDirectory = $controllerDirs[$module];
