From a78cd081e7cafd19f7a03e54e0c7a8189fe54e15 Mon Sep 17 00:00:00 2001
From: Sven Axelsson <sven@axelsson.name>
Date: Wed, 28 Apr 2010 14:38:32 +0200
Subject: [PATCH] [Fix: Need to escape backslashes.]

---
 .../CodeGenerator/Php/Property/DefaultValue.php    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/w3-html/core/extlibs/Zend/CodeGenerator/Php/Property/DefaultValue.php b/w3-html/core/extlibs/Zend/CodeGenerator/Php/Property/DefaultValue.php
index 7ff534b..e440dae 100644
--- a/w3-html/core/extlibs/Zend/CodeGenerator/Php/Property/DefaultValue.php
+++ b/w3-html/core/extlibs/Zend/CodeGenerator/Php/Property/DefaultValue.php
@@ -268,7 +268,7 @@ class Zend_CodeGenerator_Php_Property_DefaultValue extends Zend_CodeGenerator_Ph
                 $output .= ( $value ? 'true' : 'false' );
                 break;
             case self::TYPE_STRING:
-                $output .= "'" . addcslashes($value, "'") . "'";
+                $output .= "'" . addcslashes($value, "'\\") . "'";
                 break;
             case self::TYPE_NULL:
                 $output .= 'null';
-- 
1.7.1

