Index: library/Zend/Controller/Router/Rewrite.php
===================================================================
--- library/Zend/Controller/Router/Rewrite.php	(revision 706)
+++ library/Zend/Controller/Router/Rewrite.php	(working copy)
@@ -95,7 +95,7 @@
             require_once 'Zend/Controller/Router/Route/Module.php';
             $compat = new Zend_Controller_Router_Route_Module(array(), $dispatcher, $request);
 
-            $this->_routes = array_merge(array('default' => $compat), $this->_routes);
+            $this->_routes = array('default' => $compat) + $this->_routes;
         }
 
         return $this;
