Index: ../trunk/tests/Zend/Form/Element/CaptchaTest.php
===================================================================
--- ../trunk/tests/Zend/Form/Element/CaptchaTest.php	(revision 19479)
+++ ../trunk/tests/Zend/Form/Element/CaptchaTest.php	(working copy)
@@ -163,6 +163,31 @@
         require_once 'Zend/View.php';
         $this->assertFalse(array_key_exists('helper', $this->element->getAttribs()));
     }
+
+    /**
+     * @group ZF-8485
+     */
+    public function testCaptchaWordShouldNotChangeIfRenderedMultipleTimes()
+    {
+        $captcha = $this->element->getCaptcha();
+        $this->element->render();
+        $origWord = $captcha->getWord();
+        $this->element->render();
+        $this->assertEquals($origWord, $captcha->getWord(), 'Captcha word has changed!');
+    }
+
+    /**
+     * @group ZF-8485
+     */
+    public function testCaptchaWordShouldNotChangeIfDecoratorsAreRenderedIndividually()
+    {
+        $captcha = $this->element->getCaptcha();
+        $this->element->renderCaptcha();
+        $origWord = $captcha->getWord();
+        $this->element->renderCaptcha_Word();
+        $this->assertEquals($origWord, $captcha->getWord(), 'Captcha word has changed!');
+    }
+
 }
 
 /**
