--- a/library/Zend/Filter/StripTags.php
+++ b/library/Zend/Filter/StripTags.php
@@ -238,7 +238,7 @@ class Zend_Filter_StripTags implements Zend_Filter_Interface
     public function filter($value)
     {
         // Strip HTML comments first
-        $valueCopy = preg_replace('#<!--(?:[^<]+|<(?!\!--))*?(--\s*>)#us', '', (string) $value);
+        $valueCopy = preg_replace('#<!--(?:[^<]+|<(?!\!--))*?(--\s*>)#s', '', (string) $value);
 
         // Initialize accumulator for filtered data
         $dataFiltered = '';
