===================================================================
--- /Zend/Service/Amazon/S3.php	(revision 18447)
+++ /Zend/Service/Amazon/S3.php	(working copy)
@@ -474,6 +474,7 @@
         $client = self::getHttpClient();
 
         $client->resetParameters();
+        $client->setUri($endpoint);
         $client->setAuth(false);
         // Work around buglet in HTTP client - it doesn't clean headers
         // Remove when ZHC is fixed
@@ -482,7 +483,6 @@
                                   'Range'       => null,
                                   'x-amz-acl'   => null));
 
-        $client->setUri($endpoint);
         $client->setHeaders($headers);
 
         if (is_array($params)) {
