Index: tests/Zend/Http/UserAgentTest.php
===================================================================
--- tests/Zend/Http/UserAgentTest.php	(revision 24790)
+++ tests/Zend/Http/UserAgentTest.php	(working copy)
@@ -194,9 +194,6 @@
         $this->assertEquals('Zend_Http_UserAgent_Mobile', $className);
     }
 
-    /**
-     * @runInSeparateProcess
-     */
     public function testDeviceClassNameMatchesDesktopTypeIfUserAgentDoesNotMatch()
     {
         $config = array(
@@ -534,9 +531,6 @@
         $userAgent->setStorage(new Zend_Http_UserAgent_Storage_NonPersistent());
     }
 
-    /**
-     * @runInSeparateProcess
-     */
     public function testAllowsPassingStorageConfigurationOptions()
     {
         $config = $this->config;
