--- HtmlList.php.orig	2009-10-21 18:38:24.414865576 +0200
+++ HtmlList.php	2009-12-05 11:30:20.048742707 +0100
@@ -58,11 +58,13 @@
 
         foreach ($items as $item) {
             if (!is_array($item)) {
-                if ($escape) {
-                    $item = $this->view->escape($item);
+                if (trim($item) != '') {
+                    if ($escape) {
+                        $item = $this->view->escape($item);
+                    }
+                    $list .= '<li>' . $item . '</li>' . self::EOL;
                 }
-                $list .= '<li>' . $item . '</li>' . self::EOL;
-            } else {
+            } elseif (!empty($item)) {
                 if (6 < strlen($list)) {
                     $list = substr($list, 0, strlen($list) - 6)
                      . $this->htmlList($item, $ordered, $attribs, $escape) . '</li>' . self::EOL;
