Index: /opt/phplibs-zf/Zend/Db/Table/Abstract.php
===================================================================
--- /opt/phplibs-zf/Zend/Db/Table/Abstract.php	(Revision 4423)
+++ /opt/phplibs-zf/Zend/Db/Table/Abstract.php	(Arbeitskopie)
@@ -128,15 +128,15 @@
      * Each key is a mnemonic name for one reference rule.
      *
      * Each value is also an associative array, with the following keys:
-     * - columns    = array of names of column(s) in the child table.
-     * - refTable   = class name of the parent table.
-     * - refColumns = array of names of column(s) in the parent table,
-     *                in the same order as those in the 'columns' entry.
-     * - onDelete   = "cascade" means that a delete in the parent table also
-     *                causes a delete of referencing rows in the child table.
-     * - onUpdate   = "cascade" means that an update of primary key values in
-     *                the parent table also causes an update of referencing
-     *                rows in the child table.
+     * - columns       = array of names of column(s) in the child table.
+     * - refTableClass = class name of the parent table.
+     * - refColumns    = array of names of column(s) in the parent table,
+     *                   in the same order as those in the 'columns' entry.
+     * - onDelete      = "cascade" means that a delete in the parent table also
+     *                   causes a delete of referencing rows in the child table.
+     * - onUpdate      = "cascade" means that an update of primary key values in
+     *                   the parent table also causes an update of referencing
+     *                   rows in the child table.
      *
      * @var array
      */ 
