--- library/Zend/CodeGenerator/Php/Class.php	(revision 3698)
+++ library/Zend/CodeGenerator/Php/Class.php	(working copy)
@@ -111,11 +111,16 @@
         
         if ($parentClass = $reflectionClass->getParentClass()) {
             $class->setExtendedClass($parentClass->getName());
-            $interfaces = array_diff($parentClass->getInterfaces(), $reflectionClass->getInterfaces());
+            $interfaces = array_diff($reflectionClass->getInterfaces(), $parentClass->getInterfaces());
         } else {
             $interfaces = $reflectionClass->getInterfaces();
         }
         
+        foreach ($interfaces as &$interface)
+        {
+            $interface = $interface->getName();
+        }
+
         $class->setImplementedInterfaces($interfaces);