Index: library/Zend/Mail/Transport/Sendmail.php
===================================================================
--- library/Zend/Mail/Transport/Sendmail.php	(revision 20756)
+++ library/Zend/Mail/Transport/Sendmail.php	(working copy)
@@ -74,7 +74,14 @@
      */
     public function __construct($parameters = null)
     {
-        $this->parameters = $parameters;
+        if ($parameters instanceof Zend_Config) {
+            $parameters = $parameters->toArray();
+        }
+        if (is_array($parameters)) {
+            $this->parameters = implode(' ', $parameters);
+        } else if (is_string($parameters)) {
+            $this->parameters = $parameters;
+        }
     }
 
 
