Index: View/Helper/Action.php
===================================================================
--- View/Helper/Action.php	(revision 15802)
+++ View/Helper/Action.php	(working copy)
@@ -104,11 +104,13 @@
      * Retrieve rendered contents of a controller action
      *
      * If the action results in a forward or redirect, returns empty string.
+     * If the action throws an exception - we will re-throw it after setting things back.
      * 
      * @param  string $action 
      * @param  string $controller 
      * @param  string $module Defaults to default module
      * @param  array $params 
+     * @throws Exception if the underlying action throws an exception.
      * @return string
      */
     public function action($action, $controller, $module = null, array $params = array())
@@ -127,13 +129,25 @@
                       ->setControllerName($controller) 
                       ->setActionName($action) 
                       ->setDispatched(true); 
+        
+        // if the action called throws an exception - we need to put the old view 
+        // renderer back in place and reset objects before throwing it.
+        $exception = null;
+        
+        try {
+            $this->dispatcher->dispatch($this->request, $this->response);           
+        } catch (Exception $e) {
+            $exception = $e;
+        }
  
-        $this->dispatcher->dispatch($this->request, $this->response); 
- 
         // reset the viewRenderer object to it's original state
         Zend_Controller_Action_HelperBroker::addHelper($viewRendererObj);
 
-        
+        // if we caught an exception earlier - now is the time to throw it.
+        if ($exception !== null) {
+            $this->resetObjects();
+            throw $exception;
+        }
         if (!$this->request->isDispatched() 
             || $this->response->isRedirect()) 
         { 
