Index: Resource.php
===================================================================
--- Resource.php	(revision 19024)
+++ Resource.php	(working copy)
@@ -173,7 +173,7 @@
 
         $final = substr($class, strlen($lastMatch));
         $path = $this->_components[$lastMatch];
-        return include $path . '/' . str_replace('_', '/', $final) . '.php';
+        return include rtrim($path, '/') . '/' . ltrim(str_replace('_', '/', $final), '/') . '.php';
     }
 
     /**
