Index: Session.php
===================================================================
--- Session.php	(revision 4312)
+++ Session.php	(working copy)
@@ -118,7 +118,8 @@
      */
     private static $_localOptions = array(
         'strict'                => '_strict',
-        'remember_me_seconds'   => '_rememberMeSeconds'
+        'remember_me_seconds'   => '_rememberMeSeconds',
+        'ignore_save_path'      => '_ignore_save_path'
     );
 
     /**
@@ -143,6 +144,13 @@
     private static $_destroyed = false;
 
     /**
+     * Whether or not session save_path must be writable (false = must be writable)
+     *
+     * @var bool
+     */
+    private static $_ignore_save_path = false;
+
+    /**
      * Whether or not session must be initiated before usage
      *
      * @var bool
@@ -213,7 +221,7 @@
             $savePath = explode(';', $savePath);
             $savePath = realpath(array_pop($savePath));
         }
-        if (!is_writable($savePath)) {
+        if (self::$_ignore_save_path !== true && !is_writable($savePath)) {
             throw new Zend_Session_Exception("Unwritable session.save_path: $savePath");
         }
     }
