Index: Oci.php
===================================================================
--- Oci.php	(revision 7751)
+++ Oci.php	(working copy)
@@ -82,6 +82,8 @@
             $tns .= '(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)(HOST=' . $dsn['host'] . ')';
             if (isset($dsn['port'])) {
                 $tns .= '(PORT=' . $dsn['port'] . ')';
+            } else {
+                $tns .= '(PORT=1521)';
             }
             $tns .= '))';
         }
