Index: Controller/Request/Http.php
===================================================================
--- Controller/Request/Http.php	(revision 13968)
+++ Controller/Request/Http.php	(working copy)
@@ -712,11 +712,12 @@
      */
     public function getParams()
     {
+        $paramSources = $this->getParamSources();
         $return = $this->_params;
-        if (isset($_GET) && is_array($_GET)) {
+        if (in_array('_GET', $paramSources) && isset($_GET) && is_array($_GET)) {
             $return += $_GET;
         }
-        if (isset($_POST) && is_array($_POST)) {
+        if (in_array('_POST', $paramSources) && isset($_POST) && is_array($_POST)) {
             $return += $_POST;
         }
         return $return;
