--- tests/Zend/Form/Decorator/FieldsetTest.php	(revision 23320)
+++ tests/Zend/Form/Decorator/FieldsetTest.php	(working copy)
@@ -219,6 +219,21 @@
         $this->assertContains('<fieldset', $test, $test);
         $this->assertNotContains('helper="', $test);
     }
+
+    /**
+     * @group ZF-10679
+     */
+    public function testFieldsetIdOverridesFormId()
+    {
+        $form = new Zend_Form();
+        $form->setName('bar')
+             ->setAttrib('id', 'form-id')
+             ->setView($this->getView());
+        $html = $this->decorator->setElement($form)
+                                ->setOption('id', 'fieldset-id')
+                                ->render('content');
+        $this->assertContains('<fieldset id="fieldset-id"', $html);
+    }
 }
 
 // Call Zend_Form_Decorator_FieldsetTest::main() if this source file is executed directly.
Index: library/Zend/Form/Decorator/Fieldset.php
===================================================================
--- library/Zend/Form/Decorator/Fieldset.php	(revision 23320)
+++ library/Zend/Form/Decorator/Fieldset.php	(working copy)
@@ -72,7 +72,7 @@
         $options = parent::getOptions();
         if (null !== ($element = $this->getElement())) {
             $attribs = $element->getAttribs();
-            $options = array_merge($options, $attribs);
+            $options = array_merge($attribs, $options);
             $this->setOptions($options);
         }
         return $options;
