Index: library/Zend/Loader/PluginLoader.php
===================================================================
--- library/Zend/Loader/PluginLoader.php	(revision 18)
+++ library/Zend/Loader/PluginLoader.php	(working copy)
@@ -125,6 +125,10 @@
     	if($prefix == "") {
     		return $prefix;
     	}
+        $last = strlen($prefix) - 1;
+        if ($prefix{$last} == '\\') {
+            return $prefix;
+        }
         return rtrim($prefix, '_') . '_';
     }
 
