Index: Client.php
===================================================================
--- Client.php	(revision 24211)
+++ Client.php	(working copy)
@@ -371,7 +371,7 @@
             throw new Zend_Http_Client_Exception("'{$method}' is not a valid HTTP request method.");
         }
 
-        if ($method == self::POST && $this->enctype === null) {
+        if (($method == self::POST or $method == self::PUT or $method == self::DELETE) && $this->enctype === null) {
             $this->setEncType(self::ENC_URLENCODED);
         }
 
