Index: library/Zend/Tool/Framework/Client/Console/HelpSystem.php
===================================================================
--- library/Zend/Tool/Framework/Client/Console/HelpSystem.php	(revision 18303)
+++ library/Zend/Tool/Framework/Client/Console/HelpSystem.php	(working copy)
@@ -170,6 +170,18 @@
     }
 
     /**
+     * @param Zend_Tool_Framework_Action_Interface $actionInfo
+     */
+    protected function _printActionHelpDescription($actionInfo)
+    {
+        if($actionInfo->getShortDescription() != "") {
+            $helpDescription = $actionInfo->getShortDescription();
+
+            $this->_response->appendContent('    Description: '.$helpDescription.PHP_EOL);
+        }
+    }
+
+    /**
      * _respondWithSystemInformation()
      *
      * @param string $providerNameFilter
@@ -255,6 +267,8 @@
 
                     $this->_respondWithCommand($providerMetadata, $actionMetadata, $specialtyMetadata, $actionableGlobalLongParamMetadata);
 
+                    $this->_printActionHelpDescription($actionInfo);
+
                     $actionIsGlobal = true;
                 } else {
                     $actionIsGlobal = false;
@@ -295,7 +309,7 @@
                             'specialtyName' => $specialtyName,
                             'actionName'    => $actionName,
                             'clientName'    => 'console'
-                            ));
+                        ));
 
                         if($actionableSpecialtyLongMetadata) {
                             $this->_respondWithCommand($providerMetadata, $actionMetadata, $specialtyMetadata, $actionableSpecialtyLongMetadata);
