Index: tests/Zend/Feed/ImportTest.php
===================================================================
--- tests/Zend/Feed/ImportTest.php	(revision 13383)
+++ tests/Zend/Feed/ImportTest.php	(working copy)
@@ -192,7 +192,8 @@
      */
     public function testRss100Sample1()
     {
-        $this->_importRssValid('RssTest100Sample1.xml');
+        $feed = $this->_importRssValid('RssTest100Sample1.xml');
+        $this->assertEquals($feed->count(), 2);
     }
 
     /**
@@ -200,7 +201,8 @@
      */
     public function testRss100Sample2()
     {
-        $this->_importRssValid('RssTest100Sample2.xml');
+        $feed = $this->_importRssValid('RssTest100Sample2.xml');
+        $this->assertEquals($feed->count(), 1);
     }
 
     /**
@@ -423,6 +425,7 @@
 
         $feed = Zend_Feed::import('http://localhost');
         $this->assertType('Zend_Feed_Rss', $feed);
+        return $feed;
     }
 
     /**
Index: library/Zend/Feed/Rss.php
===================================================================
--- library/Zend/Feed/Rss.php	(revision 13383)
+++ library/Zend/Feed/Rss.php	(working copy)
@@ -80,7 +80,11 @@
         parent::__wakeup();
 
         // Find the base channel element and create an alias to it.
-        $this->_element = $this->_element->getElementsByTagName('channel')->item(0);
+        if ($this->_element->firstChild->nodeName == 'rdf:RDF') {
+            $this->_element = $this->_element->firstChild; 
+        } else {
+            $this->_element = $this->_element->getElementsByTagName('channel')->item(0);
+        }
         if (!$this->_element) {
             /** 
              * @see Zend_Feed_Exception
