--- tests/Zend/Service/Amazon/SimpleDb/AllTests.php	(revision 23796)
+++ tests/Zend/Service/Amazon/SimpleDb/AllTests.php	(working copy)
@@ -64,7 +64,7 @@
 
         if (defined('TESTS_ZEND_SERVICE_AMAZON_ONLINE_ENABLED')
             && constant('TESTS_ZEND_SERVICE_AMAZON_ONLINE_ENABLED')
-            && defined('TESTS_ZEND_SERVICE_AMAZON_ONLINE_ACCESSKEY')
+            && defined('TESTS_ZEND_SERVICE_AMAZON_ONLINE_ACCESSKEYID')
             && defined('TESTS_ZEND_SERVICE_AMAZON_ONLINE_SECRETKEY')
         ) {
             $suite->addTestSuite('Zend_Service_Amazon_SimpleDb_OnlineTest');
Index: tests/Zend/Service/Amazon/SimpleDb/OnlineTest.php
===================================================================
--- tests/Zend/Service/Amazon/SimpleDb/OnlineTest.php	(revision 23796)
+++ tests/Zend/Service/Amazon/SimpleDb/OnlineTest.php	(working copy)
@@ -86,7 +86,7 @@
     public function setUp()
     {
         $this->_amazon = new Zend_Service_Amazon_SimpleDb(
-            constant('TESTS_ZEND_SERVICE_AMAZON_ONLINE_ACCESSKEY'),
+            constant('TESTS_ZEND_SERVICE_AMAZON_ONLINE_ACCESSKEYID'),
             constant('TESTS_ZEND_SERVICE_AMAZON_ONLINE_SECRETKEY')
         );
 
