Index: library/Zend/Tool/Project/Provider/Module.php
===================================================================
--- library/Zend/Tool/Project/Provider/Module.php	(revision 23639)
+++ library/Zend/Tool/Project/Provider/Module.php	(working copy)
@@ -138,9 +138,27 @@
 
         $resources = self::createResources($this->_loadedProfile, $name);
 
+        //in case the user has entered all small characters for module-name
+        //we make sure the first letter is capitalized.
+        $originalName = $name;
+        $name = ucfirst($name);
+
         $response = $this->_registry->getResponse();
+        $request = $this->_registry->getRequest();
 
-        if ($this->_registry->getRequest()->isPretend()) {
+        // informing user about the name if required
+        if (($name !== $originalName)) {
+            $tense = (($request->isPretend()) ? 'would be' : 'is');
+            $response->appendContent(
+                'Note: The canonical module name that ' . $tense
+                    . ' used with other providers is "' . $name . '";'
+                    . ' not "' . $originalName . '" as supplied',
+                array('color' => array('yellow'))
+                );
+            unset($tense);
+        }
+
+        if ($request->isPretend()) {
             $response->appendContent('I would create the following module and artifacts:');
             foreach (new RecursiveIteratorIterator($resources, RecursiveIteratorIterator::SELF_FIRST) as $resource) {
                 if (is_callable(array($resource->getContext(), 'getPath'))) {
