Index: Table.php
===================================================================
--- Table.php	(revision 921)
+++ Table.php	(working copy)
@@ -230,10 +230,10 @@
     public function insert(&$data)
     {
         $this->_db->insert(
-            $this->_name,
+            $this->_db->quoteIdentifier($this->_name),
             $data
         );
-        return $this->_db->lastInsertId($this->_name, $this->_primary);
+        return $this->_db->lastInsertId($this->_db->quoteIdentifier($this->_name), $this->_primary);
     }
 
     /**
@@ -248,7 +248,7 @@
     public function update(&$data, $where)
     {
         return $this->_db->update(
-            $this->_name,
+            $this->_db->quoteIdentifier($this->_name),
             $data,
             $where
         );
@@ -264,7 +264,7 @@
      */
     public function delete($where)
     {
-        return $this->_db->delete($this->_name, $where);
+        return $this->_db->delete($this->_db->quoteIdentifier($this->_name), $where);
     }
 
 
@@ -371,7 +371,7 @@
         $select = $this->_db->select();
 
         // the FROM clause
-        $select->from($this->_name, array_keys($this->_cols));
+        $select->from($this->_db->quoteIdentifier($this->_name), array_keys($this->_cols));
 
         // the WHERE clause
         $where = (array) $where;
