--- Standard.php.orig	2009-03-22 00:58:14.000000000 +0100
+++ Standard.php	2009-03-22 01:04:59.000000000 +0100
@@ -259,7 +259,7 @@ class Zend_Controller_Dispatcher_Standar
          * arguments; throw exception if it's not an action controller
          */
         $controller = new $className($request, $this->getResponse(), $this->getParams());
-        if (!($controller instanceof Zend_Controller_Action_Interface) && 
+        if (!($controller instanceof Zend_Controller_Action_Interface) &&
             !($controller instanceof Zend_Controller_Action)) {
             require_once 'Zend/Controller/Dispatcher/Exception.php';
             throw new Zend_Controller_Dispatcher_Exception(
@@ -373,6 +373,10 @@ class Zend_Controller_Dispatcher_Standar
         $module = $request->getModuleName();
         if ($this->isValidModule($module)) {
             $this->_curModule    = $module;
+            if(!isset($controllerDirs[$module])) {
+                require_once 'Zend/Controller/Exception.php';
+                throw new Zend_Controller_Exception("Module '$module' was not registered nor found");
+            }
             $this->_curDirectory = $controllerDirs[$module];
         } elseif ($this->isValidModule($this->_defaultModule)) {
             $request->setModuleName($this->_defaultModule);
