--- library/Zend/Db/Adapter/Mysqli.php	(revision 22364)
+++ library/Zend/Db/Adapter/Mysqli.php	(working copy)
@@ -312,6 +312,35 @@
             }
         }
 
+        $realConnectFlags
+            = !empty($this->_config['driver_options']['realConnectFlags'])
+                ? $this->_config['driver_options']['realConnectFlags']
+                : 0;
+        if ( $realConnectFlags & MYSQLI_CLIENT_SSL ) {
+            $sslKey = !empty($this->_config['driver_options']['sslKey'])
+                ? $this->_config['driver_options']['sslKey']
+                : null;
+            $sslCert = !empty($this->_config['driver_options']['sslCert'])
+                ? $this->_config['driver_options']['sslCert']
+                : null;
+            $sslCa = !empty($this->_config['driver_options']['sslCa'])
+                ? $this->_config['driver_options']['sslCa']
+                : null;
+            $sslCapath = !empty($this->_config['driver_options']['sslCapath'])
+                ? $this->_config['driver_options']['sslCapath']
+                : null;
+            $sslCipher = !empty($this->_config['driver_options']['sslCipher'])
+                ? $this->_config['driver_options']['sslCipher']
+                : null;
+            @mysqli_ssl_set($this->_connection,
+                $sslKey,
+                $sslCert,
+                $sslCa,
+                $sslCapath,
+                $sslCipher
+            );
+        }
+
         // Suppress connection warnings here.
         // Throw an exception instead.
         $_isConnected = @mysqli_real_connect(
@@ -320,7 +349,9 @@
             $this->_config['username'],
             $this->_config['password'],
             $this->_config['dbname'],
-            $port
+            $port,
+            null,
+            $realConnectFlags
         );
 
         if ($_isConnected === false || mysqli_connect_errno()) {
