Index: tests/Zend/Soap/ServerTest.php
===================================================================
--- tests/Zend/Soap/ServerTest.php	(revision 22135)
+++ tests/Zend/Soap/ServerTest.php	(working copy)
@@ -64,7 +64,9 @@
                          'classmap' => array('TestData1' => 'Zend_Soap_Server_TestData1',
                                              'TestData2' => 'Zend_Soap_Server_TestData2',),
                          'encoding' => 'ISO-8859-1',
-                         'uri' => 'http://framework.zend.com/Zend_Soap_ServerTest.php'
+                         'uri' => 'http://framework.zend.com/Zend_Soap_ServerTest.php',
+                         'features' => SOAP_USE_XSI_ARRAY_TYPE,
+                         'cache_wsdl' => WSDL_CACHE_MEMORY
                         );
         $server->setOptions($options);
 
