Index: TestCommon.php
===================================================================
--- TestCommon.php	(révision 11342)
+++ TestCommon.php	(copie de travail)
@@ -1216,13 +1216,13 @@
             'Incorrect quoteInto() result for CHAR');
     }
 
-    public function testAdapterQuoteIntoCount() 
-    { 
-        $value = $this->_db->quoteInto('foo = ? and bar = ?', 1234, null, 1); 
-        $this->assertType('string', $value); 
-        $this->assertEquals('foo = 1234 and bar = ?', $value, 
-            'Incorrect quoteInto() result for count'); 
-    } 
+    public function testAdapterQuoteIntoCount()
+    {
+        $value = $this->_db->quoteInto('foo = ? and bar = ?', 1234, null, 1);
+        $this->assertType('string', $value);
+        $this->assertEquals('foo = 1234 and bar = ?', $value,
+            'Incorrect quoteInto() result for count');
+    }
 
     public function testAdapterQuoteTypeInt()
     {
@@ -1444,7 +1444,10 @@
     {
         $value       = ':0\'';
         $valueQuoted = $this->_db->quote($value);
-        $sql = 'SELECT bug_id FROM zfbugs WHERE bug_status != ' . $valueQuoted;
+        $bugs = $this->_db->quoteIdentifier('zfbugs');
+        $bug_id = $this->_db->quoteIdentifier('bug_id');
+        $bug_status = $this->_db->quoteIdentifier('bug_status');
+        $sql = "SELECT $bug_id FROM $bugs WHERE $bug_status != " . $valueQuoted;
         $results = $this->_db->fetchAll($sql);
         $this->assertEquals(4, count($results));
     }
