Index: Abstract.php
===================================================================
--- Abstract.php	(revision 7833)
+++ Abstract.php	(working copy)
@@ -544,6 +544,9 @@
     {
         if (! $this->_db) {
             $this->_db = self::getDefaultAdapter();
+            if (!$this->_db instanceof Zend_Db_Adapter_Abstract) {
+            	throw new Zend_Db_Table_Exception('No adapter found for ' . get_class($this));
+            }
         }
     }
 
