Index: library/Zend/Controller/Action/Helper/Url.php
===================================================================
--- library/Zend/Controller/Action/Helper/Url.php	(revision 24820)
+++ library/Zend/Controller/Action/Helper/Url.php	(working copy)
@@ -76,7 +76,10 @@
             $url .= '/' . $paramString;
         }
 
-        $url = '/' . ltrim($url, '/');
+        // Only prepend a slash if $baseUrl was not a fully-qualified URL
+        if (!strstr($url, '://')) {
+            $url = '/' . ltrim($url, '/');
+        }
 
         return $url;
     }
Index: tests/Zend/Controller/Action/Helper/UrlTest.php
===================================================================
--- tests/Zend/Controller/Action/Helper/UrlTest.php	(revision 24820)
+++ tests/Zend/Controller/Action/Helper/UrlTest.php	(working copy)
@@ -178,6 +178,21 @@
         $url = $this->helper->simple('action', null, null, array('foo' => 'bar'));
         $this->assertEquals('/baseurl/module/controller/action/foo/bar', $url);
     }
+
+    /**
+     * @group ZF-10923
+     */
+    public function testFullyQualifiedBaseUrlDoesNotGetSlashPrepended()
+    {
+        $this->front->setBaseUrl('http://example.com');
+
+        $request = $this->front->getRequest();
+        $request->setModuleName('module')
+                ->setControllerName('controller');
+
+        $url = $this->helper->simple('action', null, null, array('foo' => 'bar'));
+        $this->assertEquals('http://example.com/module/controller/action/foo/bar', $url);
+    }
 }
 
 // Call Zend_Controller_Action_Helper_UrlTest::main() if this source file is executed directly.
