Index: Zend_Framework/library/Zend/Amf/Server.php
===================================================================
--- Zend_Framework/library/Zend/Amf/Server.php	(Revision 14286)
+++ Zend_Framework/library/Zend/Amf/Server.php	(Arbeitskopie)
@@ -136,6 +136,11 @@
                 }
                 // Add the new loaded class to the server.
                 $this->setClass($className);
+                
+                if(!isset($this->_table[$method])) {
+                	require_once 'Zend/Amf/Server/Exception.php';
+                    throw new Zend_Amf_Server_Exception('Method "' . $className . '.' . $method . '" does not exist');
+                }
             } else {
                 require_once 'Zend/Amf/Server/Exception.php';
                 throw new Zend_Amf_Server_Exception('Method "' . $method . '" does not exist');
