--- library/ZendX/JQuery/View/Helper/JQuery/Container.php	(revision 23775)
+++ library/ZendX/JQuery/View/Helper/JQuery/Container.php	(working copy)
@@ -239,11 +239,21 @@
      */
     public function setCdnSsl($flag)
     {
-        $this->_loadSslCdnPath = $flag;
+        $this->_loadSslCdnPath = (boolean) $flag;
         return $this;
     }
 
     /**
+     * Get Flag of SSL on CDN
+     *
+     * @return boolean True if SSL is used on CDN
+     */
+    public function getCdnSsl()
+    {
+        return $this->_loadSslCdnPath;
+    }
+
+    /**
      * Are we using the CDN?
      *
      * @return boolean
@@ -829,4 +839,4 @@
         }
         return $uiPath;
     }
-}
\ No newline at end of file
+}
