--- tests/Zend/Form/FormTest.php	(revision 22797)
+++ tests/Zend/Form/FormTest.php	(working copy)
@@ -4196,8 +4196,23 @@
 
         $this->assertEquals(array('sub' => array('valid' => 1234)), $this->form->getValidValues($data));
     }
-        
+
     /**
+     * @group ZF-9275
+     */
+    public function testElementTranslatorNotOverriddenbyGlobalTranslatorDuringValidation()
+    {
+        $translator = new Zend_Translate('array', array('foo' => 'bar'));
+        Zend_Registry::set('Zend_Translate', $translator);
+
+        $this->form->addElement('text', 'foo');
+        $this->form->isValid(array());
+
+        $received = $this->form->foo->hasTranslator();
+        $this->assertSame(false, $received);
+    }
+
+    /**
      * @group ZF-9494
      */
     public function testElementTranslatorNotOveriddenbyFormTranslator()
