Index: library/Zend/View/Abstract.php
===================================================================
--- library/Zend/View/Abstract.php	(revision 4437)
+++ library/Zend/View/Abstract.php	(working copy)
@@ -461,7 +461,7 @@
      * @param  string $type
      * @return Zend_View_Abstract
      */
-    public function setPluginLoader(Zend_Loader_PluginLoader $loader, $type)
+    public function setPluginLoader(Zend_Loader_PluginLoader_Interface $loader, $type)
     {
         $type = strtolower($type);
         if (!in_array($type, $this->_loaderTypes)) {
