Index: Controller/Front.php
===================================================================
--- Controller/Front.php	(revision 1057)
+++ Controller/Front.php	(working copy)
@@ -111,6 +111,7 @@
 	 * Convenience method, passthru to Zend_Controller_Dispatcher::setControllerDirectory()
 	 *
 	 * @param string $directory
+	 * @return Zend_Controller_Front
 	 */
 	public function setControllerDirectory($directory)
 	{
