Index: library/Zend/Memory.php
===================================================================
--- library/Zend/Memory.php	(revision 20756)
+++ library/Zend/Memory.php	(working copy)
@@ -54,7 +54,9 @@
         }
 
         // because lowercase will fail
-        $backend = @ucfirst(strtolower($backend));
+        $backend = ucfirst(strtolower($backend));
+        $backend = ucwords(str_replace(array('-', '_', '.'), ' ', $backend));
+        $backend = str_replace(' ', '', $backend);
 
         if (!in_array($backend, Zend_Cache::$availableBackends)) {
             require_once 'Zend/Memory/Exception.php';
