Index: library/ZendX/JQuery/Form/Decorator/UiWidgetContainer.php
===================================================================
--- library/ZendX/JQuery/Form/Decorator/UiWidgetContainer.php	(revision 24400)
+++ library/ZendX/JQuery/Form/Decorator/UiWidgetContainer.php	(working copy)
@@ -127,12 +127,23 @@
             return $content;
         }
 
+        $placement = $this->getPlacement();
+        $separator = $this->getSeparator();
+
         $jQueryParams = $this->getJQueryParams();
-        $attribs     = $this->getOptions();
+        $attribs      = $this->getOptions();
 
         $helper      = $this->getHelper();
         $id          = $element->getId() . '-container';
 
-        return $view->$helper($id, $jQueryParams, $attribs);
+        $tabs = $view->$helper($id, $jQueryParams, $attribs);
+
+        switch ($placement) {
+            case self::PREPEND:
+                return $tabs . $separator . $content;
+
+            case self::APPEND:
+                return $content . $separator . $tabs;
+        }
     }
 }
\ No newline at end of file
